--- lib_eio/unix/dune
+++ lib_eio/unix/dune
@@ -6,6 +6,7 @@
   (language c)
   (include_dirs include)
   (names fork_action eio_unix_stubs cap pty))
+ (c_library_flags -lMacportsLegacySupport)
  (libraries eio eio.utils unix threads mtime.clock.os))
 
 (rule
--- lib_eio_posix/dune
+++ lib_eio_posix/dune
@@ -7,6 +7,7 @@
   (flags :standard -D_LARGEFILE64_SOURCE)
   (include_dirs ../lib_eio/unix/include)
   (names eio_posix_stubs))
+ (c_library_flags -lMacportsLegacySupport)
  (libraries eio eio.utils eio.unix fmt iomux))
 
 (rule
