--- meson.build	2025-10-18 19:13:44.000000000 +0800
+++ meson.build	2025-10-29 12:50:19.000000000 +0800
@@ -28,7 +28,7 @@
 pycairo_version_req = '>= 1.16.0'
 libffi_version_req = '>= 3.0'
 
-gi_dep = dependency('girepository-2.0', version : glib_version_req,
+gi_dep = dependency('girepository-3.0', version : glib_version_req,
   fallback: ['glib', 'libgirepository_dep'])
 glib_dep = dependency('glib-2.0', version : glib_version_req,
   fallback: ['glib', 'libglib_dep'], default_options: ['-Dintrospection=enabled'])
