--- ripgrep-13.0.0/crates/core/app.rs	2021-06-12 20:12:24.000000000 +0800
+++ ripgrep-13.0.0/crates/core/app.rs	2026-01-04 16:24:17.000000000 +0800
@@ -9,7 +9,7 @@
 // is where we read clap's configuration from the end user's arguments and turn
 // it into a ripgrep-specific configuration type that is not coupled with clap.
 
-use clap::{self, crate_authors, crate_version, App, AppSettings};
+use clap::{self, crate_version, App, AppSettings};
 use lazy_static::lazy_static;
 
 const ABOUT: &str = "
@@ -56,7 +56,6 @@
     }
 
     let mut app = App::new("ripgrep")
-        .author(crate_authors!())
         .version(crate_version!())
         .long_version(LONG_VERSION.as_str())
         .about(ABOUT)
