From f1594ced444f6ef3a2f028e4a7a03b3483492f3c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 7 Aug 2025 20:45:46 +0800
Subject: [PATCH] Makefile: support macOS on arm64

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git Makefile Makefile
index 7c973d0..da83c3c 100644
--- Makefile
+++ Makefile
@@ -52,6 +52,10 @@ ifeq ($(sys),Darwinppc64)
  system=mac
  f90compiler=gfortran
 endif
+ifeq ($(sys),Darwinarm64)
+ system=maci
+ f90compiler=gfortran
+endif
 ifeq ($(sys),aix)
  system=aix
  f90compiler=xlf90
