--- src/include/access/gin.h	2025-09-23 04:11:33.000000000 +0800
+++ src/include/access/gin.h	2025-10-17 05:40:47.000000000 +0800
@@ -68,7 +68,11 @@
  * This must be of the same size as a bool because some code will cast a
  * pointer to a bool to a pointer to a GinTernaryValue.
  */
-typedef char GinTernaryValue;
+#ifdef __ppc__
+    typedef int GinTernaryValue;
+#else
+    typedef char GinTernaryValue;
+#endif
 
 StaticAssertDecl(sizeof(GinTernaryValue) == sizeof(bool),
 				 "sizes of GinTernaryValue and bool are not equal");
