--- libavcodec/libsvtav1.c	2025-11-26 10:41:31.000000000 +0800
+++ libavcodec/libsvtav1.c	2026-02-10 08:06:13.000000000 +0800
@@ -190,7 +190,11 @@
     } else if (svt_enc->qp > 0) {
         param->qp                   = svt_enc->qp;
         param->rate_control_mode    = 0;
+#if SVT_AV1_CHECK_VERSION(4, 0, 0)
+        param->aq_mode = 0;
+#else
         param->enable_adaptive_quantization = 0;
+#endif
     }
 
     desc = av_pix_fmt_desc_get(avctx->pix_fmt);
