--- configure	2011-03-25 03:45:09.000000000 +0800
+++ configure	2025-04-28 02:45:14.000000000 +0800
@@ -4333,10 +4333,10 @@
 You must specify an absolute path for --with-gmp-lib.
 " "$LINENO" 5  ;;
     esac
-    GMP_LIBS="-L$withval -Wl,-rpath $withval -Wl,-rpath /usr/local/lib -lgmp"
+    GMP_LIBS="-L$withval -Wl,-rpath $withval -lgmp"
 
 else
-  GMP_LIBS="-Wl,-rpath /usr/local/lib -lgmp"
+  GMP_LIBS="-Wl,-rpath /opt/local/lib -lgmp"
 fi
 
 
@@ -4441,7 +4441,7 @@
 if test "${with_pbc_include+set}" = set; then :
   withval=$with_pbc_include; PBC_CFLAGS="-I$withval"
 else
-  PBC_CFLAGS="-I/usr/include/pbc -I/usr/local/include/pbc"
+  PBC_CFLAGS="-I/opt/local/include/pbc"
 fi
 
 
@@ -4456,10 +4456,10 @@
 You must specify an absolute path for --with-pbc-lib.
 " "$LINENO" 5  ;;
     esac
-    PBC_LIBS="-L$withval -Wl,-rpath $withval -Wl,-rpath /usr/local/lib -lpbc"
+    PBC_LIBS="-L$withval -Wl,-rpath $withval -lpbc"
 
 else
-  PBC_LIBS="-Wl,-rpath /usr/local/lib -lpbc"
+  PBC_LIBS="-Wl,-rpath /opt/local/lib -lpbc"
 fi
 
 
