--- configure.ac	2024-03-18 10:16:31.000000000 +0800
+++ configure.ac	2026-03-19 00:33:11.000000000 +0800
@@ -254,7 +254,7 @@
     abi_strl=no
     abi_strmode=no
     abi_strnstr=no
-    abi_strtonum=no
+    abi_strtonum=yes
     abi_strtox=yes
     abi_timeconv=no
     # On libmd.

--- src/setproctitle.c	2024-03-06 09:28:54.000000000 +0800
+++ src/setproctitle.c	2026-03-19 05:18:32.000000000 +0800
@@ -34,8 +34,13 @@
 #include "local-link.h"
 
 #if !HAVE_DECL_ENVIRON
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
 extern char **environ;
 #endif
+#endif
 
 static struct {
 	/* Original value. */
