From fc06eb6b15759b5886c4ee62a4b0c398d0f568a0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 29 Apr 2025 18:55:26 +0800
Subject: [PATCH] Entropy.c: fix HAVE_SYS_RANDOM macro

---
 Entropy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git Entropy.c Entropy.c
index c14a776..df1f40e 100644
--- Entropy.c
+++ Entropy.c
@@ -3,7 +3,7 @@
 #include "Hash.h"
 #include <sys/utsname.h>
 
-#ifdef HAVE_SYS_RANDOM_H
+#ifdef HAVE_SYS_RANDOM
 #include <sys/random.h>
 #endif
 
