From 5939e3ce70866ad976ef307461adb2b6ec831984 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 12 Feb 2025 07:36:16 +0800
Subject: [PATCH 3/4] Entropy.c: add a missing header for getentropy

---
 Entropy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git Entropy.c Entropy.c
index 72efc93..4d6fc9a 100644
--- Entropy.c
+++ Entropy.c
@@ -2,6 +2,7 @@
 #include "Encodings.h"
 #include "Hash.h"
 #include <sys/utsname.h>
+#include <sys/random.h>
 
 
 #ifdef HAVE_GETENTROPY
-- 
2.47.1

