From d67f638e281b80326d6b86ec5a65a7caf29b46f9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 12 Feb 2025 07:46:17 +0800
Subject: [PATCH 4/4] Fixes for Makefile

---
 Makefile.in  | 43 ++++++++++++++++++++++++++++---------------
 configure.ac |  4 ++--
 2 files changed, 30 insertions(+), 17 deletions(-)

diff --git Makefile.in Makefile.in
index 957b4ff..828b26a 100644
--- Makefile.in
+++ Makefile.in
@@ -1,9 +1,22 @@
 CC = @CC@
 AR = @AR@
+LN = gln
 VERSION = 5.42
 MAJOR=5
-LIBFILE=libUseful.so.$(VERSION)
-SONAME=libUseful.so.$(MAJOR)
+UNAME_S := $(shell uname -s)
+ifeq ($(UNAME_S),Darwin)
+	SONAME_SUFFIX = dylib
+	SONAME_VER_SUFFIX = $(MAJOR).$(SONAME_SUFFIX)
+	LIBFILE_SUFFIX = $(VERSION).$(SONAME_SUFFIX)
+	LINKER_SONAME_OPTION = -dylib_install_name
+else
+	SONAME_SUFFIX = so
+	SONAME_VER_SUFFIX = $(SONAME_SUFFIX).$(MAJOR)
+	LIBFILE_SUFFIX = $(SONAME_SUFFIX).$(VERSION)
+	LINKER_SONAME_OPTION = -soname
+endif
+LIBFILE = libUseful.$(LIBFILE_SUFFIX)
+SONAME = libUseful.$(SONAME_VER_SUFFIX)
 CFLAGS = @CFLAGS@ @SIMD_CFLAGS@ @SONAME_FLAGS@
 LDFLAGS=@LDFLAGS@
 LIBS = @LIBS@
@@ -15,13 +28,13 @@ OBJ=StrLenCache.o String.o Array.o List.o IPAddress.o Socket.o Server.o UnixSock
 
 all: $(OBJ)
 	$(CC) $(FLAGS) -shared -o $(LIBFILE) $(OBJ) $(LIBS) $(LDFLAGS)
-	-ln -s -r -f $(LIBFILE) libUseful-$(VERSION).so
-	-ln -s -r -f $(LIBFILE) libUseful-$(MAJOR).so
-	-ln -s -r -f $(LIBFILE) $(SONAME)
-	-ln -s -r -f $(LIBFILE) libUseful.so
+	-$(LN) -s -r -f $(LIBFILE) libUseful-$(VERSION).$(SONAME_SUFFIX)
+	-$(LN) -s -r -f $(LIBFILE) libUseful-$(MAJOR).$(SONAME_SUFFIX)
+	-$(LN) -s -r -f $(LIBFILE) $(SONAME)
+	-$(LN) -s -r -f $(LIBFILE) libUseful.$(SONAME_SUFFIX)
 	$(AR) rcs libUseful-$(VERSION).a $(OBJ)
-	-ln -s -r -f libUseful-$(VERSION).a libUseful-$(MAJOR).a
-	-ln -s -r -f libUseful-$(VERSION).a libUseful.a
+	-$(LN) -s -r -f libUseful-$(VERSION).a libUseful-$(MAJOR).a
+	-$(LN) -s -r -f libUseful-$(VERSION).a libUseful.a
 
 
 StrLenCache.o: StrLenCache.h StrLenCache.c
@@ -278,20 +291,20 @@ LibSettings.o: LibSettings.h LibSettings.c
 	$(CC) $(FLAGS) -c LibSettings.c
 
 clean:
-	-rm -f *.o *.so *.so.* *.a *.orig .*.swp *~
-	-rm config.log config.status 
+	-rm -f *.o *.$(SONAME_SUFFIX) *.$(SONAME_VER_SUFFIX) *.$(LIBFILE_SUFFIX) *.a *.orig .*.swp *~
+	-rm config.log config.status
 	-rm -r autom4te.cache config.cache
 	-$(MAKE) clean -C examples
 
-install: libUseful.so
-	-mkdir -p $(DESTDIR)$(prefix)/lib 
-	cp -P *.so *.so.* *.a $(DESTDIR)$(prefix)/lib  
+install: libUseful.$(SONAME_SUFFIX)
+	-mkdir -p $(DESTDIR)$(prefix)/lib
+	cp -P *.$(SONAME_SUFFIX) *.$(SONAME_VER_SUFFIX) *.$(LIBFILE_SUFFIX) *.a $(DESTDIR)$(prefix)/lib
 	-mkdir -p $(DESTDIR)$(prefix)/include/libUseful-$(VERSION)
 	cp *.h $(DESTDIR)$(prefix)/include/libUseful-$(VERSION)
-	-ln -s -r -f $(DESTDIR)$(prefix)/include/libUseful-$(VERSION) $(DESTDIR)$(prefix)/include/libUseful-5
+	-$(LN) -s -r -f $(DESTDIR)$(prefix)/include/libUseful-$(VERSION) $(DESTDIR)$(prefix)/include/libUseful-5
 	-mkdir -p $(DESTDIR)$(prefix)/etc
 	cp *.conf $(DESTDIR)$(prefix)/etc
 
 
-test: libUseful.so
+test: libUseful.$(SONAME_SUFFIX)
 	-echo "No tests written yet"
diff --git configure.ac configure.ac
index b59cfbb..dafd4ec 100644
--- configure.ac
+++ configure.ac
@@ -54,11 +54,11 @@ if test "$cf_use_soname" != "no"
 then
   if test "$GCC" = "yes"
   then
-        AC_SUBST([SONAME_FLAGS], ['-Wl,-soname,${SONAME}'])
+        AC_SUBST([SONAME_FLAGS], ['-Wl,${LINKER_SONAME_OPTION},@PREFIX@/lib/${SONAME}'])
   else
       if test "$cf_use_soname" = "yes"
       then
-        AC_SUBST([SONAME_FLAGS], ['-Wl,-soname,${SONAME}'])
+        AC_SUBST([SONAME_FLAGS], ['-Wl,${LINKER_SONAME_OPTION},@PREFIX@/lib/${SONAME}'])
       fi
   fi
 fi
-- 
2.47.1

