From 8d957963a1afe035915fde9accb80d3a31430ff4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 12 Feb 2025 10:19:12 +0800
Subject: [PATCH] Process.c: fix environ declaration for macOS

---
 Process.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git Process.c Process.c
index 4e9c426..d7e282f 100644
--- Process.c
+++ Process.c
@@ -29,7 +29,12 @@
 
 /*This is code to change the command-line of a program as visible in ps */
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#else
 extern char **environ;
+#endif
 static char *TitleBuffer=NULL;
 static int TitleLen=0;
 
