--- Makefile.in.orig	2025-05-12 22:07:51.000000000 +0800
+++ Makefile.in	2025-05-14 05:17:43.000000000 +0800
@@ -17,7 +17,7 @@
 	LINKER_SONAME_OPTION = -dylib_install_name
 	SONAME = $(prefix)/lib/libUseful-$(SONAME_MAJOR_SUFFIX)
 	LIBFILE = libUseful.$(SONAME_VERSION_SUFFIX)
-	LIBFILES = $(LIBFILE) libUseful.$(SONAME_SUFFIX) libUseful-$(MAJOR).$(SONAME_SUFFIX) libUseful-$(VERSION).$(SONAME_SUFFIX) *.a
+	LIBFILES = $(LIBFILE) libUseful-$(VERSION).a
 else
 	SONAME_SUFFIX = so
 	SONAME_MAJOR_SUFFIX = $(SONAME_SUFFIX).$(MAJOR)
@@ -42,13 +42,7 @@
 
 all: $(OBJ)
 	$(CC) $(FLAGS) -shared -o $(LIBFILE) $(OBJ) $(LIBS) $(LDFLAGS)
-	-$(LN) -s -f $(LIBFILE) libUseful-$(VERSION).$(SONAME_SUFFIX)
-	-$(LN) -s -f $(LIBFILE) libUseful-$(MAJOR).$(SONAME_SUFFIX)
-	-$(LN) -s -f $(LIBFILE) $(SONAME)
-	-$(LN) -s -f $(LIBFILE) libUseful.$(SONAME_SUFFIX)
 	$(AR) rcs libUseful-$(VERSION).a $(OBJ)
-	-$(LN) -s -f libUseful-$(VERSION).a libUseful-$(MAJOR).a
-	-$(LN) -s -f libUseful-$(VERSION).a libUseful.a
 
 StrLenCache.o: StrLenCache.h StrLenCache.c
 	$(CC) $(FLAGS) -c StrLenCache.c
@@ -316,12 +310,17 @@
 	-$(MAKE) clean -C examples
 
 
-install: libUseful.$(SONAME_SUFFIX)
+install: $(LIBFILE)
 	-mkdir -p $(DESTDIR)$(libdir)
 	cp -P $(LIBFILES) $(DESTDIR)$(libdir)
+	-$(LN) -s -f $(prefix)/lib/$(LIBFILE) $(DESTDIR)$(prefix)/lib/libUseful-$(VERSION).$(SONAME_SUFFIX)
+	-$(LN) -s -f $(prefix)/lib/$(LIBFILE) $(DESTDIR)$(prefix)/lib/libUseful-$(MAJOR).$(SONAME_SUFFIX)
+	-$(LN) -s -f $(prefix)/lib/$(LIBFILE) $(DESTDIR)$(prefix)/lib/libUseful.$(SONAME_SUFFIX)
+	-$(LN) -s -f $(prefix)/lib/libUseful-$(VERSION).a $(DESTDIR)$(prefix)/lib/libUseful-$(MAJOR).a
+	-$(LN) -s -f $(prefix)/lib/libUseful-$(VERSION).a $(DESTDIR)$(prefix)/lib/libUseful.a
 	-mkdir -p $(DESTDIR)$(prefix)/include/libUseful-$(VERSION)
 	cp *.h $(DESTDIR)$(prefix)/include/libUseful-$(VERSION)
-	-$(LN) -s -r -f $(DESTDIR)$(prefix)/include/libUseful-$(VERSION) $(DESTDIR)$(prefix)/include/libUseful-5
+	-$(LN) -s -f $(prefix)/include/libUseful-$(VERSION) $(DESTDIR)$(prefix)/include/libUseful-$(MAJOR)
 	-mkdir -p $(DESTDIR)$(sysconfdir)
 	cp *.conf $(DESTDIR)$(sysconfdir)
 
