--- third_party/nativefiledialog/src/nfd_cocoa.m	2023-11-30 18:07:12.000000000 +0800
+++ third_party/nativefiledialog/src/nfd_cocoa.m	2026-01-08 17:34:28.000000000 +0800
@@ -9,6 +9,10 @@
 #include <Availability.h>
 #include "nfd.h"
 
+#ifndef NSModalResponseOK
+#define NSModalResponseOK NSOKButton
+#endif
+
 // MacOS is deprecating the allowedFileTypes property in favour of allowedContentTypes, so we have
 // to introduce this breaking change.  Define NFD_MACOS_ALLOWEDCONTENTTYPES to 1 to have it set the
 // allowedContentTypes property of the SavePanel or OpenPanel. Define
@@ -212,7 +216,7 @@
                             nfdfiltersize_t filterCount,
                             const nfdnchar_t* defaultPath) {
     nfdresult_t result = NFD_CANCEL;
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         NSWindow* keyWindow = [[NSApplication sharedApplication] keyWindow];
 
         NSOpenPanel* dialog = [NSOpenPanel openPanel];
@@ -232,7 +236,7 @@
 
         // return focus to the key window (i.e. main window)
         [keyWindow makeKeyAndOrderFront:nil];
-    }
+    [pool release];
     return result;
 }
 
@@ -241,7 +245,7 @@
                                     nfdfiltersize_t filterCount,
                                     const nfdnchar_t* defaultPath) {
     nfdresult_t result = NFD_CANCEL;
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         NSWindow* keyWindow = [[NSApplication sharedApplication] keyWindow];
 
         NSOpenPanel* dialog = [NSOpenPanel openPanel];
@@ -266,7 +270,7 @@
 
         // return focus to the key window (i.e. main window)
         [keyWindow makeKeyAndOrderFront:nil];
-    }
+    [pool release];
     return result;
 }
 
@@ -276,7 +280,7 @@
                             const nfdnchar_t* defaultPath,
                             const nfdnchar_t* defaultName) {
     nfdresult_t result = NFD_CANCEL;
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         NSWindow* keyWindow = [[NSApplication sharedApplication] keyWindow];
 
         NSSavePanel* dialog = [NSSavePanel savePanel];
@@ -302,13 +306,13 @@
 
         // return focus to the key window (i.e. main window)
         [keyWindow makeKeyAndOrderFront:nil];
-    }
+    [pool release];
     return result;
 }
 
 nfdresult_t NFD_PickFolderN(nfdnchar_t** outPath, const nfdnchar_t* defaultPath) {
     nfdresult_t result = NFD_CANCEL;
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         NSWindow* keyWindow = [[NSApplication sharedApplication] keyWindow];
 
         NSOpenPanel* dialog = [NSOpenPanel openPanel];
@@ -328,7 +332,7 @@
 
         // return focus to the key window (i.e. main window)
         [keyWindow makeKeyAndOrderFront:nil];
-    }
+    [pool release];
     return result;
 }
 
@@ -343,12 +347,12 @@
                                  nfdnchar_t** outPath) {
     const NSArray* urls = (const NSArray*)pathSet;
 
-    @autoreleasepool {
-        // autoreleasepool needed because UTF8String method might use the pool
-        const NSURL* url = [urls objectAtIndex:index];
-        const char* utf8Path = [[url path] UTF8String];
-        return CopyUtf8String(utf8Path, outPath);
-    }
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    // autoreleasepool needed because UTF8String method might use the pool
+    const NSURL* url = [urls objectAtIndex:index];
+    const char* utf8Path = [[url path] UTF8String];
+    [pool release];
+    return CopyUtf8String(utf8Path, outPath);
 }
 
 void NFD_PathSet_Free(const nfdpathset_t* pathSet) {
@@ -359,13 +363,13 @@
 nfdresult_t NFD_PathSet_GetEnum(const nfdpathset_t* pathSet, nfdpathsetenum_t* outEnumerator) {
     const NSArray* urls = (const NSArray*)pathSet;
 
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         // autoreleasepool needed because NSEnumerator uses it
         NSEnumerator* enumerator = [urls objectEnumerator];
         [enumerator retain];
         outEnumerator->ptr = (void*)enumerator;
-    }
 
+    [pool release];
     return NFD_OKAY;
 }
 
@@ -377,15 +381,16 @@
 nfdresult_t NFD_PathSet_EnumNextN(nfdpathsetenum_t* enumerator, nfdnchar_t** outPath) {
     NSEnumerator* real_enum = (NSEnumerator*)enumerator->ptr;
 
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         // autoreleasepool needed because NSURL uses it
         const NSURL* url = [real_enum nextObject];
         if (url) {
             const char* utf8Path = [[url path] UTF8String];
+            [pool release];
             return CopyUtf8String(utf8Path, outPath);
         } else {
             *outPath = NULL;
+            [pool release];
             return NFD_OKAY;
         }
-    }
 }
