--- source/window.cpp
+++ source/window.cpp	2026-01-08 19:07:25.000000000 +0800
@@ -8,7 +8,11 @@
 #include <imgui.h>
 #include <imgui_internal.h>
 #include <imgui_impl_glfw.h>
+#ifndef GLFW_COMPAT
 #include <imgui_impl_opengl3.h>
+#else
+#include <imgui_impl_opengl2.h>
+#endif
 #include <strnatcmp.h>
 #ifdef _WIN32
 #include <windowsx.h>
@@ -62,9 +66,8 @@
   glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
   glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
 #else
-  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
   glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
-  glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
 #endif
   glfwWindowHint(GLFW_AUTO_ICONIFY, GLFW_FALSE);
   glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE);

--- source/player.cpp	2026-01-08 19:46:41.000000000 +0800
+++ source/player.cpp	2026-01-08 19:49:22.000000000 +0800
@@ -7,11 +7,28 @@
 #include <romfs/romfs.hpp>
 #include <imgui.h>
 #include <imgui_internal.h>
+#ifndef GLFW_COMPAT
 #include <imgui_impl_opengl3.h>
+#else
+#include <imgui_impl_opengl2.h>
+#endif
 #include <fonts/cascadia.h>
 #include <fonts/fontawesome.h>
 #include <fonts/unifont.h>
 #include <strnatcmp.h>
+
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1070
+#define glGenFramebuffers glGenFramebuffersEXT
+#define glBindFramebuffer glBindFramebufferEXT
+#define glDeleteFramebuffers glDeleteFramebuffersEXT
+#define glFramebufferTexture2D glFramebufferTexture2DEXT
+#define GL_FRAMEBUFFER GL_FRAMEBUFFER_EXT
+#define GL_COLOR_ATTACHMENT0 GL_COLOR_ATTACHMENT0_EXT
+#endif
+#endif
+
 #include "theme.h"
 #include "player.h"
 
@@ -118,7 +135,7 @@
       glClear(GL_COLOR_BUFFER_BIT);
       glBindFramebuffer(GL_FRAMEBUFFER, 0);
     }
-
+#ifndef GLFW_COMPAT
     if (config->FontReload) {
       loadFonts();
       ImGui_ImplOpenGL3_DestroyFontsTexture();
@@ -126,6 +143,15 @@
       config->FontReload = false;
     }
     ImGui_ImplOpenGL3_NewFrame();
+#else
+    if (config->FontReload) {
+      loadFonts();
+      ImGui_ImplOpenGL2_DestroyFontsTexture();
+      ImGui_ImplOpenGL2_CreateFontsTexture();
+      config->FontReload = false;
+    }
+    ImGui_ImplOpenGL2_NewFrame();
+#endif
   }
 
   BackendNewFrame();
@@ -148,8 +174,11 @@
 
     glClearColor(0, 0, 0, 1);
     glClear(GL_COLOR_BUFFER_BIT);
-
+#ifndef GLFW_COMPAT
     ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+#else
+    ImGui_ImplOpenGL2_RenderDrawData(ImGui::GetDrawData());
+#endif
     SetSwapInterval(config->Data.Interface.Fps > 60 ? 0 : 1);
     SwapBuffers();
     mpv->reportSwap();
@@ -214,16 +238,23 @@
 #ifdef IMGUI_IMPL_OPENGL_ES3
   ImGui_ImplOpenGL3_Init("#version 300 es");
 #elif defined(__APPLE__)
+#ifndef GLFW_COMPAT
   ImGui_ImplOpenGL3_Init("#version 150");
 #else
+  ImGui_ImplOpenGL2_Init();
+#endif
+#else
   ImGui_ImplOpenGL3_Init("#version 130");
 #endif
 }
 
 void Player::exitGui() {
   MakeContextCurrent();
-
+#ifndef GLFW_COMPAT
   ImGui_ImplOpenGL3_Shutdown();
+#else
+  ImGui_ImplOpenGL2_Shutdown();
+#endif
   glDeleteTextures(1, &tex);
   glDeleteFramebuffers(1, &fbo);
 
--- CMakeLists.txt	2023-11-30 18:07:12.000000000 +0800
+++ CMakeLists.txt	2026-01-08 20:53:00.000000000 +0800
@@ -38,9 +38,8 @@
 
 set(LIBROMFS_PROJECT_NAME ${PROJECT_NAME})
 set(LIBROMFS_RESOURCE_LOCATION "${CMAKE_SOURCE_DIR}/resources/romfs")
-set(OPENGL_LIBRARIES "glad")
+set(OPENGL_LIBRARIES "-framework OpenGL")
 
-add_subdirectory(third_party/glad)
 add_subdirectory(third_party/fmt)
 add_subdirectory(third_party/natsort)
 add_subdirectory(third_party/json)
@@ -69,7 +68,7 @@
   source/main.cpp
 )
 set(INCLUDE_DIRS include ${MPV_INCLUDE_DIRS} ${GLFW_INCLUDE_DIRS} ${OPENGL_INCLUDE_DIR})
-set(LINK_LIBS glad fmt natsort json inipp nfd imgui ${CMAKE_THREAD_LIBS_INIT} ${MPV_LIBRARIES} ${GLFW_LIBRARIES} ${LIBROMFS_LIBRARY})
+set(LINK_LIBS ${OPENGL_LIBRARIES} fmt natsort json inipp nfd imgui ${CMAKE_THREAD_LIBS_INIT} ${MPV_LIBRARIES} ${GLFW_LIBRARIES} ${LIBROMFS_LIBRARY})
 
 if(WIN32)
   configure_file(${PROJECT_SOURCE_DIR}/resources/win32/app.rc.in ${PROJECT_BINARY_DIR}/app.rc @ONLY)

--- include/player.h	2023-11-30 18:07:12.000000000 +0800
+++ include/player.h	2026-01-08 21:19:36.000000000 +0800
@@ -6,8 +6,8 @@
 #include <string>
 #include <vector>
 #include <mutex>
-#ifdef IMGUI_IMPL_OPENGL_ES3
-#include <GLES3/gl3.h>
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
 #else
 #include <GL/gl.h>
 #endif

--- source/helpers/imgui.cpp	2023-11-30 18:07:12.000000000 +0800
+++ source/helpers/imgui.cpp	2026-01-08 21:26:19.000000000 +0800
@@ -3,8 +3,8 @@
 
 #define STB_IMAGE_IMPLEMENTATION
 #include <stb_image.h>
-#ifdef IMGUI_IMPL_OPENGL_ES3
-#include <GLES3/gl3.h>
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
 #else
 #include <GL/gl.h>
 #endif


From 496250753be78fd350345c031d95872bae81d339 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 11 Jan 2026 02:41:36 +0800
Subject: [PATCH] Wipe out glad

---
 include/mpv.h     | 3 +--
 include/player.h  | 1 -
 include/window.h  | 1 -
 source/mpv.cpp    | 7 ++++---
 source/player.cpp | 4 +---
 source/window.cpp | 2 --
 6 files changed, 6 insertions(+), 12 deletions(-)

diff --git include/mpv.h include/mpv.h
index 6e196a2..55cdf06 100644
--- include/mpv.h
+++ include/mpv.h
@@ -10,7 +10,6 @@
 #include <mpv/render_gl.h>
 
 namespace ImPlay {
-typedef void *(*GLAddrLoadFunc)(const char *name);
 class Mpv {
  public:
   Mpv();
@@ -20,7 +19,7 @@ class Mpv {
   using LogHandler = std::function<void(const char *, const char *, const char *)>;
   using Callback = std::function<void(Mpv *)>;
 
-  void init(GLAddrLoadFunc load, int64_t wid = 0);
+  void init(int64_t wid = 0);
   void render(int w, int h, int fbo = 0, bool flip = true);
   bool wantRender();
   void reportSwap();
diff --git include/player.h include/player.h
index d8f7411..84e1bbb 100644
--- include/player.h
+++ include/player.h
@@ -84,7 +84,6 @@ class Player {
   bool isSubtitleFile(std::string file);
 
   virtual int64_t GetWid() { return 0; }
-  virtual GLAddrLoadFunc GetGLAddrFunc() = 0;
   virtual std::string GetClipboardString() = 0;
   virtual void GetMonitorSize(int *w, int *h) = 0;
   virtual int GetMonitorRefreshRate() = 0;
diff --git include/window.h include/window.h
index b4c771f..4b763c0 100644
--- include/window.h
+++ include/window.h
@@ -42,7 +42,6 @@ class Window : Player {
 #ifdef _WIN32
   int64_t GetWid() override;
 #endif
-  GLAddrLoadFunc GetGLAddrFunc() override;
   std::string GetClipboardString() override;
   void GetMonitorSize(int *w, int *h) override;
   int GetMonitorRefreshRate() override;
diff --git source/mpv.cpp source/mpv.cpp
index c9dec52..11270e6 100644
--- source/mpv.cpp
+++ source/mpv.cpp
@@ -6,6 +6,7 @@
 #include <thread>
 #include <cstdarg>
 #include <cstring>
+#include <dlfcn.h>
 #include <nlohmann/json.hpp>
 #include "mpv.h"
 
@@ -92,13 +93,13 @@ void Mpv::reportSwap() {
   if (renderCtx != nullptr) mpv_render_context_report_swap(renderCtx);
 }
 
-static void *get_proc_address(void *ctx, const char *name) { return ((GLAddrLoadFunc)ctx)(name); }
+static void *get_proc_address(void *, const char *name) { return dlsym(RTLD_DEFAULT, name); }
 
-void Mpv::init(GLAddrLoadFunc load, int64_t wid) {
+void Mpv::init(int64_t wid) {
   if (mpv_set_property(mpv, "wid", MPV_FORMAT_INT64, &wid) < 0) throw std::runtime_error("could not set mpv wid");
   if (mpv_initialize(mpv) < 0) throw std::runtime_error("could not initialize mpv context");
   if (wid == 0) {
-    mpv_opengl_init_params gl_init_params{get_proc_address, (void *)load};
+    mpv_opengl_init_params gl_init_params{get_proc_address, nullptr};
     mpv_render_param params[]{
         {MPV_RENDER_PARAM_API_TYPE, const_cast<char *>(MPV_RENDER_API_TYPE_OPENGL)},
         {MPV_RENDER_PARAM_OPENGL_INIT_PARAMS, &gl_init_params},
diff --git source/player.cpp source/player.cpp
index 511c0fe..4c86425 100644
--- source/player.cpp
+++ source/player.cpp
@@ -70,7 +70,7 @@ bool Player::init(std::map<std::string, std::string> &options) {
   {
     ContextGuard guard(this);
     logoTexture = ImGui::LoadTexture("icon.png");
-    mpv->init(GetGLAddrFunc(), GetWid());
+    mpv->init(GetWid());
   }
 
   SetWindowDecorated(mpv->property<int, MPV_FORMAT_FLAG>("border"));
@@ -196,8 +196,6 @@ void Player::initGui() {
 
 #ifdef IMGUI_IMPL_OPENGL_ES3
   if (!gladLoadGLES2((GLADloadfunc)GetGLAddrFunc())) throw std::runtime_error("Failed to load GLES 2!");
-#else
-  if (!gladLoadGL((GLADloadfunc)GetGLAddrFunc())) throw std::runtime_error("Failed to load GL!");
 #endif
   SetSwapInterval(1);
 
diff --git source/window.cpp source/window.cpp
index e96d7a9..44d5fe5 100644
--- source/window.cpp
+++ source/window.cpp
@@ -276,8 +276,6 @@ void Window::translateMod(std::vector<std::string>& keys, int mods) {
 int64_t Window::GetWid() { return config->Data.Mpv.UseWid ? static_cast<uint32_t>((intptr_t)hwnd) : 0; }
 #endif
 
-GLAddrLoadFunc Window::GetGLAddrFunc() { return (GLAddrLoadFunc)glfwGetProcAddress; }
-
 std::string Window::GetClipboardString() {
   const char* s = glfwGetClipboardString(window);
   return s != nullptr ? s : "";
