--- source/window.cpp
+++ source/window.cpp	2026-01-08 19:07:25.000000000 +0800
@@ -8,7 +8,11 @@
 #include <imgui.h>
 #include <imgui_internal.h>
 #include <imgui_impl_glfw.h>
+#ifndef GLFW_COMPAT
 #include <imgui_impl_opengl3.h>
+#else
+#include <imgui_impl_opengl2.h>
+#endif
 #include <strnatcmp.h>
 #ifdef _WIN32
 #include <windowsx.h>
@@ -62,9 +66,8 @@
   glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
   glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
 #else
-  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
   glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
-  glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);
 #endif
   glfwWindowHint(GLFW_AUTO_ICONIFY, GLFW_FALSE);
   glfwWindowHint(GLFW_VISIBLE, GLFW_FALSE);

--- source/player.cpp	2026-01-08 19:46:41.000000000 +0800
+++ source/player.cpp	2026-01-08 19:49:22.000000000 +0800
@@ -7,11 +7,28 @@
 #include <romfs/romfs.hpp>
 #include <imgui.h>
 #include <imgui_internal.h>
+#ifndef GLFW_COMPAT
 #include <imgui_impl_opengl3.h>
+#else
+#include <imgui_impl_opengl2.h>
+#endif
 #include <fonts/cascadia.h>
 #include <fonts/fontawesome.h>
 #include <fonts/unifont.h>
 #include <strnatcmp.h>
+
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1070
+#define glGenFramebuffers glGenFramebuffersEXT
+#define glBindFramebuffer glBindFramebufferEXT
+#define glDeleteFramebuffers glDeleteFramebuffersEXT
+#define glFramebufferTexture2D glFramebufferTexture2DEXT
+#define GL_FRAMEBUFFER GL_FRAMEBUFFER_EXT
+#define GL_COLOR_ATTACHMENT0 GL_COLOR_ATTACHMENT0_EXT
+#endif
+#endif
+
 #include "theme.h"
 #include "player.h"
 
@@ -118,7 +135,7 @@
       glClear(GL_COLOR_BUFFER_BIT);
       glBindFramebuffer(GL_FRAMEBUFFER, 0);
     }
-
+#ifndef GLFW_COMPAT
     if (config->FontReload) {
       loadFonts();
       ImGui_ImplOpenGL3_DestroyFontsTexture();
@@ -126,6 +143,15 @@
       config->FontReload = false;
     }
     ImGui_ImplOpenGL3_NewFrame();
+#else
+    if (config->FontReload) {
+      loadFonts();
+      ImGui_ImplOpenGL2_DestroyFontsTexture();
+      ImGui_ImplOpenGL2_CreateFontsTexture();
+      config->FontReload = false;
+    }
+    ImGui_ImplOpenGL2_NewFrame();
+#endif
   }
 
   BackendNewFrame();
@@ -148,8 +174,11 @@
 
     glClearColor(0, 0, 0, 1);
     glClear(GL_COLOR_BUFFER_BIT);
-
+#ifndef GLFW_COMPAT
     ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+#else
+    ImGui_ImplOpenGL2_RenderDrawData(ImGui::GetDrawData());
+#endif
     SetSwapInterval(config->Data.Interface.Fps > 60 ? 0 : 1);
     SwapBuffers();
     mpv->reportSwap();
@@ -178,11 +207,6 @@
 void Player::initGui() {
   ContextGuard guard(this);
 
-#ifdef IMGUI_IMPL_OPENGL_ES3
-  if (!gladLoadGLES2((GLADloadfunc)GetGLAddrFunc())) throw std::runtime_error("Failed to load GLES 2!");
-#else
-  if (!gladLoadGL((GLADloadfunc)GetGLAddrFunc())) throw std::runtime_error("Failed to load GL!");
-#endif
   SetSwapInterval(1);
 
   IMGUI_CHECKVERSION();
@@ -214,16 +238,23 @@
 #ifdef IMGUI_IMPL_OPENGL_ES3
   ImGui_ImplOpenGL3_Init("#version 300 es");
 #elif defined(__APPLE__)
+#ifndef GLFW_COMPAT
   ImGui_ImplOpenGL3_Init("#version 150");
 #else
+  ImGui_ImplOpenGL2_Init();
+#endif
+#else
   ImGui_ImplOpenGL3_Init("#version 130");
 #endif
 }
 
 void Player::exitGui() {
   MakeContextCurrent();
-
+#ifndef GLFW_COMPAT
   ImGui_ImplOpenGL3_Shutdown();
+#else
+  ImGui_ImplOpenGL2_Shutdown();
+#endif
   glDeleteTextures(1, &tex);
   glDeleteFramebuffers(1, &fbo);
 
--- CMakeLists.txt	2023-11-30 18:07:12.000000000 +0800
+++ CMakeLists.txt	2026-01-08 20:53:00.000000000 +0800
@@ -38,9 +38,8 @@
 
 set(LIBROMFS_PROJECT_NAME ${PROJECT_NAME})
 set(LIBROMFS_RESOURCE_LOCATION "${CMAKE_SOURCE_DIR}/resources/romfs")
-set(OPENGL_LIBRARIES "glad")
+set(OPENGL_LIBRARIES "-framework OpenGL")
 
-add_subdirectory(third_party/glad)
 add_subdirectory(third_party/fmt)
 add_subdirectory(third_party/natsort)
 add_subdirectory(third_party/json)
@@ -69,7 +68,7 @@
   source/main.cpp
 )
 set(INCLUDE_DIRS include ${MPV_INCLUDE_DIRS} ${GLFW_INCLUDE_DIRS} ${OPENGL_INCLUDE_DIR})
-set(LINK_LIBS glad fmt natsort json inipp nfd imgui ${CMAKE_THREAD_LIBS_INIT} ${MPV_LIBRARIES} ${GLFW_LIBRARIES} ${LIBROMFS_LIBRARY})
+set(LINK_LIBS ${OPENGL_LIBRARIES} fmt natsort json inipp nfd imgui ${CMAKE_THREAD_LIBS_INIT} ${MPV_LIBRARIES} ${GLFW_LIBRARIES} ${LIBROMFS_LIBRARY})
 
 if(WIN32)
   configure_file(${PROJECT_SOURCE_DIR}/resources/win32/app.rc.in ${PROJECT_BINARY_DIR}/app.rc @ONLY)

--- include/player.h	2023-11-30 18:07:12.000000000 +0800
+++ include/player.h	2026-01-08 21:19:36.000000000 +0800
@@ -6,8 +6,8 @@
 #include <string>
 #include <vector>
 #include <mutex>
-#ifdef IMGUI_IMPL_OPENGL_ES3
-#include <GLES3/gl3.h>
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
 #else
 #include <GL/gl.h>
 #endif

--- source/helpers/imgui.cpp	2023-11-30 18:07:12.000000000 +0800
+++ source/helpers/imgui.cpp	2026-01-08 21:26:19.000000000 +0800
@@ -3,8 +3,8 @@
 
 #define STB_IMAGE_IMPLEMENTATION
 #include <stb_image.h>
-#ifdef IMGUI_IMPL_OPENGL_ES3
-#include <GLES3/gl3.h>
+#ifdef __APPLE__
+#include <OpenGL/gl.h>
 #else
 #include <GL/gl.h>
 #endif

