--- third_party/imgui/CMakeLists.txt	2023-11-30 18:07:12.000000000 +0800
+++ third_party/imgui/CMakeLists.txt	2026-01-11 08:49:23.000000000 +0800
@@ -11,7 +11,7 @@
   source/imgui_draw.cpp
   source/imgui_freetype.cpp
   source/imgui_impl_glfw.cpp
-  source/imgui_impl_opengl3.cpp
+  source/imgui_impl_opengl2.cpp
   source/imgui_tables.cpp
   source/imgui_widgets.cpp
   source/fonts/cascadia.c
@@ -22,7 +22,7 @@
 target_include_directories(imgui PUBLIC include ${FREETYPE_INCLUDE_DIRS} ${GLFW_INCLUDE_DIRS})
 target_link_libraries(imgui PUBLIC
   ${CMAKE_DL_LIBS} ${FREETYPE_LIBRARIES} ${GLFW_LIBRARIES}
-  $<$<BOOL:${USE_OPENGL_ES3}>:${OPENGL_LIBRARIES}>
+  ${OPENGL_LIBRARIES}
 )
 target_compile_definitions(imgui PRIVATE
   $<$<BOOL:${USE_OPENGL_ES3}>:IMGUI_IMPL_OPENGL_ES3>
