--- third_party/imgui/source/imgui_impl_glfw.cpp	2023-11-30 18:07:12.000000000 +0800
+++ third_party/imgui/source/imgui_impl_glfw.cpp	2026-01-08 12:55:45.000000000 +0800
@@ -103,8 +103,11 @@
 #include <emscripten/html5.h>
 #endif
 
-// We gather version tests as define in order to easily see which features are version-dependent.
+// Allow to override detection:
+#ifndef GLFW_VERSION_COMBINED
 #define GLFW_VERSION_COMBINED           (GLFW_VERSION_MAJOR * 1000 + GLFW_VERSION_MINOR * 100 + GLFW_VERSION_REVISION)
+#endif
+// We gather version tests as define in order to easily see which features are version-dependent.
 #define GLFW_HAS_WINDOW_TOPMOST         (GLFW_VERSION_COMBINED >= 3200) // 3.2+ GLFW_FLOATING
 #define GLFW_HAS_WINDOW_HOVERED         (GLFW_VERSION_COMBINED >= 3300) // 3.3+ GLFW_HOVERED
 #define GLFW_HAS_WINDOW_ALPHA           (GLFW_VERSION_COMBINED >= 3300) // 3.3+ glfwSetWindowOpacity

--- third_party/imgui/CMakeLists.txt	2023-11-30 18:07:12.000000000 +0800
+++ third_party/imgui/CMakeLists.txt	2026-01-08 19:58:18.000000000 +0800
@@ -11,7 +11,7 @@
   source/imgui_draw.cpp
   source/imgui_freetype.cpp
   source/imgui_impl_glfw.cpp
-  source/imgui_impl_opengl3.cpp
+  source/imgui_impl_opengl2.cpp
   source/imgui_tables.cpp
   source/imgui_widgets.cpp
   source/fonts/cascadia.c
