--- src/meson.build	2019-07-13 14:18:43.000000000 +0800
+++ src/meson.build	2025-10-22 12:14:03.000000000 +0800
@@ -23,10 +23,6 @@
   '-DG_LOG_DOMAIN="libgtkhex-3"'
 ]
 
-libghex_link_args = [
-  '-Wl,--no-undefined'
-]
-
 install_headers(
   libghex_headers,
   subdir: 'gtkhex-@0@'.format(libghex_version_major)
@@ -39,7 +35,6 @@
   include_directories: ghex_root_dir,
   dependencies: libghex_deps,
   c_args: libghex_c_args,
-  link_args: libghex_link_args,
   install: true
 )
 
--- meson.build	2019-07-13 14:18:43.000000000 +0800
+++ meson.build	2025-10-22 12:16:55.000000000 +0800
@@ -96,5 +96,3 @@
 subdir('icons')
 subdir('po')
 subdir('help')
-
-meson.add_install_script('meson_post_install.py')
