--- mesonbuild/scripts/depfixer.py	2025-04-05 00:06:24.000000000 +0800
+++ mesonbuild/scripts/depfixer.py	2025-04-19 03:37:24.000000000 +0800
@@ -418,14 +418,6 @@
             new_rpaths.add(rpath_dir)
     try:
         args = []
-        # compute diff, translate it into -delete_rpath and -add_rpath
-        # calls
-        for path in new_rpaths:
-            if path not in old_rpaths:
-                args += ['-add_rpath', path]
-        for path in old_rpaths:
-            if path not in new_rpaths:
-                args += ['-delete_rpath', path]
         # Rewrite -install_name @rpath/libfoo.dylib to /path/to/libfoo.dylib
         if fname.endswith('dylib'):
             args += ['-id', final_path]
