--- mesonbuild/linkers/linkers.py.orig	2026-04-21 07:19:41.000000000 +0800
+++ mesonbuild/linkers/linkers.py	2026-05-16 15:02:41.000000000 +0800
@@ -952,10 +952,6 @@
     def get_thinlto_cache_args(self, path: str) -> T.List[str]:
         return ["-Wl,-cache_path_lto," + path]
 
-    def get_lto_obj_cache_path(self, path: str) -> T.List[str]:
-        # https://clang.llvm.org/docs/CommandGuide/clang.html#cmdoption-flto
-        return ["-Wl,-object_path_lto," + path]
-
     def export_dynamic_args(self) -> T.List[str]:
         if mesonlib.version_compare(self.version, '>=224.1'):
             return self._apply_prefix('-export_dynamic')
