diff --git a/uppsrc/Core/config.h b/uppsrc/Core/config.h
index b9244bd93..f11453999 100644
--- a/uppsrc/Core/config.h
+++ b/uppsrc/Core/config.h
@@ -94,6 +94,21 @@
 		#define CPU_BE 1
 		#define CPU_BIG_ENDIAN 1
 		#define CPU_ALIGNED 1
+	#elif __powerpc64__ || __ppc64__
+		#define CPU_POWERPC 1
+		#define CPU_PPC 1
+		#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+			#define CPU_BE 1
+			#define CPU_BIG_ENDIAN 1
+		#endif
+	#elif __powerpc__ || __ppc__
+		#define CPU_32 1
+		#define CPU_POWERPC 1
+		#define CPU_PPC 1
+		#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+			#define CPU_BE 1
+			#define CPU_BIG_ENDIAN 1
+		#endif
 	#elif __aarch64__
 		#define CPU_ARM 1
 		#ifdef __ARM_NEON
@@ -156,12 +171,12 @@
 #endif
 
 #ifdef CPU_BIG_ENDIAN
-#error "Big endian CPUs are not supported anymore"
+	#define CPU_BE 1
+#else
+	#define CPU_LITTLE_ENDIAN 1
+	#define CPU_LE 1
 #endif
 
-#define CPU_LITTLE_ENDIAN 1
-#define CPU_LE 1
-
 #ifndef CPU_32
 #define CPU_64 1
 #endif
