diff --git a/uppsrc/Core/Ops.h b/uppsrc/Core/Ops.h
index 65893243c..1ae4a2e8f 100644
--- a/uppsrc/Core/Ops.h
+++ b/uppsrc/Core/Ops.h
@@ -1,7 +1,3 @@
-#ifndef CPU_LE
-#error Only little endian CPUs supported
-#endif
-
 #if defined(CPU_X86) && defined(COMPILER_MSC)
 
 #ifdef COMPILER_GCC
@@ -101,7 +97,7 @@ void EndianSwap(int *v, size_t count);
 void EndianSwap(int64 *v, size_t count);
 void EndianSwap(uint64 *v, size_t count);
 
-// unligned access - memcpy converts to simple load/store with normal compilers
+// unaligned access - memcpy converts to simple load/store with normal compilers
 
 inline int    Peek16(const void *ptr)        { word x; memcpy(&x, ptr, 2); return x; }
 inline int    Peek32(const void *ptr)        { dword x; memcpy(&x, ptr, 4); return x; }
@@ -111,6 +107,7 @@ inline void   Poke16(void *ptr, word val)    { memcpy(ptr, &val, 2); }
 inline void   Poke32(void *ptr, dword val)   { memcpy(ptr, &val, 4); }
 inline void   Poke64(void *ptr, int64 val)   { memcpy(ptr, &val, 8); }
 
+#ifdef CPU_LE
 inline int    Peek16le(const void *ptr)      { return Peek16(ptr); }
 inline int    Peek32le(const void *ptr)      { return Peek32(ptr); }
 inline int64  Peek64le(const void *ptr)      { return Peek64(ptr); }
@@ -126,6 +123,23 @@ inline int64  Peek64be(const void *ptr)      { return SwapEndian64(Peek64(ptr));
 inline void   Poke16be(void *ptr, word val)  { Poke16(ptr, SwapEndian16(val)); }
 inline void   Poke32be(void *ptr, dword val) { Poke32(ptr, SwapEndian32(val)); }
 inline void   Poke64be(void *ptr, int64 val) { Poke64(ptr, SwapEndian64(val)); }
+#else
+inline int    Peek16le(const void *ptr)      { return SwapEndian16(Peek16(ptr)); }
+inline int    Peek32le(const void *ptr)      { return SwapEndian32(Peek32(ptr)); }
+inline int64  Peek64le(const void *ptr)      { return SwapEndian64(Peek64(ptr)); }
+
+inline void   Poke16le(void *ptr, word val)  { Poke16(ptr, SwapEndian16(val)); }
+inline void   Poke32le(void *ptr, dword val) { Poke32(ptr, SwapEndian32(val)); }
+inline void   Poke64le(void *ptr, int64 val) { Poke64(ptr, SwapEndian64(val)); }
+
+inline int    Peek16be(const void *ptr)      { return Peek16(ptr); }
+inline int    Peek32be(const void *ptr)      { return Peek32(ptr); }
+inline int64  Peek64be(const void *ptr)      { return Peek64(ptr); }
+
+inline void   Poke16be(void *ptr, word val)  { Poke16(ptr, val); }
+inline void   Poke32be(void *ptr, dword val) { Poke32(ptr, val); }
+inline void   Poke64be(void *ptr, int64 val) { Poke64(ptr, val); }
+#endif
 
 #define MAKE2B(b0, b1)                            MAKEWORD(b0, b1)
 #define MAKE4B(b0, b1, b2, b3)                    MAKELONG(MAKEWORD(b0, b1), MAKEWORD(b2, b3))
