diff --git a/Makefile.in b/Makefile.in
index f9193af7b..f8fc32605 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -9,12 +9,14 @@
 
 UPPOUT = .cache/upp.out/
 CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) -I$(UPPDIR4) -I$(UPPDIR5) -I$(UPPDIR6) -I$(UPPDIR7) -I$(UPPDIR8) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` -I./ -I$(UPPOUT)
+CC ?= gcc
 Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
-LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = -Wl,--gc-sections $(LINKOPTIONS)
+CXX ?= g++
+LINKER ?= $(CXX)
+CFLAGS ?= -O3 -ffunction-sections -fdata-sections 
+CXXFLAGS ?= -O3 -ffunction-sections -fdata-sections -std=c++17 @ABI_FLAG@ -isystem@PREFIX@/include/LegacySupport
+LDFLAGS ?= $(LINKOPTIONS)
+LDFLAGS += -L@PREFIX@/lib -lMacportsLegacySupport
 LIBPATH =
 AR = ar -sr
 
diff --git a/umkMakefile.in b/umkMakefile.in
index 6b2d17e5e..23287a4e2 100644
--- a/umkMakefile.in
+++ b/umkMakefile.in
@@ -9,12 +9,14 @@
 
 UPPOUT = .cache/upp.out/
 CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) -I$(UPPDIR4) -I$(UPPDIR5) -I$(UPPDIR6) -I$(UPPDIR7) -I$(UPPDIR8) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` -I./ -I$(UPPOUT)
+CC ?= gcc
 Macro  =  -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
-LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = -Wl,--gc-sections $(LINKOPTIONS)
+CXX ?= g++
+LINKER ?= $(CXX)
+CFLAGS ?= -O3 -ffunction-sections -fdata-sections
+CXXFLAGS ?= -O3 -ffunction-sections -fdata-sections -std=c++17 @ABI_FLAG@ -isystem@PREFIX@/include/LegacySupport
+LDFLAGS ?= $(LINKOPTIONS)
+LDFLAGS += -L@PREFIX@/lib -lMacportsLegacySupport
 LIBPATH =
 AR = ar -sr
 
