diff --git a/uppsrc/plugin/pcx/pcxhdr.h b/uppsrc/plugin/pcx/pcxhdr.h
index 90a51ea38..e6817cb32 100644
--- a/uppsrc/plugin/pcx/pcxhdr.h
+++ b/uppsrc/plugin/pcx/pcxhdr.h
@@ -40,17 +40,16 @@ struct PCXHeader {
 	void    SwapEndian()
 	{
 #ifdef CPU_BIG_ENDIAN
-		biSize = UPP::SwapEndian(biSize);
-		biWidth = UPP::SwapEndian(biWidth);
-		biHeight = UPP::SwapEndian(biHeight);
-		biPlanes = UPP::SwapEndian(biPlanes);
-		biBitCount = UPP::SwapEndian(biBitCount);
-		biCompression = UPP::SwapEndian(biCompression);
-		biSizeImage = UPP::SwapEndian(biSizeImage);
-		biXPelsPerMeter = UPP::SwapEndian(biXPelsPerMeter);
-		biYPelsPerMeter = UPP::SwapEndian(biYPelsPerMeter);
-		biClrUsed = UPP::SwapEndian(biClrUsed);
-		biClrImportant = UPP::SwapEndian(biClrImportant);
+		minX = UPP::SwapEndian16(minX);
+		minY = UPP::SwapEndian16(minY);
+		maxX = UPP::SwapEndian16(maxX);
+		maxY = UPP::SwapEndian16(maxY);
+		horzDpi = UPP::SwapEndian16(horzDpi);
+		vertDpi = UPP::SwapEndian16(vertDpi);
+		bytesPerLine = UPP::SwapEndian16(bytesPerLine);
+		paltype = UPP::SwapEndian16(paltype);
+		hScreenSize = UPP::SwapEndian16(hScreenSize);
+		vScreenSize = UPP::SwapEndian16(vScreenSize);
 #endif
 	}
 
