diff --git a/uppsrc/Draw/Drawing.cpp b/uppsrc/Draw/Drawing.cpp
index 088275185..295e22000 100644
--- a/uppsrc/Draw/Drawing.cpp
+++ b/uppsrc/Draw/Drawing.cpp
@@ -46,8 +46,8 @@ static void StreamUnpackPoints(Stream& stream, Point *out, int count)
 	byte *top = reinterpret_cast<byte *>(end) - count * 8;
 	stream.Get(top, count * 8);
 	for(; out < end; out++, top += 8) {
-		out -> x = (short)Peek32le(top + 0);
-		out -> y = (short)Peek32le(top + 4);
+		out -> x = Peek32le(top + 0);
+		out -> y = Peek32le(top + 4);
 	}
 }
 
@@ -68,7 +68,7 @@ static void StreamPackPoints(Stream& stream, const Point *in, int count)
 			Poke32le(pp + 0, in -> x);
 			Poke32le(pp + 4, in -> y);
 		}
-		stream.Put(part, part_count * 4);
+		stream.Put(part, part_count * 8);
 		count -= part_count;
 	}
 }
@@ -265,7 +265,10 @@ void DrawingDraw::DrawTextOp(int x, int y, int angle, const wchar *text, Font fo
 #ifdef CPU_LE
 	s.Put(text, n * sizeof(wchar));
 #else
-	#error big endiand not supported
+	Buffer<wchar> txt(n);
+	memcpy(txt, text, n * sizeof(wchar));
+	EndianSwap((dword *)(wchar *)txt, n);
+	s.Put(txt, n * sizeof(wchar));
 #endif
 	bool dxb = dx;
 	s % dxb;
@@ -558,6 +561,9 @@ void Draw::DrawDrawingOp(const Rect& target, const Drawing& w) {
 				if(cs == CHARSET_UTF32) {
 					Buffer<wchar> txt(n);
 					ps.Stream::Get(txt, n * sizeof(wchar));
+#ifdef CPU_BE
+					EndianSwap((dword *)(wchar *)txt, n);
+#endif
 					text = WString(txt, n);
 				}
 				else
