diff --git a/Makefile.in b/Makefile.in
index f9193af7b..f8fc32605 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -5,12 +5,14 @@ UPPDIR4 = /home/cxl/.config/u++/theide/UppHub/Functions4U/
 
 UPPOUT = .cache/upp.out/
 CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) -I$(UPPDIR4) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` `pkg-config --cflags xinerama` `pkg-config --cflags xrender` `pkg-config --cflags xft` `pkg-config --cflags xdmcp` `pkg-config --cflags xext` `pkg-config --cflags gtk+-3.0` `pkg-config --cflags libnotify` -I./ -I$(UPPOUT)
+CC ?= cc
 Macro  =  -DflagGUI -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
-LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = -Wl,--gc-sections $(LINKOPTIONS)
+CXX ?= c++
+LINKER ?= $(CXX)
+CFLAGS ?= -O3 -ffunction-sections -fdata-sections
+CXXFLAGS ?= -O3 -ffunction-sections -fdata-sections -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -isystem/opt/local/include/LegacySupport
+LDFLAGS ?= $(LINKOPTIONS)
+LDFLAGS += -L/opt/local/lib -lMacportsLegacySupport
 LIBPATH =
 AR = ar -sr
 
diff --git a/umkMakefile.in b/umkMakefile.in
index 6b2d17e5e..23287a4e2 100644
--- a/umkMakefile.in
+++ b/umkMakefile.in
@@ -5,12 +5,14 @@ UPPDIR4 = /home/cxl/.config/u++/theide/UppHub/Functions4U/
 
 UPPOUT = .cache/upp.out/
 CINC   =  -I$(UPPDIR1) -I$(UPPDIR2) -I$(UPPDIR3) -I$(UPPDIR4) `pkg-config --cflags libpng` `pkg-config --cflags freetype2` `pkg-config --cflags fontconfig` `pkg-config --cflags x11` `pkg-config --cflags xcb` `pkg-config --cflags expat` -I./ -I$(UPPOUT)
+CC ?= cc
 Macro  =  -DflagGCC -DflagSHARED -DflagPOSIX -DflagLINUX
-CXX = c++
-LINKER = $(CXX)
-CFLAGS = -O3 -ffunction-sections -fdata-sections 
-CXXFLAGS = -O3 -ffunction-sections -fdata-sections  -std=c++17
-LDFLAGS = -Wl,--gc-sections $(LINKOPTIONS)
+CXX ?= c++
+LINKER ?= $(CXX)
+CFLAGS ?= -O3 -ffunction-sections -fdata-sections
+CXXFLAGS ?= -O3 -ffunction-sections -fdata-sections -std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -isystem/opt/local/include/LegacySupport
+LDFLAGS ?= $(LINKOPTIONS)
+LDFLAGS += -L/opt/local/lib -lMacportsLegacySupport
 LIBPATH =
 AR = ar -sr
 
diff --git a/uppsrc/Core/HeapImp.h b/uppsrc/Core/HeapImp.h
index 12128ae7a..5a828d309 100644
--- a/uppsrc/Core/HeapImp.h
+++ b/uppsrc/Core/HeapImp.h
@@ -13,8 +13,8 @@ struct Heap;
 struct BlkPrefix { // this part is at the start of Blk allocated block, client must not touch it
 	word        prev_size;
 	word        size;
-	bool        free;
-	bool        last;
+	byte        free;  // Use byte instead of bool for consistent 1-byte size across platforms
+	byte        last;  // Use byte instead of bool for consistent 1-byte size across platforms
 	Heap       *heap; // we need this for 4KB pages and large blocks, NULL for Huge blocks
 #ifdef CPU_32
 	dword       filler;
