--- a/uppsrc/ide/Builders/Install.cpp
+++ b/uppsrc/ide/Builders/Install.cpp	2026-03-21 13:20:05.000000000 +0800
@@ -31,6 +31,33 @@
 LIB = "$LIB$";
 LINKMODE_LOCK = "0";)";
 
+const char *gcc_bm =
+R"(BUILDER = "GCC";
+COMPILER = "@CXX@";
+COMMON_OPTIONS = "$COMMON$";
+COMMON_CPP_OPTIONS = "-std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -isystem@PREFIX@/include/LegacySupport";
+COMMON_C_OPTIONS = "-isystem@PREFIX@/include/LegacySupport";
+COMMON_LINK = "$COMMON$ -L@PREFIX@/lib -lMacportsLegacySupport";
+COMMON_FLAGS = "";
+DEBUG_INFO = "2";
+DEBUG_BLITZ = "1";
+DEBUG_LINKMODE = "1";
+DEBUG_OPTIONS = "-O0";
+DEBUG_FLAGS = "";
+DEBUG_LINK = "";
+RELEASE_BLITZ = "1";
+RELEASE_LINKMODE = "1";
+RELEASE_OPTIONS = "-O3 -ffunction-sections -fdata-sections";
+RELEASE_FLAGS = "";
+RELEASE_LINK = "";
+DEBUGGER = "gdb";
+ALLOW_PRECOMPILED_HEADERS = "0";
+DISABLE_BLITZ = "0";
+PATH = "";
+INCLUDE = "$INCLUDE$";
+LIB = "$LIB$";
+LINKMODE_LOCK = "0";)";
+
 #elif PLATFORM_SOLARIS
 
 const char *gcc_bm =
@@ -148,9 +175,9 @@
 void CreateBuildMethods()
 {
 #ifdef PLATFORM_COCOA
-	String bm_path = ConfigFile("CLANG.bm");
+	String bm_path = ConfigFile("GCC.bm");
 	if(IsNull(LoadFile(bm_path))) {
-		String bm = clang_bm;
+		String bm = gcc_bm;
 		
 		auto Path = [&](const char *var, const char *path) {
 			String h;
@@ -160,8 +187,8 @@
 			bm.Replace(var, h);
 		};
 		
-		Path("$INCLUDE$", "/opt/local/include;/usr/include;/usr/local/include;/opt/homebrew/include;/opt/homebrew/opt/openssl/include");
-		Path("$LIB$", "/opt/local/lib;/usr/lib;/usr/local/lib;/opt/homebrew/lib;/opt/homebrew/opt/openssl/lib");
+		Path("$INCLUDE$", "@PREFIX@/include/LegacySupport;@PREFIX@/include;/usr/include");
+		Path("$LIB$", "@PREFIX@/lib/libgcc;@PREFIX@/lib;/usr/lib");
 
 		// No hardcoded architecture flags - let compiler use native architecture
 		bm.Replace("$COMMON$", "");
