From 0977efca1fd974ea1fab45d1d46a0f0e40fd2454 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 11 Apr 2025 16:00:42 +0800
Subject: [PATCH 01/60] KeyBindingsDialog.cpp: setSectionResizeMode to
 setResizeMode

---
 src/gui/KeyBindingsDialog.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/KeyBindingsDialog.cpp b/src/gui/KeyBindingsDialog.cpp
index 3dca5428..4c71bdd0 100644
--- a/src/gui/KeyBindingsDialog.cpp
+++ b/src/gui/KeyBindingsDialog.cpp
@@ -35,8 +35,8 @@ KeyBindingsDialog::KeyBindingsDialog(QWidget *p) :
     shortcuts->setModel(QMPlay2GUI.shortcutHandler);
     shortcuts->setFrameShape(QFrame::NoFrame);
     shortcuts->setAlternatingRowColors(true);
-    shortcuts->horizontalHeader()->setSectionResizeMode(0, QHeaderView::ResizeToContents);
-    shortcuts->horizontalHeader()->setSectionResizeMode(1, QHeaderView::Stretch);
+    shortcuts->horizontalHeader()->setResizeMode(0, QHeaderView::ResizeToContents);
+    shortcuts->horizontalHeader()->setResizeMode(1, QHeaderView::Stretch);
     shortcuts->setSelectionMode(QAbstractItemView::SingleSelection);
     shortcuts->verticalHeader()->setVisible(false);
 
-- 
2.49.0

