From 8d5149a7711a1564738f41d7e2853be9cd1967bf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 11 Apr 2025 17:00:48 +0800
Subject: [PATCH 03/60] PlaylistDock.cpp: compat

---
 src/gui/PlaylistDock.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/PlaylistDock.cpp b/src/gui/PlaylistDock.cpp
index f2872ab0..29e5de72 100644
--- a/src/gui/PlaylistDock.cpp
+++ b/src/gui/PlaylistDock.cpp
@@ -56,7 +56,7 @@ PlaylistDock::PlaylistDock() :
 
     connect(list, SIGNAL(itemDoubleClicked(QTreeWidgetItem *, int)), this, SLOT(itemDoubleClicked(QTreeWidgetItem *)));
     connect(list, SIGNAL(returnItem(QTreeWidgetItem *)), this, SLOT(addAndPlay(QTreeWidgetItem *)));
-    connect(list, &PlaylistWidget::itemExpanded, this, &PlaylistDock::maybeDoQuickSync, Qt::QueuedConnection); // Must be queued to not crash at startup in some cases
+    connect(list, SIGNAL(itemExpanded(QTreeWidgetItem *)), this, SLOT(maybeDoQuickSync(QTreeWidgetItem *)), Qt::QueuedConnection);
     connect(list, SIGNAL(visibleItemsCount(int)), this, SLOT(visibleItemsCount(int)));
     connect(list, SIGNAL(addStatus(bool)), findE, SLOT(setDisabled(bool)));
     connect(findE, SIGNAL(textChanged(const QString &)), this, SLOT(findItems(const QString &)));
-- 
2.49.0

