--- a/src/qmplay2/NotifiesMacOS.mm
+++ b/src/qmplay2/NotifiesMacOS.mm	2026-02-25 03:49:29.000000000 +0800
@@ -3,7 +3,12 @@
 #import <AppKit/AppKit.h>
 
 #include <QPixmap>
+
+#if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
 #include <QTimer>
+#endif
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
 
 @interface QMPlay2NotificationItem : NSObject<NSUserNotificationCenterDelegate>
 {}
@@ -32,26 +37,41 @@
     [notification release];
 }
 
+#endif // MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
+
 /**/
 
 NotifiesMacOS::NotifiesMacOS()
 {
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
     QMPlay2NotificationItem *notificationItem = [QMPlay2NotificationItem alloc];
     [[NSUserNotificationCenter defaultUserNotificationCenter] setDelegate:notificationItem];
     m_notificationItem = notificationItem;
+#else
+    m_notificationItem = nullptr;
+#endif
 }
 NotifiesMacOS::~NotifiesMacOS()
 {
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
     [[NSUserNotificationCenter defaultUserNotificationCenter] setDelegate:nil];
     [(QMPlay2NotificationItem *)m_notificationItem release];
     for (auto &&notification : m_notifications)
         releaseNotification((NSUserNotification *)notification);
+#endif
 }
 
 bool NotifiesMacOS::doNotify(const QString &title, const QString &message, const int ms, const QPixmap &pixmap, const int iconId)
 {
     Q_UNUSED(iconId)
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1080
+    Q_UNUSED(title)
+    Q_UNUSED(message)
+    Q_UNUSED(ms)
+    Q_UNUSED(pixmap)
+    return false;
+#else
     if (!m_notificationItem)
         return false;
 
@@ -113,6 +133,7 @@
     }
 
     return true;
+#endif
 }
 bool NotifiesMacOS::doNotify(const QString &title, const QString &message, const int ms, const QImage &image, const int iconId)
 {

--- a/src/gui/macOS/ScreenSaver.cpp.orig	2025-06-07 05:45:55.000000000 +0800
+++ b/src/gui/macOS/ScreenSaver.cpp	2026-02-25 04:48:36.000000000 +0800
@@ -20,6 +20,13 @@
 
 #include <IOKit/pwr_mgt/IOPMLib.h>
 
+#ifndef kIOPMAssertPreventUserIdleDisplaySleep
+#define kIOPMAssertPreventUserIdleDisplaySleep CFSTR("PreventUserIdleDisplaySleep")
+#endif
+#ifndef kIOPMAssertPreventUserIdleSystemSleep
+#define kIOPMAssertPreventUserIdleSystemSleep CFSTR("PreventUserIdleSystemSleep")
+#endif
+
 #define QMPLAY2_MEDIA_PLAYBACK CFSTR("QMPlay2 media playback")
 
 class ScreenSaverPriv
