From 6b3e9abe8340d2671148b83d57d90cfd34174d49 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 11 Apr 2025 17:06:20 +0800
Subject: [PATCH 04/60] PlaylistWidget.cpp: compat

---
 src/gui/PlaylistWidget.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/gui/PlaylistWidget.cpp b/src/gui/PlaylistWidget.cpp
index 87164c7c..c529fa8c 100644
--- a/src/gui/PlaylistWidget.cpp
+++ b/src/gui/PlaylistWidget.cpp
@@ -595,7 +595,7 @@ PlaylistWidget::PlaylistWidget() :
     setAnimated(true);
     header()->setStretchLastSection(false);
     setHeaderHidden(true);
-    header()->setSectionResizeMode(0, QHeaderView::Stretch);
+    header()->setResizeMode(0, QHeaderView::Stretch);
     header()->hideSection(1);
     setItemsResizeToContents(true);
     setIconSize({22, 22});
@@ -608,7 +608,7 @@ PlaylistWidget::PlaylistWidget() :
     connect(&animationTimer, SIGNAL(timeout()), this, SLOT(animationUpdate()));
     connect(&addTimer, SIGNAL(timeout()), this, SLOT(addTimerElapsed()));
     connect(&addThr, SIGNAL(status(bool)), this, SIGNAL(addStatus(bool)));
-    connect(playlistMenu(), &MenuBar::Playlist::aboutToShow, this, &PlaylistWidget::createExtensionsMenu);
+    connect(playlistMenu(), SIGNAL(aboutToShow()), this, SLOT(createExtensionsMenu()));
 }
 
 QString PlaylistWidget::getUrl(QTreeWidgetItem *tWI) const
@@ -622,7 +622,7 @@ void PlaylistWidget::setItemsResizeToContents(bool b)
 {
     const QHeaderView::ResizeMode rm = b ? QHeaderView::ResizeToContents : QHeaderView::Fixed;
     for (int i = 1; i <= 2; ++i)
-        header()->setSectionResizeMode(i, rm);
+        header()->setResizeMode(i, rm);
 }
 
 void PlaylistWidget::sortCurrentGroup(int column, Qt::SortOrder sortOrder)
-- 
2.49.0

