--- a/src/modules/PortAudio/3rdparty/CoreAudio/AudioDeviceList.cpp	2019-09-04 01:24:49.000000000 +0800
+++ b/src/modules/PortAudio/3rdparty/CoreAudio/AudioDeviceList.cpp	2025-07-25 05:01:42.000000000 +0800
@@ -48,6 +48,10 @@
 
 #include "AudioDeviceList.h"
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+    #define __Verify_noErr verify_noerr
+#endif
+
 AudioDeviceList::AudioDeviceList(bool forInput)
     : mForInput(forInput)
 {

--- a/src/modules/PortAudio/3rdparty/CoreAudio/AudioDevice.mm	2019-09-04 01:24:49.000000000 +0800
+++ b/src/modules/PortAudio/3rdparty/CoreAudio/AudioDevice.mm	2025-07-25 05:13:50.000000000 +0800
@@ -47,7 +47,11 @@
 */
 
 #include "AudioDevice.h"
-#import <Cocoa/Cocoa.h>
+#include <Cocoa/Cocoa.h>
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+    #define __Verify_noErr verify_noerr
+#endif
 
 char *OSTStr(OSType type)
 {
