--- a/src/gui/PlayClass.cpp	2026-02-26 13:20:51.000000000 +0800
+++ b/src/gui/PlayClass.cpp	2026-03-06 14:37:47.000000000 +0800
@@ -37,10 +37,15 @@
 #include <QApplication>
 #include <QVarLengthArray>
 #include <QMessageBox>
-#include <QRawFont>
 #include <QAction>
 #include <QDir>
 
+#ifdef USE_QRAWFONT
+#include <QRawFont>
+#else
+#include <QFontDatabase>
+#endif
+
 #include <cmath>
 
 PlayClass::PlayClass() :
@@ -397,9 +402,20 @@
                     {
                         const QByteArray fontData = f.readAll();
                         f.close();
+#ifdef USE_QRAWFONT
                         const QString fontName = QRawFont(fontData, 0.0).familyName();
                         if (!fontName.isEmpty())
                             ass->addFont(fontName.toUtf8(), fontData);
+#else
+						const int fontID = QFontDatabase::addApplicationFontFromData(fontData);
+						if (fontID != -1)
+						{
+							const QStringList fontFamilies = QFontDatabase::applicationFontFamilies(fontID);
+							QFontDatabase::removeApplicationFont(fontID);
+							if (!fontFamilies.isEmpty())
+								ass->addFont(fontFamilies.first().toUtf8(), fontData);
+						}
+#endif
                     }
                 }
 
