From 32251422d812bbad5445bb0f15106c3c96433201 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 19:27:39 +0800
Subject: [PATCH 13/60] StreamInfo.hpp

---
 src/qmplay2/headers/StreamInfo.hpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/qmplay2/headers/StreamInfo.hpp b/src/qmplay2/headers/StreamInfo.hpp
index 6a8951ac..ec830b88 100644
--- a/src/qmplay2/headers/StreamInfo.hpp
+++ b/src/qmplay2/headers/StreamInfo.hpp
@@ -26,6 +26,8 @@
 #include <QVector>
 #include <QPair>
 
+#include <qnumeric.h> // qQNaN
+
 using QMPlay2Tag = QPair<QString, QString>;
 
 enum QMPlay2MediaType
-- 
2.49.0

