From 56a72e35c3a720b31b9e7b7ded77deeee703a15b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 19:36:00 +0800
Subject: [PATCH 16/60] QMPlay2Core.cpp

---
 src/qmplay2/QMPlay2Core.cpp | 25 ++++++-------------------
 1 file changed, 6 insertions(+), 19 deletions(-)

diff --git a/src/qmplay2/QMPlay2Core.cpp b/src/qmplay2/QMPlay2Core.cpp
index 55f775da..56a04b28 100644
--- a/src/qmplay2/QMPlay2Core.cpp
+++ b/src/qmplay2/QMPlay2Core.cpp
@@ -28,7 +28,6 @@
 #include <Version.hpp>
 #include <Module.hpp>
 
-#include <QLoggingCategory>
 #include <QApplication>
 #include <QLibraryInfo>
 #include <QTranslator>
@@ -36,19 +35,19 @@
 #include <QLibrary>
 #include <QPointer>
 #include <QLocale>
-#include <QWindow>
 #include <QFile>
 #include <QDir>
 #if defined Q_OS_WIN
     #include <windows.h>
     #include <powrprof.h>
-#elif defined Q_OS_MACOS
-    #include <QStandardPaths>
 #endif
 
 #include <cstdarg>
 #include <cstdio>
 
+#define QT_VERSION_MAJOR 4
+#define QT_VERSION_MINOR 8 // Qt 4.8.x is the oldest supported Qt version
+
 extern "C"
 {
     #include <libavformat/avformat.h>
@@ -58,19 +57,8 @@ extern "C"
 
 /**/
 
-Q_LOGGING_CATEGORY(ffmpeglog, "FFmpegLog")
-
 static void avQMPlay2LogHandler(void *avcl, int level, const char *fmt, va_list vl)
 {
-    if (level <= AV_LOG_FATAL)
-    {
-        const QByteArray msg = QString::vasprintf(fmt, vl).trimmed().toUtf8();
-        qCCritical(ffmpeglog) << msg.constData();
-    }
-    else
-    {
-        av_log_default_callback(avcl, level, fmt, vl);
-    }
 }
 
 /**/
@@ -109,6 +97,8 @@ QMPlay2CoreClass::QMPlay2CoreClass()
 {
     qmplay2Core = this;
 
+    videoDevicePixelRatio = 1.0;
+
     QFile f(":/Languages.txt");
     if (f.open(QFile::ReadOnly))
     {
@@ -203,8 +193,6 @@ void QMPlay2CoreClass::init(bool loadModules, bool modulesInSubdirs, const QStri
     {
 #if defined(Q_OS_WIN)
         settingsDir = QFileInfo(QSettings(QSettings::IniFormat, QSettings::UserScope, QString()).fileName()).absolutePath() + "/QMPlay2/";
-#elif defined(Q_OS_MACOS)
-        settingsDir = Functions::cleanPath(QStandardPaths::standardLocations(QStandardPaths::DataLocation).value(0, settingsDir));
 #else
         settingsDir = QDir::homePath() + "/.qmplay2/";
 #endif
@@ -414,9 +402,8 @@ QStringList QMPlay2CoreClass::getModules(const QString &type, int typeLen) const
     return modules + availableModules;
 }
 
-qreal QMPlay2CoreClass::getVideoDevicePixelRatio() const
+void QMPlay2CoreClass::setVideoDevicePixelRatio()
 {
-    return getVideoDock()->devicePixelRatioF();
 }
 
 QIcon QMPlay2CoreClass::getIconFromTheme(const QString &iconName, const QIcon &fallback)
-- 
2.49.0

