From 68df29b32a0ad61b38c7647ab222773003605dbc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 20:08:15 +0800
Subject: [PATCH 17/60] QMPlay2OSD.cpp

---
 src/qmplay2/QMPlay2OSD.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/qmplay2/QMPlay2OSD.cpp b/src/qmplay2/QMPlay2OSD.cpp
index d77bc2ce..7c0ff030 100644
--- a/src/qmplay2/QMPlay2OSD.cpp
+++ b/src/qmplay2/QMPlay2OSD.cpp
@@ -17,13 +17,13 @@
 */
 
 #include <QMPlay2OSD.hpp>
+#include <QAtomicInt>
 
-#include <QAtomicInteger>
-static QAtomicInteger<quint64> g_id;
+static QAtomicInt g_id;
 
 void QMPlay2OSD::genId()
 {
-    m_id = ++g_id;
+    m_id = g_id.fetchAndAddOrdered(1) + 1;
 }
 
 void QMPlay2OSD::clear(bool all)
-- 
2.49.0

