From 5acbe88823b41cc7ae07e033b39657f0e4df7e8b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 20:06:53 +0800
Subject: [PATCH 18/60] Writer.cpp

---
 src/qmplay2/Writer.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/qmplay2/Writer.cpp b/src/qmplay2/Writer.cpp
index 5eb05afd..a09bda50 100644
--- a/src/qmplay2/Writer.cpp
+++ b/src/qmplay2/Writer.cpp
@@ -20,7 +20,7 @@
 
 #include <Functions.hpp>
 
-#include <QSaveFile>
+#include <QFile>
 #include <QBuffer>
 
 #include <memory>
@@ -52,8 +52,10 @@ class QMPlay2FileWriter : public IODeviceWriter
 {
     ~QMPlay2FileWriter()
     {
-        if (auto f = static_cast<QSaveFile *>(m_io.get()))
-            f->commit();
+        if (auto f = static_cast<QFile *>(m_io.get()))
+        {
+            f->close();
+        }
     }
 
     QString name() const override final
@@ -63,7 +65,7 @@ class QMPlay2FileWriter : public IODeviceWriter
 
     bool open() override final
     {
-        m_io.reset(new QSaveFile(getUrl().mid(7)));
+        m_io.reset(new QFile(getUrl().mid(7)));
         return IODeviceWriter::open();
     }
 };
-- 
2.49.0

