From be437ebcb95491e92d60403c69983ae990a905d9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 21:19:30 +0800
Subject: [PATCH 23/60] ColorButton.hpp

---
 src/qmplay2/headers/ColorButton.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/qmplay2/headers/ColorButton.hpp b/src/qmplay2/headers/ColorButton.hpp
index 75aa784a..a9b8c5e0 100644
--- a/src/qmplay2/headers/ColorButton.hpp
+++ b/src/qmplay2/headers/ColorButton.hpp
@@ -22,7 +22,7 @@
 
 #include <QPushButton>
 
-class QMPLAY2SHAREDLIB_EXPORT ColorButton final : public QPushButton
+class ColorButton : public QPushButton
 {
     Q_OBJECT
 public:
@@ -34,7 +34,7 @@ public:
         return m_color;
     }
 protected:
-    void paintEvent(QPaintEvent *) override;
+    void paintEvent(QPaintEvent *) override final;
 private:
     QColor m_color;
     bool m_alphaChannel;
-- 
2.49.0

