From 29a0a7dd07d09ba4b82de4505df4056421a69a3a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 21:40:22 +0800
Subject: [PATCH 24/60] Slider.hpp

---
 src/qmplay2/headers/Slider.hpp | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/qmplay2/headers/Slider.hpp b/src/qmplay2/headers/Slider.hpp
index 52acb3b0..d8d49ee9 100644
--- a/src/qmplay2/headers/Slider.hpp
+++ b/src/qmplay2/headers/Slider.hpp
@@ -22,7 +22,7 @@
 
 #include <QSlider>
 
-class QMPLAY2SHAREDLIB_EXPORT Slider final : public QSlider
+class Slider : public QSlider
 {
     Q_OBJECT
 public:
@@ -45,12 +45,12 @@ public slots:
     }
     void drawRange(int first, int second);
 protected:
-    void paintEvent(QPaintEvent *) override;
-    void mousePressEvent(QMouseEvent *) override;
-    void mouseReleaseEvent(QMouseEvent *) override;
-    void mouseMoveEvent(QMouseEvent *) override;
-    void wheelEvent(QWheelEvent *) override;
-    void enterEvent(QEvent *) override;
+    void paintEvent(QPaintEvent *) override final;
+    void mousePressEvent(QMouseEvent *) override final;
+    void mouseReleaseEvent(QMouseEvent *) override final;
+    void mouseMoveEvent(QMouseEvent *) override final;
+    void wheelEvent(QWheelEvent *) override final;
+    void enterEvent(QEvent *) override final;
 private:
     int getMousePos(const QPoint &pos);
 
-- 
2.49.0

