From c2eb80fd37bb2506624f2fcccf777390b106fb6f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 21:45:10 +0800
Subject: [PATCH 26/60] InDockW.hpp

---
 src/qmplay2/headers/InDockW.hpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/qmplay2/headers/InDockW.hpp b/src/qmplay2/headers/InDockW.hpp
index d4b51c50..67504fc5 100644
--- a/src/qmplay2/headers/InDockW.hpp
+++ b/src/qmplay2/headers/InDockW.hpp
@@ -23,7 +23,7 @@
 #include <QPointer>
 #include <QWidget>
 
-class QMPLAY2SHAREDLIB_EXPORT InDockW final : public QWidget
+class InDockW : public QWidget
 {
     Q_OBJECT
 public:
@@ -41,12 +41,12 @@ private:
     QPointer<QWidget> w;
 private slots:
     void wallpaperChanged(bool hasWallpaper, double alpha);
-public:
+public slots:
     void setWidget(QWidget *newW);
 protected:
-    void resizeEvent(QResizeEvent *) override;
-    void paintEvent(QPaintEvent *) override;
-    bool event(QEvent *) override;
+    void resizeEvent(QResizeEvent *) override final;
+    void paintEvent(QPaintEvent *) override final;
+    bool event(QEvent *) override final;
 signals:
     void resized(int, int);
     void itemDropped(const QString &);
-- 
2.49.0

