From 5341c5f5df58b624bebe779e796779198ae504b7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 15 Apr 2025 23:39:32 +0800
Subject: [PATCH 34/60] PlayClass.hpp

---
 src/gui/PlayClass.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gui/PlayClass.hpp b/src/gui/PlayClass.hpp
index 41e4e629..bc4152ed 100644
--- a/src/gui/PlayClass.hpp
+++ b/src/gui/PlayClass.hpp
@@ -44,7 +44,7 @@ enum
     SEEK_REPEAT = -3
 };
 
-class PlayClass final : public QObject
+class PlayClass : public QObject
 {
     Q_OBJECT
     friend class DemuxerThr;
@@ -53,7 +53,7 @@ class PlayClass final : public QObject
     friend class AudioThr;
 public:
     PlayClass();
-    ~PlayClass();
+    ~PlayClass() final;
 
     Q_SLOT void play(const QString &);
     Q_SLOT void stop(bool quitApp = false);
-- 
2.49.0

