From 4b5a12cac153df2fcc7b88ca751a262863047d6a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 16 Apr 2025 01:54:00 +0800
Subject: [PATCH 38/60] LastFM.hpp

---
 src/modules/Extensions/LastFM.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/Extensions/LastFM.hpp b/src/modules/Extensions/LastFM.hpp
index 3d2b3e73..c8abaa10 100644
--- a/src/modules/Extensions/LastFM.hpp
+++ b/src/modules/Extensions/LastFM.hpp
@@ -29,7 +29,7 @@
 
 class QImage;
 
-class LastFM final : public QObject, public QMPlay2Extensions
+class LastFM : public QObject, public QMPlay2Extensions
 {
     Q_OBJECT
 public:
@@ -49,7 +49,7 @@ public:
 
     LastFM(Module &module);
 private:
-    bool set() override;
+    bool set() override final;
 
     void getAlbumCover(const QString &title, const QString &artist, const QString &album, bool titleAsAlbum = false);
 
-- 
2.49.0

