From d811961faade74e552eed04b5eb6e0e567c294be Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 16 Apr 2025 01:54:58 +0800
Subject: [PATCH 39/60] Lyrics.hpp

---
 src/modules/Extensions/Lyrics.hpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/Extensions/Lyrics.hpp b/src/modules/Extensions/Lyrics.hpp
index d4f954a3..572571cc 100644
--- a/src/modules/Extensions/Lyrics.hpp
+++ b/src/modules/Extensions/Lyrics.hpp
@@ -25,15 +25,15 @@
 #include <QTextEdit>
 #include <QPointer>
 
-class Lyrics final : public QTextEdit, public QMPlay2Extensions
+class Lyrics : public QTextEdit, public QMPlay2Extensions
 {
     Q_OBJECT
 
 public:
     Lyrics(Module &module);
-    ~Lyrics();
+    ~Lyrics() final;
 
-    DockWidget *getDockWidget() override;
+    DockWidget *getDockWidget() override final;
 
 private slots:
     void visibilityChanged(bool v);
-- 
2.49.0

