From 0f23fcab4b7332ee5148f95ec572a528e4a60b74 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 16 Apr 2025 05:06:57 +0800
Subject: [PATCH 46/60] Down

---
 src/modules/Extensions/Downloader.hpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/modules/Extensions/Downloader.hpp b/src/modules/Extensions/Downloader.hpp
index 6163b945..ad6790dd 100644
--- a/src/modules/Extensions/Downloader.hpp
+++ b/src/modules/Extensions/Downloader.hpp
@@ -34,7 +34,7 @@ class QProgressBar;
 class QTreeWidgetItem;
 class DownloaderThread;
 
-class DownloadItemW public QWidget, public QObject
+class DownloadItemW : public QWidget
 {
     Q_OBJECT
 public:
@@ -103,7 +103,7 @@ private:
 
 /**/
 
-class DownloadListW public QTreeWidget
+class DownloadListW : public QTreeWidget
 {
     friend class Downloader;
 public:
@@ -117,7 +117,7 @@ private:
 
 /**/
 
-class DownloaderThread public QThread
+class DownloaderThread : public QThread
 {
     Q_OBJECT
     enum {ADD_ENTRY, NAME, SET, SET_POS, SET_SPEED, DOWNLOAD_ERROR, FINISH};
@@ -149,7 +149,7 @@ private:
 
 /**/
 
-class Downloader public QWidget, public QMPlay2Extensions
+class Downloader : public QWidget, public QMPlay2Extensions
 {
     Q_OBJECT
 
-- 
2.49.0

