From c44eb235ca348f0341607de1ded4ab4bd663a4e7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Apr 2025 13:08:02 +0800
Subject: [PATCH 52/60] SettingsWidget.cpp

---
 src/gui/SettingsWidget.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/gui/SettingsWidget.cpp b/src/gui/SettingsWidget.cpp
index cbc9990a..32809a17 100644
--- a/src/gui/SettingsWidget.cpp
+++ b/src/gui/SettingsWidget.cpp
@@ -55,6 +55,8 @@
 #include <MenuBar.hpp>
 #include <Module.hpp>
 
+#include <stdlib.h>
+
 #include "ui_SettingsGeneral.h"
 #include "ui_SettingsPlayback.h"
 #include "ui_SettingsPlaybackModulesList.h"
@@ -686,7 +688,7 @@ void SettingsWidget::applyProxy()
     Settings &QMPSettings = QMPlay2Core.getSettings();
     if (!QMPSettings.getBool("Proxy/Use"))
     {
-        qunsetenv("http_proxy");
+        unsetenv("http_proxy");
     }
     else
     {
-- 
2.49.0

