From 62a224f5ea7076947d099a54492745e7d4537caa Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Apr 2025 12:25:54 +0800
Subject: [PATCH 53/60] PlayClass

---
 src/gui/PlayClass.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/gui/PlayClass.cpp b/src/gui/PlayClass.cpp
index 1138db76..5b103071 100644
--- a/src/gui/PlayClass.cpp
+++ b/src/gui/PlayClass.cpp
@@ -311,14 +311,14 @@ void PlayClass::seek(double pos, bool allowAccurate)
 void PlayClass::chStream(const QString &s)
 {
     if (s.startsWith("audio"))
-        chosenAudioStream = s.rightRef(s.length() - 5).toInt();
+        chosenAudioStream = s.rightRef(s.length() - 5).toString().toInt();
     else if (s.startsWith("video"))
-        chosenVideoStream = s.rightRef(s.length() - 5).toInt();
+        chosenVideoStream = s.rightRef(s.length() - 5).toString().toInt();
     else if (s.startsWith("subtitles"))
-        chosenSubtitlesStream = s.rightRef(s.length() - 9).toInt();
+        chosenSubtitlesStream = s.rightRef(s.length() - 9).toString().toInt();
     else if (s.startsWith("fileSubs"))
     {
-        int idx = s.rightRef(s.length() - 8).toInt();
+        int idx = s.rightRef(s.length() - 5).toString().toInt();
         if (fileSubsList.count() > idx)
             loadSubsFile(fileSubsList[idx]);
     }
-- 
2.49.0

