From e9f2d033d62a618736dd88cd9aa2081cf1a79fda Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Apr 2025 14:11:33 +0800
Subject: [PATCH 55/60] PlaylistWidget

---
 src/gui/PlaylistWidget.cpp | 19 +++----------------
 1 file changed, 3 insertions(+), 16 deletions(-)

diff --git a/src/gui/PlaylistWidget.cpp b/src/gui/PlaylistWidget.cpp
index c529fa8c..3c3a38bc 100644
--- a/src/gui/PlaylistWidget.cpp
+++ b/src/gui/PlaylistWidget.cpp
@@ -57,20 +57,6 @@ static inline MenuBar::Playlist *playlistMenu()
     return QMPlay2GUI.menuBar->playlist;
 }
 
-/* PlaylistItem class */
-class PlaylistItem : public QTreeWidgetItem
-{
-public:
-    bool operator <(const QTreeWidgetItem &other) const override
-    {
-        if (treeWidget() && treeWidget()->sortColumn() == 2)
-        {
-            return (data(2, Qt::UserRole) < other.data(2, Qt::UserRole));
-        }
-        return QTreeWidgetItem::operator <(other);
-    }
-};
-
 /* UpdateEntryThr class */
 UpdateEntryThr::UpdateEntryThr(PlaylistWidget &pLW) :
     pendingUpdates(0),
@@ -915,7 +901,7 @@ void PlaylistWidget::setEntryFont(QTreeWidgetItem *tWI, const int flags)
 
 QTreeWidgetItem *PlaylistWidget::newGroup(const QString &name, const QString &url, QTreeWidgetItem *parent, int insertChildAt, QStringList *existingEntries)
 {
-    QTreeWidgetItem *tWI = new PlaylistItem;
+    QTreeWidgetItem *tWI = new QTreeWidgetItem;
 
     tWI->setFlags(tWI->flags() | Qt::ItemIsEditable);
     QMPlay2GUI.setTreeWidgetItemIcon(tWI, url.isEmpty() ? *QMPlay2GUI.groupIcon : *QMPlay2GUI.folderIcon, 0, this);
@@ -928,9 +914,10 @@ QTreeWidgetItem *PlaylistWidget::newGroup(const QString &name, const QString &ur
     QMetaObject::invokeMethod(this, "insertItem", Q_ARG(QTreeWidgetItem *, tWI), Q_ARG(QTreeWidgetItem *, parent), Q_ARG(int, insertChildAt));
     return tWI;
 }
+
 QTreeWidgetItem *PlaylistWidget::newEntry(const Playlist::Entry &entry, QTreeWidgetItem *parent, const Functions::DemuxersInfo &demuxersInfo, int insertChildAt, QStringList *existingEntries)
 {
-    QTreeWidgetItem *tWI = new PlaylistItem;
+    QTreeWidgetItem *tWI = new QTreeWidgetItem;
 
     QIcon icon;
     Functions::getDataIfHasPluginPrefix(entry.url, nullptr, nullptr, &icon, nullptr, demuxersInfo);
-- 
2.49.0

