From 1ad4b49c4ddd33f810836a1919e9f97f6af4ffc4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Apr 2025 14:12:58 +0800
Subject: [PATCH 56/60] AboutWidget

---
 src/gui/AboutWidget.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/gui/AboutWidget.cpp b/src/gui/AboutWidget.cpp
index dfc52ccd..0f867e67 100644
--- a/src/gui/AboutWidget.cpp
+++ b/src/gui/AboutWidget.cpp
@@ -52,7 +52,8 @@ AboutWidget::AboutWidget()
 
     QTabWidget *tabW = new QTabWidget;
 
-    const QFont font(QFontDatabase::systemFont(QFontDatabase::FixedFont));
+    QFont font("Monospace");
+    font.setStyleHint(QFont::TypeWriter);
 
     logE = new QPlainTextEdit;
     logE->setFont(font);
-- 
2.49.0

