From 6db4980638e7e5bcd7f4578e3247f7e9cec8cdd3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Apr 2025 14:50:57 +0800
Subject: [PATCH 60/60] CMakeLists

---
 src/gui/CMakeLists.txt | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/src/gui/CMakeLists.txt b/src/gui/CMakeLists.txt
index c36122a2..8bd630af 100644
--- a/src/gui/CMakeLists.txt
+++ b/src/gui/CMakeLists.txt
@@ -70,7 +70,6 @@ set(GUI_SRC
     ShortcutHandler.cpp
     KeyBindingsDialog.cpp
     Updater.cpp
-    EventFilterWorkarounds.cpp
 )
 
 set(GUI_FORMS # *.ui files
@@ -99,9 +98,11 @@ if(APPLE)
     list(APPEND GUI_HDR
         macOS/QMPlay2MacExtensions.hpp
     )
-    list(APPEND GUI_SRC
-        macOS/QMPlay2MacExtensions.mm
-    )
+    if(NOT USE_QT4)
+        list(APPEND GUI_SRC
+           macOS/QMPlay2MacExtensions.mm
+        )
+    endif()
 
     include_directories("macOS")
 endif()
-- 
2.49.0

