From 629296f380ea2890770e10f95d822611bfee68cb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 22 May 2025 08:35:58 +0800
Subject: [PATCH 64/69] YouTube.cpp: backport 77cf66c

---
 src/modules/Extensions/YouTube.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index d8ff9b94..27d1c37c 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -1111,7 +1111,7 @@ QJsonDocument YouTube::getYtInitialData(const QByteArray &data)
     if (idx2 < 0)
         return QJsonDocument();
 
-    auto jsonData = data.mid(idx, idx2 - idx);
+    auto jsonData = data.mid(idx, idx2 - idx).trimmed();
     if (jsonData.endsWith(';'))
         jsonData.chop(1);
 
-- 
2.48.0

