From 7dc42e7dcb90fbb93bf3d51c4d1773632a1286cb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 22 May 2025 08:36:57 +0800
Subject: [PATCH 65/69] YouTube.cpp: backport 0e03873

---
 src/modules/Extensions/YouTube.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index 27d1c37c..2a44f820 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -1067,7 +1067,7 @@ void YouTube::preparePlaylist(const QByteArray &data, QTreeWidgetItem *tWI)
     {
         const auto playlistRenderer = obj.toObject()["playlistVideoRenderer"].toObject();
 
-        const auto title = playlistRenderer["title"].toObject()["simpleText"].toString();
+        const auto title = playlistRenderer["title"].toObject()["runs"].toArray().at(0).toObject()["text"].toString();
         const auto videoId = playlistRenderer["videoId"].toString();
         if (title.isEmpty() || videoId.isEmpty())
             continue;
-- 
2.48.0

