From 294d1884ec4ec46bd31c75034018e30e453625a7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 22 May 2025 08:57:18 +0800
Subject: [PATCH 67/69] YouTube.cpp: backport 491894b plus 5aaaa4f

---
 src/modules/Extensions/YouTube.cpp | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index c3608d59..1f62617c 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -1113,13 +1113,14 @@ QJsonDocument YouTube::getYtInitialData(const QByteArray &data)
     if (idx < 0)
         return QJsonDocument();
 
-    int idx2 = data.indexOf("\n", idx);
-    if (idx2 < 0)
-        return QJsonDocument();
+    QJsonParseError e = {};
+    auto jsonDoc = QJsonDocument::fromJson(data.mid(idx), &e);
+
+    if (Q_UNLIKELY(e.error == QJsonParseError::NoError))
+        return jsonDoc;
 
-    auto jsonData = data.mid(idx, idx2 - idx).trimmed();
-    if (jsonData.endsWith(';'))
-        jsonData.chop(1);
+    if (e.error == QJsonParseError::GarbageAtEnd && e.offset > 0)
+        return QJsonDocument::fromJson(data.mid(idx, e.offset));
 
-    return QJsonDocument::fromJson(jsonData);
+    return QJsonDocument();
 }
-- 
2.48.0

