From 1f795286ec72363b15a68d215e1dc5c2a544da52 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 22 May 2025 08:18:23 +0800
Subject: [PATCH 69/69] YouTube.cpp: backport 2f9281b

---
 src/modules/Extensions/YouTube.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index 0d94ac51..97151af5 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -934,10 +934,10 @@ QStringList YouTube::getYouTubeVideo(const QString &param, const QString &url, I
         if (format.isEmpty())
             continue;
 
-        const auto container = format["container"].toString();
-        if (container.contains("dash", Qt::CaseInsensitive))
+        const auto protocol = format["protocol"].toString();
+        if (protocol.contains("dash", Qt::CaseInsensitive))
         {
-            // Skip MP4 DASH, because it doesn't work properly
+            // Skip MP4 DASH, because it doesn't work properly (old comment, is it still valid?)
             continue;
         }
 
-- 
2.48.0

