From 29c848199b778985ae3457cccb355dbecbb20b29 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 22 May 2025 10:12:50 +0800
Subject: [PATCH 73/73] ScreenSaver.cpp: fix IOKit defines

---
 src/gui/macOS/ScreenSaver.cpp | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/src/gui/macOS/ScreenSaver.cpp b/src/gui/macOS/ScreenSaver.cpp
index 0165d5cc..d71b25b7 100644
--- a/src/gui/macOS/ScreenSaver.cpp
+++ b/src/gui/macOS/ScreenSaver.cpp
@@ -20,6 +20,10 @@
 
 #include <IOKit/pwr_mgt/IOPMLib.h>
 
+#ifdef __APPLE__
+    #include <AvailabilityMacros.h>
+#endif
+
 #define QMPLAY2_MEDIA_PLAYBACK CFSTR("QMPlay2 media playback")
 
 class ScreenSaverPriv
@@ -32,8 +36,13 @@ public:
 
     inline void inhibit()
     {
-        m_okDisp = (IOPMAssertionCreateWithName(kIOPMAssertPreventUserIdleDisplaySleep, kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idDisp) == kIOReturnSuccess);
-        m_okSys  = (IOPMAssertionCreateWithName(kIOPMAssertPreventUserIdleSystemSleep,  kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idSys)  == kIOReturnSuccess);
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MAX_ALLOWED < 1060)
+        m_okDisp = (IOPMAssertionCreate(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, &m_idDisp) == kIOReturnSuccess);
+        m_okSys  = (IOPMAssertionCreate(kIOPMAssertionTypeNoIdleSleep, kIOPMAssertionLevelOn, &m_idSys)  == kIOReturnSuccess);
+#else
+        m_okDisp = (IOPMAssertionCreateWithName(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idDisp) == kIOReturnSuccess);
+        m_okSys  = (IOPMAssertionCreateWithName(kIOPMAssertionTypeNoIdleSleep, kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idSys) == kIOReturnSuccess);
+#endif
     }
     inline void unInhibit()
     {
-- 
2.48.0

