From 7b85391b1e2d697dc03b27937ddf2f71842ad343 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 03:53:44 +0800
Subject: [PATCH 74/74] Lyrics.cpp: fix syntax

---
 src/modules/Extensions/Lyrics.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/modules/Extensions/Lyrics.cpp b/src/modules/Extensions/Lyrics.cpp
index 87c1fc18..560d4ea1 100644
--- a/src/modules/Extensions/Lyrics.cpp
+++ b/src/modules/Extensions/Lyrics.cpp
@@ -56,8 +56,10 @@ Lyrics::Lyrics(Module &module) :
 {
     SetModule(module);
 
-    connect(&QMPlay2Core, &QMPlay2CoreClass::updatePlaying,
-            this, &Lyrics::updatePlaying);
+    connect(&QMPlay2Core,
+        SIGNAL(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &, const QString &)),
+        this,
+        SLOT(updatePlaying(bool, const QString &, const QString &, const QString &, int, bool, const QString &, const QString &)));
     connect(&m_net, SIGNAL(finished(NetworkReply *)), this, SLOT(finished(NetworkReply *)));
 
     m_dW = new DockWidget;
-- 
2.49.0

