From 2c1f19ce1f458754a947912d34ea23ebdff25d2a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 04:45:04 +0800
Subject: [PATCH 75/77] YouTube.cpp: backport a850bc5

---
 src/modules/Extensions/YouTube.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index a3c2f206..5c226093 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -983,6 +983,8 @@ QStringList YouTube::getYouTubeVideo(const QString &param, const QString &url, I
 
     const auto subtitles = o["subtitles"].toObject();
     QString lang = QMPlay2Core.getSettings().getString("SubtitlesLanguage");
+    if (lang.isEmpty()) // Default language
+        lang = QLocale::languageToString(QLocale::system().language());
     if (!audioOnly && m_allowSubtitles && !subtitles.isEmpty() && !lang.isEmpty())
     {
         // Try to convert full language name into short language code
-- 
2.49.0

