From b83ca744e123bcd4e441f376effc78fcfb2ecdc8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 04:51:15 +0800
Subject: [PATCH 76/77] YouTube.cpp: backport 6a2caa3 and b6c43f9

---
 src/modules/Extensions/YouTube.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/Extensions/YouTube.cpp b/src/modules/Extensions/YouTube.cpp
index 5c226093..c3ddb88b 100644
--- a/src/modules/Extensions/YouTube.cpp
+++ b/src/modules/Extensions/YouTube.cpp
@@ -936,8 +936,8 @@ QStringList YouTube::getYouTubeVideo(const QString &param, const QString &url, I
         const auto protocol = format["protocol"].toString();
         if (protocol.contains("dash", Qt::CaseInsensitive))
         {
-            // Skip MP4 DASH, because it doesn't work properly (old comment, is it still valid?)
-            continue;
+            if (format.contains("fragment_base_url"))
+                continue; // Skip DASH, because it doesn't work
         }
 
         const auto itag = format["format_id"].toString().toInt();
-- 
2.49.0

