From e3d3f569794fa70b242e5780707536fe0bce5c82 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 13:52:32 +0800
Subject: [PATCH 83/85] OpenGL2Common: define GL_*_VERSION when undefined

---
 src/modules/OpenGL2/OpenGL2Common.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/modules/OpenGL2/OpenGL2Common.cpp b/src/modules/OpenGL2/OpenGL2Common.cpp
index 8a43b64a..bbd48f7f 100644
--- a/src/modules/OpenGL2/OpenGL2Common.cpp
+++ b/src/modules/OpenGL2/OpenGL2Common.cpp
@@ -43,6 +43,13 @@
 
 #include <cmath>
 
+#ifndef GL_MAJOR_VERSION
+	#define GL_MAJOR_VERSION 0x821B
+#endif
+#ifndef GL_MINOR_VERSION
+	#define GL_MINOR_VERSION 0x821C
+#endif
+
 /* OpenGL|ES 2.0 doesn't have those definitions */
 #ifndef GL_MAP_WRITE_BIT
 	#define GL_MAP_WRITE_BIT 0x0002
-- 
2.48.0

