From ff533ad06307f2f43f6dd5ccc821ab4f4a45f343 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 14:09:40 +0800
Subject: [PATCH 85/85] OpenGL2: CMakeLists

---
 src/modules/OpenGL2/CMakeLists.txt | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/src/modules/OpenGL2/CMakeLists.txt b/src/modules/OpenGL2/CMakeLists.txt
index 6e23a397..43ab1d31 100644
--- a/src/modules/OpenGL2/CMakeLists.txt
+++ b/src/modules/OpenGL2/CMakeLists.txt
@@ -1,18 +1,4 @@
-cmake_minimum_required(VERSION 2.8.6)
-if(COMMAND cmake_policy)
-    if(POLICY CMP0003)
-        cmake_policy(SET CMP0003 NEW)
-    endif()
-    if(POLICY CMP0020)
-        cmake_policy(SET CMP0020 NEW)
-    endif()
-    if(POLICY CMP0042)
-        cmake_policy(SET CMP0042 NEW)
-    endif()
-    if(POLICY CMP0043)
-        cmake_policy(SET CMP0043 NEW)
-    endif()
-endif()
+cmake_minimum_required(VERSION 3.1)
 project(OpenGL2)
 
 set(OpenGL2_HDR
@@ -91,9 +77,8 @@ else() # Qt5 >= 5.6.0
     qt5_use_modules(${PROJECT_NAME} Gui Widgets)
 endif()
 
-add_dependencies(${PROJECT_NAME} libqmplay2)
 target_link_libraries(${PROJECT_NAME}
-    ${qmplay2lib}
+    libqmplay2
 )
 
 if(APPLE)
-- 
2.48.0

