From c25e954e69489ed8ee9ac29ff47e1b1a12a6be13 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 14:18:00 +0800
Subject: [PATCH 86/86] CMakeLists: fix IOKit spelling

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b7c6ed9..7166ec73 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -291,7 +291,7 @@ endif()
 
 if(APPLE)
     find_library(APPKIT_LIBRARY AppKit)
-    find_library(IOKIT_LIBRARY IoKit)
+    find_library(IOKIT_LIBRARY IOKit)
     set(SOLID_ACTIONS_DEFAULT "None")
     set(DEFAULT_INSTALL_RPATH ON)
 endif()
-- 
2.48.0

