--- myocamlbuild.ml	2024-09-10 21:46:38.000000000 +0800
+++ myocamlbuild.ml	2026-03-03 12:52:23.000000000 +0800
@@ -6,6 +6,11 @@
 
 let system_support_lib = match os with
 | "Linux\n" -> [A "-cclib"; A "-lrt"]
+| "Darwin\n" -> [A "-cclib"; A "-lMacportsLegacySupport"]
+| _ -> []
+
+let system_support_cflags = match os with
+| "Darwin\n" -> [A "-ccopt"; A "-isystem@PREFIX@/include/LegacySupport"]
 | _ -> []
 
 let lib s =
@@ -23,6 +28,7 @@
 
       (* mtime-clock *)
 
+      flag ["c"; "compile"] (S system_support_cflags);
       flag_and_dep ["link"; "ocaml"; "link_mtime_clock_stubs"]
         (P (lib "src/clock/libmtime_clock_stubs"));
 
