diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoaclipboard.m patch2-round8/after/src/video/cocoa/SDL_cocoaclipboard.m
--- a/src/video/cocoa/SDL_cocoaclipboard.m	2026-03-09 06:07:02.495556115 +0000
+++ b/src/video/cocoa/SDL_cocoaclipboard.m	2026-03-09 06:07:02.508556448 +0000
@@ -27,13 +27,14 @@
 
 int
 Cocoa_SetClipboardText(_THIS, const char *text)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *data = (SDL_VideoData *) _this->driverdata;
     NSPasteboard *pasteboard;
     NSString *format = NSPasteboardTypeString;
     NSString *nsstr = [NSString stringWithUTF8String:text];
     if (nsstr == nil) {
+        [pool drain];
         return SDL_SetError("Couldn't create NSString; is your string data in UTF-8 format?");
     }
 
@@ -41,13 +42,14 @@
     data->clipboard_count = [pasteboard declareTypes:[NSArray arrayWithObject:format] owner:nil];
     [pasteboard setString:nsstr forType:format];
 
+    [pool drain];
     return 0;
-}}
+}
 
 char *
 Cocoa_GetClipboardText(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSPasteboard *pasteboard;
     NSString *format = NSPasteboardTypeString;
     NSString *available;
@@ -70,8 +72,9 @@
         text = SDL_strdup("");
     }
 
+    [pool drain];
     return text;
-}}
+}
 
 SDL_bool
 Cocoa_HasClipboardText(_THIS)
@@ -87,8 +90,8 @@
 
 void
 Cocoa_CheckClipboardUpdate(struct SDL_VideoData * data)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSPasteboard *pasteboard;
     NSInteger count;
 
@@ -100,7 +103,8 @@
         }
         data->clipboard_count = count;
     }
-}}
+    [pool drain];
+}
 
 #endif /* SDL_VIDEO_DRIVER_COCOA */
 
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoaevents.m patch2-round8/after/src/video/cocoa/SDL_cocoaevents.m
--- a/src/video/cocoa/SDL_cocoaevents.m	2026-03-09 06:07:02.495556115 +0000
+++ b/src/video/cocoa/SDL_cocoaevents.m	2026-03-09 06:07:02.506556397 +0000
@@ -32,6 +32,9 @@
 #ifndef kIOPMAssertPreventUserIdleDisplaySleep
 #define kIOPMAssertPreventUserIdleDisplaySleep kIOPMAssertionTypePreventUserIdleDisplaySleep
 #endif
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
 #ifndef NSAppKitVersionNumber10_8
 #define NSAppKitVersionNumber10_8 1187
 #endif
@@ -232,6 +235,10 @@
 
 - (void)focusSomeWindow:(NSNotification *)aNotification
 {
+    SDL_VideoDevice *device;
+    SDL_Window *window;
+    int i;
+
     /* HACK: Ignore the first call. The application gets a
      * applicationDidBecomeActive: a little bit after the first window is
      * created, and if we don't ignore it, a window that has been created with
@@ -249,10 +256,9 @@
         return;
     }
 
-    SDL_VideoDevice *device = SDL_GetVideoDevice();
+    device = SDL_GetVideoDevice();
     if (device && device->windows) {
-        SDL_Window *window = device->windows;
-        int i;
+        window = device->windows;
         for (i = 0; i < device->num_displays; ++i) {
             SDL_Window *fullscreen_window = device->displays[i].fullscreen_window;
             if (fullscreen_window) {
@@ -458,8 +464,8 @@
 
 void
 Cocoa_RegisterApp(void)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     /* This can get called more than once! Be careful what you initialize! */
 
     if (NSApp == nil) {
@@ -478,7 +484,7 @@
          */
         if ([NSApp mainMenu] == nil) {
             bool nibLoaded;
-            
+
             nibLoaded = LoadMainMenuNibIfAvailable();
             if (!nibLoaded) {
                 CreateApplicationMenus();
@@ -513,7 +519,8 @@
             appDelegate->seenFirstActivate = YES;
         }
     }
-}}
+    [pool drain];
+}
 
 int
 Cocoa_PumpEventsUntilDate(_THIS, NSDate *expiration, bool accumulate)
@@ -552,30 +559,34 @@
 
 int
 Cocoa_WaitEventTimeout(_THIS, int timeout)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     if (timeout > 0) {
         NSDate *limitDate = [NSDate dateWithTimeIntervalSinceNow: (double) timeout / 1000.0];
+        [pool drain];
         return Cocoa_PumpEventsUntilDate(_this, limitDate, false);
     } else if (timeout == 0) {
+        [pool drain];
         return Cocoa_PumpEventsUntilDate(_this, [NSDate distantPast], false);
     } else {
         while (Cocoa_PumpEventsUntilDate(_this, [NSDate distantFuture], false) == 0) {
         }
     }
+    [pool drain];
     return 1;
-}}
+}
 
 void
 Cocoa_PumpEvents(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     Cocoa_PumpEventsUntilDate(_this, [NSDate distantPast], true);
-}}
+    [pool drain];
+}
 
 void Cocoa_SendWakeupEvent(_THIS, SDL_Window *window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
 
     NSEvent* event = [NSEvent otherEventWithType: NSEventTypeApplicationDefined
@@ -589,15 +600,17 @@
                                        data2: 0];
 
     [NSApp postEvent: event atStart: YES];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SuspendScreenSaver(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *data = (SDL_VideoData *)_this->driverdata;
 
     if (!data->screensaver_use_iopm) {
+        [pool drain];
         return;
     }
 
@@ -607,6 +620,7 @@
     }
 
     if (_this->suspend_screensaver) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
         /* FIXME: this should ideally describe the real reason why the game
          * called SDL_DisableScreenSaver. Note that the name is only meant to be
          * seen by OS X power users. there's an additional optional human-readable
@@ -617,8 +631,12 @@
                                            (CFStringRef) name,
                                            NULL, NULL, NULL, 0, NULL,
                                            &data->screensaver_assertion);
+#else
+        IOPMAssertionCreate(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, &data->screensaver_assertion);
+#endif
     }
-}}
+    [pool drain];
+}
 
 #endif /* SDL_VIDEO_DRIVER_COCOA */
 
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoakeyboard.m patch2-round8/after/src/video/cocoa/SDL_cocoakeyboard.m
--- a/src/video/cocoa/SDL_cocoakeyboard.m	2026-03-09 06:07:02.495556115 +0000
+++ b/src/video/cocoa/SDL_cocoakeyboard.m	2026-03-09 06:07:02.506556397 +0000
@@ -33,6 +33,13 @@
 /*#define DEBUG_IME NSLog */
 #define DEBUG_IME(...)
 
+/* Forward declare 10.7+ methods for compilation on older SDKs */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+@interface NSWindow (SDL_ForwardDeclare)
+- (NSRect)convertRectToScreen:(NSRect)aRect;
+@end
+#endif
+
 @interface SDLTranslatorResponder : NSView <NSTextInputClient> {
     NSString *_markedText;
     NSRange   _markedRange;
@@ -399,6 +406,7 @@
     int i;
     SDL_Scancode scancode;
     SDL_Keycode keymap[SDL_NUM_SCANCODES];
+    CFDataRef uchrDataRef;
 
     /* See if the keymap needs to be updated */
     key_layout = TISCopyCurrentKeyboardLayoutInputSource();
@@ -410,7 +418,7 @@
     SDL_GetDefaultKeymap(keymap);
 
     /* Try Unicode data first */
-    CFDataRef uchrDataRef = TISGetInputSourceProperty(key_layout, kTISPropertyUnicodeKeyLayoutData);
+    uchrDataRef = TISGetInputSourceProperty(key_layout, kTISPropertyUnicodeKeyLayoutData);
     if (uchrDataRef) {
         chr_data = CFDataGetBytePtr(uchrDataRef);
     } else {
@@ -447,7 +455,7 @@
                 keymap[scancode] = s[0];
             }
         }
-        SDL_SetKeymap(0, keymap, SDL_NUM_SCANCODES);
+        SDL_SetKeymap(0, keymap, SDL_NUM_SCANCODES, SDL_FALSE);
         if (send_event) {
             SDL_SendKeymapChangedEvent();
         }
@@ -479,16 +487,18 @@
 
 void
 Cocoa_StartTextInput(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *data = (SDL_VideoData *) _this->driverdata;
     SDL_Window *window = SDL_GetKeyboardFocus();
     NSWindow *nswindow = nil;
+    NSView *parentView;
+
     if (window) {
         nswindow = ((SDL_WindowData*)window->driverdata)->nswindow;
     }
 
-    NSView *parentView = [nswindow contentView];
+    parentView = [nswindow contentView];
 
     /* We only keep one field editor per process, since only the front most
      * window can receive text input events, so it make no sense to keep more
@@ -506,12 +516,13 @@
         [parentView addSubview: data->fieldEdit];
         [nswindow makeFirstResponder: data->fieldEdit];
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_StopTextInput(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *data = (SDL_VideoData *) _this->driverdata;
 
     if (data && data->fieldEdit) {
@@ -519,7 +530,8 @@
         [data->fieldEdit release];
         data->fieldEdit = nil;
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetTextInputRect(_THIS, SDL_Rect *rect)
@@ -537,17 +549,20 @@
 void
 Cocoa_HandleKeyEvent(_THIS, NSEvent *event)
 {
-    SDL_VideoData *data = _this ? ((SDL_VideoData *) _this->driverdata) : NULL;
-    if (!data) {
-        return;  /* can happen when returning from fullscreen Space on shutdown */
-    }
-
-    unsigned short scancode = [event keyCode];
+    SDL_VideoData *data;
+    unsigned short scancode;
     SDL_Scancode code;
 #if 0
     const char *text;
 #endif
 
+    data = _this ? ((SDL_VideoData *) _this->driverdata) : NULL;
+    if (!data) {
+        return;  /* can happen when returning from fullscreen Space on shutdown */
+    }
+
+    scancode = [event keyCode];
+
     if ((scancode == 10 || scancode == 50) && KBGetLayoutType(LMGetKbdType()) == kKeyboardISO) {
         /* see comments in SDL_cocoakeys.h */
         scancode = 60 - scancode;
@@ -614,7 +629,7 @@
 void
 Cocoa_SetWindowKeyboardGrab(_THIS, SDL_Window * window, SDL_bool grabbed)
 {
-#if SDL_MAC_NO_SANDBOX
+#ifdef SDL_MAC_NO_SANDBOX
     CGSSetGlobalHotKeyOperatingMode(_CGSDefaultConnection(), grabbed ? CGSGlobalHotKeyDisable : CGSGlobalHotKeyEnable);
 #endif
 }
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoamessagebox.m patch2-round8/after/src/video/cocoa/SDL_cocoamessagebox.m
--- a/src/video/cocoa/SDL_cocoamessagebox.m	2026-03-09 06:07:02.495556115 +0000
+++ b/src/video/cocoa/SDL_cocoamessagebox.m	2026-03-09 06:07:02.506556397 +0000
@@ -57,7 +57,7 @@
 - (void)showAlert:(NSAlert*)alert
 {
     if (nswindow) {
-#ifdef MAC_OS_X_VERSION_10_9
+#if defined(MAC_OS_X_VERSION_10_9) && defined(__clang__)
         if ([alert respondsToSelector:@selector(beginSheetModalForWindow:completionHandler:)]) {
             [alert beginSheetModalForWindow:nswindow completionHandler:^(NSModalResponse returnCode) {
                 clicked = returnCode;
@@ -65,7 +65,7 @@
         } else
 #endif
         {
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1090
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1090 || !defined(__clang__)
             [alert beginSheetModalForWindow:nswindow modalDelegate:self didEndSelector:@selector(alertDidEnd:returnCode:contextInfo:) contextInfo:nil];
 #endif
         }
@@ -88,13 +88,52 @@
 
 @end
 
+typedef struct {
+    const SDL_MessageBoxData *messageboxdata;
+    int *buttonid;
+    int *returnValue;
+} Cocoa_MessageBoxCallbackData;
+
+/* Forward declaration */
+static void Cocoa_ShowMessageBoxImpl(const SDL_MessageBoxData *messageboxdata, int *buttonid, int *returnValue);
+
+@interface SDLMessageBoxCaller : NSObject {
+    Cocoa_MessageBoxCallbackData *_data;
+}
+- (void)setData:(Cocoa_MessageBoxCallbackData *)data;
+- (Cocoa_MessageBoxCallbackData *)data;
+- (void)showMessageBox;
+@end
+
+@implementation SDLMessageBoxCaller
+- (void)setData:(Cocoa_MessageBoxCallbackData *)data
+{
+    _data = data;
+}
+
+- (Cocoa_MessageBoxCallbackData *)data
+{
+    return _data;
+}
+
+- (void)showMessageBox
+{
+    Cocoa_ShowMessageBoxImpl(self.data->messageboxdata, self.data->buttonid, self.data->returnValue);
+}
+@end
 
 static void
 Cocoa_ShowMessageBoxImpl(const SDL_MessageBoxData *messageboxdata, int *buttonid, int *returnValue)
 {
+    NSAlert* alert;
+    const SDL_MessageBoxButtonData *buttons;
+    int i;
+    SDLMessageBoxPresenter* presenter;
+    NSInteger clicked;
+
     Cocoa_RegisterApp();
 
-    NSAlert* alert = [[[NSAlert alloc] init] autorelease];
+    alert = [[[NSAlert alloc] init] autorelease];
 
     if (messageboxdata->flags & SDL_MESSAGEBOX_ERROR) {
         [alert setAlertStyle:NSAlertStyleCritical];
@@ -107,8 +146,7 @@
     [alert setMessageText:[NSString stringWithUTF8String:messageboxdata->title]];
     [alert setInformativeText:[NSString stringWithUTF8String:messageboxdata->message]];
 
-    const SDL_MessageBoxButtonData *buttons = messageboxdata->buttons;
-    int i;
+    buttons = messageboxdata->buttons;
     for (i = 0; i < messageboxdata->numbuttons; ++i) {
         const SDL_MessageBoxButtonData *sdlButton;
         NSButton *button;
@@ -129,11 +167,11 @@
         }
     }
 
-    SDLMessageBoxPresenter* presenter = [[[SDLMessageBoxPresenter alloc] initWithParentWindow:messageboxdata->window] autorelease];
+    presenter = [[[SDLMessageBoxPresenter alloc] initWithParentWindow:messageboxdata->window] autorelease];
 
     [presenter showAlert:alert];
 
-    NSInteger clicked = presenter->clicked;
+    clicked = presenter->clicked;
     if (clicked >= NSAlertFirstButtonReturn) {
         clicked -= NSAlertFirstButtonReturn;
         if (messageboxdata->flags & SDL_MESSAGEBOX_BUTTONS_RIGHT_TO_LEFT) {
@@ -149,17 +187,27 @@
 /* Display a Cocoa message box */
 int
 Cocoa_ShowMessageBox(const SDL_MessageBoxData *messageboxdata, int *buttonid)
-{ @autoreleasepool
 {
-    __block int returnValue = 0;
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    int returnValue = 0;
+    Cocoa_MessageBoxCallbackData data;
+    SDLMessageBoxCaller *caller;
 
     if ([NSThread isMainThread]) {
         Cocoa_ShowMessageBoxImpl(messageboxdata, buttonid, &returnValue);
     } else {
-        dispatch_sync(dispatch_get_main_queue(), ^{ Cocoa_ShowMessageBoxImpl(messageboxdata, buttonid, &returnValue); });
+        data.messageboxdata = messageboxdata;
+        data.buttonid = buttonid;
+        data.returnValue = &returnValue;
+
+        caller = [[SDLMessageBoxCaller alloc] init];
+        [caller setData:&data];
+        [caller performSelectorOnMainThread:@selector(showMessageBox) withObject:nil waitUntilDone:YES];
+        [caller release];
     }
+    [pool drain];
     return returnValue;
-}}
+}
 
 #endif /* SDL_VIDEO_DRIVER_COCOA */
 
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoametalview.m patch2-round8/after/src/video/cocoa/SDL_cocoametalview.m
--- a/src/video/cocoa/SDL_cocoametalview.m	2026-03-09 06:07:02.495556115 +0000
+++ b/src/video/cocoa/SDL_cocoametalview.m	2026-03-09 06:07:02.506556397 +0000
@@ -45,12 +45,12 @@
      * resizable window exits a fullscreen Space via the user pressing the OS
      * exit-space button). */
     if (event->type == SDL_WINDOWEVENT && event->window.event == SDL_WINDOWEVENT_SIZE_CHANGED) {
-        @autoreleasepool {
-            SDL_cocoametalview *view = (__bridge SDL_cocoametalview *)userdata;
-            if (view.sdlWindowID == event->window.windowID) {
-                [view updateDrawableSize];
-            }
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+        SDL_cocoametalview *view = (__bridge SDL_cocoametalview *)userdata;
+        if (view.sdlWindowID == event->window.windowID) {
+            [view updateDrawableSize];
         }
+        [pool drain];
     }
     return 0;
 }
@@ -133,7 +133,8 @@
 
 SDL_MetalView
 Cocoa_Metal_CreateView(_THIS, SDL_Window * window)
-{ @autoreleasepool {
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData* data = (__bridge SDL_WindowData *)window->driverdata;
     NSView *view = data->nswindow.contentView;
     BOOL highDPI = (window->flags & SDL_WINDOW_ALLOW_HIGHDPI) != 0;
@@ -145,6 +146,7 @@
                                                 highDPI:highDPI
                                                 windowID:windowID];
     if (newview == nil) {
+        [pool drain];
         return NULL;
     }
 
@@ -153,26 +155,33 @@
     metalview = (SDL_MetalView)CFBridgingRetain(newview);
     [newview release];
 
+    [pool drain];
     return metalview;
-}}
+}
 
 void
 Cocoa_Metal_DestroyView(_THIS, SDL_MetalView view)
-{ @autoreleasepool {
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_cocoametalview *metalview = CFBridgingRelease(view);
     [metalview removeFromSuperview];
-}}
+    [pool drain];
+}
 
 void *
 Cocoa_Metal_GetLayer(_THIS, SDL_MetalView view)
-{ @autoreleasepool {
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_cocoametalview *cocoaview = (__bridge SDL_cocoametalview *)view;
-    return (__bridge void *)cocoaview.layer;
-}}
+    void *layer = (__bridge void *)cocoaview.layer;
+    [pool drain];
+    return layer;
+}
 
 void
 Cocoa_Metal_GetDrawableSize(_THIS, SDL_Window * window, int * w, int * h)
-{ @autoreleasepool {
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *data = (__bridge SDL_WindowData *)window->driverdata;
     NSView *contentView = data->sdlContentView;
     SDL_cocoametalview* metalview = [contentView viewWithTag:SDL_METALVIEW_TAG];
@@ -202,7 +211,8 @@
             *h = viewport.size.height;
         }
     }
-}}
+    [pool drain];
+}
 
 #endif /* SDL_VIDEO_DRIVER_COCOA && (SDL_VIDEO_VULKAN || SDL_VIDEO_METAL) */
 
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoamodes.m patch2-round8/after/src/video/cocoa/SDL_cocoamodes.m
--- a/src/video/cocoa/SDL_cocoamodes.m	2026-03-09 06:07:02.497556166 +0000
+++ b/src/video/cocoa/SDL_cocoamodes.m	2026-03-09 06:07:02.508556448 +0000
@@ -40,10 +40,20 @@
 #ifndef MAC_OS_X_VERSION_10_13
 #define NSAppKitVersionNumber10_12 1504
 #endif
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
 #if (IOGRAPHICSTYPES_REV < 40)
 #define kDisplayModeNativeFlag 0x02000000
 #endif
 
+/* Forward declare 10.7+ methods for compilation on older SDKs */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+@interface NSScreen (SDL_ForwardDeclare)
+- (CGFloat)backingScaleFactor;
+@end
+#endif
+
 
 static void
 Cocoa_ToggleMenuBar(const BOOL show)
@@ -177,6 +187,9 @@
     bool interlaced = (ioflags & kDisplayModeInterlacedFlag) != 0;
     CFMutableArrayRef modes;
 
+    (void)usableForGUI;  /* unused, kept for potential future use */
+    (void)interlaced;     /* unused, kept for potential future use */
+
     if (format == SDL_PIXELFORMAT_UNKNOWN) {
         return SDL_FALSE;
     }
@@ -318,8 +331,8 @@
 
 void
 Cocoa_InitModes(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     CGDisplayErr result;
     CGDirectDisplayID *displays;
     CGDisplayCount numDisplays;
@@ -329,6 +342,7 @@
     result = CGGetOnlineDisplayList(0, NULL, &numDisplays);
     if (result != kCGErrorSuccess) {
         CG_SetError("CGGetOnlineDisplayList()", result);
+        [pool drain];
         return;
     }
     displays = SDL_small_alloc(CGDirectDisplayID, numDisplays, &isstack);
@@ -336,6 +350,7 @@
     if (result != kCGErrorSuccess) {
         CG_SetError("CGGetOnlineDisplayList()", result);
         SDL_small_free(displays, isstack);
+        [pool drain];
         return;
     }
 
@@ -399,7 +414,8 @@
         }
     }
     SDL_small_free(displays, isstack);
-}}
+    [pool drain];
+}
 
 int
 Cocoa_GetDisplayBounds(_THIS, SDL_VideoDisplay * display, SDL_Rect * rect)
@@ -422,9 +438,10 @@
     const CGDirectDisplayID cgdisplay = displaydata->display;
     NSArray *screens = [NSScreen screens];
     NSScreen *screen = nil;
+    NSScreen *i;
 
     /* !!! FIXME: maybe track the NSScreen in SDL_DisplayData? */
-    for (NSScreen *i in screens) {
+    for (i in screens) {
         const CGDirectDisplayID thisDisplay = (CGDirectDisplayID) [[[i deviceDescription] objectForKey:@"NSScreenNumber"] unsignedIntValue];
         if (thisDisplay == cgdisplay) {
             screen = i;
@@ -448,20 +465,23 @@
 
 int
 Cocoa_GetDisplayDPI(_THIS, SDL_VideoDisplay * display, float * ddpi, float * hdpi, float * vdpi)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     const float MM_IN_INCH = 25.4f;
-
     SDL_DisplayData *data = (SDL_DisplayData *) display->driverdata;
-
-    /* we need the backingScaleFactor for Retina displays, which is only exposed through NSScreen, not CGDisplay, afaik, so find our screen... */
     CGFloat scaleFactor = 1.0f;
     NSArray *screens = [NSScreen screens];
     NSSize displayNativeSize;
+    NSScreen *screen;
+    CGSize displaySize;
+    int pixelWidth;
+    int pixelHeight;
+
+    /* we need the backingScaleFactor for Retina displays, which is only exposed through NSScreen, not CGDisplay, afaik, so find our screen... */
     displayNativeSize.width = (int) CGDisplayPixelsWide(data->display);
     displayNativeSize.height = (int) CGDisplayPixelsHigh(data->display);
-    
-    for (NSScreen *screen in screens) {
+
+    for (screen in screens) {
         const CGDirectDisplayID dpyid = (const CGDirectDisplayID ) [[[screen deviceDescription] objectForKey:@"NSScreenNumber"] unsignedIntValue];
         if (dpyid == data->display) {
 #ifdef MAC_OS_X_VERSION_10_8
@@ -507,9 +527,9 @@
         }
     }
 
-    const CGSize displaySize = CGDisplayScreenSize(data->display);
-    const int pixelWidth =  displayNativeSize.width;
-    const int pixelHeight = displayNativeSize.height;
+    displaySize = CGDisplayScreenSize(data->display);
+    pixelWidth =  displayNativeSize.width;
+    pixelHeight = displayNativeSize.height;
 
     if (ddpi) {
         *ddpi = (SDL_ComputeDiagonalDPI(pixelWidth, pixelHeight, displaySize.width / MM_IN_INCH, displaySize.height / MM_IN_INCH));
@@ -521,8 +541,9 @@
         *vdpi = (pixelHeight * MM_IN_INCH / displaySize.height);
     }
 
+    [pool drain];
     return 0;
-}}
+}
 
 void
 Cocoa_GetDisplayModes(_THIS, SDL_VideoDisplay * display)
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoamouse.m patch2-round8/after/src/video/cocoa/SDL_cocoamouse.m
--- a/src/video/cocoa/SDL_cocoamouse.m	2026-03-09 06:07:02.497556166 +0000
+++ b/src/video/cocoa/SDL_cocoamouse.m	2026-03-09 06:07:02.505556371 +0000
@@ -65,8 +65,8 @@
 
 static SDL_Cursor *
 Cocoa_CreateDefaultCursor()
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSCursor *nscursor;
     SDL_Cursor *cursor = NULL;
 
@@ -80,13 +80,14 @@
         }
     }
 
+    [pool drain];
     return cursor;
-}}
+}
 
 static SDL_Cursor *
 Cocoa_CreateCursor(SDL_Surface * surface, int hot_x, int hot_y)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSImage *nsimage;
     NSCursor *nscursor = NULL;
     SDL_Cursor *cursor = NULL;
@@ -105,13 +106,14 @@
         }
     }
 
+    [pool drain];
     return cursor;
-}}
+}
 
 static SDL_Cursor *
 Cocoa_CreateSystemCursor(SDL_SystemCursor id)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSCursor *nscursor = NULL;
     SDL_Cursor *cursor = NULL;
 
@@ -152,6 +154,7 @@
         break;
     default:
         SDL_assert(!"Unknown system cursor");
+        [pool drain];
         return NULL;
     }
 
@@ -164,23 +167,25 @@
         }
     }
 
+    [pool drain];
     return cursor;
-}}
+}
 
 static void
 Cocoa_FreeCursor(SDL_Cursor * cursor)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSCursor *nscursor = (NSCursor *)cursor->driverdata;
 
     [nscursor release];
     SDL_free(cursor);
-}}
+    [pool drain];
+}
 
 static int
 Cocoa_ShowCursor(SDL_Cursor * cursor)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoDevice *device = SDL_GetVideoDevice();
     SDL_Window *window = (device ? device->windows : NULL);
     for (; window != NULL; window = window->next) {
@@ -191,8 +196,9 @@
                                                 waitUntilDone:NO];
         }
     }
+    [pool drain];
     return 0;
-}}
+}
 
 static SDL_Window *
 SDL_FindWindowAtPoint(const int x, const int y)

--- a/src/video/cocoa/SDL_cocoaopengl.m	2026-03-09 14:43:47.067108237 +0000
+++ b/src/video/cocoa/SDL_cocoaopengl.m	2026-03-09 14:44:00.548457591 +0000
@@ -36,6 +36,30 @@
 
 #define DEFAULT_OPENGL  "/System/Library/Frameworks/OpenGL.framework/Libraries/libGL.dylib"
 
+/* Define constants not available in 10.6 SDK */
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
+
+#ifndef NSOpenGLPFAOpenGLProfile
+#define NSOpenGLPFAOpenGLProfile 99
+#endif
+
+#ifndef NSOpenGLProfileVersionLegacy
+#define NSOpenGLProfileVersionLegacy 0x1000
+#endif
+
+#ifndef NSOpenGLProfileVersion3_2Core
+#define NSOpenGLProfileVersion3_2Core 0x3200
+#endif
+
+/* Forward declare 10.7+ methods for compilation on older SDKs */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+@interface NSView (SDL_ForwardDeclare)
+- (NSRect)convertRectToBacking:(NSRect)aRect;
+@end
+#endif
+
 /* We still support OpenGL as long as Apple offers it, deprecated or not, so disable deprecation warnings about it. */
 #ifdef __clang__
 #pragma clang diagnostic push
@@ -107,10 +131,14 @@
             if ([NSThread isMainThread]) {
                 [self setView:contentview];
             } else {
-                dispatch_sync(dispatch_get_main_queue(), ^{ [self setView:contentview]; });
+                [self performSelectorOnMainThread:@selector(setView:) withObject:contentview waitUntilDone:NO];
             }
             if (self == [NSOpenGLContext currentContext]) {
-                [self explicitUpdate];
+                if ([NSThread isMainThread]) {
+                    [self update];
+                } else {
+                    [self scheduleUpdate];
+                }
             } else {
                 [self scheduleUpdate];
             }
@@ -118,7 +146,11 @@
     } else {
         [self clearDrawable];
         if (self == [NSOpenGLContext currentContext]) {
-            [self explicitUpdate];
+            if ([NSThread isMainThread]) {
+                [self update];
+            } else {
+                [self scheduleUpdate];
+            }
         } else {
             [self scheduleUpdate];
         }
@@ -130,12 +162,27 @@
     return self->window;
 }
 
+- (void)doSuperUpdate
+{
+    [super update];
+}
+
+- (void)setViewAndUpdate:(NSView *)view
+{
+    [self setView:view];
+    if (self == [NSOpenGLContext currentContext]) {
+        [super update];
+    } else {
+        [self scheduleUpdate];
+    }
+}
+
 - (void)explicitUpdate
 {
     if ([NSThread isMainThread]) {
         [super update];
     } else {
-        dispatch_async(dispatch_get_main_queue(), ^{ [super update]; });
+        [self performSelectorOnMainThread:@selector(doSuperUpdate) withObject:nil waitUntilDone:NO];
     }
 }
 
@@ -176,8 +223,8 @@
 
 SDL_GLContext
 Cocoa_GL_CreateContext(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoDisplay *display = SDL_GetDisplayForWindow(window);
     SDL_DisplayData *displaydata = (SDL_DisplayData *)display->driverdata;
     SDL_bool lion_or_later = floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6;
@@ -203,18 +250,22 @@
         _this->GL_GetSwapInterval = Cocoa_GLES_GetSwapInterval;
         _this->GL_SwapWindow = Cocoa_GLES_SwapWindow;
         _this->GL_DeleteContext = Cocoa_GLES_DeleteContext;
-        
+
         if (Cocoa_GLES_LoadLibrary(_this, NULL) != 0) {
+            [pool drain];
             return NULL;
         }
+        [pool drain];
         return Cocoa_GLES_CreateContext(_this, window);
 #else
         SDL_SetError("SDL not configured with EGL support");
+        [pool drain];
         return NULL;
 #endif
     }
     if ((_this->gl_config.profile_mask == SDL_GL_CONTEXT_PROFILE_CORE) && !lion_or_later) {
         SDL_SetError ("OpenGL Core Profile is not supported on this platform version");
+        [pool drain];
         return NULL;
     }
 
@@ -284,6 +335,7 @@
     fmt = [[NSOpenGLPixelFormat alloc] initWithAttributes:attr];
     if (fmt == nil) {
         SDL_SetError("Failed creating OpenGL pixel format");
+        [pool drain];
         return NULL;
     }
 
@@ -297,12 +349,14 @@
 
     if (context == nil) {
         SDL_SetError("Failed creating OpenGL context");
+        [pool drain];
         return NULL;
     }
 
     if ( Cocoa_GL_MakeCurrent(_this, window, context) < 0 ) {
         Cocoa_GL_DeleteContext(_this, context);
         SDL_SetError("Failed making OpenGL context current");
+        [pool drain];
         return NULL;
     }
 
@@ -317,6 +371,7 @@
         if (!glGetStringFunc) {
             Cocoa_GL_DeleteContext(_this, context);
             SDL_SetError ("Failed getting OpenGL glGetString entry point");
+            [pool drain];
             return NULL;
         }
 
@@ -324,12 +379,14 @@
         if (glversion == NULL) {
             Cocoa_GL_DeleteContext(_this, context);
             SDL_SetError ("Failed getting OpenGL context version");
+            [pool drain];
             return NULL;
         }
 
         if (SDL_sscanf(glversion, "%d.%d", &glversion_major, &glversion_minor) != 2) {
             Cocoa_GL_DeleteContext(_this, context);
             SDL_SetError ("Failed parsing OpenGL context version");
+            [pool drain];
             return NULL;
         }
 
@@ -337,6 +394,7 @@
            ((glversion_major == _this->gl_config.major_version) && (glversion_minor < _this->gl_config.minor_version))) {
             Cocoa_GL_DeleteContext(_this, context);
             SDL_SetError ("Failed creating OpenGL context at version requested");
+            [pool drain];
             return NULL;
         }
 
@@ -346,13 +404,14 @@
         /*_this->gl_config.major_version = glversion_major;*/
         /*_this->gl_config.minor_version = glversion_minor;*/
     }
+    [pool drain];
     return context;
-}}
+}
 
 int
 Cocoa_GL_MakeCurrent(_THIS, SDL_Window * window, SDL_GLContext context)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     if (context) {
         SDLOpenGLContext *nscontext = (SDLOpenGLContext *)context;
         if ([nscontext window] != window) {
@@ -364,8 +423,9 @@
         [NSOpenGLContext clearCurrentContext];
     }
 
+    [pool drain];
     return 0;
-}}
+}
 
 void
 Cocoa_GL_GetDrawableSize(_THIS, SDL_Window * window, int * w, int * h)
@@ -393,13 +453,14 @@
 
 int
 Cocoa_GL_SetSwapInterval(_THIS, int interval)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSOpenGLContext *nscontext;
     GLint value;
     int status;
 
     if (interval < 0) {  /* no extension for this on Mac OS X at the moment. */
+        [pool drain];
         return SDL_SetError("Late swap tearing currently unsupported");
     }
 
@@ -412,13 +473,14 @@
         status = SDL_SetError("No current OpenGL context");
     }
 
+    [pool drain];
     return status;
-}}
+}
 
 int
 Cocoa_GL_GetSwapInterval(_THIS)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSOpenGLContext *nscontext;
     GLint value;
     int status = 0;
@@ -429,13 +491,14 @@
         status = (int)value;
     }
 
+    [pool drain];
     return status;
-}}
+}
 
 int
 Cocoa_GL_SwapWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDLOpenGLContext* nscontext = (SDLOpenGLContext*)SDL_GL_GetCurrentContext();
     SDL_VideoData *videodata = (SDL_VideoData *) _this->driverdata;
 
@@ -445,18 +508,20 @@
     [nscontext flushBuffer];
     [nscontext updateIfNeeded];
     SDL_UnlockMutex(videodata->swaplock);
+    [pool drain];
     return 0;
-}}
+}
 
 void
 Cocoa_GL_DeleteContext(_THIS, SDL_GLContext context)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDLOpenGLContext *nscontext = (SDLOpenGLContext *)context;
 
     [nscontext setWindow:NULL];
     [nscontext release];
-}}
+    [pool drain];
+}
 
 /* We still support OpenGL as long as Apple offers it, deprecated or not, so disable deprecation warnings about it. */
 #ifdef __clang__

--- a/src/video/cocoa/SDL_cocoashape.m	2026-03-09 06:07:02.497556166 +0000
+++ b/src/video/cocoa/SDL_cocoashape.m	2026-03-09 06:07:02.508556448 +0000
@@ -71,8 +71,8 @@
 
 int
 Cocoa_SetWindowShape(SDL_WindowShaper *shaper, SDL_Surface *shape, SDL_WindowShapeMode *shape_mode)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_ShapeData* data = (SDL_ShapeData*)shaper->driverdata;
     SDL_WindowData* windata = (SDL_WindowData*)shaper->window->driverdata;
     SDL_CocoaClosure closure;
@@ -95,8 +95,9 @@
     SDL_TraverseShapeTree(data->shape,&ConvertRects,&closure);
     [closure.path addClip];
 
+    [pool drain];
     return 0;
-}}
+}
 
 int
 Cocoa_ResizeWindowShape(SDL_Window *window)
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoavideo.h patch2-round8/after/src/video/cocoa/SDL_cocoavideo.h
--- a/src/video/cocoa/SDL_cocoavideo.h	2026-03-09 06:07:02.499556217 +0000
+++ b/src/video/cocoa/SDL_cocoavideo.h	2026-03-09 06:07:02.505556371 +0000
@@ -41,24 +41,23 @@
 #include "SDL_cocoawindow.h"
 
 #ifndef MAC_OS_X_VERSION_10_12
-#define DECLARE_EVENT(name) static const NSEventType NSEventType##name = NS##name
-DECLARE_EVENT(LeftMouseDown);
-DECLARE_EVENT(LeftMouseUp);
-DECLARE_EVENT(RightMouseDown);
-DECLARE_EVENT(RightMouseUp);
-DECLARE_EVENT(OtherMouseDown);
-DECLARE_EVENT(OtherMouseUp);
-DECLARE_EVENT(MouseMoved);
-DECLARE_EVENT(LeftMouseDragged);
-DECLARE_EVENT(RightMouseDragged);
-DECLARE_EVENT(OtherMouseDragged);
-DECLARE_EVENT(ScrollWheel);
-DECLARE_EVENT(KeyDown);
-DECLARE_EVENT(KeyUp);
-DECLARE_EVENT(FlagsChanged);
-#undef DECLARE_EVENT
+/* Use #define instead of static const for case label compatibility */
+#define NSEventTypeLeftMouseDown NSLeftMouseDown
+#define NSEventTypeLeftMouseUp NSLeftMouseUp
+#define NSEventTypeRightMouseDown NSRightMouseDown
+#define NSEventTypeRightMouseUp NSRightMouseUp
+#define NSEventTypeOtherMouseDown NSOtherMouseDown
+#define NSEventTypeOtherMouseUp NSOtherMouseUp
+#define NSEventTypeMouseMoved NSMouseMoved
+#define NSEventTypeLeftMouseDragged NSLeftMouseDragged
+#define NSEventTypeRightMouseDragged NSRightMouseDragged
+#define NSEventTypeOtherMouseDragged NSOtherMouseDragged
+#define NSEventTypeScrollWheel NSScrollWheel
+#define NSEventTypeKeyDown NSKeyDown
+#define NSEventTypeKeyUp NSKeyUp
+#define NSEventTypeFlagsChanged NSFlagsChanged
 
-static const NSEventMask NSEventMaskAny = NSAnyEventMask;
+#define NSEventMaskAny NSAnyEventMask
 
 #define DECLARE_MODIFIER_FLAG(name) static const NSUInteger NSEventModifierFlag##name = NS##name##KeyMask
 DECLARE_MODIFIER_FLAG(Shift);
@@ -79,7 +78,9 @@
 DECLARE_WINDOW_MASK(Resizable);
 DECLARE_WINDOW_MASK(TexturedBackground);
 DECLARE_WINDOW_MASK(UnifiedTitleAndToolbar);
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
 DECLARE_WINDOW_MASK(FullScreen);
+#endif
 /*DECLARE_WINDOW_MASK(FullSizeContentView);*/ /* Not used, fails compile on older SDKs */
 static const unsigned int NSWindowStyleMaskUtilityWindow = NSUtilityWindowMask;
 static const unsigned int NSWindowStyleMaskDocModalWindow = NSDocModalWindowMask;
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoavideo.m patch2-round8/after/src/video/cocoa/SDL_cocoavideo.m
--- a/src/video/cocoa/SDL_cocoavideo.m	2026-03-09 06:07:02.499556217 +0000
+++ b/src/video/cocoa/SDL_cocoavideo.m	2026-03-09 06:07:02.508556448 +0000
@@ -38,9 +38,6 @@
 static void
 Cocoa_DeleteDevice(SDL_VideoDevice * device)
 {
-    if (device->wakeup_lock) {
-        SDL_DestroyMutex(device->wakeup_lock);
-    }
     SDL_free(device->driverdata);
     SDL_free(device);
 }
@@ -66,7 +63,6 @@
         return NULL;
     }
     device->driverdata = data;
-    device->wakeup_lock = SDL_CreateMutex();
 
     /* Set the function pointers */
     device->VideoInit = Cocoa_VideoInit;
@@ -107,6 +103,8 @@
     device->SetWindowMouseGrab = Cocoa_SetWindowMouseGrab;
     device->SetWindowKeyboardGrab = Cocoa_SetWindowKeyboardGrab;
     device->DestroyWindow = Cocoa_DestroyWindow;
+    device->GetWindowSizeInPixels = Cocoa_GetWindowSizeInPixels;
+    device->GetWindowDisplayIndex = Cocoa_GetWindowDisplayIndex;
     device->GetWindowWMInfo = Cocoa_GetWindowWMInfo;
     device->SetWindowHitTest = Cocoa_SetWindowHitTest;
     device->AcceptDragAndDrop = Cocoa_AcceptDragAndDrop;
@@ -173,6 +171,10 @@
 };
 
 
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
+
 int
 Cocoa_VideoInit(_THIS)
 {
@@ -271,10 +273,10 @@
 
 void SDL_NSLog(const char *text)
 {
-    @autoreleasepool {
-        NSString *str = [NSString stringWithUTF8String:text];
-        NSLog(@"%@", str);
-    }
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    NSString *str = [NSString stringWithUTF8String:text];
+    NSLog(@"%@", str);
+    [pool drain];
 }
 
 #endif /* SDL_VIDEO_DRIVER_COCOA */
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoawindow.h patch2-round8/after/src/video/cocoa/SDL_cocoawindow.h
--- a/src/video/cocoa/SDL_cocoawindow.h	2026-03-09 06:07:02.499556217 +0000
+++ b/src/video/cocoa/SDL_cocoawindow.h	2026-03-09 06:07:02.506556397 +0000
@@ -157,6 +157,8 @@
 extern void Cocoa_SetWindowMouseRect(_THIS, SDL_Window * window);
 extern void Cocoa_SetWindowMouseGrab(_THIS, SDL_Window * window, SDL_bool grabbed);
 extern void Cocoa_DestroyWindow(_THIS, SDL_Window * window);
+extern void Cocoa_GetWindowSizeInPixels(_THIS, SDL_Window * window, int *w, int *h);
+extern int Cocoa_GetWindowDisplayIndex(_THIS, SDL_Window * window);
 extern SDL_bool Cocoa_GetWindowWMInfo(_THIS, SDL_Window * window, struct SDL_SysWMinfo *info);
 extern int Cocoa_SetWindowHitTest(SDL_Window *window, SDL_bool enabled);
 extern void Cocoa_AcceptDragAndDrop(SDL_Window * window, SDL_bool accept);
diff -ruN patch2-round8/before/src/video/cocoa/SDL_cocoawindow.m patch2-round8/after/src/video/cocoa/SDL_cocoawindow.m
--- a/src/video/cocoa/SDL_cocoawindow.m	2026-03-09 06:07:02.499556217 +0000
+++ b/src/video/cocoa/SDL_cocoawindow.m	2026-03-09 06:07:02.508556448 +0000
@@ -22,10 +22,6 @@
 
 #if SDL_VIDEO_DRIVER_COCOA
 
-#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
-# error SDL for Mac OS X must be built with a 10.7 SDK or above.
-#endif /* MAC_OS_X_VERSION_MAX_ALLOWED < 1070 */
-
 #include "SDL_syswm.h"
 #include "SDL_timer.h"  /* For SDL_GetTicks() */
 #include "SDL_hints.h"
@@ -61,6 +57,32 @@
 #ifndef NSAppKitVersionNumber10_14
 #define NSAppKitVersionNumber10_14      1671
 #endif
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
+
+/* 10.7+ Notification constants for backwards compatibility */
+#ifndef NSWindowDidChangeBackingPropertiesNotification
+#define NSWindowDidChangeBackingPropertiesNotification @"NSWindowDidChangeBackingPropertiesNotification"
+#endif
+#ifndef NSWindowWillEnterFullScreenNotification
+#define NSWindowWillEnterFullScreenNotification @"NSWindowWillEnterFullScreenNotification"
+#endif
+#ifndef NSWindowDidEnterFullScreenNotification
+#define NSWindowDidEnterFullScreenNotification @"NSWindowDidEnterFullScreenNotification"
+#endif
+#ifndef NSWindowWillExitFullScreenNotification
+#define NSWindowWillExitFullScreenNotification @"NSWindowWillExitFullScreenNotification"
+#endif
+#ifndef NSWindowDidExitFullScreenNotification
+#define NSWindowDidExitFullScreenNotification @"NSWindowDidExitFullScreenNotification"
+#endif
+#ifndef NSBackingPropertyOldScaleFactorKey
+#define NSBackingPropertyOldScaleFactorKey @"NSBackingPropertyOldScaleFactorKey"
+#endif
+#ifndef NSApplicationPresentationFullScreen
+#define NSApplicationPresentationFullScreen (1 << 10)
+#endif
 
 @interface NSWindow (SDL)
 #if MAC_OS_X_VERSION_MAX_ALLOWED < 101000 /* Added in the 10.10 SDK */
@@ -71,7 +93,15 @@
 @property (nonatomic) NSRect mouseConfinementRect;
 @end
 
-@interface SDLWindow : NSWindow <NSDraggingDestination>
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+/* Forward declarations for 10.7+ methods */
+@interface NSView (SDL_ForwardDeclare)
+- (void)setWantsBestResolutionOpenGLSurface:(BOOL)flag;
+- (NSRect)convertRectToBacking:(NSRect)aRect;
+@end
+#endif
+
+@interface SDLWindow : NSWindow
 /* These are needed for borderless/fullscreen windows */
 - (BOOL)canBecomeKeyWindow;
 - (BOOL)canBecomeMainWindow;
@@ -119,13 +149,15 @@
 
 - (void)sendEvent:(NSEvent *)event
 {
+    id delegate;
+
     [super sendEvent:event];
 
     if ([event type] != NSEventTypeLeftMouseUp) {
         return;
     }
 
-    id delegate = [self delegate];
+    delegate = [self delegate];
     if (![delegate isKindOfClass:[Cocoa_WindowListener class]]) {
         return;
     }
@@ -153,30 +185,43 @@
 }
 
 - (BOOL)performDragOperation:(id <NSDraggingInfo>)sender
-{ @autoreleasepool
 {
-    NSPasteboard *pasteboard = [sender draggingPasteboard];
-    NSArray *types = [NSArray arrayWithObject:NSFilenamesPboardType];
-    NSString *desiredType = [pasteboard availableTypeFromArray:types];
-    SDL_Window *sdlwindow = [self findSDLWindow];
+    NSAutoreleasePool *pool;
+    NSPasteboard *pasteboard;
+    NSArray *types;
+    NSString *desiredType;
+    SDL_Window *sdlwindow;
+    NSData *data;
+    NSArray *array;
+    NSPoint point;
+    SDL_Mouse *mouse;
+    int x, y;
+
+    pool = [[NSAutoreleasePool alloc] init];
+    pasteboard = [sender draggingPasteboard];
+    types = [NSArray arrayWithObject:NSFilenamesPboardType];
+    desiredType = [pasteboard availableTypeFromArray:types];
+    sdlwindow = [self findSDLWindow];
 
     if (desiredType == nil) {
+        [pool drain];
         return NO;  /* can't accept anything that's being dropped here. */
     }
 
-    NSData *data = [pasteboard dataForType:desiredType];
+    data = [pasteboard dataForType:desiredType];
     if (data == nil) {
+        [pool drain];
         return NO;
     }
 
     SDL_assert([desiredType isEqualToString:NSFilenamesPboardType]);
-    NSArray *array = [pasteboard propertyListForType:@"NSFilenamesPboardType"];
+    array = [pasteboard propertyListForType:@"NSFilenamesPboardType"];
 
     /* Code addon to update the mouse location */
-    NSPoint point = [sender draggingLocation];
-    SDL_Mouse *mouse = SDL_GetMouse();
-    int x = (int)point.x;
-    int y = (int)(sdlwindow->h - point.y);
+    point = [sender draggingLocation];
+    mouse = SDL_GetMouse();
+    x = (int)point.x;
+    y = (int)(sdlwindow->h - point.y);
     if (x >= 0 && x < sdlwindow->w && y >= 0 && y < sdlwindow->h) {
         SDL_SendMouseMotion(sdlwindow, mouse->mouseID, 0, x, y);
     }
@@ -206,13 +251,15 @@
         }
 
         if (!SDL_SendDropFile(sdlwindow, [[fileURL path] UTF8String])) {
+            [pool drain];
             return NO;
         }
     }
 
     SDL_SendDropComplete(sdlwindow);
+    [pool drain];
     return YES;
-}}
+}
 
 - (BOOL)wantsPeriodicDraggingUpdates
 {
@@ -250,6 +297,9 @@
 static void
 ScheduleContextUpdates(SDL_WindowData *data)
 {
+    NSOpenGLContext *currentContext;
+    NSMutableArray *contexts;
+
     if (!data || !data->nscontexts) {
         return;
     }
@@ -260,8 +310,8 @@
     #pragma clang diagnostic ignored "-Wdeprecated-declarations"
     #endif
 
-    NSOpenGLContext *currentContext = [NSOpenGLContext currentContext];
-    NSMutableArray *contexts = data->nscontexts;
+    currentContext = [NSOpenGLContext currentContext];
+    contexts = data->nscontexts;
     @synchronized (contexts) {
         for (SDLOpenGLContext *context in contexts) {
             if (context == currentContext) {
@@ -573,7 +623,11 @@
     inFullscreenTransition = YES;
 
     /* you need to be FullScreenPrimary, or toggleFullScreen doesn't work. Unset it again in windowDidExitFullScreen. */
-    [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6) {
+        [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+    }
+#endif
     [nswindow performSelectorOnMainThread: @selector(toggleFullScreen:) withObject:nswindow waitUntilDone:NO];
     return YES;
 }
@@ -723,10 +777,15 @@
 - (void)windowDidMove:(NSNotification *)aNotification
 {
     int x, y;
-    SDL_Window *window = _data->window;
-    NSWindow *nswindow = _data->nswindow;
-    BOOL fullscreen = window->flags & FULLSCREEN_MASK;
-    NSRect rect = [nswindow contentRectForFrameRect:[nswindow frame]];
+    SDL_Window *window;
+    NSWindow *nswindow;
+    BOOL fullscreen;
+    NSRect rect;
+
+    window = _data->window;
+    nswindow = _data->nswindow;
+    fullscreen = window->flags & FULLSCREEN_MASK;
+    rect = [nswindow contentRectForFrameRect:[nswindow frame]];
     ConvertNSRect([nswindow screen], fullscreen, &rect);
 
     if (inFullscreenTransition) {
@@ -759,15 +818,20 @@
 
 - (void)windowDidResize:(NSNotification *)aNotification
 {
+    SDL_Window *window;
+    NSWindow *nswindow;
+    int x, y, w, h;
+    NSRect rect;
+    BOOL zoomed;
+
     if (inFullscreenTransition) {
         /* We'll take care of this at the end of the transition */
         return;
     }
 
-    SDL_Window *window = _data->window;
-    NSWindow *nswindow = _data->nswindow;
-    int x, y, w, h;
-    NSRect rect = [nswindow contentRectForFrameRect:[nswindow frame]];
+    window = _data->window;
+    nswindow = _data->nswindow;
+    rect = [nswindow contentRectForFrameRect:[nswindow frame]];
     ConvertNSRect([nswindow screen], (window->flags & FULLSCREEN_MASK), &rect);
     x = (int)rect.origin.x;
     y = (int)rect.origin.y;
@@ -785,7 +849,7 @@
     SDL_SendWindowEvent(window, SDL_WINDOWEVENT_MOVED, x, y);
     SDL_SendWindowEvent(window, SDL_WINDOWEVENT_RESIZED, w, h);
 
-    const BOOL zoomed = [nswindow isZoomed];
+    zoomed = [nswindow isZoomed];
     if (!zoomed) {
         SDL_SendWindowEvent(window, SDL_WINDOWEVENT_RESTORED, 0, 0);
     } else if (zoomed) {
@@ -809,8 +873,14 @@
 
 - (void)windowDidBecomeKey:(NSNotification *)aNotification
 {
-    SDL_Window *window = _data->window;
-    SDL_Mouse *mouse = SDL_GetMouse();
+    SDL_Window *window;
+    SDL_Mouse *mouse;
+    NSPoint point;
+    int x, y;
+    unsigned int newflags;
+
+    window = _data->window;
+    mouse = SDL_GetMouse();
 
     /* We're going to get keyboard events, since we're key. */
     /* This needs to be done before restoring the relative mouse mode. */
@@ -822,9 +892,6 @@
 
     /* If we just gained focus we need the updated mouse position */
     if (!mouse->relative_mode) {
-        NSPoint point;
-        int x, y;
-
         point = [_data->nswindow mouseLocationOutsideOfEventStream];
         x = (int)point.x;
         y = (int)(window->h - point.y);
@@ -841,7 +908,7 @@
         [NSMenu setMenuBarVisible:NO];
     }
 
-    const unsigned int newflags = [NSEvent modifierFlags] & NSEventModifierFlagCapsLock;
+    newflags = [NSEvent modifierFlags] & NSEventModifierFlagCapsLock;
     _data->videodata->modifierFlags = (_data->videodata->modifierFlags & ~NSEventModifierFlagCapsLock) | newflags;
     SDL_ToggleModState(KMOD_CAPS, newflags != 0);
 }
@@ -870,18 +937,25 @@
 
 - (void)windowDidChangeBackingProperties:(NSNotification *)aNotification
 {
-    NSNumber *oldscale = [[aNotification userInfo] objectForKey:NSBackingPropertyOldScaleFactorKey];
+    NSNumber *oldscale;
+
+    oldscale = [[aNotification userInfo] objectForKey:NSBackingPropertyOldScaleFactorKey];
+    (void)oldscale;
 
     if (inFullscreenTransition) {
         return;
     }
 
-    if ([oldscale doubleValue] != [_data->nswindow backingScaleFactor]) {
-        /* Force a resize event when the backing scale factor changes. */
-        _data->window->w = 0;
-        _data->window->h = 0;
-        [self windowDidResize:aNotification];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6) {
+        if ([oldscale doubleValue] != [_data->nswindow backingScaleFactor]) {
+            /* Force a resize event when the backing scale factor changes. */
+            _data->window->w = 0;
+            _data->window->h = 0;
+            [self windowDidResize:aNotification];
+        }
     }
+#endif
 }
 
 - (void)windowDidChangeScreenProfile:(NSNotification *)aNotification
@@ -1013,7 +1087,11 @@
         /* Adjust the fullscreen toggle button and readd menu now that we're here. */
         if (window->flags & SDL_WINDOW_RESIZABLE) {
             /* resizable windows are Spaces-friendly: they get the "go fullscreen" toggle button on their titlebar. */
-            [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+            if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6) {
+                [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+            }
+#endif
         } else {
             [nswindow setCollectionBehavior:NSWindowCollectionBehaviorManaged];
         }
@@ -1182,16 +1260,18 @@
 
 - (void)mouseDown:(NSEvent *)theEvent
 {
-    SDL_Mouse *mouse = SDL_GetMouse();
+    SDL_Mouse *mouse;
+    int button;
+    NSRect windowRect;
+
+    mouse = SDL_GetMouse();
     if (!mouse) {
         return;
     }
 
-    int button;
-
     /* Ignore events that aren't inside the client area (i.e. title bar.) */
     if ([theEvent window]) {
-        NSRect windowRect = [[[theEvent window] contentView] frame];
+        windowRect = [[[theEvent window] contentView] frame];
         if (!NSMouseInRect([theEvent locationInWindow], windowRect, NO)) {
             return;
         }
@@ -1239,13 +1319,14 @@
 
 - (void)mouseUp:(NSEvent *)theEvent
 {
-    SDL_Mouse *mouse = SDL_GetMouse();
+    SDL_Mouse *mouse;
+    int button;
+
+    mouse = SDL_GetMouse();
     if (!mouse) {
         return;
     }
 
-    int button;
-
     if ([self processHitTest:theEvent]) {
         SDL_SendWindowEvent(_data->window, SDL_WINDOWEVENT_HIT_TEST, 0, 0);
         return;  /* stopped dragging, drop event. */
@@ -1286,15 +1367,19 @@
 
 - (void)mouseMoved:(NSEvent *)theEvent
 {
-    SDL_Mouse *mouse = SDL_GetMouse();
+    SDL_Mouse *mouse;
+    SDL_MouseID mouseID;
+    SDL_Window *window;
+    NSPoint point;
+    int x, y;
+
+    mouse = SDL_GetMouse();
     if (!mouse) {
         return;
     }
 
-    const SDL_MouseID mouseID = mouse->mouseID;
-    SDL_Window *window = _data->window;
-    NSPoint point;
-    int x, y;
+    mouseID = mouse->mouseID;
+    window = _data->window;
 
     if ([self processHitTest:theEvent]) {
         SDL_SendWindowEvent(window, SDL_WINDOWEVENT_HIT_TEST, 0, 0);
@@ -1346,9 +1431,14 @@
 
 - (void)touchesBeganWithEvent:(NSEvent *) theEvent
 {
+    BOOL istrackpad;
+    NSSet *touches;
+    SDL_TouchID touchID;
+    int existingTouchCount;
+
     /* probably a MacBook trackpad; make this look like a synthesized event.
        This is backwards from reality, but better matches user expectations. */
-    BOOL istrackpad = NO;
+    istrackpad = NO;
     @try {
         istrackpad = ([theEvent subtype] == NSEventSubtypeMouseEvent);
     }
@@ -1361,9 +1451,9 @@
          */
     }
 
-    NSSet *touches = [theEvent touchesMatchingPhase:NSTouchPhaseAny inView:nil];
-    const SDL_TouchID touchID = istrackpad ? SDL_MOUSE_TOUCHID : (SDL_TouchID)(intptr_t)[[touches anyObject] device];
-    int existingTouchCount = 0;
+    touches = [theEvent touchesMatchingPhase:NSTouchPhaseAny inView:nil];
+    touchID = istrackpad ? SDL_MOUSE_TOUCHID : (SDL_TouchID)(intptr_t)[[touches anyObject] device];
+    existingTouchCount = 0;
 
     for (NSTouch* touch in touches) {
         if ([touch phase] != NSTouchPhaseBegan) {
@@ -1406,11 +1496,14 @@
 
 - (void)handleTouches:(NSTouchPhase) phase withEvent:(NSEvent *) theEvent
 {
-    NSSet *touches = [theEvent touchesMatchingPhase:phase inView:nil];
+    NSSet *touches;
+    BOOL istrackpad;
+
+    touches = [theEvent touchesMatchingPhase:phase inView:nil];
 
     /* probably a MacBook trackpad; make this look like a synthesized event.
        This is backwards from reality, but better matches user expectations. */
-    BOOL istrackpad = NO;
+    istrackpad = NO;
     @try {
         istrackpad = ([theEvent subtype] == NSEventSubtypeMouseEvent);
     }
@@ -1424,15 +1517,22 @@
     }
 
     for (NSTouch *touch in touches) {
-        const SDL_TouchID touchId = istrackpad ? SDL_MOUSE_TOUCHID : (SDL_TouchID)(intptr_t)[touch device];
-        SDL_TouchDeviceType devtype = SDL_TOUCH_DEVICE_INDIRECT_ABSOLUTE;
+        SDL_TouchID touchId;
+        SDL_TouchDeviceType devtype;
+        SDL_Window *window;
+        SDL_FingerID fingerId;
+        float x;
+        float y;
+
+        touchId = istrackpad ? SDL_MOUSE_TOUCHID : (SDL_TouchID)(intptr_t)[touch device];
+        devtype = SDL_TOUCH_DEVICE_INDIRECT_ABSOLUTE;
 
         /* trackpad touches have no window. If we really wanted one we could
          * use the window that has mouse or keyboard focus.
          * Sending a null window currently also prevents synthetic mouse events
          * from being generated from touch events.
          */
-        SDL_Window *window = NULL;
+        window = NULL;
 
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 101202 /* Added in the 10.12.2 SDK. */
         if ([touch respondsToSelector:@selector(type)]) {
@@ -1453,9 +1553,9 @@
             return;
         }
 
-        const SDL_FingerID fingerId = (SDL_FingerID)(intptr_t)[touch identity];
-        float x = [touch normalizedPosition].x;
-        float y = [touch normalizedPosition].y;
+        fingerId = (SDL_FingerID)(intptr_t)[touch identity];
+        x = [touch normalizedPosition].x;
+        y = [touch normalizedPosition].y;
         /* Make the origin the upper left instead of the lower left */
         y = 1.0f - y;
 
@@ -1550,8 +1650,10 @@
 
 - (void)resetCursorRects
 {
+    SDL_Mouse *mouse;
+
     [super resetCursorRects];
-    SDL_Mouse *mouse = SDL_GetMouse();
+    mouse = SDL_GetMouse();
 
     if (mouse->cursor_shown && mouse->cur_cursor && !mouse->relative_mode) {
         [self addCursorRect:[self bounds]
@@ -1574,14 +1676,15 @@
 
 static int
 SetupWindowData(_THIS, SDL_Window * window, NSWindow *nswindow, NSView *nsview, SDL_bool created)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *videodata = (SDL_VideoData *) _this->driverdata;
     SDL_WindowData *data;
 
     /* Allocate the window data */
     window->driverdata = data = (SDL_WindowData *) SDL_calloc(1, sizeof(*data));
     if (!data) {
+        [pool drain];
         return SDL_OutOfMemory();
     }
     data->window = window;
@@ -1656,20 +1759,28 @@
 
     /* All done! */
     window->driverdata = data;
+    [pool drain];
     return 0;
-}}
+}
 
 int
 Cocoa_CreateWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
-    SDL_VideoData *videodata = (SDL_VideoData *) _this->driverdata;
+    NSAutoreleasePool *pool;
+    SDL_VideoData *videodata;
     NSWindow *nswindow;
-    SDL_VideoDisplay *display = SDL_GetDisplayForWindow(window);
+    SDL_VideoDisplay *display;
     NSRect rect;
     SDL_Rect bounds;
     NSUInteger style;
-    NSArray *screens = [NSScreen screens];
+    NSArray *screens;
+    NSScreen *screen;
+    SDLView *contentView;
+
+    pool = [[NSAutoreleasePool alloc] init];
+    videodata = (SDL_VideoData *) _this->driverdata;
+    display = SDL_GetDisplayForWindow(window);
+    screens = [NSScreen screens];
 
     Cocoa_GetDisplayBounds(_this, display, &bounds);
     rect.origin.x = window->x;
@@ -1681,7 +1792,7 @@
     style = GetWindowStyle(window);
 
     /* Figure out which screen to place this window */
-    NSScreen *screen = nil;
+    screen = nil;
     for (NSScreen *candidate in screens) {
         NSRect screenRect = [candidate frame];
         if (rect.origin.x >= screenRect.origin.x &&
@@ -1698,6 +1809,7 @@
         nswindow = [[SDLWindow alloc] initWithContentRect:rect styleMask:style backing:NSBackingStoreBuffered defer:NO screen:screen];
     }
     @catch (NSException *e) {
+        [pool drain];
         return SDL_SetError("%s", [[e reason] UTF8String]);
     }
 
@@ -1714,7 +1826,11 @@
         /* we put FULLSCREEN_DESKTOP windows in their own Space, without a toggle button or menubar, later */
         if (window->flags & SDL_WINDOW_RESIZABLE) {
             /* resizable windows are Spaces-friendly: they get the "go fullscreen" toggle button on their titlebar. */
-            [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+            if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6) {
+                [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+            }
+#endif
         }
     }
 
@@ -1724,7 +1840,7 @@
 
     /* Create a default view for this window */
     rect = [nswindow contentRectForFrameRect:[nswindow frame]];
-    SDLView *contentView = [[SDLView alloc] initWithFrame:rect];
+    contentView = [[SDLView alloc] initWithFrame:rect];
     [contentView setSDLWindow:window];
 
     /* We still support OpenGL as long as Apple offers it, deprecated or not, so disable deprecation warnings about it. */
@@ -1755,36 +1871,49 @@
 
     if (SetupWindowData(_this, window, nswindow, contentView, SDL_TRUE) < 0) {
         [nswindow release];
+        [pool drain];
         return -1;
     }
 
     if (!(window->flags & SDL_WINDOW_OPENGL)) {
+        [pool drain];
         return 0;
     }
-    
+
     /* The rest of this macro mess is for OpenGL or OpenGL ES windows */
 #if SDL_VIDEO_OPENGL_ES2
     if (_this->gl_config.profile_mask == SDL_GL_CONTEXT_PROFILE_ES) {
 #if SDL_VIDEO_OPENGL_EGL
         if (Cocoa_GLES_SetupWindow(_this, window) < 0) {
             Cocoa_DestroyWindow(_this, window);
+            [pool drain];
             return -1;
         }
+        [pool drain];
         return 0;
 #else
+        [pool drain];
         return SDL_SetError("Could not create GLES window surface (EGL support not configured)");
 #endif /* SDL_VIDEO_OPENGL_EGL */
     }
 #endif /* SDL_VIDEO_OPENGL_ES2 */
+    [pool drain];
     return 0;
-}}
+}
 
 int
 Cocoa_CreateWindowFrom(_THIS, SDL_Window * window, const void *data)
-{ @autoreleasepool
 {
-    NSView* nsview = nil;
-    NSWindow *nswindow = nil;
+    NSAutoreleasePool *pool;
+    NSView* nsview;
+    NSWindow *nswindow;
+    NSString *title;
+    BOOL highdpi;
+    int result;
+
+    pool = [[NSAutoreleasePool alloc] init];
+    nsview = nil;
+    nswindow = nil;
 
     if ([(id)data isKindOfClass:[NSWindow class]]) {
       nswindow = (NSWindow*)data;
@@ -1796,8 +1925,6 @@
       SDL_assert(false);
     }
 
-    NSString *title;
-
     /* Query the title from the existing window */
     title = [nswindow title];
     if (title) {
@@ -1812,42 +1939,46 @@
     /* Note: as of the macOS 10.15 SDK, this defaults to YES instead of NO when
      * the NSHighResolutionCapable boolean is set in Info.plist. */
     if ([nsview respondsToSelector:@selector(setWantsBestResolutionOpenGLSurface:)]) {
-        BOOL highdpi = (window->flags & SDL_WINDOW_ALLOW_HIGHDPI) != 0;
+        highdpi = (window->flags & SDL_WINDOW_ALLOW_HIGHDPI) != 0;
         [nsview setWantsBestResolutionOpenGLSurface:highdpi];
     }
     #ifdef __clang__
     #pragma clang diagnostic pop
     #endif
 
-    return SetupWindowData(_this, window, nswindow, nsview, SDL_FALSE);
-}}
+    result = SetupWindowData(_this, window, nswindow, nsview, SDL_FALSE);
+    [pool drain];
+    return result;
+}
 
 void
 Cocoa_SetWindowTitle(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     const char *title = window->title ? window->title : "";
     NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
     NSString *string = [[NSString alloc] initWithUTF8String:title];
     [nswindow setTitle:string];
     [string release];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowIcon(_THIS, SDL_Window * window, SDL_Surface * icon)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSImage *nsimage = Cocoa_CreateImage(icon);
 
     if (nsimage) {
         [NSApp setApplicationIconImage:nsimage];
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowPosition(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
     NSWindow *nswindow = windata->nswindow;
     NSRect rect;
@@ -1865,12 +1996,13 @@
     s_moveHack = moveHack;
 
     ScheduleContextUpdates(windata);
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowSize(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
     NSWindow *nswindow = windata->nswindow;
     NSRect rect;
@@ -1892,12 +2024,13 @@
     s_moveHack = moveHack;
 
     ScheduleContextUpdates(windata);
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowMinimumSize(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
 
     NSSize minSize;
@@ -1905,12 +2038,13 @@
     minSize.height = window->min_h;
 
     [windata->nswindow setContentMinSize:minSize];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowMaximumSize(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
 
     NSSize maxSize;
@@ -1918,12 +2052,13 @@
     maxSize.height = window->max_h;
 
     [windata->nswindow setContentMaxSize:maxSize];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_ShowWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windowData = ((SDL_WindowData *) window->driverdata);
     NSWindow *nswindow = windowData->nswindow;
 
@@ -1932,21 +2067,23 @@
         [nswindow makeKeyAndOrderFront:nil];
         [windowData->listener resumeVisibleObservation];
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_HideWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
 
     [nswindow orderOut:nil];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_RaiseWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windowData = ((SDL_WindowData *) window->driverdata);
     NSWindow *nswindow = windowData->nswindow;
 
@@ -1959,24 +2096,26 @@
         [nswindow makeKeyAndOrderFront:nil];
     }
     [windowData->listener resumeVisibleObservation];
-}}
+    [pool drain];
+}
 
 void
 Cocoa_MaximizeWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
     NSWindow *nswindow = windata->nswindow;
 
     [nswindow zoom:nil];
 
     ScheduleContextUpdates(windata);
-}}
+    [pool drain];
+}
 
 void
 Cocoa_MinimizeWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
     NSWindow *nswindow = data->nswindow;
     if ([data->listener isInFullscreenSpaceTransition]) {
@@ -1984,12 +2123,13 @@
     } else {
         [nswindow miniaturize:nil];
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_RestoreWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
 
     if ([nswindow isMiniaturized]) {
@@ -1997,23 +2137,25 @@
     } else if ((window->flags & SDL_WINDOW_RESIZABLE) && [nswindow isZoomed]) {
         [nswindow zoom:nil];
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowBordered(_THIS, SDL_Window * window, SDL_bool bordered)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     if (SetWindowStyle(window, GetWindowStyle(window))) {
         if (bordered) {
             Cocoa_SetWindowTitle(_this, window);  /* this got blanked out. */
         }
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowResizable(_THIS, SDL_Window * window, SDL_bool resizable)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     /* Don't set this if we're in a space!
      * The window will get permanently stuck if resizable is false.
      * -flibit
@@ -2028,32 +2170,45 @@
     if (videodata->allow_spaces) {
         if (resizable) {
             /* resizable windows are Spaces-friendly: they get the "go fullscreen" toggle button on their titlebar. */
-            [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+            if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_6) {
+                [nswindow setCollectionBehavior:NSWindowCollectionBehaviorFullScreenPrimary];
+            }
+#endif
         } else {
             [nswindow setCollectionBehavior:NSWindowCollectionBehaviorManaged];
         }
     }
-}}
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowAlwaysOnTop(_THIS, SDL_Window * window, SDL_bool on_top)
-{ @autoreleasepool
-    {
-        NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
-        if (on_top) {
-            [nswindow setLevel:NSFloatingWindowLevel];
-        } else {
-            [nswindow setLevel:kCGNormalWindowLevel];
-        }
-    }}
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    NSWindow *nswindow = ((SDL_WindowData *) window->driverdata)->nswindow;
+    if (on_top) {
+        [nswindow setLevel:NSFloatingWindowLevel];
+    } else {
+        [nswindow setLevel:kCGNormalWindowLevel];
+    }
+    [pool drain];
+}
 
 void
 Cocoa_SetWindowFullscreen(_THIS, SDL_Window * window, SDL_VideoDisplay * display, SDL_bool fullscreen)
-{ @autoreleasepool
 {
-    SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
-    NSWindow *nswindow = data->nswindow;
+    NSAutoreleasePool *pool;
+    SDL_WindowData *data;
+    NSWindow *nswindow;
     NSRect rect;
+    SDL_Rect bounds;
+    NSRect screenRect;
+    NSRect frameRect;
+
+    pool = [[NSAutoreleasePool alloc] init];
+    data = (SDL_WindowData *) window->driverdata;
+    nswindow = data->nswindow;
 
     /* The view responder chain gets messed with during setStyleMask */
     if ([data->sdlContentView nextResponder] == data->listener) {
@@ -2061,8 +2216,6 @@
     }
 
     if (fullscreen) {
-        SDL_Rect bounds;
-
         Cocoa_GetDisplayBounds(_this, display, &bounds);
         rect.origin.x = bounds.x;
         rect.origin.y = bounds.y;
@@ -2074,7 +2227,7 @@
            This is no longer needed as of Mac OS X 10.15, according to bug 4822.
          */
         if (floor(NSAppKitVersionNumber) <= NSAppKitVersionNumber10_14) {
-            NSRect screenRect = [[nswindow screen] frame];
+            screenRect = [[nswindow screen] frame];
             if (screenRect.size.height >= 1.0f) {
                 rect.origin.y += (screenRect.size.height - rect.size.height);
             }
@@ -2097,7 +2250,7 @@
         [nswindow setStyleMask:GetWindowWindowedStyle(window)];
 
         /* Hack to restore window decorations on Mac OS X 10.10 */
-        NSRect frameRect = [nswindow frame];
+        frameRect = [nswindow frame];
         [nswindow setFrame:NSMakeRect(frameRect.origin.x, frameRect.origin.y, frameRect.size.width + 1, frameRect.size.height) display:NO];
         [nswindow setFrame:frameRect display:NO];
     }
@@ -2133,7 +2286,8 @@
     }
 
     ScheduleContextUpdates(data);
-}}
+    [pool drain];
+}
 
 int
 Cocoa_SetWindowGammaRamp(_THIS, SDL_Window * window, const Uint16 * ramp)
@@ -2250,9 +2404,13 @@
 
 void
 Cocoa_DestroyWindow(_THIS, SDL_Window * window)
-{ @autoreleasepool
 {
-    SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
+    NSAutoreleasePool *pool;
+    SDL_WindowData *data;
+    NSArray *contexts;
+
+    pool = [[NSAutoreleasePool alloc] init];
+    data = (SDL_WindowData *) window->driverdata;
 
     if (data) {
         if ([data->listener isInFullscreenSpace]) {
@@ -2266,9 +2424,9 @@
             [data->nswindow close];
         }
 
-        NSArray *contexts = [[data->nscontexts copy] autorelease];
+        contexts = [[data->nscontexts copy] autorelease];
         for (SDLOpenGLContext *context in contexts) {
-            /* Calling setWindow:NULL causes the context to remove itself from the context list. */            
+            /* Calling setWindow:NULL causes the context to remove itself from the context list. */
             [context setWindow:NULL];
         }
         [data->nscontexts release];
@@ -2276,7 +2434,8 @@
         SDL_free(data);
     }
     window->driverdata = NULL;
-}}
+    [pool drain];
+}
 
 SDL_bool
 Cocoa_GetWindowWMInfo(_THIS, SDL_Window * window, SDL_SysWMinfo * info)
@@ -2308,12 +2467,13 @@
 
 SDL_bool
 Cocoa_SetWindowFullscreenSpace(SDL_Window * window, SDL_bool state)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_bool succeeded = SDL_FALSE;
     SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
 
     if (data->inWindowFullscreenTransition) {
+        [pool drain];
         return SDL_FALSE;
     }
 
@@ -2346,8 +2506,9 @@
     }
     data->inWindowFullscreenTransition = SDL_FALSE;
 
+    [pool drain];
     return succeeded;
-}}
+}
 
 int
 Cocoa_SetWindowHitTest(SDL_Window * window, SDL_bool enabled)
@@ -2368,8 +2529,8 @@
 
 int
 Cocoa_FlashWindow(_THIS, SDL_Window *window, SDL_FlashOperation operation)
-{ @autoreleasepool
 {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     /* Note that this is app-wide and not window-specific! */
     SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
 
@@ -2389,10 +2550,12 @@
         data->flash_request = [NSApp requestUserAttention:NSCriticalRequest];
         break;
     default:
+        [pool drain];
         return SDL_Unsupported();
     }
+    [pool drain];
     return 0;
-}}
+}
 
 int
 Cocoa_SetWindowOpacity(_THIS, SDL_Window * window, float opacity)
@@ -2402,6 +2565,68 @@
     return 0;
 }
 
+void
+Cocoa_GetWindowSizeInPixels(_THIS, SDL_Window * window, int *w, int *h)
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    SDL_WindowData *windata = (SDL_WindowData *) window->driverdata;
+    NSView *contentView = windata->sdlContentView;
+    NSRect viewport = [contentView bounds];
+
+    if (window->flags & SDL_WINDOW_ALLOW_HIGHDPI) {
+        /* This gives us the correct viewport for a Retina-enabled view. */
+        if ([contentView respondsToSelector:@selector(convertRectToBacking:)]) {
+            viewport = [contentView convertRectToBacking:viewport];
+        }
+    }
+
+    *w = viewport.size.width;
+    *h = viewport.size.height;
+    [pool drain];
+}
+
+int
+Cocoa_GetWindowDisplayIndex(_THIS, SDL_Window * window)
+{
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    NSScreen *screen;
+    SDL_WindowData *data = (SDL_WindowData *) window->driverdata;
+
+    /* Not recognized via CHECK_WINDOW_MAGIC */
+    if (data == nil) {
+        /* Don't set the error here, it hides other errors and is ignored anyway */
+        /*return SDL_SetError("Window data not set");*/
+        [pool drain];
+        return -1;
+    }
+
+    /* NSWindow.screen may be nil when the window is off-screen. */
+    screen = [data->nswindow screen];
+
+    if (screen != nil) {
+        CGDirectDisplayID displayid;
+        int i;
+
+        /* https://developer.apple.com/documentation/appkit/nsscreen/1388360-devicedescription?language=objc */
+        displayid = [[[screen deviceDescription] objectForKey:@"NSScreenNumber"] unsignedIntValue];
+
+        for (i = 0; i < _this->num_displays; i++) {
+            SDL_DisplayData *displaydata = (SDL_DisplayData *)_this->displays[i].driverdata;
+            if (displaydata != NULL && displaydata->display == displayid) {
+                [pool drain];
+                return i;
+            }
+        }
+    }
+
+    /* Other code may expect SDL_GetWindowDisplayIndex to always return a valid
+     * index for a window. The higher level GetWindowDisplayIndex code will fall
+     * back to a generic position-based query if the backend implementation
+     * fails. */
+    [pool drain];
+    return SDL_SetError("Couldn't find the display where the window is located.");
+}
+
 #endif /* SDL_VIDEO_DRIVER_COCOA */
 
 /* vi: set ts=4 sw=4 expandtab: */
