--- a/src/video/cocoa/SDL_cocoaclipboard.m	2026-03-09 08:52:41.013072727 +0000
+++ b/src/video/cocoa/SDL_cocoaclipboard.m	2026-03-09 07:44:08.266972723 +0000
@@ -25,13 +25,23 @@
 #include "SDL_cocoavideo.h"
 #include "../../events/SDL_clipboardevents_c.h"
 
+static NSString *
+GetTextFormat(_THIS)
+{
+    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_5) {
+        return NSPasteboardTypeString;
+    } else {
+        return NSStringPboardType;
+    }
+}
+
 int
 Cocoa_SetClipboardText(_THIS, const char *text)
 {
     NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     SDL_VideoData *data = (SDL_VideoData *) _this->driverdata;
     NSPasteboard *pasteboard;
-    NSString *format = NSPasteboardTypeString;
+    NSString *format = GetTextFormat(_this);
     NSString *nsstr = [NSString stringWithUTF8String:text];
     if (nsstr == nil) {
         [pool drain];
@@ -51,7 +61,7 @@
 {
     NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
     NSPasteboard *pasteboard;
-    NSString *format = NSPasteboardTypeString;
+    NSString *format = GetTextFormat(_this);
     NSString *available;
     char *text;
 
--- a/src/video/cocoa/SDL_cocoaevents.m	2026-03-09 08:52:41.015072778 +0000
+++ b/src/video/cocoa/SDL_cocoaevents.m	2026-03-09 07:45:56.744783102 +0000
@@ -212,23 +212,28 @@
      */
     for (NSWindow *window in [NSApp orderedWindows]) {
         if (window != win && [window canBecomeKeyWindow]) {
-            if (![window isOnActiveSpace]) {
-                continue;
+            if ([window respondsToSelector:@selector(isOnActiveSpace)]) {
+                if (![window isOnActiveSpace]) {
+                    continue;
+                }
             }
             [window makeKeyAndOrderFront:self];
             return;
         }
     }
 
-    /* If a window wasn't found above, iterate through all visible windows in
-     * the active Space in z-order (including the 'About' window, if it's shown)
-     * and make the first one key.
+    /* If a window wasn't found above, iterate through all visible windows
+     * (including the 'About' window, if it's shown) and make the first one key.
+     * Note that +[NSWindow windowNumbersWithOptions:] was added in 10.6.
      */
-    for (NSNumber *num in [NSWindow windowNumbersWithOptions:0]) {
-        NSWindow *window = [NSApp windowWithWindowNumber:[num integerValue]];
-        if (window && window != win && [window canBecomeKeyWindow]) {
-            [window makeKeyAndOrderFront:self];
-            return;
+    if ([NSWindow respondsToSelector:@selector(windowNumbersWithOptions:)]) {
+        /* Get all visible windows in the active Space, in z-order. */
+        for (NSNumber *num in [NSWindow windowNumbersWithOptions:0]) {
+            NSWindow *window = [NSApp windowWithWindowNumber:[num integerValue]];
+            if (window && window != win && [window canBecomeKeyWindow]) {
+                [window makeKeyAndOrderFront:self];
+                return;
+            }
         }
     }
 }
@@ -475,7 +480,16 @@
         s_bShouldHandleEventsInSDLApplication = SDL_TRUE;
 
         if (!SDL_GetHintBoolean(SDL_HINT_MAC_BACKGROUND_APP, SDL_FALSE)) {
-            [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
+#if MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_6
+            if ([NSApp respondsToSelector:@selector(setActivationPolicy:)]) {
+#endif
+                [NSApp setActivationPolicy:NSApplicationActivationPolicyRegular];
+#if MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_6
+            } else {
+                ProcessSerialNumber psn = {0, kCurrentProcess};
+                TransformProcessType(&psn, kProcessTransformToForegroundApplication);
+            }
+#endif
         }
 
         /* If there aren't already menus in place, look to see if there's
--- a/src/video/cocoa/SDL_cocoakeyboard.m	2026-03-09 08:52:41.015072778 +0000
+++ b/src/video/cocoa/SDL_cocoakeyboard.m	2026-03-09 08:35:19.979354409 +0000
@@ -40,6 +40,10 @@
 @end
 #endif
 
+#ifndef NSAppKitVersionNumber10_6
+#define NSAppKitVersionNumber10_6 1038
+#endif
+
 @interface SDLTranslatorResponder : NSView <NSTextInputClient> {
     NSString *_markedText;
     NSRange   _markedRange;
@@ -481,8 +485,11 @@
     SDL_SetScancodeName(SDL_SCANCODE_RALT, "Right Option");
     SDL_SetScancodeName(SDL_SCANCODE_RGUI, "Right Command");
 
-    data->modifierFlags = (unsigned int)[NSEvent modifierFlags];
-    SDL_ToggleModState(KMOD_CAPS, (data->modifierFlags & NSEventModifierFlagCapsLock) != 0);
+    /* On pre-10.6, you might have the initial capslock key state wrong. */
+    if (floor(NSAppKitVersionNumber) >= NSAppKitVersionNumber10_6) {
+        data->modifierFlags = (unsigned int)[NSEvent modifierFlags];
+        SDL_ToggleModState(KMOD_CAPS, (data->modifierFlags & NSEventModifierFlagCapsLock) != 0);
+    }
 }
 
 void
