--- a/src/joystick/iphoneos/SDL_mfijoystick_c.h	2026-03-09 09:26:01.760626557 +0000
+++ b/src/joystick/iphoneos/SDL_mfijoystick_c.h	2026-03-09 07:28:42.420012343 +0000
@@ -35,7 +35,7 @@
 {
     SDL_bool accelerometer;
 
-    GCController __unsafe_unretained *controller;
+    GCController *controller;
     void *rumble;
     int pause_button_index;
     Uint32 pause_button_pressed;
@@ -60,8 +60,8 @@
     SDL_bool is_backbone_one;
     int is_siri_remote;
 
-    NSArray __unsafe_unretained *axes;
-    NSArray __unsafe_unretained *buttons;
+    NSArray *axes;
+    NSArray *buttons;
 
     SDL_bool has_dualshock_touchpad;
     SDL_bool has_xbox_paddles;
--- a/src/joystick/iphoneos/SDL_mfijoystick.m	2026-03-09 09:26:02.174637282 +0000
+++ b/src/joystick/iphoneos/SDL_mfijoystick.m	2026-03-09 09:41:33.909724399 +0000
@@ -787,22 +787,22 @@
     }
 
 #ifdef SDL_JOYSTICK_MFI
-    @autoreleasepool {
-        /* These were explicitly retained in the struct, so they should be explicitly released before freeing the struct. */
-        if (device->controller) {
-            GCController *controller = CFBridgingRelease((__bridge CFTypeRef)(device->controller));
-            controller.controllerPausedHandler = nil;
-            device->controller = nil;
-        }
-        if (device->axes) {
-            CFRelease((__bridge CFTypeRef)device->axes);
-            device->axes = nil;
-        }
-        if (device->buttons) {
-            CFRelease((__bridge CFTypeRef)device->buttons);
-            device->buttons = nil;
-        }
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+    /* These were explicitly retained in the struct, so they should be explicitly released before freeing the struct. */
+    if (device->controller) {
+        GCController *controller = CFBridgingRelease((__bridge CFTypeRef)(device->controller));
+        controller.controllerPausedHandler = nil;
+        device->controller = nil;
+    }
+    if (device->axes) {
+        CFRelease((__bridge CFTypeRef)device->axes);
+        device->axes = nil;
+    }
+    if (device->buttons) {
+        CFRelease((__bridge CFTypeRef)device->buttons);
+        device->buttons = nil;
     }
+    [pool drain];
 #endif /* SDL_JOYSTICK_MFI */
 
     --numjoysticks;
@@ -819,19 +819,24 @@
 static void SDLCALL SDL_AppleTVRemoteRotationHintChanged(void *udata, const char *name, const char *oldValue, const char *newValue)
 {
     BOOL allowRotation = newValue != NULL && *newValue != '0';
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
 
-    @autoreleasepool {
-        for (GCController *controller in [GCController controllers]) {
-            if (controller.microGamepad) {
-                controller.microGamepad.allowsRotation = allowRotation;
-            }
+    for (GCController *controller in [GCController controllers]) {
+        if (controller.microGamepad) {
+            controller.microGamepad.allowsRotation = allowRotation;
         }
     }
+    [pool drain];
 }
 #endif /* TARGET_OS_TV */
 
 static int IOS_JoystickInit(void)
 {
+    NSAutoreleasePool *pool;
+#ifdef SDL_JOYSTICK_MFI
+    NSNotificationCenter *center;
+#endif
+
     if (!SDL_GetHintBoolean(SDL_HINT_JOYSTICK_MFI, SDL_TRUE)) {
         return 0;
     }
@@ -849,22 +854,22 @@
 #endif
 #endif
 
-    @autoreleasepool {
+    pool = [[NSAutoreleasePool alloc] init];
 #ifdef SDL_JOYSTICK_MFI
-        NSNotificationCenter *center;
 #endif
 #ifdef SDL_JOYSTICK_iOS_ACCELEROMETER
-        if (SDL_GetHintBoolean(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, SDL_TRUE)) {
-            /* Default behavior, accelerometer as joystick */
-            IOS_AddJoystickDevice(nil, SDL_TRUE);
-        }
+    if (SDL_GetHintBoolean(SDL_HINT_ACCELEROMETER_AS_JOYSTICK, SDL_TRUE)) {
+        /* Default behavior, accelerometer as joystick */
+        IOS_AddJoystickDevice(nil, SDL_TRUE);
+    }
 #endif
 
 #ifdef SDL_JOYSTICK_MFI
-        /* GameController.framework was added in iOS 7. */
-        if (![GCController class]) {
-            return 0;
-        }
+    /* GameController.framework was added in iOS 7. */
+    if (![GCController class]) {
+        [pool drain];
+        return 0;
+    }
 
         /* For whatever reason, this always returns an empty array on
          macOS 11.0.1 */
@@ -905,7 +910,7 @@
                                                SDL_UnlockJoysticks();
                                              }];
 #endif /* SDL_JOYSTICK_MFI */
-    }
+    [pool drain];
 
     return 0;
 }
@@ -976,7 +981,10 @@
 
 static int IOS_JoystickOpen(SDL_Joystick *joystick, int device_index)
 {
-    SDL_JoystickDeviceItem *device = GetDeviceForIndex(device_index);
+    SDL_JoystickDeviceItem *device;
+    NSAutoreleasePool *pool;
+
+    device = GetDeviceForIndex(device_index);
     if (device == NULL) {
         return SDL_SetError("Could not open Joystick: no hardware device for the specified index");
     }
@@ -994,9 +1002,9 @@
     }
 
     device->joystick = joystick;
+    pool = [[NSAutoreleasePool alloc] init];
 
-    @autoreleasepool {
-        if (device->accelerometer) {
+    if (device->accelerometer) {
 #ifdef SDL_JOYSTICK_iOS_ACCELEROMETER
             if (motionManager == nil) {
                 motionManager = [[CMMotionManager alloc] init];
@@ -1043,8 +1051,8 @@
 #endif /* ENABLE_MFI_SYSTEM_GESTURE_STATE */
 
 #endif /* SDL_JOYSTICK_MFI */
-        }
     }
+    [pool drain];
     if (device->is_siri_remote) {
         ++SDL_AppleTVRemoteOpenedAsJoystick;
     }
@@ -1058,15 +1066,16 @@
     const float maxgforce = SDL_IPHONE_MAX_GFORCE;
     const SInt16 maxsint16 = 0x7FFF;
     CMAcceleration accel;
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
 
-    @autoreleasepool {
-        if (!motionManager.isAccelerometerActive) {
-            return;
-        }
-
-        accel = motionManager.accelerometerData.acceleration;
+    if (!motionManager.isAccelerometerActive) {
+        [pool drain];
+        return;
     }
 
+    accel = motionManager.accelerometerData.acceleration;
+    [pool drain];
+
     /*
      Convert accelerometer data from floating point to Sint16, which is what
      the joystick system expects.
@@ -1123,7 +1132,7 @@
 static void IOS_MFIJoystickUpdate(SDL_Joystick *joystick)
 {
 #ifdef SDL_JOYSTICK_MFI
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         SDL_JoystickDeviceItem *device = joystick->hwdata;
         GCController *controller = device->controller;
         Uint8 hatstate = SDL_HAT_CENTERED;
@@ -1205,6 +1214,7 @@
 
             if (buttons == NULL) {
                 SDL_OutOfMemory();
+                [pool drain];
                 return;
             }
 
@@ -1262,6 +1272,7 @@
 
             if (buttons == NULL) {
                 SDL_OutOfMemory();
+                [pool drain];
                 return;
             }
 
@@ -1398,6 +1409,7 @@
             }
         }
 #endif /* ENABLE_MFI_BATTERY */
+    [pool drain];
     }
 #endif /* SDL_JOYSTICK_MFI */
 }
@@ -1416,7 +1428,7 @@
 
 - (void)cleanup
 {
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         if (@available(macOS 10.16, iOS 14.0, tvOS 14.0, *)) {
             if (self.player != nil) {
                 [self.player cancelAndReturnError:nil];
@@ -1427,17 +1439,19 @@
                 self.engine = nil;
             }
         }
+    [pool drain];
     }
 }
 
 - (int)setIntensity:(float)intensity
 {
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         if (@available(macOS 10.16, iOS 14.0, tvOS 14.0, *)) {
             NSError *error = nil;
             CHHapticDynamicParameter *param;
 
             if (self.engine == nil) {
+                [pool drain];
                 return SDL_SetError("Haptics engine was stopped");
             }
 
@@ -1446,6 +1460,7 @@
                     [self.player stopAtTime:0 error:&error];
                 }
                 self.active = false;
+                [pool drain];
                 return 0;
             }
 
@@ -1454,11 +1469,13 @@
                 CHHapticEvent *event = [[CHHapticEvent alloc] initWithEventType:CHHapticEventTypeHapticContinuous parameters:[NSArray arrayWithObjects:event_param, nil] relativeTime:0 duration:GCHapticDurationInfinite];
                 CHHapticPattern *pattern = [[CHHapticPattern alloc] initWithEvents:[NSArray arrayWithObject:event] parameters:[[NSArray alloc] init] error:&error];
                 if (error != nil) {
+                    [pool drain];
                     return SDL_SetError("Couldn't create haptic pattern: %s", [error.localizedDescription UTF8String]);
                 }
 
                 self.player = [self.engine createPlayerWithPattern:pattern error:&error];
                 if (error != nil) {
+                    [pool drain];
                     return SDL_SetError("Couldn't create haptic player: %s", [error.localizedDescription UTF8String]);
                 }
                 self.active = false;
@@ -1467,6 +1484,7 @@
             param = [[CHHapticDynamicParameter alloc] initWithParameterID:CHHapticDynamicParameterIDHapticIntensityControl value:intensity relativeTime:0];
             [self.player sendParameters:[NSArray arrayWithObject:param] atTime:0 error:&error];
             if (error != nil) {
+                [pool drain];
                 return SDL_SetError("Couldn't update haptic player: %s", [error.localizedDescription UTF8String]);
             }
 
@@ -1475,14 +1493,14 @@
                 self.active = true;
             }
         }
-
+    [pool drain];
         return 0;
     }
 }
 
 - (id)initWithController:(GCController *)controller locality:(GCHapticsLocality)locality API_AVAILABLE(macos(10.16), ios(14.0), tvos(14.0))
 {
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         NSError *error;
         __weak __typeof(self) weakSelf;
         self = [super init];
@@ -1491,12 +1509,14 @@
         self.engine = [controller.haptics createEngineWithLocality:locality];
         if (self.engine == nil) {
             SDL_SetError("Couldn't create haptics engine");
+            [pool drain];
             return nil;
         }
 
         [self.engine startAndReturnError:&error];
         if (error != nil) {
             SDL_SetError("Couldn't start haptics engine");
+            [pool drain];
             return nil;
         }
 
@@ -1518,7 +1538,7 @@
           _this.player = nil;
           [_this.engine startAndReturnError:nil];
         };
-
+    [pool drain];
         return self;
     }
 }
@@ -1581,19 +1601,22 @@
 
 static SDL_RumbleContext *IOS_JoystickInitRumble(GCController *controller)
 {
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         if (@available(macOS 10.16, iOS 14.0, tvOS 14.0, *)) {
             SDL_RumbleMotor *low_frequency_motor = [[SDL_RumbleMotor alloc] initWithController:controller locality:GCHapticsLocalityLeftHandle];
             SDL_RumbleMotor *high_frequency_motor = [[SDL_RumbleMotor alloc] initWithController:controller locality:GCHapticsLocalityRightHandle];
             SDL_RumbleMotor *left_trigger_motor = [[SDL_RumbleMotor alloc] initWithController:controller locality:GCHapticsLocalityLeftTrigger];
             SDL_RumbleMotor *right_trigger_motor = [[SDL_RumbleMotor alloc] initWithController:controller locality:GCHapticsLocalityRightTrigger];
             if (low_frequency_motor && high_frequency_motor) {
-                return [[SDL_RumbleContext alloc] initWithLowFrequencyMotor:low_frequency_motor
+                SDL_RumbleContext *result = [[SDL_RumbleContext alloc] initWithLowFrequencyMotor:low_frequency_motor
                                                          HighFrequencyMotor:high_frequency_motor
                                                            LeftTriggerMotor:left_trigger_motor
                                                           RightTriggerMotor:right_trigger_motor];
+                [pool drain];
+                return result;
             }
         }
+    [pool drain];
     }
     return nil;
 }
@@ -1663,10 +1686,11 @@
     Uint32 result = 0;
 
 #if defined(ENABLE_MFI_LIGHT) || defined(ENABLE_MFI_RUMBLE)
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         SDL_JoystickDeviceItem *device = joystick->hwdata;
 
         if (device == NULL) {
+            [pool drain];
             return 0;
         }
 
@@ -1690,6 +1714,7 @@
             }
 #endif
         }
+    [pool drain];
     }
 #endif /* ENABLE_MFI_LIGHT || ENABLE_MFI_RUMBLE */
 
@@ -1699,10 +1724,11 @@
 static int IOS_JoystickSetLED(SDL_Joystick *joystick, Uint8 red, Uint8 green, Uint8 blue)
 {
 #ifdef ENABLE_MFI_LIGHT
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         SDL_JoystickDeviceItem *device = joystick->hwdata;
 
         if (device == NULL) {
+            [pool drain];
             return SDL_SetError("Controller is no longer connected");
         }
 
@@ -1713,9 +1739,11 @@
                 light.color = [[GCColor alloc] initWithRed:(float)red / 255.0f
                                                      green:(float)green / 255.0f
                                                       blue:(float)blue / 255.0f];
+                [pool drain];
                 return 0;
             }
         }
+    [pool drain];
     }
 #endif /* ENABLE_MFI_LIGHT */
 
@@ -1730,10 +1758,11 @@
 static int IOS_JoystickSetSensorsEnabled(SDL_Joystick *joystick, SDL_bool enabled)
 {
 #ifdef ENABLE_MFI_SENSORS
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         SDL_JoystickDeviceItem *device = joystick->hwdata;
 
         if (device == NULL) {
+            [pool drain];
             return SDL_SetError("Controller is no longer connected");
         }
 
@@ -1742,9 +1771,11 @@
             GCMotion *motion = controller.motion;
             if (motion) {
                 motion.sensorsActive = enabled ? YES : NO;
+                [pool drain];
                 return 0;
             }
         }
+    [pool drain];
     }
 #endif /* ENABLE_MFI_SENSORS */
 
@@ -1768,7 +1799,10 @@
 
 static void IOS_JoystickClose(SDL_Joystick *joystick)
 {
-    SDL_JoystickDeviceItem *device = joystick->hwdata;
+    SDL_JoystickDeviceItem *device;
+    NSAutoreleasePool *pool;
+
+    device = joystick->hwdata;
 
     if (device == NULL) {
         return;
@@ -1776,7 +1810,7 @@
 
     device->joystick = NULL;
 
-    @autoreleasepool {
+    pool = [[NSAutoreleasePool alloc] init];
 #ifdef ENABLE_MFI_RUMBLE
         if (device->rumble) {
             SDL_RumbleContext *rumble = (__bridge SDL_RumbleContext *)device->rumble;
@@ -1810,15 +1844,15 @@
 
 #endif /* SDL_JOYSTICK_MFI */
         }
-    }
     if (device->is_siri_remote) {
         --SDL_AppleTVRemoteOpenedAsJoystick;
     }
+    [pool drain];
 }
 
 static void IOS_JoystickQuit(void)
 {
-    @autoreleasepool {
+    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
 #ifdef SDL_JOYSTICK_MFI
         NSNotificationCenter *center = [NSNotificationCenter defaultCenter];
 
@@ -1845,9 +1879,9 @@
 #ifdef SDL_JOYSTICK_iOS_ACCELEROMETER
         motionManager = nil;
 #endif
-    }
 
     numjoysticks = 0;
+    [pool drain];
 }
 
 static SDL_bool IOS_JoystickGetGamepadMapping(int device_index, SDL_GamepadMapping *out)
