--- configure.orig	2026-08-17 23:30:13.000000000 -0600
+++ configure	2026-08-17 23:34:45.000000000 -0600
@@ -29304,11 +29304,7 @@
         fi
         # Set up files for the audio library
         if test x$enable_audio = xyes; then
-
-printf "%s\n" "#define SDL_AUDIO_DRIVER_COREAUDIO 1" >>confdefs.h
-
-            SOURCES="$SOURCES $srcdir/src/audio/coreaudio/*.m"
-            SUMMARY_audio="${SUMMARY_audio} coreaudio"
+            CheckPulseAudio
             have_audio=yes
         fi
         # Set up files for the joystick library
@@ -29450,12 +29446,7 @@
         fi
         # Set up files for the audio library
         if test x$enable_audio = xyes; then
-
-printf "%s\n" "#define SDL_AUDIO_DRIVER_COREAUDIO 1" >>confdefs.h
-
-            SOURCES="$SOURCES $srcdir/src/audio/coreaudio/*.m"
-            EXTRA_LDFLAGS="$EXTRA_LDFLAGS -Wl,-framework,CoreAudio -Wl,-framework,AudioToolbox"
-            SUMMARY_audio="${SUMMARY_audio} coreaudio"
+            CheckPulseAudio
             have_audio=yes
         fi
         # Set up files for the joystick library
