--- yuvfilters/Makefile.in	2021-09-05 14:14:13.000000000 +0800
+++ yuvfilters/Makefile.in	2025-04-12 16:57:47.000000000 +0800
@@ -350,7 +350,9 @@
 noinst_LTLIBRARIES = libyuvfilters.la
 AM_CPPFLAGS = -I$(top_srcdir)/utils
 # needed for yuvkineco.c - removing the abs() calls breaks the program
-AM_CFLAGS = -Wno-absolute-value
+# AM_CFLAGS = -Wno-absolute-value
+# This is just a flag to disable a warning, but it is not supported by gcc-4.2,
+# drop it.
 MJPEGLIB = $(top_builddir)/utils/libmjpegutils.la $(am__append_1)
 libyuvfilters_la_SOURCES = \
 	addtask.c \
