--- include/st_config.h.in	2025-01-11 04:58:36.000000000 +0800
+++ include/st_config.h.in	2026-02-20 11:02:38.000000000 +0800
@@ -56,11 +56,15 @@
 #   define ST_AVAILABILITY(...)
 #endif
 
+#ifdef __apple_build_version__
 #define ST_FILESYSTEM_AVAILABILITY \
     ST_AVAILABILITY(macosx,strict,introduced=10.15) \
     ST_AVAILABILITY(ios,strict,introduced=13.0) \
     ST_AVAILABILITY(tvos,strict,introduced=13.0) \
     ST_AVAILABILITY(watchos,strict,introduced=6.0)
+#else
+#define ST_FILESYSTEM_AVAILABILITY
+#endif
 
 #if defined(_MSC_VER)
 #   define ST_DEPRECATED(message) __declspec(deprecated(message))
