From 18835a91d1c675ee3a943b8579c8a6cc04fd809d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 23 Oct 2025 23:54:40 +0800
Subject: [PATCH] prov:tcp: TCP_KEEPINTVL, TCP_KEEPCNT: use only when available

---
 prov/tcp/src/xnet_ep.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/prov/tcp/src/xnet_ep.c b/prov/tcp/src/xnet_ep.c
index b28fc7104c4..ec174b57c5e 100644
--- prov/tcp/src/xnet_ep.c
+++ prov/tcp/src/xnet_ep.c
@@ -196,6 +196,7 @@ xnet_enable_keepalive(struct xnet_ep *ep)
 		goto out;
 	}
 
+#ifdef TCP_KEEPINTVL
 	ret = setsockopt(ep->bsock.sock, IPPROTO_TCP, TCP_KEEPINTVL, (const void *)&keep_intvl,
 			 sizeof(keep_intvl));
 	if (ret) {
@@ -203,7 +204,9 @@ xnet_enable_keepalive(struct xnet_ep *ep)
 		FI_WARN(&xnet_prov, FI_LOG_EP_CTRL, "set TCP_KEEPINTVL failed %d", ret);
 		goto out;
 	}
+#endif
 
+#ifdef TCP_KEEPCNT
 	ret = setsockopt(ep->bsock.sock, IPPROTO_TCP, TCP_KEEPCNT, (const void *)&keep_cnt,
 			 sizeof(keep_cnt));
 	if (ret) {
@@ -211,6 +214,7 @@ xnet_enable_keepalive(struct xnet_ep *ep)
 		FI_WARN(&xnet_prov, FI_LOG_EP_CTRL, "set SO_KEEPALIVE failed %d", ret);
 		goto out;
 	}
+#endif
 
 	FI_INFO(&xnet_prov, FI_LOG_EP_CTRL, "%p KEEPALIVE idle %d intvl %d cnt %d\n",
 		ep, idle_time, keep_intvl, keep_cnt);

From e3d5238f29c28802aef94df14c2f62eff58665f7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 24 Oct 2025 00:06:44 +0800
Subject: [PATCH] osd.h: use correct form of macOS availability macro

---
 include/osx/osd.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/osx/osd.h b/include/osx/osd.h
index 7b7742c64b2..12aa43b9f31 100644
--- include/osx/osd.h
+++ include/osx/osd.h
@@ -183,7 +183,7 @@ ssize_t ofi_recvmsg_tcp(SOCKET fd, struct msghdr *msg, int flags);
  * Fallback: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/spinlock.3.html
  */
 
-#if __MAC_OS_X_VERSION_MIN_REQUIRED > 101100
+#if MAC_OS_X_VERSION_MIN_REQUIRED > 101100
 
 #include <os/lock.h>
 
