From 7d578378a7152842293178b5af5a6d3fb4c18039 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 7 Aug 2025 22:37:31 +0800
Subject: [PATCH] event.c: add a fallback for pthread_threadid_np for macOS

---
 src/event.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git src/event.c src/event.c
index 304b7e3..d414253 100644
--- src/event.c
+++ src/event.c
@@ -36,6 +36,10 @@
 #include <Winsock2.h>
 #endif
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 void zlog_event_profile(zlog_event_t * a_event, int flag)
 {
 	zc_assert(a_event,);
@@ -103,9 +107,13 @@ zlog_event_t *zlog_event_new(int time_cache_count)
 
 #ifdef __linux__
 	a_event->ktid = syscall(SYS_gettid);
-#elif __APPLE__
+#elif defined(__APPLE__)
     uint64_t tid64;
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060 || defined(__POWERPC__)
+    tid64 = pthread_mach_thread_np(pthread_self());
+#else
     pthread_threadid_np(NULL, &tid64);
+#endif
     a_event->tid = (pthread_t)tid64;
 #endif
 
