--- src/Makefile.orig	2026-08-30 03:32:55.000000000 +0800
+++ src/Makefile	2026-09-10 11:43:44.000000000 +0800
@@ -34,6 +34,7 @@
 ORIG_SRC   = $(ORIG_GEN) $(ORIG_NORM)
 ORIG_FILES = $(ORIG_SRC) $(ORIG_HDR)
 
+CFLAGS_ADDTL += -std=c99
 
 .PHONY: boxes.static check_dir clean build cov debug package static flags_unix flags_static flags_windows flags_
 
@@ -109,10 +110,10 @@
 
 # Bison generates `yynerrs` bookkeeping that newer MinGW GCC warns about, even though
 # the variable lives entirely in generated code.
-parser.o: CFLAGS += -Wno-unused-but-set-variable
+# parser.o: CFLAGS += -Wno-unused-but-set-variable
 
 # Flex 2.6.x emits REJECT bookkeeping that Clang warns about in generated code.
-lex.yy.o: CFLAGS += -Wno-misleading-indentation
+# lex.yy.o: CFLAGS += -Wno-misleading-indentation
 
 lex.yy.c lex.yy.h: lexer.l | check_dir
 	$(LEX) --header-file=lex.yy.h $<
