From c53bc82558c1933b58a43b2b9d3034a23918dc7d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 8 Jan 2026 14:15:50 +0800
Subject: [PATCH 3/6] Backport a90ee65f7b0264a672634fc4f5891a2ab57a47ae

---
 include/GLFW/glfw3.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git include/GLFW/glfw3.h include/GLFW/glfw3.h
index fa245711..b6412161 100644
--- include/GLFW/glfw3.h
+++ include/GLFW/glfw3.h
@@ -88,6 +88,12 @@ extern "C" {
  #endif
 #endif /* APIENTRY */
 
+/* Some OpenGL related headers use GLAPIENTRY instead.
+ */
+#ifndef GLAPIENTRY
+ #define GLAPIENTRY APIENTRY
+#endif /* GLAPIENTRY */
+
 /* Some Windows OpenGL headers need this.
  */
 #if !defined(WINGDIAPI) && defined(_WIN32)
-- 
2.52.0

