--- src/cocoa_monitor.m	2016-08-18 19:32:27.000000000 +0800
+++ src/cocoa_monitor.m.new.m	2026-01-11 00:39:08.000000000 +0800
@@ -62,6 +62,11 @@
         return strdup("Unknown");
     }
 
+    if (!value) {
+        CFRelease(info);
+        return strdup("Unknown");
+    }
+
     size = CFStringGetMaximumSizeForEncoding(CFStringGetLength(value),
                                              kCFStringEncodingUTF8);
     name = calloc(size + 1, 1);
@@ -87,6 +92,9 @@
         return GLFW_FALSE;
 
     CFStringRef format = CGDisplayModeCopyPixelEncoding(mode);
+    if (!format)
+        return GLFW_FALSE;
+
     if (CFStringCompare(format, CFSTR(IO16BitDirectPixels), 0) &&
         CFStringCompare(format, CFSTR(IO32BitDirectPixels), 0))
     {
@@ -117,6 +125,13 @@
 
     CFStringRef format = CGDisplayModeCopyPixelEncoding(mode);
 
+    if (!format) {
+        result.redBits = 8;
+        result.greenBits = 8;
+        result.blueBits = 8;
+        return result;
+    }
+
     if (CFStringCompare(format, CFSTR(IO16BitDirectPixels), 0) == 0)
     {
         result.redBits = 5;
