From 20eb5e948ba2a4908eb8b74009e5b9b547d43d1c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 8 Jan 2026 16:14:06 +0800
Subject: [PATCH] Do not fail on non-standard GL

diff --git src/nsgl_context.m src/nsgl_context.m
index 9cc91d11..b0f1f20b 100644
--- src/nsgl_context.m
+++ src/nsgl_context.m
@@ -65,6 +65,8 @@ int _glfwCreateContext(_GLFWwindow* window,
 {
     unsigned int attributeCount = 0;
 
+// There is no good in breaking deliberately what otherwise works.
+#if 0
     if (ctxconfig->api == GLFW_OPENGL_ES_API)
     {
         _glfwInputError(GLFW_API_UNAVAILABLE,
@@ -106,6 +108,7 @@ int _glfwCreateContext(_GLFWwindow* window,
     }
 #endif /*MAC_OS_X_VERSION_MAX_ALLOWED*/
 
+#endif
     // Context robustness modes (GL_KHR_robustness) are not yet supported on
     // OS X but are not a hard constraint, so ignore and continue
 
