From 20eb5e948ba2a4908eb8b74009e5b9b547d43d1c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 8 Jan 2026 16:14:06 +0800
Subject: [PATCH 7/7] Do not fail on non-standard GL

---
 include/GLFW/glfw3.h | 30 ++++++++++++++++++++++++++++++
 src/nsgl_context.m   |  3 +++
 2 files changed, 33 insertions(+)

diff --git include/GLFW/glfw3.h include/GLFW/glfw3.h
index 90e1324b..b175e2e2 100644
--- include/GLFW/glfw3.h
+++ include/GLFW/glfw3.h
@@ -1372,6 +1372,36 @@ GLFWAPI void glfwGetMonitorPos(GLFWmonitor* monitor, int* xpos, int* ypos);
  */
 GLFWAPI void glfwGetMonitorPhysicalSize(GLFWmonitor* monitor, int* widthMM, int* heightMM);
 
+/*! @brief Retrieves the content scale for the specified monitor.
+ *
+ *  This function retrieves the content scale for the specified monitor.  The
+ *  content scale is the ratio between the current DPI and the platform's
+ *  default DPI.  If you scale all pixel dimensions by this scale then your
+ *  content should appear at an appropriate size.  This is especially important
+ *  for text and any UI elements.
+ *
+ *  The content scale may depend on both the monitor resolution and pixel
+ *  density and on user settings.  It may be very different from the raw DPI
+ *  calculated from the physical size and current resolution.
+ *
+ *  @param[in] monitor The monitor to query.
+ *  @param[out] xscale Where to store the x-axis content scale, or `NULL`.
+ *  @param[out] yscale Where to store the y-axis content scale, or `NULL`.
+ *
+ *  @errors Possible errors include @ref GLFW_NOT_INITIALIZED and @ref
+ *  GLFW_PLATFORM_ERROR.
+ *
+ *  @thread_safety This function must only be called from the main thread.
+ *
+ *  @sa @ref monitor_scale
+ *  @sa @ref glfwGetWindowContentScale
+ *
+ *  @since Added in version 3.3.
+ *
+ *  @ingroup monitor
+ */
+GLFWAPI void glfwGetMonitorContentScale(GLFWmonitor* monitor, float* xscale, float* yscale);
+
 /*! @brief Returns the name of the specified monitor.
  *
  *  This function returns a human-readable name, encoded as UTF-8, of the
diff --git src/nsgl_context.m src/nsgl_context.m
index 9cc91d11..b0f1f20b 100644
--- src/nsgl_context.m
+++ src/nsgl_context.m
@@ -65,6 +65,8 @@ int _glfwCreateContext(_GLFWwindow* window,
 {
     unsigned int attributeCount = 0;
 
+// There is no good in breaking deliberately what otherwise works.
+#if 0
     if (ctxconfig->api == GLFW_OPENGL_ES_API)
     {
         _glfwInputError(GLFW_API_UNAVAILABLE,
@@ -106,6 +108,7 @@ int _glfwCreateContext(_GLFWwindow* window,
     }
 #endif /*MAC_OS_X_VERSION_MAX_ALLOWED*/
 
+#endif
     // Context robustness modes (GL_KHR_robustness) are not yet supported on
     // OS X but are not a hard constraint, so ignore and continue
 
-- 
2.52.0

