--- CorsixTH/CMakeLists.txt	2026-07-05 23:41:49.000000000 +0800
+++ CorsixTH/CMakeLists.txt	2026-08-08 21:47:50.000000000 +0800
@@ -199,7 +199,7 @@
     if(APPLE)
       target_link_libraries(CorsixTH PRIVATE libz.dylib)
       # These LDFLAGS are necessary for static ffmpeg, harmless for dynamic ffmpeg
-      set(CMAKE_EXE_LINKER_FLAGS "-liconv -framework CoreMedia -framework CoreVideo -framework VideoToolbox")
+      set(CMAKE_EXE_LINKER_FLAGS "-liconv @ATOMIC@")
     endif()
     message("  FFmpeg found")
   else()
@@ -403,7 +403,6 @@
     install(CODE "
       INCLUDE(BundleUtilities)
       SET(BU_CHMOD_BUNDLE_ITEMS ON)
-      FIXUP_BUNDLE(\"${CMAKE_INSTALL_PREFIX}/CorsixTH.app\" \"\" \"\")
       ")
     if(WITH_LUAROCKS)
       install(CODE "execute_process(
