From b7d2100571903526dd2c26e4522dfd5a2720184a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 11 Nov 2025 00:00:04 +0800
Subject: [PATCH] Restore LLVM-7 compatibility

---
 src/rustllvm/PassWrapper.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/rustllvm/PassWrapper.cpp b/src/rustllvm/PassWrapper.cpp
index 0cda3465dc0..5abc02912c4 100644
--- a/src/rustllvm/PassWrapper.cpp
+++ b/src/rustllvm/PassWrapper.cpp
@@ -110,8 +110,10 @@ void LLVMRustAddLastExtensionPasses(
   // so they are run for optimized and non-optimized builds.
   unwrap(PMBR)->addExtension(PassManagerBuilder::EP_OptimizerLast,
                              AddExtensionPasses);
+#if LLVM_VERSION_GE(9, 0)
   unwrap(PMBR)->addExtension(PassManagerBuilder::EP_EnabledOnOptLevel0,
                              AddExtensionPasses);
+#endif
 }
 
 #ifdef LLVM_COMPONENT_X86
@@ -294,7 +296,11 @@ extern "C" void LLVMRustPrintTargetCPUs(LLVMTargetMachineRef TM) {
   const MCSubtargetInfo *MCInfo = Target->getMCSubtargetInfo();
   const Triple::ArchType HostArch = Triple(sys::getProcessTriple()).getArch();
   const Triple::ArchType TargetArch = Target->getTargetTriple().getArch();
+#if LLVM_VERSION_GE(9, 0)
   const ArrayRef<SubtargetSubTypeKV> CPUTable = MCInfo->getCPUTable();
+#else
+  const ArrayRef<SubtargetFeatureKV> CPUTable = MCInfo->getCPUTable();
+#endif
   unsigned MaxCPULen = getLongestEntryLength(CPUTable);
 
   printf("Available CPUs for this target:\n");
-- 
2.51.1

