From 21bedbb14739809ba4ec247f22632ee3cca64c55 Mon Sep 17 00:00:00 2001
From: "Kirill A. Korinsky" <kirill@korins.ky>
Date: Wed, 9 Feb 2022 05:47:47 +0100
Subject: [PATCH] `curl-sys`: never try to link with `/lib/darwin`

diff --git a/vendor/curl-sys/build.rs b/vendor/curl-sys/build.rs
index 64df41e995..3213c66b33 100644
--- a/src/vendor/curl-sys/build.rs
+++ b/src/vendor/curl-sys/build.rs
@@ -26,9 +26,9 @@
     let dst = PathBuf::from(env::var_os("OUT_DIR").unwrap());
     let windows = target.contains("windows");
 
-    // OSX and Haiku ships libcurl by default, so we just use that version
+    // Haiku ships libcurl by default, so we just use that version
     // unconditionally.
-    if target.contains("apple") || target.contains("haiku") {
+    if target.contains("haiku") {
         return println!("cargo:rustc-flags=-l curl");
     }
 
