--- /dev/null
+++ b/vendor/rustc-ap-rustc_target/spec/aarch64_apple_darwin.rs
@@ -0,0 +1,29 @@
+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    let mut base = super::apple_base::opts();
+    base.cpu = "apple-a12".to_string();
+    base.max_atomic_width = Some(128);
+    base.pre_link_args.insert(LinkerFlavor::Gcc, vec!["-arch".to_string(), "arm64".to_string()]);
+    base.link_env_remove.extend(super::apple_base::macos_link_env_remove());
+
+    // Clang automatically chooses a more specific target based on
+    // MACOSX_DEPLOYMENT_TARGET.  To enable cross-language LTO to work
+    // correctly, we do too.
+    let arch = "aarch64";
+    let llvm_target = super::apple_base::macos_llvm_target(&arch);
+
+    Ok(Target {
+        llvm_target,
+        target_endian: "little".to_string(),
+        target_pointer_width: "64".to_string(),
+        target_c_int_width: "32".to_string(),
+        data_layout: "e-m:o-i64:64-i128:128-n32:64-S128".to_string(),
+        arch: arch.to_string(),
+        target_os: "macos".to_string(),
+        target_env: String::new(),
+        target_vendor: "apple".to_string(),
+        linker_flavor: LinkerFlavor::Gcc,
+        options: TargetOptions { target_mcount: "\u{1}mcount".to_string(), ..base },
+    })
+}

--- /dev/null
+++ b/vendor/rustc-ap-rustc_target/spec/powerpc_apple_darwin.rs
@@ -0,0 +1,33 @@
+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    let mut base = super::apple_base::opts();
+    base.cpu = "g4".to_string();
+    base.max_atomic_width = Some(32);
+    base.pre_link_args.insert(LinkerFlavor::Gcc, vec!["-arch".to_string(), "ppc".to_string()]);
+    base.link_env_remove.extend(super::apple_base::macos_link_env_remove());
+    base.eliminate_frame_pointer = false;
+
+    // Clang automatically chooses a more specific target based on
+    // MACOSX_DEPLOYMENT_TARGET.  To enable cross-language LTO to work
+    // correctly, we do too.
+    let arch = "powerpc";
+    let llvm_target = super::apple_base::macos_llvm_target(&arch);
+
+    Ok(Target {
+        llvm_target: llvm_target,
+        target_endian: "big".to_string(),
+        target_pointer_width: "32".to_string(),
+        target_c_int_width: "32".to_string(),
+        data_layout: "E-m:e-p:32:32-i64:64-n32".to_string(),
+        arch: "powerpc".to_string(),
+        target_os: "macos".to_string(),
+        target_env: String::new(),
+        target_vendor: "apple".to_string(),
+        linker_flavor: LinkerFlavor::Gcc,
+        options: TargetOptions {
+            target_mcount: "\u{1}mcount".to_string(),
+            .. base
+        },
+    })
+}

--- /dev/null
+++ b/vendor/rustc-ap-rustc_target/spec/powerpc64_apple_darwin.rs
@@ -0,0 +1,33 @@
+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    let mut base = super::apple_base::opts();
+    base.cpu = "ppc64".to_string();
+    base.max_atomic_width = Some(64);
+    base.pre_link_args.insert(LinkerFlavor::Gcc, vec!["-arch".to_string(), "ppc64".to_string()]);
+    base.link_env_remove.extend(super::apple_base::macos_link_env_remove());
+    base.eliminate_frame_pointer = false;
+
+    // Clang automatically chooses a more specific target based on
+    // MACOSX_DEPLOYMENT_TARGET.  To enable cross-language LTO to work
+    // correctly, we do too.
+    let arch = "powerpc64";
+    let llvm_target = super::apple_base::macos_llvm_target(&arch);
+
+    Ok(Target {
+        llvm_target: llvm_target,
+        target_endian: "big".to_string(),
+        target_pointer_width: "64".to_string(),
+        target_c_int_width: "32".to_string(),
+        data_layout: "E-m:e-i64:64-n32:64".to_string(),
+        arch: "powerpc64".to_string(),
+        target_os: "macos".to_string(),
+        target_env: String::new(),
+        target_vendor: "apple".to_string(),
+        linker_flavor: LinkerFlavor::Gcc,
+        options: TargetOptions {
+            target_mcount: "\u{1}mcount".to_string(),
+            .. base
+        },
+    })
+}

--- a/vendor/rustc-ap-rustc_target/spec/mod.rs	2019-11-05 00:34:46
+++ b/vendor/rustc-ap-rustc_target/spec/mod.rs	2025-11-23 14:53:20
@@ -416,8 +416,11 @@
     ("i686-unknown-haiku", i686_unknown_haiku),
     ("x86_64-unknown-haiku", x86_64_unknown_haiku),
 
-    ("x86_64-apple-darwin", x86_64_apple_darwin),
+    ("aarch64-apple-darwin", aarch64_apple_darwin),
     ("i686-apple-darwin", i686_apple_darwin),
+    ("powerpc-apple-darwin", powerpc_apple_darwin),
+    ("powerpc64-apple-darwin", powerpc64_apple_darwin),
+    ("x86_64-apple-darwin", x86_64_apple_darwin),
 
     ("aarch64-fuchsia", aarch64_fuchsia),
     ("x86_64-fuchsia", x86_64_fuchsia),
