--- a/vendor/backtrace-sys/src/libbacktrace/macho.c	2019-11-05 01:34:41.000000000 +0800
+++ b/vendor/backtrace-sys/src/libbacktrace/macho.c	2025-11-11 16:02:12.000000000 +0800
@@ -58,6 +58,10 @@
 #define NATIVE_CPU_TYPE CPU_TYPE_ARM64
 #elif defined(__arm__)
 #define NATIVE_CPU_TYPE CPU_TYPE_ARM
+#elif defined(__ppc64__)
+#define NATIVE_CPU_TYPE CPU_TYPE_POWERPC64
+#elif defined(__ppc__)
+#define NATIVE_CPU_TYPE CPU_TYPE_POWERPC
 #else
 #error Could not detect native Mach-O cpu_type_t
 #endif
