From the context, both ppc and ppc64 are meant here.

--- a/vendor/jemalloc-sys/jemalloc/include/jemalloc/internal/jemalloc_internal_types.h	2019-11-05 01:34:41.000000000 +0800
+++ b/vendor/jemalloc-sys/jemalloc/include/jemalloc/internal/jemalloc_internal_types.h	2025-11-12 08:22:51.000000000 +0800
@@ -91,7 +91,7 @@
 #  ifdef __or1k__
 #    define LG_QUANTUM		3
 #  endif
-#  ifdef __powerpc__
+#  if defined(__powerpc__) || defined(__POWERPC__)
 #    define LG_QUANTUM		4
 #  endif
 #  if defined(__riscv) || defined(__riscv__)
