--- a/vendor/libc/src/unix/bsd/apple/mod.rs	2021-07-26 23:20:38.000000000 +0800
+++ b/vendor/libc/src/unix/bsd/apple/mod.rs	2025-12-15 21:03:34.000000000 +0800
@@ -4081,10 +4081,10 @@
 }
 
 cfg_if! {
-    if #[cfg(any(target_arch = "arm", target_arch = "x86"))] {
+    if #[cfg(any(target_arch = "arm", target_arch = "x86", target_arch = "powerpc"))] {
         mod b32;
         pub use self::b32::*;
-    } else if #[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))] {
+    } else if #[cfg(any(target_arch = "x86_64", target_arch = "aarch64", target_arch = "powerpc64"))] {
         mod b64;
         pub use self::b64::*;
     } else {
