--- a/vendor/openssl-src/src/lib.rs	2019-11-05 01:34:46.000000000 +0800
+++ b/vendor/openssl-src/src/lib.rs	2025-11-12 08:52:38.000000000 +0800
@@ -158,7 +158,9 @@
             "mips64-unknown-linux-gnuabi64" => "linux64-mips64",
             "mips64el-unknown-linux-gnuabi64" => "linux64-mips64",
             "mipsel-unknown-linux-gnu" => "linux-mips32",
+            "powerpc-apple-darwin" => "darwin-ppc-cc",
             "powerpc-unknown-linux-gnu" => "linux-ppc",
+            "powerpc64-apple-darwin" => "darwin-ppc64-cc",
             "powerpc64-unknown-linux-gnu" => "linux-ppc64",
             "powerpc64le-unknown-linux-gnu" => "linux-ppc64le",
             "s390x-unknown-linux-gnu" => "linux64-s390x",
