--- a/vendor/backtrace/src/symbolize/gimli/macho.rs	2021-07-26 23:20:38.000000000 +0800
+++ b/vendor/backtrace/src/symbolize/gimli/macho.rs	2025-12-15 21:10:01.000000000 +0800
@@ -103,6 +103,10 @@
             Some(macho::CPU_TYPE_ARM)
         } else if cfg!(target_arch = "aarch64") {
             Some(macho::CPU_TYPE_ARM64)
+        } else if cfg!(target_arch = "powerpc") {
+            Some(macho::CPU_TYPE_POWERPC)
+        } else if cfg!(target_arch = "powerpc64") {
+            Some(macho::CPU_TYPE_POWERPC64)
         } else {
             None
         }
