Non-functional, but not to forget:

--- a/vendor/libgit2-sys/libgit2/src/hash/sha1/generic.c	2019-11-05 01:34:46.000000000 +0800
+++ b/vendor/libgit2-sys/libgit2/src/hash/sha1/generic.c	2025-11-12 08:44:24.000000000 +0800
@@ -65,6 +65,7 @@
  * and is faster on architectures with memory alignment issues.
  */
 
+/* FIXME: should we add __POWERPC__ here or let it keep using a generic fallback below? */
 #if defined(__i386__) || defined(__x86_64__) || \
 	defined(_M_IX86) || defined(_M_X64) || \
 	defined(__ppc__) || defined(__ppc64__) || \
