--- a/vendor/openssl-src/src/lib.rs	2021-07-26 23:20:39.000000000 +0800
+++ b/vendor/openssl-src/src/lib.rs	2025-12-15 21:19:42.000000000 +0800
@@ -248,7 +248,9 @@
             "mips64el-unknown-linux-gnuabi64" => "linux64-mips64",
             "mipsel-unknown-linux-gnu" => "linux-mips32",
             "mipsel-unknown-linux-musl" => "linux-mips32",
+            "powerpc-apple-darwin" => "darwin-ppc-cc",
             "powerpc-unknown-linux-gnu" => "linux-ppc",
+            "powerpc64-apple-darwin" => "darwin-ppc64-cc",
             "powerpc64-unknown-freebsd" => "BSD-generic64",
             "powerpc64-unknown-linux-gnu" => "linux-ppc64",
             "powerpc64le-unknown-freebsd" => "BSD-generic64",
