--- a/vendor/crossbeam-utils/src/lib.rs	2021-07-26 23:20:38.000000000 +0800
+++ b/vendor/crossbeam-utils/src/lib.rs	2025-12-20 19:54:43.000000000 +0800
@@ -75,8 +75,6 @@
                 AtomicBool, AtomicI16, AtomicI32, AtomicI8, AtomicIsize, AtomicU16, AtomicU32,
                 AtomicU8, AtomicUsize,
             };
-            #[cfg(not(crossbeam_no_atomic_64))]
-            pub(crate) use core::sync::atomic::{AtomicI64, AtomicU64};
         }
 
         #[cfg(feature = "std")]

--- a/vendor/crossbeam-utils/src/atomic/atomic_cell.rs	2021-07-26 23:20:38.000000000 +0800
+++ b/vendor/crossbeam-utils/src/atomic/atomic_cell.rs	2025-12-20 19:55:22.000000000 +0800
@@ -537,13 +537,6 @@
 impl_arithmetic!(i16, atomic::AtomicI16, "let a = AtomicCell::new(7i16);");
 impl_arithmetic!(u32, atomic::AtomicU32, "let a = AtomicCell::new(7u32);");
 impl_arithmetic!(i32, atomic::AtomicI32, "let a = AtomicCell::new(7i32);");
-#[cfg(not(crossbeam_no_atomic_64))]
-impl_arithmetic!(u64, atomic::AtomicU64, "let a = AtomicCell::new(7u64);");
-#[cfg(not(crossbeam_no_atomic_64))]
-impl_arithmetic!(i64, atomic::AtomicI64, "let a = AtomicCell::new(7i64);");
-// TODO: AtomicU128 is unstable
-// impl_arithmetic!(u128, atomic::AtomicU128, "let a = AtomicCell::new(7u128);");
-// impl_arithmetic!(i128, atomic::AtomicI128, "let a = AtomicCell::new(7i128);");
 
 impl_arithmetic!(
     usize,
@@ -838,10 +831,6 @@
             atomic!(@check, $t, atomic::AtomicU8, $a, $atomic_op);
             atomic!(@check, $t, atomic::AtomicU16, $a, $atomic_op);
             atomic!(@check, $t, atomic::AtomicU32, $a, $atomic_op);
-            #[cfg(not(crossbeam_no_atomic_64))]
-            atomic!(@check, $t, atomic::AtomicU64, $a, $atomic_op);
-            // TODO: AtomicU128 is unstable
-            // atomic!(@check, $t, atomic::AtomicU128, $a, $atomic_op);
 
             #[cfg(crossbeam_loom)]
             unimplemented!("loom does not support non-atomic atomic ops");
@@ -859,10 +848,6 @@
         | can_transmute::<T, atomic::AtomicU8>()
         | can_transmute::<T, atomic::AtomicU16>()
         | can_transmute::<T, atomic::AtomicU32>();
-    #[cfg(not(crossbeam_no_atomic_64))]
-    let is_lock_free = is_lock_free | can_transmute::<T, atomic::AtomicU64>();
-    // TODO: AtomicU128 is unstable
-    // let is_lock_free = is_lock_free | can_transmute::<T, atomic::AtomicU128>();
     is_lock_free
 }
 
--- a/vendor/crossbeam-utils/src/atomic/consume.rs	2021-07-26 23:20:38.000000000 +0800
+++ b/vendor/crossbeam-utils/src/atomic/consume.rs	2025-12-20 19:54:51.000000000 +0800
@@ -74,10 +74,6 @@
 impl_atomic!(AtomicI16, i16);
 impl_atomic!(AtomicU32, u32);
 impl_atomic!(AtomicI32, i32);
-#[cfg(not(crossbeam_no_atomic_64))]
-impl_atomic!(AtomicU64, u64);
-#[cfg(not(crossbeam_no_atomic_64))]
-impl_atomic!(AtomicI64, i64);
 
 #[cfg(not(crossbeam_no_atomic))]
 impl<T> AtomicConsume for core::sync::atomic::AtomicPtr<T> {
