From ea68befaf70089b7ff6ee97c54cb26547b38f2f7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 13 Apr 2026 22:10:27 +0700
Subject: [PATCH] Dirty hack to broken hir

---
 src/hir/hir.cpp | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/hir/hir.cpp b/src/hir/hir.cpp
index e4485c94..44bac422 100644
--- a/src/hir/hir.cpp
+++ b/src/hir/hir.cpp
@@ -127,17 +127,17 @@ namespace HIR {
             }
 
             // If the MIR is populated
-            if( this->expr->m_mir )
-            {
-                TODO(Span(), "Compare non-expanded array sizes - (w/ MIR) " << *this << " and " << x);
-            }
-            else {
+//            if( this->expr->m_mir )
+//            {
+//                TODO(Span(), "Compare non-expanded array sizes - (w/ MIR) " << *this << " and " << x);
+//            }
+//            else {
                 // EVIL OPTION: Just compare the string representations
                 // - Hopefully there's no pointers printed involved.
                 auto v_t = FMT(*this);
                 auto v_x = FMT(x);
                 return ::ord(v_t, v_x);
-            }
+//            }
         }
         if(auto cmp = this->params_impl.ord(x.params_impl)) return cmp;
         if(auto cmp = this->params_item.ord(x.params_item)) return cmp;
-- 
2.53.0

