From 58416cad6c3f54046ab49a5cf72518529943347a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 20 Dec 2025 18:12:37 +0800
Subject: [PATCH] target.cpp: enable 16-bit atomics on powerpc

---
 src/trans/target.cpp | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/trans/target.cpp b/src/trans/target.cpp
index cc91e0fb..9e2d05bc 100644
--- a/src/trans/target.cpp
+++ b/src/trans/target.cpp
@@ -59,18 +59,18 @@ const TargetArch ARCH_POWERPC64 = {
     { /*atomic(u8)=*/true, true, true, true,  true },
     TargetArch::Alignments(2, 4, 8, 16, 4, 8, 8)
 };
-const TargetArch ARCH_POWERPC32 = {
-    "powerpc",
-    32, true,
-    { /*atomic(u8)=*/true, false, true, false,  true },
-    TargetArch::Alignments(2, 4, 8, 8, 4, 8, 4)
-};
 const TargetArch ARCH_POWERPC64LE = {
     "powerpc64",
     64, false,
     { /*atomic(u8)=*/true, true, true, true,  true },
     TargetArch::Alignments(2, 4, 8, 16, 4, 8, 8)
 };
+const TargetArch ARCH_POWERPC = {
+    "powerpc",
+    32, true,
+    { /*atomic(u8)=*/true, true, true, false,  true },
+    TargetArch::Alignments(2, 4, 8, 8, 4, 8, 4)
+};
 const TargetArch ARCH_RISCV64 = {
     "riscv64",
     64, false,
@@ -155,9 +155,9 @@ namespace
                         {
                             rv.m_arch = ARCH_M68K;
                         }
-                        else if( key_val.value.as_string() == ARCH_POWERPC32.m_name )
+                        else if( key_val.value.as_string() == ARCH_POWERPC.m_name )
                         {
-                            rv.m_arch = ARCH_POWERPC32;
+                            rv.m_arch = ARCH_POWERPC;
                         }
                         else if( key_val.value.as_string() == ARCH_POWERPC64.m_name )
                         {
@@ -634,7 +634,7 @@ namespace
             // NOTE: OSX uses Mach-O binaries, which don't fully support the defaults used for GNU targets
             return TargetSpec {
                 "unix", "macos", "gnu", {CodegenMode::Gnu11, true, "powerpc-apple-darwin", {}, {}},
-                ARCH_POWERPC32
+                ARCH_POWERPC
                 };
         }
         else if(target_name == "powerpc64-apple-darwin")
