--- tools/standalone_miri/module_tree.cpp.orig	2025-01-01 00:00:00.000000000 +0000
+++ tools/standalone_miri/module_tree.cpp	2025-01-01 00:00:00.000000000 +0000
@@ -654,7 +654,7 @@
                     else {
                     }
                     auto val = state.parse_lvalue(*this);
-                    src_rval = ::MIR::RValue::make_Borrow({ bt, ::std::move(val) });
+                    src_rval = ::MIR::RValue::make_Borrow({ bt, false, ::std::move(val) });
                 }
                 // Composites
                 else if( lex.consume_if('(') ) {
