From 46344493e1766eb27d2cea78a1417900acc65a4e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 7 Nov 2025 20:03:31 +0800
Subject: [PATCH] minicargo.mk, Makefile: use external llvm

---
 minicargo.mk       | 14 ++++++++------
 run_rustc/Makefile |  3 ++-
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/minicargo.mk b/minicargo.mk
index e6c427a5..bb4ac5df 100644
--- a/minicargo.mk
+++ b/minicargo.mk
@@ -96,7 +96,7 @@ ifeq ($(RUSTC_CHANNEL),nightly)
 else
   RUSTCSRC := rustc-$(RUSTC_VERSION)-src/
 endif
-RUSTC_SRC_DL := $(RUSTCSRC)/dl-version
+RUSTC_SRC_DL := $(RUSTCSRC)dl-version
 ifeq ($(RUSTC_VERSION),1.19.0)
   VENDOR_DIR := $(RUSTCSRC)src/vendor
 else ifeq ($(RUSTC_VERSION),1.29.0)
@@ -137,7 +137,8 @@ ifeq ($(RUSTC_VERSION),1.74.0)
 SRCDIR_RUST_TESTS := $(RUSTCSRC)tests/
 endif
 
-LLVM_CONFIG := $(RUSTCSRC)build/bin/llvm-config
+LLVM_CONFIG := @PREFIX@/libexec/llvm-@LLVM_V@/bin/llvm-config
+
 ifeq ($(shell uname -s || echo not),Darwin)
  # /usr/bin/uname because uname might call coreutils
  # which can make the arm64 uname called when
@@ -222,7 +223,7 @@ $(RUSTC_SRC_DL): rustc-$(RUSTC_VERSION)-src/extracted rustc-$(RUSTC_VERSION)-src
 # - libstd, libpanic_unwind, libtest and libgetopts
 # - libproc_macro (mrustc)
 ifeq ($(USE_MERGED_BUILD),1)
-$(RUSTCSRC)mrustc-stdlib/Cargo.toml: $(RUSTC_SRC_DL) minicargo.mk
+$(RUSTCSRC)mrustc-stdlib/Cargo.toml: minicargo.mk
 	@mkdir -p $(dir $@)
 	@echo "#![no_core]" > $(dir $@)/lib.rs
 	@echo "[package]" > $@
@@ -238,7 +239,7 @@ LIBS: $(RUSTCSRC)mrustc-stdlib/Cargo.toml $(MRUSTC) $(MINICARGO)
 	+$(MINICARGO) --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(OUTDIR) $(MINICARGO_FLAGS) $(RUSTCSRC)mrustc-stdlib/
 	+$(MINICARGO) --output-dir $(OUTDIR) $(MINICARGO_FLAGS) lib/libproc_macro
 else
-LIBS: $(MRUSTC) $(MINICARGO) $(RUSTC_SRC_DL)
+LIBS: $(MRUSTC) $(MINICARGO)
 	+$(MINICARGO) --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(OUTDIR) $(MINICARGO_FLAGS) $(RUSTCSRC)$(RUST_LIB_PREFIX)std
 	+$(MINICARGO) --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(OUTDIR) $(MINICARGO_FLAGS) $(RUSTCSRC)$(RUST_LIB_PREFIX)panic_unwind
 	+$(MINICARGO) --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(OUTDIR) $(MINICARGO_FLAGS) $(RUSTCSRC)$(RUST_LIB_PREFIX)test
@@ -246,7 +247,7 @@ LIBS: $(MRUSTC) $(MINICARGO) $(RUSTC_SRC_DL)
 endif
 
 # Dynamically linked version of the standard library
-$(OUTDIR)test/libtest.so: $(RUSTC_SRC_DL)
+$(OUTDIR)test/libtest.so:
 	mkdir -p $(dir $@)
 	+MINICARGO_DYLIB=1 $(MINICARGO) $(RUSTCSRC)$(RUST_LIB_PREFIX)std          --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(dir $@) $(MINICARGO_FLAGS)
 	+MINICARGO_DYLIB=1 $(MINICARGO) $(RUSTCSRC)$(RUST_LIB_PREFIX)panic_unwind --vendor-dir $(VENDOR_DIR) --script-overrides $(OVERRIDE_DIR) --output-dir $(dir $@) $(MINICARGO_FLAGS)
@@ -255,6 +256,7 @@ $(OUTDIR)test/libtest.so: $(RUSTC_SRC_DL)
 
 RUSTC_ENV_VARS := CFG_COMPILER_HOST_TRIPLE=$(RUSTC_TARGET)
 RUSTC_ENV_VARS += LLVM_CONFIG=$(abspath $(LLVM_CONFIG))
+RUSTC_ENV_VARS += CFG_LLVM_ROOT=@PREFIX@/libexec/llvm-@LLVM_V@
 RUSTC_ENV_VARS += CFG_RELEASE=$(RUSTC_VERSION)	# Claiming stable
 RUSTC_ENV_VARS += CFG_RELEASE_CHANNEL=$(RUSTC_CHANNEL)
 RUSTC_ENV_VARS += CFG_VERSION=$(RUSTC_VERSION)-$(RUSTC_CHANNEL)-mrustc
@@ -264,7 +266,7 @@ RUSTC_ENV_VARS += LD_LIBRARY_PATH=$(abspath $(OUTDIR))
 RUSTC_ENV_VARS += REAL_LIBRARY_PATH_VAR=LD_LIBRARY_PATH
 RUSTC_ENV_VARS += RUSTC_INSTALL_BINDIR=bin
 
-$(OUTDIR)rustc: $(MRUSTC) $(MINICARGO) LIBS $(LLVM_CONFIG)
+$(OUTDIR)rustc: $(MRUSTC) $(MINICARGO) LIBS
 	mkdir -p $(OUTDIR)rustc-build
 	+$(RUSTC_ENV_VARS) $(MINICARGO) $(RUSTCSRC)$(SRCDIR_RUSTC) --vendor-dir $(VENDOR_DIR) --output-dir $(OUTDIR)rustc-build -L $(OUTDIR) $(MINICARGO_FLAGS) $(MINICARGO_FLAGS_$@)
 	test -e $@ -a ! $(OUTDIR)rustc-build/$(RUSTC_OUT_BIN) -nt $@ || cp $(OUTDIR)rustc-build/$(RUSTC_OUT_BIN) $@
diff --git a/run_rustc/Makefile b/run_rustc/Makefile
index a97ce927..1b812596 100644
--- a/run_rustc/Makefile
+++ b/run_rustc/Makefile
@@ -90,11 +90,12 @@ BINDIR := $(PREFIX)bin/
 LIBDIR := $(PREFIX)lib/rustlib/$(RUSTC_TARGET)/lib/
 CARGO_HOME := $(PREFIX)cargo_home/
 
-LLVM_CONFIG ?= $(RUST_SRC)build/bin/llvm-config
+LLVM_CONFIG := @PREFIX@/libexec/llvm-@LLVM_V@/bin/llvm-config
 LLVM_TARGETS ?= X86;ARM;AArch64#;Mips;PowerPC;SystemZ;JSBackend;MSP430;Sparc;NVPTX
 
 RUSTC_ENV_VARS := CFG_COMPILER_HOST_TRIPLE=$(RUSTC_TARGET)
 RUSTC_ENV_VARS += LLVM_CONFIG=$(abspath $(LLVM_CONFIG))
+RUSTC_ENV_VARS += CFG_LLVM_ROOT=@PREFIX@/libexec/llvm-@LLVM_V@
 RUSTC_ENV_VARS += CFG_RELEASE=$(RUSTC_VERSION)	# Claiming stable
 RUSTC_ENV_VARS += CFG_RELEASE_CHANNEL=$(RUSTC_CHANNEL)
 RUSTC_ENV_VARS += CFG_VERSION=$(RUSTC_VERSION)-stable-mrustc
