From 03a413728106ea757cb1aeecada4df0a74c27fb0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 7 Nov 2025 23:53:54 +0800
Subject: [PATCH] codegen_c.cpp: clang does not accept -fno-tree-sra

---
 src/trans/codegen_c.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/trans/codegen_c.cpp b/src/trans/codegen_c.cpp
index 660d22c2..a1350330 100644
--- a/src/trans/codegen_c.cpp
+++ b/src/trans/codegen_c.cpp
@@ -1294,11 +1294,13 @@ namespace {
                     args.push_back("-O1");  // HACK: Work around mrustc #347 by reducing the optimisation level
                     break;
                 }
+#ifndef __clang__
                 // HACK: Work around [https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117423] by disabling an optimisation stage
                 if( opt.opt_level > 0 )
                 {
                     args.push_back("-fno-tree-sra");
                 }
+#endif
                 if( opt.emit_debug_info )
                 {
                     args.push_back("-g");
-- 
2.51.0

