--- a/library/std/src/sys/pal/unix/sync/condvar.rs
+++ b/library/std/src/sys/pal/unix/sync/condvar.rs
@@ -162,7 +162,7 @@
     #[inline]
     fn drop(&mut self) {
         let r = unsafe { libc::pthread_cond_destroy(self.raw()) };
-        if cfg!(target_os = "dragonfly") {
+        if cfg!(any(target_os = "dragonfly", target_os = "macos")) {
             // On DragonFly pthread_cond_destroy() returns EINVAL if called on
             // a condvar that was just initialized with
             // libc::PTHREAD_COND_INITIALIZER. Once it is used or
