--- a/library/core/src/ffi/va_list.rs
+++ b/library/core/src/ffi/va_list.rs
@@ -38,7 +38,7 @@ crate::cfg_select! {
             _marker: PhantomInvariantLifetime<'f>,
         }
     }
-    all(target_arch = "powerpc", not(target_os = "uefi"), not(windows)) => {
+    all(target_arch = "powerpc", not(target_os = "uefi"), not(target_os = "macos"), not(windows)) => {
         /// PowerPC ABI implementation of a `va_list`.
         #[cfg_attr(not(doc), repr(C))] // work around https://github.com/rust-lang/rust/issues/66401
         #[derive(Debug)]
@@ -117,6 +117,7 @@ crate::cfg_select! {
         ),
         not(target_arch = "xtensa"),
         any(not(target_arch = "aarch64"), not(target_vendor = "apple")),
+        any(not(target_arch = "powerpc"), not(target_os = "macos")),
         not(target_family = "wasm"),
         not(target_os = "uefi"),
         not(windows),
