--- Makefile	2025-11-23 00:45:55.000000000 +0800
+++ Makefile	2026-01-13 13:39:04.000000000 +0800
@@ -132,6 +132,9 @@
 ifeq ($(uname_S),OpenBSD)
 	LDLIBS += -liconv
 endif
+ifeq ($(uname_S),Darwin)
+	LDLIBS += -liconv
+endif
 
 LDLIBS := -lunistring $(WX_LIBS) $(GTK_LIBS) $(BOTAN_LIBS) $(CAPSTONE_LIBS) $(JANSSON_LIBS) $(LUA_LIBS) $(LDLIBS)
 
--- src/buffer.cpp	2026-01-13 13:45:37.000000000 +0800
+++ src/buffer.cpp	2026-01-13 13:43:56.000000000 +0800
@@ -319,7 +319,7 @@
 	reload();
 }
 
-#ifdef __APPLE__
+#ifdef __WXCOCOA__
 REHex::Buffer::Buffer(MacFileName &&filename, off_t block_size):
 	Buffer(filename.GetFileName().GetFullPath().ToStdString())
 {
