--- wxLua/bindings/wxwidgets/wxbase_file.i	2025-11-23 00:45:55.000000000 +0800
+++ wxLua/bindings/wxwidgets/wxbase_file.i	2026-01-13 13:26:50.000000000 +0800
@@ -207,7 +207,6 @@
     virtual wxString GetDataDir() const;
     virtual wxString GetDocumentsDir() const;
     virtual wxString GetExecutablePath() const;
-    %wxchkver_3_0_0 && %gtk wxString GetInstallPrefix() const;
     virtual wxString GetLocalDataDir() const;
     virtual wxString GetPluginsDir() const;
     virtual wxString GetResourcesDir() const;
@@ -218,7 +217,6 @@
     %wxchkver_3_0_0 && %win void IgnoreAppSubDir(const wxString& subdirPattern);
     %wxchkver_3_0_0 && %win void IgnoreAppBuildSubDirs();
     %wxchkver_3_0_0 && %win static wxString MSWGetShellDir(int csidl);
-    %wxchkver_3_0_0 && %gtk void SetInstallPrefix(const wxString& prefix);
     %wxchkver_3_0_0 void UseAppInfo(int info);
     %wxchkver_3_1_1 wxStandardPaths::FileLayout GetFileLayout() const; // %override return type
     %wxchkver_3_1_0 wxString GetUserDir(wxStandardPaths::Dir userDir) const; // %override parameter type
