From e811c1a8719561d823599e9706ea1a003915ef1c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 13 Jan 2026 15:13:14 +0800
Subject: [PATCH] search.cpp: use std::isinf and std::isnan

---
 src/search.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/search.cpp b/src/search.cpp
index d89efd6b..4d2c033a 100644
--- src/search.cpp
+++ src/search.cpp
@@ -37,6 +37,9 @@
 #include "search.hpp"
 #include "util.hpp"
 
+using std::isinf;
+using std::isnan;
+
 /* This MUST come after the wxWidgets headers have been included, else we pull in windows.h BEFORE the wxWidgets
  * headers when building on Windows and this causes unicode-flavoured pointer conversion errors.
 */
