--- src/SettingsDialogKeyboard.cpp.orig	2025-11-23 00:45:55.000000000 +0800
+++ src/SettingsDialogKeyboard.cpp	2026-01-13 12:58:09.000000000 +0800
@@ -184,7 +184,7 @@
 #endif
 		}
 		
-#ifdef __APPLE__
+#ifdef __WXCOCOA__
 		if((combination.modifiers & wxACCEL_RAW_CTRL) != 0)
 		{
 			prompt += "CTRL + ";
@@ -225,7 +225,7 @@
 			combination.modifiers |= wxACCEL_SHIFT;
 			break;
 			
-#ifdef __APPLE__
+#ifdef __WXCOCOA__
 		case WXK_RAW_CONTROL:
 			combination.modifiers |= wxACCEL_RAW_CTRL;
 			break;
@@ -268,7 +268,7 @@
 			combination.modifiers &= ~wxACCEL_SHIFT;
 			break;
 			
-#ifdef __APPLE__
+#ifdef __WXCOCOA__
 		case WXK_RAW_CONTROL:
 			combination.modifiers &= ~wxACCEL_RAW_CTRL;
 			break;
