--- yt_dlp/networking/_curlcffi.py.orig	2026-07-05 06:23:27.000000000 +0800
+++ yt_dlp/networking/_curlcffi.py	2026-08-19 08:41:05.000000000 +0800
@@ -33,9 +33,9 @@
 
 curl_cffi_version = tuple(map(int, re.split(r'[^\d]+', curl_cffi.__version__)[:3]))
 
-if curl_cffi_version != (0, 5, 10) and not (0, 10) <= curl_cffi_version < (0, 16):
+if curl_cffi_version != (0, 5, 10) and not (0, 10) <= curl_cffi_version < (0, 17):
     curl_cffi._yt_dlp__version = f'{curl_cffi.__version__} (unsupported)'
-    raise ImportError('Only curl_cffi versions 0.5.10 and 0.10.x through 0.15.x are supported')
+    raise ImportError('Only curl_cffi versions 0.5.10 and 0.10.x through 0.16.x are supported')
 
 import curl_cffi.requests
 from curl_cffi.const import CurlECode, CurlOpt
