--- configure.ac	2023-06-05 21:30:53.000000000 +0800
+++ configure.ac	2026-06-03 00:34:25.000000000 +0800
@@ -907,7 +907,7 @@
 			hdf5_include_dir="$withval"
 		], [hdf5_include_dir=""])
 
-	hdf5_inc_dir="/usr/include/hdf5 /usr/include/hdf5/serial ${hdf5_include_dir}"
+	hdf5_inc_dir="${hdf5_include_dir}"
 
 	enable_include_hdf5=no
 	for f in $hdf5_inc_dir; do
@@ -934,7 +934,7 @@
 			hdf5_libraries_dir="$withval"
 		], [hdf5_libraries_dir=""])
 
-	hdf5_lib_dir="/usr/lib/x86_64-linux-gnu/hdf5 /usr/lib/x86_64-linux-gnu/hdf5/serial ${hdf5_libraries_dir}"
+	hdf5_lib_dir="${hdf5_libraries_dir}"
 
 	enable_libraries_hdf5=no
 	for f in $hdf5_lib_dir; do
