--- src/atlas.nim	2026-02-22 00:00:00.000000000 +0000
+++ src/atlas.nim	2026-03-16 00:00:00.000000000 +0000
@@ -32,7 +32,7 @@
         if line.startsWith("version ="):
           ver = line.split("=")[1].replace("\"", "").replace(" ", "")
       assert ver != ""
-      ver & " (sha: " & staticExec("git log -n 1 --format=%H") & ")"
+      ver & " (sha: @GITHASH@)"

 proc writeHelp() =
   stdout.write """
