--- build/config/compiler/BUILD.gn	2026-09-27 15:23:26.000000000 +0800
+++ build/config/compiler/BUILD.gn	2026-09-27 15:28:58.000000000 +0800
@@ -2010,33 +2010,6 @@
   if (false) {
     ldflags += [ "-Wl,-no_warn_duplicate_libraries" ]
   }
-
-  if (enable_rust_clippy) {
-    configs = [ ":default_clippy_lints" ]
-  }
-}
-
-# default_clippy_lints --------------------------------------------------------
-
-config("default_clippy_lints") {
-  rustflags = [
-    # Treat all default Clippy lints as warnings.
-    # See also `config("treat_warnings_as_errors")`.
-    "-Wclippy::all",
-
-    # Additional lints that we enable beyond the defaults
-    "-Wclippy::undocumented_unsafe_blocks",
-
-    # Lints disabled during the initial rollout period (when any objection
-    # coming from a Chromium engineer would get a lint disabled):
-    "-Aclippy::needless_return",
-
-    # TODO(crbug.com/534628766): Fix and re-enable.
-    "-Aclippy::chunks_exact_to_as_chunks",
-
-    # TODO(crbug.com/556789085): Fix and re-enable.
-    "-Aclippy::needless_range_loop",
-  ]
 }
 
 # prevent_unsafe_narrowing ----------------------------------------------------
