--- build/mac/find_sdk.py	2026-08-17 15:19:51.000000000 +0800
+++ build/mac/find_sdk.py	2026-08-28 08:21:38.000000000 +0800
@@ -82,7 +82,7 @@
         print(err, file=sys.stderr)
         raise Exception('Error %d running xcode-select' % job.returncode)
     dev_dir = out.decode('UTF-8').rstrip()
-    sdk_dir = os.path.join(dev_dir, 'Platforms/MacOSX.platform/Developer/SDKs')
+    sdk_dir = os.path.join(dev_dir, 'SDKs')
 
     if not os.path.isdir(sdk_dir):
         raise SdkError(
@@ -110,7 +110,7 @@
         print(sdk_path)
 
     if options.print_bin_path:
-        bin_path = 'Toolchains/XcodeDefault.xctoolchain/usr/bin/'
+        bin_path = '/usr/bin/'
         print(os.path.join(dev_dir, bin_path))
 
     if options.print_sdk_build:

--- build/config/apple/sdk_info.py	2026-08-17 15:19:51.000000000 +0800
+++ build/config/apple/sdk_info.py	2026-08-28 08:25:00.000000000 +0800
@@ -112,11 +112,11 @@
         .decode('UTF-8')
         .strip()
     )
-    settings['toolchains_path'] = os.path.join(
-        subprocess.check_output(['xcode-select', '-print-path'])
+    settings['toolchains_path'] = (
+        subprocess.check_output(
+            ['xcode-select', '-print-path'])
         .decode('UTF-8')
         .strip(),
-        'Toolchains/XcodeDefault.xctoolchain',
     )
 
 
