--- build/mac/find_sdk.py.orig	2025-10-27 10:10:03.000000000 +0800
+++ build/mac/find_sdk.py	2025-11-04 02:34:59.000000000 +0800
@@ -70,7 +70,7 @@
     raise Exception('Error %d running xcode-select' % job.returncode)
   dev_dir = out.decode('UTF-8').rstrip()
   sdk_dir = os.path.join(
-      dev_dir, 'Platforms/MacOSX.platform/Developer/SDKs')
+      dev_dir, 'SDKs')
 
   if not os.path.isdir(sdk_dir):
     raise SdkError('Install Xcode, launch it, accept the license ' +
@@ -92,7 +92,7 @@
     print(sdk_path)
 
   if options.print_bin_path:
-    bin_path = 'Toolchains/XcodeDefault.xctoolchain/usr/bin/'
+    bin_path = '/usr/bin/'
     print(os.path.join(dev_dir, bin_path))
 
   if options.print_sdk_build:

--- build/config/apple/sdk_info.py.orig	2025-10-27 10:10:03.000000000 +0800
+++ build/config/apple/sdk_info.py	2025-11-04 02:38:56.000000000 +0800
@@ -87,10 +87,8 @@
   settings['sdk_build'] = subprocess.check_output(
       ['xcrun', '-sdk', platform,
        '--show-sdk-build-version']).decode('UTF-8').strip()
-  settings['toolchains_path'] = os.path.join(
-      subprocess.check_output(['xcode-select',
-                               '-print-path']).decode('UTF-8').strip(),
-      'Toolchains/XcodeDefault.xctoolchain')
+  settings['toolchains_path'] =  subprocess.check_output(
+      ['xcode-select', '-print-path']).decode('UTF-8').strip()
 
 
 def CreateXcodeSymlinkAt(src, dst, root_build_dir):
