--- build/config/BUILDCONFIG.gn.orig	2025-09-27 12:42:34.000000000 +0800
+++ build/config/BUILDCONFIG.gn	2025-11-03 22:28:25.000000000 +0800
@@ -384,10 +382,6 @@
   ]
 }
 
-if (is_apple) {
-  default_compiler_configs += [ "//build/config/compiler:enable_arc" ]
-}
-
 if (is_posix) {
   if (current_os != "aix") {
     default_compiler_configs +=

--- build/config/mac/BUILD.gn.orig	2025-09-27 12:42:34.000000000 +0800
+++ build/config/mac/BUILD.gn	2025-11-03 23:47:51.000000000 +0800
@@ -74,7 +78,7 @@
 
 # On Mac, this is used for everything except static libraries.
 config("mac_dynamic_flags") {
-  ldflags = [ "-Wl,-ObjC" ]  # Always load Objective-C categories and classes.
+  ldflags = [ "-Wl,-lobjc" ]  # Always load Objective-C categories and classes.
 
   if (is_component_build) {
     ldflags += [
