--- build/config/host_byteorder.gni.orig	2025-11-28 15:30:25.000000000 +0800
+++ build/config/host_byteorder.gni	2026-01-11 03:36:58.000000000 +0800
@@ -10,7 +10,6 @@
 }
 
 # Detect host byteorder
-assert(current_cpu != "ppc" && current_cpu != "s390")
 if ((host_cpu == "ppc64" && current_os == "aix") || host_cpu == "s390x" ||
     host_cpu == "mips" || host_cpu == "mips64") {
   host_byteorder = "big"
