--- src/common/debug.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/common/debug.cpp	2025-10-27 06:55:57.000000000 +0800
@@ -26,7 +26,11 @@
 #endif
 
 #if defined(ANGLE_PLATFORM_APPLE)
-#    include <os/log.h>
+#    include <AvailabilityMacros.h>
+#    if MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
+#        define MAC_HAS_LOG 1
+#        include <os/log.h>
+#    endif
 #endif
 
 #if defined(ANGLE_PLATFORM_WINDOWS)
@@ -272,7 +276,7 @@
         // Note: we also log to stdout/stderr below.
 #endif
 
-#if defined(ANGLE_PLATFORM_APPLE)
+#if MAC_HAS_LOG
         os_log_type_t apple_log_type = OS_LOG_TYPE_DEFAULT;
         switch (severity)
         {
