--- src/common/debug.cpp	2026-07-21 11:31:12.000000000 +0800
+++ src/common/debug.cpp	2026-07-25 21:40:42.000000000 +0800
@@ -26,10 +26,14 @@
 #endif
 
 #if defined(ANGLE_PLATFORM_APPLE)
-#    pragma GCC diagnostic push
-#    pragma GCC diagnostic ignored "-Wignored-attributes"
-#    include <os/log.h>
-#    pragma GCC diagnostic pop
+#    include <AvailabilityMacros.h>
+#    if MAC_OS_X_VERSION_MIN_REQUIRED >= 110000
+#        pragma GCC diagnostic push
+#        pragma GCC diagnostic ignored "-Wignored-attributes"
+#        define MAC_HAS_LOG 1
+#        include <os/log.h>
+#        pragma GCC diagnostic pop
+#    endif
 #endif
 
 #if defined(ANGLE_PLATFORM_WINDOWS)
@@ -275,7 +279,7 @@
         // Note: we also log to stdout/stderr below.
 #endif
 
-#if defined(ANGLE_PLATFORM_APPLE)
+#if MAC_HAS_LOG
         os_log_type_t apple_log_type = OS_LOG_TYPE_DEFAULT;
         switch (severity)
         {
