--- build/config/compiler/BUILD.gn	2026-07-21 17:42:35.000000000 +0800
+++ build/config/compiler/BUILD.gn	2026-07-21 19:17:52.000000000 +0800
@@ -528,6 +528,10 @@
       cflags_cc += [ "-D_GLIBCXX_USE_CXX11_ABI=0" ]
     }
 
+    if (!is_clang) {
+      ldflags += [ "-latomic" ]
+    }
+
     # dsymutil is not available in the system, on bots, for rustc to call. Our
     # linker_driver.py script runs dsymutil itself, which is set to be the
     # linker for Rust targets as well.
@@ -1080,15 +1084,6 @@
     ldflags += [ "-Wl,--icf=all" ]
   }
 
-  # Note: `use_lld=false` on Apple platforms is experimental and may not
-  # run properly on bots due to toolchain limitations. It is mainly
-  # intended to unblock local builds for debugging.
-  if (is_apple && !use_lld) {
-    if (!use_icf) {
-      ldflags += [ "-Wl,-no_deduplicate" ]
-    }
-  }
-
   if (is_chromeos_device && current_toolchain == default_toolchain) {
     # Flags added automatically by ChromeOS's custom toolchain that are
     # not already added elsewhere in the configuration. Add these here,
