--- src/libANGLE/renderer/gl/cgl/IOSurfaceSurfaceCGL.cpp.orig
+++ src/libANGLE/renderer/gl/cgl/IOSurfaceSurfaceCGL.cpp
@@ -7,6 +7,12 @@
 // PBufferSurfaceCGL.cpp: an implementation of PBuffers created from IOSurfaces using
 //                        EGL_ANGLE_iosurface_client_buffer
 
+#define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
+
+#define __gl_h_
+#define __gltypes_h_
+#define __glext_h_
+
 #include "libANGLE/renderer/gl/cgl/IOSurfaceSurfaceCGL.h"
 
 #include <IOSurface/IOSurface.h>
