--- src/libANGLE/renderer/gl/cgl/IOSurfaceSurfaceCGL.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/renderer/gl/cgl/IOSurfaceSurfaceCGL.cpp	2025-11-04 09:28:39.000000000 +0800
@@ -7,6 +7,12 @@
 // PBufferSurfaceCGL.cpp: an implementation of PBuffers created from IOSurfaces using
 //                        EGL_ANGLE_iosurface_client_buffer
 
+#define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
+
+#define __gl_h_
+#define __gltypes_h_
+#define __glext_h_
+
 #ifdef UNSAFE_BUFFERS_BUILD
 #    pragma allow_unsafe_buffers
 #endif
