--- src/libANGLE/renderer/gl/cgl/WindowSurfaceCGL.mm	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/renderer/gl/cgl/WindowSurfaceCGL.mm	2025-11-04 08:18:33.000000000 +0800
@@ -6,6 +6,12 @@
 
 // WindowSurfaceCGL.cpp: CGL implementation of egl::Surface for windows
 
+#define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
+
+#define __gl_h_
+#define __gltypes_h_
+#define __glext_h_
+
 #ifdef UNSAFE_BUFFERS_BUILD
 #    pragma allow_unsafe_buffers
 #endif
@@ -16,6 +22,8 @@
 #import <OpenGL/OpenGL.h>
 #import <QuartzCore/QuartzCore.h>
 
+#include <AvailabilityMacros.h>
+
 #import "common/debug.h"
 #import "common/gl/cgl/FunctionsCGL.h"
 #import "libANGLE/Context.h"
@@ -68,8 +76,11 @@
 - (CGLPixelFormatObj)copyCGLPixelFormatForDisplayMask:(uint32_t)mask
 {
     CGLPixelFormatAttribute attribs[] = {
-        kCGLPFADisplayMask, static_cast<CGLPixelFormatAttribute>(mask), kCGLPFAOpenGLProfile,
+        kCGLPFADisplayMask, static_cast<CGLPixelFormatAttribute>(mask),
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
+        kCGLPFAOpenGLProfile,
         static_cast<CGLPixelFormatAttribute>(kCGLOGLPVersion_3_2_Core),
+#endif
         static_cast<CGLPixelFormatAttribute>(0)};
 
     CGLPixelFormatObj pixelFormat = nullptr;
@@ -160,7 +171,11 @@
     : SurfaceGL(state),
       mSwapLayer(nil),
       mCurrentSwapId(0),
+#ifdef __clang__
       mLayer((__bridge CALayer *)layer),
+#else
+      mLayer((CALayer *)layer),
+#endif
       mContext(context),
       mFunctions(renderer->getFunctions()),
       mStateManager(renderer->getStateManager()),
@@ -231,7 +246,11 @@
                                                       withContext:mContext
                                                     withFunctions:mFunctions];
     [mLayer addSublayer:mSwapLayer];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
     [mSwapLayer setContentsScale:[mLayer contentsScale]];
+#else
+    [mSwapLayer setContentsScale:1.0];
+#endif
 
     mFunctions->genRenderbuffers(1, &mDSRenderbuffer);
     mStateManager->bindRenderbuffer(GL_RENDERBUFFER, mDSRenderbuffer);
@@ -334,7 +353,11 @@
                                          EGLint *height) const
 {
     CGRect frame  = [mLayer frame];
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
     CGFloat scale = [mLayer contentsScale];
+#else
+    CGFloat scale = 1.0;
+#endif
     if (width != nullptr)
     {
         *width = static_cast<EGLint>(CGRectGetWidth(frame) * scale);
