--- src/libANGLE/renderer/gl/cgl/DisplayCGL.mm	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/renderer/gl/cgl/DisplayCGL.mm	2025-11-04 09:26:45.000000000 +0800
@@ -6,6 +6,12 @@
 
 // DisplayCGL.mm: CGL implementation of egl::Display
 
+#define GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED
+
+#define __gl_h_
+#define __gltypes_h_
+#define __glext_h_
+
 #import "libANGLE/renderer/gl/cgl/DisplayCGL.h"
 
 #import <Cocoa/Cocoa.h>
@@ -179,8 +185,10 @@
     {
         // TODO(cwallez) investigate which pixel format we want
         std::vector<CGLPixelFormatAttribute> attribs;
+#if defined(MAC_OS_X_VERSION_10_7) && MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_7
         attribs.push_back(kCGLPFAOpenGLProfile);
         attribs.push_back(static_cast<CGLPixelFormatAttribute>(kCGLOGLPVersion_3_2_Core));
+#endif
         attribs.push_back(kCGLPFAAllowOfflineRenderers);
         attribs.push_back(static_cast<CGLPixelFormatAttribute>(0));
         GLint nVirtualScreens = 0;
