--- build/build_config.h.orig	2025-09-27 12:42:34.000000000 +0800
+++ build/build_config.h	2025-10-27 08:30:38.000000000 +0800
@@ -306,7 +306,12 @@
 #define ARCH_CPU_S390 1
 #define ARCH_CPU_31_BITS 1
 #define ARCH_CPU_BIG_ENDIAN 1
-#elif (defined(__PPC64__) || defined(__PPC__)) && defined(__BIG_ENDIAN__)
+#elif defined(__ppc__)
+#define ARCH_CPU_PPC_FAMILY 1
+#define ARCH_CPU_PPC 1
+#define ARCH_CPU_32_BITS 1
+#define ARCH_CPU_BIG_ENDIAN 1
+#elif defined(__ppc64__) || ((defined(__PPC64__) || defined(__PPC__)) && defined(__BIG_ENDIAN__))
 #define ARCH_CPU_PPC64_FAMILY 1
 #define ARCH_CPU_PPC64 1
 #define ARCH_CPU_64_BITS 1
