--- src/common/apple/SoftLinking.h.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/common/apple/SoftLinking.h	2025-11-04 01:38:59.000000000 +0800
@@ -15,9 +15,9 @@
 
 #    include "common/debug.h"
 
-#    import <dispatch/dispatch.h>
-#    import <dlfcn.h>
-#    import <objc/runtime.h>
+#    include <dlfcn.h>
+#    include <objc/runtime.h>
+#    include <pthread.h>
 
 #    define RELEASE_ASSERT(expression, message)                                               \
         (expression                                                                           \
@@ -36,16 +36,18 @@
 #    define SOFT_LINK_FRAMEWORK_HEADER(framework) extern void *framework##Library();
 
 #    define SOFT_LINK_FRAMEWORK_SOURCE(framework)                                               \
+        static void *framework##LibraryHandle = NULL;                                           \
+        static void init##framework##Library(void)                                              \
+        {                                                                                       \
+            framework##LibraryHandle = dlopen(                                                  \
+                "/System/Library/Frameworks/" #framework ".framework/" #framework, RTLD_NOW);   \
+            RELEASE_ASSERT(framework##LibraryHandle, "Unable to load " #framework ".framework");\
+        }                                                                                       \
         void *framework##Library()                                                              \
         {                                                                                       \
-            static dispatch_once_t once   = 0;                                                  \
-            static void *frameworkLibrary = NULL;                                               \
-            dispatch_once(&once, ^{                                                             \
-              frameworkLibrary = dlopen(                                                        \
-                  "/System/Library/Frameworks/" #framework ".framework/" #framework, RTLD_NOW); \
-              RELEASE_ASSERT(frameworkLibrary, "Unable to load " #framework ".framework");      \
-            });                                                                                 \
-            return frameworkLibrary;                                                            \
+            static pthread_once_t once = PTHREAD_ONCE_INIT;                                     \
+            pthread_once(&once, init##framework##Library);                                      \
+            return framework##LibraryHandle;                                                    \
         }
 
 #    define SOFT_LINK_FUNCTION_HEADER(framework, functionName, resultType, parameterDeclarations, \
@@ -62,15 +64,17 @@
 
 #    define SOFT_LINK_FUNCTION_SOURCE(framework, functionName, resultType, parameterDeclarations,  \
                                       parameterNames)                                              \
+        static void init##framework##functionName##Ptr(void)                                       \
+        {                                                                                          \
+            softLink##framework##functionName =                                                    \
+                (resultType(*)parameterDeclarations)dlsym(framework##Library(), #functionName);    \
+        }                                                                                          \
         resultType(*softLink##framework##functionName) parameterDeclarations =                     \
             init##framework##functionName;                                                         \
         resultType init##framework##functionName parameterDeclarations                             \
         {                                                                                          \
-            static dispatch_once_t once;                                                           \
-            dispatch_once(&once, ^{                                                                \
-              softLink##framework##functionName =                                                  \
-                  (resultType(*) parameterDeclarations)dlsym(framework##Library(), #functionName); \
-            });                                                                                    \
+            static pthread_once_t once = PTHREAD_ONCE_INIT;                                        \
+            pthread_once(&once, init##framework##functionName##Ptr);                               \
             return softLink##framework##functionName parameterNames;                               \
         }
 
@@ -80,23 +84,25 @@
         className *alloc##className##Instance(); \
         inline className *alloc##className##Instance() { return [get##className##Class() alloc]; }
 
-#    define SOFT_LINK_CLASS(framework, className)                                       \
+# define SOFT_LINK_CLASS(framework, className)                                          \
         @class className;                                                               \
+        static Class class##className;                                                  \
+        static void init##className##Ptr(void)                                          \
+        {                                                                               \
+            framework##Library();                                                       \
+            class##className = objc_getClass(#className);                               \
+            RELEASE_ASSERT(class##className, "objc_getClass failed for " #className);   \
+            get##className##Class = className##Function;                                \
+        }                                                                               \
         static Class init##className();                                                 \
         Class (*get##className##Class)() = init##className;                             \
-        static Class class##className;                                                  \
                                                                                         \
         static Class className##Function() { return class##className; }                 \
                                                                                         \
         static Class init##className()                                                  \
         {                                                                               \
-            static dispatch_once_t once;                                                \
-            dispatch_once(&once, ^{                                                     \
-              framework##Library();                                                     \
-              class##className = objc_getClass(#className);                             \
-              RELEASE_ASSERT(class##className, "objc_getClass failed for " #className); \
-              get##className##Class = className##Function;                              \
-            });                                                                         \
+            static pthread_once_t once = PTHREAD_ONCE_INIT;                             \
+            pthread_once(&once, init##className##Ptr);                                  \
             return class##className;                                                    \
         }                                                                               \
         _Pragma("clang diagnostic push")                                                \
