--- src/common/gl/cgl/FunctionsCGL.h.orig
+++ src/common/gl/cgl/FunctionsCGL.h
@@ -10,6 +10,14 @@
 #define CGL_FUNCTIONS_H_
 
 #include <OpenGL/OpenGL.h>
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
+#    include <IOSurface/IOSurface.h>
+#else
+// There is no IOSurface.framework before Mac OS X 10.6; only the IOSurfaceRef type is needed here.
+struct __IOSurface;
+typedef __IOSurface *IOSurfaceRef;
+#endif
 
 #include "common/apple/SoftLinking.h"
 
