--- build/config/compiler/BUILD.gn.orig	2025-11-04 19:03:50.000000000 +0800
+++ build/config/compiler/BUILD.gn	2025-11-04 19:03:08.000000000 +0800
@@ -544,6 +544,10 @@
       cflags_cc += [ "-D_GLIBCXX_USE_CXX11_ABI=0" ]
     }
 
+    if (!is_clang) {
+      ldflags += [ "-latomic" ]
+    }
+
     # dsymutil is not available in the system, on bots, for rustc to call. Our
     # linker_driver.py script runs dsymutil itself, which is set to be the
     # linker for Rust targets as well.
