--- scripts/roll_aosp.sh.orig	2025-09-30 05:58:34.000000000 +0800
+++ scripts/roll_aosp.sh	2025-11-04 01:56:03.000000000 +0800
@@ -111,7 +111,7 @@
     # variable is set to {rolling_to} git hash, and that can be used by below
     # script commit_id.py as the ANGLE_COMMIT_HASH written to the angle_commit.h.
     # See b/348044346.
-    python3 src/commit_id.py \
+    @PYTHON@ src/commit_id.py \
         gen \
         angle_commit.h
 }
