--- src/libANGLE/renderer/gl/BUILD.gn.orig
+++ src/libANGLE/renderer/gl/BUILD.gn
@@ -65,10 +65,9 @@
     deps += [ "$angle_root/src/common/linux:angle_dma_buf" ]
   }
   if (is_apple) {
-    frameworks = [
-      "IOSurface.framework",
-      "QuartzCore.framework",
-    ]
+    # There is no IOSurface.framework before Mac OS X 10.6; the IOSurface code is compiled out
+    # there (see IOSurfaceSurfaceCGL.cpp).
+    frameworks = [ "QuartzCore.framework" ]
     if (angle_enable_cgl) {
       frameworks += [ "OpenGL.framework" ]
     }
