--- src/libGLESv2/global_state.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libGLESv2/global_state.cpp	2025-11-04 06:18:20.000000000 +0800
@@ -18,7 +18,7 @@
 
 #include <atomic>
 #if defined(ANGLE_PLATFORM_APPLE)
-#    include <dispatch/dispatch.h>
+#    include <pthread.h>
 #endif
 namespace egl
 {
@@ -91,15 +91,22 @@
 // local storage instead.
 // https://bugs.webkit.org/show_bug.cgi?id=228240
 
+namespace
+{
+angle::TLSIndex g_CurrentThreadIndex = TLS_INVALID_INDEX;
+pthread_once_t g_CurrentThreadIndexOnce = PTHREAD_ONCE_INIT;
+
+void InitCurrentThreadTLSIndex()
+{
+    ASSERT(g_CurrentThreadIndex == TLS_INVALID_INDEX);
+    g_CurrentThreadIndex = angle::CreateTLSIndex(nullptr);
+}
+} // anonymous namespace
+
 static angle::TLSIndex GetCurrentThreadTLSIndex()
 {
-    static angle::TLSIndex CurrentThreadIndex = TLS_INVALID_INDEX;
-    static dispatch_once_t once;
-    dispatch_once(&once, ^{
-      ASSERT(CurrentThreadIndex == TLS_INVALID_INDEX);
-      CurrentThreadIndex = angle::CreateTLSIndex(nullptr);
-    });
-    return CurrentThreadIndex;
+    pthread_once(&g_CurrentThreadIndexOnce, InitCurrentThreadTLSIndex);
+    return g_CurrentThreadIndex;
 }
 Thread *GetCurrentThreadTLS()
 {
