--- src/libANGLE/queryutils.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/queryutils.cpp	2025-11-04 09:24:32.000000000 +0800
@@ -1261,7 +1261,7 @@
             break;
 
         case GL_FRAMEBUFFER_ATTACHMENT_OBJECT_NAME:
-            *params = attachmentObject->id();
+            *params = attachmentObject->getId();
             break;
 
         case GL_FRAMEBUFFER_ATTACHMENT_TEXTURE_LEVEL:

--- src/libANGLE/Framebuffer.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/Framebuffer.cpp	2025-11-04 07:34:59.000000000 +0800
@@ -1068,7 +1068,7 @@
                                            GLenum matchType,
                                            GLuint matchId)
 {
-    if (attachment->isAttached() && attachment->type() == matchType && attachment->id() == matchId)
+    if (attachment->isAttached() && attachment->type() == matchType && attachment->getId() == matchId)
     {
         const State &contextState = context->getState();
         if (contextState.getPixelLocalStorageActivePlanes() != 0 &&

--- src/libANGLE/FramebufferAttachment.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/FramebufferAttachment.cpp	2025-11-04 07:35:12.000000000 +0800
@@ -180,7 +180,7 @@
     return getFormat().info->colorEncoding;
 }
 
-GLuint FramebufferAttachment::id() const
+GLuint FramebufferAttachment::getId() const
 {
     return mResource->getId();
 }

--- src/libANGLE/FramebufferAttachment.h.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/FramebufferAttachment.h	2025-11-04 07:29:31.000000000 +0800
@@ -91,7 +91,7 @@
 
     bool isTextureWithId(TextureID textureId) const
     {
-        return mType == GL_TEXTURE && id() == textureId.value;
+        return mType == GL_TEXTURE && getId() == textureId.value;
     }
     bool isExternalTexture() const
     {
@@ -99,11 +99,11 @@
     }
     bool isRenderbufferWithId(GLuint renderbufferId) const
     {
-        return mType == GL_RENDERBUFFER && id() == renderbufferId;
+        return mType == GL_RENDERBUFFER && getId() == renderbufferId;
     }
 
     GLenum getBinding() const { return mTarget.binding(); }
-    GLuint id() const;
+    GLuint getId() const;
 
     // These methods are only legal to call on Texture attachments
     const ImageIndex &getTextureImageIndex() const;

--- src/libANGLE/capture/serialize.cpp.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/libANGLE/capture/serialize.cpp	2025-11-04 11:43:42.000000000 +0800
@@ -280,7 +280,7 @@
     if (framebufferAttachment.type() == GL_TEXTURE ||
         framebufferAttachment.type() == GL_RENDERBUFFER)
     {
-        json->addScalar("AttachedResourceID", framebufferAttachment.id());
+        json->addScalar("AttachedResourceID", framebufferAttachment.getId());
     }
     json->addCString(
         "Type", gl::GLenumToString(gl::GLESEnum::ObjectIdentifier, framebufferAttachment.type()));
