--- src/gpu_info_util/SystemInfo_macos.mm.orig	2025-09-30 05:58:34.000000000 +0800
+++ src/gpu_info_util/SystemInfo_macos.mm	2025-10-27 07:49:26.000000000 +0800
@@ -12,9 +12,18 @@
 
 #include "gpu_info_util/SystemInfo_internal.h"
 
+#include <AvailabilityMacros.h>
+
 #import <Cocoa/Cocoa.h>
 #import <IOKit/IOKitLib.h>
-#import <Metal/Metal.h>
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
+#import    <Metal/Metal.h>
+#endif
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 120000
+#    define kIOMainPortDefault kIOMasterPortDefault
+#endif
 
 #include "common/apple_platform_utils.h"
 
@@ -196,6 +205,7 @@
 // Used with permission.
 uint64_t GetGpuIDFromDisplayID(uint32_t displayID)
 {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101100
     // First attempt to use query the registryID from a Metal device before falling back to CGL.
     // This avoids loading the OpenGL framework when possible.
     id<MTLDevice> device = CGDirectDisplayCopyCurrentMetalDevice(displayID);
@@ -205,6 +215,7 @@
         [device release];
         return registryId;
     }
+#endif
 #if ANGLE_ENABLE_CGL
     return GetGpuIDFromOpenGLDisplayMask(CGDisplayIDToOpenGLDisplayMask(displayID));
 #else
