--- ffc.h	2026-06-03 22:09:52.000000000 +0800
+++ ffc.h	2026-06-22 07:24:25.000000000 +0800
@@ -883,8 +883,8 @@
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wfloat-equal"
 #elif defined(__GNUC__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wfloat-equal"
+// #pragma GCC diagnostic push
+// #pragma GCC diagnostic ignored "-Wfloat-equal"
 #endif
   return (fmini + 1.0f == 1.0f - fmini);
 #ifdef FFC_VISUAL_STUDIO
@@ -892,7 +892,7 @@
 #elif defined(__clang__)
 #pragma clang diagnostic pop
 #elif defined(__GNUC__)
-#pragma GCC diagnostic pop
+// #pragma GCC diagnostic pop
 #endif
 }
 
