--- GNUmakefile.orig	2025-10-31 21:40:42.000000000 +0800
+++ GNUmakefile	2025-11-02 16:20:31.000000000 +0800
@@ -1,12 +1,12 @@
 # Linux makefile
 SHELL = /bin/sh
 
-PREFIX = /usr
-CC = cc
+PREFIX ?= /usr
+CC ?= cc
 
-CFLAGS = -ansi -std=c89 -O2 -Wall -Wpedantic -Wextra \
+CFLAGS += -ansi -std=c89 -O2 -Wall -Wpedantic -Wextra \
 		-I./include -D_POSIX_C_SOURCE=200809L -D_GNU_SOURCE
-LDFLAGS = -s -L./lib -ltls -lssl -lcrypto -lm -lpthread -ldl
+LDFLAGS += -s -L./lib -ltls -lssl -lcrypto -lm -lpthread -ldl
 
 FLAGS = -DENABLE_SECCOMP_FILTER #-DENABLE_PKEY
 
