--- lib/expat_external.h.orig	2026-09-22 19:07:50.000000000 +0800
+++ lib/expat_external.h	2026-09-27 10:58:20.000000000 +0800
@@ -128,9 +128,13 @@
 
 /* Marks a function that Expat still provides but that callers should move off
    of. */
-#  if defined(__clang__) || defined(__GNUC__)
+#  if defined(__clang__)                                                       \
+      || (defined(__GNUC__)                                                    \
+          && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)))
 #    define XML_ATTR_DEPRECATED(message)                                       \
       __attribute__((__deprecated__(message)))
+#  elif defined(__GNUC__)
+#    define XML_ATTR_DEPRECATED(message) __attribute__((__deprecated__))
 #  elif defined(_MSC_VER)
 #    define XML_ATTR_DEPRECATED(message) __declspec(deprecated(message))
 #  else
