From 1eb45b27f67853101a6ff27a008923c3ddc8fe39 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 29 Jan 2025 08:49:25 +0800
Subject: [PATCH] CMakeLists: use correct flags with native build on
 powerpc

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 039262a2b..b766111f2 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -214,7 +214,7 @@ endif()
 
 if(NOT APPLE)
   if(NATIVE_BUILD)
-    if(CMAKE_SYSTEM_PROCESSOR STREQUAL ppc64le)
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "ppc*|power*")
       add_compile_options(-mcpu=native -mtune=native)
     else()
       add_compile_options(-march=native -mtune=native)
