--- CMakeLists.txt	2026-02-28 15:07:47.000000000 +0800
+++ CMakeLists.txt	2026-03-29 12:15:50.000000000 +0800
@@ -260,7 +260,7 @@
 # Includes
 target_include_directories(nmail PRIVATE ${LIBETPAN_INCLUDE_DIR} ${XAPIAN_INCLUDE_DIR}
                            ${MAGIC_HEADERS} ${CYRUS_SASL_INCLUDE_DIR} ${CURSES_INCLUDE_DIR} ${OPENSSL_INCLUDE_DIR}
-                           "ext/sqlite_modern_cpp/hdr" "ext/cereal/include" ${LIBUUID_HEADERS}
+                           "ext/sqlite_modern_cpp/hdr" ${LIBUUID_HEADERS}
                            "ext/cyrus-imap/lib")
 
 # Linking
