--- CMakeLists.txt	2026-02-28 15:07:47.000000000 +0800
+++ CMakeLists.txt	2026-06-28 03:14:11.000000000 +0800
@@ -236,8 +236,7 @@
                            ${LIBETPAN_INCLUDE_DIR} ${XAPIAN_INCLUDE_DIR} ${MAGIC_HEADERS}
                            ${CYRUS_SASL_INCLUDE_DIR} ${CURSES_INCLUDE_DIR} ${OPENSSL_INCLUDE_DIR}
                            ${LIBUUID_HEADERS}
-                           "ext/cereal/include" "ext/cyrus-imap/lib" "ext/sqlite_modern_cpp/hdr"
-                           "ext/utfcpp/source")
+                           "@PREFIX@/include/utf8cpp" "ext/cyrus-imap/lib" "ext/sqlite_modern_cpp/hdr")
 
 # Linking
 target_link_libraries(nmail PUBLIC
