--- CMakeLists.txt	2025-01-25 16:52:16.000000000 +0800
+++ CMakeLists.txt	2025-03-24 00:13:44.000000000 +0800
@@ -104,15 +104,6 @@
 # Platform specifics
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   add_definitions(-D_XOPEN_SOURCE_EXTENDED)
-  execute_process(COMMAND sh "-c"
-                  "command -v brew &> /dev/null && brew --prefix ncurses 2> /dev/null | tr -d '\n'"
-                  OUTPUT_VARIABLE CMAKE_PREFIX_PATH)
-  if (EXISTS "${CMAKE_PREFIX_PATH}")
-    message(STATUS "Ncurses cmake prefix '${CMAKE_PREFIX_PATH}' (detected).")
-  else()
-    set(CMAKE_PREFIX_PATH /opt/local)
-    message(STATUS "Ncurses cmake prefix '${CMAKE_PREFIX_PATH}' (default).")
-  endif()
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Android")
   add_compile_definitions(_XOPEN_SOURCE_EXTENDED)
 endif()
