--- CMakeLists.txt	2026-02-28 15:07:47.000000000 +0800
+++ CMakeLists.txt	2026-03-29 12:05:10.000000000 +0800
@@ -106,32 +106,6 @@
 # Platform specifics
 if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   add_definitions(-D_XOPEN_SOURCE_EXTENDED)
-
-# ncurses
-  execute_process(COMMAND sh "-c"
-                  "command -v brew &> /dev/null && brew --prefix ncurses 2> /dev/null | tr -d '\n'"
-                  OUTPUT_VARIABLE NCURSES_ROOT_DIR)
-  if (EXISTS "${NCURSES_ROOT_DIR}")
-    message(STATUS "Ncurses cmake prefix '${NCURSES_ROOT_DIR}' (homebrew).")
-    list(APPEND CMAKE_PREFIX_PATH ${NCURSES_ROOT_DIR})
-  elseif (EXISTS "/opt/local/bin/port")
-    set(NCURSES_ROOT_DIR /opt/local)
-    message(STATUS "Ncurses cmake prefix '${NCURSES_ROOT_DIR}' (macports).")
-    list(APPEND CMAKE_PREFIX_PATH ${NCURSES_ROOT_DIR})
-  endif()
-
-  # sqlite
-  execute_process(COMMAND sh "-c"
-                  "command -v brew &> /dev/null && brew --prefix sqlite 2> /dev/null | tr -d '\n'"
-                  OUTPUT_VARIABLE SQLITE_ROOT_DIR)
-  if (EXISTS "${SQLITE_ROOT_DIR}")
-    message(STATUS "SQLite cmake prefix '${SQLITE_ROOT_DIR}' (homebrew).")
-    list(APPEND CMAKE_PREFIX_PATH ${SQLITE_ROOT_DIR})
-  elseif (EXISTS "/opt/local/bin/port")
-    set(SQLITE_ROOT_DIR /opt/local)
-    message(STATUS "SQLite cmake prefix '${SQLITE_ROOT_DIR}' (macports).")
-    list(APPEND CMAKE_PREFIX_PATH ${SQLITE_ROOT_DIR})
-  endif()
 elseif (${CMAKE_SYSTEM_NAME} MATCHES "Android")
   add_compile_definitions(_XOPEN_SOURCE_EXTENDED)
   if (DEFINED ENV{PREFIX} AND "$ENV{PREFIX}" MATCHES "^/data/data/com\\.termux/files/usr$")
