--- meson.build.orig	2026-05-15 21:09:38.000000000 +0800
+++ meson.build	2026-06-02 03:00:12.000000000 +0800
@@ -218,7 +218,7 @@
 
 # Python plugins
 if get_option('python-plugins').enabled()
-  python_dep = dependency('python3-embed', required: false)
+  python_dep = dependency('python-@PY_V@-embed', required: false)
   if not python_dep.found()
     python_dep = dependency('python-embed', required: true)
   else
