--- src/regex_vm.hh
+++ src/regex_vm.hh	2026-04-23 15:03:46.000000000 +0800
@@ -105,14 +105,14 @@
         struct Literal
         {
             uint32_t codepoint : 24;
-            bool ignore_case : 1;
+            uint8_t ignore_case : 1;
         } literal;
         struct CharRange
         {
             uint8_t min;
             uint8_t max;
-            bool ignore_case : 1;
-            bool negative;
+            uint8_t ignore_case : 1;
+            uint8_t negative;
         } range;
         CharacterType character_type;
         int16_t character_class_index;
@@ -121,17 +121,17 @@
         struct Split
         {
             int16_t offset;
-            bool prioritize_parent : 1;
+            uint8_t prioritize_parent : 1;
         } split;
-        bool line_start;
-        bool subject_begin;
-        bool word_boundary_positive;
+        uint8_t line_start;
+        uint8_t subject_begin;
+        uint8_t word_boundary_positive;
         struct Lookaround
         {
             int16_t index;
-            bool ahead : 1;
-            bool positive : 1;
-            bool ignore_case : 1;
+            uint8_t ahead : 1;
+            uint8_t positive : 1;
+            uint8_t ignore_case : 1;
         } lookaround;
     };
     static_assert(sizeof(Param) == 4);
@@ -142,9 +142,7 @@
         mutable uint16_t last_step; // mutable as used during execution
         Param param;
     };
-#ifndef __ppc__
     static_assert(sizeof(Instruction) == 8);
-#endif
 
     explicit operator bool() const { return not instructions.empty(); }
 
@@ -167,7 +165,7 @@
         using OffsetLimits = std::numeric_limits<uint8_t>;
         char start_byte = 0;
         uint8_t offset = 0;
-        bool map[count];
+        uint8_t map[count];
     };
 
     UniquePtr<StartDesc> forward_start_desc;
