diff --git a/uppsrc/plugin/tif/lib/tif_config.h b/uppsrc/plugin/tif/lib/tif_config.h
index 8734ffa9d..3690049e5 100644
--- a/uppsrc/plugin/tif/lib/tif_config.h
+++ b/uppsrc/plugin/tif/lib/tif_config.h
@@ -144,6 +144,8 @@
 # if defined __BIG_ENDIAN__
 #  define WORDS_BIGENDIAN 1
 # endif
+#elif defined(CPU_BIG_ENDIAN) || defined(__BIG_ENDIAN__) || defined(__ppc__) || defined(__ppc64__)
+# define WORDS_BIGENDIAN 1
 #else
 # ifndef WORDS_BIGENDIAN
 #  undef WORDS_BIGENDIAN
