--- setup.py	2024-08-31 20:12:24.000000000 +0800
+++ setup.py	2025-07-10 05:09:58.000000000 +0800
@@ -63,7 +63,7 @@
 
     # Nothing wrong with that if you know what you are doing
     # (which Cython does)
-    compile_args.append('-Wno-implicit-fallthrough')
+    # compile_args.append('-Wno-implicit-fallthrough')
 
     # Due to platform specific conditions, these are unavoidable
     compile_args.append('-Wno-unused-parameter')
