--- nng/src/platform/posix/posix_tcpdial.c
+++ nng/src/platform/posix/posix_tcpdial.c	2025-08-30 08:04:22.000000000 +0800
@@ -155,10 +155,14 @@
 		(void) setsockopt(nni_posix_pfd_fd(pfd), IPPROTO_TCP,
 		    TCP_KEEPALIVE, &keepidle, sizeof(int));
 #endif
+#ifdef TCP_KEEPINTVL
 		(void) setsockopt(nni_posix_pfd_fd(pfd), IPPROTO_TCP,
 		    TCP_KEEPINTVL, &keepintvl, sizeof(int));
+#endif
+#ifdef TCP_KEEPCNT
 		(void) setsockopt(nni_posix_pfd_fd(pfd), IPPROTO_TCP,
 		    TCP_KEEPCNT, &keepcnt, sizeof(int));
+#endif
 		(void) setsockopt(nni_posix_pfd_fd(pfd), SOL_SOCKET,
 		    SO_SNDTIMEO, (char *) &sendtimeo, sizeof(struct timeval));
 		(void) setsockopt(nni_posix_pfd_fd(pfd), SOL_SOCKET,
