From 62c04eb981f94d661a1e34f1c5e0f13cd3438aeb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 3 Oct 2025 02:35:27 +0800
Subject: [PATCH] Support external lunasvg

---
 CMakeLists.txt         |  1 +
 library/CMakeLists.txt | 13 ++++++++++---
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index a73152a..3c1b4f8 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -42,6 +42,7 @@ option(USE_SYSTEM_TINYXML2 "" ${USE_SHARED_LIB})
 option(USE_SYSTEM_SDL2 "" ${USE_SHARED_LIB})
 option(USE_SYSTEM_MONGOOSE "" ${USE_SHARED_LIB})
 option(USE_SYSTEM_GLFW "Not recommended because we used a modified GLFW" OFF)
+option(USE_SYSTEM_LUNASVG "" OFF)
 
 option(DISABLE_OPENCC "Disable Chinese Simplified and Chinese Traditional Conversion" OFF)
 option(DISABLE_WEBP "Disable webp" OFF)
diff --git library/CMakeLists.txt library/CMakeLists.txt
index 3dd570b..af21513 100644
--- library/CMakeLists.txt
+++ library/CMakeLists.txt
@@ -77,9 +77,16 @@ else ()
 endif ()
 
 # add lunasvg
-set(BUILD_SHARED_LIBS OFF)
-set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
-add_subdirectory(lunasvg EXCLUDE_FROM_ALL)
+if (USE_SYSTEM_LUNASVG)
+    find_package(lunasvg REQUIRED)
+    if (lunasvg_FOUND)
+        message(STATUS "Found lunasvg: ${lunasvg_CONFIG} (found version ${lunasvg_VERSION})")
+    endif ()
+else ()
+    set(BUILD_SHARED_LIBS OFF)
+    set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
+    add_subdirectory(lunasvg EXCLUDE_FROM_ALL)
+endif ()
 
 # add mongoose
 if (USE_SYSTEM_MONGOOSE)
