From 79e892df6f17b6f0515ef09785d750d19e95fe51 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 3 Oct 2025 05:29:09 +0800
Subject: [PATCH] qr_image.hpp: add a missing header

---
 wiliwili/include/view/qr_image.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git wiliwili/include/view/qr_image.hpp wiliwili/include/view/qr_image.hpp
index 6d15061..9496f55 100644
--- wiliwili/include/view/qr_image.hpp
+++ wiliwili/include/view/qr_image.hpp
@@ -8,6 +8,7 @@
 #include <qrcodegen.hpp>
 #include <pystring.h>
 #include <cstdlib>
+#include <sstream>
 #include <fmt/format.h>
 
 #include "view/svg_image.hpp"
-- 
2.51.0

