From 34dee85cc0c5fe848ed70f5e0cba1f32e849e069 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 3 Oct 2025 09:16:37 +0800
Subject: [PATCH] Allow customized builds on macOS

---
 CMakeLists.txt    | 1 +
 cmake/extra.cmake | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 3c1b4f8..df31036 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -19,6 +19,7 @@ cmake_dependent_option(MAC_IntelChip "Build for x86_64 (10.15)" OFF "APPLE;NOT P
 cmake_dependent_option(MAC_AppleSilicon "Build for arm64 (11.0)" OFF "APPLE;NOT PLATFORM_IOS" OFF)
 cmake_dependent_option(MAC_Universal "Support x86_64;arm64 at the same time" OFF "APPLE;NOT PLATFORM_IOS" OFF)
 cmake_dependent_option(MAC_DOWNLOAD_DYLIB "Using precompiled dylib instead of system provided" OFF "APPLE;NOT PLATFORM_IOS" OFF)
+cmake_dependent_option(MAC_CUSTOM_BUILD "Allow specifying build config for macOS" OFF "APPLE;NOT PLATFORM_IOS" OFF)
 
 # Linux
 # Using CMAKE_INSTALL_PREFIX to specify the location where the application is installed.
diff --git cmake/extra.cmake cmake/extra.cmake
index e31f12f..f88a1b4 100644
--- cmake/extra.cmake
+++ cmake/extra.cmake
@@ -50,7 +50,7 @@ if (APPLE AND PLATFORM_DESKTOP)
         message(STATUS "build for modern system, using curl provided by system")
         set(USE_SYSTEM_CURL ON)
         set(CPR_FORCE_DARWINSSL_BACKEND ON)
-    else ()
+    elseif (NOT MAC_CUSTOM_BUILD)
         message(STATUS "build for old system, using boost filesystem and compile curl")
         set(USE_BOOST_FILESYSTEM ON)
         set(USE_SYSTEM_CURL OFF)
