From 3b4cb74425a7403d994bf2d68f88e91c3d92b369 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 3 Oct 2025 10:40:15 +0800
Subject: [PATCH] CMakeLists.txt: use correct frameworks on macOS

---
 library/CMakeLists.txt | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git library/CMakeLists.txt library/CMakeLists.txt
index 060292a9..c0c46c47 100644
--- library/borealis/library/CMakeLists.txt
+++ library/borealis/library/CMakeLists.txt
@@ -108,8 +108,16 @@ if (PLATFORM_DESKTOP)
         endif ()
         list(APPEND BRLS_PLATFORM_LIBS wlanapi iphlpapi ws2_32)
     elseif (APPLE)
-        list(APPEND BRLS_PLATFORM_LIBS "-framework CoreWLAN" "-framework SystemConfiguration" "-framework Cocoa")
-        list(APPEND BRLS_PLATFORM_LIBS "-framework IOKit" "-framework CoreFoundation" "-framework CoreGraphics")
+        if (CMAKE_OSX_DEPLOYMENT_TARGET VERSION_GREATER_EQUAL "10.6")
+            list(APPEND BRLS_PLATFORM_LIBS "-framework CoreWLAN")
+        endif ()
+        if (CMAKE_OSX_DEPLOYMENT_TARGET VERSION_GREATER_EQUAL "10.7")
+            list(APPEND BRLS_PLATFORM_LIBS "-framework CoreGraphics")
+        else ()
+            list(APPEND BRLS_PLATFORM_LIBS "-framework ApplicationServices")
+        endif ()
+        list(APPEND BRLS_PLATFORM_LIBS "-framework SystemConfiguration" "-framework Cocoa")
+        list(APPEND BRLS_PLATFORM_LIBS "-framework IOKit" "-framework CoreFoundation")
         list(APPEND BRLS_PLATFORM_LIBS "-framework AppKit" "-framework Foundation" "-framework CoreServices" "-lobjc")
         list(APPEND BOREALIS_SRC ${BOREALIS_PATH}/lib/platforms/desktop/desktop_darwin.mm)
     endif ()
