From ce2087bd4dfac587e8f59077e9de237363ca8a29 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 6 Oct 2025 18:53:05 +0800
Subject: [PATCH] Fix colors on Big-endian

---
 wiliwili/include/view/mpv_core.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/wiliwili/include/view/mpv_core.hpp b/wiliwili/include/view/mpv_core.hpp
index 06078f237..331659690 100644
--- wiliwili/include/view/mpv_core.hpp
+++ wiliwili/include/view/mpv_core.hpp
@@ -405,7 +405,11 @@ class MPVCore : public brls::Singleton<MPVCore> {
 #ifdef MPV_SW_RENDER
     const int PIXCEL_SIZE          = 4;
     int nvg_image                  = 0;
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+    const char *sw_format          = "abgr";
+#else
     const char *sw_format          = "rgba";
+#endif
     int sw_size[2]                 = {1920, 1080};
     size_t pitch                   = PIXCEL_SIZE * sw_size[0];
     void *pixels                   = nullptr;
