--- library/borealis/library/lib/platforms/desktop/desktop_platform.cpp
+++ library/borealis/library/lib/platforms/desktop/desktop_platform.cpp	2025-10-04 21:04:45.000000000 +0800
@@ -554,6 +554,8 @@
     if (globalRef)
     {
         CFDictionaryGetValueIfPresent(globalRef, kSCDynamicStorePropNetPrimaryInterface, &primaryIf);
+            if (primaryIf)
+                CFRetain(primaryIf);
         CFRelease(globalRef);
     }
     CFRelease(storeRef);
@@ -577,6 +579,7 @@
         }
         CFRelease(iflist);
     }
+    CFRelease(primaryIf);
 #elif defined(_WIN32)
     PIP_ADAPTER_ADDRESSES addrs = nullptr;
     ULONG outlen = sizeof(IP_ADAPTER_ADDRESSES), ret = 0, curindex = 0;
