--- tools/tiff2ps.c	2026-06-27 20:36:34.000000000 +0800
+++ tools/tiff2ps.c	2026-07-04 07:34:47.000000000 +0800
@@ -698,17 +698,12 @@
             "%%ImageData: %" PRIu32 " %" PRIu32 " %" PRIu16
             " %d 0 %" TIFF_SSIZE_FORMAT " 2 \"",
             w, h, bitspersample, nc, bs);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wformat-nonliteral"
 #endif
     fprintf(fd, startline, nc);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic pop
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
     fprintf(fd, "\"\n");
@@ -2873,17 +2868,12 @@
             break;
         case PHOTOMETRIC_MINISBLACK:
         case PHOTOMETRIC_MINISWHITE:
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wformat-nonliteral"
 #endif
             PhotoshopBanner(fd, w, h, 1, 1, imageOp);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic pop
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
             fprintf(fd, "/scanLine %" TIFF_SSIZE_FORMAT " string def\n",

--- tools/tiffcrop.c	2026-06-27 20:36:34.000000000 +0800
+++ tools/tiffcrop.c	2026-07-04 07:41:52.000000000 +0800
@@ -3504,17 +3504,12 @@
         va_list ap;
         va_start(ap, msg);
         fprintf(dumpfile, "%s ", prefix);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wformat-nonliteral"
 #endif
         vfprintf(dumpfile, msg, ap);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic pop
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
         fprintf(dumpfile, "\n");

--- tools/tiffdump.c	2026-06-27 20:36:34.000000000 +0800
+++ tools/tiffdump.c	2026-07-04 07:46:13.000000000 +0800
@@ -952,17 +952,12 @@
 static void vError(FILE *fd, const char *fmt, va_list ap)
 {
     fprintf(fd, "%s: ", curfile);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wformat-nonliteral"
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wformat-nonliteral"
 #endif
     vfprintf(fd, fmt, ap);
-#if defined(__GNUC__) && !defined(__clang__)
-#pragma GCC diagnostic pop
-#elif defined(__clang__)
+#if defined(__clang__)
 #pragma clang diagnostic pop
 #endif
     fprintf(fd, ".\n");
