--- libemf/libemf.cpp.orig	2020-06-07 21:10:20.000000000 +0800
+++ libemf/libemf.cpp	2026-02-02 03:42:48.000000000 +0800
@@ -21,6 +21,13 @@
 #include <iostream>
 #include <climits>
 
+#ifdef __APPLE__
+#include <libkern/OSByteOrder.h>
+#define bswap_32(x) OSSwapInt32(x)
+#else
+#include <byteswap.h>
+#endif
+
 #include "libemf.h"
 
 namespace EMF {
@@ -72,7 +79,6 @@
     if ( not bigEndian() ) {
       return a;
     }
-#include <byteswap.h>
     return bswap_32(a);
   }
 
