From e594d16bab49aaccd2fd5e4ce92087b516b2e529 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 31 May 2026 15:14:52 +0800
Subject: [PATCH 1/2] cocoa.mm: fix availability macros usage

---
 vncviewer/cocoa.mm | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git vncviewer/cocoa.mm vncviewer/cocoa.mm
index e6bd4491..23b1023b 100644
--- vncviewer/cocoa.mm
+++ vncviewer/cocoa.mm
@@ -26,6 +26,8 @@
 #include <FL/Fl_Window.H>
 #include <FL/x.H>
 
+#include <AvailabilityMacros.h>
+
 #import <Cocoa/Cocoa.h>
 #import <ApplicationServices/ApplicationServices.h>
 
@@ -39,7 +41,7 @@ void cocoa_prevent_native_fullscreen(Fl_Window *win)
   NSWindow *nsw;
   nsw = (NSWindow*)fl_xid(win);
   assert(nsw);
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 100700
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
   nsw.collectionBehavior |= NSWindowCollectionBehaviorFullScreenNone;
 #endif
 }
@@ -52,8 +54,7 @@ bool cocoa_is_trusted(bool prompt)
 
   Boolean trusted;
 
-#if !defined(MAC_OS_X_VERSION_10_9) || MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_9
-  // FIXME: Raise system requirements so this isn't needed
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1090
   void *lib;
   typedef Boolean (*AXIsProcessTrustedWithOptionsRef)(CFDictionaryRef);
   AXIsProcessTrustedWithOptionsRef AXIsProcessTrustedWithOptions;
-- 
2.54.0

