From f455d9e3e4858a9821ddc88a8af5bf3ed0d211c0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 31 May 2026 15:21:45 +0800
Subject: [PATCH 2/2] cocoa.mm: unbreak build on 10.5

---
 vncviewer/cocoa.mm | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git vncviewer/cocoa.mm vncviewer/cocoa.mm
index 23b1023b..60b9b72d 100644
--- vncviewer/cocoa.mm
+++ vncviewer/cocoa.mm
@@ -31,6 +31,10 @@
 #import <Cocoa/Cocoa.h>
 #import <ApplicationServices/ApplicationServices.h>
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1060
+#include <Carbon/Carbon.h>
+#endif
+
 #include "cocoa.h"
 
 static CFMachPortRef event_tap;
@@ -135,6 +139,7 @@ bool cocoa_is_trusted(bool prompt)
     }
 
     if (pid != 0) {
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
       NSRunningApplication* authApp;
 
       authApp = [NSRunningApplication runningApplicationWithProcessIdentifier:pid];
@@ -143,6 +148,12 @@ bool cocoa_is_trusted(bool prompt)
         // or NSApplicationActivateIgnoringOtherApps
         [authApp activateWithOptions:0];
       }
+#else
+      ProcessSerialNumber psn = {0, kNoProcess};
+      if (GetProcessForPID((pid_t)pid, &psn) == noErr) {
+        SetFrontProcess(&psn);
+      }
+#endif
     }
   }
 
-- 
2.54.0

