--- build/cmake/CondorConfigure.cmake	2019-02-19 12:08:49.000000000 +0800
+++ build/cmake/CondorConfigure.cmake	2025-08-16 06:48:37.000000000 +0800
@@ -18,10 +18,7 @@
 
 # OS pre mods
 if(${OS_NAME} STREQUAL "DARWIN")
-	# All recent versions of Mac OS X are 64-bit, but 'uname -p'
-	# (the source for SYS_ARCH) reports 'i386'.
-	# Override that to set the actual architecture.
-	set (SYS_ARCH "X86_64")
+#	set (SYS_ARCH ${CMAKE_SYSTEM_PROCESSOR})
 elseif(${OS_NAME} MATCHES "WIN")
 	cmake_minimum_required(VERSION 2.8.3)
 	set(WINDOWS ON)
