--- data/appdata/meson.build	2021-06-26 00:49:29.000000000 +0800
+++ data/appdata/meson.build	2024-07-02 18:22:05.000000000 +0800
@@ -1,5 +1,5 @@
 appdata_name = app_id + '.appdata.xml'
-appdata = i18n.merge_file(appdata_name,
+appdata = i18n.merge_file(
   input: appdata_name + '.in',
   output: appdata_name,
   po_dir: '../../po',

--- data/meson.build	2021-06-26 00:49:29.000000000 +0800
+++ data/meson.build	2024-07-02 18:24:47.000000000 +0800
@@ -2,7 +2,7 @@
 subdir('icons')
 
 desktop_filename = app_id + '.desktop'
-desktop_file = i18n.merge_file(desktop_filename,
+desktop_file = i18n.merge_file(
   input: desktop_filename + '.in',
   output: desktop_filename,
   po_dir: '../po',
