--- event.c.orig	2026-06-21 22:50:37.000000000 +0800
+++ event.c	2026-09-03 09:29:46.000000000 +0800
@@ -1353,15 +1353,6 @@
 static inline void
 event_signal_closure(struct event_base *base, struct event *ev)
 {
-#if defined(__clang__)
-#elif defined(__GNUC__)
-#pragma GCC diagnostic push
-/* NOTE: it is better to avoid such code all together, by using separate
- * variable to break the loop in the event structure, but now this code is safe
- * */
-#pragma GCC diagnostic ignored "-Wdangling-pointer"
-#endif
-
 	short ncalls;
 	int should_break;
 
@@ -1387,11 +1378,6 @@
 			return;
 		}
 	}
-
-#if defined(__clang__)
-#elif defined(__GNUC__)
-#pragma GCC diagnostic pop
-#endif
 }
 
 /* Common timeouts are special timeouts that are handled as queues rather than
