From 0caae4dc89a799302e2cf23f7ac664ee91197006 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 1 Jun 2025 00:23:35 +0800
Subject: [PATCH] Use prefix for datadir

---
 llarp/constants/files.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llarp/constants/files.hpp b/llarp/constants/files.hpp
index 167cc08b..f2e6a574 100644
--- a/llarp/constants/files.hpp
+++ b/llarp/constants/files.hpp
@@ -20,7 +20,7 @@
   inline fs::path
   GetDefaultDataDir()
   {
-    fs::path datadir{"/var/lib/llarpd"};
+    fs::path datadir{"@PREFIX@/var/lib/llarpd"};
     if (auto uid = geteuid())
     {
       if (auto* pw = getpwuid(uid))
