From 5ab701b39ee20a77e3198e890b8cfffd8836b8e7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 30 May 2025 16:54:08 +0800
Subject: [PATCH 4/7] logging.cpp: fix possible crash on launch

---
 llarp/util/logging.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/llarp/util/logging.cpp b/llarp/util/logging.cpp
index 237c934b..c0b8fafc 100644
--- a/llarp/util/logging.cpp
+++ b/llarp/util/logging.cpp
@@ -100,7 +100,11 @@ namespace llarp::log
     format_sl(const llarp::util::source_location& loc)
     {
       static constexpr std::string_view source_prefix = LOGGING_SOURCE_ROOT;
-      std::string_view filename{loc.file_name()};
+      std::string_view filename;
+      if (auto fn = loc.file_name())
+        filename = fn;
+      else
+        filename = "<unknown>";
       if (filename.substr(0, source_prefix.size()) == source_prefix)
       {
         filename.remove_prefix(source_prefix.size());
-- 
2.49.0

