From aa20343ba9f1d54dde613349b6d47a4661bfcb67 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 28 May 2025 09:12:14 +0800
Subject: [PATCH 6/7] crypto/CMakeLists: do not use x86 flags on arm64

---
 crypto/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/crypto/CMakeLists.txt b/crypto/CMakeLists.txt
index c73a6ed1..053918d7 100644
--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -47,7 +47,8 @@ set(NTRU_AVX_SRC
 include(CheckCXXCompilerFlag)
 check_cxx_compiler_flag(-mavx2 COMPILER_SUPPORTS_AVX2)
 check_cxx_compiler_flag(-mfma COMPILER_SUPPORTS_FMA)
-if(COMPILER_SUPPORTS_AVX2 AND COMPILER_SUPPORTS_FMA AND (NOT ANDROID))
+if(COMPILER_SUPPORTS_AVX2 AND COMPILER_SUPPORTS_FMA AND (NOT ANDROID)
+  AND (NOT CMAKE_OSX_ARCHITECTURES MATCHES "arm64"))
   target_sources(lokinet-cryptography PRIVATE ${NTRU_AVX_SRC})
   set_property(SOURCE ${NTRU_AVX_SRC} APPEND PROPERTY COMPILE_FLAGS "-mavx2 -mfma")
   message(STATUS "Building libntrup with runtime AVX2/FMA support")
-- 
2.49.0

