From f6ed17b0774383f71697b541dbefcd7c3c819cfb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 1 Jun 2025 02:20:37 +0800
Subject: [PATCH 7/7] dht/key.hpp: remove template to fix linking

---
 llarp/dht/key.hpp | 2 --
 1 file changed, 2 deletions(-)

diff --git a/llarp/dht/key.hpp b/llarp/dht/key.hpp
index fa9ee929..9155da33 100644
--- a/llarp/dht/key.hpp
+++ b/llarp/dht/key.hpp
@@ -71,6 +71,4 @@ namespace llarp
       }
     };
   }  // namespace dht
-  template <>
-  constexpr bool IsToStringFormattable<dht::Key_t> = true;
 }  // namespace llarp
-- 
2.49.0

