--- src/sdk/configmanager.cpp.orig	2025-04-30 20:18:35.000000000 +0800
+++ src/sdk/configmanager.cpp	2025-04-30 20:23:12.000000000 +0800
@@ -1502,7 +1502,7 @@
             ConfigManager::plugin_path_global = data_path_global + _T("/plugins");
         else
         {
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
             // It seems we can not longer rely on wxStandardPathsBase::Get().GetPluginsDir(),
             // because its behaviour has changed on some systems (at least Fedora 14 64-bit).
             // So we create the pathname manually
