--- packaging/pep517_backend/_backend.py	2025-10-09 02:09:05.000000000 +0800
+++ packaging/pep517_backend/_backend.py	2026-02-20 12:02:00.000000000 +0800
@@ -376,7 +376,7 @@
     if is_pure_python_build:
         c_ext_build_deps = []
     else:
-        c_ext_build_deps = ['Cython ~= 3.1.0']
+        c_ext_build_deps = ['Cython >= 3.1.0']
 
     return _setuptools_get_requires_for_build_wheel(
         config_settings=config_settings,
