--- core/include/prometheus/registry.h	2021-08-29 21:04:00.000000000 +0800
+++ core/include/prometheus/registry.h	2026-03-13 14:43:15.000000000 +0800
@@ -5,6 +5,7 @@
 #include <mutex>
 #include <string>
 #include <vector>
+#include <stdexcept>
 
 #include "prometheus/collectable.h"
 #include "prometheus/family.h"

--- core/include/prometheus/family.h	2021-08-29 21:04:00.000000000 +0800
+++ core/include/prometheus/family.h	2026-03-13 14:43:04.000000000 +0800
@@ -8,6 +8,7 @@
 #include <unordered_map>
 #include <vector>
 #include <cassert>
+#include <stdexcept>
 
 #include "prometheus/collectable.h"
 #include "prometheus/metric.h"
