--- psutil/arch/osx/mem.c.orig	2025-11-03 05:47:52.000000000 +0800
+++ psutil/arch/osx/mem.c	2026-09-12 14:02:28.000000000 +0800
@@ -18,11 +18,27 @@
 
 #include "../../arch/all/init.h"
 
+#ifdef HOST_VM_INFO64
+    #define PSUTIL_VM_STAT_T vm_statistics64_data_t
+    #define PSUTIL_VM_STAT_PTR_T vm_statistics64_t
+    #define PSUTIL_HOST_VM_INFO HOST_VM_INFO64
+    #define PSUTIL_HOST_VM_INFO_COUNT HOST_VM_INFO64_COUNT
+    #define PSUTIL_HOST_STAT host_statistics64
+    #define PSUTIL_HOST_INFO_T host_info64_t
+#else
+    #define PSUTIL_VM_STAT_T vm_statistics_data_t
+    #define PSUTIL_VM_STAT_PTR_T vm_statistics_t
+    #define PSUTIL_HOST_VM_INFO HOST_VM_INFO
+    #define PSUTIL_HOST_VM_INFO_COUNT HOST_VM_INFO_COUNT
+    #define PSUTIL_HOST_STAT host_statistics
+    #define PSUTIL_HOST_INFO_T host_info_t
+#endif
+
 
 static int
-psutil_sys_vminfo(vm_statistics64_t vmstat) {
+psutil_sys_vminfo(PSUTIL_VM_STAT_PTR_T vmstat) {
     kern_return_t ret;
-    mach_msg_type_number_t count = HOST_VM_INFO64_COUNT;
+    mach_msg_type_number_t count = PSUTIL_HOST_VM_INFO_COUNT;
     mach_port_t mport;
 
     mport = mach_host_self();
@@ -31,13 +47,13 @@
         return -1;
     }
 
-    ret = host_statistics64(
-        mport, HOST_VM_INFO64, (host_info64_t)vmstat, &count
+    ret = PSUTIL_HOST_STAT(
+        mport, PSUTIL_HOST_VM_INFO, (PSUTIL_HOST_INFO_T)vmstat, &count
     );
     mach_port_deallocate(mach_task_self(), mport);
     if (ret != KERN_SUCCESS) {
         psutil_runtime_error(
-            "host_statistics64(HOST_VM_INFO64) syscall failed: %s",
+            "PSUTIL_HOST_STAT(PSUTIL_HOST_VM_INFO) syscall failed: %s",
             mach_error_string(ret)
         );
         return -1;
@@ -56,7 +72,7 @@
 psutil_virtual_mem(PyObject *self, PyObject *args) {
     int mib[2];
     uint64_t total;
-    vm_statistics64_data_t vm;
+    PSUTIL_VM_STAT_T vm;
     long pagesize = psutil_getpagesize();
     // physical mem
     mib[0] = CTL_HW;
@@ -77,7 +93,11 @@
         (unsigned long long)vm.inactive_count * pagesize,  // inactive
         (unsigned long long)vm.wire_count * pagesize,  // wired
         (unsigned long long)vm.free_count * pagesize,  // free
+#if  defined(HOST_VM_INFO64) || defined(HOST_VM_INFO_REV1_COUNT)
         (unsigned long long)vm.speculative_count * pagesize  // speculative
+#else
+        0ULL  /* speculative_count not available on this platform */
+#endif
     );
 }
 
@@ -89,7 +109,7 @@
 psutil_swap_mem(PyObject *self, PyObject *args) {
     int mib[2];
     struct xsw_usage totals;
-    vm_statistics64_data_t vmstat;
+    PSUTIL_VM_STAT_T vmstat;
     long pagesize = psutil_getpagesize();
 
     mib[0] = CTL_VM;
