From 15777726951341393e8fb2614268f1d26f14e08b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 30 Dec 2025 01:48:35 +0800
Subject: [PATCH] macOS: fix heap.c for < 10.7

---
 psutil/arch/osx/heap.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/psutil/arch/osx/heap.c b/psutil/arch/osx/heap.c
index 95238ba31..5f32bab75 100644
--- psutil/arch/osx/heap.c
+++ psutil/arch/osx/heap.c
@@ -9,6 +9,8 @@
 #include <mach/mach.h>
 #include <mach/vm_map.h>
 
+#include <AvailabilityMacros.h>
+
 #include "../../arch/all/init.h"
 
 
@@ -96,8 +98,11 @@ psutil_heap_trim(PyObject *self, PyObject *args) {
     if (ok == -1)
         return NULL;
 
+// malloc_zone_pressure_relief added in macOS 10.7.
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
     for (unsigned int i = 0; i < count; i++)
         malloc_zone_pressure_relief(zones[i], 0);
+#endif
 
     if (!ok)
         free(zones);
