--- psutil/arch/osx/proc.c	2025-10-30 09:57:46.000000000 +0800
+++ psutil/arch/osx/proc.c	2025-11-30 12:03:26.000000000 +0800
@@ -100,7 +100,7 @@
         (long)kp.kp_eproc.e_pcred.p_rgid,  // (long) real gid
         (long)kp.kp_eproc.e_ucred.cr_groups[0],  // (long) effective gid
         (long)kp.kp_eproc.e_pcred.p_svgid,  // (long) saved gid
-        (long long)kp.kp_eproc.e_tdev,  // (long long) tty nr
+        (int)kp.kp_eproc.e_tdev,  // (int) tty nr
         PSUTIL_TV2DOUBLE(kp.kp_proc.p_starttime),  // (double) create time
         status,  // (int) status
         py_name  // (pystr) name
