--- psutil/arch/osx/cpu.c	2025-10-28 02:21:13.000000000 +0800
+++ psutil/arch/osx/cpu.c	2025-11-30 12:07:34.000000000 +0800
@@ -30,6 +30,7 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/IOKitLib.h>
 #endif
+#include <AvailabilityMacros.h>
 
 #include "../../arch/all/init.h"
 
@@ -119,8 +120,7 @@
         vmstat.v_swtch,
         vmstat.v_intr,
         vmstat.v_soft,
-#if defined(__MAC_OS_X_VERSION_MIN_REQUIRED) \
-    && __MAC_OS_X_VERSION_MIN_REQUIRED__ >= 120000
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 120000
         0,
 #else
         vmstat.v_syscall,
