--- psutil/arch/osx/proc.c	2025-10-30 09:57:46.000000000 +0800
+++ psutil/arch/osx/proc.c	2025-11-30 12:03:26.000000000 +0800
@@ -254,6 +254,14 @@
             base = SHARED_REGION_BASE_X86_64;
             size = SHARED_REGION_SIZE_X86_64;
             break;
+        case CPU_TYPE_POWERPC:
+            base = SHARED_REGION_BASE_PPC;
+            size = SHARED_REGION_SIZE_PPC;
+            break;
+        case CPU_TYPE_POWERPC64:
+            base = SHARED_REGION_BASE_PPC64;
+            size = SHARED_REGION_SIZE_PPC64;
+            break;
         default:
             return false;
     }
