--- configure	2025-05-18 07:24:39.000000000 +0800
+++ configure	2025-06-18 16:17:48.000000000 +0800
@@ -15336,7 +15336,7 @@
   LIBS="$PGSQL_FE_LDFLAGS"
 
   CC_SAVE="$CC"
-  CC=`"$PG_CONFIG" --cc`
+  CC="$CC"
 
   CFLAGS_SAVE="$CFLAGS"
   CFLAGS=`"$PG_CONFIG" --cflags`
@@ -20841,7 +20841,7 @@
 
 		if test "x$PG_CONFIG" != "xno"; then
 			CFLAGS=`"$PG_CONFIG" --cflags`
-			CC=`"$PG_CONFIG" --cc`
+			CC="$CC"
 		fi
 
 		LIBS_SAVE="$LIBS"
@@ -21054,7 +21054,7 @@
     LDFLAGS_SAVE="$LDFLAGS"
     LIBS_SAVE="$LIBS"
 
-    WAGYU_CXX=`"$PG_CONFIG" --cc`
+    WAGYU_CXX="$CXX"
     CPPFLAGS="-x c++"
     CFLAGS=""
     LDFLAGS=""
@@ -21157,7 +21157,7 @@
 done
 
   if test "x$ac_ct_CXX" = x; then
-    CXX="g++"
+    CXX="$CXX"
   else
     case $cross_compiling:$ac_tool_warned in
 yes:)
@@ -25007,7 +25007,7 @@
 
     WAGYU_CXX="$CXX -x c++"
 
-                CC="$WAGYU_CXX"
+                CC="$CXX"
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lc++" >&5
 $as_echo_n "checking for main in -lc++... " >&6; }
@@ -25234,7 +25234,7 @@
 done
 
   if test "x$ac_ct_CXX" = x; then
-    CXX="g++"
+    CXX="CXX"
   else
     case $cross_compiling:$ac_tool_warned in
 yes:)

--- deps/flatgeobuf/Makefile.in	2025-05-18 07:24:39.000000000 +0800
+++ deps/flatgeobuf/Makefile.in	2025-06-18 16:30:10.000000000 +0800
@@ -28,7 +28,7 @@
 srcdir = @srcdir@
 libdir = @libdir@
 
-CXX = @FLATGEOBUF_CXX@
+CXX = @CXX@
 # This ugly -Dflatbuffers=postgis_flatbuffers is to prevent potential subtle
 # clash with other flatbuffers copies that could come from GDAL or its
 # dependencies.

--- deps/wagyu/Makefile.in	2025-05-18 07:24:39.000000000 +0800
+++ deps/wagyu/Makefile.in	2025-06-18 16:32:51.000000000 +0800
@@ -28,7 +28,7 @@
 top_srcdir = @top_srcdir@
 libdir = @libdir@
 
-CXX = @WAGYU_CXX@
+CXX = @CXX@
 CXXFLAGS =-I$(top_srcdir)/liblwgeom -I$(top_builddir)/liblwgeom -I$(srcdir)/include @CPPFLAGS@ @CXXFLAGS@ @PICFLAGS@
 LDFLAGS = @LDFLAGS@
 LIBTOOL = @LIBTOOL@

--- postgis/Makefile.in	2025-05-18 07:24:39.000000000 +0800
+++ postgis/Makefile.in	2025-06-18 16:54:52.000000000 +0800
@@ -171,8 +171,9 @@
 srcdir := @srcdir@
 VPATH := $(srcdir)
 
-# Set PERL _after_ the include of PGXS
+# Set PERL and CC _after_ the include of PGXS
 PERL=@PERL@
+CC=@CC@
 
 # This is to workaround a bug in PGXS 8.4 win32 link line,
 # see http://trac.osgeo.org/postgis/ticket/1158#comment:57

--- topology/Makefile.in	2025-05-18 07:24:39.000000000 +0800
+++ topology/Makefile.in	2025-06-18 17:07:35.000000000 +0800
@@ -85,8 +85,9 @@
 exec_prefix = @exec_prefix@
 bindir = @bindir@
 
-# Set PERL _after_ the include of PGXS
+# Set PERL and CC _after_ the include of PGXS
 PERL=@PERL@
+CC=@CC@
 
 # This is to workaround a bug in PGXS 8.4 win32 link line,
 # see http://trac.osgeo.org/postgis/ticket/1158#comment:57

--- raster/rt_pg/Makefile.in	2025-05-18 07:24:39.000000000 +0800
+++ raster/rt_pg/Makefile.in	2025-06-18 17:14:20.000000000 +0800
@@ -74,7 +74,6 @@
 LIBGDAL_LDFLAGS=@LIBGDAL_LDFLAGS@
 LIBPROJ_CFLAGS=@PROJ_CPPFLAGS@
 
-CC = @CC@
 PG_CPPFLAGS += \
 	$(LIBLWGEOM_CFLAGS) \
 	@CPPFLAGS@ \
@@ -104,8 +103,9 @@
 # Utility rule for use from outside
 sql_objs: $(DATA_built)
 
-# Set PERL _after_ the include of PGXS
+# Set PERL and CC _after_ the include of PGXS
 PERL=@PERL@
+CC = @CC@
 
 # This is to workaround a bug in PGXS 8.4 win32 link line,
 # see http://trac.osgeo.org/postgis/ticket/1158#comment:57
