From 851fb14c27f7cc66bcacbd8b54939be7c7c3121d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 14 May 2025 19:29:28 +0800
Subject: [PATCH] rest_profile.cc: add missing <filesystem>

---
 tiledb/sm/rest/rest_profile.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tiledb/sm/rest/rest_profile.cc b/tiledb/sm/rest/rest_profile.cc
index 06457991f..8a8ade22f 100644
--- tiledb/sm/rest/rest_profile.cc
+++ tiledb/sm/rest/rest_profile.cc
@@ -31,6 +31,7 @@
  */
 
 #include <iostream>
+#include <filesystem>
 
 #include "rest_profile.h"
 #include "tiledb/common/random/random_label.h"
