--- tiledb/CMakeLists.txt	2025-09-22 23:21:21.000000000 +0800
+++ tiledb/CMakeLists.txt	2025-10-02 14:06:54.000000000 +0800
@@ -582,6 +582,7 @@
     spdlog::spdlog
     ZLIB::ZLIB
     zstd::libzstd
+    MacportsLegacySupport
 )
 
 if(TILEDB_WEBP)
@@ -593,6 +594,13 @@
   add_definitions(-DTILEDB_WEBP)
 endif()
 
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
+    INTERFACE
+    atomic
+  )
+endif()
+
 if (NOT WIN32)
   target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
     INTERFACE

--- test/CMakeLists.txt	2025-09-22 23:21:21.000000000 +0800
+++ test/CMakeLists.txt	2025-10-02 14:04:04.000000000 +0800
@@ -231,6 +231,7 @@
     tiledb_test_support_lib
     configuration_definitions
     rapidcheck
+    MacportsLegacySupport
 )
 
 target_link_libraries(tiledb_unit PRIVATE $<BUILD_INTERFACE:common>)
