--- src/CMakeLists.txt.orig	2026-02-03 16:56:24.000000000 +0800
+++ src/CMakeLists.txt	2026-05-16 13:05:54.000000000 +0800
@@ -114,6 +114,7 @@
   file (GLOB xournalpp-sources-osx exe/osx/*.cpp exe/osx/*.h)
   target_sources(xournalpp PRIVATE ${xournalpp-sources-osx})
   target_include_directories(xournalpp PRIVATE exe)
+  target_link_libraries(xournalpp PRIVATE "-framework CoreFoundation")
 endif()
 
 if (WIN32)
