--- widgets/detailswidget.cpp	2013-08-12 04:07:23.000000000 +0800
+++ widgets/detailswidget.cpp	2023-08-23 11:09:31.000000000 +0800
@@ -17,6 +17,8 @@
 #include "detailswidget.h"
 #include <iostream>
 
+#include <exiv2/exiv2.hpp>
+
 // This class provides a widget for displaying meta (Exif) information
 Details::Details(QWidget *parent, QMap<QString, QVariant> set, bool v): QWidget(parent) {
 
@@ -492,7 +494,7 @@
 
 				}
 
-			} catch(const Exiv2::AnyError&) { }
+			} catch(const Exiv2::Error &e) { }
 
 
 		}

--- widgets/detailswidget.cpp	2013-08-12 04:07:23.000000000 +0800
+++ widgets/detailswidget.cpp	2026-01-25 13:58:13.000000000 +0800
@@ -393,7 +393,7 @@
 	if(exiv2Supported) {
 
 
-		Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(currentfile.toUtf8().constData());
+		Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open(currentfile.toUtf8().constData());
 		image->readMetadata();
 		Exiv2::ExifData &exifData = image->exifData();
 
