--- packager/file/local_file.cc
+++ packager/file/local_file.cc
@@ -83,7 +83,7 @@
   }
 
   std::error_code ec;
-  auto file_path = std::filesystem::u8path(file_name());
+  auto file_path = std::filesystem::path(file_name());
   int64_t file_size = std::filesystem::file_size(file_path, ec);
   if (ec) {
     LOG(ERROR) << "Cannot get file size, error: " << ec;
@@ -121,7 +121,7 @@
 LocalFile::~LocalFile() {}
 
 bool LocalFile::Open() {
-  auto file_path = std::filesystem::u8path(file_name());
+  auto file_path = std::filesystem::path(file_name());
 
   // Create upper level directories for write mode.
   if (file_mode_.find("w") != std::string::npos) {
@@ -140,12 +140,12 @@
     }
   }
 
-  internal_file_ = fopen(file_path.u8string().c_str(), file_mode_.c_str());
+  internal_file_ = fopen(file_path.string().c_str(), file_mode_.c_str());
   return (internal_file_ != NULL);
 }
 
 bool LocalFile::Delete(const char* file_name) {
-  auto file_path = std::filesystem::u8path(file_name);
+  auto file_path = std::filesystem::path(file_name);
   std::error_code ec;
   // On error (ec truthy), remove() will return false anyway.
   return std::filesystem::remove(file_path, ec);
