--- packager/media/test/test_web_server.h	2026-07-28 02:28:24.000000000 +0800
+++ packager/media/test/test_web_server.h	2026-08-29 21:35:02.000000000 +0800
@@ -74,8 +74,7 @@
 
   static void HandleEvent(struct mg_connection* connection,
                           int event,
-                          void* event_data,
-                          void* callback_data);
+                          void* event_data);
 
   bool HandleStatus(struct mg_http_message* message,
                     struct mg_connection* connection);

--- packager/media/test/test_web_server.cc	2026-07-28 02:28:24.000000000 +0800
+++ packager/media/test/test_web_server.cc	2026-08-29 21:36:53.000000000 +0800
@@ -40,15 +40,15 @@
 
 // Get a string_view on mongoose's mg_string, which may not be nul-terminated.
 std::string_view MongooseStringView(const mg_str& mg_string) {
-  return std::string_view(mg_string.ptr, mg_string.len);
+  return std::string_view(mg_string.buf, mg_string.len);
 }
 
 bool IsMongooseStringNull(const mg_str& mg_string) {
-  return mg_string.ptr == NULL;
+  return mg_string.buf == NULL;
 }
 
 bool IsMongooseStringNullOrBlank(const mg_str& mg_string) {
-  return mg_string.ptr == NULL || mg_string.len == 0;
+  return mg_string.buf == NULL || mg_string.len == 0;
 }
 
 // Get a string query parameter from a mongoose HTTP message.
@@ -173,9 +173,8 @@
 // static
 void TestWebServer::HandleEvent(struct mg_connection* connection,
                                 int event,
-                                void* event_data,
-                                void* callback_data) {
-  TestWebServer* instance = static_cast<TestWebServer*>(callback_data);
+                                void* event_data) {
+  TestWebServer* instance = static_cast<TestWebServer*>(connection->fn_data);
 
   if (event == MG_EV_POLL) {
     std::vector<struct mg_connection*> to_delete;
@@ -207,13 +206,13 @@
 
   struct mg_http_message* message =
       static_cast<struct mg_http_message*>(event_data);
-  if (mg_http_match_uri(message, "/reflect")) {
+  if (mg_match(message->uri, mg_str("/reflect"), NULL)) {
     if (instance->HandleReflect(message, connection))
       return;
-  } else if (mg_http_match_uri(message, "/status")) {
+  } else if (mg_match(message->uri, mg_str("/status"), NULL)) {
     if (instance->HandleStatus(message, connection))
       return;
-  } else if (mg_http_match_uri(message, "/delay")) {
+  } else if (mg_match(message->uri, mg_str("/delay"), NULL)) {
     if (instance->HandleDelay(message, connection))
       return;
   }
