--- packager/file/udp_file.cc	2026-07-28 02:28:24.000000000 +0800
+++ packager/file/udp_file.cc	2026-08-29 20:24:20.000000000 +0800
@@ -15,12 +15,12 @@
 #define close closesocket
 #define EINTR_CODE WSAEINTR
 #else
+#include <sys/socket.h>
+#include <sys/time.h>
 #include <arpa/inet.h>
 #include <errno.h>
 #include <netinet/in.h>
 #include <string.h>
-#include <sys/socket.h>
-#include <sys/time.h>
 #include <unistd.h>
 #define INVALID_SOCKET -1
 #define EINTR_CODE EINTR
@@ -29,6 +29,14 @@
 #ifndef IP_MULTICAST_ALL
 #define IP_MULTICAST_ALL 49
 #endif
+#ifndef IP_ADD_SOURCE_MEMBERSHIP
+#define IP_ADD_SOURCE_MEMBERSHIP 70
+struct ip_mreq_source {
+  struct in_addr imr_multiaddr;
+  struct in_addr imr_sourceaddr;
+  struct in_addr imr_interface;
+};
+#endif
 #endif  // defined(OS_WIN)
 
 #include <absl/log/check.h>
