--- configure	2026-07-30 20:30:50.000000000 +0800
+++ configure	2026-08-04 21:58:13.000000000 +0800
@@ -7621,9 +7621,6 @@
   if test "x$target_win32" = "xyes"; then
       FLRIG_BUILD_CPPFLAGS="$FLRIG_BUILD_CPPFLAGS -D_WINDOWS"
   fi
-  if test "x$target_darwin" = "xyes"; then
-      FLRIG_BUILD_CPPFLAGS="$FLRIG_BUILD_CPPFLAGS -fno-stack-check -mmacosx-version-min=10.11"
-  fi
 # CXXFLAGS
   FLRIG_BUILD_CXXFLAGS="$FLTK_CFLAGS -I\$(srcdir) -I\$(srcdir)/include  \
 $X_CFLAGS -pipe -Wall -fexceptions $OPT_CFLAGS $DEBUG_CFLAGS $PTW32_CFLAGS"
@@ -7640,10 +7637,6 @@
       FLRIG_BUILD_CXXFLAGS="-mthreads $FLRIG_BUILD_CXXFLAGS"
   fi
 
-  if test "x$target_darwin" = "xyes"; then
-      FLRIG_BUILD_CXXFLAGS="$FLRIG_BUILD_CXXFLAGS -fno-stack-check -mmacosx-version-min=10.11"
-  fi
-
 # LDFLAGS
   FLRIG_BUILD_LDFLAGS=
 # LDADD

--- m4/build.m4	2026-04-11 06:57:56.000000000 +0800
+++ m4/build.m4	2026-08-04 21:59:01.000000000 +0800
@@ -12,9 +12,6 @@
   if test "x$target_win32" = "xyes"; then
       FLRIG_BUILD_CPPFLAGS="$FLRIG_BUILD_CPPFLAGS -D_WINDOWS"
   fi
-  if test "x$target_darwin" = "xyes"; then
-      FLRIG_BUILD_CPPFLAGS="$FLRIG_BUILD_CPPFLAGS -fno-stack-check -mmacosx-version-min=10.11"
-  fi
 # CXXFLAGS
   FLRIG_BUILD_CXXFLAGS="$FLTK_CFLAGS -I\$(srcdir) -I\$(srcdir)/include  \
 $X_CFLAGS -pipe -Wall -fexceptions $OPT_CFLAGS $DEBUG_CFLAGS $PTW32_CFLAGS"
@@ -31,10 +28,6 @@
       FLRIG_BUILD_CXXFLAGS="-mthreads $FLRIG_BUILD_CXXFLAGS"
   fi
 
-  if test "x$target_darwin" = "xyes"; then
-      FLRIG_BUILD_CXXFLAGS="$FLRIG_BUILD_CXXFLAGS -fno-stack-check -mmacosx-version-min=10.11"
-  fi
-
 # LDFLAGS
   FLRIG_BUILD_LDFLAGS=
 # LDADD
