--- src/plugins/mac/wscript.orig	2026-03-17 05:49:25.000000000 +0800
+++ src/plugins/mac/wscript	2026-05-19 09:55:40.000000000 +0800
@@ -5,19 +5,18 @@
     if not conf.env.CXX:
         return
 
-    conf.check_cxx(header_name='mac/All.h')
-    conf.check_cc(lib="mac", uselib_store="mac")
-    conf.check_cxx(lib='mac', fragment = _pointer_test)
+    conf.check_cxx(header_name='MAC/All.h')
+    conf.check_cc(lib="MAC", uselib_store="mac")
+    conf.check_cxx(lib='MAC', fragment = _pointer_test)
 
 configure, build = plugin('mac', configure=plugin_configure,
                           source=['mac.cpp', 'source_adapter.cpp'],
-                          libs=["mac"], tool="cxx")
-
+                          libs=["MAC"], tool="cxx")
 
 
 _pointer_test = """
-#include <mac/All.h>
-#include <mac/MACLib.h>
+#include <MAC/All.h>
+#include <MAC/MACLib.h>
 
 typedef char __CHECK__[1-2*!(  sizeof(CreateIAPEDecompressEx2 (0, -1, -1, 0)->GetInfo (APE_INFO_TAG)) >= sizeof (void *)  )];
 
