--- src/plugins/sc68/wscript.orig	2026-03-17 05:49:25.000000000 +0800
+++ src/plugins/sc68/wscript	2026-05-19 10:01:01.000000000 +0800
@@ -1,7 +1,12 @@
+from waflib import Errors
 from waftools.plugin import plugin
 
 def plugin_configure(conf):
-    conf.check_cfg(path="sc68-config", package="sc68", uselib_store="sc68",
-            atleast_version="2.3.0", args="--cflags --libs")
+    try:
+        conf.check_cfg(path="sc68-config", package="sc68", uselib_store="sc68",
+                atleast_version="2.3.0", args="--cflags --libs")
+    except Errors.ConfigurationError:
+        conf.check_cc(header_name="sc68.h")
+        conf.check_cc(lib="sc68", uselib_store="sc68")
 
 configure, build = plugin('sc68', configure=plugin_configure, libs=["sc68"])
