From d050519bec14411215f0e7a23e9b733611995a76 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 25 Jun 2025 16:00:56 +0800
Subject: [PATCH] Makefile: fixes for macOS

---
 Makefile | 35 ++++++++++++++++++++++++++---------
 1 file changed, 26 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index 087c600..b2795f6 100644
--- Makefile
+++ Makefile
@@ -1,10 +1,10 @@
 # Compiler and flags
 # clang is also supported
-CC = gcc
-LDFLAGS = -lpcap -lm
+CC ?= gcc
+LDFLAGS += -lpcap -lm
 LDFLAGS += -L. -lnetdump
 
-CFLAGS = -g -O2
+CFLAGS += -g -O2
 #CFLAGS += -std=c99
 #CFLAGS += -Wpedantic -pedantic-errors
 CFLAGS += -Werror
@@ -39,6 +39,8 @@ CFLAGS += -Wconversion -Wsign-conversion
 #CFLAGS += -fsanitize=address,undefined
 #CFLAGS += -O0
 
+UNAME_S := $(shell uname -s)
+
 # Source files (recursive find using shell, for portability)
 ifeq ($(origin MAKE_VERSION), undefined)
   # BSD Make
@@ -52,13 +54,28 @@ endif
 
 LIB_SRC = utils/protocol.c utils/visualizer.c
 LIB_OBJ = ${LIB_SRC:.c=.o}
-LIB_EXT = .so
+
+ifeq ($(UNAME_S),Darwin)
+	LIB_EXT = .dylib
+	SHAREDFLAG = -dynamiclib
+	INSTALLNAME = -Wl,-install_name,$(PREFIX)/lib/$(LIB_TARGET)
+else
+	LIB_EXT = .so
+	SHAREDFLAG = -shared
+	RPATH = -Wl,-rpath,\$$ORIGIN:${LIBDIR}
+endif
+
 LIB_TARGET = libnetdump${LIB_EXT}
 TARGET = netdump
 
-PREFIX = /usr/local
+PREFIX ?= /usr/local
 BINDIR = $(PREFIX)/bin
-LIBDIR = $(PREFIX)/lib64
+
+ifeq ($(UNAME_S),Linux)
+	LIBDIR = $(PREFIX)/lib64
+else
+	LIBDIR = $(PREFIX)/lib
+endif
 
 # Default target
 all: ${TARGET} ${LIB_TARGET}
@@ -75,11 +92,11 @@ utils/visualizer.o: utils/visualizer.c
 
 # Create shared library
 ${LIB_TARGET}: ${LIB_OBJ}
-	${CC} -shared -o $@ ${LIB_OBJ}
+	${CC} ${SHAREDFLAG} -o $@ ${LIB_OBJ} ${INSTALLNAME}
 
 # Link target with shared library
 ${TARGET}: ${OBJ} ${LIB_TARGET}
-	${CC} -o $@ ${OBJ} ${LDFLAGS} -Wl,-rpath,\$$ORIGIN:${LIBDIR}
+	${CC} -o $@ ${OBJ} ${LDFLAGS} ${RPATH}
 
 show-config:
 	@echo "Configuration:"
@@ -111,4 +128,4 @@ install: ${TARGET} ${LIB_TARGET}
 # Remove
 remove: clean
 	rm -f ${DESTDIR}${BINDIR}/${TARGET}
-	rm -f ${DESTDIR}${LIBDIR}/${LIB_TARGET}
\ No newline at end of file
+	rm -f ${DESTDIR}${LIBDIR}/${LIB_TARGET}
