--- Makefile
+++ Makefile	2025-04-27 23:40:28.000000000 +0800
@@ -1,15 +1,16 @@
 SRC := $(wildcard *.c **/*.c ***/**/*.c)
 OBJ := $(SRC:.c=.o)
+PREFIX ?= /opt/local
 
 # Compiler and flags
-CC := gcc
-CFLAGS := -fPIC -g -w
-LDFLAGS := -lpcap -lm
+# CC := gcc
+CFLAGS += -fPIC -g -w
+LDFLAGS += -lpcap -lm
 
 # Library source files
 LIB_SRC := utils/protocol.c utils/visualizer.c protocols/proto_tables_nums.c
 LIB_OBJ := $(LIB_SRC:.c=.o)
-LIB_TARGET := libnetdump.so
+LIB_TARGET := libnetdump.dylib
 
 # Binary output
 TARGET := netdump
@@ -23,11 +24,11 @@
 
 # Create shared library
 $(LIB_TARGET): $(LIB_OBJ)
-	$(CC) -shared -o $@ $^ -fPIC
+	$(CC) -dynamiclib -undefined dynamic_lookup -o $@ $^ -fPIC -Wl,-install_name,$(PREFIX)/lib/$(LIB_TARGET)
 
 # Compile netdump, linking dynamically to libnetdump.so
 $(TARGET): $(OBJ) $(LIB_TARGET)
-	$(CC) -o $(TARGET) $(OBJ) -L. -lnetdump $(LDFLAGS) -Wl,-rpath,\$$ORIGIN:/usr/local/lib64
+	$(CC) -o $(TARGET) $(OBJ) -L. -lnetdump $(LDFLAGS)
 
 # Clean target
 clean:
@@ -44,13 +45,13 @@
 
 # Install target: install netdump and libnetdump.so
 install: $(TARGET) $(LIB_TARGET)
-	mkdir -p $(DESTDIR)/usr/local/bin
-	mkdir -p $(DESTDIR)/usr/local/lib64
-	install -m 0755 $(TARGET) $(DESTDIR)/usr/local/bin/$(TARGET)
-	install -m 0755 $(LIB_TARGET) $(DESTDIR)/usr/local/lib64/$(LIB_TARGET)
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/lib
+	install -m 0755 $(TARGET) $(DESTDIR)$(PREFIX)/bin/$(TARGET)
+	install -m 0755 $(LIB_TARGET) $(DESTDIR)$(PREFIX)/lib/$(LIB_TARGET)
 
 # Remove target: remove netdump, libnetdump.so and .o files
 remove:
 	make clean
-	rm -f $(DESTDIR)/usr/local/bin/$(TARGET)
-	rm -f $(DESTDIR)/usr/local/lib64/$(LIB_TARGET)
+	rm -f $(DESTDIR)$(PREFIX)/bin/$(TARGET)
+	rm -f $(DESTDIR)$(PREFIX)/lib/$(LIB_TARGET)
