--- osx/memory.cc.orig	2026-06-26 02:37:41.000000000 +0800
+++ osx/memory.cc	2026-09-27 17:32:14.000000000 +0800
@@ -20,10 +20,23 @@
 #include <sstream>
 #include <mach/mach.h>
 #include <sys/sysctl.h> // for sysctl
+#include <AvailabilityMacros.h>
 
 #include "memory.h"
 #include "conversions.h"
 
+#ifdef HOST_VM_INFO64
+#define OSX_HOST_STATISTICS host_statistics64
+#define OSX_HOST_VM_INFO HOST_VM_INFO64
+#define OSX_HOST_VM_INFO_COUNT HOST_VM_INFO64_COUNT
+#define OSX_VM_STATISTICS_DATA_T vm_statistics64_data_t
+#else
+#define OSX_HOST_STATISTICS host_statistics
+#define OSX_HOST_VM_INFO HOST_VM_INFO
+#define OSX_HOST_VM_INFO_COUNT HOST_VM_INFO_COUNT
+#define OSX_VM_STATISTICS_DATA_T vm_statistics_data_t
+#endif
+
 void mem_status( MemoryStatus & status )
 {
   // These values are in bytes
@@ -37,7 +50,7 @@
   //u_int64_t unused_mem;
 
   vm_size_t page_size;
-  vm_statistics64_data_t vm_stats; // Use 64-bit VM statistics
+  OSX_VM_STATISTICS_DATA_T vm_stats; // Use 64-bit VM statistics
 
   // Get total physical memory
   int mib[] = { CTL_HW, HW_MEMSIZE };
@@ -47,17 +60,21 @@
   mach_port_t mach_port = mach_host_self();
 
   // Use HOST_VM_INFO64_COUNT as the count for vm_statistics64_data_t
-  mach_msg_type_number_t count = HOST_VM_INFO64_COUNT;
+  mach_msg_type_number_t count = OSX_HOST_VM_INFO_COUNT;
 
   if( KERN_SUCCESS == host_page_size( mach_port, &page_size ) &&
-      KERN_SUCCESS == host_statistics64( mach_port, HOST_VM_INFO64,
+      KERN_SUCCESS == OSX_HOST_STATISTICS( mach_port, OSX_HOST_VM_INFO,
         ( host_info_t )&vm_stats, &count )
     )
   {
     //unused_mem = static_cast<u_int64_t>( vm_stats.free_count * page_size );
     // app_mem = static_cast<float>(( vm_stats.internal_page_count - vm_stats.purgeable_count ) * page_size );
     // cached_file_mem = static_cast<float>(( vm_stats.purgeable_count + vm_stats.external_page_count ) * page_size );
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1090
+    available_mem = static_cast<float>(( vm_stats.free_count - vm_stats.speculative_count ) * page_size );
+#else
     available_mem = static_cast<float>(( vm_stats.free_count + vm_stats.external_page_count - vm_stats.speculative_count ) * page_size );
+#endif
   }
 
   status.used_mem = convert_unit(static_cast< float >(total_mem - available_mem ), MEGABYTES );
