--- wsutil/crash_info.c	2025-06-05 02:25:24.000000000 +0800
+++ wsutil/crash_info.c	2025-06-24 16:59:31.000000000 +0800
@@ -68,6 +68,11 @@
  *
  * It still appears to work as of OS X 10.8 (Mountain Lion).
  */
+
+#if defined(__GNUC__) && !defined(__private_extern__)
+#define __private_extern__ __attribute__((visibility("hidden")))
+#endif
+
 __private_extern__ char *__crashreporter_info__ = NULL;
 
 #if 0
