--- CMakeLists.txt	2026-06-11 02:07:51.000000000 +0800
+++ CMakeLists.txt	2026-06-22 15:16:13.000000000 +0800
@@ -802,24 +802,8 @@
 else() # ! MSVC
 	if(APPLE)
 		# MIN_MACOS_VERSION is used to set LSMinimumSystemVersion
-		# in Info.plist, so start with something low.
-		set(MIN_MACOS_VERSION 10.13)
-		if(CMAKE_OSX_DEPLOYMENT_TARGET)
-			if(CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS MIN_MACOS_VERSION)
-				message(FATAL_ERROR "We don't support building for macOS < ${MIN_MACOS_VERSION}")
-			endif()
-			set(MIN_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET})
-		endif()
-		find_program(ACTOOL_EXECUTABLE actool)
-		execute_process(
-			COMMAND ${ACTOOL_EXECUTABLE} --version
-			COMMAND xmllint --xpath "//dict/key[text()=\"short-bundle-version\"]/following-sibling::string/text()" -
-			OUTPUT_VARIABLE _actool_version
-		)
-		if(_actool_version VERSION_GREATER_EQUAL "26.0")
-			set(HAVE_LIQUID_GLASS_ICONS True)
-		endif()
-		unset(_actool_version)
+		# in Info.plist, so start with something correct.
+		set(MIN_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET})
 	endif()
 
 	#
