--- CMakeLists.txt	2025-10-09 00:55:14.000000000 +0800
+++ CMakeLists.txt	2025-10-22 07:25:24.000000000 +0800
@@ -676,15 +676,8 @@
 else() # ! MSVC
 	if(APPLE)
 		# MIN_MACOS_VERSION is used to set LSMinimumSystemVersion
-		# in Info.plist, so start with something low.
-		set(MIN_MACOS_VERSION 10.13)
-		if(CMAKE_OSX_DEPLOYMENT_TARGET)
-			if(CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS MIN_MACOS_VERSION)
-				message(FATAL_ERROR "We don't support building for macOS < ${MIN_MACOS_VERSION}")
-			endif()
-			set(MIN_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET})
-		endif()
-		find_program(ICONUTIL_EXECUTABLE iconutil)
+		# in Info.plist, so start with something correct.
+		set(MIN_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET})
 	endif()
 
 	#
