From ce2c9e8a4ad1e78df6dcd82188d247990be00f0c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 13 May 2025 03:16:34 +0800
Subject: [PATCH 1/4] Unbreak bundle

---
 hw/xquartz/bundle/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/hw/xquartz/bundle/meson.build b/hw/xquartz/bundle/meson.build
index 22941203b..29d0e5fce 100644
--- a/hw/xquartz/bundle/meson.build
+++ b/hw/xquartz/bundle/meson.build
@@ -55,5 +55,3 @@ install_data('PkgInfo',
 install_data('X11.sh',
     install_dir: join_paths(bundle_root, 'Contents/MacOS'),
     install_mode: 'rwxr-xr-x')
-
-meson.add_install_script('chown-bundle.sh', bundle_root)
-- 
2.49.0

