From 10ce447f6d9ce7b45a33ef37f6703522301c2dae Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 29 Aug 2024 13:49:34 +0800
Subject: [PATCH 2/4] x-list.h: replace __private_extern__ with working
 __attribute__

---
 hw/xquartz/xpr/x-list.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xquartz/xpr/x-list.h b/hw/xquartz/xpr/x-list.h
index 28385fd99..317be0a4a 100644
--- a/hw/xquartz/xpr/x-list.h
+++ b/hw/xquartz/xpr/x-list.h
@@ -45,7 +45,7 @@ struct x_list_struct {
 #endif
 
 #ifndef X_EXTERN
-#define X_EXTERN __private_extern__
+#define X_EXTERN __attribute__((visibility("hidden")))
 #endif
 
 X_EXTERN void X_PFX(list_free_1) (x_list * node);
-- 
2.49.0

