From 6197a6063794bcab77d7469839b22b01de43355a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 30 May 2025 01:30:59 +0800
Subject: [PATCH 4/9] include/glx_extinit.h: fix for pre-C11 compilers

---
 include/glx_extinit.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/glx_extinit.h b/include/glx_extinit.h
index 07f3cc855..a0ecd47e5 100644
--- a/include/glx_extinit.h
+++ b/include/glx_extinit.h
@@ -30,7 +30,11 @@
 /* XXX this comment no longer makes sense i think */
 #ifdef GLXEXT
 typedef struct __GLXprovider __GLXprovider;
+
+#ifndef _GLX_screens_h_
 typedef struct __GLXscreen __GLXscreen;
+#endif
+
 struct __GLXprovider {
     __GLXscreen *(*screenProbe) (ScreenPtr pScreen);
     const char *name;
@@ -44,5 +48,4 @@ Bool xorgGlxCreateVendor(void);
 static inline Bool xorgGlxCreateVendor(void) { return TRUE; }
 #endif
 
-
 #endif
-- 
2.24.3 (Apple Git-128)

