From 55edd6c805e3a4c8ca666416f36d943c4780c3b6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 30 May 2025 02:01:35 +0800
Subject: [PATCH 5/9] os/inputthread.c: fix compiling

---
 os/inputthread.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/os/inputthread.c b/os/inputthread.c
index bd0a8335f..3469cfc1c 100644
--- a/os/inputthread.c
+++ b/os/inputthread.c
@@ -479,12 +479,6 @@ InputThreadInit(void)
     if (pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM) != 0)
         ErrorF("input-thread: error setting thread scope\n");
 
-#ifdef __APPLE__
-    if (&pthread_attr_set_qos_class_np) {
-        pthread_attr_set_qos_class_np(&attr, QOS_CLASS_USER_INTERACTIVE, 0);
-    }
-#endif
-
     DebugF("input-thread: creating thread\n");
     pthread_create(&inputThreadInfo->thread, &attr,
                    &InputThreadDoWork, NULL);
-- 
2.24.3 (Apple Git-128)

