From ad240045484ad92c5a26fa58ebb75117fab5752e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 30 May 2025 03:39:26 +0800
Subject: [PATCH 7/9] x-list.h: fix for private_extern

---
 hw/xquartz/xpr/x-list.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hw/xquartz/xpr/x-list.h b/hw/xquartz/xpr/x-list.h
index 28385fd99..317be0a4a 100644
--- a/hw/xquartz/xpr/x-list.h
+++ b/hw/xquartz/xpr/x-list.h
@@ -45,7 +45,7 @@ struct x_list_struct {
 #endif
 
 #ifndef X_EXTERN
-#define X_EXTERN __private_extern__
+#define X_EXTERN __attribute__((visibility("hidden")))
 #endif
 
 X_EXTERN void X_PFX(list_free_1) (x_list * node);
-- 
2.24.3 (Apple Git-128)

