From f4815b7b39e364d478a708361c68e2ffb7a25cfc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 20 Dec 2024 01:47:54 +0800
Subject: [PATCH 1/6] prefix.xcconfig: add ppc

---
 prefix.xcconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git prefix.xcconfig prefix.xcconfig
index 51edead..e6272fd 100644
--- prefix.xcconfig
+++ prefix.xcconfig
@@ -35,4 +35,4 @@ DATDIR = $(PREFIX)/share
 INCDIR = $(PREFIX)/include
 DEPLOYMENT_LOCATION = YES
 DSTROOT = $(BUILT_PRODUCTS_DIR)
-ARCHS = x86_64 i386
+ARCHS = x86_64 i386 ppc
-- 
2.48.0-rc0

