From fa651ef1761e4ca5a669f46eb7d2a84d622e4231 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 22 Dec 2024 03:38:13 +0800
Subject: [PATCH 4/6] darwinbuild.in: fix for renamed sources from Apple
 releases

---
 darwinbuild/darwinbuild.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git darwinbuild/darwinbuild.in darwinbuild/darwinbuild.in
index ad791d4..f59087f 100755
--- darwinbuild/darwinbuild.in
+++ darwinbuild/darwinbuild.in
@@ -485,7 +485,7 @@ chmod 1777 "$BuildRoot/$vartmp"
 ### the BuildRoot prefix omitted because the chroot
 ### will make all paths relative to that point.
 ###
-REAL_SRCROOT="$BuildRoot/SourceCache/$projnam/$project"
+REAL_SRCROOT="$BuildRoot/SourceCache/$projnam/$projnam-$project"
 REAL_OBJROOT="$BuildRoot/$vartmp/$projnam/$project.obj"
 REAL_SYMROOT="$BuildRoot/$vartmp/$projnam/$project.sym"
 REAL_DSTROOT="$BuildRoot/$vartmp/$projnam/$project.root"
@@ -500,7 +500,7 @@ if [ "$nosource" != "YES" ]; then
 	rm -Rf "$REAL_SRCROOT" "$REAL_OBJROOT" "$REAL_SYMROOT" "$REAL_DSTROOT"
 	mkdir -p "$REAL_SRCROOT" "$REAL_OBJROOT" "$REAL_SYMROOT" "$REAL_DSTROOT"
 	chown root:wheel "$REAL_SRCROOT" "$REAL_OBJROOT" "$REAL_SYMROOT" "$REAL_DSTROOT"
-	
+
 	###
 	### Install the sources and patches into the BuildRoot
 	###
-- 
2.48.0-rc0

