From 5a6eb0f24887b9ea9e6050f390cf74ca7f7864cf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 24 Dec 2024 06:44:37 +0800
Subject: [PATCH 6/6] darwinbuild.in: fix alias projects

---
 darwinbuild/darwinbuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git darwinbuild/darwinbuild.in darwinbuild/darwinbuild.in
index 4263814..c24be03 100755
--- darwinbuild/darwinbuild.in
+++ darwinbuild/darwinbuild.in
@@ -515,7 +515,7 @@ if [ "$nosource" != "YES" ]; then
 	elif [ "$alias" != "" ]; then
 		tar xzf "$SourceCache/$alias-$version.tar.gz"
 		rmdir "$REAL_SRCROOT"
-		ln -fhs "$alias-$version" "$project"
+		ln -fhs "$alias-$alias-$version" "$projnam-$project"
 	else
 		tar xzf "$SourceCache/$filename"
 	fi
