From d72ddb76c9ea2e5734cdc93cc55e91b34cdeef11 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 22 Dec 2024 03:55:32 +0800
Subject: [PATCH] darwinup: a hack for 10a190

---
 darwinup/Database.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git darwinup/Database.cpp darwinup/Database.cpp
index c787d3d..a016696 100644
--- darwinup/Database.cpp
+++ darwinup/Database.cpp
@@ -31,6 +31,7 @@
  */
 
 #include "Database.h"
+#include <AvailabilityMacros.h>
 
 /**
  * sqlite3_trace callback for debugging
@@ -931,13 +932,20 @@ int Database::step_all(sqlite3_stmt* stmt, void** output, uint32_t size,
  */
 void Database::init_cache() {
 	cache_attributes_t attrs;
+#if MAC_OS_X_VERSION_MAX_ALLOWED > 1060
+    /* Defined in 10.6.8, not defined in 10a190. */
 	attrs.version = CACHE_ATTRIBUTES_VERSION_2;
+#else
+	attrs.version = CACHE_ATTRIBUTES_VERSION_1;
+#endif
 	attrs.key_hash_cb = cache_key_hash_cb_cstring;
 	attrs.key_is_equal_cb = cache_key_is_equal_cb_cstring;
 	attrs.key_retain_cb = cache_key_retain;
 	attrs.key_release_cb = cache_release_cb_free;
 	attrs.value_release_cb = cache_statement_release;
+#if MAC_OS_X_VERSION_MAX_ALLOWED > 1060
 	attrs.value_retain_cb = NULL;
+#endif
 	attrs.value_make_purgeable_cb = NULL;
 	attrs.value_make_nonpurgeable_cb = NULL;
 	attrs.user_data = NULL;
