--- include/grpc/event_engine/memory_request.h	2026-02-01 01:20:31.000000000 +0800
+++ include/grpc/event_engine/memory_request.h	2026-02-01 01:55:33.000000000 +0800
@@ -16,6 +16,7 @@
 
 #include <grpc/support/port_platform.h>
 #include <stddef.h>
+#include <string>
 
 #include "absl/strings/string_view.h"
 
--- src/core/util/glob.cc	2026-02-01 01:20:31.000000000 +0800
+++ src/core/util/glob.cc	2026-02-01 03:02:44.000000000 +0800
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <algorithm>
+
 #include "absl/strings/string_view.h"
 
 namespace grpc_core {

--- src/core/channelz/v2tov1/property_list.cc	2026-02-01 01:20:31.000000000 +0800
+++ src/core/channelz/v2tov1/property_list.cc	2026-02-01 03:30:58.000000000 +0800
@@ -15,6 +15,7 @@
 #include "src/core/channelz/v2tov1/property_list.h"
 
 #include <cstdint>
+#include <limits>
 #include <optional>
 #include <string>
 
