From 89a40634a79fa2d6502b4d75c5595786c9de3456 Mon Sep 17 00:00:00 2001
From: Nils Werner <nils@hey.com>
Date: Thu, 13 Jun 2024 10:31:09 +0200
Subject: [PATCH] Fix netifaces and configparser for Python 3

---
 setup.py      | 2 +-
 zget/utils.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index e041784..6fbfd6a 100644
--- setup.py
+++ setup.py
@@ -42,7 +42,7 @@ def run(self):
         install_requires=[
             'six',
             'zeroconf<0.17.6',
-            'netifaces<0.10.5',
+            'netifaces>=0.11',
             'progressbar2',
             'requests',
         ],
diff --git a/zget/utils.py b/zget/utils.py
index 3f13abe..6e3f12d 100644
--- zget/utils.py
+++ zget/utils.py
@@ -108,7 +108,7 @@
 def config():
     """ Reads config values from zget.cfg or zget.ini
     """
-    config = configparser.SafeConfigParser(
+    config = configparser.ConfigParser(
         defaults={
             'port': '0',
             'interface': None,
