From c286db15dac442655c82bd43267fae2fc2d3a3c5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 25 Sep 2025 03:53:08 +0800
Subject: [PATCH] vm_dump.c: unbreak unwind on powerpc

---
 vm_dump.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/vm_dump.c b/vm_dump.c
index 2a863ddef955a0..f8597a252129dc 100644
--- a/vm_dump.c
+++ b/vm_dump.c
@@ -511,7 +511,7 @@ rb_vmdebug_thread_dump_state(FILE *errout, VALUE self)
 #  include <sys/mman.h>
 #  undef backtrace
 
-#  if defined(__arm64__)
+#  if defined(__arm64__) || defined(__POWERPC__)
 static bool
 is_coroutine_start(unw_word_t ip)
 {
@@ -637,6 +637,16 @@ backtrace(void **trace, int size)
         if (is_coroutine_start(ip)) break;
     }
     return n;
+
+#  elif defined(__POWERPC__)
+    while (unw_step(&cursor) > 0) {
+        unw_get_reg(&cursor, UNW_REG_IP, &ip);
+        trace[n++] = (void *)ip;
+
+        // Apple's libunwind can't handle our coroutine switching code
+        if (is_coroutine_start(ip)) break;
+    }
+    return n;
 #  else
 #    error unsupported architecture
 #  endif
