From 39c086c50fbd89b0006a031b999340085c406a06 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 25 Sep 2025 04:03:37 +0800
Subject: [PATCH] version.c: define undefined RUBY_API_VERSION_NAME

---
 version.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/version.c b/version.c
index d634755efd16c0..56cfa67eb41955 100644
--- a/version.c
+++ b/version.c
@@ -36,6 +36,7 @@
 #else
 # define RUBY_REVISION "HEAD"
 # define RUBY_REVISION_STR ""
+# define RUBY_API_VERSION_NAME "master"
 #endif
 #if !defined RUBY_RELEASE_DATETIME || RUBY_PATCHLEVEL != -1
 # undef RUBY_RELEASE_DATETIME
