+++ libltdl/configure	2024-09-25 21:52:26.000000000 +0800
+++ libltdl/configure	2025-01-01 23:38:04.000000000 +0800
@@ -16390,7 +16390,7 @@
 lock_old_archive_extraction=$lock_old_archive_extraction
 
 # A C compiler.
-LTCC=$lt_CC
+LTCC=\$CC
 
 # LTCC compiler flags.
 LTCFLAGS=$lt_CFLAGS
@@ -16533,7 +16533,7 @@
 
 
 # The linker used to build libraries.
-LD=$lt_LD
+LD=\$LD
 
 # How to create reloadable object files.
 reload_flag=$lt_reload_flag
@@ -16543,7 +16543,7 @@
 old_archive_cmds=$lt_old_archive_cmds
 
 # A language specific compiler.
-CC=$lt_compiler
+CC=\$CC
 
 # Is the compiler the GNU compiler?
 with_gcc=$GCC
