--- Makefile	2025-03-12 04:03:43.000000000 +0800
+++ Makefile	2025-03-15 08:38:55.000000000 +0800
@@ -117,7 +117,7 @@
 	${CC} -o $@ -c sfeed_curses.c ${SFEED_CURSES_CFLAGS} ${SFEED_CURSES_CPPFLAGS}
 
 sfeed_curses: ${LIB} sfeed_curses.o
-	${CC} -o $@ sfeed_curses.o ${LIB} ${SFEED_CURSES_LDFLAGS}
+	${CC} -o $@ sfeed_curses.o ${LIB} ${SFEED_CURSES_LDFLAGS} ${SFEED_LDFLAGS}
 
 ${LIBUTIL}: ${LIBUTILOBJ}
 	${AR} -rc $@ $?
