--- gcc/config/darwin.h.orig	2026-08-04 00:30:54.000000000 -0600
+++ gcc/config/darwin.h	2026-08-04 00:31:07.000000000 -0600
@@ -396,7 +396,7 @@
 
 #define DARWIN_NOCOMPACT_UNWIND \
 "%{!fuse-ld=lld: \
-    %:version-compare(>= 10.6 mmacosx-version-min= -no_compact_unwind)}"
+    -no_compact_unwind}"
 
 /* In Darwin linker specs we can put -lcrt0.o and ld will search the library
    path for crt0.o or -lcrtx.a and it will search for libcrtx.a.  As for
