--- a/meson.build	2025-03-08 20:35:35.000000000 +0800
+++ b/meson.build	2026-04-30 22:07:24.000000000 +0800
@@ -458,6 +458,7 @@
 
     feature_conf.set('CAIRO_HAS_QUARTZ_SURFACE', 1)
     feature_conf.set('CAIRO_HAS_QUARTZ_IMAGE_SURFACE', 1)
+    feature_conf.set('CAIRO_HAS_QUARTZ_FONT', 1)
 
     built_features += [
       {
@@ -469,17 +470,16 @@
         'name': 'cairo-quartz-image',
         'description': 'Quartz Image surface backend',
         'deps': quartz_deps,
+      },
+      {
+        'name': 'cairo-quartz-font',
+        'description': 'Quartz font backend',
+        'deps': quartz_deps,
       }]
     compiler = meson.get_compiler('c')
     if compiler.has_function('CTFontDrawGlyphs', prefix: '#include <ApplicationServices/ApplicationServices.h>',
                              dependencies: quartz_deps)
-      built_features += [
-           {
-               'name': 'cairo-quartz-font',
-               'description': 'Quartz font backend',
-               'deps': quartz_deps,
-           }]
-      feature_conf.set('CAIRO_HAS_QUARTZ_FONT', 1)
+      feature_conf.set('CAIRO_HAS_CORE_TEXT', 1)
     endif
   endif
 endif
