From 3c3b3d7c52e088172bd5ef492b1a9065feec26f4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 3 Jan 2025 13:57:58 +0800
Subject: [PATCH] make-config.sh: enable dispatch on 10.6 ppc

---
 make-config.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git make-config.sh make-config.sh
index dfd2b6125..a03a22c7c 100755
--- make-config.sh
+++ make-config.sh
@@ -642,7 +642,7 @@
         printf ' :unix :bsd :darwin :mach-o' >> $ltf
         darwin_version=`uname -r`
         darwin_version_major=${DARWIN_VERSION_MAJOR:-${darwin_version%%.*}}
-        if (( 10 > $darwin_version_major )) || [ $sbcl_arch = "ppc" ]; then
+        if (( 10 > $darwin_version_major )); then
             printf ' :use-darwin-posix-semaphores' >> $ltf
         else
             printf ' :os-provides-pthread-setname-np' >> $ltf
