--- libarchive/archive_disk_acl_darwin.c.orig	2026-08-07 20:53:02.000000000 -0600
+++ libarchive/archive_disk_acl_darwin.c	2026-08-07 23:01:40.000000000 -0600
@@ -26,7 +26,8 @@
 #include "archive_platform.h"
 
 #if ARCHIVE_ACL_DARWIN

+/* Detect that Mac OS X 10.4  ACL support  does not have acl_set_link_np */
+#include <AvailabilityMacros.h>
 #ifdef HAVE_FCNTL_H
 #include <fcntl.h>
 #endif
@@ -473,6 +474,11 @@
 				ret = ARCHIVE_WARN;
 			}
 		}
+/* Mac OS X 10.4 does not support setting ACLs on symlinks */
+/* Reference: https://www.shoup.net/xbup/doc.pdf */
+#if MAC_OS_X_VERSION_MIN_REQUIRED == 1040
+        }
+#else
 	} else if (acl_set_link_np(name, ACL_TYPE_EXTENDED, acl) != 0) {
 		if (errno == EOPNOTSUPP) {
 			/* Filesystem doesn't support ACLs */
@@ -483,6 +489,7 @@
 			ret = ARCHIVE_WARN;
 		}
 	}
+#endif
 exit_free:
 	acl_free(acl);
 	return (ret);
