commit 04f2cbc98ce50ee1ea86801b6c40eac7f27bf2d0
Author: Dag-Erling Smørgrav <des@des.dev>
Date:   Wed Jun 24 17:55:41 2026 +0200

    archive_write_disk_posix: Symlinks can have AppleDouble data

diff --git b/libarchive/archive_write_disk_posix.c a/libarchive/archive_write_disk_posix.c
index a27e7318..9a6098da 100644
--- libarchive/archive_write_disk_posix.c
+++ libarchive/archive_write_disk_posix.c
@@ -4350,7 +4350,7 @@ copy_metadata(struct archive_write_disk *a, const char *metadata,
 		 * with at least a writing mode(O_RDWR or O_WRONLY). it
 		 * makes the data fork uncompressed.
 		 */
-		dffd = open(datafork, O_SYMLINK);
+		dffd = open(datafork, 0);
 		if (dffd == -1) {
 			archive_set_error(&a->archive, errno,
 			    "Failed to open the data fork for metadata");
@@ -4459,9 +4459,7 @@ fixup_appledouble(struct archive_write_disk *a, const char *pathname)
 	if (la_stat(datafork.s, &st) == -1)
 		goto skip_appledouble;
 #endif
-	if (!S_ISREG(st.st_mode) &&
-	    !S_ISDIR(st.st_mode) &&
-	    !S_ISLNK(st.st_mode))
+	if (!S_ISREG(st.st_mode) && !S_ISDIR(st.st_mode))
 		goto skip_appledouble;
 
 	/*
