--- build-aux/meson/postinstall.py	2018-03-25 18:31:26.000000000 +0800
+++ build-aux/meson/postinstall.py	2025-07-01 09:05:02.000000000 +0800
@@ -1,9 +1,9 @@
-#!/usr/bin/env python3
+#!@PYTHON@
 
 import os
 import subprocess
 
-prefix = os.environ.get('MESON_INSTALL_PREFIX', '/usr/local')
+prefix = os.environ.get('MESON_INSTALL_PREFIX', '@PREFIX@')
 datadir = os.path.join(prefix, 'share')
 
 # Packaging tools define DESTDIR and this isn't needed for them
