--- meson.build	2018-03-25 18:31:26.000000000 +0800
+++ meson.build	2025-07-01 08:47:33.000000000 +0800
@@ -53,7 +53,7 @@
 gsettings_desktop_schemas = dependency ('gsettings-desktop-schemas',
                                         version: schemas_ver)
 
-libm = cc.find_library ('libm')
+libm = cc.find_library ('m')
 
 conf = configuration_data ()
 conf.set_quoted ('VERSION', meson.project_version ())

--- data/meson.build	2018-03-25 18:31:26.000000000 +0800
+++ data/meson.build	2025-07-01 08:59:28.000000000 +0800
@@ -6,24 +6,21 @@
                                  'lineup-parameters.c',
                                  dependencies: [gio, gio_unix])
 
-desktop = i18n.merge_file ('desktop',
-                           input: 'org.gnome.Nautilus.desktop.in',
+desktop = i18n.merge_file (input: 'org.gnome.Nautilus.desktop.in',
                            output: 'org.gnome.Nautilus.desktop',
                            install: true,
                            install_dir: desktopdir,
                            po_dir: po_dir,
                            type: 'desktop')
 
-desktop_autorun_software = i18n.merge_file ('desktop-autorun-software',
-                                            input: 'nautilus-autorun-software.desktop.in',
+desktop_autorun_software = i18n.merge_file (input: 'nautilus-autorun-software.desktop.in',
                                             output: 'nautilus-autorun-software.desktop',
                                             install: true,
                                             install_dir: desktopdir,
                                             po_dir: po_dir,
                                             type: 'desktop')
 
-desktop_autostart = i18n.merge_file ('desktop-autostart',
-                                     input: 'nautilus-autostart.desktop.in',
+desktop_autostart = i18n.merge_file (input: 'nautilus-autostart.desktop.in',
                                      output: 'nautilus-autostart.desktop',
                                      install: true,
                                      install_dir: join_paths (sysconfdir, 'xdg', 'autostart'),
@@ -31,8 +28,7 @@
                                      type: 'desktop')
 
 if get_option ('enable-desktop')
-    desktop_classic = i18n.merge_file ('desktop-classic',
-                                       input: 'nautilus-classic.desktop.in',
+    desktop_classic = i18n.merge_file (input: 'nautilus-classic.desktop.in',
                                        output: 'nautilus-classic.desktop',
                                        install: true,
                                        install_dir: desktopdir,
@@ -40,8 +36,7 @@
                                        type: 'desktop')
 endif
 
-appdata = i18n.merge_file ('appdata',
-                           input: 'org.gnome.Nautilus.appdata.xml.in',
+appdata = i18n.merge_file (input: 'org.gnome.Nautilus.appdata.xml.in',
                            output: 'org.gnome.Nautilus.appdata.xml',
                            install: true,
                            install_dir: appdatadir,
