--- src/nautilus-filename-utilities.c	2026-04-12 08:10:06.000000000 +0800
+++ src/nautilus-filename-utilities.c	2026-05-07 14:58:01.000000000 +0800
@@ -336,7 +336,15 @@
     /* basename must have at least one char */
     const char *start = g_utf8_next_char (filename);
     size_t search_length = strlen (start);
-    const char *extension = memrchr (start, '.', search_length);
+    const char *extension = NULL;
+    for (const char *p = start + search_length; p > start; p--)
+    {
+        if (*(p - 1) == '.')
+        {
+            extension = p - 1;
+            break;
+        }
+    }
 
     if (extension == NULL || *g_utf8_next_char (extension) == '\0')
     {
