From ae858b4fc418c5e59fa84a2cbce2ac2c187ba596 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 1 May 2025 22:36:27 +0800
Subject: [PATCH 3/3] Disable building html help

---
 Build/GNUmakefile | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git Build/GNUmakefile Build/GNUmakefile
index 0262e5ce..3f2a6145 100644
--- Build/GNUmakefile
+++ Build/GNUmakefile
@@ -191,7 +191,6 @@ IconForStackWindows.icns \
 IconForTerminal.icns \
 English.lproj/Credits.html
 APP_RESOURCES_BUNDLES := \
-English.lproj/Help \
 English.lproj/AlertMessages.nib \
 English.lproj/ColorBox.nib \
 English.lproj/GenericDialog.nib \
@@ -451,11 +450,11 @@ clean-app-dot-strings:
 
 # Component: Help
 
-$(DEST_APP_RESOURCES_TOP)/English.lproj/Help:
-	$(call banner,$@)
-	$(CURDIR)/HTMLHelpBuild.command
-	$(MKDIR_P) $(dir $@)
-	$(RSYNC) --archive $(CURDIR)/_Generated/Help_English/Help/ $@
+# $(DEST_APP_RESOURCES_TOP)/English.lproj/Help:
+# 	$(call banner,$@)
+# 	$(CURDIR)/HTMLHelpBuild.command
+# 	$(MKDIR_P) $(dir $@)
+# 	$(RSYNC) --archive $(CURDIR)/_Generated/Help_English/Help/ $@
 
 # Component: Bug Reporter
 
-- 
2.49.0

