From 1065ce7fddd2e29d66685da59fdb80f36304ab4b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 2 May 2025 00:13:20 +0800
Subject: [PATCH] Fixes to makefile

---
 Build/GNUmakefile           | 20 ++++----------------
 Build/HTMLHelpBuild.command |  2 +-
 2 files changed, 5 insertions(+), 17 deletions(-)

diff --git Build/GNUmakefile Build/GNUmakefile
index db9cc5a0..2b1cc7fc 100644
--- Build/GNUmakefile
+++ Build/GNUmakefile
@@ -195,7 +195,6 @@ IconForStackWindows.icns \
 IconForTerminal.icns \
 English.lproj/Credits.html
 APP_RESOURCES_BUNDLES := \
-MacTerm.help \
 English.lproj/AlertMessages.nib \
 English.lproj/ColorBox.nib \
 English.lproj/GenericDialog.nib \
@@ -453,24 +452,13 @@ install-app-dot-strings: $(SRC_APP_CODE_TOP)/UIStrings.cp
 clean-app-dot-strings:
 	$(foreach FILE,$(wildcard $(DEST_APP_RESOURCES_TOP)/English.lproj/*.strings),$(RM) $(FILE) ;)
 
-# Component: Help
-
-$(CURDIR)/_Generated/MacTerm.help:
-	$(call banner,$@)
-	$(CURDIR)/HTMLHelpBuild.command
-
-$(DEST_APP_RESOURCES_TOP)/MacTerm.help: $(CURDIR)/_Generated/MacTerm.help
-	$(call banner,$@)
-	$(MKDIR_P) $(dir $@)
-	$(RSYNC) --archive $(CURDIR)/_Generated/MacTerm.help/ $@
-
 # Component: Bug Reporter
 
 .PHONY: install-bug-reporter
 install-bug-reporter:
 	$(call banner,$@)
 	@# keep in sync with the matching clean rule
-	$(XCB4) -project BugReporter.xcodeproj
+	$(XCB) -project BugReporter.xcodeproj
 	@# avoid rsyncing from /
 	@if [ "x$(SRC_BUGREPORTER_TOP)" = "x" ] ; then \
 		echo "SRC_BUGREPORTER_TOP variable unset" >&2 ; \
@@ -481,7 +469,7 @@ install-bug-reporter:
 
 .PHONY: clean-bug-reporter
 clean-bug-reporter:
-	$(XCB4) clean -project BugReporter.xcodeproj
+	$(XCB) clean -project BugReporter.xcodeproj
 	$(RM) -R $(DEST_BUGREPORTER_TOP)
 
 # Component: Preferences Converter
@@ -490,7 +478,7 @@ clean-bug-reporter:
 install-prefs-converter:
 	$(call banner,$@)
 	@# keep in sync with the matching clean rule
-	$(XCB4) -project PrefsConverter.xcodeproj
+	$(XCB) -project PrefsConverter.xcodeproj
 	@# avoid rsyncing from /
 	@if [ "x$(SRC_PREFSCONVERTER_TOP)" = "x" ] ; then \
 		echo "SRC_PREFSCONVERTER_TOP variable unset" >&2 ; \
@@ -501,7 +489,7 @@ install-prefs-converter:
 
 .PHONY: clean-prefs-converter
 clean-prefs-converter:
-	$(XCB4) clean -project PrefsConverter.xcodeproj
+	$(XCB) clean -project PrefsConverter.xcodeproj
 	$(RM) -R $(DEST_PREFSCONVERTER_TOP)
 
 # Component: Miscellaneous Resources (NIBs, Images, etc.)
diff --git Build/HTMLHelpBuild.command Build/HTMLHelpBuild.command
index c1407a04..39c5503f 100755
--- Build/HTMLHelpBuild.command
+++ Build/HTMLHelpBuild.command
@@ -3,7 +3,7 @@
 # when double-clicked in the Finder, this will run a
 # terminal (preferably, MacTerm!) to run "make" for you
 
-PATH=/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:$PATH
+PATH=/Developer/usr/bin:$PATH
 export PATH
 
 cd `dirname $0` && make -f GNUmakefile_HTMLHelp
-- 
2.49.0

