--- meson.build	2025-08-20 16:14:44.000000000 +0800
+++ meson.build	2025-08-30 13:11:12.000000000 +0800
@@ -38,7 +38,6 @@
 
 fs = import('fs')
 pkg = import('pkgconfig')
-python_module = import('python')
 
 #### Variables
 
@@ -120,7 +119,7 @@
 PDFLATEX = find_program('pdflatex', required: false)
 PIP_COMPILE = find_program('pip-compile', required: false)
 PLANTUML = find_program('plantuml', required: false)
-PYTHON = find_program('python3', 'python', required: true)
+PYTHON = find_program('@PYTHON@', required: true)
 SPHINX = find_program('sphinx-build', 'sphinx-build-3', required: false)
 SUDO = find_program('sudo', required: false)
 XMLLINT = find_program('xmllint', required: false)
@@ -132,16 +131,16 @@
 
 #### sudo
 
-PASSWORDLESS_SUDO_SET_UP = false
-if SUDO.found()
-    result = run_command(SUDO, '-k', check: false)
-    if result.returncode() == 0
-        result = run_command(SUDO, '-n', 'true', check: false)
-        if result.returncode() == 0
-            PASSWORDLESS_SUDO_SET_UP = true
-        endif
-    endif
-endif
+#PASSWORDLESS_SUDO_SET_UP = false
+#if SUDO.found()
+#    result = run_command(SUDO, '-k', check: false)
+#    if result.returncode() == 0
+#        result = run_command(SUDO, '-n', 'true', check: false)
+#        if result.returncode() == 0
+#            PASSWORDLESS_SUDO_SET_UP = true
+#        endif
+#    endif
+#endif
 
 #### Configuration Data
 
@@ -338,7 +337,7 @@
 
 # Kea shell
 PKGPYTHONDIR = 'unknown'
-py_installation = python_module.find_installation('python3', required: false)
+py_installation = import('python').find_installation('@PYTHON@', required: false)
 if py_installation.found()
     PKGPYTHONDIR = py_installation.get_install_dir(pure: true) / 'kea'
 else
