--- src/a68g/single-math.c	2026-08-10 06:02:28.000000000 +0800
+++ src/a68g/single-math.c	2026-08-11 11:22:02.000000000 +0800
@@ -801,7 +801,7 @@
 // Neumaier's improved Kahan-Babuska algorithm.
 // Sum terms while preserving precision loss. 
   BOOL_T ascend = ABS (terms[0]) < ABS (terms[n - 1]); // Guess.
-  REAL_T sum = 0.0q, lost = 0.0q;
+  REAL_T sum = 0.0, lost = 0.0;
   for (int k = 0; k < n; k ++) {
     volatile REAL_T u = terms[ascend ? k : n - k -1];
     volatile REAL_T v = sum + u;
