--- src/Makefile.am.orig	2025-02-13 03:25:07.000000000 +0800
+++ src/Makefile.am	2025-04-05 16:37:35.000000000 +0800
@@ -616,9 +616,9 @@
 
 if !CROSS_COMPILING
 all-local: $(LNAV_BUILT_FILES) lnav $(RUST_DEPS)
-	if test -w $(srcdir)/internals; then \
-		env DUMP_INTERNALS_DIR=$(srcdir)/internals DUMP_CRASH=1 ./lnav Makefile; \
-		mv $(srcdir)/internals/*.schema.json $(top_srcdir)/docs/schemas; \
+	if test -w $(subdir)/internals; then \
+		env DUMP_INTERNALS_DIR=$(subdir)/internals DUMP_CRASH=1 ./lnav Makefile; \
+		mv $(subdir)/internals/*.schema.json $(top_srcdir)/docs/schemas; \
 	fi
 else
 all-local: $(LNAV_BUILT_FILES) $(RUST_DEPS)
