From c94800cf0a74f0fa1c1034bef12564da23e97d00 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 20 Mar 2025 20:38:38 +0800
Subject: [PATCH 2/2] Fix macOS headers

---
 src/tun/macos.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/tun/macos.cc src/tun/macos.cc
index ab63480..1657f4c 100644
--- src/tun/macos.cc
+++ src/tun/macos.cc
@@ -7,6 +7,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <memory>
+#include <sys/socket.h>
 #include <net/if.h>
 #include <net/if_utun.h>
 #include <net/route.h>
@@ -17,7 +18,6 @@
 #include <sys/ioctl.h>
 #include <sys/kern_control.h>
 #include <sys/select.h>
-#include <sys/socket.h>
 #include <sys/sys_domain.h>
 #include <sys/uio.h>
 #include <unistd.h>
-- 
2.48.0

