--- src/wallet/rpc/coins.cpp	2026-07-06 21:09:19.000000000 +0800
+++ src/wallet/rpc/coins.cpp	2026-08-05 19:07:37.000000000 +0800
@@ -406,9 +406,9 @@
         {},
         RPCResult{
             RPCResult::Type::OBJ, "", "",
-            {
+            std::vector<RPCResult>{
                 {RPCResult::Type::OBJ, "mine", "balances from outputs that the wallet can sign",
-                {
+                std::vector<RPCResult>{
                     {RPCResult::Type::STR_AMOUNT, "trusted", "trusted balance (outputs created by the wallet or confirmed outputs)"},
                     {RPCResult::Type::STR_AMOUNT, "untrusted_pending", "untrusted pending balance (outputs created by others that are in the mempool)"},
                     {RPCResult::Type::STR_AMOUNT, "immature", "balance from immature coinbase outputs"},

--- src/wallet/rpc/wallet.cpp	2026-07-06 21:09:19.000000000 +0800
+++ src/wallet/rpc/wallet.cpp	2026-08-05 18:33:37.000000000 +0800
@@ -38,7 +38,7 @@
                 {},
                 RPCResult{
                     RPCResult::Type::OBJ, "", "",
-                    {
+                    std::vector<RPCResult>{
                         {
                         {RPCResult::Type::STR, "walletname", "the wallet name"},
                         {RPCResult::Type::NUM, "walletversion", "(DEPRECATED) only related to unsupported legacy wallet, returns the latest version 169900 for backwards compatibility"},
