From e8c671bdb7f7fc56367ffb0b9e1ef525fcdc2ea0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 8 Jul 2026 21:33:20 +0800
Subject: [PATCH 4/5] vpx_ports update

---
 vpx_ports/ppc.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vpx_ports/ppc.h b/vpx_ports/ppc.h
index a11f4e873..8a80c0bd2 100644
--- a/vpx_ports/ppc.h
+++ b/vpx_ports/ppc.h
@@ -19,6 +19,7 @@ extern "C" {
 #endif
 
 #define HAS_VSX 0x01
+#define HAS_ALTIVEC 0x02
 
 int ppc_simd_caps(void);
 
-- 
2.55.0

