diff --git a/vp8/vp8_common.mk b/vp8/vp8_common.mk
index 842133eff..7f118937c 100644
--- a/vp8/vp8_common.mk
+++ b/vp8/vp8_common.mk
@@ -153,9 +153,9 @@ VP8_COMMON_SRCS-$(HAVE_NEON)  += common/arm/neon/sixtappredict_neon.c
 VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/idct_altivec.c
 VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/copy_altivec.c
 VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/loopfilter_altivec.c
-VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/loopfilter_filters_altivec.s
-VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/filter_altivec.s
-VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/filter_bilinear_altivec.s
-VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/recon_altivec.s
+VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/loopfilter_filters_altivec.S
+VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/filter_altivec.S
+VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/filter_bilinear_altivec.S
+VP8_COMMON_SRCS-$(HAVE_ALTIVEC) += common/ppc/recon_altivec.S
 
 $(eval $(call rtcd_h_template,vp8_rtcd,vp8/common/rtcd_defs.pl))
