--- CMakeLists.txt.orig	2025-01-09 06:02:14.000000000 +0800
+++ CMakeLists.txt	2025-10-09 09:58:12.000000000 +0800
@@ -32,6 +32,9 @@
 option(ENABLE_GOCLEAR "Enable the possibility to send and receive GoClear" YES)
 option(ENABLE_PQCRYPTO "Enable Post Quantum Cryptography key agreements algorithms" NO)
 
+# This is a transitive dependency from bctoolbox, but CMake cannot find it otherwise.
+option(ENABLE_OPENSSL "Use OpenSSL." NO)
+
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_EXTENSIONS NO)
 
@@ -53,6 +56,9 @@
 if(ENABLE_ZIDCACHE)
 	bc_find_package(SQLite3 SQLite::SQLite3 sqlite3 REQUIRED)
 endif()
+if(ENABLE_OPENSSL)
+	find_package(OpenSSL REQUIRED)
+endif()
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
 set_source_files_properties(${CMAKE_CURRENT_BINARY_DIR}/config.h PROPERTIES GENERATED ON)
