--- cmake/MainConfig.cmake	2026-07-21 22:07:17.000000000 +0800
+++ cmake/MainConfig.cmake	2026-07-21 22:27:05.000000000 +0800
@@ -3,7 +3,7 @@
 
 IF (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   SET(CMAKE_Fortran_ARCHIVE_CREATE   "<CMAKE_AR> -r -LS <TARGET> <LINK_FLAGS> <OBJECTS>")
-  SET(CMAKE_Fortran_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
+  #SET(CMAKE_Fortran_ARCHIVE_FINISH   "<CMAKE_RANLIB> -no_warning_for_no_symbols -c <TARGET>")
 ENDIF()
 
 STRING(TOLOWER "${CMAKE_Fortran_COMPILER_ID}" FC_ID)
