--- perf_tests/utils.cpp.orig	2025-02-01 05:52:00.000000000 +0800
+++ perf_tests/utils.cpp	2025-04-01 08:05:39.000000000 +0800
@@ -5,7 +5,8 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#include <faiss/perf_tests/utils.h>
+#include <perf_tests/utils.h>
+
 namespace faiss::perf_tests {
 std::map<std::string, faiss::ScalarQuantizer::QuantizerType> sq_types() {
     static std::map<std::string, faiss::ScalarQuantizer::QuantizerType>

--- perf_tests/bench_scalar_quantizer_accuracy.cpp.orig	2024-10-09 23:26:22.000000000 +0800
+++ perf_tests/bench_scalar_quantizer_accuracy.cpp	2024-10-29 14:17:50.000000000 +0800
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#include <faiss/perf_tests/utils.h>
+#include <perf_tests/utils.h>
 #include <gflags/gflags.h>
 #include <cstdio>
 #include <map>

--- perf_tests/bench_scalar_quantizer_decode.cpp.orig	2024-10-09 23:26:22.000000000 +0800
+++ perf_tests/bench_scalar_quantizer_decode.cpp	2024-10-29 14:17:37.000000000 +0800
@@ -5,7 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-#include <faiss/perf_tests/utils.h>
+#include <perf_tests/utils.h>
 #include <gflags/gflags.h>
 #include <omp.h>
 #include <cstdio>

--- perf_tests/bench_scalar_quantizer_distance.cpp.orig	2024-10-09 23:26:22.000000000 +0800
+++ perf_tests/bench_scalar_quantizer_distance.cpp	2024-10-29 14:17:26.000000000 +0800
@@ -12,7 +12,7 @@
 
 #include <benchmark/benchmark.h>
 #include <faiss/impl/ScalarQuantizer.h>
-#include <faiss/perf_tests/utils.h>
+#include <perf_tests/utils.h>
 #include <faiss/utils/random.h>
 #include <faiss/utils/utils.h>
 
--- perf_tests/bench_scalar_quantizer_encode.cpp.orig	2024-10-09 23:26:22.000000000 +0800
+++ perf_tests/bench_scalar_quantizer_encode.cpp	2024-10-29 14:17:09.000000000 +0800
@@ -12,7 +12,7 @@
 
 #include <benchmark/benchmark.h>
 #include <faiss/impl/ScalarQuantizer.h>
-#include <faiss/perf_tests/utils.h>
+#include <perf_tests/utils.h>
 #include <faiss/utils/distances.h>
 #include <faiss/utils/random.h>
 #include <faiss/utils/utils.h>
