--- src/base/Headers.hpp	2026-07-08 07:07:34.000000000 +0800
+++ src/base/Headers.hpp	2026-07-11 15:50:51.000000000 +0800
@@ -44,6 +44,7 @@
 using uid_t = int;
 using gid_t = int;
 #else
+#include <sys/socket.h>
 #include <arpa/inet.h>
 #include <grp.h>
 #include <netdb.h>
@@ -54,7 +55,6 @@
 #include <pwd.h>
 #include <resolv.h>
 #include <sys/ioctl.h>
-#include <sys/socket.h>
 #include <sys/un.h>
 #include <sys/wait.h>
 #include <termios.h>
@@ -97,11 +97,10 @@
 // Need to check for Boost first because Mojave *has* std::filesystem,
 // but won't let you use it.
 #ifdef __APPLE__
-#include <Availability.h>
+#include <AvailabilityMacros.h>
 #endif
 
-#if defined(__APPLE__) && \
-    __MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_15
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MAX_ALLOWED < 101500 && defined(_LIBCPP_VERSION)
 #if __has_include(<boost/filesystem.hpp>)
 #include <boost/filesystem.hpp>
 namespace fs = boost::filesystem;
@@ -110,7 +109,7 @@
 namespace fs = std::filesystem;
 #endif
 #else
-// For non-Apple systems, prefer std::filesystem
+// For other systems, prefer std::filesystem
 // Otherwise, older versions of boost cause failures.
 #if __has_include(<filesystem>)
 #include <filesystem>
