--- CMakeLists.txt.orig	2025-07-22 09:37:18.000000000 +0800
+++ CMakeLists.txt	2025-12-24 09:37:25.000000000 +0800
@@ -91,7 +91,6 @@
 find_package(Threads REQUIRED)
 find_package(sodium REQUIRED)
 find_package(Protobuf REQUIRED)
-find_package(Unwind)
 
 if(DISABLE_VCPKG)
   add_subdirectory(${EXTERNAL_DIR}/Catch2)
@@ -140,7 +139,7 @@
 
 IF(APPLE)
   # Proxy test for Mojave? (Clang that doesn't have std::filesystem)
-  IF(CMAKE_HOST_SYSTEM_VERSION VERSION_LESS "19.0.0")
+  IF(CMAKE_HOST_SYSTEM_VERSION VERSION_LESS "19.0.0" AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
     # So use Boost's version
     find_package(Boost 1.76 COMPONENTS filesystem REQUIRED)
   ENDIF()
