From 0e5cdaf5e1177b4ee782f3c59f7992414ab7baa6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Jul 2025 14:20:48 +0800
Subject: [PATCH] Fix build on macOS: add missing header for
 _NSGetExecutablePath

---
 build-aux/yuck.c | 3 +++
 test/clitoris.c  | 3 +++
 2 files changed, 6 insertions(+)

diff --git build-aux/yuck.c build-aux/yuck.c
index a4e0754..40fb383 100644
--- build-aux/yuck.c
+++ build-aux/yuck.c
@@ -70,6 +70,9 @@
 #if defined WITH_SCMVER
 # include <yuck-scmver.h>
 #endif	/* WITH_SCMVER */
+#ifdef __APPLE__
+# include <mach-o/dyld.h>
+#endif
 
 #if !defined LIKELY
 # define LIKELY(_x)	__builtin_expect((_x), 1)
diff --git test/clitoris.c test/clitoris.c
index 255f9c4..61e3f0d 100644
--- test/clitoris.c
+++ test/clitoris.c
@@ -62,6 +62,9 @@
 #endif	/* HAVE_PTY_H */
 /* check for me */
 #include <wordexp.h>
+#ifdef __APPLE__
+# include <mach-o/dyld.h>
+#endif
 
 #if !defined LIKELY
 # define LIKELY(_x)	__builtin_expect((_x), 1)
