From 478fe29ce7a0c2c86182fd996d2270d008f9ad9b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Jul 2025 14:35:22 +0800
Subject: [PATCH] yuck.c: fix _DARWIN_C_SOURCE macro

---
 build-aux/yuck.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git build-aux/yuck.c build-aux/yuck.c
index 40fb383..219afb6 100644
--- build-aux/yuck.c
+++ build-aux/yuck.c
@@ -44,9 +44,9 @@
 #if !defined _NETBSD_SOURCE
 # define _NETBSD_SOURCE
 #endif	/* !_NETBSD_SOURCE */
-#if !defined _DARWIN_SOURCE
-# define _DARWIN_SOURCE
-#endif	/* !_DARWIN_SOURCE */
+#if !defined _DARWIN_C_SOURCE
+# define _DARWIN_C_SOURCE
+#endif	/* !_DARWIN_C_SOURCE */
 #if !defined _ALL_SOURCE
 # define _ALL_SOURCE
 #endif	/* !_ALL_SOURCE */
