From 3bfae3af389bf849e8f4a2cc0719775097b76dd3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Jul 2025 14:50:47 +0800
Subject: [PATCH 3/3] Use -D_DARWIN_C_SOURCE on Darwin

---
 configure.ac    | 8 ++++++++
 src/Makefile.am | 3 +++
 2 files changed, 11 insertions(+)

diff --git configure.ac configure.ac
index eae2289..0428e0c 100644
--- configure.ac
+++ configure.ac
@@ -35,6 +35,14 @@ SXE_CHECK_CC([c11 c1x c99 gnu99])
 SXE_CHECK_CFLAGS
 AC_C_BIGENDIAN
 
+case $host_os in
+  darwin*)
+    AM_CONDITIONAL([DARWIN], [true])
+    ;;
+  *)
+    AM_CONDITIONAL([DARWIN], [false])
+esac
+
 ## check if yuck is globally available
 AX_CHECK_YUCK
 AX_CHECK_CLITORIS
diff --git src/Makefile.am src/Makefile.am
index b170196..504ef2a 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -3,6 +3,9 @@ include $(top_builddir)/version.mk
 
 AM_CFLAGS = $(EXTRA_CFLAGS)
 AM_CPPFLAGS = -D_POSIX_C_SOURCE=201001L -D_XOPEN_SOURCE=700 -D_BSD_SOURCE
+if DARWIN
+AM_CPPFLAGS += -D_DARWIN_C_SOURCE
+endif
 AM_LDFLAGS = $(XCCLDFLAGS)
 
 bin_PROGRAMS =
-- 
2.48.0

