--- modules/asdf/asdf.lisp	2025-04-29 09:34:50.000000000 +0800
+++ modules/asdf/asdf.lisp	2025-04-29 14:55:58.000000000 +0800
@@ -7662,9 +7662,9 @@
                  ;; So, when running Genera on macOS, only search /usr/local/share.
                  ((os-genera-p)
                   #+Genera (sys:system-case
-                            (darwin-vlm (mapcar 'parse-unix-namestring '("/usr/local/share/")))
+                            (darwin-vlm (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/")))
                             (otherwise (mapcar 'parse-unix-namestring '("/usr/local/share/" "/usr/share/")))))
-                 (t (mapcar 'parse-unix-namestring '("/usr/local/share/" "/usr/share/")))))))
+                 (t (mapcar 'parse-unix-namestring '("@@PREFIX@@/share/" "/usr/local/share/" "/usr/share/")))))))
 
   (defun xdg-config-dirs (&rest more)
     "The preference-ordered set of additional base paths to search for configuration files.
