From b5fafb0742e5a99bb303bc079ef7c5e96ead1b91 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 29 Apr 2025 18:36:14 +0800
Subject: [PATCH] main.cpp: add a missing <map> header

---
 src/main.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/main.cpp src/main.cpp
index 8f9f427..0d9c4ff 100644
--- src/main.cpp
+++ src/main.cpp
@@ -12,6 +12,7 @@
 #include <ftxui/component/screen_interactive.hpp> // for ScreenInteractive
 #include <ftxui/screen/color.hpp>
 #include <iostream>
+#include <map>
 #include <string>
 #include <vector>
 
