--- CMakeLists.txt	2025-01-14 23:56:59.000000000 +0800
+++ CMakeLists.txt	2025-04-28 10:39:04.000000000 +0800
@@ -1,13 +1,5 @@
 cmake_minimum_required(VERSION 3.11)
  
-include(FetchContent)
-set(FETCHCONTENT_UPDATES_DISCONNECTED TRUE)
-FetchContent_Declare(ftxui
-  GIT_REPOSITORY https://github.com/ArthurSonzogni/ftxui
-  GIT_TAG main # Important: Specify a version or a commit hash here.
-)
-FetchContent_MakeAvailable(ftxui)
- 
 project(tuisic LANGUAGES C CXX VERSION 1.0.0)
 add_executable(tuisic src/main.cpp)
 
@@ -19,8 +11,11 @@
     C_STANDARD 11
     C_STANDARD_REQUIRED ON
 )
+
+find_package(PkgConfig REQUIRED)
+
 # Link against FFTW3
-find_package(FFTW3 REQUIRED)
+pkg_check_modules(FFTW3 REQUIRED fftw3)
 target_include_directories(cavacore PUBLIC 
     ${CMAKE_CURRENT_SOURCE_DIR}/cava
     ${FFTW3_INCLUDE_DIRS}
@@ -32,8 +27,8 @@
 
 # Find CURL package
 find_package(CURL REQUIRED)
-find_package(PkgConfig REQUIRED)
 find_package(fmt REQUIRED)
+find_package(ftxui REQUIRED)
 
 # Use pkg-config to find MPV
 pkg_check_modules(MPV REQUIRED mpv)
