--- gio/glocalfile.c	2026-05-02 20:26:52.000000000 +0800
+++ gio/glocalfile.c	2026-05-07 10:44:20.000000000 +0800
@@ -1278,7 +1278,7 @@
  * https://docs.oracle.com/cd/E86824_01/html/E54765/faccessat-2.html
  */
 #if defined(HAVE_FACCESSAT) && !defined(__FreeBSD__) && !defined(__ANDROID__) && \
-    !defined(__OpenBSD__) && !defined(__sun__)
+    !defined(__OpenBSD__) && !defined(__sun__) && !defined(__APPLE__)
 static gboolean
 g_local_file_query_exists (GFile        *file,
                            GCancellable *cancellable)
@@ -3280,7 +3280,7 @@
   iface->monitor_file = g_local_file_monitor_file;
   iface->measure_disk_usage = g_local_file_measure_disk_usage;
 #if defined(HAVE_FACCESSAT) && !defined(__FreeBSD__) && !defined(__ANDROID__) && \
-    !defined(__OpenBSD__) && !defined(__sun__)
+    !defined(__OpenBSD__) && !defined(__sun__) && !defined(__APPLE__)
   iface->query_exists = g_local_file_query_exists;
 #endif
 
