diff --git src/video/wayland/SDL_waylandevents.c src/video/wayland/SDL_waylandevents.c
index 496bb5e8f..370cd38f4 100644
--- src/video/wayland/SDL_waylandevents.c
+++ src/video/wayland/SDL_waylandevents.c
@@ -2277,13 +2277,13 @@ uint32_t tablet_tool_btn_to_sdl_button(struct SDL_WaylandTabletInput *input)
 {
     unsigned int tool_btn = input->btn_stylus3 << 2 | input->btn_stylus2 << 1 | input->btn_stylus << 0;
     switch (tool_btn) {
-    case 0b000:
+    case 0:
         return SDL_BUTTON_LEFT;
-    case 0b001:
+    case 1:
         return SDL_BUTTON_RIGHT;
-    case 0b010:
+    case 2:
         return SDL_BUTTON_MIDDLE;
-    case 0b100:
+    case 4:
         return SDL_BUTTON_X1;
     default:
         return SDL_BUTTON_LEFT;
