--- src/audio/pulseaudio/SDL_pulseaudio.c.orig	2026-09-23 21:42:14.000000000 -0600
+++ src/audio/pulseaudio/SDL_pulseaudio.c	2026-09-23 22:07:01.000000000 -0600
@@ -664,7 +664,8 @@
     paattr.fragsize = this->spec.size;
     paattr.tlength = h->mixlen;
     paattr.prebuf = -1;
-    paattr.maxlength = -1;
+    /* Patch was found in Cygwin: https://cygwin.com/git/?p=git/cygwin-packages/SDL2.git;a=blob;f=SDL2-2.30.3-1.src.patch;h=5ebdc7307116ad118e11ebef076efa6d106fde2b */
+    paattr.maxlength = iscapture ? -1 : (h->mixlen * 2);
     paattr.minreq = -1;
     flags |= PA_STREAM_ADJUST_LATENCY;
 
