--- src/audio/pulseaudio/SDL_pulseaudio.c.orig	2026-09-09 10:13:30.000000000 -0600
+++ src/audio/pulseaudio/SDL_pulseaudio.c	2026-09-09 10:24:40.000000000 -0600
@@ -663,8 +663,8 @@
     /* Reduced prebuffering compared to the defaults. */
     paattr.fragsize = this->spec.size;
     paattr.tlength = h->mixlen;
-    paattr.prebuf = -1;
-    paattr.maxlength = -1;
+    paattr.prebuf = iscapture ? -1 : 0;
+    paattr.maxlength = iscapture ? -1 : (h->mixlen * 2);
     paattr.minreq = -1;
     flags |= PA_STREAM_ADJUST_LATENCY;
 
