--- postgis/lwgeom_functions_basic.c	2020-08-16 02:43:44.000000000 +0800
+++ postgis/lwgeom_functions_basic.c	2025-04-14 09:36:21.000000000 +0800
@@ -1563,10 +1563,7 @@
 		holes = lwalloc(sizeof(LWLINE *)*nholes);
 		for (i=0; i<nholes; i++)
 		{
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wsign-compare"
 			GSERIALIZED *g = (GSERIALIZED *)(ARR_DATA_PTR(array)+offset);
-#pragma GCC diagnostic pop
 			LWLINE *hole;
 			offset += INTALIGN(VARSIZE(g));
 			if ( gserialized_get_type(g) != LINETYPE )

--- postgis/lwgeom_geos.c	2020-08-16 02:43:44.000000000 +0800
+++ postgis/lwgeom_geos.c	2025-04-14 09:39:15.000000000 +0800
@@ -393,10 +393,7 @@
 	/* One geom, good geom? Return it */
 	if ( count == 1 && nelems == 1 )
 	{
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wsign-compare"
 		PG_RETURN_POINTER((GSERIALIZED *)(ARR_DATA_PTR(array)));
-#pragma GCC diagnostic pop
 	}
 
 	/* Ok, we really need GEOS now ;) */
