--- configure.ac	2026-07-12 13:20:29
+++ configure.ac	2026-07-12 13:19:30
@@ -79,7 +79,7 @@
 AM_CONDITIONAL([USE_FTXUI], [test x$with_ftxui == xyes])
 AS_IF([test "x${with_ftxui}" == xyes], [
    AC_DEFINE([HAVE_FTXUI], [1], [Build to use FINAL CUT FTXUI library])
-   PKG_CHECK_MODULES([FTXUI], [ftxui >= 5.0.0 ftxui < 7.0.0])
+   PKG_CHECK_MODULES([FTXUI], [ftxui >= 5.0.0 ftxui < 8.0.0])
 ])
 
 dnl https://wiki.wxwidgets.org/Autoconf

--- src/main.cpp	2026-07-10 07:51:40
+++ src/main.cpp	2026-07-10 07:51:55
@@ -91,13 +91,19 @@
   auto usage = [=] {
     simple_usage();
 #ifdef HAVE_FINALCUT
+// --finalcut parameter only needed if there are multiple front-ends
+#ifdef HAVE_FTXUI
     std::cout << "  -t, --finalcut\t\t";
     std::cout << "run with FINALCUT user interface\n";
 #endif
+#endif
 #ifdef HAVE_FTXUI
+// --ftxui parameter only needed if there are multiple front-ends
+#ifdef HAVE_FINALCUT
     std::cout << "  -x, --ftxui\t\t\t";
     std::cout << "run with FTXUI user interface\n";
 #endif
+#endif
   };
 
   std::string filename;
@@ -215,7 +221,7 @@
         usage();
         exit(0xFF);
       }
-      KitListFtxui ftxui_app;
+      fdsd::ftxui::KitListFtxui ftxui_app;
       ftxui_app.run(filename);
     }
 #endif // HAVE_FTXUI
