From 8c9283aaafbea7524dee01b192c8d4202b240556 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 10 Jul 2025 13:25:31 +0800
Subject: [PATCH 1/2] zlib: add missing header

---
 TTKThirdParty/Zlib/zlib/gzguts.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/TTKThirdParty/Zlib/zlib/gzguts.h b/TTKThirdParty/Zlib/zlib/gzguts.h
index eba72085b..4e08d51ed 100644
--- a/TTKThirdParty/Zlib/zlib/gzguts.h
+++ b/TTKThirdParty/Zlib/zlib/gzguts.h
@@ -34,6 +34,10 @@
 #  include <stddef.h>
 #endif
 
+#ifdef __APPLE__
+#include <unistd.h> /* close, read, write etc. */
+#endif
+
 #if defined(__TURBOC__) || defined(_MSC_VER) || defined(_WIN32)
 #  include <io.h>
 #endif
-- 
2.48.0

