--- CMakeLists.txt.orig	2025-01-20 12:09:04
+++ CMakeLists.txt	2025-06-27 18:24:52
@@ -152,8 +152,8 @@
 )
 
 
-set(TTK_BUILD_DIR "${CMAKE_CACHEFILE_DIR}/bin")
-set(TTK_GENERATE_DIR "${CMAKE_CACHEFILE_DIR}/Generate")
+set(TTK_BUILD_DIR "@BUILDDIR@/bin")
+set(TTK_GENERATE_DIR "${TTK_BUILD_DIR}/Generate")
 
 include_directories(${TTK_GENERATE_DIR})
 
@@ -243,7 +243,7 @@
 endif()
 
 
-if(UNIX)
+if(UNIX AND NOT APPLE)
   find_program(RENAME_EXECUTABLE rename /usr/bin)
   if(RENAME_EXECUTABLE)
     message(STATUS "Message Found rename executable: " ${RENAME_EXECUTABLE})
