From 302391e8b8ea70a18942a085df8745f4b517597d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 6 Jul 2025 05:13:27 +0800
Subject: [PATCH 2/3] qdevicewatcher_mac.cpp: handle error

---
 TTKThirdParty/TTKWatcher/qdevicewatcher_mac.cpp | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git TTKThirdParty/TTKWatcher/qdevicewatcher_mac.cpp TTKThirdParty/TTKWatcher/qdevicewatcher_mac.cpp
index 4d9dc1956..9b786dad4 100644
--- TTKThirdParty/TTKWatcher/qdevicewatcher_mac.cpp
+++ TTKThirdParty/TTKWatcher/qdevicewatcher_mac.cpp
@@ -40,7 +40,7 @@ bool QDeviceWatcherPrivate::stop()
 {
     mStop = true;
     wait();
-    //DAUnregisterApprovalCallback
+    // DAUnregisterApprovalCallback
     DAUnregisterCallback(mSession, (void*)onDiskAppear, this);
     DAUnregisterCallback(mSession, (void*)onDiskDisappear, this);
 }
@@ -51,9 +51,13 @@ void QDeviceWatcherPrivate::parseDeviceInfo()
 
 bool QDeviceWatcherPrivate::initialize()
 {
-    //get sDevices
-    //FSGetVolumeInfo()
+    // get sDevices
+    // FSGetVolumeInfo()
     mSession = DASessionCreate(kCFAllocatorDefault);
+    if (!mSession) {
+        qWarning("Failed to create Disk Arbitration session");
+        return false;
+    }
 
     DARegisterDiskAppearedCallback(mSession, nullptr, onDiskAppear, this);
     DARegisterDiskDisappearedCallback(mSession, nullptr, onDiskDisappear, this);
-- 
2.49.0

