From 0be478506f5d017b3829607e6a8642a0825ece2f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 06:25:47 +0800
Subject: [PATCH 2/2] CMakeLists: no QtMobility on Apple

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 1b1b21cc5..1335db748 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -272,7 +272,7 @@ find_library(TTK_QMMP_LIBRARY
 
 # qmmp include
 include_directories(${TTK_THIRDPARTY_DIR}/TTKqmmp)
-if(TTK_QT_VERSION VERSION_EQUAL "4" AND UNIX)
+if(TTK_QT_VERSION VERSION_EQUAL "4" AND UNIX AND NOT APPLE)
   include_directories(
     /usr/include/QtMultimediaKit
     /usr/include/QtMobility)
-- 
2.49.0

