From 24b6f1ef0f61cc866e66a34311dfc08ca160b638 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 10 Jul 2025 13:31:23 +0800
Subject: [PATCH 2/2] qglobalshortcut_mac: link to Carbon

---
 TTKThirdParty/TTKExtras/CMakeLists.txt | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/TTKThirdParty/TTKExtras/CMakeLists.txt b/TTKThirdParty/TTKExtras/CMakeLists.txt
index 49978cfa4..8805c208c 100644
--- a/TTKThirdParty/TTKExtras/CMakeLists.txt
+++ b/TTKThirdParty/TTKExtras/CMakeLists.txt
@@ -223,6 +223,14 @@ endif()
 
 set(QT_LINK_LIBS TTKLibrary TTKUi)
 
+# qglobalshortcut_mac.cpp needs this.
+# Notice, it cannot work on recent macOS versions,
+# so either additional implementation is needed, using
+# modern API, or at least a dummy fallback.
+if(APPLE)
+  list(APPEND QT_LINK_LIBS "-framework Carbon")
+endif()
+
 if(TTK_QT_VERSION VERSION_EQUAL "4")
   qt4_wrap_cpp(MOC_FILES ${HEADER_FILES})
 
-- 
2.48.0

