From 1399d8db4e787de21eaabb948a7e0c4fd91cd249 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 06:50:27 +0800
Subject: [PATCH 3/3] TTKLibrary: do not use X11 code on macOS without X11

--- TTKCommon/TTKLibrary/ttkplatformsystem.cpp	2025-01-20 12:09:04
+++ TTKCommon/TTKLibrary/ttkplatformsystem.cpp	2025-07-05 06:58:06
@@ -5,8 +5,15 @@
 #  include <qt_windows.h>
 #  include <QScreen>
 #  include <QApplication>
-#elif defined Q_OS_UNIX
+#elif defined Q_OS_UNIX && defined Q_WS_X11
 #  include <X11/Xlib.h>
+#elif defined(Q_OS_MAC) || defined(Q_OS_MACOS)
+#  include <QApplication>
+#  if QT_VERSION < 0x050000
+#    include <QDesktopWidget>
+#  else
+#    include <QScreen>
+#  endif
 #endif
 #include <QSize>
 #include <QFile>
@@ -36,7 +43,7 @@
 
     return QSize(x, y);
 #  endif
-#elif defined Q_OS_UNIX
+#elif defined Q_OS_UNIX && defined Q_WS_X11
     Display *dp = XOpenDisplay(nullptr);
     if(!dp)
     {
@@ -49,6 +56,24 @@
 
     XCloseDisplay(dp);
     return QSize(x + 0.5, y + 0.5);
+#elif defined(Q_OS_MAC) || defined(Q_OS_MACOS)
+    if(!qApp)
+    {
+        int count = 0;
+        QApplication(count, nullptr);
+    }
+
+#  if QT_VERSION >= 0x050000
+    // Qt5: Prefer QScreen
+    QScreen *screen = QApplication::primaryScreen();
+    if(screen)
+        return QSize(screen->logicalDotsPerInchX(), screen->logicalDotsPerInchY());
+    // fallback below if no screen
+#  endif
+    // Qt4 or fallback: Use QDesktopWidget
+    QDesktopWidget* desktop = QApplication::desktop();
+    if(desktop)
+        return QSize(desktop->logicalDpiX(), desktop->logicalDpiY());
 #endif
     return defaultSize;
 }
