From 351fcfaf4ea36095adb2588e395a50683907fe77 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 6 Jul 2025 05:19:00 +0800
Subject: [PATCH 3/3] TTKRoutine.sh: use DYLD_LIBRARY_PATH on macOS

---
 TTKResource/_extras/TTKRoutine.sh | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git TTKResource/_extras/TTKRoutine.sh TTKResource/_extras/TTKRoutine.sh
index 1bf8a2bfd..1843abfa8 100644
--- TTKResource/_extras/TTKRoutine.sh
+++ TTKResource/_extras/TTKRoutine.sh
@@ -14,7 +14,14 @@ if [ "${dirpath%${tmp}}" != "/" ]; then
   dirpath=$PWD/${dirpath}
 fi
 
-export LD_LIBRARY_PATH=${dirpath}:${dirpath}/lib:${dirpath}/4.1.3.0
+case "$(uname)" in
+  Darwin)
+    export DYLD_LIBRARY_PATH="${dirpath}:${dirpath}/lib:${dirpath}/4.1.3.0:$DYLD_LIBRARY_PATH"
+    ;;
+  *)
+    export LD_LIBRARY_PATH="${dirpath}:${dirpath}/lib:${dirpath}/4.1.3.0"
+    ;;
+esac
 export QT_PLUGIN_PATH=${dirpath}/plugins:$QT_PLUGIN_PATH
 
 ${appname} "$@"
-- 
2.49.0

