From b1922df230c793ba34a6e008cd05cc787f4b2fd8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 18:06:09 +0800
Subject: [PATCH 4/5] musictimerautomodule.cpp: no RB_POWER_OFF on macOS

---
 TTKModule/TTKCore/musicToolsSetsKits/musictimerautomodule.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git TTKModule/TTKCore/musicToolsSetsKits/musictimerautomodule.cpp TTKModule/TTKCore/musicToolsSetsKits/musictimerautomodule.cpp
index 9db5ac653..5672ab9c5 100644
--- TTKModule/TTKCore/musicToolsSetsKits/musictimerautomodule.cpp
+++ TTKModule/TTKCore/musicToolsSetsKits/musictimerautomodule.cpp
@@ -138,7 +138,11 @@ void MusicTimerAutoModule::setShutdown()
     sync();
     TTK::Core::sleep(3 * TTK_DN_S2MS);
     /* shutdown */
+#if defined(__APPLE__)
+    system("shutdown -h now");
+#else
     reboot(RB_POWER_OFF);
+#endif
 #endif
     TTK_INFO_STREAM("Shutdown now");
 }
-- 
2.49.0

