From df55dc618bd4a7392df15e0da48410c33791a88b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 18:10:19 +0800
Subject: [PATCH 5/5] Use QtMultimedia

---
 TTKModule/CMakeLists.txt | 6 +-----
 TTKMusicPlayer.pri       | 2 +-
 2 files changed, 2 insertions(+), 6 deletions(-)

diff --git TTKModule/CMakeLists.txt TTKModule/CMakeLists.txt
index aeee1bdc6..557469e59 100644
--- TTKModule/CMakeLists.txt
+++ TTKModule/CMakeLists.txt
@@ -135,11 +135,7 @@ if(TTK_QT_VERSION VERSION_EQUAL "4")
   qt4_wrap_ui(UIC_FILES ${UI_FILES})
   qt4_wrap_cpp(MOC_FILES ${HEADER_FILES})
 
-  if(UNIX)
-    list(APPEND QT_LINK_LIBS ${QT_QTDBUS_LIBRARY} QtMultimediaKit)
-  endif()
-
-  list(APPEND QT_LINK_LIBS ${QT_QTCORE_LIBRARY} ${QT_QTMULTIMEDIA_LIBRARY} ${QT_QTXML_LIBRARY} ${QT_QTGUI_LIBRARY} ${QT_QTNETWORK_LIBRARY})
+  list(APPEND QT_LINK_LIBS Qt4::QtCore Qt4::QtMultimedia Qt4::QtXml Qt4::QtGui Qt4::QtNetwork Qt4::QtDBus)
 elseif(TTK_QT_VERSION VERSION_EQUAL "5")
   qt5_wrap_ui(UIC_FILES ${UI_FILES})
   qt5_wrap_cpp(MOC_FILES ${HEADER_FILES})
diff --git TTKMusicPlayer.pri TTKMusicPlayer.pri
index f2a67e572..e94c103e4 100644
--- TTKMusicPlayer.pri
+++ TTKMusicPlayer.pri
@@ -18,7 +18,7 @@
 
 QT += core gui
 equals(QT_MAJOR_VERSION, 4){ #Qt4
-    QT += network
+    QT += network multimedia
     CONFIG += gcc
 }
 
-- 
2.49.0

