From 250fb9353cafa8054be80fd0374d5a8c1786f0bf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 18:45:40 +0800
Subject: [PATCH 6/6] volumehandler.h: fix declaration

diff --git TTKThirdParty/TTKqmmp/qmmp/volumehandler.h TTKThirdParty/TTKqmmp/qmmp/volumehandler.h
index c4f582ba6..1e22dc85b 100644
--- TTKThirdParty/TTKqmmp/qmmp/volumehandler.h
+++ TTKThirdParty/TTKqmmp/qmmp/volumehandler.h
@@ -53,7 +53,7 @@ public:
      * @param left Left channel volume level. It should be \b 0..100
      * @param right Right channel volume level. It should be \b 0..100
      */
-    void setVolume(int left, int right);
+    void setVolumePerChannel(int left, int right);
     /*!
      * Changes volume by \b delta percent
      */
