From d2093919c8d7c287e9737f13063b81ebdfa0a815 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Jul 2025 19:22:02 +0800
Subject: [PATCH 7/7] musicservicemain.cpp: do not use Linux-specific stuff on
 macOS

---
 TTKService/musicservicemain.cpp | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git TTKService/musicservicemain.cpp TTKService/musicservicemain.cpp
index 53236fac7..73c038fbf 100644
--- TTKService/musicservicemain.cpp
+++ TTKService/musicservicemain.cpp
@@ -8,7 +8,9 @@
 #include "ttkglobalinterface.h"
 
 #ifdef Q_OS_UNIX
-#  include <malloc.h>
+#  ifndef __APPLE__
+#    include <malloc.h>
+#  endif
 #  include "musicmprisplayer.h"
 #endif
 
@@ -118,7 +120,7 @@ int main(int argc, char *argv[])
     mpris.run();
 #endif
 
-#ifdef Q_OS_UNIX
+#if defined(Q_OS_UNIX) && !defined(__APPLE__)
     // memory free
     mallopt(M_MMAP_THRESHOLD, 1024 * 1024);   // 1MB mmap
     mallopt(M_TRIM_THRESHOLD, 2 * 1024 * 1024); // 2MB brk
-- 
2.49.0

