From dc0ba16f7d95335f547f5b5f54263ec5685906a6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 16 Aug 2025 00:36:11 +0800
Subject: [PATCH 2/2] verilatedos_c.h: do not include cpuid.h on powerpc

---
 include/verilatedos_c.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git include/verilatedos_c.h include/verilatedos_c.h
index 2b4619816..849419701 100644
--- include/verilatedos_c.h
+++ include/verilatedos_c.h
@@ -38,7 +38,7 @@
 #if defined(__linux)
 # include <sched.h>  // For sched_getcpu()
 #endif
-#if defined(__APPLE__) && !defined(__arm64__)
+#if defined(__APPLE__) && !defined(__arm64__) && !defined(__POWERPC__)
 # include <cpuid.h>  // For __cpuid_count()
 #endif
 // clang-format on
@@ -88,7 +88,7 @@ double DeltaWallTime::gettime() VL_MT_SAFE {
 uint16_t getcpu() VL_MT_SAFE {
 #if defined(__linux)
     return sched_getcpu();  // TODO: this is a system call. Not exactly cheap.
-#elif defined(__APPLE__) && !defined(__arm64__)
+#elif defined(__APPLE__) && !defined(__arm64__) && !defined(__POWERPC__)
     uint32_t info[4];
     __cpuid_count(1, 0, info[0], info[1], info[2], info[3]);
     // info[1] is EBX, bits 24-31 are APIC ID
