--- common/Xregion/Xregion.h	2008-10-14 21:28:36.000000000 +0800
+++ common/Xregion/Xregion.h	2026-06-11 09:14:01.000000000 +0800
@@ -51,7 +51,7 @@
 #ifndef _XREGION_H_
 #define _XREGION_H_
 
-// - Faked defines to fool the X11 region code
+/* Faked defines to fool the X11 region code */
 
 #include <stdlib.h>
 #include <string.h>
@@ -71,7 +71,7 @@
 
 #define NeedFunctionPrototypes 1
 
-// - Cribbed from Xlib.h
+/* Cribbed from Xlib.h */
 
 typedef struct {
     short x, y;

--- common/Xregion/Region.c	2008-10-14 21:28:36.000000000 +0800
+++ common/Xregion/Region.c	2026-06-11 09:14:35.000000000 +0800
@@ -73,10 +73,10 @@
  */
 
 #include "Xregion.h"
-//#include "Xlibint.h"
-//#include "Xutil.h"
+/* #include "Xlibint.h" */
+/* #include "Xutil.h" */
 #include "region.h"
-//#include "poly.h"
+/* #include "poly.h" */
 
 #ifdef DEBUG
 #include <stdio.h>
