From 25423b0ec435c3684e03e8ce130aff13eec87680 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Sun, 28 Dec 2025 09:59:56 +0000
Subject: [PATCH] Fix macppx build error

---
 libr/debug/p/native/xnu/xnu_excthreads.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libr/debug/p/native/xnu/xnu_excthreads.c b/libr/debug/p/native/xnu/xnu_excthreads.c
index e28587bd571f7..eee940717d3c4 100644
--- libr/debug/p/native/xnu/xnu_excthreads.c
+++ libr/debug/p/native/xnu/xnu_excthreads.c
@@ -344,7 +344,11 @@ static int handle_exception_message(RDebug *dbg, exc_msg *msg, int *ret_code) {
 		R_LOG_ERROR ("EXC_EMULATION");
 		break;
 	case EXC_SOFTWARE:
+#if __POWERPC__
+		R_LOG_ERROR ("EXC_SOFTWARE retry dcu?");
+#else
 		R_LOG_ERROR ("EXC_SOFTWARE code %d retry dcu?", msg->code);
+#endif
 		ret = R_DEBUG_REASON_BREAKPOINT;
 		*ret_code = KERN_SUCCESS;
 		kr = task_suspend (msg->task.name);
