From 6bce93de57ad23ca2a53319dcebced686d2dd109 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 1 Sep 2025 02:16:31 +0800
Subject: [PATCH] sys.c: use environ correctly on macOS

---
 libr/util/sys.c | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/libr/util/sys.c b/libr/util/sys.c
index cdd239af9e14c..1b9854f559efb 100644
--- a/libr/util/sys.c
+++ b/libr/util/sys.c
@@ -43,7 +43,9 @@ static R_TH_LOCAL bool Gunsignable = false; // OK
 #endif
 #if __APPLE__
 #include <errno.h>
-#ifdef __MAC_10_8
+#include <TargetConditionals.h>
+// iOS don't have this
+#if !TARGET_OS_IPHONE
 #define HAVE_ENVIRON 1
 #else
 #define HAVE_ENVIRON 0
@@ -51,10 +53,9 @@ static R_TH_LOCAL bool Gunsignable = false; // OK
 
 #if HAVE_ENVIRON
 #include <execinfo.h>
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
 #endif
-// iOS don't have this we can't hardcode
-// #include <crt_externs.h>
-extern char ***_NSGetEnviron(void);
 # ifndef PROC_PIDPATHINFO_MAXSIZE
 #  define PROC_PIDPATHINFO_MAXSIZE 1024
 int proc_pidpath(int pid, void * buffer, ut32 buffersize);
@@ -70,7 +71,9 @@ int proc_pidpath(int pid, void * buffer, ut32 buffersize);
 # include <sys/wait.h>
 #endif
 # include <signal.h>
+#ifndef __APPLE__
 extern char **environ;
+#endif
 
 #ifdef __HAIKU__
 # define Sleep sleep
