--- configure	2026-06-03 06:12:24.000000000 +0800
+++ configure	2026-06-04 15:19:25.000000000 +0800
@@ -37,9 +37,9 @@
 USE_CS5=0
 USE_CS4=0
 WITH_CAPSTONE=0
-WITHOUT_SYSCAPSTONE=1
+WITHOUT_SYSCAPSTONE=0
 WITH_ZYDIS=0
-WITHOUT_SYSZYDIS=1
+WITHOUT_SYSZYDIS=0
 WITH_SQSH=1
 WITH_SYSLZ4=0
 WANT_ZIP=1
@@ -404,9 +404,9 @@
 "--with-capstone5") USE_CS5="1"; ;;
 "--with-capstone4") USE_CS4="1"; ;;
 "--with-syscapstone") WITH_CAPSTONE="1"; ;;
-"--without-syscapstone") WITHOUT_SYSCAPSTONE="0"; ;;
+"--without-syscapstone") WITHOUT_SYSCAPSTONE="1"; ;;
 "--with-syszydis") WITH_ZYDIS="1"; ;;
-"--without-syszydis") WITHOUT_SYSZYDIS="0"; ;;
+"--without-syszydis") WITHOUT_SYSZYDIS="1"; ;;
 "--without-sqsh") WITH_SQSH="0"; ;;
 "--with-syslz4") WITH_SYSLZ4="1"; ;;
 "--without-zip") WANT_ZIP="0"; ;;
@@ -711,31 +711,17 @@
 USE_CAPSTONE="1"
 else
 USE_CAPSTONE="0"; fi
-if [ "$WITHOUT_SYSCAPSTONE" = "0" ]; then
+if [ "$WITHOUT_SYSCAPSTONE" = "1" ]; then
 USE_CAPSTONE="0"
 HAVE_PKGCFG_CAPSTONE="0"
 CAPSTONE_CFLAGS=""
 CAPSTONE_LDFLAGS=""; fi
 if [ "$WITH_ZYDIS" = "1" ]; then
-if [ -z "${PKGCONFIG}" ]; then pkg-config --version >/dev/null 2>&1 ; if [ 0 = 0 ]; then PKGCONFIG=pkg-config ; else PKGCONFIG=pkgconf ; fi; fi
-printf 'checking pkg-config flags for zydis... '
-tmp=`${PKGCONFIG} --cflags zydis 2>/dev/null`
-if [ $? != 0 ]; then echo no ; HAVE_PKGCFG_ZYDIS=0;
-ZYDIS_CFLAGS='';ZYDIS_LDFLAGS='';
-else
-ZYDIS_CFLAGS=$tmp;
-tmp=`${PKGCONFIG} --libs zydis 2>/dev/null`
-if [ $? = 0 ]; then
-echo yes; HAVE_PKGCFG_ZYDIS=1;
-ZYDIS_LDFLAGS=$tmp; fi; fi; fi
-if [ 11 = "$WITH_ZYDIS$HAVE_PKGCFG_ZYDIS" ]; then
-
 USE_ZYDIS="1"
 USE_SYS_ZYDIS="1"
-else
-USE_ZYDIS="0"
-USE_SYS_ZYDIS="0"; fi
-if [ "$WITHOUT_SYSZYDIS" = "0" ]; then
+ZYDIS_CFLAGS=""
+ZYDIS_LDFLAGS="-lZydis"; fi
+if [ "$WITHOUT_SYSZYDIS" = "1" ]; then
 USE_ZYDIS="0"
 USE_SYS_ZYDIS="0"
 HAVE_PKGCFG_ZYDIS="0"
