--- CIrrDeviceOSX.h.orig	2023-03-06 04:45:09.000000000 +0800
+++ CIrrDeviceOSX.h	2025-08-16 19:09:55.000000000 +0800
@@ -15,8 +15,9 @@
 #include "IGUIEnvironment.h"
 #include "ICursorControl.h"
 
-#import <AppKit/NSWindow.h>
-#import <AppKit/NSBitmapImageRep.h>
+#include <Cocoa/Cocoa.h>
+#include <AppKit/NSWindow.h>
+#include <AppKit/NSBitmapImageRep.h>
 
 #include <map>
 
@@ -25,12 +26,14 @@
     class CIrrDeviceMacOSX;
 }
 
-@interface CIrrDelegateOSX : NSObject
-
+@interface CIrrDelegateOSX : NSObject <NSWindowDelegate, NSApplicationDelegate>
+{
+    irr::CIrrDeviceMacOSX* Device;
+    BOOL Quit;
+}
 - (id)initWithDevice:(irr::CIrrDeviceMacOSX*)device;
 - (void)terminate:(id)sender;
 - (BOOL)isQuit;
-
 @end
 
 namespace irr
