--- CIrrDeviceOSX.mm.orig	2023-03-06 04:45:09.000000000 +0800
+++ CIrrDeviceOSX.mm	2025-08-16 19:05:40.000000000 +0800
@@ -464,10 +464,6 @@
 static bool firstLaunch = true;
 
 @implementation CIrrDelegateOSX
-{
-    irr::CIrrDeviceMacOSX* Device;
-    bool Quit;
-}
 
 - (id)initWithDevice:(irr::CIrrDeviceMacOSX*)device
 {
@@ -476,14 +472,14 @@
     if (self)
         Device = device;
     
-    Quit = false;
+    Quit = NO;
     
     return (self);
 }
 
 - (void)applicationDidFinishLaunching:(NSNotification*)notification
 {
-    Quit = false;
+    Quit = NO;
 }
 
 - (void)orderFrontStandardAboutPanel:(id)sender
@@ -508,13 +504,13 @@
 
 - (void)terminate:(id)sender
 {
-    Quit = true;
+    Quit = YES;
 }
 
 - (void)windowWillClose:(id)sender
 {
     Device->setWindow(nil);
-    Quit = true;
+    Quit = YES;
 }
 
 - (NSSize)windowWillResize:(NSWindow *)window toSize:(NSSize)proposedFrameSize
