From b38acb9ea9e2d9583e2219bc75d5e1ba16632f98 Mon Sep 17 00:00:00 2001
From: Richard Russon <rich@flatcap.org>
Date: Fri, 26 Jun 2026 09:37:36 +0100
Subject: [PATCH] build: fix includes for SIZE_MAX

---
 mutt/charset.c | 1 +
 mutt/memory.c  | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/mutt/charset.c b/mutt/charset.c
index fd3fe2b3209..81538b529fc 100644
--- mutt/charset.c
+++ mutt/charset.c
@@ -35,6 +35,7 @@
 #include <langinfo.h>
 #include <limits.h>
 #include <stdbool.h>
+#include <stdint.h>
 #include <stdio.h>
 #include <string.h>
 #include "charset.h"
diff --git a/mutt/memory.c b/mutt/memory.c
index 35573f0b463..7d8fc2cdfed 100644
--- mutt/memory.c
+++ mutt/memory.c
@@ -37,6 +37,9 @@
 #include "memory.h"
 #include "exit.h"
 #include "logging2.h"
+#ifndef HAVE_REALLOCARRAY
+#include <stdint.h>
+#endif
 
 /**
  * reallocarray - reallocarray(3) implementation
