From d7065e3ec2864b516ea2b41bc9fde334f4cae1ef Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 27 Mar 2026 06:57:32 +0800
Subject: [PATCH] socket.cpp: define MSG_NOSIGNAL if undefined

---
 src/util/socket.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/util/socket.cpp src/util/socket.cpp
index d770270..9a1d1ac 100644
--- src/util/socket.cpp
+++ src/util/socket.cpp
@@ -29,6 +29,10 @@
 #include "os.hpp"
 #include "util.hpp"
 
+#ifndef MSG_NOSIGNAL
+#define MSG_NOSIGNAL 0
+#endif
+
 namespace fs = std::filesystem;
 
 UnixSocket::UnixSocket(const std::string_view endpoint)
