From 48bb6f2edbfdca5ff17db522d7cfa71de321560c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 21 Jan 2024 05:05:05 +0800
Subject: [PATCH] gfx/qcms/transform-altivec.c: fix load_aligned_float
 declaration

---
 gfx/qcms/transform-altivec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git gfx/qcms/transform-altivec.c gfx/qcms/transform-altivec.c
index 9dc17bc13f..f8b60c7901 100644
--- platform/gfx/qcms/transform-altivec.c
+++ platform/gfx/qcms/transform-altivec.c
@@ -29,7 +29,7 @@
 static const ALIGN float floatScaleX4 = FLOATSCALE;
 static const ALIGN float clampMaxValueX4 = CLAMPMAXVAL;
 
-inline vector float load_aligned_float(float *dataPtr)
+static inline vector float load_aligned_float(float *dataPtr)
 {
 	vector float data = vec_lde(0, dataPtr);
 	vector unsigned char moveToStart = vec_lvsl(0, dataPtr);
