From 8f381309f13760cd2fb31e339818507fdf6d651f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 14 Jan 2024 02:53:44 +0800
Subject: [PATCH 02/10] compiler-opts.m4: drop forcing of broken clang

---
 build/autoconf/compiler-opts.m4 | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git build/autoconf/compiler-opts.m4 build/autoconf/compiler-opts.m4
index 29fde4f832..295b4c7b8c 100644
--- build/autoconf/compiler-opts.m4
+++ build/autoconf/compiler-opts.m4
@@ -45,23 +45,6 @@ case "$target" in
         HOST_LDFLAGS=" "
     fi
     ;;
-*-darwin*)
-    # GCC on darwin is based on gcc 4.2 and we don't support it anymore.
-    if test -z "$CC"; then
-        MOZ_PATH_PROGS(CC, clang)
-    fi
-    if test -z "$CXX"; then
-        MOZ_PATH_PROGS(CXX, clang++)
-    fi
-    IS_GCC=$($CC -v 2>&1 | grep gcc)
-    if test -n "$IS_GCC"
-    then
-      echo gcc is known to be broken on OS X, please use clang.
-      echo see http://developer.mozilla.org/en-US/docs/Developer_Guide/Build_Instructions/Mac_OS_X_Prerequisites
-      echo for more information.
-      exit 1
-    fi
-    ;;
 esac
 fi
 ])
-- 
2.43.0

