From 4abdfcd90daa681dc48d672e7ea4f1e874d4bf72 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 21 Jan 2024 05:21:49 +0800
Subject: [PATCH 2/7] js/src/jsscript.h: restrict padding hack to older gcc

---
 js/src/jsscript.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git js/src/jsscript.h js/src/jsscript.h
index d801661a86..3885ec3707 100644
--- platform/js/src/jsscript.h
+++ platform/js/src/jsscript.h
@@ -2068,7 +2068,7 @@ class LazyScript : public gc::TenuredCell
     // Add padding so LazyScript is gc::Cell aligned. Make padding protected
     // instead of private to suppress -Wunused-private-field compiler warnings.
   protected:
-#if JS_BITS_PER_WORD == 32 && !(defined(XP_DARWIN) && defined(__ppc__))
+#if JS_BITS_PER_WORD == 32 && !(defined(XP_DARWIN) && defined(__ppc__) && (__GNUC__ < 13))
     uint32_t padding;
 #endif
 
