From c334909ab2246d33c016fff282337b74c7d0c1ff Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Sun, 14 Jan 2024 08:38:06 +0800
Subject: [PATCH 04/10] dom/html/nsGenericHTMLElement.h: fix for gcc

---
 dom/html/nsGenericHTMLElement.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git dom/html/nsGenericHTMLElement.h dom/html/nsGenericHTMLElement.h
index 45770b99d9..4b488281ac 100644
--- dom/html/nsGenericHTMLElement.h
+++ dom/html/nsGenericHTMLElement.h
@@ -410,9 +410,9 @@ public:
     return SetTokenList(nsGkAtoms::itemtype, aType);
   }
   NS_IMETHOD GetItemId(nsAString& aId) final override {
-    nsString id;
-    GetItemId(id);
-    aId.Assign(aId);
+    nsString hid;
+    GetItemId(hid);
+    aId.Assign(hid);
     return NS_OK;
   }
   NS_IMETHOD SetItemId(const nsAString& aId) final override {
-- 
2.43.0

