--- platform/python/mozbuild/mozbuild/configure/constants.py.orig	2024-07-14 19:44:36.000000000 +0800
+++ platform/python/mozbuild/mozbuild/configure/constants.py	2024-07-14 22:33:36.000000000 +0800
@@ -78,8 +78,8 @@
     ('ia64', '__ia64__'),
     ('s390x', '__s390x__'),
     ('s390', '__s390__'),
-    ('ppc64', '__powerpc64__'),
-    ('ppc', '__powerpc__'),
+    ('ppc64', '__powerpc64__ || __ppc64__'),
+    ('ppc', '__powerpc__ || __ppc__'),
     ('Alpha', '__alpha__'),
     ('hppa', '__hppa__'),
     ('sparc64', '__sparc__ && __arch64__'),
