From 7d548f70be1cdc915096b050cef5d9c899bbae11 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 21 Jan 2024 05:50:12 +0800
Subject: [PATCH 4/7] sqlite3: fix pointer size for ppc64

---
 db/sqlite3/src/sqlite3.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git db/sqlite3/src/sqlite3.c db/sqlite3/src/sqlite3.c
index dd3b5c5757..a6499f47e3 100644
--- db/sqlite3/src/sqlite3.c
+++ db/sqlite3/src/sqlite3.c
@@ -14547,7 +14547,7 @@ typedef INT16_TYPE LogEst;
 #   define SQLITE_PTRSIZE __SIZEOF_POINTER__
 # elif defined(i386)     || defined(__i386__)   || defined(_M_IX86) ||    \
        defined(_M_ARM)   || defined(__arm__)    || defined(__x86)   ||    \
-      (defined(__APPLE__) && defined(__POWERPC__)) ||                     \
+      (defined(__APPLE__) && defined(__ppc__))  ||                        \
       (defined(__TOS_AIX__) && !defined(__64BIT__))
 #   define SQLITE_PTRSIZE 4
 # else
-- 
2.43.0

