--- platform/toolkit/moz.configure.orig	2024-07-14 19:44:45.000000000 +0800
+++ platform/toolkit/moz.configure	2024-07-15 13:05:47.000000000 +0800
@@ -59,7 +59,7 @@
     if target.os == 'WINNT':
         platform_choices = ('cairo-windows',)
     elif target.os == 'OSX':
-        platform_choices = ('cairo-cocoa',)
+        platform_choices = ('cairo-gtk3', 'cairo-gtk2', 'cairo-gtk2-x11', 'cairo-cocoa',)
     elif target.os == 'iOS':
         platform_choices = ('cairo-uikit',)
     else:
