--- python/pcaptools/setup.py.in.orig	2022-08-26 05:40:38.000000000 +0800
+++ python/pcaptools/setup.py.in	2025-10-28 11:52:47.000000000 +0800
@@ -18,7 +18,7 @@
     author      = "Joshua Wright",
     url         = "https://github.com/greatscottgadgets/libbtbb",
     license     = "GPL",
-    version     = '${PACKAGE_VERSION}',
+    version     = '0.8.0',
     package_dir = { '': '${CMAKE_CURRENT_SOURCE_DIR}' },
     packages    = ['pcapdump'],
     classifiers=[
