--- python/pcaptools/setup.py.orig	2025-10-28 11:23:10.000000000 +0800
+++ python/pcaptools/setup.py	2025-10-28 11:53:54.000000000 +0800
@@ -14,7 +14,7 @@
 
 setup(
     name        = "pcapdump",
-    version     = "0.8",
+    version     = "0.8.0",
     description = "A reader and dump utility for Pcap files",
     author      = "Joshua Wright",
     url         = "https://github.com/greatscottgadgets/libbtbb",
