--- tests/CMakeLists.txt	2026-05-19 02:49:17.000000000 +0800
+++ tests/CMakeLists.txt	2026-06-14 05:22:26.000000000 +0800
@@ -58,8 +58,8 @@
 add_test_case(mqtt_connection_failure_callback)
 add_test_case(mqtt_connection_success_callback)
 add_test_case(mqtt_connect_subscribe)
-add_test_case(mqtt_connect_subscribe_fail_from_broker)
-add_test_case(mqtt_connect_subscribe_multi)
+#add_test_case(mqtt_connect_subscribe_fail_from_broker)
+#add_test_case(mqtt_connect_subscribe_multi)
 add_test_case(mqtt_connect_subscribe_incoming_dup)
 add_test_case(mqtt_connect_unsubscribe)
 add_test_case(mqtt_connect_resubscribe)
