--- ni/src/db/rgb/rgb.c	2019-02-28 07:44:39.000000000 +0800
+++ ni/src/db/rgb/rgb.c	2025-08-15 21:52:22.000000000 +0800
@@ -40,6 +40,7 @@
 #include <ncarg/c.h>
 #include <ctype.h>
 #include <errno.h>
+#include <stdlib.h>
 #include <string.h>
 
 char *ProgramName;

--- ni/src/mkwrap/wrapit.y	2019-02-28 07:44:39.000000000 +0800
+++ ni/src/mkwrap/wrapit.y	2025-08-15 21:54:46.000000000 +0800
@@ -1,5 +1,6 @@
 %{
 #include <stdio.h>
+#include <ctype.h>
 #include <ncarg/hlu/hlu.h>
 #include <ncarg/hlu/NresDB.h>
 #include <ncarg/ncl/defs.h>

--- ni/src/lib/nio/nio_path.c	2019-02-28 07:44:39.000000000 +0800
+++ ni/src/lib/nio/nio_path.c	2025-08-15 22:25:26.000000000 +0800
@@ -37,6 +37,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <pwd.h>
+#include <ctype.h>
 #include <errno.h>
 #include "nio_path.h"
 #include "nioError.h"

--- ni/src/ncl/DataSupport.c	2019-02-28 07:44:39.000000000 +0800
+++ ni/src/ncl/DataSupport.c	2025-08-15 22:28:53.000000000 +0800
@@ -29,6 +29,7 @@
 #include <ncarg/hlu/NresDB.h>
 #include <ncarg/hlu/Callbacks.h>
 #endif
+#include <ctype.h>
 #include "defs.h"
 #include "Symbol.h"
 #include "NclMdInc.h"
