--- tools/pybind11Common.cmake	2026-03-31 11:38:29.000000000 +0800
+++ tools/pybind11Common.cmake	2026-05-14 11:36:14.000000000 +0800
@@ -352,7 +352,7 @@
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND NOT APPLE)
       # Clang Gold plugin does not support -Os; append -O3 to MinSizeRel builds to override it
       set(linker_append ";$<$<CONFIG:MinSizeRel>:-O3>")
-    elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT MINGW)
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT APPLE)
       set(cxx_append ";-fno-fat-lto-objects")
     endif()
 
