--- build.d/extensions/osxfuse.sh	2018-12-18 13:34:11.000000000 +0800
+++ build.d/extensions/osxfuse.sh	2026-02-22 07:19:24.000000000 +0800
@@ -52,7 +52,7 @@
     fi
 
     local value="$(
-/bin/cat <<EOF | /usr/bin/xcrun clang -E -I"${BUILD_SOURCE_DIRECTORY}/common" -mmacosx-version-min=10.0 - | /usr/bin/tail -1
+/bin/cat <<EOF | /usr/bin/xcrun gcc-4.2 -E -I"${BUILD_SOURCE_DIRECTORY}/common" -mmacosx-version-min=10.0 - | /usr/bin/tail -1
 #include "${file}"
 ${name}
 EOF
