--- src/google/protobuf/generated_message_tctable_decl.h	2025-09-12 01:24:52.000000000 +0800
+++ src/google/protobuf/generated_message_tctable_decl.h	2026-06-28 08:31:26.000000000 +0800
@@ -180,9 +180,9 @@
 
  private:
   uint8_t tag_;
-  bool is_signed_ : 1;
-  bool is_zigzag_ : 1;
-  bool is_utf8_ : 1;
+  uint8_t is_signed_ : 1;
+  uint8_t is_zigzag_ : 1;
+  uint8_t is_utf8_ : 1;
 };
 
 // Make the map entry type card for a specified field type.
