--- auxil/broker/caf/libcaf_core/src/telemetry/importer/process.cpp	2025-08-19 02:02:16.000000000 +0800
+++ auxil/broker/caf/libcaf_core/src/telemetry/importer/process.cpp	2025-08-20 13:09:31.000000000 +0800
@@ -94,6 +94,7 @@
   sys_stats result;
   // Fetch memory usage.
   {
+#ifdef MACH_TASK_BASIC_INFO
     mach_task_basic_info info;
     mach_msg_type_number_t count = MACH_TASK_BASIC_INFO_COUNT;
     if (task_info(mach_task_self(), MACH_TASK_BASIC_INFO,
@@ -102,6 +103,16 @@
       result.rss = info.resident_size;
       result.vms = info.virtual_size;
     }
+#else
+    task_basic_info info;
+    mach_msg_type_number_t count = TASK_BASIC_INFO_COUNT;
+    if (task_info(mach_task_self(), TASK_BASIC_INFO,
+                  reinterpret_cast<task_info_t>(&info), &count)
+        == KERN_SUCCESS) {
+      result.rss = info.resident_size;
+      result.vms = info.virtual_size;
+    }
+#endif
   }
   // Fetch CPU time.
   {
