--- src/packet_analysis/protocol/icmp/ICMP.cc	2025-08-27 01:59:56.000000000 +0800
+++ src/packet_analysis/protocol/icmp/ICMP.cc	2025-08-28 04:10:24.000000000 +0800
@@ -15,6 +15,13 @@
 #include "zeek/packet_analysis/protocol/icmp/events.bif.h"
 #include "zeek/session/Manager.h"
 
+#ifndef MLD_LISTENER_QUERY
+#define MLD_LISTENER_QUERY MLD6_LISTENER_QUERY
+#endif
+#ifndef MLD_LISTENER_REPORT
+#define MLD_LISTENER_REPORT MLD6_LISTENER_REPORT
+#endif
+
 using namespace zeek::packet_analysis::ICMP;
 using namespace zeek::packet_analysis::IP;
 
