--- src/include/duckdb/common/platform.hpp	2025-05-21 19:38:48.000000000 +0800
+++ src/include/duckdb/common/platform.hpp	2025-05-25 16:00:06.000000000 +0800
@@ -62,13 +62,19 @@
 #if defined(__aarch64__) || defined(__ARM_ARCH_ISA_A64)
 	arch = "arm64";
 #endif
+// Big-endian systems are not supported by upstream,
+// and some functionality is broken, as of now.
+#if defined(__ppc64__) || defined(__powerpc64__)
+	arch = "ppc64";
+#elif defined(__ppc__) || defined(__powerpc__)
+	arch = "ppc";
+#endif
 
 #if defined(__MUSL__)
 	if (os == "linux") {
 		postfix = "_musl";
 	}
-#elif (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 5) ||                                                    \
-    (defined(_GLIBCXX_USE_CXX11_ABI) && _GLIBCXX_USE_CXX11_ABI == 0)
+#elif (!defined(__clang__) && defined(__GNUC__) && __GNUC__ < 5)
 #error                                                                                                                 \
     "DuckDB does not provide extensions for this (legacy) CXX ABI - Explicitly set DUCKDB_PLATFORM (Makefile) / DUCKDB_EXPLICIT_PLATFORM (CMake) to build anyway. "
 #endif
