From c0fb7c1dec87b0eb97f7945faf122781ec6c8b2a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 May 2025 15:19:18 +0800
Subject: [PATCH 02/10] src.pro: allow qt4

---
 src/src.pro | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git src/src.pro src/src.pro
index 46e43ab..f4d5cf4 100644
--- src/src.pro
+++ src/src.pro
@@ -5,12 +5,12 @@ GIT_EXEC_DIR = "$$(ProgramFiles)\\Git\\bin"
 # Under Windows uncomment following line to enable console messages
 #CONFIG += ENABLE_CONSOLE_MSG
 
-# check for Qt >= 5.11.0
+# check for Qt >= 4.8.0
 CUR_QT = $$[QT_VERSION]
 
 # WARNING greaterThan is an undocumented function
-!greaterThan(CUR_QT, 5.11) {
-        error("Sorry I need Qt 5.11.0 or later, you seem to have Qt $$CUR_QT instead")
+!greaterThan(CUR_QT, 4.8) {
+        error("Sorry I need Qt 4.8.0 or later, you seem to have Qt $$CUR_QT instead")
 }
 
 # check for g++ compiler
-- 
2.49.0

