From 3c04cbe76bb029665d15ac2227e3266e7c5dc6ac Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 May 2025 14:42:07 +0800
Subject: [PATCH 2/2] listview.cpp: fix enums

---
 src/listview.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/listview.cpp src/listview.cpp
index 24bb584..851b0a9 100644
--- src/listview.cpp
+++ src/listview.cpp
@@ -428,13 +428,13 @@ struct ListView::DropInfo {
 		PATCHES  = 1 << 0,
 		REV_LIST = 1 << 1,
 		REV_RANGE = 1 << 2,
-		SAME_REPO     = 1 << 3,
+		SAME_REPO     = 1 << 3
 	};
 	enum Action {
 		PatchAction = Qt::CopyAction,
 		RebaseAction = Qt::MoveAction,
 		MoveRefAction = (Qt::LinkAction << 1) | Qt::MoveAction,
-		MergeAction = Qt::LinkAction,
+		MergeAction = Qt::LinkAction
 	};
 
 	QString sourceRepo;
-- 
2.49.0

