From 3677dd36cd1c30a905b8d9e8d47c721424764aae Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 May 2025 16:16:43 +0800
Subject: [PATCH 07/10] revsview.cpp: use QString with hash

---
 src/revsview.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/revsview.cpp src/revsview.cpp
index 2e69278..7885cee 100644
--- src/revsview.cpp
+++ src/revsview.cpp
@@ -322,7 +322,7 @@ void RevsView::updateLineEditSHA(bool clear) {
 		l->setText(""); // clears history
 
 	else if (l->text() != st.sha()) {
-		auto hash = QGit::testFlag(QGit::ENABLE_SHORTREF_F)
+		QString hash = QGit::testFlag(QGit::ENABLE_SHORTREF_F)
 			? st.sha().left(git->shortHashLength())
 			: st.sha();
 
-- 
2.49.0

