From 90ab3486ecf15a438fa326bd1069a118b75d2dd5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 May 2025 16:03:30 +0800
Subject: [PATCH 09/10] common.h, listview.cpp: fix enums

---
 src/common.h     | 4 ++--
 src/listview.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git src/common.h src/common.h
index 44e79bf..ed1f26b 100644
--- src/common.h
+++ src/common.h
@@ -223,7 +223,7 @@ namespace QGit {
 	enum FileDoubleClickAction {
 		VIEW_PATCH,
 		OPEN_IN_EDITOR,
-		OPEN_IN_DIFFER,
+		OPEN_IN_DIFFER
 	};
 
 	// settings booleans
@@ -246,7 +246,7 @@ namespace QGit {
 		USE_CMT_MSG_F   = 1 << 15,
 // 		OPEN_IN_EDITOR_F = 1 << 16,  //  not used anymore; subject to be replaced
 		ENABLE_DRAGNDROP_F = 1 << 17,
-		ENABLE_SHORTREF_F = 1 << 18,
+		ENABLE_SHORTREF_F = 1 << 18
 	};
 	const int FLAGS_DEF = USE_CMT_MSG_F | RANGE_SELECT_F | SMART_LBL_F | VERIFY_CMT_F | SIGN_PATCH_F | LOG_DIFF_TAB_F | MSG_ON_NEW_F | ENABLE_DRAGNDROP_F;
 
diff --git src/listview.cpp src/listview.cpp
index 31a6e15..576c836 100644
--- src/listview.cpp
+++ src/listview.cpp
@@ -438,13 +438,13 @@ struct ListView::DropInfo {
 		PATCHES  = 1 << 0,
 		REV_LIST = 1 << 1,
 		REV_RANGE = 1 << 2,
-		SAME_REPO     = 1 << 3,
+		SAME_REPO     = 1 << 3
 	};
 	enum Action {
 		PatchAction = Qt::CopyAction,
 		RebaseAction = Qt::MoveAction,
 		MoveRefAction = (Qt::LinkAction << 1) | Qt::MoveAction,
-		MergeAction = Qt::LinkAction,
+		MergeAction = Qt::LinkAction
 	};
 
 	QString sourceRepo;
-- 
2.49.0

