From 74d2e3a8c0b70ef1bb87c3799004dce90d07036a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 4 May 2025 13:52:57 +0800
Subject: [PATCH 10/10] mainimpl.cpp: compat code for Qt4

---
 src/mainimpl.cpp | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git src/mainimpl.cpp src/mainimpl.cpp
index 954360c..6353c32 100644
--- src/mainimpl.cpp
+++ src/mainimpl.cpp
@@ -112,10 +112,19 @@ MainImpl::MainImpl(SCRef cd, QWidget* p) : QMainWindow(p) {
 	tabWdg->addTab(rv->tabPage(), "&Rev list");
 
 	// hide close button for rev list tab
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+	// For Qt4: Use findChild to access QTabBar as tabBar() is protected
+	QTabBar* const tabBar = tabWdg->findChild<QTabBar*>();
+#else
+	// For Qt5+: Directly access tabBar()
 	QTabBar* const tabBar = tabWdg->tabBar();
+#endif
+
+if (tabBar) {
 	tabBar->setTabButton(0, QTabBar::RightSide, NULL);
 	tabBar->setTabButton(0, QTabBar::LeftSide, NULL);
-	connect(tabWdg, SIGNAL(tabCloseRequested(int)), SLOT(tabBar_tabCloseRequested(int)));
+}
+connect(tabWdg, SIGNAL(tabCloseRequested(int)), SLOT(tabBar_tabCloseRequested(int)));
 
 	// set-up file names loading progress bar
 	pbFileNamesLoading = new QProgressBar(statusBar());
@@ -1343,7 +1352,11 @@ static void prepareRefSubmenu(QMenu* menu, const QStringList& refs, const QChar
 		QMenu* add_here = menu;
 		FOREACH_SL (pit, parts) {
 			if (pit == parts.end() - 1) break;
+	#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
+			QMenu* found = add_here->findChild<QMenu*>(*pit);
+	#else
 			QMenu* found = add_here->findChild<QMenu*>(*pit, Qt::FindDirectChildrenOnly);
+	#endif
 			if(!found) {
 				found = add_here->addMenu(*pit);
 				found->setObjectName(*pit);
-- 
2.49.0

