From 9303df18e17eb3ca26f34319e1cab212dd54a92d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 16 Jul 2025 08:36:04 +0800
Subject: [PATCH] Fix type for is_detached

---
 quickjs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git quickjs.c quickjs.c
index fb42ff8..ce0099d 100644
--- quickjs.c
+++ quickjs.c
@@ -374,7 +374,7 @@ typedef struct JSVarRef {
         struct {
             int __gc_ref_count; /* corresponds to header.ref_count */
             uint8_t __gc_mark; /* corresponds to header.mark/gc_obj_type */
-            bool is_detached;
+            uint8_t is_detached;
         };
     };
     JSValue *pvalue; /* pointer to the value, either on the stack or
-- 
2.48.0

