--- binrz/rizin/meson.build	2026-02-02 01:52:21.000000000 +0800
+++ binrz/rizin/meson.build	2026-02-04 08:49:50.000000000 +0800
@@ -25,29 +25,3 @@
   install_rpath: rpath_exe,
   implicit_include_directories: false,
 )
-
-if host_machine.system() == 'darwin'
-  # CodeSign the rizin binary to make debugger work by applying the
-  # entitlements
-  entitlements = files('rizin_macos.xml')
-  macos_sign_sh = meson.current_source_dir() / 'macos_sign.sh'
-  custom_target('rizin-signing',
-    build_by_default: true,
-    build_always_stale: true,
-    command: [
-      macos_sign_sh,
-      'false',
-      entitlements,
-      rizin_exe.full_path(),
-    ],
-    depends: rizin_exe,
-    output: 'rizin-signing',
-    install: false,
-  )
-
-  meson.add_install_script(meson.current_source_dir() / 'macos_sign.sh',
-    'true',
-    entitlements,
-    rizin_bindir / 'rizin',
-  )
-endif
\ No newline at end of file
