From 8ff60c8c70dd89f87939967e5401465a82782a7c Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:29:25 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_tydi.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ppx_tydi.ml b/src/ppx_tydi.ml
index 2c80a1b..97bcd3d 100644
--- src/ppx_tydi.ml
+++ src/ppx_tydi.ml
@@ -7,7 +7,7 @@ let extension =
     Ast_pattern.t
     =
     let open Ast_pattern in
-    single_expr_payload (pexp_let __ (many (pack2 (value_binding ~pat:__ ~expr:__))) __)
+    single_expr_payload (pexp_let __ (many (pack2 (value_binding ~constraint_:drop ~pat:__ ~expr:__))) __)
     |> map2 ~f:(fun rec_flag bindings ->
          match rec_flag with
          | Recursive -> Location.raise_errorf "[let rec] is not supported."
