--- test/test-stress.c	2026-08-02 01:54:31.000000000 +0800
+++ test/test-stress.c	2026-08-02 22:52:37.000000000 +0800
@@ -60,7 +60,7 @@
 static int ITER    = 50;
 #define ALLOW_LARGE true
 #else
-static int THREADS = NTHREADS;      // more repeatable if THREADS <= #processors
+static int THREADS = NTHREADS/4;    // more repeatable if THREADS <= #processors
 static int SCALE   = 50;            // scaling factor
 static int ITER    = 50;            // N full iterations destructing and re-creating all threads
 #endif
