--- libsrc/Dk/Dkutil.c.orig	2023-06-07 19:59:11.000000000 +0800
+++ libsrc/Dk/Dkutil.c	2025-09-08 15:30:43.000000000 +0800
@@ -182,10 +182,8 @@
       clockid_t cop = CLOCK_MONOTONIC;	/* Linux and macOS */
 #endif
 
-#if defined (__APPLE__)
-      have_clock_gettime = 0;
-      if (__builtin_available (macOS 10.12, iOS 10, tvOS 10, watchOS 3, *))
-	have_clock_gettime = 1;
+#if defined (__APPLE__) /* We know legacy-support has this. */
+      have_clock_gettime = 1;
 #endif
 
       if (have_clock_gettime && clock_gettime (cop, &ts) == 0)
