From 8f69318ed8344f278ffb6b46bc254633044bc325 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 19 Jan 2025 12:50:16 +0800
Subject: [PATCH] iwjsreg.c: use fsync for fdatasync on macOS

---
 src/json/iwjsreg.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/json/iwjsreg.c src/json/iwjsreg.c
index 70e9be4..c59c819 100644
--- src/json/iwjsreg.c
+++ src/json/iwjsreg.c
@@ -12,6 +12,10 @@
 #include <string.h>
 #include <stdlib.h>
 
+#ifdef __APPLE__
+#define fdatasync fsync
+#endif
+
 struct iwjsreg {
   struct iwpool    *pool;
   struct jbl_node  *root;
