--- a/configure.ac	2026-06-29 00:34:10.000000000 +0800
+++ b/configure.ac	2026-06-29 00:36:05.000000000 +0800
@@ -69,7 +69,6 @@
   AS_HELP_STRING([--disable-protoc], [Disable building protoc_c (also disables tests)]))
 if test "x$enable_protoc" != "xno"; then
   AC_LANG_PUSH([C++])
-  AX_CXX_COMPILE_STDCXX(17, noext, mandatory)
   PKG_CHECK_MODULES([protobuf], [protobuf >= 3.0.0])
 
   save_CPPFLAGS="$CPPFLAGS"
