--- configure.ac	2026-05-17 03:55:42.000000000 +0800
+++ configure.ac	2026-06-23 07:22:29.000000000 +0800
@@ -123,13 +123,6 @@
 ])
 AM_CONDITIONAL(ENABLE_LOCAL_HELP, [test "x$with_local_help" != "xno"])
 
-if test "$build_mac" = "yes"; then
-  PKG_CHECK_MODULES(GTK_MAC, [gtk-mac-integration-gtk3 >= 3.0.1])
-  AC_SUBST(GTK_MAC_CFLAGS)
-  AC_SUBST(GTK_MAC_LIBS)
-  AC_DEFINE([HAVE_GTK_MAC], [1], [Define if gtk-mac-integration is available])
-fi
-
 PKG_CHECK_MODULES(QALCULATE, [
 	libqalculate >= 5.6.0
 	])

--- src/Makefile.am	2026-02-13 15:09:27.000000000 +0800
+++ src/Makefile.am	2026-06-23 07:21:47.000000000 +0800
@@ -11,10 +11,6 @@
 	@GIOUNIX_CFLAGS@ \
 	@QALCULATE_CFLAGS@
 
-if PLATFORM_MACOS
-AM_CPPFLAGS += $(GTK_MAC_CFLAGS)
-endif
-
 if ENABLE_SEARCH_PROVIDER
 searchproviderdir=@libexecdir@
 searchprovider_PROGRAMS = qalculate-search-provider
@@ -86,10 +82,6 @@
 	@WEBKITGTK_LIBS@ \
 	@QALCULATE_LIBS@
 
-if PLATFORM_MACOS
-qalculate_gtk_LDADD += $(GTK_MAC_LIBS)
-endif
-
 if ENABLE_SEARCH_PROVIDER
 gnome-search-provider2.c: $(top_srcdir)/data/org.gnome.ShellSearchProvider2.xml
 	$(AM_V_GEN)gdbus-codegen \

Just use a macro which is sensible but not defined:

--- src/main.cc	2026-03-25 18:29:16.000000000 +0800
+++ src/main.cc	2026-06-23 07:20:54.000000000 +0800
@@ -20,7 +20,7 @@
 #ifdef G_OS_UNIX
 #	include <glib-unix.h>
 #endif
-#ifdef HAVE_GTK_MAC
+#ifdef GTK_USE_QUARTZ
 # include <gtkosxapplication.h>
 #endif
 #ifndef _MSC_VER
@@ -139,7 +139,7 @@
 
 	g_application_set_default(G_APPLICATION(app));
 	gtk_window_set_application(main_window(), app);
-#ifdef HAVE_GTK_MAC 
+#ifdef GTK_USE_QUARTZ 
   GtkosxApplication *osx_app = (GtkosxApplication *)g_object_new(GTKOSX_TYPE_APPLICATION, NULL);
   GtkWidget *menubar = GTK_WIDGET(gtk_builder_get_object(main_builder, "menubar"));
   if (menubar) {
