From 485eb432c5b47bf346e6915700ac6d41747b7c76 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:47:16 +0800
Subject: [PATCH 1/4] strlcat.h, strlcpy.h: use declarations only when needed

---
 missing/strlcat.h | 2 ++
 missing/strlcpy.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/missing/strlcat.h b/missing/strlcat.h
index 702f250..78b2c92 100644
--- a/missing/strlcat.h
+++ b/missing/strlcat.h
@@ -1 +1,3 @@
+#if !HAVE_STRLCAT
 size_t strlcat(char *dst, const char *src, size_t siz);
+#endif
diff --git a/missing/strlcpy.h b/missing/strlcpy.h
index ebf5b5f..1cd37c4 100644
--- a/missing/strlcpy.h
+++ b/missing/strlcpy.h
@@ -1 +1,3 @@
+#if !HAVE_STRLCPY
 size_t strlcpy(char *dst, const char *src, size_t siz);
+#endif
-- 
2.49.0

