From 337d6234aa4a84c9228091fd133d094390363eda Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:49:03 +0800
Subject: [PATCH 2/4] Makefile.in: do not hardcode prefix

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 1e1b052..e031b1f 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -17,7 +17,7 @@ mandir		= @mandir@
 
 VER=2.5a1
 CC	= @CC@
-CFLAGS	= @CFLAGS@ -DDSNIFF_LIBDIR=\"$(libdir)/\" -I/usr/include/tirpc -DVERSION=\"$(VER)\"
+CFLAGS	= @CFLAGS@ -DDSNIFF_LIBDIR=\"$(libdir)/\" -I$(prefix)/include/tirpc -DVERSION=\"$(VER)\"
 CPPFLAGS = @CPPFLAGS@
 LDFLAGS	= @LDFLAGS_STATIC@ @LDFLAGS@
 
-- 
2.49.0

