From 40dda070fb0c3496017ba20f0cd24e9f95e932bc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:50:57 +0800
Subject: [PATCH 3/4] pcaputil.c: fix for Apple

---
 pcaputil.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pcaputil.c b/pcaputil.c
index de3b50e..eae16ed 100644
--- a/pcaputil.c
+++ b/pcaputil.c
@@ -12,7 +12,7 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 
-#if defined(BSD) && !defined(__FreeBSD__)
+#if defined(BSD) && !defined(__FreeBSD__) && !defined(__APPLE__)
 # define WITH_BPF   1
 #endif
 
-- 
2.49.0

