From 0f35845995e40a6ad4681689e5eb7c490423ba10 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:52:06 +0800
Subject: [PATCH 4/4] arpspoof.c: fix for Apple

---
 arpspoof.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arpspoof.c b/arpspoof.c
index f4754be..2ee88c1 100644
--- a/arpspoof.c
+++ b/arpspoof.c
@@ -31,7 +31,11 @@
 
 #include "arp.h"
 
+#ifdef __APPLE__
+extern char *ether_ntoa(const struct ether_addr *);
+#else
 extern char *ether_ntoa(struct ether_addr *);
+#endif
 
 struct host {
 	in_addr_t ip;
-- 
2.49.0

