From 5b783693abcaecf94035ec68db53b2d054d6197a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 17:37:54 +0800
Subject: [PATCH] nfs_prot.x: fix for Apple

---
 nfs_prot.x | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/nfs_prot.x b/nfs_prot.x
index 9bc5ab9..8ffc5b1 100644
--- a/nfs_prot.x
+++ b/nfs_prot.x
@@ -9,6 +9,10 @@
 
 %#include "config.h"
 
+#ifdef __APPLE__
+%#define xdr_uint64_t xdr_u_int64_t
+#endif
+
 const NFS_PORT          = 2049;
 const NFS_MAXDATA       = 8192;
 const NFS_MAXPATHLEN    = 1024;
