--- a/decode_pptp.c	2025-12-10 17:56:28.000000000 +0800
+++ b/decode_pptp.c	2026-01-18 09:46:32.000000000 +0800
@@ -186,15 +186,12 @@
 			chapch = (struct ppp_chap_challenge *)(chap + 1);
 			if ((p = strchr(name, '\\')) == NULL)
 				p = name;
-			
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+
 			SHA1_Init(&ctx);
 			SHA1_Update(&ctx, chapch->value.response_v2.peer_challenge, 16);
 			SHA1_Update(&ctx, save_challenge.challenge, 16);
 			SHA1_Update(&ctx, p, strlen(p));
 			SHA1_Final(digest, &ctx);
-#pragma GCC diagnostic pop
 
 			for (i = 0; i < 8; i++) {
 				buf_putf(&outbuf, "%02X", digest[i]);
