From 372e76d580359edb30f4ec520d3a0cd8ba9b4785 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 2 Aug 2025 09:17:23 +0800
Subject: [PATCH] util.c: add missing header, fixes macOS build

---
 tools/blisp/src/util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git tools/blisp/src/util.c tools/blisp/src/util.c
index 906985a..7263dcf 100644
--- tools/blisp/src/util.c
+++ tools/blisp/src/util.c
@@ -10,6 +10,7 @@
 // to free an allocated buffer on the caller.nn
 
 #include <stdlib.h>
+#include <mach-o/dyld.h>
 
 static void util_get_executable_path(char* buffer_out, uint32_t max_size) {
   assert(max_size >= PATH_MAX);  // n.b. 1024 on MacOS. 4K on most Linux.
-- 
2.48.0

