--- tools.ml.orig	2026-02-22 13:34:42.000000000 +0800
+++ tools.ml	2026-08-08 00:27:30.000000000 +0800
@@ -276,7 +276,7 @@
     external length : 'a array -> int = "%array_length"
     external get : 'a array -> int -> 'a = "%array_safe_get"
     external set : 'a array -> int -> 'a -> unit = "%array_safe_set"
-    external make : int -> 'a -> 'a t = "caml_make_vect"
+    val make : int -> 'a -> 'a t
     val filter : ('a -> bool) -> 'a array -> 'a array
     val filter_row_by_colmatch :
       ('a -> bool) -> 'a array array -> 'a array array
