--- harfbuzz/src/hb-coretext.cc	2026-06-04 05:17:28.000000000 +0800
+++ harfbuzz/src/hb-coretext.cc	2026-07-04 12:22:06.000000000 +0800
@@ -523,6 +523,8 @@
   CFDictionaryRef variations = CTFontCopyVariation (ct_font);
   if (variations)
   {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
     HB_SCOPE_GUARD (CFRelease (variations));
 
     hb_vector_t<hb_variation_t> vars;
@@ -546,6 +548,7 @@
       }
       hb_font_set_variations (font, vars.arrayZ, vars.length);
     }
+#pragma GCC diagnostic pop
   }
 
   /* Let there be dragons here... */

--- harfbuzz/src/hb-coretext-font.cc	2026-06-04 05:17:28.000000000 +0800
+++ harfbuzz/src/hb-coretext-font.cc	2026-07-04 12:21:37.000000000 +0800
@@ -299,6 +299,8 @@
 			       hb_glyph_extents_t *extents,
 			       void *user_data HB_UNUSED)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wnarrowing"
   CTFontRef ct_font = (CTFontRef) (const void *) font->data.coretext;
 
   CGFloat ct_font_size = CTFontGetSize (ct_font);
@@ -313,6 +315,7 @@
   extents->y_bearing = round ((bounds.origin.y + bounds.size.height) * y_mult);
   extents->width = round (bounds.size.width * x_mult);
   extents->height = round (bounds.origin.y * y_mult) - extents->y_bearing;
+#pragma GCC diagnostic pop
 
   return true;
 }
