From 092f273f96e79e1a281b0bf6d404aa2d3a7cb579 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 22 Jan 2026 04:27:42 +0800
Subject: [PATCH 1/5] main.m: fix macOS version threshold

---
 XLD/main.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/XLD/main.m b/XLD/main.m
index 0077631..8e5ab8b 100644
--- a/XLD/main.m
+++ b/XLD/main.m
@@ -45,7 +45,7 @@
 			Method toMethod   = class_getInstanceMethod([NSBundle class],@selector(mod_infoDictionary2));
 			method_exchangeImplementations(fromMethod, toMethod);
 			XLDDarkModeSupportEnabled = 1;
-			[pref registerDefaults:@{@"NSLayerPerformanceUpdates": @(YES)}];
+			[NSDictionary dictionaryWithObject:[NSNumber numberWithBool:YES] forKey:@"NSLayerPerformanceUpdates"];
 		}
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 101400
 		else {
