From dff367a20c429fbb0612e21864adadac63ffce42 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 22 Jan 2026 06:43:32 +0800
Subject: [PATCH 5/5] XLDFlacOutput: adjust macOS version threshold

---
 XLDFlacOutput/XLDFlacOutputTask.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/XLDFlacOutput/XLDFlacOutputTask.m b/XLDFlacOutput/XLDFlacOutputTask.m
index 2a70a7c..b735897 100644
--- a/XLDFlacOutput/XLDFlacOutputTask.m
+++ b/XLDFlacOutput/XLDFlacOutputTask.m
@@ -492,7 +492,7 @@ extern FLAC__bool FLAC__stream_encoder_disable_verbatim_subframes(FLAC__StreamEn
 	
 	if(![[configurations objectForKey:@"SetOggS"] boolValue]) return;
 	
-#if MAC_OS_X_VERSION_MAX_ALLOWED < 1050
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
 	FSRef ref;
 	OSErr err;
 	FSCatalogInfoBitmap	myInfoWanted = kFSCatInfoFinderInfo;
-- 
2.52.0

