--- a/XLD/XLDController.m	2023-04-01 11:55:08.000000000 +0800
+++ b/XLD/XLDController.m	2023-04-18 10:48:35.000000000 +0800
@@ -2828,7 +2828,7 @@
 		[task setTrack:track];
 	}
 	if(removeFlag && [o_warnBeforeConversion state] == NSOnState) {
-#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060 && !defined(__ppc__)
 		__block int ret;
 		if (![NSThread isMainThread]) {
 			dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);

