diff --git a/player/clipboard/clipboard.c b/player/clipboard/clipboard.c
index 9a21ff6ddd..056470b5af 100644
--- a/player/clipboard/clipboard.c
+++ b/player/clipboard/clipboard.c
@@ -30,7 +30,9 @@ struct clipboard_opts {
 
 // backend list
 extern const struct clipboard_backend clipboard_backend_win32;
+#if HAVE_COCOA && HAVE_SWIFT
 extern const struct clipboard_backend clipboard_backend_mac;
+#endif
 extern const struct clipboard_backend clipboard_backend_wayland;
 extern const struct clipboard_backend clipboard_backend_x11;
 extern const struct clipboard_backend clipboard_backend_vo;
@@ -39,7 +41,7 @@ static const struct clipboard_backend *const clipboard_backend_list[] = {
 #if HAVE_WIN32_DESKTOP
     &clipboard_backend_win32,
 #endif
-#if HAVE_COCOA
+#if HAVE_COCOA && HAVE_SWIFT
     &clipboard_backend_mac,
 #endif
 #if HAVE_WAYLAND && HAVE_WAYLAND_PROTOCOLS_1_39
