--- audio/out/ao_coreaudio_chmap.c	2025-03-26 01:33:27.000000000 +0800
+++ audio/out/ao_coreaudio_chmap.c	2025-05-12 05:09:09.000000000 +0800
@@ -22,7 +22,10 @@
 #include "ao_coreaudio_utils.h"
 
 #include "ao_coreaudio_chmap.h"
-#include <CoreAudioTypes/CoreAudioTypes.h>
+
+#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
+    #include <CoreAudioTypes/CoreAudioTypes.h>
+#endif
 
 static const int speaker_map[][2] = {
     { kAudioChannelLabel_Left,                 MP_SPEAKER_ID_FL   },
