--- meson.build	2025-03-26 01:33:27.000000000 +0800
+++ meson.build	2025-05-12 03:57:26.000000000 +0800
@@ -361,13 +361,10 @@
 dependencies += cc.find_library('atomic', required: false)
 
 cplugins = get_option('cplugins').require(
-    win32 or (features['libdl'] and cc.has_link_argument('-rdynamic')),
+    win32 or (features['libdl']),
     error_message: 'cplugins not supported by the os or compiler!',
 )
 features += {'cplugins': cplugins.allowed()}
-if features['cplugins'] and not win32
-    link_flags += '-rdynamic'
-endif
 
 win32_threads = get_option('win32-threads').require(win32)
 
