--- python/pyproject.toml.in	2026-02-05 18:34:28.000000000 +0800
+++ python/pyproject.toml.in	2026-02-08 02:05:36.000000000 +0800
@@ -17,6 +17,6 @@
 [[tool.setuptools.ext-modules]]
 name = "dnet"
 sources = ["dnet.pyx"]
-include-dirs = ["../include"@WIN32_INCLUDES@]
-library-dirs = ["../src/.libs"@WIN32_LIBRARY_DIRS@]
-libraries = ["dnet"@WIN32_LIBRARIES@]
+include-dirs = ["../include"]
+library-dirs = ["@prefix@/lib"]
+libraries = ["dnet"]
