From f313aa15196cc974d9ac39e4e10e10a367378411 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 00:35:22 +0800
Subject: [PATCH 01/12] deadbeef.h: add missing header for macOS

---
 include/deadbeef/deadbeef.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/deadbeef/deadbeef.h b/include/deadbeef/deadbeef.h
index 8f8cf9c94..a11f2bd26 100644
--- a/include/deadbeef/deadbeef.h
+++ b/include/deadbeef/deadbeef.h
@@ -31,6 +31,10 @@
 #include <stdarg.h>
 #include <unistd.h>
 
+#ifdef __APPLE__
+#include <sys/types.h> // ssize_t
+#endif
+
 struct scriptableItem_s;
 typedef struct scriptableItem_s ddb_scriptable_item_t;
 
-- 
2.49.0

