From b7ec4fc84412daf5bc4bec4dd679d50997e29977 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 8 Jun 2025 16:35:53 +0800
Subject: [PATCH 02/12] streamer.c: fix include

---
 src/streamer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/streamer.c b/src/streamer.c
index 033da0d19..762f1af0f 100644
--- a/src/streamer.c
+++ b/src/streamer.c
@@ -60,7 +60,7 @@
 #include "viz.h"
 #include "fft.h"
 #ifdef __APPLE__
-#    include "coreaudio.h"
+#    include "plugins/coreaudio/coreaudio.h"
 #endif
 
 #ifdef trace
-- 
2.49.0

