From 60af2f5bc91955883ab5c3719c7aeb8089aea8d7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 8 Jun 2025 20:39:05 +0800
Subject: [PATCH 03/12] moduleconf.h: unbreak coreaudio

---
 src/moduleconf.h | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/moduleconf.h b/src/moduleconf.h
index 667cde5b5..b73ff1703 100644
--- a/src/moduleconf.h
+++ b/src/moduleconf.h
@@ -2,10 +2,6 @@ PLUG(stdio)
 #ifdef HAVE_COCOAUI
 PLUG(cocoaui)
 #endif
-#ifdef HAVE_COREAUDIO
-PLUG(coreaudio)
-PLUG(nullout)
-#endif
 #ifdef HAVE_XGUI
 PLUG(xgui)
 #endif
-- 
2.49.0

