From efe5178e530ea5568455938fe5da914acb98d17d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 8 Jun 2025 19:04:04 +0800
Subject: [PATCH 04/12] junklib.c: unbreak iconv

---
 src/junklib.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/junklib.c b/src/junklib.c
index 6529aa03e..92a4e7fbc 100644
--- a/src/junklib.c
+++ b/src/junklib.c
@@ -32,9 +32,6 @@
 #include <stdlib.h>
 #include <string.h>
 #if HAVE_ICONV
-  #if !defined(__MINGW32__) && !defined(__OpenBSD__)
-  #define LIBICONV_PLUG
-  #endif
   #include <iconv.h>
 #elif HAVE_ICU
   #warning icu
-- 
2.49.0

