From f6b8a96e7d3a59aa5cf0129499abb1ccf0efe802 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 03:21:49 +0800
Subject: [PATCH 14/19] bswap.h: define G_GINT64_CONSTANT

---
 plugins/mms/libmms/bswap.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/plugins/mms/libmms/bswap.h b/plugins/mms/libmms/bswap.h
index afa80e080..76e650f67 100644
--- a/plugins/mms/libmms/bswap.h
+++ b/plugins/mms/libmms/bswap.h
@@ -20,6 +20,8 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
+/* How was it supposed to work without defining G_GINT64_CONSTANT? */
+#define G_GINT64_CONSTANT(val) (val##L)
 
 /* NOTE:
  * Now, to clear up confusion: LE_XX means "from LE to native, XX bits wide"
-- 
2.49.0

