From b0d820558eae824df41cc7a7bc86a2db9c985bc6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 03:22:35 +0800
Subject: [PATCH 15/19] coreaudio.h: define kAudioDeviceTransportTypeAirPlay

---
 plugins/coreaudio/coreaudio.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/plugins/coreaudio/coreaudio.h b/plugins/coreaudio/coreaudio.h
index 2fea2414e..1e6aefd21 100644
--- a/plugins/coreaudio/coreaudio.h
+++ b/plugins/coreaudio/coreaudio.h
@@ -26,4 +26,8 @@
 
 #define DDB_COREAUDIO_FLAG_AIRPLAY 0x00010000
 
+#ifndef kAudioDeviceTransportTypeAirPlay
+#define kAudioDeviceTransportTypeAirPlay 'airp'
+#endif
+
 #endif /* coreaudio_h */
-- 
2.49.0

