From 7c5debd1f6d23c369df1afbf361a4b7d85ec99d2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 03:25:10 +0800
Subject: [PATCH 16/19] main.c: fix Apple macros

---
 src/main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index c4808b521..45a28c92d 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1332,7 +1332,7 @@ main (int argc, char *argv[]) {
     }
     (void)mkdir_recursive (confdir);
 
-#if __APPLE__
+#if OSX_APPBUNDLE
     char appcachepath[PATH_MAX];
     cocoautil_get_cache_path (appcachepath, sizeof (appcachepath));
     if (snprintf (dbcachedir, sizeof (dbcachedir), "%s/Deadbeef", appcachepath) > (int)sizeof (dbcachedir)) {
@@ -1454,7 +1454,7 @@ main (int argc, char *argv[]) {
     }
 
     // State directory
-#if __APPLE__
+#if OSX_APPBUNDLE
     char statedir[PATH_MAX];
     cocoautil_get_application_support_path (statedir, sizeof (statedir));
     if (snprintf (dbstatedir, sizeof (dbstatedir), "%s/Deadbeef/State", statedir) > (int)sizeof (dbstatedir)) {
-- 
2.49.0

