From 647cfc3d0ece43de56736eca4d15c13327e7a555 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 03:49:43 +0800
Subject: [PATCH 18/19] gtkui: restore 10b8f9f but blockless

---
 plugins/gtkui/gtkui.c | 19 +++++++++++++++++++
 plugins/gtkui/gtkui.h |  3 +++
 2 files changed, 22 insertions(+)

diff --git a/plugins/gtkui/gtkui.c b/plugins/gtkui/gtkui.c
index cc0352b82..3b4ac3794 100644
--- a/plugins/gtkui/gtkui.c
+++ b/plugins/gtkui/gtkui.c
@@ -144,6 +144,25 @@ init_widget_layout (void);
 static int sb_context_id = -1;
 static char sb_text[512];
 
+typedef void (*gtkui_dispatch_func_t)(void *userdata);
+
+static gboolean
+_dispatch_on_main_wrapper(void *context) {
+    gtkui_dispatch_func_t func = ((gtkui_dispatch_func_t *)context)[0];
+    void *userdata = ((void **)context)[1];
+    func(userdata);
+    free(context);
+    return FALSE;
+}
+
+void
+gtkui_dispatch_on_main(gtkui_dispatch_func_t func, void *userdata) {
+    void **context = malloc(2 * sizeof(void *));
+    context[0] = (void *)func;
+    context[1] = userdata;
+    g_idle_add(_dispatch_on_main_wrapper, context);
+}
+
 static void
 format_timestr (char *buf, int sz, float time) {
     time = roundf (time);
diff --git a/plugins/gtkui/gtkui.h b/plugins/gtkui/gtkui.h
index de10c6bfc..864e91639 100644
--- a/plugins/gtkui/gtkui.h
+++ b/plugins/gtkui/gtkui.h
@@ -196,4 +196,7 @@ show_file_chooser (const gchar *title, enum GtkuiFileChooserType type, gboolean
 char *
 gtkui_trim_whitespace (char *p, size_t len);
 
+typedef void (*gtkui_dispatch_func_t)(void *userdata);
+void gtkui_dispatch_on_main(gtkui_dispatch_func_t func, void *userdata);
+
 #endif
-- 
2.49.0

