From 52245751c1b4d495a68d5e74d11db9ccbc7ce231 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 03:54:18 +0800
Subject: [PATCH 19/19] gtkui: undo_process blockless

---
 plugins/gtkui/gtkui.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/plugins/gtkui/gtkui.c b/plugins/gtkui/gtkui.c
index 3b4ac3794..da91af8d1 100644
--- a/plugins/gtkui/gtkui.c
+++ b/plugins/gtkui/gtkui.c
@@ -978,6 +978,10 @@ trackfocus_cb (gpointer data) {
     return FALSE;
 }
 
+static void _undo_process_cb(void *userdata) {
+    deadbeef->undo_process();
+}
+
 static int
 gtkui_message (uint32_t id, uintptr_t ctx, uint32_t p1, uint32_t p2) {
     if (!gtkui_accept_messages) {
@@ -985,9 +989,7 @@ gtkui_message (uint32_t id, uintptr_t ctx, uint32_t p1, uint32_t p2) {
     }
 
     if (id == DB_EV_PLAYLISTCHANGED) {
-        gtkui_dispatch_on_main(^{
-            deadbeef->undo_process();
-        });
+        gtkui_dispatch_on_main(_undo_process_cb, NULL);
     }
 
     switch (id) {
-- 
2.49.0

