From cde70fac4d14a54210d949844cc6aea741e344e6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 9 Jun 2025 05:32:07 +0800
Subject: [PATCH 21/23] main.c: quick hack for undefined *_deinit functions

---
 src/main.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index 45a28c92d..81b5ac6f6 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1067,7 +1067,8 @@ _async_exit_handler(void) {
 
     conf_free ();
 
-    tf_deinit ();
+// FIXME: this is a hack: tf_deinit and metacache_deinit are not found.
+//    tf_deinit ();
 
     trace ("messagepump_free\n");
     messagepump_free ();
@@ -1075,7 +1076,7 @@ _async_exit_handler(void) {
     plug_cleanup ();
     trace ("logger_free\n");
 
-    metacache_deinit ();
+//    metacache_deinit ();
 
     trace ("💛💙\n");
     ddb_logger_free ();
-- 
2.49.0

