--- src/gn/setup.cc	2026-04-01 06:56:53
+++ src/gn/setup.cc	2026-04-27 19:55:01
@@ -859,7 +859,7 @@
     }
     build_settings_.SetPythonPath(std::move(python_path));
 #else
-    build_settings_.SetPythonPath(base::FilePath("python"));
+    build_settings_.SetPythonPath(base::FilePath("@PYTHON@"));
 #endif
   }
   return true;

--- src/gn/exec_process_unittest.cc
+++ src/gn/exec_process_unittest.cc
@@ -33,7 +33,7 @@
   args.push_back(L"-c");
   args.push_back(base::UTF8ToUTF16(command));
 #else
-  args.push_back("python3");
+  args.push_back("@PYTHON@");
   args.push_back("-c");
   args.push_back(command);
 #endif
