--- src/libs/JustFastUi/JustFastUi.h	2023-12-18 16:50:27.000000000 +0800
+++ src/libs/JustFastUi/JustFastUi.h	2025-05-27 21:20:39.000000000 +0800
@@ -50,6 +50,6 @@
 
     void setQuitFunction(std::function<void()>);
 
-    ftxui::Element Render() override;
+    ftxui::Element OnRender() override;
     bool OnEvent(ftxui::Event) override;
 };

--- src/libs/JustFastUi/JustFastUi.cpp	2023-12-18 16:50:27.000000000 +0800
+++ src/libs/JustFastUi/JustFastUi.cpp	2025-05-27 21:20:31.000000000 +0800
@@ -253,7 +253,7 @@
 }
 
 // clang-format off
-ftxui::Element JustFastUi::Render()
+ftxui::Element JustFastUi::OnRender()
 {
     using namespace ftxui;
 
@@ -266,9 +266,9 @@
       
     auto mainView =
         hbox({
-            parentFolder->Render() | frame,
+            parentFolder->OnRender() | frame,
             separator(),
-            currentFolder->Render() | flex | frame
+            currentFolder->OnRender() | flex | frame
 	});
 
     auto statusLine =
