--- fd-8.4.0/build.rs	2022-05-29 17:21:45.000000000 +0800
+++ fd-8.4.0/build.rs	2026-01-07 00:10:36.000000000 +0800
@@ -7,15 +7,12 @@
 include!("src/app.rs");
 
 fn main() {
-    let min_version = "1.56";
+    let min_version = "1.54";
 
     match version_check::is_min_version(min_version) {
         Some(true) => {}
         // rustc version too small or can't figure it out
-        _ => {
-            eprintln!("'fd' requires rustc >= {}", min_version);
-            std::process::exit(1);
-        }
+        _ => {}
     }
 
     let var = std::env::var_os("SHELL_COMPLETIONS_DIR").or_else(|| std::env::var_os("OUT_DIR"));
