--- .home/.cargo/macports/proc-macro-error-1.0.4/build.rs	2026-01-06 22:47:13.000000000 +0800
+++ .home/.cargo/macports/proc-macro-error-1.0.4/build.rs	2026-01-05 00:08:18.000000000 +0800
@@ -2,10 +2,4 @@
     if !version_check::is_feature_flaggable().unwrap_or(false) {
         println!("cargo:rustc-cfg=use_fallback");
     }
-
-    if version_check::is_max_version("1.38.0").unwrap_or(false)
-        || !version_check::Channel::read().unwrap().is_stable()
-    {
-        println!("cargo:rustc-cfg=skip_ui_tests");
-    }
 }
