From bf3db3206dbcc559380184ceea3958413a29858e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 23 May 2026 15:45:03 +0800
Subject: [PATCH 1/9] xquartz/X11Controller.h: define strong to retain if
 undefined

---
 hw/xquartz/X11Controller.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hw/xquartz/X11Controller.h b/hw/xquartz/X11Controller.h
index a6b5d87c9..d1f95fb6b 100644
--- a/hw/xquartz/X11Controller.h
+++ b/hw/xquartz/X11Controller.h
@@ -42,6 +42,10 @@
 #undef BOOL
 #endif
 
+#ifndef strong
+#define strong retain
+#endif
+
 @interface X11Controller : NSObject <NSTableViewDataSource>
 @property (nonatomic, readwrite, strong) IBOutlet NSPanel *prefs_panel;
 
-- 
2.54.0

