From 38a51ef58515e40677bd2c377c4a2fe403b95c70 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 23 May 2026 21:44:23 +0800
Subject: [PATCH] xquartz: work around meson bug

diff --git a/hw/xquartz/bundle/meson.build b/hw/xquartz/bundle/meson.build
index 22941203b..29d0e5fce 100644
--- a/hw/xquartz/bundle/meson.build
+++ b/hw/xquartz/bundle/meson.build
@@ -55,5 +55,3 @@ install_data('PkgInfo',
 install_data('X11.sh',
     install_dir: join_paths(bundle_root, 'Contents/MacOS'),
     install_mode: 'rwxr-xr-x')
-
-meson.add_install_script('chown-bundle.sh', bundle_root)

