From 2e61c42e752c432d72397263623b7d603dc1cdc0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 28 May 2026 16:47:30 +0800
Subject: [PATCH 3/8] xquartz: xprFrame.c: drop duplicate variable in fallback
 code

---
 hw/xquartz/xpr/xprFrame.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/hw/xquartz/xpr/xprFrame.c b/hw/xquartz/xpr/xprFrame.c
index 9e28625b2..68d054a09 100644
--- a/hw/xquartz/xpr/xprFrame.c
+++ b/hw/xquartz/xpr/xprFrame.c
@@ -576,7 +576,6 @@ xprGetXWindow(xp_window_id wid)
 
     dispatch_sync_f(window_hash_serial_q, &ctx, windowGet);
 #else
-    RootlessWindowRec *winRec;
     pthread_rwlock_rdlock(&window_hash_rwlock);
     winRec = x_hash_table_lookup(window_hash, x_cvt_uint_to_vptr(wid), NULL);
     pthread_rwlock_unlock(&window_hash_rwlock);
-- 
2.54.0

