From 532c7fe0f9609bcaf1cc9880c8e6b3632d6ea4c8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 28 May 2026 21:19:06 +0800
Subject: [PATCH 7/8] xquartz: X11Controller.m: add one missing synthesize call

---
 hw/xquartz/X11Controller.m | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/xquartz/X11Controller.m b/hw/xquartz/X11Controller.m
index bf1b34fdd..d828bcf5f 100644
--- a/hw/xquartz/X11Controller.m
+++ b/hw/xquartz/X11Controller.m
@@ -113,6 +113,7 @@ extern char *bundle_id_prefix;
 @synthesize dock_apps_menu = _dock_apps_menu;
 @synthesize apps_table = _apps_table;
 @synthesize dock_menu = _dock_menu;
+@synthesize can_quit = _can_quit;
 
 - (void) awakeFromNib
 {
-- 
2.54.0

