From c1defbddeaaf47f82c82234a68540956d7043555 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 23 May 2026 16:59:51 +0800
Subject: [PATCH 7/9] xquartz: objc_autoreleasePool* only supported with clang
 and > 10.6

---
 hw/xquartz/quartz.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/hw/xquartz/quartz.c b/hw/xquartz/quartz.c
index 06d1b09a5..2feab69f4 100644
--- a/hw/xquartz/quartz.c
+++ b/hw/xquartz/quartz.c
@@ -64,6 +64,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
+#include <AvailabilityMacros.h>
 #include <IOKit/pwr_mgt/IOPMLib.h>
 #include <libkern/OSAtomic.h>
 #include <signal.h>
@@ -77,8 +78,10 @@
 // in the OS without major bincompat ramifications.
 //
 // These were added in macOS 10.7.
+#if defined(__clang__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
 void * _Nonnull objc_autoreleasePoolPush(void);
 void objc_autoreleasePoolPop(void * _Nonnull context);
+#endif
 
 DevPrivateKeyRec quartzScreenKeyRec;
 int aquaMenuBarHeight = 0;
@@ -160,12 +163,14 @@ QuartzSetupScreen(int index,
 static void
 QuartzBlockHandler(void *blockData, void *pTimeout)
 {
+#if defined(__clang__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
     static void *poolToken = NULL;
 
     if (poolToken) {
         objc_autoreleasePoolPop(poolToken);
     }
     poolToken = objc_autoreleasePoolPush();
+#endif
 }
 
 /*
-- 
2.54.0

