From 9fe2a60754b53c0546aba95e95bf72e4d4a97d39 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 28 May 2026 23:41:36 +0800
Subject: [PATCH 8/8] xquartz: x-hash.h: a hack for ppc64

---
 hw/xquartz/xpr/x-hash.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hw/xquartz/xpr/x-hash.h b/hw/xquartz/xpr/x-hash.h
index 2cdae7c5d..628ef11b5 100644
--- a/hw/xquartz/xpr/x-hash.h
+++ b/hw/xquartz/xpr/x-hash.h
@@ -82,8 +82,9 @@ X_PFX(cvt_vptr_to_uint) (void * val) {
 
     /* If this assert fails, chances are val actually is a pointer,
        or there's been memory corruption */
+#ifndef __ppc64__
     assert(sv == uv);
-
+#endif
     return uv;
 }
 
-- 
2.54.0

