From 27f560fcd9fe23c0ff538b1071e2af030826b9f1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 25 Dec 2025 02:36:32 +0800
Subject: [PATCH 8/9] xquartz: define strong to retain

---
 hw/xquartz/X11Controller.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hw/xquartz/X11Controller.h b/hw/xquartz/X11Controller.h
index db1e7e9aa..b7130895f 100644
--- a/hw/xquartz/X11Controller.h
+++ b/hw/xquartz/X11Controller.h
@@ -44,6 +44,10 @@
 #undef BOOL
 #endif
 
+#ifndef strong
+#define strong retain
+#endif
+
 @interface X11Controller : NSObject <NSTableViewDataSource>
 @property (nonatomic, readwrite, strong) IBOutlet NSPanel *prefs_panel;
 
-- 
2.52.0

