diff --git a/hw/xquartz/bundle/meson.build b/hw/xquartz/bundle/meson.build
index 22941203b..29d0e5fce 100644
--- a/hw/xquartz/bundle/meson.build
+++ b/hw/xquartz/bundle/meson.build
@@ -55,5 +55,3 @@ install_data('PkgInfo',
 install_data('X11.sh',
     install_dir: join_paths(bundle_root, 'Contents/MacOS'),
     install_mode: 'rwxr-xr-x')
-
-meson.add_install_script('chown-bundle.sh', bundle_root)
