diff --git a/hw/xquartz/bundle/meson.build b/hw/xquartz/bundle/meson.build
index 22941203b..29d0e5fce 100644
--- a/hw/xquartz/bundle/meson.build
+++ b/hw/xquartz/bundle/meson.build
@@ -55,5 +55,3 @@ install_data('PkgInfo',
 install_data('X11.sh',
     install_dir: join_paths(bundle_root, 'Contents/MacOS'),
     install_mode: 'rwxr-xr-x')
-
-meson.add_install_script('chown-bundle.sh', bundle_root)

diff --git a/hw/xquartz/pbproxy/meson.build b/hw/xquartz/pbproxy/meson.build
index 206cbdca3..ffdfbcac7 100644
--- a/hw/xquartz/pbproxy/meson.build
+++ b/hw/xquartz/pbproxy/meson.build
@@ -20,13 +20,19 @@ libxpbproxy = static_library('xpbproxy',
     objc_args: pbproxy_defs,
 )
 
-cocoa = dependency('Cocoa', method: 'extraframework')
+if cc.get_id() == 'clang'
+  cocoa = dependency('Cocoa', method: 'extraframework')
+  xpbproxy_linkargs = []
+else
+  xpbproxy_linkargs = ['-framework Cocoa']
+endif
 
 # standalone xpbproxy
 if build_standalone_pbproxy
     executable('xpbproxy',
         'app-main.m',
         link_with: libxpbproxy,
+        link_args: xpbproxy_linkargs,
         dependencies: [cocoa, dependency('x11')],
         objc_args: pbproxy_defs,
         install: true,

diff --git a/hw/xquartz/mach-startup/meson.build b/hw/xquartz/mach-startup/meson.build
index 2f7694252..edb43e52e 100644
--- a/hw/xquartz/mach-startup/meson.build
+++ b/hw/xquartz/mach-startup/meson.build
@@ -28,14 +28,32 @@ x11appdir = join_paths(bundle_root, 'Contents/MacOS')
 
 x11_bin_deps = [
      meson.get_compiler('c').find_library('Xplugin'),
+]
+
+# Meson is a broken build system. It cannot find frameworks with gcc.
+if cc.get_id() == 'clang'
+  x11_bin_deps += [
      dependency('Carbon', method: 'extraframework'),
      cocoa,
      dependency('CoreAudio', method: 'extraframework'),
      dependency('IOKit', method: 'extraframework')
-]
+  ]
+  binlinkargs = ['-Objc']
+else
+  binlinkargs = [
+    '-framework', 'Carbon',
+    '-framework', 'Cocoa',
+    '-framework', 'CoreAudio',
+    '-framework', 'IOKit'
+  ]
+endif
 
 if build_glx
+  if cc.get_id() == 'clang'
     x11_bin_deps += [dependency('OpenGL', method: 'extraframework')]
+  else
+    binlinkargs += ['-framework', 'OpenGL']
+  endif
 endif
 
 if build_sparkle
@@ -65,26 +83,38 @@ x11_bin = executable('X11.bin',
      dependencies: [xproto_dep, x11_bin_deps, mach_startup_dep],
      include_directories: [inc, '..', top_dir_inc],
      c_args: xquartz_defs,
-     link_args: ['-Objc'],
+     link_args: binlinkargs,
      install: true,
      install_dir: x11appdir,
 )
 
+if cc.get_id() == 'clang'
+     execlinkargs = ['-Objc']
+else
+     execlinkargs = []
+endif
+
 # X11 (Bundle trampoline)
 x11 = executable('X11',
      [
       'bundle_trampoline.c',
      ],
      c_args: xquartz_defs,
-     link_args: ['-Objc'],
+     link_args: execlinkargs,
      install: true,
      install_dir: x11appdir,
 )
 
 # Xquartz
-xquartz_deps = [
+if cc.get_id() == 'clang'
+  xquartz_deps = [
     dependency('CoreServices', method: 'extraframework'),
-]
+  ]
+  xquartz_linkargs = []
+else
+  xquartz_deps = []
+  xquartz_linkargs = ['-framework', 'CoreServices']
+endif
 
 xquartz = executable('Xquartz',
     [
@@ -94,6 +124,7 @@ xquartz = executable('Xquartz',
     ],
     include_directories: inc,
     c_args: xquartz_defs,
+    link_args: xquartz_linkargs,
     dependencies: [xquartz_deps, mach_startup_dep],
     install: true,
 )
