--- src/idle-server-connection.c.orig	2020-11-01 07:11:34.000000000 +0800
+++ src/idle-server-connection.c	2026-08-23 13:48:08.000000000 +0800
@@ -23,9 +23,9 @@
 #include "idle-server-connection.h"
 
 #include <string.h>
+#include <sys/socket.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <sys/socket.h>
 
 #include <telepathy-glib/telepathy-glib.h>
 
@@ -590,10 +590,7 @@
          * with null bytes gives us cleaner debug messages, without
          * affecting the readability of the code.
          */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wstringop-truncation"
 	strncpy(priv->output_buffer, cmd, output_buffer_size);
-#pragma GCC diagnostic pop
 
 	priv->nwritten = 0;
 
