--- src/dune.orig	2026-08-08 21:56:03.606885162 +0000
+++ src/dune	2026-08-08 22:39:40.732197377 +0000
@@ -23,18 +23,8 @@
     )
     (link_flags (:standard
         -cclib -lcurl
-        -cclib -lnghttp2
-        -cclib -lssl
-        -cclib -lcrypto
-        -cclib -lcares
-        -cclib -lbrotlidec
-        -cclib -lbrotlicommon
         -cclib -lz
         -cclib -lbz2
-        -cclib -lzstd
-        -cclib -lidn2
-        -cclib -lunistring
-        -cclib -lpsl
         -cclib -lcryptopp
         -cclib -lminiupnpc
         -cclib -lnatpmp
@@ -42,5 +32,7 @@
         -cclib -lgd
         -cclib -lpng
         -cclib -ljpeg
+        -cclib -liconv
+        -cclib @CXXLIB@
     ))
 )
