--- src/utils/cdk/dune.orig	2026-08-08 22:43:35.340248233 +0000
+++ src/utils/cdk/dune	2026-08-08 22:43:35.414250139 +0000
@@ -20,7 +20,7 @@
     (targets tar.ml)
     (deps tar.mlcpp)
     (action
-        (run cpp -x c -DOCAMLVERSION_MAJOR=%{env:OCAMLVERSION_MAJOR=4122} -I%{project_root}/../.. -P tar.mlcpp -o tar.ml)
+        (run %{env:MLDONKEY_CPP=cpp} -E -x c -DOCAMLVERSION_MAJOR=%{env:OCAMLVERSION_MAJOR=4122} -I%{project_root}/../.. -P tar.mlcpp -o tar.ml)
     )
 )
 
--- src/utils/lib/dune.orig	2026-08-08 22:43:35.347248414 +0000
+++ src/utils/lib/dune	2026-08-08 22:43:35.415250165 +0000
@@ -63,7 +63,7 @@
     (targets misc2.ml)
     (deps misc2.mlcpp)
     (action
-        (run cpp -x c -DOCAMLVERSION_MAJOR=%{env:OCAMLVERSION_MAJOR=4122} -I%{project_root}/../.. -P misc2.mlcpp -o misc2.ml)
+        (run %{env:MLDONKEY_CPP=cpp} -E -x c -DOCAMLVERSION_MAJOR=%{env:OCAMLVERSION_MAJOR=4122} -I%{project_root}/../.. -P misc2.mlcpp -o misc2.ml)
     )
 )
 
