--- configure	2026-01-14 02:38:53.000000000 +0800
+++ configure	2026-01-16 03:39:05.000000000 +0800
@@ -13271,10 +13271,6 @@
 printf "%s\n" "#define SIZEOF_TIME_T $ac_cv_sizeof_time_t" >>confdefs.h
 
 
-if test "$ac_cv_sizeof_time_t" -lt "8"; then
-	as_fn_error $? "rpki-client doesn't support broken OS with small time_t" "$LINENO" 5
-fi
-
 
 # Check whether --with-openssl was given.
 if test ${with_openssl+y}
