--- Makefile	2024-12-09 19:48:56.000000000 +0800
+++ Makefile	2025-12-14 01:22:16.000000000 +0800
@@ -16,6 +16,8 @@
 STATIC ?= 1
 SHARED ?= 0
 
+PREFIX ?= /opt/local
+
 SRCROOT = .
 TESTDIR = $(SRCROOT)/tests
 FTESTDIR = $(TESTDIR)/fortran
@@ -27,7 +29,7 @@
 SO_EXT := $(if $(DARWIN),dylib,so)
 
 ifneq (,$(strip $(DESTDIR)))
-  INSTALL_ROOT = $(DESTDIR)
+  INSTALL_ROOT = $(DESTDIR)$(PREFIX)
 else
   INSTALL_ROOT = $(SRCROOT)/build
 endif
@@ -44,7 +46,7 @@
     override CFLAGS += -fPIC
   endif
   ifneq ($(DARWIN),)
-    override LDFLAGS += -install_name @rpath/lib$(LIBNAME).$(SO_EXT)
+    override LDFLAGS += -install_name $(PREFIX)/lib/lib$(LIBNAME).$(SO_EXT)
   endif
 endif
 
