--- cffi-tests.asd.orig	2025-03-11 21:07:10.000000000 +0800
+++ cffi-tests.asd	2025-05-03 14:22:04.000000000 +0800
@@ -63,7 +63,6 @@
      (:file "bindings" :depends-on ("package" "libtest" "libtest2" "libfsbv"))
      (:file "funcall" :depends-on ("bindings"))
      (:file "defcfun" :depends-on ("bindings"))
-     (:file "callbacks" :depends-on ("bindings"))
      (:file "foreign-globals" :depends-on ("package"))
      (:file "memory" :depends-on ("package"))
      (:file "strings" :depends-on ("package"))
@@ -72,7 +71,6 @@
      (:file "union" :depends-on ("package"))
      (:file "enum" :depends-on ("package"))
      (:file "fsbv" :depends-on ("bindings" "enum"))
-     (:file "misc-types" :depends-on ("bindings"))
      (:file "misc" :depends-on ("bindings"))
      (:file "test-asdf" :depends-on ("package"))
      (:file "grovel" :depends-on ("package")))))
