--- meson.build	2024-05-04 04:12:25
+++ meson.build	2024-11-27 06:07:18
@@ -12,18 +12,18 @@
 
 message('Looking for dependencies')
 # We need pykeepass>=4.1.1 and pyotp>=2.4.0.
-python_bin = python.find_installation('python3', modules:[
+python_bin = python.find_installation('@PYTHON@', modules:[
   'pykeepass',
   'pyotp',
   'validators',
-  'zxcvbn_rs_py',
+  'zxcvbn',
   'PyKCS11',
   'yubico',
 ])
 if not python_bin.found()
-  error('No valid python3 binary found')
+  error('No valid python binary found')
 else
-  message('Found python3 binary')
+  message('Found python binary')
 endif
 
 if not python_bin.language_version().version_compare('>= 3.8')
@@ -39,7 +39,7 @@
 dependency('pygobject-3.0', version: '>= 3.52')
 dependency('gtksourceview-5', version: '>= 5.0')
 
-python_dir = python_bin.get_install_dir()
+python_dir = '@PY_PACKAGES@'
 DATA_DIR = join_paths(get_option('prefix'), get_option('datadir'), meson.project_name())
 bindir = join_paths(get_option('prefix'), get_option('bindir'))
 
