--- include/openssl/target.h	2023-11-17 07:31:50.000000000 +0800
+++ include/openssl/target.h	2026-01-27 07:19:02.000000000 +0800
@@ -45,6 +45,12 @@
 #define OPENSSL_RISCV64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 4
 #define OPENSSL_32_BIT
+#elif defined(__ppc64__) || defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#define OPENSSL_PPC64
+#elif defined(__ppc__) || defined(__powerpc__)
+#define OPENSSL_32_BIT
+#define OPENSSL_PPC
 #elif defined(__pnacl__)
 #define OPENSSL_32_BIT
 #define OPENSSL_PNACL


While legacy-support provides getentropy, it does not want to link statically, and then we need
to propagate dependency on it down the pipeline. For now, use urandom instead.

--- crypto/bcm_support.h	2025-03-01 07:12:34.000000000 +0800
+++ crypto/bcm_support.h	2026-01-29 13:09:48.000000000 +0800
@@ -49,7 +49,7 @@
 #define OPENSSL_RAND_TRUSTY
 #elif defined(OPENSSL_WINDOWS)
 #define OPENSSL_RAND_WINDOWS
-#elif defined(OPENSSL_LINUX)
+#elif defined(OPENSSL_LINUX) || defined(OPENSSL_APPLE)
 #define OPENSSL_RAND_URANDOM
 #elif defined(OPENSSL_APPLE) && !defined(OPENSSL_MACOS)
 // Unlike macOS, iOS and similar hide away getentropy().

--- crypto/rand/sysrand_internal.h	2025-03-01 07:12:34.000000000 +0800
+++ crypto/rand/sysrand_internal.h	2026-01-29 13:09:56.000000000 +0800
@@ -23,7 +23,7 @@
 #define OPENSSL_RAND_TRUSTY
 #elif defined(OPENSSL_WINDOWS)
 #define OPENSSL_RAND_WINDOWS
-#elif defined(OPENSSL_LINUX)
+#elif defined(OPENSSL_LINUX) || defined(OPENSSL_APPLE)
 #define OPENSSL_RAND_URANDOM
 #elif defined(OPENSSL_APPLE) && !defined(OPENSSL_MACOS)
 // Unlike macOS, iOS and similar hide away getentropy().
