From c0d909e2695b8503c0491a81274babcefcd749b9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 28 Feb 2026 00:50:45 +0800
Subject: [PATCH] Fix makefile

---
 Makefile.in | 134 +++++++++-------------------------------------------
 1 file changed, 22 insertions(+), 112 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 9a99f98..37880c1 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -29,7 +29,8 @@ LIBUNISTRING_VERSION := 1.1
 LIBUNISTRING_URL := https://ftp.gnu.org/gnu/libunistring/libunistring-$(LIBUNISTRING_VERSION).tar.gz
 LIBIDN2_VERSION := 2.3.7
 LIBIDN2_URL := https://ftp.gnu.org/gnu/libidn/libidn2-$(LIBIDN2_VERSION).tar.gz
-CURL_VERSION := curl-8_15_0
+CURL_TAG := 8_15_0
+CURL_DIST := curl-$(CURL_TAG)
 
 # The following variables will be set by the configure script.
 prefix = @prefix@
@@ -95,15 +96,13 @@ help: ## Show this help message
 .PHONY: help
 .DEFAULT_GOAL := help
 
-build: $(CURL_VERSION)/.chrome
-	cd $(CURL_VERSION)
-	# Don't pass this Makefile's MAKEFLAGS
-	$(MAKE) MAKEFLAGS=-j$(SUBJOBS)
+build: $(CURL_DIST)/.chrome
+	cd $(CURL_DIST) && $(MAKE) MAKEFLAGS=-j$(SUBJOBS)
 .PHONY: build
 
 checkbuild: ## Run basic checks on the built binary
 ifeq ($(host),$(build))
-	cd $(CURL_VERSION)
+	cd $(CURL_DIST)
 	# Make sure all needed features were compiled in
 	./src/curl-impersonate -V | grep -q zlib
 	./src/curl-impersonate -V | grep -q zstd
@@ -122,24 +121,13 @@ endif
 .PHONY: checkbuild
 
 install:
-	cd $(CURL_VERSION)
-	$(MAKE) install-exec DESTDIR="$(DESTDIR)" MAKEFLAGS=
-	mkdir -p "$(DESTDIR)@includedir@"
-	cp -R include/curl "$(DESTDIR)@includedir@"
-	# Wrapper scripts for the cli shortcut (e.g. 'curl_chrome99')
-	install $(srcdir)/bin/curl_* "$(DESTDIR)@bindir@"
+	cd $(CURL_DIST) && $(MAKE) install-exec DESTDIR="$(DESTDIR)" MAKEFLAGS= && install $(srcdir)/bin/curl_* "$(DESTDIR)@bindir@"
+
 .PHONY: install
 
 install-strip: ## Like 'install', but strip binaries for smaller size
-	cd $(CURL_VERSION)
-	$(MAKE) install-exec DESTDIR="$(DESTDIR)" MAKEFLAGS=
-	mkdir -p "$(DESTDIR)@includedir@"
-	cp -R include/curl "$(DESTDIR)@includedir@"
-	# We could have used 'install-strip' but then the docs would be installed as well.
-	# Instead strip manually.
-	$(STRIP) "$(DESTDIR)@bindir@/curl-impersonate"
-	# Wrapper scripts for the cli shortcut (e.g. 'curl_chrome99')
-	install $(srcdir)/bin/curl_* "$(DESTDIR)@bindir@"
+	cd $(CURL_DIST) && $(MAKE) install-exec DESTDIR="$(DESTDIR)" MAKEFLAGS= && \
+		$(STRIP) "$(DESTDIR)@bindir@/curl-impersonate" && install $(srcdir)/bin/curl_* "$(DESTDIR)@bindir@"
 .PHONY: install-strip
 
 uninstall: ## Uninstall the Chrome version of curl-impersonate after 'make install'
@@ -440,94 +428,16 @@ $(CURL_VERSION).tar.gz:
 	curl -L "https://github.com/curl/curl/archive/$(CURL_VERSION).tar.gz" \
 		-o "$(CURL_VERSION).tar.gz"
 
-# Apply the "Chorme version" patches and mark using a dummy file
-$(CURL_VERSION)/.patched: $(srcdir)/patches/curl.patch
-	rm -Rf $(CURL_VERSION)
-	tar -xf $(CURL_VERSION).tar.gz
-	mv curl-$(CURL_VERSION) $(CURL_VERSION)  # fix directory name
-	cd $(CURL_VERSION)
-	for p in $^; do patch -p1 < $$p; done
-	# Re-generate the configure script
-	autoreconf -fi
-	touch .patched
-
-# This is a small hack that flags that curl was patched and configured in the "chrome" version
-$(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.patched
-	cd $(CURL_VERSION)
-
-	# Set up the configure flags to curl.
-	# If the user provided the --host flag to our configure script
-	# (for cross compilation), then pass it on to curl.
-	
-	# XXX: psl should be enabled in the future:
-	# https://daniel.haxx.se/blog/2024/01/10/psl-in-curl/
-	{ \
-	  config_flags="--prefix=@prefix@"; \
-	  config_flags="$$config_flags --with-brotli=$(brotli_install_dir)"; \
-	  config_flags="$$config_flags --with-nghttp2=$(nghttp2_install_dir)"; \
-	  config_flags="$$config_flags --with-ngtcp2=$(ngtcp2_install_dir)"; \
-	  config_flags="$$config_flags --with-nghttp3=$(nghttp3_install_dir)"; \
-	  case "$(host_os)" in \
-	    darwin*|ios*|tvos*|watchos*) \
-	      config_flags="$$config_flags --with-apple-idn --without-libidn2"; \
-	      ;; \
-	    *android*) \
-	      config_flags="$$config_flags --without-libidn2"; \
-	      ;; \
-	    *) \
-	      config_flags="$$config_flags --with-libidn2=$(libidn2_install_dir)"; \
-	      ;; \
-	  esac; \
-	  config_flags="$$config_flags --with-openssl=$(boringssl_dir)"; \
-	  config_flags="$$config_flags --with-zlib=$(with_zlib)"; \
-	  config_flags="$$config_flags --with-zstd=$(with_zstd)"; \
-	  config_flags="$$config_flags --without-librtmp"; \
-	  config_flags="$$config_flags --disable-ldap --disable-ldaps"; \
-	  config_flags="$$config_flags --without-libpsl"; \
-	  config_flags="$$config_flags --enable-websockets"; \
-	  config_flags="$$config_flags --enable-ech"; \
-	  config_flags="$$config_flags --enable-ssls-export"; \
-	  config_flags="$$config_flags --enable-ipv6"; \
-	  config_flags="$$config_flags USE_CURL_SSLKEYLOGFILE=true"; \
-	  if test "$(static_build)" = "yes"; then \
-	    config_flags="$$config_flags --enable-static --disable-shared"; \
-	  fi; \
-	  if test -n "$(host_alias)"; then \
-	    config_flags="$$config_flags --host=$(host_alias)"; \
-	  fi; \
-	  if test -n "$(with_ca_bundle)"; then \
-		config_flags="$$config_flags --with-ca-bundle=$(with_ca_bundle)"; \
-	  fi; \
-	  if test -n "$(with_ca_path)"; then \
-		config_flags="$$config_flags --with-ca-path=$(with_ca_path)"; \
-	  fi; \
-	  case "$(host_os)" in \
-	    darwin*|ios*|tvos*|watchos*) add_libs="-pthread -lc++" ;; \
-	    *android*) add_libs="-pthread -lc++" ;; \
-	    *) add_libs="-pthread -lstdc++ -L$(libunistring_install_dir)/lib -lunistring" ;; \
-	  esac; \
-	}
-
-	echo PKG_CONFIG_PATH="$(ngtcp2_install_dir)/lib/pkgconfig"
-	echo "Configuring curl with: $$config_flags"
-
-	{ \
-		lib_path="$(nghttp2_install_dir)/lib:$(ngtcp2_install_dir)/lib:$(nghttp3_install_dir)/lib:$(zstd_install_dir)/lib:$(zlib_install_dir)/lib:$(brotli_install_dir)/lib"; \
-		case "$(host_os)" in \
-		  darwin*|ios*|tvos*|watchos*) ;; \
-		  *android*) ;; \
-		  *) lib_path="$$lib_path:$(libidn2_install_dir)/lib:$(libunistring_install_dir)/lib" ;; \
-		esac; \
-	}
-
-	DYLD_LIBRARY_PATH="$$lib_path" \
-	LD_LIBRARY_PATH="$$lib_path" \
-	PKG_CONFIG_LIBDIR= \
-	PKG_CONFIG_PATH="$(ngtcp2_install_dir)/lib/pkgconfig" \
-		./configure \
-			 $$config_flags \
-			 LIBS="$$add_libs"
-
-	# Remove possible leftovers from a previous compilation
-	$(MAKE) clean MAKEFLAGS=
-	touch .chrome
+$(CURL_DIST)/.patched:
+	cd $(CURL_DIST) && autoreconf -fi && touch .patched
+
+$(CURL_DIST)/.chrome: $(CURL_DIST)/.patched
+	cd ./$(CURL_DIST) && CC=@CC@ LDFLAGS="$(LDFLAGS) -Wl,-rpath,$(ngtcp2_install_dir)/lib" \
+		DYLD_LIBRARY_PATH="$(nghttp2_install_dir)/lib:$(ngtcp2_install_dir)/lib:$(nghttp3_install_dir)/lib:$(zstd_install_dir)/lib:$(zlib_install_dir)/lib:$(brotli_install_dir)/lib" \
+		PKG_CONFIG_LIBDIR= PKG_CONFIG_PATH="$(ngtcp2_install_dir)/lib/pkgconfig" \
+		./configure --prefix=@prefix@ $(CURL_CONFIG_FLAGS) --with-ca-bundle=$(with_ca_bundle) \
+		--with-brotli=$(brotli_install_dir) --with-libidn2=$(libidn2_install_dir) --with-openssl=$(boringssl_dir) \
+		--with-nghttp2=$(nghttp2_install_dir) --with-nghttp3=$(nghttp3_install_dir) --with-ngtcp2=$(ngtcp2_install_dir) \
+		--with-zlib=$(with_zlib) --with-zstd==$(with_zstd) --without-apple-idn --without-libgsasl --without-libpsl --without-librtmp \
+		--enable-websockets --enable-ech --enable-ssls-export --enable-ipv6 --enable-static --disable-shared --disable-ldap --disable-ldaps \
+		LIBS="-pthread -l@CXXLIB@" && touch .chrome
