--- a/cmake/libutils.cmake	2025-12-14 18:46:49.000000000 +0800
+++ b/cmake/libutils.cmake	2025-12-15 00:54:25.000000000 +0800
@@ -521,7 +521,7 @@
     STRING(REGEX REPLACE "[ ]+" ";" STATIC_LIBS_STRING "${STATIC_LIBS_STRING}" )
     ADD_CUSTOM_COMMAND(TARGET ${TARGET} POST_BUILD
       COMMAND rm $<TARGET_FILE:${TARGET}>
-      COMMAND /usr/bin/libtool -static -no_warning_for_no_symbols
+      COMMAND /usr/bin/libtool -static
       -o $<TARGET_FILE:${TARGET}>
       ${STATIC_LIBS_STRING}
       )

--- a/cmake/os/Darwin.cmake	2025-09-24 21:35:03.000000000 +0800
+++ b/cmake/os/Darwin.cmake	2025-12-14 19:04:06.000000000 +0800
@@ -84,9 +90,9 @@
 
 IF(WITH_LIBTOOL)
   SET(CMAKE_C_CREATE_STATIC_LIBRARY
-    "/usr/bin/libtool -static -no_warning_for_no_symbols -o <TARGET> <LINK_FLAGS> <OBJECTS> ")
+    "/usr/bin/libtool -static -o <TARGET> <LINK_FLAGS> <OBJECTS> ")
   SET(CMAKE_CXX_CREATE_STATIC_LIBRARY
-    "/usr/bin/libtool -static -no_warning_for_no_symbols -o <TARGET> <LINK_FLAGS> <OBJECTS> ")
+    "/usr/bin/libtool -static -o <TARGET> <LINK_FLAGS> <OBJECTS> ")
 ELSE()
   # This did not fix the "library.a(filename.cc.o) has no symbols" warnings.
   # 'ranlib' has the -no_warning_for_no_symbols option, but 'ar' does not.
