--- a/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/CMakeLists.txt	2025-12-14 20:07:33.000000000 +0800
@@ -695,7 +695,7 @@
 # even if we have configured for 64bit build....
 SET(CMAKE_SIZEOF_VOID_P ${SIZEOF_VOIDP})
 IF(NOT SIZEOF_VOIDP EQUAL 8)
-  MESSAGE(FATAL_ERROR "MySQL supports only 64-bit platforms.")
+  MESSAGE(WARNING "MySQL upstream does not test their code.")
 ENDIF()
 
 INCLUDE(compile_flags)
@@ -2043,25 +2043,9 @@
 MYSQL_CHECK_FIDO_DLLS()
 
 IF(APPLE)
-  GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
-  IF(EXISTS ${HOMEBREW_BASE}/include/boost)
-    FOREACH(SYSTEM_LIB ICU LZ4 PROTOBUF ZSTD FIDO)
-      IF(WITH_${SYSTEM_LIB} STREQUAL "system")
-        MESSAGE(FATAL_ERROR
-          "WITH_${SYSTEM_LIB}=system is not compatible with Homebrew boost\n"
-          "MySQL depends on ${BOOST_PACKAGE_NAME} with a set of patches.\n"
-          "Including headers from ${HOMEBREW_BASE}/include "
-          "will break the build.\n"
-          "Please use WITH_${SYSTEM_LIB}=bundled\n"
-          "or do 'brew uninstall boost' or 'brew unlink boost'"
-          )
-      ENDIF()
-    ENDFOREACH()
-  ENDIF()
-  # Ensure that we look in /usr/local/include or /opt/homebrew/include
   FOREACH(SYSTEM_LIB ICU LZ4 PROTOBUF ZSTD FIDO)
     IF(WITH_${SYSTEM_LIB} STREQUAL "system")
-      INCLUDE_DIRECTORIES(SYSTEM ${HOMEBREW_BASE}/include)
+      INCLUDE_DIRECTORIES(SYSTEM @PREFIX@/include)
       BREAK()
     ENDIF()
   ENDFOREACH()
