--- a/components/test_server_telemetry_traces/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/components/test_server_telemetry_traces/CMakeLists.txt	2025-12-14 22:14:15.000000000 +0800
@@ -29,4 +29,5 @@
   server_telemetry_helpers.cc
   MODULE_ONLY
   TEST_ONLY
+  LINK_LIBRARIES atomic
   )

--- a/components/test_server_telemetry_metrics/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/components/test_server_telemetry_metrics/CMakeLists.txt	2025-12-14 22:13:26.000000000 +0800
@@ -28,4 +28,5 @@
   server_metrics_helpers.cc
   MODULE_ONLY
   TEST_ONLY
+  LINK_LIBRARIES atomic
   )

--- a/components/libminchassis/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/components/libminchassis/CMakeLists.txt	2025-12-15 02:38:45.000000000 +0800
@@ -40,4 +40,4 @@
 
 ADD_LIBRARY(minchassis STATIC ${LIBMINCHASSIS_SOURCES})
 ADD_DEPENDENCIES(minchassis GenError)
-TARGET_LINK_LIBRARIES(minchassis ${CMAKE_DL_LIBS})
+TARGET_LINK_LIBRARIES(minchassis ${CMAKE_DL_LIBS} atomic)

--- a/plugin/connection_control/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/connection_control/CMakeLists.txt	2025-12-15 05:30:48.000000000 +0800
@@ -27,7 +27,7 @@
   security_context_wrapper.cc
   connection_delay.cc
   MODULE_ONLY
-  LINK_LIBRARIES extra::rapidjson
+  LINK_LIBRARIES extra::rapidjson atomic
   )
 
 ADD_DEFINITIONS(-DMYSQL_SERVER)

--- a/plugin/rewriter/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/rewriter/CMakeLists.txt	2025-12-15 06:29:02.000000000 +0800
@@ -30,6 +30,7 @@
   MODULE_ONLY
   MODULE_OUTPUT_NAME "rewriter"
   WIN_DEF_FILE "${CMAKE_CURRENT_SOURCE_DIR}/rewriter_udf.def"
+  LINK_LIBRARIES atomic
   )
 
 # MYSQL_ADD_PLUGIN may have decided not to build it.

--- a/plugin/replication_observers_example/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/replication_observers_example/CMakeLists.txt	2025-12-15 06:24:06.000000000 +0800
@@ -33,7 +33,7 @@
   MODULE_ONLY
   MODULE_OUTPUT_NAME "replication_observers_example_plugin"
   TEST_ONLY
-  LINK_LIBRARIES extra::rapidjson mysql_gtid
+  LINK_LIBRARIES extra::rapidjson mysql_gtid atomic
   )
 
 ADD_DEFINITIONS(-DMYSQL_SERVER)

--- a/plugin/group_replication/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/group_replication/CMakeLists.txt	2025-12-15 06:34:36.000000000 +0800
@@ -225,6 +225,7 @@
   mysqlgcs
   extra::rapidjson
   mysql_gtid
+  atomic
   MODULE_ONLY
   MODULE_OUTPUT_NAME "group_replication"
   )

--- a/plugin/version_token/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/version_token/CMakeLists.txt	2025-12-15 06:41:11.000000000 +0800
@@ -24,7 +24,7 @@
 MYSQL_ADD_PLUGIN(version_token
   version_token.cc
   MODULE_ONLY
-  LINK_LIBRARIES extra::rapidjson
+  LINK_LIBRARIES extra::rapidjson atomic
   )
 
 ADD_DEFINITIONS(-DMYSQL_SERVER)

--- a/plugin/clone/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/clone/CMakeLists.txt	2025-12-15 05:35:39.000000000 +0800
@@ -38,4 +38,5 @@
   src/clone_hton.cc
   MODULE_ONLY
   MODULE_OUTPUT_NAME "mysql_clone"
+  LINK_LIBRARIES atomic
   )
