From 7e4c6d4590414405c3c22b90b309a69044ac8dab Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 5 May 2025 09:35:39 +0800
Subject: [PATCH 1/2] mainboard: fix for missing kIOMainPortDefault

---
 src/apple/mainboard.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/apple/mainboard.cpp src/apple/mainboard.cpp
index 1ec5766..b534dba 100644
--- src/apple/mainboard.cpp
+++ src/apple/mainboard.cpp
@@ -9,6 +9,10 @@
 
 #include "hwinfo/mainboard.h"
 
+#ifndef kIOMainPortDefault
+#define kIOMainPortDefault kIOMasterPortDefault
+#endif
+
 namespace hwinfo {
 
 std::string get_mainboard_property(CFStringRef property_name) {
-- 
2.48.0

