From 518052c5a6667c6cc09cfcaf8af0ffc088b38827 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 5 May 2025 09:57:57 +0800
Subject: [PATCH 2/2] battery: include CoreFoundation

---
 src/apple/battery.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/apple/battery.cpp src/apple/battery.cpp
index ca13811..1596259 100644
--- src/apple/battery.cpp
+++ src/apple/battery.cpp
@@ -5,6 +5,7 @@
 
 #ifdef HWINFO_APPLE
 
+#include <CoreFoundation/CoreFoundation.h>
 #include <IOKit/ps/IOPSKeys.h>
 #include <IOKit/ps/IOPowerSources.h>
 
-- 
2.48.0

