From 012b4fc4121ff42778d53a1e8396ae177f6fb86b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 3 Jul 2025 12:18:15 +0800
Subject: [PATCH 3/3] Fix arch bitness on macOS

---
 src/apple/os.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git src/apple/os.cpp src/apple/os.cpp
index 9e058a9..3d7fea2 100644
--- src/apple/os.cpp
+++ src/apple/os.cpp
@@ -52,8 +52,11 @@ OS::OS() {
   _bigEndian = (byteorder == 4321);
   _littleEndian = (byteorder == 1234);
 
-  // TODO: Actually check
+#if defined(__x86_64__) || defined(__aarch64__) || defined(__ppc64__)
   _64bit = true;
+#else
+  _64bit = false;
+#endif
   _32bit = !_64bit;
 }
 
-- 
2.48.0

