--- Makefile.orig	2026-07-06 03:48:22.000000000 +0800
+++ Makefile	2026-07-31 23:56:56.000000000 +0800
@@ -125,7 +125,7 @@
 
 $(LIBGIFSO): $(OBJECTS) $(HEADERS)
 ifeq ($(UNAME), Darwin)
-	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(OBJECTS) -o $(LIBGIFSO)
+	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) -Wl,-install_name,$(PREFIX)/lib/$(LIBGIFSO) $(OBJECTS) -o $(LIBGIFSO)
 else
 	$(CC) $(CFLAGS) $(CPPFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBGIFSOMAJOR) -o $(LIBGIFSO) $(OBJECTS)
 endif
@@ -135,7 +135,7 @@
 
 $(LIBUTILSO): $(UOBJECTS) $(UHEADERS)
 ifeq ($(UNAME), Darwin)
-	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) $(UOBJECTS) -o $(LIBUTILSO)
+	$(CC) $(CFLAGS) -dynamiclib -current_version $(LIBVER) -Wl,-install_name,$(PREFIX)/lib/$(LIBUTILSO) $(UOBJECTS) -o $(LIBUTILSO)
 else
 	$(CC) $(CFLAGS) $(CPPLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,$(LIBUTILSOMAJOR) -o $(LIBUTILSO) $(UOBJECTS)
 endif
