--- a/src/transfer/devicebroadcaster.cpp
+++ b/src/transfer/devicebroadcaster.cpp	2026-03-05 21:05:19.000000000 +0800
@@ -43,13 +43,12 @@
     int port = Settings::instance()->getBroadcastPort();
     Device dev = Settings::instance()->getMyDevice();
 
-    QVariantMap map;
-    map["id"] = dev.getId();
-    map["name"] = dev.getName();
-    map["os"] = dev.getOSName();
-    map["port"] = port;
+    QJsonObject obj;
+    obj.insert("id", QJsonValue(dev.getId()));
+    obj.insert("name", QJsonValue(dev.getName()));
+    obj.insert("os", QJsonValue(dev.getOSName()));
+    obj.insert("port", QJsonValue(static_cast<double>(port)));
 
-    QJsonObject obj = QJsonObject::fromVariantMap(map);
     QVector<QHostAddress> addresses = getBroadcastAddressFromInterfaces();
     QByteArray data = QJsonDocument(obj).toJson(QJsonDocument::Compact);
     foreach (QHostAddress address, addresses) {
