diff --git a/src/model/devicelistmodel.h b/src/model/devicelistmodel.h
--- a/src/model/devicelistmodel.h
+++ b/src/model/devicelistmodel.h
@@ -26,6 +26,7 @@

 class DeviceListModel : public QAbstractListModel
 {
+    Q_OBJECT
 public:
     DeviceListModel(DeviceBroadcaster* deviceBC, QObject* parent = 0);

diff --git a/src/ui/mainwindow.cpp b/src/ui/mainwindow.cpp
--- a/src/ui/mainwindow.cpp
+++ b/src/ui/mainwindow.cpp
@@ -146,16 +146,16 @@ void MainWindow::onAboutActionTriggered()

 void MainWindow::connectSignals()
 {
-    connect(mTransServer, SIGNAL(newReceiverAdded(Receiver*)), this, SLOT(onNewReceiverAdded(Receiver*())));
+    connect(mTransServer, SIGNAL(newReceiverAdded(Receiver*)), this, SLOT(onNewReceiverAdded(Receiver*)));

     QItemSelectionModel* senderSel = ui->senderTableView->selectionModel();
     connect(senderSel, SIGNAL(selectionChanged(QItemSelection,QItemSelection)),
-            this, SLOT(onSenderTableSelectionChanged(QItemSelection,QItemSelection())));
+            this, SLOT(onSenderTableSelectionChanged(QItemSelection,QItemSelection)));

     QItemSelectionModel* receiverSel = ui->receiverTableView->selectionModel();
     connect(receiverSel, SIGNAL(selectionChanged(QItemSelection,QItemSelection)),
-            this, SLOT(onReceiverTableSelectionChanged(QItemSelection,QItemSelection())));
+            this, SLOT(onReceiverTableSelectionChanged(QItemSelection,QItemSelection)));
 }

 void MainWindow::sendFiles()
