--- src/mesh/meshGFacePack.cpp
+++ src/mesh/meshGFacePack.cpp	2025-03-26 02:02:46.000000000 +0800
@@ -8,6 +8,7 @@
 #include <queue>
 #include <map>
 #include <vector>
+#include <cstdint>
 #include "gmsh.h"
 #include "SPoint2.h"
 #include "SVector3.h"

--- contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h
+++ contrib/QuadMeshingTools/qmtMeshGeometryOptimization.h	2025-03-26 03:19:52.000000000 +0800
@@ -8,6 +8,7 @@
 #pragma once
 
 #include <float.h>
+#include <cstdint>
 #include "qmtMeshUtils.h"
 
 class SurfaceProjector;
