From d4b1ebb2d3178396e72bef2fba057ca32a3ec3ac Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 4 Feb 2025 11:40:57 +0800
Subject: [PATCH] include/SDL3: fix wrong headers

---
 include/SDL3/SDL_platform_defines.h | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git include/SDL3/SDL_platform_defines.h include/SDL3/SDL_platform_defines.h
index 7e9a0a92a..3fe083756 100644
--- include/SDL3/SDL_platform_defines.h
+++ include/SDL3/SDL_platform_defines.h
@@ -221,8 +221,8 @@
  */
 #define SDL_PLATFORM_MACOS 1
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1070
-    #error SDL for macOS only supports deploying on 10.7 and above.
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+    #error SDL for macOS only supports deploying on 10.5 and above.
 #endif /* MAC_OS_X_VERSION_MIN_REQUIRED < 1070 */
 #endif /* TARGET_OS_IPHONE */
 #endif /* defined(__APPLE__) */
