--- src/misc/macos/SDL_sysurl.m.orig	2026-01-02 01:48:19.000000000 +0800
+++ src/misc/macos/SDL_sysurl.m	2026-06-01 18:10:26.000000000 +0800
@@ -24,19 +24,21 @@
 
 #include "../SDL_sysurl.h"
 
-#import <Cocoa/Cocoa.h>
+#include <Cocoa/Cocoa.h>
 
 bool SDL_SYS_OpenURL(const char *url)
 {
-    @autoreleasepool {
+        NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
         CFURLRef cfurl = CFURLCreateWithBytes(NULL, (const UInt8 *)url, SDL_strlen(url), kCFStringEncodingUTF8, NULL);
         OSStatus status = LSOpenCFURLRef(cfurl, NULL);
         CFRelease(cfurl);
         if (status != noErr) {
-            return SDL_SetError("LSOpenCFURLRef() failed: %d", status);
+            [pool release];
+            return SDL_SetError("LSOpenCFURLRef() failed: %d", status);
         }
+
+        [pool release];
         return true;
-    }
 }
 
 #endif // SDL_PLATFORM_MACOS
