--- src/stdlib/SDL_memset.c
+++ src/stdlib/SDL_memset.c
@@ -82,7 +82,7 @@
 // Note that memset() is a byte assignment and this is a 32-bit assignment, so they're not directly equivalent.
 void *SDL_memset4(void *dst, Uint32 val, size_t dwords)
 {
-#if defined(__APPLE__) && defined(HAVE_STRING_H)
+#if defined(__APPLE__) && false
     memset_pattern4(dst, &val, dwords * 4);
 #elif defined(__GNUC__) && defined(__i386__)
     int u0, u1, u2;
