--- src/stdlib/SDL_memset.c
+++ src/stdlib/SDL_memset.c
@@ -19,7 +19,7 @@
   3. This notice may not be removed or altered from any source distribution.
 */
 #include "SDL_internal.h"
+#include <AvailabilityMacros.h>
-

 #ifdef SDL_memset
 #undef SDL_memset
@@ -82,7 +82,7 @@
 // Note that memset() is a byte assignment and this is a 32-bit assignment, so they're not directly equivalent.
 void *SDL_memset4(void *dst, Uint32 val, size_t dwords)
 {
-#if defined(__APPLE__) && defined(HAVE_STRING_H)
+#if defined(__APPLE__) && defined(HAVE_STRING_H) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1050)
     memset_pattern4(dst, &val, dwords * 4);
 #elif defined(__GNUC__) && defined(__i386__)
     int u0, u1, u2;
