--- gotwebd/gotweb.c	2025-02-16 21:45:16.000000000 +0800
+++ gotwebd/gotweb.c	2025-02-17 03:46:25.000000000 +0800
@@ -37,6 +37,10 @@
 #include <unistd.h>
 #include <vis.h>
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 #include "got_error.h"
 #include "got_object.h"
 #include "got_reference.h"
@@ -181,6 +185,7 @@
 
 	/* Log the request. */
 	if (gotwebd_env->gotwebd_verbose > 0) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 130000
 		char *server_name = NULL;
 		char *querystring = NULL;
 		char *document_uri = NULL;
@@ -209,6 +214,7 @@
 		free(server_name);
 		free(querystring);
 		free(document_uri);
+#endif
 	}
 
 	/*
