--- configure	2026-04-30 02:03:47.000000000 +0800
+++ configure	2026-07-03 01:46:17.000000000 +0800
@@ -17400,8 +17400,8 @@
            if test "$with_coregraphics" != no
 then :
 
-             ac_fn_c_check_header_compile "$LINENO" "CoreGraphics/CoreGraphics.h" "ac_cv_header_CoreGraphics_CoreGraphics_h" "$ac_includes_default"
-if test "x$ac_cv_header_CoreGraphics_CoreGraphics_h" = xyes
+             ac_fn_c_check_header_compile "$LINENO" "ApplicationServices/ApplicationServices.h" "ac_cv_header_ApplicationServices_ApplicationServices_h" "$ac_includes_default"
+if test "x$ac_cv_header_ApplicationServices_ApplicationServices_h" = xyes
 then :
   have_coregraphics=yes
 else case e in #(
@@ -17409,26 +17409,11 @@
 esac
 fi
 
-             if test "$have_coregraphics" = yes
-then :
-
-               ac_fn_c_check_header_compile "$LINENO" "ImageIO/ImageIO.h" "ac_cv_header_ImageIO_ImageIO_h" "$ac_includes_default"
-if test "x$ac_cv_header_ImageIO_ImageIO_h" = xyes
-then :
-
-else case e in #(
-  e) have_coregraphics=no ;;
-esac
-fi
-
-
-fi
-
 fi
            if test "$have_coregraphics" = yes
 then :
 
-             COREGRAPHICS_LIBS="-framework CoreGraphics -framework ImageIO -framework CoreFoundation"
+             COREGRAPHICS_LIBS="-framework ApplicationServices -framework CoreFoundation"
 
 printf '%s\n' "#define HAVE_COREGRAPHICS 1" >>confdefs.h
 

--- tools/chafa/chicle-coregraphics-loader.c	2025-11-08 07:23:30.000000000 +0800
+++ tools/chafa/chicle-coregraphics-loader.c	2026-07-03 01:57:51.000000000 +0800
@@ -22,8 +22,8 @@
 #include "config.h"
 
 #include <CoreFoundation/CoreFoundation.h>
-#include <CoreGraphics/CoreGraphics.h>
-#include <ImageIO/ImageIO.h>
+#include <ApplicationServices/ApplicationServices.h>
+#include <AvailabilityMacros.h>
 
 #include <glib.h>
 #include <chafa.h>
@@ -433,7 +433,11 @@
     if (dict_value && CFGetTypeID (dict_value) == CFDictionaryGetTypeID ())
     {
         delay = get_delay_from_dictionary ((CFDictionaryRef) dict_value,
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
                                            kCGImagePropertyGIFUnclampedDelayTime,
+#else
+                                           kCGImagePropertyGIFDelayTime,
+#endif
                                            kCGImagePropertyGIFDelayTime);
         if (delay > 0.0)
             return delay;
