--- a/kdeui/itemviews/kselectionproxymodel.cpp	2020-01-14 18:30:24.000000000 +0800
+++ b/kdeui/itemviews/kselectionproxymodel.cpp	2026-03-02 14:04:37.000000000 +0800
@@ -2144,7 +2144,7 @@
     if (!proxyIndex.isValid() || !sourceModel() || d->m_rootIndexList.isEmpty())
         return QModelIndex();
 
-    Q_ASSERT(proxyIndex.internalPointer() >= 0);
+    // Q_ASSERT(proxyIndex.internalPointer() >= 0);
     Q_ASSERT(proxyIndex.model() == this);
 
     if (proxyIndex.internalPointer() == 0)

--- a/kdecore/localization/klocale_kde.cpp	2020-01-14 18:30:24.000000000 +0800
+++ b/kdecore/localization/klocale_kde.cpp	2026-03-02 13:50:42.000000000 +0800
@@ -2438,8 +2438,8 @@
 
 bool KLocalePrivate::use12Clock() const
 {
-    if ((timeFormat().contains(QString::fromLatin1("%I")) > 0) ||
-        (timeFormat().contains(QString::fromLatin1("%l")) > 0)) {
+    if ((timeFormat().contains(QString::fromLatin1("%I")) != 0) ||
+        (timeFormat().contains(QString::fromLatin1("%l")) != 0)) {
         return true;
     } else {
         return false;

--- a/kimgio/jp2.cpp	2020-01-14 18:30:24.000000000 +0800
+++ b/kimgio/jp2.cpp	2026-03-02 14:39:04.000000000 +0800
@@ -103,13 +103,13 @@
         stream->bufmode_ |= bufmode & JAS_STREAM_BUFMODEMASK;
 }
 
-static int qiodevice_read(jas_stream_obj_t *obj, char *buf, int cnt)
+int qiodevice_read(jas_stream_obj_t *obj, char *buf, unsigned int cnt)
 {
         QIODevice *io = (QIODevice*) obj;
         return io->read(buf, cnt);
 }
 
-static int qiodevice_write(jas_stream_obj_t *obj, char *buf, int cnt)
+int qiodevice_write(jas_stream_obj_t *obj, const char *buf, unsigned int cnt)
 {
         QIODevice *io = (QIODevice*) obj;
         return io->write(buf, cnt);

--- a/kio/kfile/kpropertiesdialog.cpp	2020-01-14 18:30:24.000000000 +0800
+++ b/kio/kfile/kpropertiesdialog.cpp	2026-03-12 20:30:37.000000000 +0800
@@ -3306,7 +3306,7 @@
 
     if (preferredTerminal == "konsole")
     {
-        terminalCloseBool = (d->m_terminalOptionStr.contains( "--noclose" ) > 0);
+        terminalCloseBool = (d->m_terminalOptionStr.contains( "--noclose" ) != 0);
         w.terminalCloseCheck->setChecked(terminalCloseBool);
         d->m_terminalOptionStr.remove( "--noclose");
     }

--- a/kio/kfile/kopenwithdialog.cpp	2020-01-14 18:30:24.000000000 +0800
+++ b/kio/kfile/kopenwithdialog.cpp	2026-03-12 20:31:01.000000000 +0800
@@ -713,7 +713,7 @@
         // ### indicate that default value was restored
         d->terminal->setChecked(d->curService->terminal());
         QString terminalOptions = d->curService->terminalOptions();
-        d->nocloseonexit->setChecked((terminalOptions.contains(QLatin1String("--noclose")) > 0));
+        d->nocloseonexit->setChecked((terminalOptions.contains(QLatin1String("--noclose")) != 0));
         d->m_terminaldirty = false; // slotTerminalToggled changed it
     }
 }
