From c0662c5d266a65b86c76c83d902674403c654c35 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 20 Jul 2025 22:47:19 +0800
Subject: [PATCH 01/31] CoreLibraries.gmk: do not hardcode arch

---
 jdk/make/lib/CoreLibraries.gmk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jdk/make/lib/CoreLibraries.gmk b/jdk/make/lib/CoreLibraries.gmk
index 5bb9394208..dc8cbc2218 100644
--- a/jdk/make/lib/CoreLibraries.gmk
+++ b/jdk/make/lib/CoreLibraries.gmk
@@ -82,7 +82,7 @@ else
       OUTPUT_DIR := $(SUPPORT_OUTPUTDIR)/native/$(MODULE)/libfdlibm, \
       SRC := $(LIBFDLIBM_SRC), \
       CFLAGS := $(CFLAGS_JDKLIB) $(LIBFDLIBM_CFLAGS), \
-      LDFLAGS := -nostdlib -r -arch x86_64, \
+      LDFLAGS := -nostdlib -r \
       OBJECT_DIR := $(SUPPORT_OUTPUTDIR)/native/$(MODULE)/libfdlibm, \
   ))
 
-- 
2.24.3 (Apple Git-128)

