From 03810a645f6b1d0a6105e72a9c972f27b1b20f72 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 20 Jul 2025 23:33:25 +0800
Subject: [PATCH 06/31] Awt2dLibraries: add -fnested-functions

---
 jdk/make/lib/Awt2dLibraries.gmk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/jdk/make/lib/Awt2dLibraries.gmk b/jdk/make/lib/Awt2dLibraries.gmk
index 3eafa4a47e..c6e3cf590c 100644
--- a/jdk/make/lib/Awt2dLibraries.gmk
+++ b/jdk/make/lib/Awt2dLibraries.gmk
@@ -994,6 +994,7 @@ ifeq ($(OPENJDK_TARGET_OS), macosx)
           $(X_CFLAGS) \
           $(X_LIBS) \
           $(LIBAWT_LWAWT_CFLAGS), \
+          -fnested-functions \
       DISABLED_WARNINGS_clang := incomplete-implementation enum-conversion \
           deprecated-declarations objc-method-access bitwise-op-parentheses \
           incompatible-pointer-types parentheses-equality extra-tokens, \
-- 
2.24.3 (Apple Git-128)

