From 446824474319531df15395272669eb799ba5743c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 21 Jul 2025 02:43:02 +0800
Subject: [PATCH 27/31] HeadlessToolkit.c: minor fix

---
 .../unix/native/libawt_headless/awt/HeadlessToolkit.c          | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/jdk/src/java.desktop/unix/native/libawt_headless/awt/HeadlessToolkit.c b/jdk/src/java.desktop/unix/native/libawt_headless/awt/HeadlessToolkit.c
index 8aa34f4967..fb4f0ccdde 100644
--- a/jdk/src/java.desktop/unix/native/libawt_headless/awt/HeadlessToolkit.c
+++ b/jdk/src/java.desktop/unix/native/libawt_headless/awt/HeadlessToolkit.c
@@ -33,7 +33,10 @@ extern JavaVM *jvm;
 JNIEXPORT jint JNICALL
 DEF_JNI_OnLoad(JavaVM *vm, void *reserved)
 {
+// barracuda: why do we need this?
+#ifndef __APPLE__
     jvm = vm;
+#endif
     return JNI_VERSION_1_2;
 }
 
-- 
2.24.3 (Apple Git-128)

