From 30f1c2ccb5d58b97f6eb08ed784be002d59b2efd Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 26 Mar 2025 19:26:46 +0800
Subject: [PATCH] Fallback for missing pthread_threadid_np in macOS

---
 base/hplatform.h | 1 +
 base/hthread.h   | 5 +++++
 2 files changed, 6 insertions(+)

diff --git base/hplatform.h base/hplatform.h
index 3487c25..7a9d555 100644
--- base/hplatform.h
+++ base/hplatform.h
@@ -17,6 +17,7 @@
 #elif defined(__APPLE__) && (defined(__GNUC__) || defined(__xlC__) || defined(__xlc__))
     #include <TargetConditionals.h>
     #if defined(TARGET_OS_MAC) && TARGET_OS_MAC
+        #include <AvailabilityMacros.h>
         #define OS_MAC
     #elif defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
         #define OS_IOS
diff --git base/hthread.h base/hthread.h
index b43e0c6..0094df6 100644
--- base/hthread.h
+++ base/hthread.h
@@ -19,7 +19,12 @@
 #elif defined(OS_DARWIN)
 static inline long hv_gettid() {
     uint64_t tid = 0;
+/* pthread_threadid_np is not available before 10.6 and in 10.6 for ppc */
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060 || defined(__POWERPC__)
+    tid = pthread_mach_thread_np(pthread_self());
+#else
     pthread_threadid_np(NULL, &tid);
+#endif
     return tid;
 }
 #elif HAVE_PTHREAD_H
