From 969cdec8a463f3427bfaec23f6a508023df118c2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 26 Mar 2025 19:30:18 +0800
Subject: [PATCH] hplatform.h: add powerpc

---
 base/hplatform.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git base/hplatform.h base/hplatform.h
index 7a9d555..592d4e7 100644
--- base/hplatform.h
+++ base/hplatform.h
@@ -60,6 +60,12 @@
     #define ARCH_MIPS64
 #elif defined(__mips__)
     #define ARCH_MIPS
+#elif defined(__riscv)
+    #define ARCH_RISCV
+#elif defined(__ppc64__) || defined(__powerpc64__)
+    #define ARCH_PPC64
+#elif defined(__ppc__) || defined(__powerpc__)
+    #define ARCH_PPC
 #else
     #warning "Untested hardware architecture!"
 #endif
@@ -260,7 +264,8 @@
       defined(__MIPSEL)  || defined(__MIPS64EL)
     #define BYTE_ORDER  LITTLE_ENDIAN
 #elif defined(__ARMEB__) || defined(__AARCH64EB__) || \
-      defined(__MIPSEB)  || defined(__MIPS64EB)
+      defined(__MIPSEB)  || defined(__MIPS64EB) || \
+      defined(__ppc__)  || defined(__ppc64__)
     #define BYTE_ORDER  BIG_ENDIAN
 #elif defined(OS_WIN)
     #define BYTE_ORDER  LITTLE_ENDIAN
