--- MenuExtras/MenuMeterMem/MenuMeterMemStats.m.orig	2014-10-23 11:39:42.000000000 +0800
+++ MenuExtras/MenuMeterMem/MenuMeterMemStats.m	2025-02-09 08:48:14.000000000 +0800
@@ -29,6 +29,7 @@
 //
 ///////////////////////////////////////////////////////////////
 
+#ifndef __ppc__
 struct vm_statistics64 {
 	natural_t	free_count;
 	natural_t	active_count;
@@ -56,6 +57,8 @@
 	uint64_t	total_uncompressed_pages_in_compressor;
 } __attribute__((aligned(8)));
 typedef struct vm_statistics64	vm_statistics64_data_t;
+#endif
+
 typedef integer_t	*host_info64_t;
 typedef kern_return_t (*host_statistics64_Ptr)(host_t host_priv,
 											   host_flavor_t flavor,
@@ -218,6 +221,7 @@
 				nil];
 } // memStats32
 
+#ifndef __ppc__
 - (NSDictionary *)memStats64 {
 	// Get the data using the 64-bit API.
 	vm_statistics64_data_t vmStats64;
@@ -284,6 +288,7 @@
 				[NSNumber numberWithUnsignedLongLong:deltaPageOut], @"deltapageouts",
 				nil];
 } // memStats64
+#endif
 
 - (NSDictionary *)memStats {
 	if (isMavericksOrLater) {
