--- cmake/SetCompilerFlags.cmake.orig	2025-06-24 03:36:13.000000000 +0800
+++ cmake/SetCompilerFlags.cmake	2025-11-29 07:31:02.000000000 +0800
@@ -97,15 +97,13 @@
             endif(HAVE_STACK_PROTECTOR)
         endif(HAVE_STACK_PROTECTOR_ALL)
         # Support address space layout randomization (ASLR)
-        if(NOT (MINGW OR CYGWIN OR CMAKE_C_COMPILER_ID STREQUAL "AppleClang"
-            OR ((CMAKE_SYSTEM_NAME MATCHES Darwin) AND (CMAKE_C_COMPILER_ID MATCHES Clang))))
+        if(NOT (MINGW OR CYGWIN OR APPLE))
             target_compile_options(${target} PRIVATE -fPIE)
             target_link_options(${target} PRIVATE -fPIE -pie)
             target_link_options(${target} PRIVATE -Wl,-z,relro)
             target_link_options(${target} PRIVATE -Wl,-z,now)
             target_link_options(${target} PRIVATE -Wl,-z,noexecstack)
-        endif(NOT (MINGW OR CYGWIN OR CMAKE_C_COMPILER_ID STREQUAL "AppleClang"
-            OR ((CMAKE_SYSTEM_NAME MATCHES Darwin) AND (CMAKE_C_COMPILER_ID MATCHES Clang))))
+        endif(NOT (MINGW OR CYGWIN OR APPLE))
         target_link_options(${target} PRIVATE -fstack-check)
         add_compile_flag_to_targets(${target})
     endif(MSVC)
