--- qtads.pro.orig	2016-01-21 20:09:42.000000000 +0800
+++ qtads.pro	2026-08-15 05:57:46.000000000 +0800
@@ -32,10 +32,10 @@
 
 macx {
     QMAKE_INFO_PLIST = Info.plist
-    QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.7
+    # QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.6
     CONFIG += link_pkgconfig
     PKGCONFIG += SDL2_mixer
-    LIBS += -lSDL_sound
+    LIBS += -lSDL2_sound
     QMAKE_CFLAGS += -std=gnu11 -fvisibility=hidden
     QMAKE_CXXFLAGS += -fvisibility=hidden
     QMAKE_LFLAGS += -dead_strip
@@ -45,7 +45,7 @@
     # Normally we would use pkg-config for SDL_mixer too, but it has to appear
     # in the linker flags before SDL_sound, which lacks pkg-config support, or
     # else we crash.
-    LIBS += -lSDL_mixer -lSDL_sound
+    LIBS += -lSDL2_mixer -lSDL2_sound
 }
 win32 {
     *-g++* {
