--- create.mk.xae	2021-06-12 16:26:59.000000000 +0800
+++ create.mk.xae	2025-09-23 21:36:17.000000000 +0800
@@ -107,53 +107,13 @@
 	five_lib=""
 fi
 
-# check location of X11 headers
+# location of X11 headers
 
-X_headers=""
+X_headers="-I@PREFIX@/include"
 
-if [ -d /usr/include/X11R5 ]
-then
-	X_headers="-I/usr/include/X11R5"
-fi
-
-if [ -d /usr/include/X11R6 ]
-then
-	X_headers="-I/usr/include/X11R6"
-fi
-
-if [ -d /usr/X11R6/include ]
-then
-	X_headers="-I/usr/X11R6/include"
-fi
-
-if [ -d /usr/openwin/include ]
-then
-	X_headers="${X_headers} -I/usr/openwin/include"
-fi
-
-# check location of X11 libraries
+# location of X11 libraries
 
-X_libs=""
-
-if [ -d /usr/lib/X11R5 ]
-then
-	X_libs="-L/usr/lib/X11R5"
-fi
-
-if [ -d /usr/lib/X11R6 ]
-then
-	X_libs="-L/usr/lib/X11R6"
-fi
-
-if [ -d /usr/X11R6/lib ]
-then
-	X_libs="${X_libs} -L/usr/X11R6/lib"
-fi
-
-if [ -d /usr/openwin/lib ]
-then
-	X_libs="${X_libs} -L/usr/openwin/lib"
-fi
+X_libs="-L@PREFIX@/lib"
 
 if [ -n "$CFLAGS" ]
 then
