--- localize.c	2021-06-12 16:26:59.000000000 +0800
+++ localize.c	2025-09-23 21:38:26.000000000 +0800
@@ -64,10 +64,9 @@
 	catalog = catopen("aee", 0);
 #endif /* NO_CATGETS */
 
-	ae_help_file = catgetlocal( 1, "/usr/local/lib/help.ae");
+	ae_help_file = catgetlocal( 1, "@PREFIX@/share/doc/aee/help.ae");
 	main_buffer_name = catgetlocal( 2, "main");
 
-
 	/*
 	 |	allocate space for strings here for settings menu
 	 */
