From 6b2af6718a635d10fd9c0a6dd2ab084b349e8d76 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 16:17:27 +0800
Subject: [PATCH 1/7] CMakeLists: set CMAKE_OSX_ARCHITECTURES only if not set
 already

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index e8f7243..efd724e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -5,7 +5,9 @@ cmake_minimum_required(VERSION 3.22)
 
 # Project setup
 
-set(CMAKE_OSX_ARCHITECTURES arm64;x86_64)
+if(APPLE AND NOT CMAKE_OSX_ARCHITECTURES)
+    set(CMAKE_OSX_ARCHITECTURES arm64;x86_64)
+endif()
 
 project(phosg)
 include(GNUInstallDirs)
-- 
2.48.0

