From 6af9c6ac9286134f33cd22cd2ea13b666e1406f9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 16:18:18 +0800
Subject: [PATCH 2/7] Hash.hh: use C++ header

---
 src/Hash.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/Hash.hh src/Hash.hh
index 0a9be52..3eaa4a0 100644
--- src/Hash.hh
+++ src/Hash.hh
@@ -2,7 +2,7 @@
 
 #include <string>
 
-#include <stdint.h>
+#include <cstdint>
 
 namespace phosg {
 
-- 
2.48.0

