From 66247b62653f0f1a57192638e9af1916d5e0697c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 16:19:57 +0800
Subject: [PATCH 3/7] Encoding.hh: use C++ headers, define __STDC_FORMAT_MACROS

---
 src/Encoding.hh | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git src/Encoding.hh src/Encoding.hh
index a21a31e..8b12cf6 100644
--- src/Encoding.hh
+++ src/Encoding.hh
@@ -1,7 +1,11 @@
 #pragma once
 
-#include <inttypes.h>
-#include <stdint.h>
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
+#include <cinttypes>
+#include <cstdint>
 
 #include <string>
 
-- 
2.48.0

