From bef877225f79e3599d41759e461ce1218642ed72 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 16:23:29 +0800
Subject: [PATCH 4/7] Platform.hh: minor fix to endianness macros

---
 src/Platform.hh | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/Platform.hh src/Platform.hh
index 51047d6..d80fca0 100644
--- src/Platform.hh
+++ src/Platform.hh
@@ -26,9 +26,9 @@ namespace phosg {
 // Try to determine endianess from GCC defines first. If they aren't available,
 // use some constants to try to figure it out
 // clang-format off
-#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
   #define PHOSG_LITTLE_ENDIAN
-#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#elif defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
   #define PHOSG_BIG_ENDIAN
 
 #else
-- 
2.48.0

