From d118441f5bfd8ba97dcdcff4530354ab7c2c778b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 16:30:01 +0800
Subject: [PATCH 5/7] Image.hh: add a missing header

---
 src/Image.hh | 1 +
 1 file changed, 1 insertion(+)

diff --git src/Image.hh src/Image.hh
index 1dea5a3..6fe82bb 100644
--- src/Image.hh
+++ src/Image.hh
@@ -2,6 +2,7 @@
 
 #include <stdint.h>
 #include <stdio.h>
+#include <sys/types.h>
 
 #include <functional>
 #include <stdexcept>
-- 
2.48.0

