From 4429ace0d85369fc45436082d9758da023163e0e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 20:14:18 +0800
Subject: [PATCH 6/7] CMakeLists: ppc also needs libatomic

---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index efd724e..7cd4482 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -57,7 +57,8 @@ target_include_directories(
 # but this library does not exist on others (e.g. Ubuntu) nor on macOS
 message(STATUS "Target architecture is ${CMAKE_HOST_SYSTEM_PROCESSOR}")
 string(FIND ${CMAKE_HOST_SYSTEM_PROCESSOR} "armv" IS_LINUX_ARMV)
-if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64" OR ${IS_LINUX_ARMV} GREATER_EQUAL 0)
+if(${CMAKE_HOST_SYSTEM_PROCESSOR} STREQUAL "aarch64" OR ${IS_LINUX_ARMV} GREATER_EQUAL 0
+  OR ${CMAKE_OSX_ARCHITECTURES} STREQUAL "ppc")
   target_link_libraries(phosg atomic)
 endif()
 
-- 
2.48.0

