--- cpp/cmake_modules/ThirdpartyToolchain.cmake	2025-10-20 22:48:39.000000000 +0800
+++ cpp/cmake_modules/ThirdpartyToolchain.cmake	2025-11-02 07:00:36.000000000 +0800
@@ -4621,7 +4621,7 @@
     # We can remove this once bundled Apache ORC is 2.2.1 or later.
     list(APPEND ORC_PATCHES ${CMAKE_CURRENT_LIST_DIR}/orc-2345.patch)
   endif()
-  if(Protobuf_VERSION VERSION_GREATER_EQUAL 32.0)
+  if(APPLE OR (Protobuf_VERSION VERSION_GREATER_EQUAL 32.0))
     # We can remove this once bundled Apache ORC is 2.2.1 or later.
     list(APPEND ORC_PATCHES ${CMAKE_CURRENT_LIST_DIR}/orc-2357.patch)
   endif()
