--- dialogs.h	2015-05-15 17:20:49.000000000 +0800
+++ dialogs.h	2025-05-03 06:19:46.000000000 +0800
@@ -21,10 +21,13 @@
 #define DIALOGS_H
 
 #include <mainwindow.h>
+#include <QLabel>
+#include <QLayout>
 #include <QMessageBox>
 #include <QInputDialog>
 #include <QApplication>
 #include <QMutex>
+#include <QPushButton>
 
 /* These input dialogs work from a different to main thread */
 class MyInputDialog:public QObject {
@@ -163,9 +166,11 @@
         mutex.tryLock();
         mutex.unlock();
     }
+
     void show() {
         QCoreApplication::postEvent(this, new QEvent(QEvent::User));
     }
+
     virtual bool event(QEvent * ev) {
         res = false;
         if (ev->type() == QEvent::User) {
@@ -174,9 +179,7 @@
 
             msgBox->setText(t1);
             msgBox->setInformativeText(t2);
-            msgBox->setStandardButtons(QMessageBox::
-                                       Cancel | QMessageBox::Help |
-                                       QMessageBox::Ok);
+            msgBox->setStandardButtons(QMessageBox::Cancel | QMessageBox::Help | QMessageBox::Ok);
             msgBox->setDefaultButton(QMessageBox::Cancel);
             msgBox->setButtonText(QMessageBox::Ok, oktxt);
             msgBox->setButtonText(QMessageBox::Help, tr("View certificate"));
@@ -184,16 +187,21 @@
             do {
                 ret = msgBox->exec();
                 if (ret == QMessageBox::Help) {
-                    QMessageBox helpBox;
-                    helpBox.setTextInteractionFlags(Qt::
-                                                    TextSelectableByMouse |
-                                                    Qt::TextSelectableByKeyboard
-                                                    |
-                                                    Qt::LinksAccessibleByMouse);
-                    helpBox.setText(details);
-                    helpBox.setTextFormat(Qt::PlainText);
-                    helpBox.setStandardButtons(QMessageBox::Ok);
-                    helpBox.exec();
+                    QDialog helpDialog(w);
+                    helpDialog.setWindowTitle(tr("Certificate Details"));
+
+                    QVBoxLayout *layout = new QVBoxLayout(&helpDialog);
+                    QLabel *label = new QLabel(details, &helpDialog);
+                    label->setTextInteractionFlags(Qt::TextSelectableByMouse | Qt::LinksAccessibleByMouse);
+                    label->setWordWrap(true); // Ensure proper wrapping of text
+                    layout->addWidget(label);
+
+                    QPushButton *okButton = new QPushButton(tr("OK"), &helpDialog);
+                    layout->addWidget(okButton);
+                    QObject::connect(okButton, SIGNAL(clicked()), &helpDialog, SLOT(accept()));
+
+                    helpDialog.setLayout(layout);
+                    helpDialog.exec();
                 }
             } while (ret == QMessageBox::Help);
 

--- mainwindow.cpp	2015-05-15 17:20:49.000000000 +0800
+++ mainwindow.cpp	2025-05-03 05:50:26.000000000 +0800
@@ -23,7 +23,7 @@
 #include <stdarg.h>
 #include <stdio.h>
 }
-#include <QtConcurrent/QtConcurrentRun>
+#include <QtConcurrentRun>
 #include <QDateTime>
 #include <QMessageBox>
 #include <vpninfo.h>

--- editdialog.cpp	2015-05-15 17:20:49.000000000 +0800
+++ editdialog.cpp	2025-05-03 06:24:07.000000000 +0800
@@ -44,9 +44,9 @@
 }
 
 int token_rtab[] = {
-    [0] = OC_TOKEN_MODE_HOTP,
-    [1] = OC_TOKEN_MODE_TOTP,
-    [2] = OC_TOKEN_MODE_STOKEN
+    OC_TOKEN_MODE_HOTP,
+    OC_TOKEN_MODE_TOTP,
+    OC_TOKEN_MODE_STOKEN
 };
 
 void EditDialog::load_win_certs()
