From cd3b73c134a442b8b46bd346edac9ab6813b079c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 24 Mar 2025 03:22:03 +0800
Subject: [PATCH] Fix ssize_t for Darwin again

---
 lobster/src/lobster/tools.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git lobster/src/lobster/tools.h lobster/src/lobster/tools.h
index dbfe1e1..a43c54b 100644
--- lobster/src/lobster/tools.h
+++ lobster/src/lobster/tools.h
@@ -29,8 +29,8 @@
 // on Darwin, where it breaks the build. Undefine ssize_t if defined.
 #ifdef ssize_t
     #undef ssize_t
+    typedef ptrdiff_t ssize_t;
 #endif
-typedef ptrdiff_t ssize_t;
 
 // Custom _L suffix, since neither L (different size on win/nix) or LL
 // (does not convert to int64_t on nix!) is portable.
