From 9b0925820975fe7bb050a2aa8ae7a039ccd5c37b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 24 Mar 2025 03:40:23 +0800
Subject: [PATCH] Fix linking on macOS

---
 CMakeLists.txt | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git CMakeLists.txt CMakeLists.txt
index c3a9eff..e240892 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -162,6 +162,21 @@ if(WIN32)
         lobster-impl
         StackWalker
     )
+elseif(APPLE)
+    target_link_libraries(
+        TreeSheets
+        PRIVATE
+        $<IF:$<BOOL:${TREESHEETS_WITH_STATIC_WXWIDGETS}>,
+            wx::aui
+            wx::adv
+            wx::core
+            wx::xml
+            wx::net,
+            ${wxWidgets_LIBRARIES}
+        >
+        lobster-impl
+        PRIVATE "-framework CoreFoundation"
+    )
 else()
     target_link_libraries(
         TreeSheets
