From 0d6a07d572c97c82f59abef22a71129d70010934 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 24 Mar 2025 04:19:57 +0800
Subject: [PATCH] Do not produce a bundle

---
 CMakeLists.txt | 29 ++---------------------------
 1 file changed, 2 insertions(+), 27 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index c3a9eff..0ff683e 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -115,19 +115,7 @@ if(WIN32)
     target_include_directories(StackWalker PUBLIC src/StackWalker)
 endif()
 
-if(APPLE)
-    add_executable(
-        TreeSheets
-        MACOSX_BUNDLE
-        src/main.cpp
-        osx/App.icns
-    )
-    set_source_files_properties(osx/App.icns PROPERTIES
-        MACOSX_PACKAGE_LOCATION "Resources")
-    set_target_properties(TreeSheets PROPERTIES
-        MACOSX_BUNDLE_INFO_PLIST "${CMAKE_SOURCE_DIR}/osx/Info.plist"
-    )
-elseif(WIN32)
+if(WIN32)
     add_executable(
         TreeSheets
         WIN32
@@ -184,7 +172,7 @@ if(LINUX)
     OPTION(TREESHEETS_RELOCATABLE_INSTALLATION "Install data relative to the TreeSheets binary, instead of respecting the Filesystem Hierarchy Standard" OFF)
 endif()
 
-if(LINUX AND NOT TREESHEETS_RELOCATABLE_INSTALLATION)
+if(NOT WIN32 AND NOT TREESHEETS_RELOCATABLE_INSTALLATION)
     include(GNUInstallDirs)
 
     set(TREESHEETS_BINDIR ${CMAKE_INSTALL_BINDIR})
@@ -201,11 +189,6 @@ if(LINUX AND NOT TREESHEETS_RELOCATABLE_INSTALLATION)
     install(FILES linux/com.strlen.TreeSheets.svg DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/scalable/apps)
     install(FILES linux/com.strlen.TreeSheets.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
     install(FILES linux/com.strlen.TreeSheets.xml DESTINATION ${CMAKE_INSTALL_DATADIR}/mime/packages)
-elseif(APPLE)
-    # Paths must be relative to use with CPack
-    set(TREESHEETS_BINDIR .)
-    set(TREESHEETS_DOCDIR TreeSheets.app/Contents/Resources)
-    set(TREESHEETS_PKGDATADIR TreeSheets.app/Contents/Resources)
 else()
     set(TREESHEETS_BINDIR .)
     set(TREESHEETS_DOCDIR .)
@@ -240,14 +223,6 @@ if(WIN32 OR TREESHEETS_RELOCATABLE_INSTALLATION)
             DESTINATION "translations/${locale}"
         )
     endforeach()
-elseif(APPLE)
-    foreach(locale ${locales})
-        install(
-            FILES "TS/translations/${locale}/ts.mo"
-            # Paths must be relative to use with CPack
-            DESTINATION "TreeSheets.app/Contents/Resources/translations/${locale}.lproj"
-        )
-    endforeach()
 else()
     # Falling back to GNU scheme
     foreach(locale ${locales})
