From 990d459112c672656125f3f0914091ce810d8bdc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 7 Jun 2025 05:46:13 +0800
Subject: [PATCH] PixelFormats.hpp: fix HEVC playback bug

From https://github.com/zaps166/QMPlay2/issues/776#issuecomment-2950779114
---
 src/qmplay2/headers/PixelFormats.hpp | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git src/qmplay2/headers/PixelFormats.hpp src/qmplay2/headers/PixelFormats.hpp
index ce761f8c..11499f1d 100644
--- src/qmplay2/headers/PixelFormats.hpp
+++ src/qmplay2/headers/PixelFormats.hpp
@@ -20,14 +20,19 @@
 
 #include <QVector>
 
+extern "C"
+{
+	#include <libavutil/pixfmt.h>
+}
+
 enum class QMPlay2PixelFormat //Compatible with FFmpeg
 {
-	YUV420P =  0,
-	YUV422P =  4,
-	YUV444P =  5,
-	YUV410P =  6,
-	YUV411P =  7,
-	YUV440P = 33,
+	YUV420P = AV_PIX_FMT_YUV420P,
+	YUV422P = AV_PIX_FMT_YUV422P,
+	YUV444P = AV_PIX_FMT_YUV444P,
+	YUV410P = AV_PIX_FMT_YUV410P,
+	YUV411P = AV_PIX_FMT_YUV411P,
+	YUV440P = AV_PIX_FMT_YUV440P,
 
 	Count   =  6
 };
