--- a/src/modules/OpenGL2/OpenGL2Common.cpp
+++ b/src/modules/OpenGL2/OpenGL2Common.cpp
@@ -257,15 +257,15 @@
     if (shaderProgramVideo->bind())
     {
         texCoordYCbCrLoc = shaderProgramVideo->attributeLocation("aTexCoord");
         positionYCbCrLoc = shaderProgramVideo->attributeLocation("aPosition");
-        shaderProgramVideo->setUniformValue((numPlanes == 1) ? "uRGB" : "uY" , 0);
+        shaderProgramVideo->setUniformValue((numPlanes == 1) ? "uRGB" : "uY" , static_cast<GLint>(0));
         if (numPlanes == 2)
-            shaderProgramVideo->setUniformValue("uCbCr", 1);
+            shaderProgramVideo->setUniformValue("uCbCr", static_cast<GLint>(1));
         else if (numPlanes == 3)
         {
-            shaderProgramVideo->setUniformValue("uCb", 1);
-            shaderProgramVideo->setUniformValue("uCr", 2);
+            shaderProgramVideo->setUniformValue("uCb", static_cast<GLint>(1));
+            shaderProgramVideo->setUniformValue("uCr", static_cast<GLint>(2));
         }
         shaderProgramVideo->release();
     }
     else
@@ -281,9 +281,9 @@
     if (shaderProgramOSD->bind())
     {
         texCoordOSDLoc = shaderProgramOSD->attributeLocation("aTexCoord");
         positionOSDLoc = shaderProgramOSD->attributeLocation("aPosition");
-        shaderProgramOSD->setUniformValue("uTex", 3);
+        shaderProgramOSD->setUniformValue("uTex", static_cast<GLint>(3));
         shaderProgramOSD->release();
     }
     else
     {
@@ -570,9 +570,9 @@
             const qreal dpr = widget()->devicePixelRatioF();
             if (!resetDone)
                 maybeSetMipmaps(dpr);
             const bool useBicubic = (W * dpr > m_textureSize.width() || H * dpr > m_textureSize.height());
-            shaderProgramVideo->setUniformValue("uBicubic", useBicubic ? 1 : 0);
+            shaderProgramVideo->setUniformValue("uBicubic", useBicubic ? static_cast<GLint>(1) : static_cast<GLint>(0));
         }
         shaderProgramVideo->setUniformValue("uTextureSize", m_textureSize);
 
         doReset = !resetDone;
--- a/src/gui/MacOS/ScreenSaver.cpp
+++ b/src/gui/MacOS/ScreenSaver.cpp
@@ -31,57 +31,32 @@
 public:
     inline ~ScreenSaverPriv()
     {
-        unInhibit();
     }
 
     inline void inhibit()
     {
-#if defined(__APPLE__) && (MAC_OS_X_VERSION_MAX_ALLOWED < 1060)
-        m_okDisp = (IOPMAssertionCreate(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, &m_idDisp) == kIOReturnSuccess);
-        m_okSys  = (IOPMAssertionCreate(kIOPMAssertionTypeNoIdleSleep, kIOPMAssertionLevelOn, &m_idSys)  == kIOReturnSuccess);
-#else
-        m_okDisp = (IOPMAssertionCreateWithName(kIOPMAssertionTypeNoDisplaySleep, kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idDisp) == kIOReturnSuccess);
-        m_okSys  = (IOPMAssertionCreateWithName(kIOPMAssertionTypeNoIdleSleep, kIOPMAssertionLevelOn, QMPLAY2_MEDIA_PLAYBACK, &m_idSys) == kIOReturnSuccess);
-#endif
     }
     inline void unInhibit()
     {
-        if (m_okDisp)
-        {
-            IOPMAssertionRelease(m_idDisp);
-            m_okDisp = false;
-        }
-        if (m_okSys)
-        {
-            IOPMAssertionRelease(m_idSys);
-            m_okSys = false;
-        }
     }
 
 private:
-    IOPMAssertionID m_idDisp;
-    IOPMAssertionID m_idSys;
     bool m_okDisp = false;
     bool m_okSys  = false;
 };
 
 /**/
 
 ScreenSaver::ScreenSaver() :
     m_priv(new ScreenSaverPriv)
 {}
 ScreenSaver::~ScreenSaver()
 {
-    delete m_priv;
 }
 
 void ScreenSaver::inhibit(int context)
 {
-    if (inhibitHelper(context))
-        m_priv->inhibit();
 }
 void ScreenSaver::unInhibit(int context)
 {
-    if (unInhibitHelper(context))
-        m_priv->unInhibit();
 }
