--- src/whisper.cpp
+++ src/whisper.cpp     2025-09-26 15:21:41
@@ -35,6 +35,8 @@
 #include <string>
 #include <thread>
 #include <vector>
+#include <dirent.h>
+#include <sys/types.h>

 #if defined(WHISPER_BIG_ENDIAN)
 template<typename T>

--- examples/server/server.cpp	2026-06-16 18:28:23.000000000 +0800
+++ examples/server/server.cpp	2026-09-03 10:21:59.000000000 +0800
@@ -19,6 +19,7 @@
 #include <atomic>
 #include <functional>
 #include <cstdlib>
+#include <filesystem>
 #if defined (_WIN32)
 #include <windows.h>
 #endif
