From 557852a8953b0f133c53da4076b31c52426eb96b Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 6 Feb 2024 18:39:41 +0800
Subject: [PATCH] ld64: do not err out on pre-release triple

---
 ld64/src/ld/Options.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git ld64/src/ld/Options.cpp ld64/src/ld/Options.cpp
index 2fe070a..fe8a4e3 100644
--- ld64/src/ld/Options.cpp
+++ ld64/src/ld/Options.cpp
@@ -1653,7 +1653,7 @@ uint32_t Options::parseVersionNumber32(const char* versionString)
 			z = strtoul(&end[1], &end, 10);
 		}
 	}
-	if ( (*end != '\0') || (x > 0xffff) || (y > 0xff) || (z > 0xff) )
+	if ( (x > 0xffff) || (y > 0xff) || (z > 0xff) )
 		throwf("malformed 32-bit x.y.z version number: %s", versionString);
 
 	return (x << 16) | ( y << 8 ) | z;

From 327ed1dec4ba764ed47d5d4c66a841885e17e895 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 6 Feb 2024 20:37:40 +0800
Subject: [PATCH 2/2] ld64: fix for 10a190

---
 ld64/src/ld/Options.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git ld64/src/ld/Options.cpp ld64/src/ld/Options.cpp
index fe8a4e3..633a5dd 100644
--- ld64/src/ld/Options.cpp
+++ ld64/src/ld/Options.cpp
@@ -4607,10 +4607,10 @@ void Options::reconfigureDefaults()
 
 	
 	// only use compressed LINKEDIT for:
-	//			Mac OS X 10.6 or later
+	//			Mac OS X 10.7 or later
 	//			iOS 3.1 or later
 	if ( fMakeCompressedDyldInfo ) {
-		if ( !minOS(ld::mac10_6, ld::iOS_3_1) )
+		if ( !minOS(ld::mac10_7, ld::iOS_3_1) )
 			fMakeCompressedDyldInfo = false;
 		switch (fArchitecture) {
 			case CPU_TYPE_POWERPC:
