--- setup.py.orig	2025-09-19 02:38:33.000000000 +0800
+++ setup.py	2025-09-22 01:53:52.000000000 +0800
@@ -40,10 +40,6 @@
         base_compile_args.append('-mno-avx2')
     elif have_avx2:
         base_compile_args.append('-mavx2')
-# On macOS, force libc++ in case the system tries to use `stdlibc++`.
-# The latter is often absent from modern macOS systems.
-if sys.platform == 'darwin':
-    base_compile_args.append('-stdlib=libc++')
 
 
 def info(*msg):
