--- Source/PSMTabBarControl.h	2013-10-27 11:59:46.000000000 +0800
+++ Source/PSMTabBarControl.h	2025-12-01 17:33:14.000000000 +0800
@@ -42,17 +42,17 @@
 @class PSMTabBarController;
 @protocol PSMTabStyle;
 
-typedef enum PSMTabBarOrientation : NSUInteger {
+typedef enum {
 	PSMTabBarHorizontalOrientation,
 	PSMTabBarVerticalOrientation
 } PSMTabBarOrientation;
 
-typedef enum PSMTabBarTearOffStyle : NSUInteger {
+typedef enum {
 	PSMTabBarTearOffAlphaWindow,
 	PSMTabBarTearOffMiniwindow
 } PSMTabBarTearOffStyle;
 
-typedef enum PSMTabStateMask : NSUInteger {
+typedef enum {
 	PSMTab_SelectedMask				= 1 << 1,
 	PSMTab_LeftIsSelectedMask		= 1 << 2,
 	PSMTab_RightIsSelectedMask		= 1 << 3,
