From 188666e6db135c799d27a9ddb388e674c8bd43b5 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 27 Jan 2023 16:26:55 -0800
Subject: [PATCH 05/18] darwin: Compile with -D_DARWIN_C_SOURCE

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index 9fd5416..c8274a0 100644
--- a/meson.build
+++ b/meson.build
@@ -19,6 +19,10 @@ cc_args = []
 if host_machine.system() not in ['freebsd', 'openbsd']
 	cc_args += ['-D_POSIX_C_SOURCE=200809L']
 endif
+if host_machine.system() == 'darwin'
+	# For CMSG_LEN(), which is an RFC 2292 addition and not part of POSIX.1-2008
+	cc_args += ['-D_DARWIN_C_SOURCE']
+endif
 add_project_arguments(cc_args, language: 'c')
 
 compiler_flags = [
-- 
2.24.3 (Apple Git-128)

