From d08e32621c7a13d0f16b1def40df9bc5d6683dd0 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 27 Jan 2023 17:17:06 -0800
Subject: [PATCH 09/18] darwin: Use epoll-shim

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index eb5614a..1659f5e 100644
--- a/meson.build
+++ b/meson.build
@@ -74,7 +74,7 @@ endif
 config_h.set10('HAVE_BROKEN_MSG_CMSG_CLOEXEC', have_broken_msg_cmsg_cloexec)
 
 if get_option('libraries')
-	if host_machine.system() in ['freebsd', 'openbsd']
+	if host_machine.system() in ['darwin', 'freebsd', 'openbsd']
 		# When building for FreeBSD, epoll(7) is provided by a userspace
 		# wrapper around kqueue(2).
 		epoll_dep = dependency('epoll-shim')
-- 
2.24.3 (Apple Git-128)

