From 925a666874f3328b427bc6f646009f6201557c3e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 6 Nov 2025 02:08:27 +0800
Subject: [PATCH 16/18] wayland-client.c: use epoll-shim

---
 src/wayland-client.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/wayland-client.c b/src/wayland-client.c
index 2464e7b..e07ce7b 100644
--- a/src/wayland-client.c
+++ b/src/wayland-client.c
@@ -41,6 +41,10 @@
 #include <poll.h>
 #include <pthread.h>
 
+#ifdef __APPLE__
+#include <epoll-shim/detail/poll.h>
+#endif
+
 #include "wayland-util.h"
 #include "wayland-os.h"
 #include "wayland-client.h"
-- 
2.24.3 (Apple Git-128)

