From 416565a3989438d50cb1277cdac407f3a6cf60e4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 11 Feb 2026 06:31:16 +0800
Subject: [PATCH] Fix build with modern ocaml

diff --git dune dune
index 4b0678a..a304575 100644
--- dune
+++ dune
@@ -1,4 +1,4 @@
 (executable
  (name fzf)
- (libraries core async)
+ (libraries core async core_unix.command_unix)
  (preprocess (pps ppx_jane)))
diff --git fzf.ml fzf.ml
index 7336f3f..e590518 100644
--- fzf.ml
+++ fzf.ml
@@ -146,7 +146,7 @@ let run user_input tty_text stdin =
 ;;
 
 let () =
-  Command.run @@
+  Command_unix.run @@
   let open Command.Let_syntax in
   Command.async ~summary:"fzf" [%map_open
     let () = return () in
diff --git render.mli render.mli
index 683eb54..ae0b751 100644
--- render.mli
+++ render.mli
@@ -14,7 +14,7 @@ open! Async
     occur on the event of a new line being read.
 *)
 val every
-  :  how_often_to_render:Time.Span.t
+  :  how_often_to_render:Time_float.Span.t
   -> render:(unit -> unit Deferred.t)
   -> (unit -> [ `Finished of 'a | `Repeat of unit ] Deferred.t)
   -> 'a Deferred.t
diff --git spinner.mli spinner.mli
index 601834e..fbee342 100644
--- spinner.mli
+++ spinner.mli
@@ -2,7 +2,7 @@ open Core
 
 type t
 
-val create  : spin_every : Time.Span.t -> t
+val create  : spin_every : Time_float.Span.t -> t
 val finish  : t -> unit
 val to_char : t -> char option
 val advance : t -> unit
diff --git tty_text.ml tty_text.ml
index 3a5c101..243c871 100644
--- tty_text.ml
+++ tty_text.ml
@@ -43,9 +43,8 @@ module Configure_terminal = struct
     let%bind () = setattr_out output ~attr_out:t.attr_out in
     return ()
 
-  let map_termio (attrs : Core.Unix.Terminal_io.t) =
+  let map_termio (attrs : Core_unix.Terminal_io.t) =
     { attrs with
-      Core.Unix.Terminal_io.
       c_ignbrk = false;
       c_brkint = false;
       c_parmrk = false;
