diff --git tty_text.ml tty_text.ml
index 3a5c101..243c871 100644
--- tty_text.ml
+++ tty_text.ml
@@ -144,9 +143,8 @@ let with_rendering f =
       Process.run_exn ()
         ~prog:"stty"
         (* NOTE: for people on Mac OS X, use ~args:[ "-f"; "/dev/tty"; "size" ] *)
-        ~args:[ "size"
-              ; "-F"
-              ; "/dev/tty"
+        ~args:[ "-f"
+              ; "/dev/tty"; "size"
               ]
     in
     match
