--- src/stubs-rng.c	2024-07-08 01:04:35.000000000 +0800
+++ src/stubs-rng.c	2025-10-19 00:25:27.000000000 +0800
@@ -24,6 +24,7 @@
 
 #include <unistd.h>
 #ifdef __APPLE__
+#include <sys/types.h>
 #include <sys/random.h>
 #endif
 
--- src/dune	2024-07-08 01:04:35.000000000 +0800
+++ src/dune	2025-10-19 00:26:24.000000000 +0800
@@ -6,7 +6,7 @@
   (language c)
   (flags -DCAML_NAME_SPACE -DEXPORT=static
          -DBLAKE3_NO_SSE2 -DBLAKE3_NO_SSE41 -DBLAKE3_NO_AVX2 -DBLAKE3_NO_AVX512
-         -DBLAKE3_USE_NEON=0
+         -DBLAKE3_USE_NEON=0 -isystem@PREFIX@/include/LegacySupport
          (:include flags.sexp))
   (names stubs-arcfour
          stubs-blowfish
@@ -47,7 +47,7 @@
     blake3.c
     blake3_dispatch.c
     blake3_portable.c))
-  (c_library_flags (:include library_flags.sexp))
+  (c_library_flags -lMacportsLegacySupport (:include library_flags.sexp))
   (flags :standard -safe-string -w -7 -w -27 -w -37))
 
 ; compute flags
