--- daemon/ratelimiting.c	2025-05-29 18:42:25.000000000 +0800
+++ daemon/ratelimiting.c	2025-05-31 20:15:46.000000000 +0800
@@ -25,8 +25,8 @@
 	uint32_t rate_limit;
 	uint32_t log_period;
 	uint16_t slip;
-	bool dry_run;
-	bool using_avx2;
+	uint8_t dry_run;
+	uint8_t using_avx2;
 	_Atomic uint32_t log_time;
 	kru_price_t v4_prices[V4_PREFIXES_CNT];
 	kru_price_t v6_prices[V6_PREFIXES_CNT];

--- lib/rules/api.h	2025-05-29 18:42:25.000000000 +0800
+++ lib/rules/api.h	2025-05-31 20:09:00.000000000 +0800
@@ -24,7 +24,7 @@
 	/// Degree of severity for the rule;  FIXME: granularity, defaults, etc.
 	uint8_t score : 4;
 
-	bool log_ip : 1, log_name : 1;
+	uint8_t log_ip : 1, log_name : 1;
 	// +maybe log rule/QNAME/something
 	/// Log level: 0 = debug, 1 = info, ...
 	uint8_t log_level : 2;
