--- daemon/ratelimiting.c	2025-04-24 17:32:24.000000000 +0800
+++ daemon/ratelimiting.c	2025-04-30 11:07:04.000000000 +0800
@@ -63,6 +63,7 @@
 	};
 
 	size_t header_size = offsetof(struct ratelimiting, using_avx2) + sizeof(header.using_avx2);
+#ifndef __ppc__
 	static_assert(  // no padding up to .using_avx2
 		offsetof(struct ratelimiting, using_avx2) ==
 			sizeof(header.capacity) +
@@ -72,7 +73,7 @@
 			sizeof(header.slip) +
 			sizeof(header.dry_run),
 		"detected padding with undefined data inside mmapped header");
-
+#endif
 	int ret = mmapped_init(&ratelimiting_mmapped, mmap_file, size, &header, header_size);
 	if (ret == MMAPPED_WAS_FIRST) {
 		kr_log_info(SYSTEM, "Initializing rate-limiting...\n");
