--- lib/mmapped.c	2025-12-02 21:33:31.000000000 +0800
+++ lib/mmapped.c	2025-12-03 20:12:47.000000000 +0800
@@ -44,12 +44,12 @@
 
 	if (!size) { // reset not allowed
 		kr_log_crit(SYSTEM, "File %s does not contain data in required format.\n", mmap_file);
-		return fail(mmapped, ENOTRECOVERABLE);
+		return fail(mmapped, EINVAL);
 	}
 
 	if (!mmapped->write_lock) {
 		kr_log_crit(SYSTEM, "Another instance of kresd uses file %s with different configuration.\n", mmap_file);
-		return fail(mmapped, ENOTRECOVERABLE);
+		return fail(mmapped, EINVAL);
 	}
 
 	if (mmapped->mem) {
