--- meson.build	2026-08-08 02:26:08.000000000 +0800
+++ meson.build	2026-09-12 20:57:07.000000000 +0800
@@ -611,7 +611,7 @@
 endif
 subdir('bin')
 
-if host_machine.system() != 'emscripten'
+if not platform_osx and host_machine.system() != 'emscripten'
   nm = find_program('nm', required: false)
 
   # Verify .def files for Windows linking (and for macOS too when -Wl,-exported_symbols_list is used).
