--- modules/gui/qt4/dialogs/messages.hpp	2014-08-14 15:20:04.000000000 +0800
+++ modules/gui/qt4/dialogs/messages.hpp	2025-10-09 18:03:46.000000000 +0800
@@ -28,7 +28,7 @@
 #include "util/singleton.hpp"
 #include "ui/messages_panel.h"
 #include <stdarg.h>
-#include <vlc_atomic.h>
+#include <QAtomicInt>
 #include <QMutex>
 
 class QTabWidget;
@@ -55,7 +55,7 @@
     void sinkMessage( const MsgEvent * );
     bool matchFilter( const QString& );
 
-    atomic_uint verbosity;
+    QAtomicInt verbosity;
     static void MsgCallback( void *, int, const vlc_log_t *, const char *,
                              va_list );
 
--- modules/gui/qt4/dialogs/messages.cpp	2016-07-29 06:35:50.000000000 +0800
+++ modules/gui/qt4/dialogs/messages.cpp	2025-10-09 18:03:14.000000000 +0800
@@ -37,7 +37,6 @@
 #include <QTabWidget>
 #include <QTreeWidget>
 #include <QTreeWidgetItem>
-#include <QMutex>
 #include <QLineEdit>
 #include <QScrollBar>
 #include <QMutex>
@@ -145,7 +144,7 @@
 
 void MessagesDialog::changeVerbosity( int i_verbosity )
 {
-    atomic_store( &this->verbosity, i_verbosity );
+    verbosity = i_verbosity;
 }
 
 void MessagesDialog::updateConfig()
@@ -339,7 +338,7 @@
 {
     MessagesDialog *dialog = (MessagesDialog *)self;
     char *str;
-    int verbosity = atomic_load( &dialog->verbosity );
+    int verbosity = dialog->verbosity;
 
     if( verbosity < 0 || verbosity < (type - VLC_MSG_ERR)
      || unlikely(vasprintf( &str, format, ap ) == -1) )
