--- configure	2025-11-17 11:32:30.000000000 +0800
+++ configure	2025-12-02 17:08:58.000000000 +0800
@@ -96,6 +96,8 @@
 		export CC="$val" ;;
 	CFLAGS)
 		export CFLAGS="$val" ;;
+	LDFLAGS)
+		export LDFLAGS="$val" ;;
 	*)
 		;;
 	esac
@@ -109,6 +111,7 @@
 AR=`printf "all:\\n\\t@echo \\\$(AR)\\n" | make ${MAKE_FLAGS} -sf -`
 CC=`printf "all:\\n\\t@echo \\\$(CC)\\n" | make ${MAKE_FLAGS} -sf -`
 CFLAGS=`printf "all:\\n\\t@echo \\\$(CFLAGS)\\n" | make ${MAKE_FLAGS} -sf -`
+LDFLAGS=`printf "all:\\n\\t@echo \\\$(LDFLAGS)\\n" | make ${MAKE_FLAGS} -sf -`
 
 # If there are no passed-in CFLAGS, so the CFLAGS are coming from `make,
 # add some additional useful defaults.
@@ -130,7 +133,6 @@
 LDADD_SCAN_SCALED=
 LDADD_STATIC=
 CPPFLAGS=
-LDFLAGS=
 LINKER_SOFLAG=
 LINKER_SONAME=
 LINKER_SOSUFFIX=
