diff --git a/fuse-ext2/fuse-ext2.h b/fuse-ext2/fuse-ext2.h
index 07fd1db..2bc3331 100644
--- fuse-ext2/fuse-ext2.h
+++ fuse-ext2/fuse-ext2.h
@@ -57,6 +57,14 @@ errcode_t ext2fs_file_close2(ext2_file_t file, void (*close_callback) (struct ex
 /* max timeout to flush bitmaps, to reduce inconsistencies */
 #define FLUSH_BITMAPS_TIMEOUT 10
 
+#ifndef UTIME_NOW
+#define UTIME_NOW -1
+#endif
+
+#ifndef UTIME_OMIT
+#define UTIME_OMIT -2
+#endif
+
 struct extfs_data {
 	unsigned char debug;
 	unsigned char silent;
