--- gedit/meson.build	2026-10-05 09:32:00.922490288 -0400
+++ gedit/meson.build	2026-10-05 09:34:59.376860643 -0400
@@ -128,7 +128,7 @@
   libgd_dep,
 ]
 
-if host_machine.system() == 'darwin'
+if host_machine.system() == 'mydarwin'
   libgedit_private_c_files += [
     'gedit-app-osx.m',
     'gedit-recent-osx.c',

--- gedit/gedit-dirs.c	2026-10-07 10:54:29.000000000 +0800
+++ gedit/gedit-dirs.c	2026-10-07 11:10:12.000000000 +0800
@@ -20,7 +20,7 @@
 #include "config.h"
 #include "gedit-dirs.h"
 
-#if OS_MACOS
+#ifdef GDK_WINDOWING_QUARTZ
 #include <gtkosxapplication.h>
 #endif
 
@@ -57,7 +57,7 @@
 	g_free (win32_dir);
 #endif /* G_OS_WIN32 */
 
-#if OS_MACOS
+#ifdef GDK_WINDOWING_QUARTZ
 	if (gtkosx_application_get_bundle_id () != NULL)
 	{
 		const gchar *bundle_resource_dir = gtkosx_application_get_resource_path ();

--- gedit/gedit.c	2026-10-07 10:54:29.000000000 +0800
+++ gedit/gedit.c	2026-10-07 11:14:54.000000000 +0800
@@ -95,7 +95,7 @@
 	GeditApp *app;
 	gint status;
 
-#if OS_MACOS
+#ifdef GDK_WINDOWING_QUARTZ
 	type = GEDIT_TYPE_APP_OSX;
 #elif defined G_OS_WIN32
 	type = GEDIT_TYPE_APP_WIN32;
