From 431bedb2471dbaffb626554ca9df82fbeb21c56d Mon Sep 17 00:00:00 2001
From: Dirk Farin <dirk.farin@gmail.com>
Date: Mon, 17 Feb 2025 19:49:30 +0100
Subject: [PATCH] fix pointer arithmetic (fixes #3)

---
 libvideogfx/utility/bitstream/bitreader.cc | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libvideogfx/utility/bitstream/bitreader.cc b/libvideogfx/utility/bitstream/bitreader.cc
index 49de4cf..442f9d9 100644
--- libvideogfx/utility/bitstream/bitreader.cc
+++ libvideogfx/utility/bitstream/bitreader.cc
@@ -21,6 +21,7 @@
 
 #include "libvideogfx/types.hh"
 #include "libvideogfx/utility/bitstream/bitreader.hh"
+#include <cassert>
 
 
 namespace videogfx {
@@ -63,7 +64,8 @@ namespace videogfx {
   {
 #if WORDS_BIGENDIAN
     assert(sizeof(uint32)==4);
-    uint32 val = *((uint32*)d_ptr)++;
+    uint32 val = *((uint32*)d_ptr);
+    d_ptr += 4;
 
     uint64 val64 = val;
     val64 <<= (64-d_bitsleft)-32;
