--- source/audioio/inc/AOCoreAudioMacOS.h	2026-03-14 20:57:59.000000000 +0800
+++ source/audioio/inc/AOCoreAudioMacOS.h	2026-06-11 07:10:24.000000000 +0800
@@ -26,6 +26,15 @@
 typedef UInt32 AudioFormatFlags;
 #endif
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+typedef Component AudioComponent;
+typedef ComponentDescription AudioComponentDescription;
+typedef ComponentInstance AudioComponentInstance;
+#define AudioComponentFindNext FindNextComponent
+#define AudioComponentInstanceNew OpenAComponent
+#define AudioComponentInstanceDispose CloseComponent
+#endif
+
 //-------------------------------------------------------------------------------------------
 namespace omega
 {

--- source/audioio/src/AOCoreAudioMacOS.mm	2026-03-14 20:57:59.000000000 +0800
+++ source/audioio/src/AOCoreAudioMacOS.mm	2026-06-11 07:21:01.000000000 +0800
@@ -2445,8 +2445,10 @@
             desc = "AC-3 (IEC 60958)"; break;
         case kAudioFormatAC3:
             desc = "AC-3"; break;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
         case kAudioFormatAES3:
             desc = "AES3-2003"; break;
+#endif
         case kAudioFormatALaw:
             desc = "A-law"; break;
         case kAudioFormatAMR:
@@ -2459,10 +2461,12 @@
             desc = "IMA 4:1 ADPCM"; break;
         case kAudioFormatAppleLossless:
             desc = "Apple Lossless"; break;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
         case kAudioFormatAudible:
             desc = "Audible"; break;
         case kAudioFormatDVIIntelIMA:
             desc = "DVI/Intel IMA ADPCM"; break;
+#endif
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
         case kAudioFormatEnhancedAC3:
             desc = "Enhanced AC-3"; break;
@@ -2511,8 +2515,10 @@
             desc = "MPEG-1/2, Layer II"; break;
         case kAudioFormatMPEGLayer3:
             desc = "MPEG-1/2, Layer III"; break;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
         case kAudioFormatMicrosoftGSM:
             desc = "Microsoft GSM 6.10"; break;
+#endif
 #if MAC_OS_X_VERSION_MIN_REQUIRED >= 101500
         case kAudioFormatOpus:
             desc = "Opus"; break;
@@ -2529,8 +2535,10 @@
             desc = "timestamp"; break;
         case kAudioFormatULaw:
             desc = "μ-Law"; break;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
         case kAudioFormatiLBC:
             desc = "Low Bitrate Codec (iLBC)"; break;
+#endif
         default:
             {
                 tuint8 c[5];

--- source/audioio/src/AOQueryCoreAudio.mm	2026-03-14 20:57:59.000000000 +0800
+++ source/audioio/src/AOQueryCoreAudio.mm	2026-06-11 07:18:24.000000000 +0800
@@ -14,6 +14,15 @@
 #define kAudioObjectPropertyElementMain kAudioObjectPropertyElementMaster
 #endif
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+typedef Component AudioComponent;
+typedef ComponentDescription AudioComponentDescription;
+typedef ComponentInstance AudioComponentInstance;
+#define AudioComponentFindNext FindNextComponent
+#define AudioComponentInstanceNew OpenAComponent
+#define AudioComponentInstanceDispose CloseComponent
+#endif
+
 //-------------------------------------------------------------------------------------------
 namespace omega
 {

--- source/track/db/inc/SandboxBookmarkService.h	2026-03-14 20:57:59.000000000 +0800
+++ source/track/db/inc/SandboxBookmarkService.h	2026-06-11 07:41:08.000000000 +0800
@@ -22,6 +22,10 @@
 #import <Foundation/Foundation.h>
 #import <Cocoa/Cocoa.h>
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+typedef NSUInteger NSURLBookmarkCreationOptions;
+#endif
+
 //-------------------------------------------------------------------------------------------
 namespace omega
 {
