--- gsk/gl/gskglprofiler.c
+++ gsk/gl/gskglprofiler.c	2026-08-20 06:57:04.000000000 +0800
@@ -1,5 +1,5 @@
 #include "config.h"
-
+#include "gdk/gdkglcontextprivate.h"
 #include "gskglprofilerprivate.h"
 
 #include <epoxy/gl.h>

--- gsk/gl/gskglrenderer.c
+++ gsk/gl/gskglrenderer.c	2026-08-20 06:52:44.000000000 +0800
@@ -303,8 +303,6 @@
   graphene_rect_t viewport;
   GskGLRenderJob *job;
   GdkSurface *surface;
-  graphene_rect_t opaque_tmp;
-  const graphene_rect_t *opaque;
   float scale;
 
   g_assert (GSK_IS_GL_LEGACY_RENDERER (renderer));
@@ -324,15 +322,10 @@
   viewport.size.width = gdk_surface_get_width (surface) * scale;
   viewport.size.height = gdk_surface_get_height (surface) * scale;
 
-  if (gsk_render_node_get_opaque_rect (root, &opaque_tmp))
-    opaque = &opaque_tmp;
-  else
-    opaque = NULL;
   gdk_draw_context_begin_frame_full (GDK_DRAW_CONTEXT (self->context),
                                      NULL,
-                                     gsk_render_node_get_preferred_depth (root),
-                                     update_area,
-                                     opaque);
+                                     root,
+                                     update_area);
 
   gdk_gl_context_make_current (self->context);
 
