--- gsk/gskpathbuilder.c
+++ gsk/gskpathbuilder.c
@@ -1706,7 +1706,7 @@
 gsk_path_builder_add_op (GskPathBuilder         *builder,
                          GskPathOperation        op,
                          const graphene_point_t *pts,
-                         size_t                  n_pts,
+                         unsigned int            n_pts,
                          float                   weight)
 {
   switch (op)
