--- gtk/gtkcolorpicker.c.orig	2026-02-11 13:20:44
+++ gtk/gtkcolorpicker.c	2026-02-11 13:21:18
@@ -22,10 +22,6 @@
 #include "gtkcolorpickershellprivate.h"
 #include "gtkcolorpickerkwinprivate.h"
 
-#ifdef __APPLE__
-#include "gtkcolorpickerquartzprivate.h"
-#endif
-
 #ifdef G_OS_WIN32
 #include "gtkcolorpickerwin32private.h"
 #endif
@@ -62,16 +58,13 @@
 {
   GtkColorPicker *picker = NULL;
 
-#if defined (G_OS_UNIX) && !defined(__APPLE__)
+#if defined(G_OS_UNIX)
   if (!picker)
     picker = gtk_color_picker_portal_new ();
   if (!picker)
     picker = gtk_color_picker_shell_new ();
   if (!picker)
     picker = gtk_color_picker_kwin_new ();
-#elif defined (__APPLE__)
-  if (!picker)
-    picker = gtk_color_picker_quartz_new ();
 #elif defined (G_OS_WIN32)
   if (!picker)
     picker = gtk_color_picker_win32_new ();
