--- a/admin/Makefile.def
+++ b/admin/Makefile.def	2025-07-30 18:33:41.000000000 +0800
@@ -238,6 +238,9 @@
 GLOBAL_CPPFLAGS += -P -traditional -C
 
 # Preprocessor MACHINE dependent flags
+ifeq ($(GAG_MACHINE),powerpc)
+  GLOBAL_CPPFLAGS += -DEEEI="eeei"
+endif
 ifeq ($(GAG_MACHINE),i386)
   GLOBAL_CPPFLAGS += -DIEEE="ieee"
 endif
@@ -369,7 +372,7 @@
     # but useful in some cases:
     # GLOBAL_FFLAGS += -ffpe-trap=invalid,overflow,zero
   endif
-  OPTION_FFLAGS += -fimplicit-none
+  # OPTION_FFLAGS += -fimplicit-none
   ifeq ($(RELEASE_MODE),no)
     OPTION_FFLAGS += -Wuninitialized -Wunused-variable -Wunused-label
   else
@@ -452,8 +455,8 @@
     # are linked to the Gildas libraries which provide only one
     # architecture. Setting $ARCHFLAGS overrides this multi-architecture
     # feature.
-    ifeq ($(GAG_MACHINE),arm64)
-      GAG_PYTHON_DISTFLAGS = ARCHFLAGS='-arch arm64'
+    ifeq ($(GAG_MACHINE),powerpc)
+      GAG_PYTHON_DISTFLAGS = ARCHFLAGS='-arch ppc'
     else
       ifeq ($(GAG_MACHINE),x86_64)
         GAG_PYTHON_DISTFLAGS = ARCHFLAGS='-arch x86_64'
