--- a/admin/define-system.sh
+++ b/admin/define-system.sh	2025-07-30 17:19:34.000000000 +0800
@@ -300,6 +300,14 @@
                 elif which gfortran > /dev/null 2>&1; then
                     DEFAULT_FCOMPILER=gfortran
                 fi
+            elif [ `uname -p` = "powerpc" ]; then
+                GAG_MACHINE=powerpc
+                if which @CC@ > /dev/null 2>&1; then
+                    DEFAULT_CCOMPILER=@CC@
+                fi
+                if which @CXX@ > /dev/null 2>&1; then
+                    DEFAULT_CXXCOMPILER=@CXX@
+                fi
             fi ;;
         CYGWIN*)
             if [ `uname -m | grep -c "x86_64"` -ne 0 ]; then
