--- external/oxen-logging/fmt/CMakeLists.txt
+++ external/oxen-logging/fmt/CMakeLists.txt	2026-06-12 17:58:20.000000000 +0800
@@ -145,9 +145,9 @@
        OFF)
 
 # Options that control generation of various targets.
-option(FMT_DOC "Generate the doc target." ${FMT_MASTER_PROJECT})
-option(FMT_INSTALL "Generate the install target." ON)
-option(FMT_TEST "Generate the test target." ${FMT_MASTER_PROJECT})
+option(FMT_DOC "Generate the doc target." OFF)
+option(FMT_INSTALL "Generate the install target." OFF)
+option(FMT_TEST "Generate the test target." OFF)
 option(FMT_FUZZ "Generate the fuzz target." OFF)
 option(FMT_CUDA_TEST "Generate the cuda-test target." OFF)
 option(FMT_OS "Include core requiring OS (Windows/Posix) " ON)
