--- src/OpenColorIO/CMakeLists.txt	2026-01-13 11:55:53.000000000 +0800
+++ src/OpenColorIO/CMakeLists.txt	2026-05-11 03:20:17.000000000 +0800
@@ -401,7 +401,7 @@
         # Add the path to CMake list of search paths for libraries.
         list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_expat_LIBDIR}")
         # Hide the expat symbols.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-lexpat")
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-lexpat")
     endif()
 
 
@@ -418,7 +418,7 @@
         # Add the path to CMake list of search paths for libraries.
         list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-L${_minizip-ng_LIBDIR}")
         # Hide the minizip-ng symbols.
-        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-hidden-l${_minizip-ng_NAME}")
+        list(APPEND _OCIO_LINK_FLAGS_LIST_ "-Wl,-l${_minizip-ng_NAME}")
     endif()
 
     if (_OCIO_LINK_FLAGS_LIST_)
