--- Makefile.orig	2022-01-26 03:45:21
+++ Makefile	2025-06-28 20:15:51
@@ -5,6 +5,7 @@
 
 export PYTHON ?= python3
 RST2MAN ?= rst2man
+SED ?= sed
 
 version ?= $(shell git describe --dirty 2> /dev/null | cut -b2-)
 version := $(if $(version),$(version),devel)
@@ -26,7 +27,7 @@
 man: git-hub.1
 
 git-hub.1: man.rst git-hub
-	sed 's/^:Version: devel$$/:Version: $(version)/' $< | \
+	$(SED) 's/^:Version: devel$$/:Version: $(version)/' $< | \
 		$(RST2MAN) --exit-status=1 > $@ || ($(RM) $@ && false)
 
 bash-completion: generate-bash-completion git-hub
@@ -34,16 +35,16 @@
 
 .PHONY: install
 install: $(install-deps)
-	install -m 755 -D git-hub $(DESTDIR)$(prefix)/bin/git-hub
-	sed -i 's/^VERSION = "git-hub devel"$$/VERSION = "git-hub $(version)"/' \
+	install -m 755 git-hub $(DESTDIR)$(prefix)/bin/git-hub
+	$(SED) -i 's/^VERSION = "git-hub devel"$$/VERSION = "git-hub $(version)"/' \
 			$(DESTDIR)$(prefix)/bin/git-hub
-	sed -i 's|^#!/usr/bin/env python3$$|#!/usr/bin/env $(PYTHON)|' \
+	$(SED) -i 's|^#!/usr/bin/env python3$$|#!/usr/bin/env $(PYTHON)|' \
 			$(DESTDIR)$(prefix)/bin/git-hub
-	install -m 644 -D git-hub.1 $(DESTDIR)$(prefix)/share/man/man1/git-hub.1
-	install -m 644 -D ftdetect.vim \
+	install -m 644 git-hub.1 $(DESTDIR)$(prefix)/share/man/man1/git-hub.1
+	install -m 644 ftdetect.vim \
 		$(DESTDIR)$(prefix)/share/vim/addons/ftdetect/githubmsg.vim
-	install -m 644 -D bash-completion $(DESTDIR)$(sysconfdir)/bash_completion.d/git-hub
-	install -m 644 -D README.rst $(DESTDIR)$(prefix)/share/doc/git-hub/README.rst
+	install -m 644 bash-completion $(DESTDIR)$(sysconfdir)/bash_completion.d/git-hub
+	install -m 644 README.rst $(DESTDIR)$(prefix)/share/doc/git-hub/README.rst
 
 .PHONY: release
 release:
