--- src/aniworld/autodeps.py	2026-05-31 01:28:03.000000000 +0800
+++ src/aniworld/autodeps.py	2026-05-31 16:12:40.000000000 +0800
@@ -399,8 +399,8 @@
     use_iina = os.getenv("ANIWORLD_USE_IINA") == "1"
     # TODO: check if aniskip is selected in future for IINA to fallback to mpv for functionality if issue #200 is fixed
 
-    if PLATFORM == "Darwin" and use_iina:
-        return manager.fetch_binary("iina")
+#     if PLATFORM == "Darwin" and use_iina:
+#         return manager.fetch_binary("iina")
 
     return manager.fetch_binary("mpv")
 
