--- src/aniworld/autodeps.py	2026-05-31 01:28:03.000000000 +0800
+++ src/aniworld/autodeps.py	2026-05-31 16:12:40.000000000 +0800
@@ -215,8 +215,8 @@
     use_iina = os.getenv("ANIWORLD_USE_IINA") == "1"
     use_aniskip = os.getenv("ANIWORLD_ANISKIP") == "1"
 
-    if PLATFORM == "Darwin" and use_iina and not use_aniskip:
-        return manager.fetch_binary("iina")
+#     if PLATFORM == "Darwin" and use_iina and not use_aniskip:
+#         return manager.fetch_binary("iina")
 
     return manager.fetch_binary("mpv")
 
