--- Makefile	2024-02-25 11:18:31.000000000 +0800
+++ Makefile	2025-03-24 08:35:44.000000000 +0800
@@ -4,7 +4,7 @@
 FC = gfortran
 FCFLAGS = -Wall -Wextra -std=f2018 -pedantic -fcheck=all
 
-OBJS = mod_dill.o libdill.a
+OBJS = mod_dill.o
 
 .PHONY: all clean
 .SUFFIXES: .f90 .o
@@ -12,16 +12,10 @@
 all: client server
 
 client: client.f90 $(OBJS)
-	$(FC) $(FCFLAGS) $< -o $@ libdill.a -pthread
+	$(FC) $(FCFLAGS) $< -o $@ $(PREFIX)/lib/libdill.a -pthread
 
 server: server.f90 $(OBJS)
-	$(FC) $(FCFLAGS) $< -o $@ libdill.a -pthread
-
-libdill:
-	git clone https://github.com/sustrik/libdill
-
-libdill.a: libdill
-	cd libdill && ./autogen.sh && ./configure && $(MAKE) && cp .libs/libdill.a ..
+	$(FC) $(FCFLAGS) $< -o $@ $(PREFIX)/lib/libdill.a -pthread
 
 .f90.o:
 	$(FC) $(FCFLAGS) -c $<
