--- src/nvim/CMakeLists.txt
+++ src/nvim/CMakeLists.txt	2025-01-27 13:35:29.000000000 +0800
@@ -157,10 +157,12 @@
   # Actually export symbols - symbols may not be visible even though
   # ENABLE_EXPORTS is set to true. See
   # https://github.com/neovim/neovim/issues/25295
+if("${DARWIN_MAJOR_VERSION}" GREATER 10)
   target_link_options(nvim_bin PRIVATE "-Wl,-export_dynamic")
   # Apple Clang 16's new deduplication pass leads to lock-up on start
   # TODO(clason): verify that problem remains after release or remove
   target_link_options(nvim_bin PRIVATE "-Wl,-no_deduplicate")
+endif()
 elseif(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
   target_link_libraries(main_lib INTERFACE pthread c++abi)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
