From 025a08698643d8ae426f7512a09e4ec53d931164 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 27 Jan 2025 13:32:29 +0800
Subject: [PATCH] Unbreak linking on macOS < 10.7

---
 src/nvim/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git src/nvim/CMakeLists.txt src/nvim/CMakeLists.txt
index 5b9946db39..033e5b82c1 100644
--- a/src/nvim/CMakeLists.txt
+++ b/src/nvim/CMakeLists.txt
@@ -159,7 +159,9 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
   # Actually export symbols - symbols may not be visible even though
   # ENABLE_EXPORTS is set to true. See
   # https://github.com/neovim/neovim/issues/25295
-  target_link_options(nvim_bin PRIVATE "-Wl,-export_dynamic")
+  if("${DARWIN_MAJOR_VERSION}" GREATER 10)
+    target_link_options(nvim_bin PRIVATE "-Wl,-export_dynamic")
+  endif()
 elseif(CMAKE_SYSTEM_NAME MATCHES "OpenBSD")
   target_link_libraries(main_lib INTERFACE pthread c++abi)
 elseif(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
