--- dep/premake-modules/ninja/ninja.lua	2020-04-12 10:36:40.000000000 +0800
+++ dep/premake-modules/ninja/ninja.lua	2025-05-18 00:12:11.000000000 +0800
@@ -113,7 +113,7 @@
 		if system_name == "windows" then
 			toolset_name = "msc"
 		elseif system_name == "macosx" then
-			toolset_name = "clang"
+			toolset_name = "gcc"
 		elseif system_name == "linux" then
 			toolset_name = "gcc"
 		else

--- premake5.lua	2020-04-12 10:36:40.000000000 +0800
+++ premake5.lua	2025-05-18 04:35:33.000000000 +0800
@@ -1,4 +1,3 @@
-require "dep/premake-modules/xcode-alt"
 require "dep/premake-modules/ninja"
 
 MAIN_VER = '0'
@@ -49,10 +48,7 @@
 workspace "otfcc"
 	configurations { "release", "debug" }
 	
-	platforms { "x64", "x86" }
-	filter "action:xcode4"
-		platforms { "x64" }
-	filter {}
+	platforms { "x64", "x86", "ppc64", "ppc" }
 	
 	location "build"
 	includedirs { "include" }
@@ -67,30 +63,14 @@
 		architecture "x86"
 	filter "platforms:x64"
 		architecture "x64"
+	filter "platforms:ppc"
+		architecture "ppc"
+	filter "platforms:ppc64"
+		architecture "ppc64"
 	filter {}
-	
-	filter "action:vs2017"
-		location "build/vs"
-		toolset "v141_clang_c2"
-		defines { '_CRT_SECURE_NO_WARNINGS', '_CRT_NONSTDC_NO_DEPRECATE' }
-		flags { "StaticRuntime" }
-		includedirs { "dep/polyfill-msvc" }
-	filter "action:vs2015"
-		location "build/vs"
-		toolset "msc-llvm-vs2014"
-		defines { '_CRT_SECURE_NO_WARNINGS', '_CRT_NONSTDC_NO_DEPRECATE' }
-		flags { "StaticRuntime" }
-		includedirs { "dep/polyfill-msvc" }
-	filter "action:ninja"
-		location "build/ninja"
-	filter {"system:windows", "action:ninja"}
-		defines { '_CRT_SECURE_NO_WARNINGS', '_CRT_NONSTDC_NO_DEPRECATE' }
-		flags { "StaticRuntime" }
-		includedirs { "dep/polyfill-msvc" }
+
 	filter "action:gmake"
 		location "build/gmake"
-	filter "action:xcode4"
-		location "build/xcode"
 	filter {}
 	
 	filter "configurations:Debug"
