From f2811f8bad49becff71974b7c2c90e9b766f24c7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 9 Sep 2025 17:23:23 +0800
Subject: [PATCH] test.h: define __STDC_FORMAT_MACROS if not defined

---
 test/test.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/test.h b/test/test.h
index b584c2b37..a372a6407 100644
--- test/test.h
+++ test/test.h
@@ -6,6 +6,12 @@
 #include "../simde/simde-f16.h"
 #include "../simde/simde-bf16.h"
 
+#if (!defined(__clang__) || SIMDE_DETECT_CLANG_VERSION_CHECK(13, 0, 0))
+  #ifndef __STDC_FORMAT_MACROS
+    #define __STDC_FORMAT_MACROS
+  #endif
+#endif
+
 #include <time.h>
 #include <stdlib.h>
 #include <math.h>
