From a787e7313e3f0b45b8e8eef6e282788491b41301 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 2 Jun 2025 17:46:56 +0800
Subject: [PATCH] Minor fix to CMakeLists

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d17927b8..26ae28cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ message(STATUS "Using install prefix ${CMAKE_INSTALL_PREFIX} ...")
 
 find_package(PkgConfig REQUIRED)
 
-set(SEARCH_LIBS /usr/lib /usr/local/lib)
+set(SEARCH_LIBS @PREFIX@/lib /usr/lib)
 
 # [sl 2020-05-23] according to https://cmake.org/cmake/help/v3.0/module/FindwxWidgets.html
 set(wxWidgets_CONFIG_OPTIONS --toolkit=gtk3)
