--- a/src/MainApp.cpp
+++ b/src/MainApp.cpp	2025-06-02 13:07:27.000000000 +0800
@@ -30,7 +30,7 @@
 #include "Settings.h"
 #include "Utils.h"
 
-#include "wx/clipbrd.h"
+#include <wx/clipbrd.h>
 #include <wx/image.h>
 #include <wx/tooltip.h>
 #include <wx/stdpaths.h>


From f3da58ba2fbeb066267c6312574b8de41a1b30f8 Mon Sep 17 00:00:00 2001
From: Thothix <thothix@protonmail.com>
Date: Tue, 3 Jun 2025 17:41:16 -0300
Subject: [PATCH] #63 Make the wx qa component optional (some systems don't
 support it). Builds without it will have the fatal exception report
 generation disabled

---
 CMakeLists.txt  | 17 +++++++++++++----
 Changelog.md    |  2 ++
 src/MainApp.cpp |  9 ++++++++-
 3 files changed, 23 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d17927b8..25fb72b6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,9 +27,17 @@ set(SEARCH_LIBS /usr/lib /usr/local/lib)
 # [sl 2020-05-23] according to https://cmake.org/cmake/help/v3.0/module/FindwxWidgets.html
 set(wxWidgets_CONFIG_OPTIONS --toolkit=gtk3)
 
-find_package(wxWidgets COMPONENTS base core adv net html xml aui qa)
-
-if (NOT wxWidgets_FOUND)
+find_package(wxWidgets COMPONENTS base core adv net html xml aui OPTIONAL_COMPONENTS qa)
+
+if (wxWidgets_FOUND)
+    if (wxWidgets_LIBRARIES)
+        if (wxWidgets_LIBRARIES MATCHES "wx_.*_qa-")   # wxWidgets_qa_FOUND does not work!
+            add_definitions(-DUSE_WXQA=1)
+        else()
+            message(STATUS "wxWidgets QA component not found!")
+        endif()
+    endif()
+else()
     message(FATAL_ERROR "wxWidgets not found!")
 endif()
 
@@ -37,6 +45,7 @@ find_package(ICU 60.0 COMPONENTS uc i18n REQUIRED)
 
 pkg_check_modules(GSTREAMER gstreamer-1.0>=1.20)
 if (NOT GSTREAMER_FOUND)
+    check_required_components()
     pkg_check_modules(GSTREAMER gstreamer-1.0>=1.0)
     if (NOT GSTREAMER_FOUND)
         message(FATAL_ERROR "gstreamer 1.0 not found!")
@@ -100,7 +109,7 @@ if (ENABLE_IPOD)
         add_definitions(-DWITH_LIBGPOD_SUPPORT=1)
     endif()
 else()
-    message(" *** Disabled iPod support!")
+    message(" *** iPod support disabled!")
 endif()
 
 # Include wxWidgets macros
diff --git a/src/MainApp.cpp b/src/MainApp.cpp
index 3ee9b304..855e940c 100644
--- a/src/MainApp.cpp
+++ b/src/MainApp.cpp
@@ -34,7 +34,9 @@
 #include <wx/stdpaths.h>
 #include <wx/utils.h>
 
-#include <wx/debugrpt.h>
+#if USE_WXQA
+    #include <wx/debugrpt.h>
+#endif
 
 IMPLEMENT_APP( Guayadeque::guMainApp );
 
@@ -489,6 +491,8 @@ int guMainApp::OnExit()
 // -------------------------------------------------------------------------------- //
 void guMainApp::OnFatalException()
 {
+#if USE_WXQA
+    guLogMessage(wxT("wxQA is enabled. Generating exception report..."));
     wxDebugReport Report;
     wxDebugReportPreviewStd Preview;
 
@@ -496,6 +500,9 @@ void guMainApp::OnFatalException()
 
     if( Preview.Show( Report ) )
         Report.Process();
+#else
+    guLogMessage(wxT("wxQA is disabled"));
+#endif
 }
 
 }
