--- gegl/meson.build	2024-11-03 10:39:32.000000000 -0600
+++ gegl/meson.build	2024-11-06 07:44:10.000000000 -0600
@@ -89,7 +89,7 @@
   subdir: api_name
 )
 
-gegl_ldflags = os_osx ? ['-framework', 'OpenCL'] : []
+gegl_ldflags = []
 
 opencl_dep = declare_dependency(
   sources : [opencl_headers],

--- gegl/opencl/gegl-cl-init.c	2026-09-07 10:51:56.000000000 +0800
+++ gegl/opencl/gegl-cl-init.c	2026-09-12 21:36:08.000000000 +0800
@@ -303,7 +303,7 @@
 
 #if defined(__APPLE__)
 #define GL_LIBRARY_NAME "/System/Library/Frameworks/OpenGL.framework/Versions/Current/OpenGL"
-#define CL_LIBRARY_NAME "/System/Library/Frameworks/OpenCL.framework/Versions/Current/OpenCL"
+#define CL_LIBRARY_NAME "libOpenCL.dylib"
 #elif defined(G_OS_WIN32)
 #define CL_LIBRARY_NAME "OpenCL.dll"
 #else
@@ -311,7 +311,7 @@
 #define CL_LIBRARY_NAME "libOpenCL.so.1"
 #endif
 
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
 #define CL_LOAD_FUNCTION(func)                                                    \
 G_STMT_START                                                                      \
   {                                                                               \
@@ -376,7 +376,7 @@
 static gboolean
 gegl_cl_init_load_functions (void)
 {
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   GModule *module = g_module_open (CL_LIBRARY_NAME, G_MODULE_BIND_LAZY);
 #else
   void *module = dlopen(CL_LIBRARY_NAME, RTLD_LAZY / RTLD_LOCAL);
@@ -406,12 +406,12 @@
   CL_LOAD_FUNCTION (clGetContextInfo);
   CL_LOAD_FUNCTION (clReleaseContext);
   CL_LOAD_FUNCTION (clRetainContext);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clSetContextDestructorCallback);
 #endif
 
   /* Command-Queues */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreateCommandQueueWithProperties);
 #else
   CL_LOAD_FUNCTION (clCreateCommandQueue);
@@ -419,13 +419,13 @@
   CL_LOAD_FUNCTION (clGetCommandQueueInfo);
   CL_LOAD_FUNCTION (clReleaseCommandQueue);
   CL_LOAD_FUNCTION (clRetainCommandQueue);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clSetDefaultDeviceCommandQueue);
 #endif
 
   /* Buffer Objects */
   CL_LOAD_FUNCTION (clCreateBuffer);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreateBufferWithProperties);
 #endif
   CL_LOAD_FUNCTION (clCreateSubBuffer);
@@ -440,7 +440,7 @@
 
   /* Image Objects */
   CL_LOAD_FUNCTION (clCreateImage);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreateImageWithProperties);
 #endif
   CL_LOAD_FUNCTION (clEnqueueReadImage);
@@ -462,7 +462,7 @@
   CL_LOAD_FUNCTION (clSetMemObjectDestructorCallback);
 
   /* Sampler Objects */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreateSamplerWithProperties);
 #endif
   CL_LOAD_FUNCTION (clReleaseSampler);
@@ -475,7 +475,7 @@
   CL_LOAD_FUNCTION (clCreateProgramWithSource);
   CL_LOAD_FUNCTION (clCreateProgramWithBinary);
   CL_LOAD_FUNCTION (clCreateProgramWithBuiltInKernels);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreateProgramWithIL);
 #endif
   CL_LOAD_FUNCTION (clGetProgramBuildInfo);
@@ -483,27 +483,27 @@
   CL_LOAD_FUNCTION (clLinkProgram);
   CL_LOAD_FUNCTION (clReleaseProgram);
   CL_LOAD_FUNCTION (clRetainProgram);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clSetProgramSpecializationConstant);
 #endif
   CL_LOAD_FUNCTION (clUnloadPlatformCompiler);
 
   /* Kernel Objects */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCloneKernel);
 #endif
   CL_LOAD_FUNCTION (clCreateKernel);
   CL_LOAD_FUNCTION (clCreateKernelsInProgram);
   CL_LOAD_FUNCTION (clGetKernelInfo);
   CL_LOAD_FUNCTION (clGetKernelArgInfo);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clGetKernelSubGroupInfo);
 #endif
   CL_LOAD_FUNCTION (clGetKernelWorkGroupInfo);
   CL_LOAD_FUNCTION (clReleaseKernel);
   CL_LOAD_FUNCTION (clRetainKernel);
   CL_LOAD_FUNCTION (clSetKernelArg);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clSetKernelArgSVMPointer);
   CL_LOAD_FUNCTION (clSetKernelExecInfo);
 #endif
@@ -526,11 +526,11 @@
   CL_LOAD_FUNCTION (clEnqueueMarkerWithWaitList);
 
   /* Profiling Operations on Memory Objects and Kernels */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clGetDeviceAndHostTimer);
 #endif
   CL_LOAD_FUNCTION (clGetEventProfilingInfo);
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clGetHostTimer);
 #endif
 
@@ -539,13 +539,13 @@
   CL_LOAD_FUNCTION (clFinish);
 
   /* Pipes */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clCreatePipe);
   CL_LOAD_FUNCTION (clGetPipeInfo);
 #endif
 
   /* Shared Virtual Memory (SVM) */
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
   CL_LOAD_FUNCTION (clSVMAlloc);
   CL_LOAD_FUNCTION (clSVMFree);
   CL_LOAD_FUNCTION (clEnqueueSVMFree);
@@ -722,7 +722,7 @@
 
   if (!cl_state.is_loaded)
     {
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
       cl_queue_properties queue_props[] = { CL_QUEUE_PROPERTIES, 0, 0 };
 #else
       cl_command_queue_properties command_queue_flags = 0;
@@ -751,17 +751,17 @@
 
       cl_state.ctx = ctx;
 
-#ifdef __APPLE__
+#ifdef __APPLE_OPENCL__
       command_queue_flags = 0;
 #endif
       if (cl_state.enable_profiling)
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
         queue_props[1] = CL_QUEUE_PROFILING_ENABLE;
 #else
         command_queue_flags |= CL_QUEUE_PROFILING_ENABLE;
 #endif
 
-#ifndef __APPLE__
+#ifndef __APPLE_OPENCL__
       cl_state.cq = gegl_clCreateCommandQueueWithProperties (cl_state.ctx, cl_state.device, (const cl_queue_properties *)&queue_props, &err);
 #else
       cl_state.cq = gegl_clCreateCommandQueue (cl_state.ctx, cl_state.device, command_queue_flags, &err);
