--- bin/confgen/tsig-keygen.c	2026-06-08 22:12:40.000000000 +0800
+++ bin/confgen/tsig-keygen.c	2026-06-22 14:05:35.000000000 +0800
@@ -100,7 +100,7 @@
 
 	result = isc_file_progname(*argv, program, sizeof(program));
 	if (result != ISC_R_SUCCESS) {
-		memmove(program, "tsig-keygen", 11);
+		memmove(program, "tsig-keygen-orig", 11);
 	}
 	progname = program;
 
@@ -115,10 +115,10 @@
 #define PROGCMP(X) \
 	(strcasecmp(progname, X) == 0 || strcasecmp(progname, X ".exe") == 0)
 
-	if (PROGCMP("tsig-keygen")) {
+	if (PROGCMP("tsig-keygen-orig")) {
 		progmode = progmode_keygen;
 		quiet = true;
-	} else if (PROGCMP("ddns-confgen")) {
+	} else if (PROGCMP("ddns-confgen-orig")) {
 		progmode = progmode_confgen;
 	} else {
 		UNREACHABLE();
