From f94b853f5dc6c24a804b1776929d81fae4b729d8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 7 Aug 2025 21:06:17 +0800
Subject: [PATCH] WServerGLWidget.C: fix Apple GL for < 10.7

---
 src/Wt/WServerGLWidget.C | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git src/Wt/WServerGLWidget.C src/Wt/WServerGLWidget.C
index df7daf9e..e9285c44 100644
--- src/Wt/WServerGLWidget.C
+++ src/Wt/WServerGLWidget.C
@@ -47,7 +47,7 @@ typedef GLXContext (*glXCreateContextAttribsARBProc)(Display*, GLXFBConfig, GLXC
 #include <OpenGL/CGLCurrent.h>
 #include <OpenGL/CGLRenderers.h>
 #include <OpenGL/CGLTypes.h>
-#include <OpenGL/OpenGL.h>
+#include <AvailabilityMacros.h>
 #endif
 
 namespace {
@@ -254,9 +254,11 @@ WServerGLWidgetImpl::WServerGLWidgetImpl(bool antialiasingEnabled):
 {
   CGLPixelFormatAttribute attributes[4] = {
     kCGLPFAAccelerated,   // no software rendering
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
     kCGLPFAOpenGLProfile, // core profile with the version stated below
     (CGLPixelFormatAttribute) kCGLOGLPVersion_Legacy,
     //    (CGLPixelFormatAttribute) kCGLOGLPVersion_3_2_Core,
+#endif
     (CGLPixelFormatAttribute) 0
   };
   CGLPixelFormatObj pix;
