--- nss/coreconf/Darwin.mk
+++ nss/coreconf/Darwin.mk	2025-12-06 00:22:23.000000000 +0800
@@ -31,15 +31,26 @@
 CC              += -arch arm64
 CCC             += -arch arm64
 override CPU_ARCH	= aarch64
-else ifeq (powerpc,$(CPU_ARCH))
+else ifeq (ppc,$(CPU_ARCH))
 OS_REL_CFLAGS	= -Dppc
 CC              += -arch ppc
 CCC             += -arch ppc
 override CPU_ARCH	= ppc
+else ifeq (ppc64,$(CPU_ARCH))
+OS_REL_CFLAGS	= -Dppc
+CC              += -arch ppc64
+CCC             += -arch ppc64
+override CPU_ARCH	= ppc64
 else
    $(error Unknown CPU architecture)
 endif
 
+ifeq (,$(filter-out ppc ppc64,$(CPU_ARCH)))
+ifneq ($(NSS_DISABLE_CRYPTO_VSX),0)
+	NSS_DISABLE_CRYPTO_VSX=1
+endif
+endif
+
 ifneq (,$(MACOS_SDK_DIR))
     GCC_VERSION_FULL := $(shell $(CC) -dumpversion)
     GCC_VERSION_MAJOR := $(shell echo $(GCC_VERSION_FULL) | awk -F. '{ print $$1 }')
