--- SConstruct
+++ SConstruct	2025-04-11 06:30:35.000000000 +0800
@@ -4,7 +4,7 @@
 import eol_scons
 
 
-env = Environment(tools = ['default', 'prefixoptions', 'openmotif', 'netcdfcxx', 'raf'])
+env = Environment(CC = '@CC@', CXX = '@CXX@', tools = ['default', 'prefixoptions', 'openmotif', 'netcdfcxx', 'raf'])
 
 env.Append(CXXFLAGS='-std=c++11 -g -Wall -Wno-write-strings -Wstrict-aliasing -Wno-deprecated-register')
 

--- raf/tool_raf.py
+++ raf/tool_raf.py	2025-04-11 06:54:13.000000000 +0800
@@ -10,7 +10,7 @@
 # requiring the 'raf' tool will instead load the raf tool from eol_scons,
 # and that tool links against the installed raf library.
 
-env = Environment(tools=['default', 'prefixoptions'])
+env = Environment(CC = '@CC@', CXX = '@CXX@', tools=['default', 'prefixoptions'])
 
 env.Append(CCFLAGS='-g -Wall -Wno-write-strings')
 env.Append(CXXFLAGS='-g -Wall -std=c++11 -Wno-write-strings -Wno-deprecated-register ')
