--- iconv/iconv.ml.orig	2023-03-21 18:21:51.000000000 +0800
+++ iconv/iconv.ml	2025-10-19 02:31:45.000000000 +0800
@@ -9,17 +9,17 @@
 let iconv_opt : iconv option typ = ptr_opt void
 
 let iconv_open = foreign ~check_errno:true
-  "iconv_open" (string @-> string @-> (returning iconv))
+  "libiconv_open" (string @-> string @-> (returning iconv))
 
 let iconv_open ~fromcode ~tocode = iconv_open tocode fromcode
 
 let iconv_close = foreign ~check_errno:true
-  "iconv_close" (iconv @-> returning int)
+  "libiconv_close" (iconv @-> returning int)
 
 let iconv_close c = ignore(iconv_close c)
 
 let iconv = foreign ~check_errno:true
-  "iconv" (iconv @-> ptr string @-> ptr int @->
+  "libiconv" (iconv @-> ptr string @-> ptr int @->
    ptr (ptr char) @-> ptr int @-> (returning int))
 
 let iconv c input =
