--- src/lxml/etree.c	2024-03-29 03:55:13.000000000 +0800
+++ src/lxml/etree.c	2026-07-17 21:31:43.000000000 +0800
@@ -27691,7 +27691,10 @@
  *                 _fixThreadDictPtr(&c_attribute.name, c_src_dict, c_dict)
  *                 _fixThreadDictPtr(&c_attribute.prefix, c_src_dict, c_dict)
  */
+    #pragma GCC diagnostic push
+    #pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
         __pyx_f_4lxml_5etree__fixThreadDictPtr((&__pyx_v_c_attribute->defaultValue), __pyx_v_c_src_dict, __pyx_v_c_dict);
+    #pragma GCC diagnostic pop
 
         /* "src/lxml/proxy.pxi":578
  *             while c_attribute:
