--- CMakeLists.txt	2026-05-01 07:44:06.000000000 +0800
+++ CMakeLists.txt	2026-05-16 10:42:34.000000000 +0800
@@ -420,6 +420,14 @@
     set(PA_USE_ALSA OFF)
 endif()
 
+option(PA_ENABLE_CXX "Install C++ bindings" ON)
+if(PA_ENABLE_CXX)
+  if(PA_BUILD_SHARED_LIBS)
+    set(BUILD_SHARED_LIBS ON CACHE BOOL "" FORCE)
+  endif()
+  add_subdirectory(bindings/cpp)
+endif()
+
 # Add public headers to sources of PortAudio (used by some IDEs to list them in project tree)
 source_group("Public Header Files" FILES ${PORTAUDIO_PUBLIC_HEADERS})
 target_sources(portaudio PRIVATE ${PORTAUDIO_PUBLIC_HEADERS})

--- bindings/cpp/CMakeLists.txt	2026-05-01 07:44:06.000000000 +0800
+++ bindings/cpp/CMakeLists.txt	2026-05-17 07:35:39.000000000 +0800
@@ -1,6 +1,3 @@
-cmake_minimum_required(VERSION 3.13)
-cmake_policy(VERSION 3.13)
-
 project(PortAudioCpp VERSION 19.8 LANGUAGES CXX)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules")
 
@@ -68,9 +65,7 @@
 add_library(portaudiocpp ${portaudiocpp-sources})
 add_library(PortAudio::portaudiocpp ALIAS portaudiocpp) # For subdirectory build
 
-find_package(PortAudio MODULE REQUIRED)
-
-target_link_libraries(portaudiocpp PUBLIC PortAudio::portaudio)
+target_link_libraries(portaudiocpp PUBLIC portaudio)
 target_include_directories(
   portaudiocpp PUBLIC $<INSTALL_INTERFACE:include>
                       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
