--- CMakeLists.txt	2026-05-01 07:44:06.000000000 +0800
+++ CMakeLists.txt	2026-05-16 10:42:34.000000000 +0800
@@ -420,6 +420,14 @@
     set(PA_USE_ALSA OFF)
 endif()
 
+option(PA_ENABLE_CXX "Install C++ bindings" ON)
+if(PA_ENABLE_CXX)
+  if(PA_BUILD_SHARED_LIBS)
+    set(BUILD_SHARED_LIBS ON CACHE BOOL "" FORCE)
+  endif()
+  add_subdirectory(bindings/cpp)
+endif()
+
 # Add public headers to sources of PortAudio (used by some IDEs to list them in project tree)
 source_group("Public Header Files" FILES ${PORTAUDIO_PUBLIC_HEADERS})
 target_sources(portaudio PRIVATE ${PORTAUDIO_PUBLIC_HEADERS})
