--- CMakeLists.txt	2013-10-17 22:18:16.000000000 +0800
+++ CMakeLists.txt	2025-02-06 10:34:51.000000000 +0800
@@ -89,10 +89,10 @@
   endif()
 
   set(CMAKE_CXX_FLAGS
-    " -pedantic-errors -Wall -Wextra -Winit-self -Woverloaded-virtual -Wnon-virtual-dtor -Wsign-promo -Wpadded -Weffc++ -Wno-long-long -std=c++98")
+    " ${CMAKE_CXX_FLAGS} -pedantic-errors -Wall -Wextra -Winit-self -Woverloaded-virtual -Wnon-virtual-dtor -Wsign-promo -Wpadded -Weffc++ -Wno-long-long")
 
   set(CMAKE_C_FLAGS
-    " -pedantic-errors -Wall -Wextra -std=c99")
+    " ${CMAKE_C_FLAGS} -pedantic-errors -Wall -Wextra -std=c99")
 
   set(CMAKE_CXX_FLAGS_RELEASE "-O3 -DNDEBUG ")
   set(CMAKE_CXX_FLAGS_DEBUG "-g3 -DDEBUG ")
