--- Makefile.in	2024-12-30 07:42:49.000000000 +0800
+++ Makefile.in	2025-07-03 18:00:32.000000000 +0800
@@ -1,7 +1,7 @@
 TOPDIR=./
 include $(TOPDIR)Rules.make
 
-SUBDIRS=boot cpiface dev doc devp devw filesel goodies help medialib playay playcda playflac playgmd playtimidity playhvl playit playogg playopl plugins
+SUBDIRS=boot cpiface dev doc devp devw filesel goodies help medialib playay playcda playflac playgmd playtimidity playhvl playit playogg plugins
 ifeq ($(HAVE_MAD),1)
 SUBDIRS+=playmp2
 endif

--- configure	2024-12-30 07:42:49.000000000 +0800
+++ configure	2025-07-03 18:32:56.000000000 +0800
@@ -14193,7 +14193,7 @@
 
 
 
-HAVE_ADPLUG=1
+HAVE_ADPLUG=0
 ADPLUG_INI=opltype
 VERSION="2.3.4-beta"
  ac_ext=c
@@ -18912,8 +18912,6 @@
 
 ac_config_files="$ac_config_files playopl/libbinio-git/src/binio.h"
 
-ac_config_files="$ac_config_files playopl/adplug-git/src/version.h"
-
 ac_config_files="$ac_config_files haiku/ocp.rdef"
 
 
@@ -19620,7 +19618,6 @@
     "playsid/libsidplayfp-git/src/builders/resid-builder/resid/siddefs.h") CONFIG_FILES="$CONFIG_FILES playsid/libsidplayfp-git/src/builders/resid-builder/resid/siddefs.h" ;;
     "playsid/libsidplayfp-git/src/builders/residfp-builder/residfp/siddefs-fp.h") CONFIG_FILES="$CONFIG_FILES playsid/libsidplayfp-git/src/builders/residfp-builder/residfp/siddefs-fp.h" ;;
     "playopl/libbinio-git/src/binio.h") CONFIG_FILES="$CONFIG_FILES playopl/libbinio-git/src/binio.h" ;;
-    "playopl/adplug-git/src/version.h") CONFIG_FILES="$CONFIG_FILES playopl/adplug-git/src/version.h" ;;
     "haiku/ocp.rdef") CONFIG_FILES="$CONFIG_FILES haiku/ocp.rdef" ;;
     "mingw/ocp.rc") CONFIG_FILES="$CONFIG_FILES mingw/ocp.rc" ;;
 
@@ -20211,8 +20208,6 @@
 else $as_nop
   echo "debug:             OFF"
 fi
-                                           echo "builtin:           $builtin"
-                                           echo "adplug:            GIT/builtin"
 if test "x$with_mad"       = "xno"
 then :
   echo "mad:               OFF"
