--- a/src/src.pri
+++ b/src/src.pri
@@ -102,11 +102,11 @@ win32:RC_FILE = $$TMP_DIR/icon.rc

 mac {
     system($$QMAKE_MKDIR $$TMP_DIR/icon.iconset)
-    SIZES = 16 32 128 256
+    SIZES = 16 32 128 256 512
     for(size, SIZES) {
         system($$CONVERT_CMD $$SRC_DIR/icons/icon.svg -density 72 -resize $${size}x$${size} -units PixelsPerInch $$TMP_DIR/icon.iconset/icon_$${size}x$${size}.png)
     }
-    system(iconutil -c icns -o $$TMP_DIR/icon.icns $$TMP_DIR/icon.iconset)
+    system(makeicns -16 $$TMP_DIR/icon.iconset/icon_16x16.png -32 $$TMP_DIR/icon.iconset/icon_32x32.png -128 $$TMP_DIR/icon.iconset/icon_128x128.png -256 $$TMP_DIR/icon.iconset/icon_256x256.png -512 $$TMP_DIR/icon.iconset/icon_512x512.png -out $$TMP_DIR/icon.icns)
     ICON = $$TMP_DIR/icon.icns
 }

