--- nglib/CMakeLists.txt	2026-04-24 21:35:17.000000000 +0800
+++ nglib/CMakeLists.txt	2026-05-14 12:58:46.000000000 +0800
@@ -6,7 +6,7 @@
 endif(USE_OCC)
 
 target_link_libraries(nglib PRIVATE ${ZLIB_LIBRARIES})
-target_link_libraries( nglib PUBLIC ngcore PRIVATE ${CMAKE_THREAD_LIBS_INIT} ${JPEG_LIBRARIES} ${MKL_LIBRARIES} occ_libs netgen_cgns )
+target_link_libraries( nglib PUBLIC ngcore PRIVATE ${CMAKE_THREAD_LIBS_INIT} ${JPEG_LIBRARIES} ${MKL_LIBRARIES} occ_libs netgen_cgns atomic)
 
 install(TARGETS nglib netgen_cgns ${NG_INSTALL_DIR})
 install(FILES nglib.h DESTINATION ${NG_INSTALL_DIR_INCLUDE} COMPONENT netgen_devel)
