--- DAEValidator/library/src/DaeValidator.cpp	2018-11-27 06:43:10.000000000 +0800
+++ DAEValidator/library/src/DaeValidator.cpp	2025-04-30 18:34:05.000000000 +0800
@@ -4,6 +4,7 @@
 #include "Strings.h"
 #include "StringUtil.h"
 #include <cmath>
+#include <cstdint>
 #include "no_warning_iomanip"
 #include "no_warning_iostream"
 #include <set>

--- DAEValidator/library/src/XmlDoc.cpp	2018-11-27 06:43:10.000000000 +0800
+++ DAEValidator/library/src/XmlDoc.cpp	2025-04-30 18:41:29.000000000 +0800
@@ -3,6 +3,7 @@
 #if IS_GNUC_AND_GNUC_VERSION_LT(5,1,1)
 #include <cstring>
 #endif
+#include <cstdint>
 
 #include "XmlDoc.h"
 #include "XmlNode.h"
