From 862dcca30d654ed7356db34fc18052f8ceaa1a14 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 30 Apr 2025 01:51:32 +0800
Subject: [PATCH] CMakeLists: do not use clang flag with gcc

---
 src/cb/CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git src/cb/CMakeLists.txt src/cb/CMakeLists.txt
index e975488..bfd8610 100644
--- src/cb/CMakeLists.txt
+++ src/cb/CMakeLists.txt
@@ -59,7 +59,9 @@ elseif(APPLE)
     src/platforms/macos.mm
   )
   target_link_libraries(cb "-framework AppKit")
-  target_compile_options(cb PRIVATE -fobjc-arc)
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    target_compile_options(cb PRIVATE -fobjc-arc)
+  endif()
 elseif(HAIKU)
   target_sources(cb PRIVATE
     src/platforms/haiku.cpp
