From f3fc947a61c24c37028b44cf3c5d44b466379d43 Mon Sep 17 00:00:00 2001
From: Daniel Kettle <initial.dann@gmail.com>
Date: Tue, 14 Apr 2020 17:23:43 -0700
Subject: [PATCH 1/7] Added configure-generated files into .gitignore

---
 .gitignore | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git .gitignore .gitignore
index e43b0f988..acdc6c171 100644
--- .gitignore
+++ .gitignore
@@ -1 +1,18 @@
 .DS_Store
+*.o
+*.d
+Makefile
+.qmake.cache
+.qmake.stash
+qtbase/bin/qmake
+qtbase/bin/qt.conf
+config.status
+qtbase/config.tests/*
+.qmake.super
+config.summary
+qtbase/mkspecs/qconfig.pri
+qtbase/mkspecs/qdevice.pri
+qtbase/mkspecs/qmodule.pri
+qtbase/src/corelib/global/qconfig.cpp
+qtbase/src/corelib/global/qconfig.h
+
-- 
2.42.0

