From ab4646fb4405c211402cd0ae385bc47d2a96c6cf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 5 Apr 2025 14:58:21 +0800
Subject: [PATCH 2/2] gemma3-cli.cpp: fix inttypes include

---
 examples/llava/gemma3-cli.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git examples/llava/gemma3-cli.cpp examples/llava/gemma3-cli.cpp
index 7813ac19..8cc75854 100644
--- examples/llava/gemma3-cli.cpp
+++ examples/llava/gemma3-cli.cpp
@@ -10,7 +10,10 @@
 
 #include <vector>
 #include <limits.h>
-#include <inttypes.h>
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
 
 #if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
 #include <signal.h>
-- 
2.48.0

