--- tools/server/server-models.cpp	2026-07-11 04:07:29.000000000 +0800
+++ tools/server/server-models.cpp	2026-07-11 22:44:53.000000000 +0800
@@ -29,7 +29,7 @@
 #include <sstream>
 #include <cstring>
 
-#ifndef _WIN32
+#ifndef __APPLE__
 extern char **environ;
 #endif
 
@@ -37,6 +37,8 @@
 // macOS: use _NSGetExecutablePath to get the executable path
 #include <mach-o/dyld.h>
 #include <limits.h>
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
 #endif
 
 #define DEFAULT_STOP_TIMEOUT 10 // seconds

--- vendor/sheredom/subprocess.h	2026-06-13 21:44:15.000000000 +0800
+++ vendor/sheredom/subprocess.h	2026-06-14 00:55:19.000000000 +0800
@@ -44,6 +44,11 @@
 #include <stdio.h>
 #include <string.h>
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #if defined(_MSC_VER)
 #pragma warning(pop)
 #endif
@@ -773,7 +778,9 @@
   int stdoutfd[2];
   int stderrfd[2];
   pid_t child;
+#ifndef __APPLE__
   extern char **environ;
+#endif
   char *const empty_environment[1] = {SUBPROCESS_NULL};
   posix_spawn_file_actions_t actions;
   char *const *used_environment;
