--- tests/test-rset-release.cpp.orig	2026-08-07 01:46:40.000000000 +0800
+++ tests/test-rset-release.cpp	2026-08-07 12:14:34.000000000 +0800
@@ -17,7 +17,7 @@
     if (host_statistics64(mach_host_self(), HOST_VM_INFO64, (host_info64_t)&vmstat, &count) != KERN_SUCCESS) {
         return UINT64_MAX;
     }
-    return static_cast<uint64_t>(vmstat.wire_count) * vm_kernel_page_size;
+    return static_cast<uint64_t>(vmstat.wire_count) * vm_page_size;
 }
 
 int main(int argc, char ** argv) {
