From fa4eca2f9134c721c311e281549c348998b67219 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 4 Jun 2025 22:05:34 +0800
Subject: [PATCH] stream-dump.cpp: add missing cinttypes

---
 src/librepgp/stream-dump.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/librepgp/stream-dump.cpp src/librepgp/stream-dump.cpp
index 0aedf573..ba71f3aa 100644
--- src/librepgp/stream-dump.cpp
+++ src/librepgp/stream-dump.cpp
@@ -49,6 +49,10 @@
 #include "key.hpp"
 #include "json-utils.h"
 #include <algorithm>
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
 
 static const id_str_pair packet_tag_map[] = {
   {PGP_PKT_RESERVED, "Reserved"},
-- 
2.48.0

