From eedc71c822245b8676cedfecc4225e7fb73c9187 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 18 Mar 2025 07:30:37 +0800
Subject: [PATCH 3/3] lib: rnp.cpp: fix inttypes

---
 src/lib/rnp.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git src/lib/rnp.cpp src/lib/rnp.cpp
index 0e93aabf..28702394 100644
--- src/lib/rnp.cpp
+++ src/lib/rnp.cpp
@@ -45,10 +45,13 @@
 #include <stdlib.h>
 #ifdef _MSC_VER
 #include "uniwin.h"
-#include <inttypes.h>
 #else
 #include <unistd.h>
 #endif
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
 #include <string.h>
 #include <sys/stat.h>
 #include <stdexcept>
-- 
2.48.0

