From ea5e1f67ab7e2b578463fec6e5c5173b33763ece Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 20 Mar 2025 01:59:07 +0800
Subject: [PATCH] Tests support: use POSIX syntax for cp

---
 src/tests/support.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/tests/support.cpp src/tests/support.cpp
index 66316769..393da0a0 100644
--- src/tests/support.cpp
+++ src/tests/support.cpp
@@ -290,7 +290,7 @@ copy_recursively(const char *src, const char *dst)
     // TODO: maybe use fts or something less hacky
     char buf[2048];
 #ifndef _WIN32
-    snprintf(buf, sizeof(buf), "cp -a '%s' '%s'", src, dst);
+    snprintf(buf, sizeof(buf), "cp -pRP '%s' '%s'", src, dst);
 #else
     snprintf(buf, sizeof(buf), "xcopy \"%s\" \"%s\" /I /Q /E /Y", src, dst);
 #endif // _WIN32
