--- src/ppx_stable.ml	2023-04-25 21:12:25.000000000 +0800
+++ src/ppx_stable.ml	2025-10-13 19:50:34.000000000 +0800
@@ -243,7 +243,7 @@
         case ~guard:None ~lhs:pattern ~rhs:value)
     in
     let expr =
-      List.fold_right ~init:(pexp_function ~loc cases) cdl ~f:(fun cd acc ->
+      List.fold_right ~init:(pexp_function_cases ~loc cases) cdl ~f:(fun cd acc ->
         let name = String.lowercase cd.pcd_name.txt in
         pexp_fun ~loc (Labelled name) None (pvar ~loc (alias_fun_label cd)) acc)
     in
