--- libuuid/src/uuid_time.c	2026-04-28 17:11:39.000000000 +0800
+++ libuuid/src/uuid_time.c	2026-05-24 18:46:22.000000000 +0800
@@ -143,7 +143,7 @@
 }
 #if defined(__USE_TIME_BITS64) && defined(__GLIBC__)
 extern time_t uuid_time64(const uuid_t uu, struct timeval *ret_tv) __attribute__((weak, alias("__uuid_time")));
-#elif defined(__clang__) && defined(__APPLE__)
+#elif defined(__APPLE__)
 __asm__(".globl _uuid_time");
 __asm__(".set _uuid_time, ___uuid_time");
 extern time_t uuid_time(const uuid_t uu, struct timeval *ret_tv);
