From 5ae8505de834345e02474fc5874f54d86778c266 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 28 Aug 2026 11:31:21 +0800
Subject: [PATCH 2/2] yt_http.h: define MSG_NOSIGNAL when undefined

---
 include/ytcui-dl/yt_http.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git include/ytcui-dl/yt_http.h include/ytcui-dl/yt_http.h
index 3fc485b..2565295 100644
--- include/ytcui-dl/yt_http.h
+++ include/ytcui-dl/yt_http.h
@@ -44,6 +44,10 @@
 #include <openssl/ssl.h>
 #include <zlib.h>
 
+#ifndef MSG_NOSIGNAL
+#define MSG_NOSIGNAL 0
+#endif
+
 namespace ytfast {
 
 // ---------------------------------------------------------------------------
-- 
2.55.0

