From 69d899402ed50125ba192350f0569afe1857f0d2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 16 Aug 2025 17:33:41 +0800
Subject: [PATCH] futex.h: unbreak build on macOS < 10.12

---
 hwy/contrib/thread_pool/futex.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git hwy/contrib/thread_pool/futex.h hwy/contrib/thread_pool/futex.h
index a6c79760..5848fe96 100644
--- hwy/contrib/thread_pool/futex.h
+++ hwy/contrib/thread_pool/futex.h
@@ -43,6 +43,13 @@
 #include <windows.h>
 #endif
 
+#if HWY_OS_APPLE
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 101200
+#define HWY_DISABLE_FUTEX
+#endif
+#endif
+
 #if HWY_ARCH_WASM
 #include <emscripten/threading.h>
 #include <math.h>  // INFINITY
