--- Makefile.orig	2017-07-26 02:44:57.000000000 +0800
+++ Makefile	2024-06-10 17:54:58.000000000 +0800
@@ -11,7 +11,7 @@
 #
 # CFLAGS settings for 64 bit Linux/unix systems.
 #
-export CFLAGS=-m64 -march=native -mtune=native -Ofast -fopenmp -ansi -Wall -DBIT64 -DUSEOPENMP -DSETNUMTHREADS -DUSESIGTERM -DUSEGETTIME -I../include
+export CFLAGS= -Os -fopenmp -ansi -Wall -DUSEOPENMP -DSETNUMTHREADS -DUSESIGTERM -DUSEGETTIME -I../include
 #
 # LIBS settings for 64 bit Linux/unix systems.
 #
