--- src/dune	2024-05-08 02:46:09.000000000 +0800
+++ src/dune	2026-08-07 18:01:46.000000000 +0800
@@ -1,8 +1,10 @@
 (library
  (foreign_stubs
   (language c)
-  (names base_bigstring_stubs))
+  (names base_bigstring_stubs)
+  (flags (:standard -I@PREFIX@/include/LegacySupport)))
  (name base_bigstring)
+ (c_library_flags -lMacportsLegacySupport)
  (public_name base_bigstring)
  (js_of_ocaml
   (javascript_files runtime.js))
