From 5f57684ff3119d296ff28e3d9bdcc7a4c89a5dee Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Sat, 15 Oct 2022 08:42:40 +0800
Subject: [PATCH 04/10] portsandbox: fix for 10a190

---
 src/port1.0/portsandbox.tcl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/port1.0/portsandbox.tcl src/port1.0/portsandbox.tcl
index cff38a6d9..6b3114421 100644
--- src/port1.0/portsandbox.tcl
+++ src/port1.0/portsandbox.tcl
@@ -120,7 +120,7 @@ proc portsandbox::set_profile {target} {
     foreach dir $allow_dirs {
         foreach perm $perms {
             append portsandbox_profile " (allow $perm ("
-            if {${os.major} > 9} {
+            if {${os.major} > 10} {
                 append portsandbox_profile "subpath \"${dir}\"))"
             } else {
                 append portsandbox_profile "regex #\"^${dir}/\"))"
-- 
2.47.0

