From 58882605f5d46c51bb4271f4d5107006cfca48d1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 13 Aug 2024 22:47:03 +0800
Subject: [PATCH 01/12] macports.tcl: fix build_arch choice for ppc

---
 src/macports1.0/macports.tcl | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 4f0513317..8cdb997d6 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1565,7 +1565,9 @@ match macports.conf.default."
     # Default arch to build for
     if {![info exists build_arch]} {
         if {$os_platform eq "darwin"} {
-            if {$os_major >= 20} {
+            if {$os_arch eq "powerpc"} {
+                set build_arch ppc
+            } elseif {$os_major >= 20} {
                 if {$os_arch eq "arm" || (![catch {sysctl sysctl.proc_translated} translated] && $translated)} {
                     set build_arch arm64
                 } else {
@@ -1578,11 +1580,7 @@ match macports.conf.default."
                     set build_arch i386
                 }
             } else {
-                if {$os_arch eq "powerpc"} {
-                    set build_arch ppc
-                } else {
-                    set build_arch i386
-                }
+                set build_arch i386
             }
         } else {
             switch -glob $tcl_platform(machine) {
-- 
2.47.1

