From a33611c5efe10523cfa4c63d173a8ebf317e2b6b Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 10 May 2024 08:08:54 +0800
Subject: [PATCH 02/16] macports.tcl: use libstdc++ on 10.6

---
 src/macports1.0/macports.tcl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 62cbdbe7d..7c5539666 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1650,7 +1650,7 @@ match macports.conf.default."
         }
     }
     if {![info exists cxx_stdlib]} {
-        if {$os_platform eq "darwin" && $os_major >= 10} {
+        if {$os_platform eq "darwin" && $os_major >= 11} {
             set cxx_stdlib libc++
         } elseif {$os_platform eq "darwin"} {
             set cxx_stdlib libstdc++
-- 
2.49.0

