From 805646788d4ff2f73b8ca55472c988bf459c5002 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 10 May 2024 07:50:38 +0800
Subject: [PATCH 03/12] Fix universal archs choices for Darwin

---
 configure | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 74095f44b..368696f32 100755
--- a/configure
+++ b/configure
@@ -7325,10 +7325,16 @@ fi
 			10.1[4-9]*)
 				UNIVERSAL_ARCHS="x86_64"
 			;;
-			10.[0-5]*)
+			10.[0-4]*)
 				UNIVERSAL_ARCHS="i386 ppc"
 			;;
-			10.[6-9]*)
+			10.5*)
+				UNIVERSAL_ARCHS="ppc ppc64"
+			;;
+			10.6*)
+				UNIVERSAL_ARCHS="i386 ppc"
+			;;
+			10.[7-9]*)
 				UNIVERSAL_ARCHS="x86_64 i386"
 			;;
 			*)
-- 
2.47.1

