From 565701382b664622957053a0bc968f736cf20139 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Fri, 10 May 2024 08:16:49 +0800
Subject: [PATCH 05/16] portconfigure.tcl: use libcxx-powerpc if libc++ set

---
 src/port1.0/portconfigure.tcl | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 015871a1d..faad640d9 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -1758,9 +1758,9 @@ proc portconfigure::add_compiler_port_dependencies {compiler} {
                 depends_lib-append "path:lib/libgcc/libgcc_s.1.dylib:libgcc"
             } elseif {${configure.cxx_stdlib} eq "libc++" && ${os.major} < 11} {
                 # libc++ does not exist on these systems
-                ui_debug "Adding depends_lib libcxx"
-                depends_lib-delete "port:libcxx"
-                depends_lib-append "port:libcxx"
+                ui_debug "Adding depends_lib libcxx-powerpc"
+                depends_lib-delete "port:libcxx-powerpc"
+                depends_lib-append "port:libcxx-powerpc"
             }
             if {${compiler.openmp_version} ne ""} {
                 ui_debug "Adding depends_lib port:libomp"
-- 
2.49.0

