From d439686b8b033ae258ace6d41af1fcf23e1acf39 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 31 Oct 2024 09:37:24 +0800
Subject: [PATCH 08/12] Default portimage mode set to archive

---
 src/macports1.0/macports.tcl | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 06916de5c..798f4eb9e 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -1442,13 +1442,8 @@ Please edit sources.conf and change '$url' to '[string range $url 0 26]macports/
         unset portimage_mode
     }
     if {![info exists portimage_mode]} {
-        # Using an extracted directory is usually only a good idea if
-        # the filesystem supports COW clones.
-        if {![catch {fs_clone_capable [file join $portdbpath software]} result] && $result} {
-            set portimage_mode directory
-        } else {
-            set portimage_mode archive
-        }
+        # Default to archive mode
+        set portimage_mode archive
     }
     set portimage::keep_imagedir [expr {$portimage_mode ne "archive"}]
     set portimage::keep_archive [expr {$portimage_mode ne "directory"}]
-- 
2.47.1

