From a8b2f67fbd3a6ea7ad66598b76722c8ced9f4c3b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 7 Jan 2025 06:12:57 +0800
Subject: [PATCH 09/16] No silent

---
 Mk/macports.tea.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Mk/macports.tea.mk b/Mk/macports.tea.mk
index ce20a9f31..18c87e9e4 100644
--- a/Mk/macports.tea.mk
+++ b/Mk/macports.tea.mk
@@ -9,7 +9,7 @@ $(SHLIB_NAME): ${OBJS}
 	${SHLIB_LD} ${OBJS} -o ${SHLIB_NAME} ${TCL_STUB_LIB_SPEC} ${SHLIB_LDFLAGS} ${LIBS}
 
 pkgIndex.tcl: $(SHLIB_NAME)
-	$(SILENT) ../pkg_mkindex.sh . || ( rm -rf $@ && exit 1 )
+	../pkg_mkindex.sh . || ( rm -rf $@ && exit 1 )
 
 clean::
 	rm -f ${OBJS} ${SHLIB_NAME} so_locations pkgIndex.tcl
-- 
2.49.0

