From e0c891e8723e3878748e84e2d57958314ed483b1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 7 Jan 2025 06:36:09 +0800
Subject: [PATCH 12/16] archive_sites: add archives for 10.5.8 ppc64

---
 doc/archive_sites.conf | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/doc/archive_sites.conf b/doc/archive_sites.conf
index 8cfdfa156..11246a62b 100644
--- a/doc/archive_sites.conf
+++ b/doc/archive_sites.conf
@@ -50,9 +50,14 @@
 #prefix              	/opt/mysource
 #applications_dir    	/opt/mysource/Applications
 
+# Pre-built ports for 10.6.8 ppc:
 name                	MacOS-PowerPC
 urls                	http://macos-powerpc.org/packages/
 
+# Pre-built ports for 10.5.8 ppc64:
+name                	MacOS-PowerPC ppc64
+urls                	http://macos-powerpc.org/packages_ppc64/
+
 # To disable the default archive source, uncomment the following dummy
 # entry; it will shadow the one from the ports tree.
 name                	macports_archives
