From 39a70580e6e15c3ef702558d531d4f7fc42fcbf8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 18 Jan 2025 08:37:07 +0800
Subject: [PATCH] Separate sources for ports

---
 doc/sources.conf | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git doc/sources.conf doc/sources.conf
index 41b50d2c9..67c9c44d2 100644
--- a/doc/sources.conf
+++ b/doc/sources.conf
@@ -27,7 +27,11 @@
 # sites, etc.), the primary MacPorts source must always be tagged
 # "[default]", even if switched from the default "rsync://" URL.
 
-rsync://macos-powerpc.org/macos-powerpc/PPCPorts/ports.tar [default]
+# Dedicated repo for R ports:
+# rsync://macos-powerpc.org/macos-powerpc/R/R-ports.tar
 
-# If you want to 
-# rsync://rsync.macports.org/macports/release/tarballs/ports.tar.gz [default]
+# MacOS PowerPC ports:
+rsync://macos-powerpc.org/macos-powerpc/PPCPorts/powerpc-ports.tar
+
+# MacPorts upstream: 
+rsync://rsync.macports.org/macports/release/tarballs/ports.tar.gz [default]
