From f3e81b1b185821d6e3d6042b4aa3cfcafeb30833 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 22 Jan 2025 20:58:55 +0800
Subject: [PATCH 15/16] Support rsync sources in PG search

---
 src/macports1.0/macports.tcl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index ffef884c5..ba774c12d 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -2405,6 +2405,8 @@ proc macports::getlocalporttreelist {} {
     foreach source $sources {
         if {[macports::getprotocol $source] eq "file"} {
             lappend sourcetreelist [string range [lindex ${source} 0] 7 end]
+        } elseif {[macports::getprotocol $source] eq "rsync"} {
+            lappend sourcetreelist [getsourcepath $source]
         }
     }
     return ${sourcetreelist}
-- 
2.49.0

