From cd174d4c557609ae9b20874b0c5b550684d58d85 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 22 Jan 2025 20:58:55 +0800
Subject: [PATCH] Support rsync sources in PG search

---
 src/macports1.0/macports.tcl | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index 798f4eb9e..a8447dbb3 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -2347,6 +2347,8 @@ proc macports::getlocalporttreelist {} {
     foreach source $sources {
         if {[macports::getprotocol $source] eq "file"} {
             lappend sourcetreelist [string range [lindex ${source} 0] 7 end]
+        } elseif {[macports::getprotocol $source] eq "rsync"} {
+            lappend sourcetreelist [getsourcepath $source]
         }
     }
     return ${sourcetreelist}
