From 8eb571b4fdd503a665744ea2ab1bd9e98cdbb697 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 14 Jul 2025 22:12:45 +0800
Subject: [PATCH] portpkg.tcl: minor fix

---
 src/package1.0/portpkg.tcl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/package1.0/portpkg.tcl b/src/package1.0/portpkg.tcl
index e7c2c1baf..34fef717d 100644
--- a/src/package1.0/portpkg.tcl
+++ b/src/package1.0/portpkg.tcl
@@ -60,7 +60,7 @@ proc portpkg::pkg_start {args} {
     global packagemaker_path xcodeversion porturl \
            package.resources package.scripts package.flat \
            subport version revision description long_description \
-           homepage workpath
+           homepage workpath os.arch
     variable packagemaker
     variable pkgbuild
     variable language
