From e66cefb5353e2051414785c6f09be2189166d2f7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 3 May 2025 00:14:42 +0800
Subject: [PATCH 16/16] portconfigure: drop clangs

---
 src/port1.0/portconfigure.tcl | 56 +----------------------------------
 1 file changed, 1 insertion(+), 55 deletions(-)

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index faad640d9..81eb21976 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -1224,61 +1224,7 @@ proc portconfigure::get_apple_compilers_os_version {} {
 }
 # utility procedure: get Clang compilers based on os.major
 proc portconfigure::get_clang_compilers {} {
-    global os.major os.platform porturl
-    set compilers [list]
-    set compiler_file [getportresourcepath $porturl "port1.0/compilers/clang_compilers.tcl"]
-    if {[file exists ${compiler_file}]} {
-        source ${compiler_file}
-    } else {
-        ui_debug "clang_compilers.tcl not found in ports tree, using built-in selections"
-
-        if {${os.major} >= 11 || ${os.platform} ne "darwin"} {
-            if {[option compiler.cxx_standard] >= 2014 && ${os.major} >= 22} {
-                # For now limit exposure of clang-17 to ports needing c++14 or newer
-                # and only on macOS13 or newer due to issues like
-                # https://github.com/macports/macports-ports/pull/21051
-                # https://trac.macports.org/ticket/68640
-                lappend compilers macports-clang-17
-            }
-            lappend compilers macports-clang-16 \
-                              macports-clang-15 \
-                              macports-clang-14
-            if {${os.major} < 23 || ${os.platform} ne "darwin"} {
-                # https://trac.macports.org/ticket/68257
-                # Versions of clang older than clang-14 probably have build issues on
-                # macOS14+. Until resolved do not append to fallback list.
-                lappend compilers macports-clang-13 \
-                                  macports-clang-12
-            }
-        }
-
-        if {${os.platform} eq "darwin"} {
-
-            if {${os.major} >= 10} {
-                lappend compilers macports-clang-11
-                if {[option build_arch] ne "arm64"} {
-                    lappend compilers macports-clang-10 macports-clang-9.0
-                    if {${os.major} < 20} {
-                        lappend compilers macports-clang-8.0
-                    }
-                }
-            }
-
-            if {${os.major} < 20} {
-                lappend compilers macports-clang-7.0 \
-                    macports-clang-6.0 \
-                    macports-clang-5.0
-            }
-
-            if {${os.major} < 16} {
-                # The Sierra SDK requires a toolchain that supports class properties
-                lappend compilers macports-clang-3.7 \
-                        compilers macports-clang-3.4
-            }
-
-        }
-    }
-    return ${compilers}
+    return
 }
 # utility procedure: get GCC compilers based on os.major
 proc portconfigure::get_gcc_compilers {} {
-- 
2.49.0

