--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl	2025-11-21 17:37:24
@@ -1236,9 +1236,9 @@
     global os.arch os.major porturl
     set compilers [list]
     set compiler_file [getportresourcepath $porturl "port1.0/compilers/gcc_compilers.tcl"]
-    if {[file exists ${compiler_file}]} {
-        source ${compiler_file}
-    } else {
+#    if {[file exists ${compiler_file}]} {
+#        source ${compiler_file}
+#    } else {
         ui_debug "gcc_compilers.tcl not found in ports tree, using built-in selections"
 
         lappend compilers macports-gcc-14 macports-gcc-13 macports-gcc-12 macports-gcc-11 macports-gcc-10
@@ -1258,7 +1258,7 @@
         if {${os.arch} eq "powerpc"} {
             lappend compilers macports-gcc-powerpc
         }
-    }
+#    }
     return ${compilers}
 }
 # utility procedure: get MPI wrapper for a given compiler
