--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl	2025-11-21 17:37:24
@@ -1244,12 +1244,12 @@
     global os.arch os.major porturl
     set compilers [list]
     set compiler_file [getportresourcepath $porturl "port1.0/compilers/gcc_compilers.tcl"]
-    if {[file exists ${compiler_file}]} {
-        source ${compiler_file}
-    } else {
+#    if {[file exists ${compiler_file}]} {
+#        source ${compiler_file}
+#    } else {
         ui_debug "gcc_compilers.tcl not found in ports tree, using built-in selections"
 
-        lappend compilers macports-gcc-14 macports-gcc-13 macports-gcc-12 macports-gcc-11 macports-gcc-10
+        lappend compilers macports-gcc-16 macports-gcc-14 macports-gcc-13 macports-gcc-12 macports-gcc-11 macports-gcc-10
 
         # GCC 9 and older only on OSX10.10 and older
         if {${os.major} < 15} {
@@ -1266,7 +1266,7 @@
         if {${os.arch} eq "powerpc"} {
             lappend compilers macports-gcc-powerpc
         }
-    }
+#    }
     return ${compilers}
 }
 # utility procedure: get MPI wrapper for a given compiler
