--- src/adw-settings-impl-macos.c	2025-10-10 02:37:48.000000000 +0800
+++ src/adw-settings-impl-macos.c	2025-10-26 04:06:13.000000000 +0800
@@ -8,6 +8,7 @@
 
 #include "adw-settings-impl-private.h"
 
+#include <AvailabilityMacros.h>
 #include <AppKit/AppKit.h>
 
 struct _AdwSettingsImplMacOS
@@ -41,6 +42,7 @@
 static AdwAccentColor
 get_accent_color (void)
 {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
   GdkRGBA rgba;
   NSColor *accentColor = [NSColor.controlAccentColor colorUsingColorSpace:[
                           NSColorSpace sRGBColorSpace]];
@@ -54,6 +56,9 @@
   rgba.alpha = alpha;
 
   return adw_accent_color_nearest_from_rgba (&rgba);
+#else
+  return ADW_ACCENT_COLOR_BLUE;
+#endif
 }
 
 -(void)appDidChangeAccentColor:(NSNotification *)notification
@@ -65,6 +70,7 @@
 static AdwSystemColorScheme
 get_ns_color_scheme (void)
 {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
   NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
   NSString *style = [userDefaults stringForKey:@"AppleInterfaceStyle"];
   BOOL isDark = [style isEqualToString:@"Dark"];
@@ -72,20 +78,27 @@
   return isDark ?
     ADW_SYSTEM_COLOR_SCHEME_PREFER_DARK :
     ADW_SYSTEM_COLOR_SCHEME_DEFAULT;
+#else
+  return ADW_SYSTEM_COLOR_SCHEME_DEFAULT;
+#endif
 }
 
 -(void)appDidChangeTheme:(NSNotification *)notification
 {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
   if (self->impl != NULL)
     adw_settings_impl_set_color_scheme (self->impl, get_ns_color_scheme ());
+#endif
 }
 
 -(void)appDidChangeHighContrast:(NSNotification *)notification
 {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
   if (self->impl != NULL) {
     gboolean high_contrast = [[NSWorkspace sharedWorkspace] accessibilityDisplayShouldIncreaseContrast];
     adw_settings_impl_set_high_contrast (self->impl, high_contrast);
   }
+#endif
 }
 @end
 
@@ -113,6 +126,7 @@
   observer = [[SettingsChangedObserver alloc] initWithSettings:(AdwSettingsImpl *)self];
 
   if (enable_accent_colors) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
     [[NSDistributedNotificationCenter defaultCenter]
       addObserver:observer
         selector:@selector(appDidChangeAccentColor:)
@@ -120,9 +134,13 @@
           object:nil];
 
     [observer appDidChangeAccentColor:nil];
+#else
+    enable_accent_colors = FALSE;
+#endif
   }
 
   if (enable_high_contrast) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
     [[[NSWorkspace sharedWorkspace] notificationCenter]
     addObserver:observer
       selector:@selector(appDidChangeHighContrast:)
@@ -130,9 +148,13 @@
         object:nil];
 
     [observer appDidChangeHighContrast:nil];
+#else
+    enable_high_contrast = FALSE;
+#endif
   }
 
   if (enable_color_scheme) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 101400
     [[NSDistributedNotificationCenter defaultCenter]
     addObserver:observer
       selector:@selector(appDidChangeTheme:)
@@ -140,6 +162,9 @@
         object:nil];
 
     [observer appDidChangeTheme:nil];
+#else
+    enable_color_scheme = FALSE;
+#endif
   }
 
   adw_settings_impl_set_features (ADW_SETTINGS_IMPL (self),
