--- mfbt/Result.h	2019-07-01 17:07:45.000000000 +0800
+++ mfbt/Result.h	2026-08-23 11:01:00.000000000 +0800
@@ -149,12 +149,12 @@
   struct VEbool {
     V v;
     E e;
-    bool ok;
+    uint8_t ok;
   };
   struct EVbool {
     E e;
     V v;
-    bool ok;
+    uint8_t ok;
   };
 
   using Impl = typename Conditional<sizeof(VEbool) <= sizeof(EVbool), VEbool,

--- js/src/frontend/NameAnalysisTypes.h
+++ js/src/frontend/NameAnalysisTypes.h
@@ -128,7 +128,7 @@
   // If the declared name is a binding, whether the binding is closed
   // over. Its value is meaningless if the declared name is not a binding
   // (i.e., a 'var' declared name in a non-var scope).
-  bool closedOver_;
+  uint8_t closedOver_;
 
  public:
   explicit DeclaredNameInfo(DeclarationKind kind, uint32_t pos)

--- js/src/vm/RegExpShared.h
+++ js/src/vm/RegExpShared.h
@@ -107,7 +107,7 @@
   GCPtr<JSAtom*> source;
 
   RegExpFlag flags;
-  bool canStringMatch;
+  uint8_t canStringMatch;
   size_t parenCount;
 
   RegExpCompilation compilationArray[4];

--- js/src/jit/Ion.h	2019-07-01 17:07:41.000000000 +0800
+++ js/src/jit/Ion.h	2026-08-23 11:04:04.000000000 +0800
@@ -41,10 +41,12 @@
 using mozilla::Err;
 using mozilla::Ok;
 
+#ifndef __ppc__
 static_assert(sizeof(AbortReasonOr<Ok>) <= sizeof(uintptr_t),
               "Unexpected size of AbortReasonOr<Ok>");
 static_assert(sizeof(AbortReasonOr<bool>) <= sizeof(uintptr_t),
               "Unexpected size of AbortReasonOr<bool>");
+#endif
 
 // A JIT context is needed to enter into either an JIT method or an instance
 // of a JIT compiler. It points to a temporary allocator and the active
