--- test/x11comp.c.orig	2026-08-17 15:58:17.000000000 -0600
+++ test/x11comp.c	2026-08-17 16:28:16.000000000 -0600
@@ -24,7 +24,14 @@
 #include "config.h"
 
 #include <stdio.h>
-#include <spawn.h>
+#if defined(__APPLE__)
+    #include "AvailabilityMacros.h"
+    #if MAC_OS_X_VERSION_MIN_REQUIRED > 1040
+        #include <spawn.h>
+    #endif
+#else
+    #include <spawn.h>
+#endif
 #include <unistd.h>
 #include <assert.h>
 #include <signal.h>
@@ -37,6 +44,9 @@
 
 X11_TEST(test_basic)
 {
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+    return 0;
+#else
     struct xkb_keymap *keymap;
     xcb_connection_t *conn;
     int32_t device_id;
@@ -114,6 +124,7 @@
     xcb_disconnect(conn);
 err_conn:
     return ret;
+#endif
 }
 
 int main(void) {
