--- test/xvfb-wrapper.c.orig	2026-08-17 16:14:33.000000000 -0600
+++ test/xvfb-wrapper.c	2026-08-17 16:29:02.000000000 -0600
@@ -25,7 +25,14 @@
 #include "config.h"
 
 #include <stdio.h>
-#include <spawn.h>
+#if defined(__APPLE__)
+    #include "AvailabilityMacros.h"
+    #if MAC_OS_X_VERSION_MIN_REQUIRED > 1040
+        #include <spawn.h>
+    #endif
+#else
+    #include <spawn.h>
+#endif
 #include <assert.h>
 #include <signal.h>
 #include <sys/types.h>
@@ -46,6 +53,9 @@
 int
 xvfb_wrapper(int (*test_func)(char* display))
 {
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+    return 0;
+#else
     int ret = 0;
     FILE * display_fd;
     char display_fd_string[32];
@@ -134,6 +144,7 @@
     fclose(display_fd);
 err_display_fd:
     return ret;
+#endif
 }
 
 /* All X11_TEST functions are in the test_func_sec ELF section.
