From 78696f1532c0f46e832cb27251deb8258c97d6ad Mon Sep 17 00:00:00 2001
From: seeyebe <dawsha.upp@gmail.com>
Date: Tue, 26 Aug 2025 20:52:53 +0300
Subject: [PATCH] fix: make install path configurable with PREFIX and DESTDIR
 for portability

---
 Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d3b3972..60af53f 100644
--- Makefile
+++ Makefile
@@ -3,6 +3,10 @@ CXXFLAGS = -std=c++17 -O3 -Wall -Wextra
 TARGET = sip
 SOURCE = sip.cpp
 
+PREFIX ?= /usr/local
+DESTDIR ?=
+BINDIR ?= $(PREFIX)/bin
+
 ifeq ($(OS),Windows_NT)
 	TARGET = sip.exe
 	RM = del
@@ -22,6 +26,7 @@ ifeq ($(OS),Windows_NT)
 endif
 
 install: $(TARGET)
-	cp $(TARGET) /usr/local/bin/
+	mkdir -p $(DESTDIR)$(BINDIR)
+	cp $(TARGET) $(DESTDIR)$(BINDIR)/
 
 .PHONY: all clean install
\ No newline at end of file
