--- wmh/Makefile.in	2007-01-27 04:58:58.000000000 +0800
+++ wmh/Makefile.in	2025-04-23 14:30:51.000000000 +0800
@@ -214,7 +214,7 @@
 	$(am__append_4)
 AM_CPPFLAGS = -DBUILDING_LIBWMH  -DDEBUG_NAMESPACE='"libggiwmh"'
 INCLUDES = -I$(top_srcdir)/include $(extra_includes)
-libggiwmh_la_LIBADD = $(builtins_libs) -lggi -lgg
+libggiwmh_la_LIBADD = $(builtins_libs) -lggi -lgg -lgii
 lib_LTLIBRARIES = libggiwmh.la
 libggiwmh_la_SOURCES = builtins.c init.c stubs.c
 libggiwmh_la_LDFLAGS = $(extra_libraries) -version-info 0:3:0 -no-undefined \
