diff -ruN a/QJsonValue.cpp b/QJsonValue.cpp
--- QJsonValue.cpp	2026-08-14 00:30:30.170702729 +0000
+++ QJsonValue.cpp	2026-08-14 00:30:30.173702804 +0000
@@ -307,6 +307,20 @@
 }
 
 //------------------------------------------------------------------------------
+// Name: operator[] (Qt 5.10)
+//------------------------------------------------------------------------------
+const QJsonValue QJsonValue::operator[](const QString &key) const {
+	return toObject().value(key);
+}
+
+//------------------------------------------------------------------------------
+// Name: operator[] (Qt 5.10)
+//------------------------------------------------------------------------------
+const QJsonValue QJsonValue::operator[](int i) const {
+	return toArray().at(i);
+}
+
+//------------------------------------------------------------------------------
 // Name: toString
 //------------------------------------------------------------------------------
 QString QJsonValue::toString(const QString &defaultValue) const {
diff -ruN a/QJsonValue.h b/QJsonValue.h
--- QJsonValue.h	2026-08-14 00:30:30.170702729 +0000
+++ QJsonValue.h	2026-08-14 00:30:30.173702804 +0000
@@ -93,6 +93,10 @@
 	QString toString(const QString &defaultValue = QString()) const;
 	QVariant toVariant() const;
 
+	// QJsonValue::operator[] (Qt 5.10)
+	const QJsonValue operator[](const QString &key) const;
+	const QJsonValue operator[](int i) const;
+
 public:
 	Type type() const;
 
