--- cpp_world_clock.cpp	2025-10-31 11:33:10.000000000 +0800
+++ cpp_world_clock.cpp	2025-11-15 16:28:59.000000000 +0800
@@ -112,7 +112,7 @@
   // for each time zone from tz_list.csv:
   // Note: much of this input code was based on the examples
   // found at https://en.cppreference.com/w/cpp/string/basic_string/getline .
-  std::ifstream tz_list_file{"../tz_list.csv"};
+  std::ifstream tz_list_file{"@TZDIR@/tz_list.csv"};
   // Retrieving each line within the .csv file:
   int line_count = 0;
   for (std::string tz_list_pair;
