--- GNUmakefile.orig	2023-08-26 05:00:19.000000000 +0800
+++ GNUmakefile	2023-10-18 09:16:30.000000000 +0800
@@ -26,7 +26,7 @@
     include make.inc
 endif
 
-python ?= python3
+python ?= @PYTHON@
 
 force: ;
 
@@ -35,7 +35,7 @@
 
 # Defaults if not given in make.inc. GNU make doesn't have defaults for these.
 RANLIB   ?= ranlib
-prefix   ?= /opt/slate
+prefix   ?= @PREFIX@
 
 # Default LD=ld won't work; use CXX. Can override in make.inc or environment.
 ifeq ($(origin LD),default)
