--- configure.py.orig	2023-08-26 05:00:19.000000000 +0800
+++ configure.py	2023-10-18 09:39:34.000000000 +0800
@@ -1,4 +1,4 @@
-#!/usr/bin/env python3
+#!@PYTHON@
 #
 # Copyright (c) 2017-2022, University of Tennessee. All rights reserved.
 # SPDX-License-Identifier: BSD-3-Clause
@@ -38,7 +38,7 @@
 
 #-------------------------------------------------------------------------------
 def main():
-    config.init( namespace='LAPACK', prefix='/opt/slate' )
+    config.init( namespace='LAPACK', prefix='@PREFIX@' )
     config.prog_cxx()
 
     print_header( 'C++ compiler flags' )
@@ -94,7 +94,7 @@
 
     blaspp = config.get_package(
         'BLAS++',
-        ['../blaspp', './blaspp'],
+        ['@PREFIX@'],
         'include/blas.hh',
         'https://github.com/icl-utk-edu/blaspp',
         'https://github.com/icl-utk-edu/blaspp/tarball/master',
