--- nss/coreconf/config.gypi	2023-07-27 17:15:00.000000000 +0800
+++ nss/coreconf/config.gypi	2026-01-27 12:04:28.000000000 +0800
@@ -362,6 +362,8 @@
         },
         'xcode_settings': {
           'CLANG_CXX_LANGUAGE_STANDARD': 'c++0x',
+          'GCC_CW_ASM_SYNTAX': 'NO',
+          'GCC_ENABLE_PASCAL_STRINGS': 'NO',
           'OTHER_CFLAGS': [
             '-fPIC',
             '-fno-common',
@@ -502,6 +504,9 @@
             'defines': [
               'DARWIN',
             ],
+            'cflags_c': [
+              '-std=c99',
+            ],
             'conditions': [
               [ 'target_arch=="ia32"', {
                 'xcode_settings': {
@@ -513,6 +518,16 @@
                   'ARCHS': ['x86_64'],
                 },
               }],
+              [ 'target_arch=="ppc"', {
+                'xcode_settings': {
+                  'ARCHS': ['ppc'],
+                },
+              }],
+              [ 'target_arch=="ppc64"', {
+                'xcode_settings': {
+                  'ARCHS': ['ppc64'],
+                },
+              }],
               [ 'target_arch=="arm64"', {
                 'xcode_settings': {
                   'ARCHS': ['arm64'],
