--- include/openssl/target.h	2023-11-17 07:31:50.000000000 +0800
+++ include/openssl/target.h	2026-01-27 07:19:02.000000000 +0800
@@ -45,6 +45,12 @@
 #define OPENSSL_RISCV64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 4
 #define OPENSSL_32_BIT
+#elif defined(__ppc64__) || defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#define OPENSSL_PPC64
+#elif defined(__ppc__) || defined(__powerpc__)
+#define OPENSSL_32_BIT
+#define OPENSSL_PPC
 #elif defined(__pnacl__)
 #define OPENSSL_32_BIT
 #define OPENSSL_PNACL

--- CMakeLists.txt	2023-11-17 07:31:50.000000000 +0800
+++ CMakeLists.txt	2026-01-27 08:59:24.000000000 +0800
@@ -16,7 +16,6 @@
 endif()
 
 include(sources.cmake)
-include(cmake/go.cmake)
 include(cmake/paths.cmake)
 include(cmake/perlasm.cmake)
 
@@ -640,22 +629,6 @@
   add_custom_target(fips_specific_tests_if_any)
 endif()
 
-file(STRINGS util/go_tests.txt GO_TESTS)
-set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS
-             util/go_tests.txt)
-
-add_custom_target(
-    run_tests
-    COMMAND ${GO_EXECUTABLE} test ${GO_TESTS}
-    COMMAND ${GO_EXECUTABLE} run util/all_tests.go -build-dir
-            ${CMAKE_CURRENT_BINARY_DIR}
-    COMMAND cd ssl/test/runner &&
-            ${GO_EXECUTABLE} test -shim-path $<TARGET_FILE:bssl_shim>
-              ${HANDSHAKER_ARGS} ${RUNNER_ARGS}
-    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    DEPENDS all_tests bssl_shim handshaker fips_specific_tests_if_any
-    USES_TERMINAL)
-
 install_if_enabled(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 install_if_enabled(EXPORT OpenSSLTargets

--- crypto/CMakeLists.txt	2023-11-17 07:31:50.000000000 +0800
+++ crypto/CMakeLists.txt	2026-01-27 09:07:12.000000000 +0800
@@ -33,37 +33,6 @@
 perlasm(CRYPTO_SOURCES x86_64 cipher_extra/chacha20_poly1305_x86_64 cipher_extra/asm/chacha20_poly1305_x86_64.pl)
 perlasm(CRYPTO_SOURCES x86_64 test/trampoline-x86_64 test/asm/trampoline-x86_64.pl)
 
-add_custom_command(
-  OUTPUT err_data.c
-  COMMAND ${GO_EXECUTABLE} run err_data_generate.go > ${CMAKE_CURRENT_BINARY_DIR}/err_data.c
-  DEPENDS
-  err/err_data_generate.go
-  err/asn1.errordata
-  err/bio.errordata
-  err/bn.errordata
-  err/cipher.errordata
-  err/conf.errordata
-  err/dh.errordata
-  err/digest.errordata
-  err/dsa.errordata
-  err/ecdh.errordata
-  err/ecdsa.errordata
-  err/ec.errordata
-  err/engine.errordata
-  err/evp.errordata
-  err/hkdf.errordata
-  err/obj.errordata
-  err/pem.errordata
-  err/pkcs7.errordata
-  err/pkcs8.errordata
-  err/rsa.errordata
-  err/ssl.errordata
-  err/trust_token.errordata
-  err/x509.errordata
-  err/x509v3.errordata
-  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/err
-)
-
 add_library(
   crypto
 
--- sources.cmake	2023-11-17 07:31:50.000000000 +0800
+++ sources.cmake	2026-01-27 21:03:12.000000000 +0800
@@ -70,10 +70,6 @@
   crypto/siphash/siphash_test.cc
   crypto/spx/spx_test.cc
   crypto/thread_test.cc
-  # TODO(crbug.com/boringssl/542): This should be in TEST_SUPPORT_SOURCES, so
-  # that all tests can use it. But it depends on GetTestData, which is not
-  # currently usable outside of crypto_test.
-  crypto/test/file_test_gtest.cc
   crypto/test/gtest_main.cc
   crypto/trust_token/trust_token_test.cc
   crypto/x509/x509_test.cc
@@ -430,6 +426,9 @@
 
   crypto/test/abi_test.cc
   crypto/test/file_test.cc
+  crypto/test/file_util.cc
+  crypto/test/file_test_gtest.cc
+  crypto/test/test_data.cc
   crypto/test/test_util.cc
   crypto/test/wycheproof_util.cc
 )

