From b9445ba75528d0672b6d5674345fbf4e1ebe043a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 26 Jan 2026 23:30:53 +0800
Subject: [PATCH] Fix Makefile

---
 Makefile.in | 234 +++++++++++++---------------------------------------
 1 file changed, 56 insertions(+), 178 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index daed8c3..46a0df7 100644
--- Makefile.in
+++ Makefile.in
@@ -8,23 +8,20 @@ SHELL := bash
 # MAKEFLAGS += --warn-undefined-variables
 # MAKEFLAGS += --no-builtin-rules
 
-BROTLI_VERSION := 1.0.9
- # In case this is changed, update build-and-test-make.yml as well
-NSS_VERSION := nss-3.92
-NSS_URL := https://ftp.mozilla.org/pub/security/nss/releases/NSS_3_92_RTM/src/nss-3.92-with-nspr-4.35.tar.gz
- # In case this is changed, update build-and-test-make.yml as well
-BORING_SSL_COMMIT := 1b7fdbd9101dedc3e0aa3fcf4ff74eacddb34ecc
-NGHTTP2_VERSION := nghttp2-1.56.0
-NGHTTP2_URL := https://github.com/nghttp2/nghttp2/releases/download/v1.56.0/nghttp2-1.56.0.tar.bz2
-CURL_VERSION := curl-8.1.1
+BROTLI_VERSION ?= 1.0.9
+NSS_VERSION ?= 3.92
+# NSPR_VERSION ?= 4.35
+BORING_SSL_COMMIT ?= 1b7fdbd9101dedc3e0aa3fcf4ff74eacddb34ecc
+NGHTTP2_VERSION ?= 1.56.0
+CURL_VERSION ?= 8.1.1
 
 brotli_install_dir := $(abspath brotli-$(BROTLI_VERSION)/out/installed)
 brotli_static_libs := $(brotli_install_dir)/lib/libbrotlicommon-static.a $(brotli_install_dir)/lib/libbrotlidec-static.a
-nss_install_dir := $(abspath $(NSS_VERSION)/dist/Release)
+nss_install_dir := $(abspath nss-$(NSS_VERSION)/dist/Release)
 nss_static_libs := $(nss_install_dir)/lib/libnss_static.a
 boringssl_install_dir := $(abspath boringssl/build)
 boringssl_static_libs := $(boringssl_install_dir)/lib/libssl.a $(boringssl_install_dir)/lib/libcrypto.a
-nghttp2_install_dir := $(abspath $(NGHTTP2_VERSION)/installed)
+nghttp2_install_dir := $(abspath nghttp2-$(NGHTTP2_VERSION)/installed)
 nghttp2_static_libs := $(nghttp2_install_dir)/lib/libnghttp2.a
 
 # Dependencies needed to compile the Firefox version
@@ -67,15 +64,13 @@ help: ## Show this help message
 .PHONY: help
 .DEFAULT_GOAL := help
 
-firefox-build: $(CURL_VERSION)/.firefox ## Build the Firefox version of curl-impersonate
-	cd $(CURL_VERSION)
-	# Don't pass this Makefile's MAKEFLAGS
-	$(MAKE) MAKEFLAGS=
+firefox-build: curl-$(CURL_VERSION)/.firefox ## Build the Firefox version of curl-impersonate
+	cd curl-$(CURL_VERSION) && $(MAKE) MAKEFLAGS=
 .PHONY: firefox-build
 
 firefox-checkbuild: ## Run basic checks on the built binary
 ifeq ($(host),$(build))
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	# Make sure all needed features were compiled in
 	./src/curl-impersonate-ff -V | grep -q zlib
 	./src/curl-impersonate-ff -V | grep -q brotli
@@ -88,14 +83,11 @@ endif
 .PHONY: firefox-checkbuild
 
 firefox-install: ## Install the Firefox version of curl-impersonate after build
-	cd $(CURL_VERSION)
-	$(MAKE) install-exec MAKEFLAGS=
-	# Wrapper scripts for the Firefox version (e.g. 'curl_ff98')
-	install $(srcdir)/firefox/curl_ff* @bindir@
+	cd curl-$(CURL_VERSION) && $(MAKE) install-exec MAKEFLAGS= && install $(srcdir)/firefox/curl_ff* @bindir@
 .PHONY: firefox-install
 
 firefox-install-strip: ## Like 'firefox-install', but strip binaries for smaller size
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) install-exec MAKEFLAGS=
 	# We could have used 'install-strip' but then the docs would be installed as well.
 	# Instead strip manually.
@@ -105,26 +97,24 @@ firefox-install-strip: ## Like 'firefox-install', but strip binaries for smaller
 .PHONY: firefox-install-strip
 
 firefox-uninstall: ## Uninstall the Firefox version of curl-impersonate after 'make install'
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) uninstall MAKEFLAGS=
 	rm -Rf @bindir@/curl_ff*
 .PHONY: firefox-uninstall
 
 firefox-clean: ## Clean build artifacts of the Firefox version. Use after re-running './configure'
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) clean MAKEFLAGS=
 	rm -f .firefox
 .PHONY: firefox-clean
 
-chrome-build: $(CURL_VERSION)/.chrome ## Build the Chrome version of curl-impersonate
-	cd $(CURL_VERSION)
-	# Don't pass this Makefile's MAKEFLAGS
-	$(MAKE) MAKEFLAGS=
+chrome-build: curl-$(CURL_VERSION)/.chrome ## Build the Chrome version of curl-impersonate
+	cd curl-$(CURL_VERSION) && $(MAKE) MAKEFLAGS=
 .PHONY: chrome-build
 
 chrome-checkbuild: ## Run basic checks on the built binary
 ifeq ($(host),$(build))
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	# Make sure all needed features were compiled in
 	./src/curl-impersonate-chrome -V | grep -q zlib
 	./src/curl-impersonate-chrome -V | grep -q brotli
@@ -137,14 +127,12 @@ endif
 .PHONY: chrome-checkbuild
 
 chrome-install: ## Install the Chrome version of curl-impersonate after build
-	cd $(CURL_VERSION)
-	$(MAKE) install-exec MAKEFLAGS=
-	# Wrapper scripts for the Chrome version (e.g. 'curl_chrome99')
-	install $(srcdir)/chrome/curl_chrome* $(srcdir)/chrome/curl_edge* $(srcdir)/chrome/curl_safari* @bindir@
+	cd curl-$(CURL_VERSION) && $(MAKE) install-exec MAKEFLAGS= && \
+		install $(srcdir)/chrome/curl_chrome* $(srcdir)/chrome/curl_edge* $(srcdir)/chrome/curl_safari* @bindir@
 .PHONY: chrome-install
 
 chrome-install-strip: ## Like 'chrome-install', but strip binaries for smaller size
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) install-exec MAKEFLAGS=
 	# We could have used 'install-strip' but then the docs would be installed as well.
 	# Instead strip manually.
@@ -154,33 +142,28 @@ chrome-install-strip: ## Like 'chrome-install', but strip binaries for smaller s
 .PHONY: chrome-install-strip
 
 chrome-uninstall: ## Uninstall the Chrome version of curl-impersonate after 'make install'
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) uninstall MAKEFLAGS=
 	rm -Rf @bindir@/curl_chrome* @bindir@/curl_edge* @bindir@/curl_safari*
 .PHONY: chrome-uninstall
 
 chrome-clean: ## Clean build artifacts of the Chrome version. Use after re-running './configure'
-	cd $(CURL_VERSION)
+	cd curl-$(CURL_VERSION)
 	$(MAKE) clean MAKEFLAGS=
 	rm -f .chrome
 .PHONY: chrome-clean
 
 clean: ## Remove all build artifacts, including dependencies
 	rm -Rf brotli-$(BROTLI_VERSION).tar.gz brotli-$(BROTLI_VERSION)
-	rm -Rf $(NSS_VERSION).tar.gz $(NSS_VERSION)
-	rm -Rf boringssl.zip boringssl
-	rm -Rf $(NGHTTP2_VERSION).tar.bz2 $(NGHTTP2_VERSION)
-	rm -Rf $(CURL_VERSION).tar.xz $(CURL_VERSION)
-
-brotli-$(BROTLI_VERSION).tar.gz:
-	curl -L "https://github.com/google/brotli/archive/refs/tags/v${BROTLI_VERSION}.tar.gz" \
-		-o "brotli-${BROTLI_VERSION}.tar.gz"
-
-$(brotli_static_libs): brotli-$(BROTLI_VERSION).tar.gz
-	tar xf brotli-$(BROTLI_VERSION).tar.gz
-	cd brotli-$(BROTLI_VERSION)
+	rm -Rf nss-$(NSS_VERSION).tar.gz nss-$(NSS_VERSION)
+	rm -Rf boringssl.tar.gz boringssl
+	rm -Rf nghttp2-$(NGHTTP2_VERSION).tar.gz nghttp2-$(NGHTTP2_VERSION)
+	rm -Rf curl-$(CURL_VERSION).tar.gz curl-$(CURL_VERSION)
+
+$(brotli_static_libs):
+	cd ./brotli-$(BROTLI_VERSION)
 	mkdir -p out
-	cd out
+	cd ./out
 
 	# Convert autoconf style os name to CMake style os name.
 	case $(host_os) in           \
@@ -201,22 +184,17 @@ $(brotli_static_libs): brotli-$(BROTLI_VERSION).tar.gz
 	        -DCMAKE_CXX_COMPILER=$(CXX) \
 	        -DCMAKE_C_COMPILER=$(CC) \
 	        -DCMAKE_SYSTEM_NAME=$$system_name \
-	        -DCMAKE_SYSTEM_PROCESSOR=$(host_cpu) \
-	        ..
+	        -DCMAKE_SYSTEM_PROCESSOR=$(host_cpu)
 
 	@cmake@ --build . --config Release --target install
 
 
-$(NSS_VERSION).tar.gz:
-	curl -L -o $(NSS_VERSION).tar.gz $(NSS_URL)
-
-$(nss_static_libs): $(NSS_VERSION).tar.gz
-	tar xf $(NSS_VERSION).tar.gz
+$(nss_static_libs):
 
 ifeq ($(host),$(build))
 	# Native build, use NSS' build script.
-	cd $(NSS_VERSION)/nss
-	./build.sh -o --disable-tests --static --python=python3
+	cd nss-$(NSS_VERSION)/nss
+	./build.sh -o --disable-tests --static --python=@python@
 else
 	# We are cross compiling.
 	# Cross compiling NSS is not supported by its build script and is poorly
@@ -233,7 +211,7 @@ else
 	esac
 
 	# Cross-compile nspr separately
-	cd $(NSS_VERSION)/nspr
+	cd nss-$(NSS_VERSION)/nspr
 	./configure --prefix=$(nss_install_dir) \
 	            --disable-debug --enable-optimize \
 	            --target=$(host_alias) \
@@ -244,7 +222,7 @@ else
 	# Now we can run ./build.sh with the already built nspr
 	cd ../nss
 	CC=$(CC) CXX=$(CXX) CCC=$(CXX) \
-	   ./build.sh -o --disable-tests --static --python=python3 \
+	   ./build.sh -o --disable-tests --static --python=@python@ \
 	   --with-nspr=$(nss_install_dir)/include/nspr:$(nss_install_dir)/lib \
 	   --target=$(host_cpu) \
 	   -Duse_system_zlib=0 \
@@ -255,19 +233,13 @@ endif
 	rm -Rf $(nss_install_dir)/lib/*.dylib
 
 
-boringssl.zip:
-	curl -L https://github.com/google/boringssl/archive/$(BORING_SSL_COMMIT).zip \
-		-o boringssl.zip
-
 # Patch boringssl and use a dummy '.patched' file to mark it patched
 boringssl/.patched: $(srcdir)/chrome/patches/boringssl-*.patch
-	unzip -q -o boringssl.zip
-	mv boringssl-$(BORING_SSL_COMMIT) boringssl
 	cd boringssl/
 	for p in $^; do patch -p1 < $$p; done
 	touch .patched
 
-$(boringssl_static_libs): boringssl.zip boringssl/.patched
+$(boringssl_static_libs): boringssl/.patched
 	mkdir -p $(boringssl_install_dir)
 	cd $(boringssl_install_dir)
 
@@ -295,8 +267,7 @@ $(boringssl_static_libs): boringssl.zip boringssl/.patched
 			-DCMAKE_C_COMPILER=$(CC) \
 			-DCMAKE_SYSTEM_NAME=$$system_name \
 			-DCMAKE_SYSTEM_PROCESSOR=$(host_cpu) \
-			-GNinja \
-			..
+			-GNinja
 	@ninja@
 	# Fix the directory structure so that curl can compile against it.
 	# See https://everything.curl.dev/source/build/tls/boringssl
@@ -306,12 +277,8 @@ $(boringssl_static_libs): boringssl.zip boringssl/.patched
 	cp -Rf ../include .
 
 
-$(NGHTTP2_VERSION).tar.bz2:
-	curl -L $(NGHTTP2_URL) -o $(NGHTTP2_VERSION).tar.bz2
-
-$(nghttp2_static_libs): $(NGHTTP2_VERSION).tar.bz2
-	tar -xf $(NGHTTP2_VERSION).tar.bz2
-	cd $(NGHTTP2_VERSION)
+$(nghttp2_static_libs):
+	cd nghttp2-$(NGHTTP2_VERSION)
 
 	# Set up the configure flags to nghttp2.
 	# If the user provided the --host flag to our configure script
@@ -329,115 +296,26 @@ $(nghttp2_static_libs): $(NGHTTP2_VERSION).tar.bz2
 	$(MAKE) MAKEFLAGS=
 	$(MAKE) install MAKEFLAGS=
 
-$(CURL_VERSION).tar.xz:
-	curl -L "https://curl.se/download/$(CURL_VERSION).tar.xz" \
-		-o "$(CURL_VERSION).tar.xz"
 
 # Apply the "Firefox version" patches and mark using a dummy file
-$(CURL_VERSION)/.patched-ff: $(srcdir)/firefox/patches/curl-*.patch
-	rm -Rf $(CURL_VERSION)
-	tar -xf $(CURL_VERSION).tar.xz
-	cd $(CURL_VERSION)
-	for p in $^; do patch -p1 < $$p; done
-	# Re-generate the configure script
-	autoreconf -fi
-	touch .patched-ff
-	rm -f .patched-chrome
+curl-$(CURL_VERSION)/.patched-ff: $(srcdir)/firefox/patches/curl-*.patch
+	cd ./curl-$(CURL_VERSION) && autoreconf -fi && touch .patched-ff && rm -f .patched-chrome
 
 # Apply the "Chorme version" patches and mark using a dummy file
-$(CURL_VERSION)/.patched-chrome: $(srcdir)/chrome/patches/curl-*.patch
-	rm -Rf $(CURL_VERSION)
-	tar -xf $(CURL_VERSION).tar.xz
-	cd $(CURL_VERSION)
-	for p in $^; do patch -p1 < $$p; done
-	# Re-generate the configure script
-	autoreconf -fi
-	touch .patched-chrome
-	rm -f .patched-ff
+curl-$(CURL_VERSION)/.patched-chrome: $(srcdir)/chrome/patches/curl-*.patch
+	cd ./curl-$(CURL_VERSION) && autoreconf -fi && touch .patched-chrome && rm -f .patched-ff
 
 # This is a small hack that flags that curl was patched and configured in the "firefox" version
-$(CURL_VERSION)/.firefox: $(firefox_libs) $(CURL_VERSION).tar.xz $(CURL_VERSION)/.patched-ff
-	cd $(CURL_VERSION)
-
-	# Set up the configure flags to curl.
-	# If the user provided the --host flag to our configure script
-	# (for cross compilation), then pass it on to curl.
-	{ \
-	  config_flags="$(CURL_CONFIG_FLAGS)"; \
-	  config_flags+=" --prefix=@prefix@"; \
-	  config_flags+=" --with-nghttp2=$(nghttp2_install_dir)"; \
-	  config_flags+=" --with-brotli=$(brotli_install_dir)"; \
-	  config_flags+=" --with-nss=$(nss_install_dir) --with-nss-deprecated"; \
-	  config_flags+=" --enable-websockets"; \
-	  config_flags+=" USE_CURL_SSLKEYLOGFILE=true"; \
-	  if test "$(static_build)" = "yes"; then \
-	    config_flags+=" --enable-static --disable-shared"; \
-	  fi; \
-	  if test -n "$(host_alias)"; then \
-	    config_flags+=" --host=$(host_alias)"; \
-	  fi; \
-	  if test -n "$(with_zlib)"; then \
-	    config_flags+=" --with-zlib=$(with_zlib)"; \
-	  else \
-	    config_flags+=" --with-zlib"; \
-	  fi; \
-	  if test -n "$(with_libnssckbi)"; then \
-		config_flags+=" --with-libnssckbi=$(with_libnssckbi)"; \
-	  fi; \
-	  add_cflags="-I$(nss_install_dir)/../public/nss"; \
-	  add_cflags+=" -I$(nss_install_dir)/include/nspr"; \
-	}
-
-	echo "Configuring curl with: $$config_flags"
-
-	./configure $$config_flags CFLAGS="$$add_cflags"
-	# Remove possible leftovers from a previous compilation
-	$(MAKE) clean MAKEFLAGS=
-	touch .firefox
-	# Remove the Chrome flag if it exists
-	rm -f .chrome
+curl-$(CURL_VERSION)/.firefox: curl-$(CURL_VERSION)/.patched-ff
+	cd curl-$(CURL_VERSION) && CC=@CC@ ./configure $(CURL_CONFIG_FLAGS) --prefix=@prefix@ --with-nghttp2=$(nghttp2_install_dir) --with-brotli=$(brotli_install_dir) \
+		--with-nss=$(nss_install_dir) --with-nss-deprecated --enable-websockets --disable-ldap USE_CURL_SSLKEYLOGFILE=true \
+		--enable-static --disable-shared --with-zlib=$(with_zlib) \
+		CFLAGS="-I$(nss_install_dir)/../public/nss -I$(nss_install_dir)/include/nspr -I__PREFIX__/include" LDFLAGS="-L$(nss_install_dir)/lib -L$(nghttp2_install_dir)/lib -L__PREFIX__/lib" \
+		&& touch .firefox && rm -f .chrome
 
 # This is a small hack that flags that curl was patched and configured in the "chrome" version
-$(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.xz $(CURL_VERSION)/.patched-chrome
-	cd $(CURL_VERSION)
-
-	# Set up the configure flags to curl.
-	# If the user provided the --host flag to our configure script
-	# (for cross compilation), then pass it on to curl.
-	{ \
-	  config_flags="$(CURL_CONFIG_FLAGS)"; \
-	  config_flags="$$config_flags --prefix=@prefix@"; \
-	  config_flags="$$config_flags --with-nghttp2=$(nghttp2_install_dir)"; \
-	  config_flags="$$config_flags --with-brotli=$(brotli_install_dir)"; \
-	  config_flags="$$config_flags --with-openssl=$(boringssl_install_dir)"; \
-	  config_flags="$$config_flags --enable-websockets"; \
-	  config_flags="$$config_flags USE_CURL_SSLKEYLOGFILE=true"; \
-	  if test "$(static_build)" = "yes"; then \
-	    config_flags="$$config_flags --enable-static --disable-shared";
-	  fi; \
-	  if test -n "$(host_alias)"; then \
-	    config_flags="$$config_flags --host=$(host_alias)"; \
-	  fi; \
-	  if test -n "$(with_zlib)"; then \
-	    config_flags="$$config_flags --with-zlib=$(with_zlib)"; \
-	  else \
-	    config_flags+=" --with-zlib"; \
-	  fi; \
-	  if test -n "$(with_ca_bundle)"; then \
-		config_flags="$$config_flags --with-ca-bundle=$(with_ca_bundle)"; \
-	  fi; \
-	  if test -n "$(with_ca_path)"; then \
-		config_flags="$$config_flags --with-ca-path=$(with_ca_path)"; \
-	  fi; \
-	  add_libs="-pthread"; \
-	}
-
-	echo "Configuring curl with: $$config_flags"
-
-	./configure $$config_flags LIBS="$$add_libs"
-
-	# Remove possible leftovers from a previous compilation
-	$(MAKE) clean MAKEFLAGS=
-	touch .chrome
-	# Remove the Firefox flag if it exists
-	rm -f .firefox
+curl-$(CURL_VERSION)/.chrome: curl-$(CURL_VERSION)/.patched-chrome
+	cd curl-$(CURL_VERSION) && CC=@CC@ ./configure $(CURL_CONFIG_FLAGS) --prefix=@prefix@ --with-nghttp2=$(nghttp2_install_dir) --with-brotli=$(brotli_install_dir) \
+		--with-openssl=$(boringssl_install_dir) --enable-websockets --enable-static --disable-shared --disable-ldap USE_CURL_SSLKEYLOGFILE=true \
+		--with-zlib=$(with_zlib) --with-ca-bundle=$(with_ca_bundle) \
+		LIBS="-pthread" && touch .chrome && rm -f .firefox
