From f2cb8cb8143e70abc53858c94f835baded9a950a Mon Sep 17 00:00:00 2001
From: xfangfang <2553041586@qq.com>
Date: Tue, 18 Jul 2023 23:01:41 +0800
Subject: [PATCH 2/5] Add arpa/inet.h for switch

---
 mongoose.h | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git mongoose.h mongoose.h
index d3b8128b..1c5e14f6 100644
--- mongoose.h
+++ mongoose.h
@@ -247,24 +247,25 @@ static inline int mg_mkdir(const char *path, mode_t mode) {
 #ifndef _POSIX_TIMERS
 #define _POSIX_TIMERS
 #endif
+#include <arpa/inet.h>
 #include <ctype.h>
+#include <dirent.h>
 #include <errno.h>
+#include <fcntl.h>
+#include <netinet/in.h>
+#include <netinet/tcp.h>
+#include <poll.h>
 #include <stdarg.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/socket.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/types.h>
-#include <sys/socket.h>
 #include <time.h>
 #include <unistd.h>
-#include <fcntl.h>
-#include <netinet/in.h>
-#include <netinet/tcp.h>
-#include <poll.h>
-#include <dirent.h>
 
 #define MG_PATH_MAX NAME_MAX
 #define MG_ENABLE_POLL 1
-- 
2.51.0

