--- lib/krb5/krb/x-deltat.y	2025-08-21 03:44:32.000000000 +0800
+++ lib/krb5/krb/x-deltat.y	2025-10-11 07:44:20.000000000 +0800
@@ -44,8 +44,10 @@
 #ifdef __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wuninitialized"
+#ifndef __clang__
 #pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 #endif
+#endif
 
 #include "k5-int.h"
 #include <ctype.h>

--- lib/krb5/krb/deltat.c	2025-08-21 03:44:32.000000000 +0800
+++ lib/krb5/krb/deltat.c	2025-10-11 07:46:09.000000000 +0800
@@ -77,8 +77,10 @@
 #ifdef __GNUC__
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wuninitialized"
+#ifndef __clang__
 #pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 #endif
+#endif
 
 #include "k5-int.h"
 #include <ctype.h>
