--- configure	2025-02-10 16:17:38.000000000 +0800
+++ configure	2025-02-10 23:57:08.000000000 +0800
@@ -15,7 +15,7 @@
 export MACOSX_DEPLOYMENT_TARGET=`echo $CC | sed -En 's/.*-version-min=([0-9][0-9.]*).*/\1/p'`
 fi
 
-# Detect -latomic linker flag for ARM architectures (Raspberry Pi etc.)
+# Detect -latomic linker flag for 32-bit architectures (Raspberry Pi etc.)
 echo "#include <stdint.h>
 uint64_t v;
 int main() {
@@ -30,27 +30,16 @@
 # Force build bundled libs
 if [ -z "$WATCHER_LIBS" ]; then
 
-# Find MbedTLS and compile if necessary
 if [ "$INCLUDE_DIR" ] || [ "$LIB_DIR" ]
 then
   PKG_CFLAGS="-I$INCLUDE_DIR $PKG_CFLAGS"
   PKG_LIBS="-L$LIB_DIR $PKG_LIBS"
   echo "Found INCLUDE_DIR $INCLUDE_DIR"
   echo "Found LIB_DIR $LIB_DIR"
-elif [ -d "/usr/local/include/libfswatch" ]
+elif [ -d "@PREFIX@/include/libfswatch" ]
 then
-  PKG_CFLAGS="-I/usr/local/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/local/lib $PKG_LIBS"
-  echo "Found 'libfswatch' $PKG_CFLAGS"
-elif [ -d "/usr/include/libfswatch" ]
-then
-  PKG_CFLAGS="-I/usr/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/lib $PKG_LIBS"
-  echo "Found 'libfswatch' $PKG_CFLAGS"
-elif [ -d "/usr/local/opt/libfswatch" ]
-then
-  PKG_CFLAGS="-I/usr/local/opt/fswatch/include $PKG_CFLAGS"
-  PKG_LIBS="-L/usr/local/opt/fswatch/lib $PKG_LIBS"
+  PKG_CFLAGS="-I@PREFIX@/include $PKG_CFLAGS"
+  PKG_LIBS="-L@PREFIX@/lib $PKG_LIBS"
   echo "Found 'libfswatch' $PKG_CFLAGS"
 else
   echo "Existing 'libfswatch' not found"
