From dc4bf1742571a02249ea3bc6e7a4246f66060088 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 30 Apr 2025 10:08:12 +0800
Subject: [PATCH] net.c: fix include order for Apple

---
 src/contrib/net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/contrib/net.c src/contrib/net.c
index a9fdd3961..58b59e44d 100644
--- src/contrib/net.c
+++ src/contrib/net.c
@@ -6,12 +6,12 @@
 #include <assert.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <sys/socket.h>
 #include <sys/types.h>   // OpenBSD
 #include <netinet/tcp.h> // TCP_FASTOPEN
 #include <netinet/in.h>
 #include <poll.h>
 #include <stdbool.h>
-#include <sys/socket.h>
 #include <sys/uio.h>
 #include <unistd.h>
 
