From b2f959bc0ad12772b146ec486d74c0c9bab2dc63 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 30 Apr 2025 11:18:11 +0800
Subject: [PATCH] netio.c: fix for legacy macOS

---
 src/utils/common/netio.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git src/utils/common/netio.c src/utils/common/netio.c
index 209f92f1b..bd8cb8897 100644
--- src/utils/common/netio.c
+++ src/utils/common/netio.c
@@ -17,6 +17,10 @@
 #include <sys/uio.h>
 #endif
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 #include "utils/common/netio.h"
 #include "utils/common/msg.h"
 #include "utils/common/tls.h"
@@ -347,7 +351,7 @@ static int fastopen_connect(int sockfd, const struct addrinfo *srv)
 #if defined( __FreeBSD__)
 	const int enable = 1;
 	return setsockopt(sockfd, IPPROTO_TCP, TCP_FASTOPEN, &enable, sizeof(enable));
-#elif defined(__APPLE__)
+#elif defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 101100)
 	// connection is performed lazily when first data are sent
 	struct sa_endpoints ep = {0};
 	ep.sae_dstaddr = srv->ai_addr;
