From 56aff45984caa81d528410495e29c2608f8ba29e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 30 Apr 2025 11:33:40 +0800
Subject: [PATCH] server.c: fallback for missing IPV6_RECVPKTINFO

---
 src/knot/server/server.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/knot/server/server.c src/knot/server/server.c
index c446916cf..89f6fe7b4 100644
--- src/knot/server/server.c
+++ src/knot/server/server.c
@@ -192,7 +192,11 @@ static bool enable_pktinfo(int sock, int family)
 		break;
 	case AF_INET6:
 		level = IPPROTO_IPV6;
+#if defined(IPV6_RECVPKTINFO)
 		option = IPV6_RECVPKTINFO; /* Multiplatform */
+#else
+		option = IPV6_PKTINFO;
+#endif
 		break;
 	default:
 		assert(0);
