--- src/tsdl_image.ml.orig	2023-08-05 04:17:07.000000000 +0800
+++ src/tsdl_image.ml	2025-10-18 13:53:58.000000000 +0800
@@ -50,7 +50,7 @@
     let env = try Sys.getenv "LIBSDL2_PATH" with Not_found -> "" in
     let filename, path =
       match Build_config.system with
-      | "macosx" -> ("libSDL2_image-2.0.0.dylib", [ "/opt/homebrew/lib/" ])
+      | "macosx" -> ("libSDL2_image-2.0.0.dylib", [ "@PREFIX@/lib/" ])
       | "win32" | "win64" ->
           (* On native Windows DLLs are loaded from the PATH *)
           ("SDL2_image.dll", [ "" ])
