--- folly/json/dynamic.h	2025-01-13 17:54:29.000000000 +0800
+++ folly/json/dynamic.h	2025-01-29 06:11:31.000000000 +0800
@@ -285,6 +285,14 @@
   static dynamic array_range(Range&& range) {
     return dynamic(array_range_construct, std::forward<Range>(range));
   }
+  /**
+   * Create a dynamic that is an array of the values from the supplied
+   * iterator range. Deprecated constructor since it is too easily confused.
+   */
+  template <class Iterator>
+  [[deprecated("use array_range_construct ctor or array_range factory")]]
+  dynamic(Iterator first, Iterator last)
+      : dynamic(array_range_construct, first, last) {}
 
   dynamic(dynamic const&);
   dynamic(dynamic&&) noexcept;
