From 2ebe465822b50fd56af261152004ad6651fdc2ad Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 23 Mar 2025 22:17:24 +0800
Subject: [PATCH] Subprocess: fix constructor for macOS

---
 folly/Subprocess.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git folly/Subprocess.cpp folly/Subprocess.cpp
index af9fc711a..0b922da45 100644
--- folly/Subprocess.cpp
+++ folly/Subprocess.cpp
@@ -73,6 +73,12 @@
   __attribute__((noinline, no_sanitize("address", "undefined", "thread")))
 #endif
 
+#if defined(__APPLE__) && !defined(__clang__)
+#define FOLLY_DETAIL_SUBPROCESS_CONSTRUCTOR __attribute__((constructor))
+#else
+#define FOLLY_DETAIL_SUBPROCESS_CONSTRUCTOR __attribute__((constructor(101)))
+#endif
+
 constexpr int kExecFailure = 127;
 constexpr int kChildFailure = 126;
 
@@ -175,7 +181,7 @@ struct subprocess_libc {
 
 FOLLY_DETAIL_SUBPROCESS_LIBC_X(FOLLY_DETAIL_SUBPROCESS_LIBC_FIELD_DEFN)
 
-__attribute__((constructor(101))) static void subprocess_libc_init() {
+FOLLY_DETAIL_SUBPROCESS_CONSTRUCTOR static void subprocess_libc_init() {
   auto handle = !kIsSanitize
       ? nullptr
       : ::dlopen(subprocess_libc_soname, RTLD_LAZY | RTLD_LOCAL | RTLD_NOLOAD);
