--- folly/io/IOBuf.cpp	2025-04-28 19:42:41.000000000 +0800
+++ folly/io/IOBuf.cpp	2025-04-29 21:57:34.000000000 +0800
@@ -137,7 +137,9 @@
   // This function exists only to have a scope to do the static_asserts().
   static void checkInvariants() {
     // Make sure jemalloc allocates from the 64-byte class.
+#ifndef __ppc__
     static_assert(sizeof(HeapPrefix) == 8);
+#endif
     static_assert(sizeof(IOBuf) <= 56);
     static_assert(sizeof(HeapStorage) <= 64);
   }
