From 3383709c6385bba21be63307c1dab7ee93b21992 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 16 Oct 2025 06:14:17 +0800
Subject: [PATCH] CMakeLists.txt: use external gtest, fallback to fetching the
 source

---
 CMakeLists.txt | 17 +++++++++++------
 1 file changed, 11 insertions(+), 6 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index ef0d46e..57625fc 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -96,12 +96,17 @@ endif()
 if (PHMAP_BUILD_TESTS)
 
     if (NOT PHMAP_GTEST_LIBS)
-        include(cmake/DownloadGTest.cmake)
-
-        check_target(gtest)
-        check_target(gtest_main)
-        check_target(gmock)
-        set(PHMAP_GTEST_LIBS gmock_main)
+        find_package(GTest CONFIG)
+        if (GTest_FOUND)
+            set(PHMAP_GTEST_LIBS GTest::gmock_main)
+        else()
+            include(cmake/DownloadGTest.cmake)
+
+            check_target(gtest)
+            check_target(gtest_main)
+            check_target(gmock)
+            set(PHMAP_GTEST_LIBS gmock_main)
+        endif()
     endif()
 
     enable_testing()
-- 
2.51.0

