--- Makefile	2026-06-02 02:50:07.000000000 +0800
+++ Makefile	2026-06-03 02:01:25.000000000 +0800
@@ -4,7 +4,7 @@
 OBJDIR ?= .obj
 OUTDIR ?= target
 # These paths are quoted in recipes.
-PREFIX ?= /usr/local
+PREFIX ?= @PREFIX@
 MANPREFIX ?= $(PREFIX)/share/man
 MANPREFIX1 ?= $(MANPREFIX)/man1
 MANPREFIX5 ?= $(MANPREFIX)/man5
@@ -225,9 +225,9 @@
 $(OUTDIR_LIBEXEC)/urldec: $(OUTDIR_LIBEXEC)/urlenc
 	(cd "$(OUTDIR_LIBEXEC)" && ln -sf urlenc urldec)
 
-# Do not add FLAGS here, because that breaks cross-compilation.
+# Need to specify compiler here, otherwise it breaks compilation.
 $(OBJDIR)/chac: src/chac.nim lib/monoucha0/monoucha/* lib/monoucha0/monoucha/qjs/*
-	$(NIMC) -o:$@ $<
+	$(NIMC) --gcc.path=@PREFIX@/bin --gcc.exe=@CC@ --gcc.linkerexe=@CC@ -o:$@ $<
 
 $(OUTDIR_LIBEXEC)/%.jsb: src/%.js $(OBJDIR)/chac
 	$(OBJDIR)/chac $(chac_flags) $< $@
