From 46b98d2c3f2a8d5c7e7998a7295d74de615518a8 Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:30:04 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_typerep_conv.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ppx_typerep_conv.ml b/src/ppx_typerep_conv.ml
index ae2edd1..9796263 100644
--- src/ppx_typerep_conv.ml
+++ src/ppx_typerep_conv.ml
@@ -505,7 +505,7 @@
           in
           List.mapi ~f:mapi variants
         in
-        pexp_function ~loc match_cases
+        pexp_function_cases ~loc match_cases
     end
   end
 
