--- plugins/in_node_exporter_metrics/ne_powersupply_darwin.c	2026-08-15 11:52:07.000000000 +0800
+++ plugins/in_node_exporter_metrics/ne_powersupply_darwin.c	2026-08-17 22:30:55.000000000 +0800
@@ -21,6 +21,16 @@
 #include <IOKit/ps/IOPowerSources.h>
 #include <IOKit/ps/IOPSKeys.h>
 
+#ifndef kIOPSTemperatureKey
+#define kIOPSTemperatureKey "Temperature"
+#endif
+#ifndef kIOPSNominalCapacityKey
+#define kIOPSNominalCapacityKey "Nominal Capacity"
+#endif
+#ifndef kIOPSInternalFailureKey
+#define kIOPSInternalFailureKey "Internal Failure"
+#endif
+
 #include "ne.h"
 
 static int cf_number_to_double(CFTypeRef value, double *out)
