--- compiler_opt.py	2024-10-01 02:09:41.000000000 +0800
+++ compiler_opt.py	2025-04-20 12:48:33.000000000 +0800
@@ -333,6 +333,8 @@
     clang = compiler_is_clang()
     gcc = compiler_is_gcc()
 
+    extra_cc_options = ['-isystem@PREFIX@/include/LegacySupport', '-Wl,-lMacportsLegacySupport']
+
     if has_stdint_h():
         extra_macros.append(("HAVE_STDINT_H", None))
 
@@ -359,10 +361,7 @@
         extra_macros.append(("HAVE_INTRIN_H", None))
 
     # Platform-specific call for getting a block of aligned memory
-    if compiler_has_posix_memalign():
-        extra_macros.append(("HAVE_POSIX_MEMALIGN", None))
-    elif compiler_has_memalign():
-        extra_macros.append(("HAVE_MEMALIGN", None))
+    extra_macros.append(("HAVE_POSIX_MEMALIGN", None))
 
     # SSE2
     sse2_result = compiler_supports_sse2()
