From 86b59ac2ca3366e417344e5c1fa5b0489007cf94 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 5 Jan 2024 19:36:54 +0800
Subject: [PATCH] ppc.h: use correct cacheline size for macOS
--- include/urcu/arch/ppc.h.orig	2024-11-20 09:54:04
+++ include/urcu/arch/ppc.h	2024-12-20 16:25:34
@@ -43,8 +43,13 @@
  * specific environments is a concern for ABI compatibility between
  * applications and liburcu.
  */
-#define CAA_CACHE_LINE_SIZE	256
 
+#ifdef __APPLE__
+#define CAA_CACHE_LINE_SIZE	128
+#else
+ #define CAA_CACHE_LINE_SIZE	256
+#endif
+
 #ifdef __NO_LWSYNC__
 #define LWSYNC_OPCODE	"sync\n"
 #else


https://github.com/urcu/userspace-rcu/commit/21966a0d00b48f813c154d98db0e2285af33ebcb

From 80eaa5e04fb31d99f41dc5046985e8c06bd47a52 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 16 May 2025 00:38:53 +0800
Subject: [PATCH] ppc.h: improve ppc64 macro

---
 include/urcu/arch/ppc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git include/urcu/arch/ppc.h include/urcu/arch/ppc.h
index 1285b6d..edcec1b 100644
--- include/urcu/arch/ppc.h
+++ include/urcu/arch/ppc.h
@@ -98,7 +98,7 @@ extern "C" {
 
 typedef uint64_t caa_cycles_t;
 
-#ifdef __powerpc64__
+#if defined(__powerpc64__) || defined(__ppc64__)
 static inline caa_cycles_t caa_get_cycles(void)
 {
 	return (caa_cycles_t) mftb();
