From 5e217eb200623813bce4edb1cf6b565415867986 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 26 Jun 2025 23:41:01 +0800
Subject: [PATCH] tls.py: do not error out on unsupported dtls

---
 mitmproxy/net/tls.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git mitmproxy/net/tls.py mitmproxy/net/tls.py
index d87dc1a67..d5f59d9db 100644
--- mitmproxy/net/tls.py
+++ mitmproxy/net/tls.py
@@ -18,8 +18,8 @@ class Method(Enum):
     TLS_SERVER_METHOD = SSL.TLS_SERVER_METHOD
     TLS_CLIENT_METHOD = SSL.TLS_CLIENT_METHOD
     # Type-pyopenssl does not know about these DTLS constants.
-    DTLS_SERVER_METHOD = SSL.DTLS_SERVER_METHOD   # type: ignore
-    DTLS_CLIENT_METHOD = SSL.DTLS_CLIENT_METHOD   # type: ignore
+    DTLS_SERVER_METHOD = getattr(SSL, "DTLS_SERVER_METHOD", None)
+    DTLS_CLIENT_METHOD = getattr(SSL, "DTLS_CLIENT_METHOD", None)
 
 
 try:
