--- meson.build	2025-01-30 04:12:29.000000000 +0800
+++ meson.build	2025-03-22 08:28:31.000000000 +0800
@@ -25,8 +25,7 @@
 libdl = cc.find_library('dl', required: false)
 pygobject_dep = dependency('pygobject-3.0', fallback: ['pygobject', 'pygobject_dep'], version : '>= 3.8')
 
-pymod = import('python')
-python = pymod.find_installation(get_option('python'))
+python = import('python').find_installation('@PYTHON@')
 pythonver = python.language_version()
 if pythonver.version_compare('<3.0')
   error('Python2 is not supported anymore, please port your code to python3 (@0@ specified)'.format(python.language_version()))
