--- meson.build	2026-04-08 03:02:23.000000000 +0800
+++ meson.build	2026-05-07 07:21:33.000000000 +0800
@@ -23,8 +23,7 @@
 libdl = cc.find_library('dl', required: false)
 pygobject_dep = dependency('pygobject-3.0', allow_fallback: true, version: '>= 3.8')
 
-pymod = import('python')
-python = pymod.find_installation(get_option('python-exe'), required: true)
+python = import('python').find_installation('@PYTHON@')
 pythonver = python.language_version()
 if pythonver.version_compare('<3.0')
   error('Python2 is not supported anymore, please port your code to python3 (@0@ specified)'.format(python.language_version()))
