--- meson.build	2025-02-05 01:33:24.000000000 +0800
+++ meson.build	2025-03-20 17:36:06.000000000 +0800
@@ -14,15 +14,14 @@
 endif
 
 i18n = import('i18n')
-py_mod = import('python')
 project_id = 'org.gnome.Lollypop'
 
 message('Looking for dependencies')
-py_installation = py_mod.find_installation('python3')
+py_installation = import('python').find_installation('@PYTHON@', modules: 'gi')
 if not py_installation.found()
-    error('No valid python3 binary found')
+    error('No valid python binary found')
 else
-    message('Found python3 binary')
+    message('Found python binary')
 endif
 
 # Dependencies
@@ -30,9 +29,9 @@
 dependency('gtk+-3.0', version: '>= 3.22')
 dependency('libsoup-3.0')
 dependency('glib-2.0')
-dependency('pygobject-3.0', version: '>= 3.29.1')
+#dependency('pygobject-3.0', version: '>= 3.29.1')
 
-python_dir = join_paths(get_option('prefix'), py_installation.get_install_dir())
+python_dir = '@PY_PACKAGES@'
 LIBEXEC_DIR = join_paths(get_option('prefix'), get_option('libexecdir'))
 DATA_DIR = join_paths(get_option('prefix'), get_option('datadir'), meson.project_name())
 bindir = join_paths(get_option('prefix'), get_option('bindir'))
