From aafdc7b022291c1afdbc7a3ec28202d4425daad9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 4 Nov 2025 12:35:50 +0800
Subject: [PATCH] llarp/vpn/apple.hpp: use ToHost to fix build error

---
 llarp/vpn/apple.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/llarp/vpn/apple.hpp b/llarp/vpn/apple.hpp
index 2bfb3bdf2..df8447906 100644
--- a/llarp/vpn/apple.hpp
+++ b/llarp/vpn/apple.hpp
@@ -94,8 +94,8 @@ namespace llarp::vpn
       {
         if (ifaddr.fam == AF_INET)
         {
-          const huint32_t addr = net::TruncateV6(ifaddr.range.addr);
-          const huint32_t netmask = net::TruncateV6(ifaddr.range.netmask_bits);
+          const huint32_t addr = net::ToHost(net::TruncateV6(ifaddr.range.addr));
+          const huint32_t netmask = net::ToHost(net::TruncateV6(ifaddr.range.netmask_bits));
           const huint32_t daddr = addr & netmask;
           Exec(
               "/sbin/ifconfig " + m_IfName + " " + addr.ToString() + " " + daddr.ToString()
