From 36d65a7eb1fa7d80fe00105808cde338792e5bb0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 23 Jun 2025 13:12:02 +0800
Subject: [PATCH] ioctl.hpp: add missing <fstream> header

---
 llarp/util/ioctl.hpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/llarp/util/ioctl.hpp b/llarp/util/ioctl.hpp
index 2da213ee6..6fcd9d082 100644
--- a/llarp/util/ioctl.hpp
+++ b/llarp/util/ioctl.hpp
@@ -7,6 +7,7 @@
 #include <cerrno>
 #include <cstring>
 #include <stdexcept>
+#include <fstream>
 #include <fmt/format.h>
 
 namespace llarp::util
