--- a/llarp/vpn/apple.hpp
+++ b/llarp/vpn/apple.hpp	2025-08-14 15:04:49.000000000 +0800
@@ -6,6 +6,8 @@
 #include <llarp/util/fd.hpp>
 #include <llarp/util/non_blocking.hpp>
 
+#include <AvailabilityMacros.h>
+
 #include <sys/kern_control.h>
 #include <sys/sys_domain.h>
 #include <sys/kern_event.h>
@@ -20,6 +22,11 @@
 #include <net/if.h>
 #include <net/if_var.h>
 #include <net/if_types.h>
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
+#include <net/if_utun.h>
+#else
+#define UTUN_CONTROL_NAME "com.apple.net.utun_control"
+#endif
 #include <net/route.h>
 #include <netinet/if_ether.h>
 #include <netinet/in.h>
@@ -55,7 +62,7 @@
         throw std::invalid_argument{"cannot open control socket: " + std::string{strerror(errno)}};
 
       ctl_info cinfo{};
-      const std::string apple_utun = "com.apple.net.utun_control";
+      const std::string apple_utun = UTUN_CONTROL_NAME;
       std::copy_n(apple_utun.c_str(), apple_utun.size(), cinfo.ctl_name);
 
       {
