From ee794fe68f784597e9579c7223bb5acb2e189aef Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 2 Mar 2026 02:18:06 +0800
Subject: [PATCH 2/7] CMakeLists.txt: Darwin needs _XOPEN_SOURCE for ucontext

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 736717d..f0fd6d5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFREEOCL_OS_LINUX")
 	file(WRITE ${FreeOCL_SOURCE_DIR}/icd/freeocl.icd "libFreeOCL.so")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFREEOCL_OS_DARWIN")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFREEOCL_OS_DARWIN -D_XOPEN_SOURCE")
 	file(WRITE ${FreeOCL_SOURCE_DIR}/icd/freeocl.icd "libFreeOCL.dylib")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DFREEOCL_OS_WINDOWS")
