From f53df2b7bda5f995e934bc607dffa1ef42555ae7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 2 Mar 2026 02:20:19 +0800
Subject: [PATCH 3/7] icd_loader.h: add missing macros

---
 src/icd/icd_loader.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/icd/icd_loader.h b/src/icd/icd_loader.h
index bdb44b9..7002662 100644
--- a/src/icd/icd_loader.h
+++ b/src/icd/icd_loader.h
@@ -24,6 +24,13 @@
 #include <CL/cl_ext.h>
 #endif
 
+#ifndef CL_PLATFORM_ICD_SUFFIX_KHR
+#define CL_PLATFORM_ICD_SUFFIX_KHR 0x0920
+#endif
+#ifndef CL_PLATFORM_NOT_FOUND_KHR
+#define CL_PLATFORM_NOT_FOUND_KHR -1001
+#endif
+
 #include <cstring>
 #include <string>
 #include <deque>
