--- pidgin/getopt.h	2024-09-17 16:52:23.000000000 +0800
+++ pidgin/getopt.h	2026-06-08 03:04:19.000000000 +0800
@@ -102,7 +102,7 @@
 #define optional_argument	2
 
 #if __STDC__
-#if defined(__GNU_LIBRARY__)
+#if defined(__GNU_LIBRARY__) || defined(__APPLE__)
 /* Many other libraries have conflicting prototypes for getopt, with
    differences in the consts, in stdlib.h.  To avoid compilation
    errors, only prototype getopt for the GNU C library.  */

--- finch/getopt.h	2024-09-17 16:52:23.000000000 +0800
+++ finch/getopt.h	2026-06-08 03:03:32.000000000 +0800
@@ -102,7 +102,7 @@
 #define optional_argument	2
 
 #if __STDC__
-#if defined(__GNU_LIBRARY__)
+#if defined(__GNU_LIBRARY__) || defined(__APPLE__)
 /* Many other libraries have conflicting prototypes for getopt, with
    differences in the consts, in stdlib.h.  To avoid compilation
    errors, only prototype getopt for the GNU C library.  */
