--- lang/sql/sqlite/src/shell.c.orig	2014-01-25 03:09:46.000000000 +0800
+++ lang/sql/sqlite/src/shell.c	2025-01-05 11:57:03.000000000 +0800
@@ -59,7 +59,7 @@
 # include <sys/types.h>
 #endif
 
-#ifdef HAVE_EDITLINE && HAVE_EDITLINE==1
+#if defined(HAVE_EDITLINE) && HAVE_EDITLINE==1
 # include <editline/editline.h>
 #elif defined(HAVE_READLINE) && HAVE_READLINE==1
 # include <readline/readline.h>
