--- a/Infovis/Core/vtkWordCloud.cxx	2026-03-25 04:22:33.000000000 +0800
+++ b/Infovis/Core/vtkWordCloud.cxx	2026-05-01 07:46:57.000000000 +0800
@@ -654,7 +654,11 @@
 
   constexpr std::size_t N = 10000;
   constexpr auto pi = 3.141592653589793238462643383279502884L; /* pi */
+#if defined(__POWERPC__) || defined(__powerpc__)
+  const double deltaAngle = pi * 20 / N;
+#else
   constexpr double deltaAngle = pi * 20 / N;
+#endif
   double maxX = -1000.0;
   double minX = 1000.0;
   double maxY = -1000.0;
