From 7d3620d14e837415b5a3646959bee91fe615a260 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 1 Apr 2025 02:39:57 +0800
Subject: [PATCH] CMakeLists: no librt on Apple

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index f06cd44..1fc9d77 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -41,8 +41,13 @@ pkg_check_modules(deps REQUIRED IMPORTED_TARGET pugixml)
 
 file(GLOB_RECURSE SRCFILES CONFIGURE_DEPENDS "src/*.cpp")
 add_executable(hyprwayland-scanner ${SRCFILES})
-target_link_libraries(hyprwayland-scanner PRIVATE rt Threads::Threads
+if(APPLE)
+  target_link_libraries(hyprwayland-scanner PRIVATE Threads::Threads
                                                   PkgConfig::deps)
+else()
+  target_link_libraries(hyprwayland-scanner PRIVATE rt Threads::Threads
+                                                  PkgConfig::deps)
+endif()
 
 configure_package_config_file(
   hyprwayland-scanner-config.cmake.in hyprwayland-scanner-config.cmake
