--- bee.lua/binding/lua_platform.cpp	2025-08-08 01:26:53.000000000 +0800
+++ bee.lua/binding/lua_platform.cpp	2025-12-31 03:27:55.000000000 +0800
@@ -125,6 +125,10 @@
         lua_pushstring(L, "arm");
 #elif defined(__riscv)
         lua_pushstring(L, "riscv");
+#elif defined(__ppc64__) || defined(__powerpc64__)
+        lua_pushstring(L, "ppc64");
+#elif defined(__ppc__) || defined(__powerpc__)
+        lua_pushstring(L, "ppc");
 #elif defined(__wasm32__)
         lua_pushstring(L, "wasm32");
 #elif defined(__wasm64__)
