This test is not supported on < 10.7

--- bee.lua/test/test_filewatch.lua
+++ bee.lua/test/test_filewatch.lua	2025-08-08 03:03:18.000000000 +0800
@@ -34,46 +34,6 @@
     end)
 end
 
-function test_fw:test_2()
-    test(function(fw, root)
-        fs.create_directories(root / 'test1')
-        create_file(root / 'test1.txt')
-        fs.rename(root / 'test1.txt', root / 'test2.txt')
-        fs.remove(root / 'test2.txt')
-
-        local list = {}
-        local n = 100
-        while true do
-            local w, v = fw:select()
-            if w then
-                n = 100
-                if type(v) == 'userdata' or type(v) == 'table' then
-                    list[#list+1] = v
-                else
-                    list[#list+1] = fs.path(v)
-                end
-            else
-                n = n - 1
-                if n < 0 then
-                    break
-                end
-                thread.sleep(0.001)
-            end
-        end
-        local function assertHas(path)
-            for _, v in ipairs(list) do
-                if v == path then
-                    return
-                end
-            end
-            lt.assertEquals(path, nil)
-        end
-        assertHas(root / 'test1')
-        assertHas(root / 'test1.txt')
-        assertHas(root / 'test2.txt')
-    end)
-end
-
 -- test unexist symlink link to self
 -- test directory symlink link to parent
 function test_fw:test_symlink()
