--- src/dune	2023-05-04 21:18:43.000000000 +0800
+++ src/dune	2025-10-18 12:06:57.000000000 +0800
@@ -10,6 +10,7 @@
             notty
             notty.unix
             nottui
+            nottui-unix
             oseq
  )
 )

--- src/docfd.ml	2023-05-04 21:18:43.000000000 +0800
+++ src/docfd.ml	2025-10-18 12:07:30.000000000 +0800
@@ -176,7 +176,7 @@
       let rec loop () =
         Ui_base.Vars.file_to_open := None;
         Lwd.set Ui_base.Vars.quit false;
-        Nottui.Ui_loop.run
+        Nottui_unix.run
           ~term:Ui_base.(get_term ())
           ~renderer
           ~quit:Ui_base.Vars.quit
