--- Makefile.in.orig	2020-04-22 02:21:11.000000000 -0500
+++ Makefile.in	2020-06-14 19:18:20.000000000 -0500
@@ -3298,7 +3298,7 @@
 
 # Leading 'am--fnord' is there to ensure the list of targets does not
 # expand to empty, as could happen e.g. with make check TESTS=''.
-am--fnord $(TEST_LOGS) $(TEST_LOGS:.log=.trs): $(am__force_recheck)
+am--fnord $(TEST_LOGS): $(am__force_recheck)
 am--force-recheck:
 	@:
 
--- Makefile.in.orig	2020-04-22 15:21:08.000000000 +0800
+++ Makefile.in	2025-06-01 13:11:00.000000000 +0800
@@ -261,6 +261,7 @@
 	$(CODE_COVERAGE_CPPFLAGS)
 
 i3_LDADD = \
+	$(AM_LDFLAGS) \
 	$(libi3_LIBS) \
 	$(LIBSN_LIBS) \
 	$(XCB_LIBS) \
