--- include/libtorrent/config.hpp	2025-01-28 20:13:22.000000000 +0800
+++ include/libtorrent/config.hpp	2025-04-30 11:53:43.000000000 +0800
@@ -144,7 +144,7 @@
 #define TORRENT_USE_EXECINFO 1
 #endif
 
-#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 // this is used for the ip_change_notifier on macOS, which isn't supported on
 // 10.6 and earlier
 #define TORRENT_USE_SYSTEMCONFIGURATION 1
