From df231ccaea9a3712cf5a380b74d063c33f0db09a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 4 Feb 2025 11:29:56 +0800
Subject: [PATCH] CMakeLists: fix silly bug

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index 0681a286c..a590760d8 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -3281,7 +3281,7 @@ if(ANDROID)
   sdl_include_directories(PRIVATE SYSTEM "${ANDROID_NDK}/sources/android/cpufeatures")
 endif()
 
-if(APPLE)
+if(APPLE AND CMAKE_C_COMPILER_ID MATCHES "Clang")
   cmake_push_check_state(RESET)
   check_c_compiler_flag(-fobjc-arc COMPILER_SUPPORTS_FOBJC_ARC)
   cmake_pop_check_state()
