From 579b717246a381699b918b115e00bbe1c2c3c6d7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 4 Feb 2025 14:21:41 +0800
Subject: [PATCH 08/10] Do not smuggle in a non-existing framework

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index e2f8ef0bc..108e89f33 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -2196,7 +2196,7 @@ elseif(APPLE)
     set(HAVE_SDL_AUDIO TRUE)
     set(SDL_FRAMEWORK_COREAUDIO 1)
     set(SDL_FRAMEWORK_AUDIOTOOLBOX 1)
-    set(SDL_FRAMEWORK_AVFOUNDATION 1)
+    # set(SDL_FRAMEWORK_AVFOUNDATION 1)
     CheckJACK()
     CheckPulseAudio()
   endif()
@@ -2402,7 +2402,7 @@ elseif(APPLE)
   if(SDL_FRAMEWORK_COCOA)
     sdl_link_dependency(cocoa LIBS "$<LINK_LIBRARY:FRAMEWORK,Cocoa>" PKG_CONFIG_LINK_OPTIONS "-Wl,-framework,Cocoa")
     # macOS 11.0+ iOS 14.0+ tvOS 14.0+
-    sdl_link_dependency(uniformtypeidentifiers LIBS "$<LINK_LIBRARY:WEAK_FRAMEWORK,UniformTypeIdentifiers>" PKG_CONFIG_LINK_OPTIONS "-Wl,-weak_framework,UniformTypeIdentifiers")
+    # sdl_link_dependency(uniformtypeidentifiers LIBS "$<LINK_LIBRARY:WEAK_FRAMEWORK,UniformTypeIdentifiers>" PKG_CONFIG_LINK_OPTIONS "-Wl,-weak_framework,UniformTypeIdentifiers")
   endif()
   if(SDL_FRAMEWORK_IOKIT)
     sdl_link_dependency(iokit LIBS "$<LINK_LIBRARY:FRAMEWORK,IOKit>" PKG_CONFIG_LINK_OPTIONS "-Wl,-framework,IOKit")
-- 
2.48.0

