From 721354befd59e0c88ebe4b53e0a144b72c4ee20e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 4 Feb 2025 15:23:47 +0800
Subject: [PATCH 11/11] And fix linking

---
 CMakeLists.txt | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git CMakeLists.txt CMakeLists.txt
index 1fa498ccb..54f45c402 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -2487,6 +2488,14 @@
     set(HAVE_SDL_MAIN_CALLBACKS TRUE)
   endif()
 
+  if(NOT SDL_COCOA)
+    sdl_sources(
+      "${SDL3_SOURCE_DIR}/src/core/unix/SDL_appid.c"
+      "${SDL3_SOURCE_DIR}/src/core/unix/SDL_poll.c"
+      "${SDL3_SOURCE_DIR}/src/dialog/unix/SDL_zenitymessagebox.c"
+    )
+  endif()
+
   if(SDL_CAMERA)
     if(MACOS OR IOS)
       set(SDL_CAMERA_DRIVER_COREMEDIA 1)
