--- src/filesystem/cocoa/SDL_sysfilesystem.m
+++ src/filesystem/cocoa/SDL_sysfilesystem.m	2025-07-08 11:49:14.000000000 +0800
@@ -28,6 +28,7 @@
 #include "../SDL_sysfilesystem.h"
 
 #include <Foundation/Foundation.h>
+#include <AvailabilityMacros.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 
@@ -165,6 +166,7 @@
             dir = NSDownloadsDirectory;
             break;
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
         case SDL_FOLDER_MUSIC:
             dir = NSMusicDirectory;
             break;
@@ -192,6 +194,7 @@
         case SDL_FOLDER_VIDEOS:
             dir = NSMoviesDirectory;
             break;
+#endif
 
         default:
             SDL_SetError("Invalid SDL_Folder: %d", (int) folder);
