--- OPCODE/Opcode.h	2025-01-17 06:48:38.000000000 +0800
+++ OPCODE/Opcode.h	2025-08-13 04:49:59.000000000 +0800
@@ -28,16 +28,6 @@
 ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // Things to help us compile on non-windows platforms
 
-#if defined(__APPLE__) || defined(__MACOSX__)
-#if __APPLE_CC__ < 1495
-#define sqrtf sqrt
-#define sinf sin
-#define cosf cos
-#define acosf acos
-#define asinf asin
-#endif
-#endif
-
 #ifndef _MSC_VER
 #ifndef __int64
 #define __int64 long long int
