From 96f9395a168f15a393a732b1aaf86cb058e52b7a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fran=C3=A7ois=20Degros?= <fdegros@chromium.org>
Date: Mon, 18 Aug 2025 11:09:42 +1000
Subject: [PATCH] Fix build error on macOS

Bug: https://github.com/google/fuse-archive/issues/54
---
 fuse-archive.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/fuse-archive.cc b/fuse-archive.cc
index 8849ee9..ba1fab6 100644
--- fuse-archive.cc
+++ fuse-archive.cc
@@ -2579,7 +2579,12 @@ int GetAttr(const char* const path,
 int GetXattr(const char* const path,
              const char* const xattr_name,
              char* const dst_ptr,
+#ifdef __APPLE__
+             size_t const dst_len,
+             uint32_t) {
+#else
              size_t const dst_len) {
+#endif
   assert(path);
   assert(xattr_name);
 
