From 0ebd0008e34f2e894924407e6dae88452818ca80 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 18 Feb 2026 10:50:13 +0800
Subject: [PATCH 2/2] Fix build without AVFoundation

---
 modules/imgcodecs/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/imgcodecs/CMakeLists.txt b/modules/imgcodecs/CMakeLists.txt
index c322645b20..37d7951565 100644
--- a/modules/imgcodecs/CMakeLists.txt
+++ b/modules/imgcodecs/CMakeLists.txt
@@ -143,7 +143,7 @@ file(GLOB imgcodecs_ext_hdrs
      "${CMAKE_CURRENT_LIST_DIR}/include/opencv2/${name}/legacy/*.h"
      )
 
-if(APPLE OR APPLE_FRAMEWORK)
+if((APPLE OR APPLE_FRAMEWORK) AND WITH_AVFOUNDATION)
   list(APPEND imgcodecs_srcs ${CMAKE_CURRENT_LIST_DIR}/src/apple_conversions.h)
   list(APPEND imgcodecs_srcs ${CMAKE_CURRENT_LIST_DIR}/src/apple_conversions.mm)
 endif()
@@ -151,7 +151,7 @@ if(IOS OR XROS)
   list(APPEND imgcodecs_srcs ${CMAKE_CURRENT_LIST_DIR}/src/ios_conversions.mm)
   list(APPEND IMGCODECS_LIBRARIES "-framework UIKit")
 endif()
-if(APPLE AND (NOT IOS) AND (NOT XROS))
+if(APPLE AND (NOT IOS) AND (NOT XROS) AND WITH_AVFOUNDATION)
   list(APPEND imgcodecs_srcs ${CMAKE_CURRENT_LIST_DIR}/src/macosx_conversions.mm)
   list(APPEND IMGCODECS_LIBRARIES "-framework AppKit")
 endif()
