From e1e170afc210aef025547b42330361c431e19f07 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 18 Feb 2026 04:35:41 +0800
Subject: [PATCH] parallel_impl.cpp: fix assembler syntax for powerpc*-darwin

---
 modules/core/src/parallel_impl.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/core/src/parallel_impl.cpp b/modules/core/src/parallel_impl.cpp
index fc7c4c2b6e..e203569ac1 100644
--- a/modules/core/src/parallel_impl.cpp
+++ b/modules/core/src/parallel_impl.cpp
@@ -53,6 +53,8 @@ DECLARE_CV_PAUSE
 #   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { asm volatile("" ::: "memory"); } } while (0)
 # elif defined __GNUC__ && defined __mips__ && __mips_isa_rev >= 2
 #   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { asm volatile("pause" ::: "memory"); } } while (0)
+# elif defined __APPLE__ && defined __POWERPC__
+#   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { asm volatile("or r27,r27,r27" ::: "memory"); } } while (0)
 # elif defined __GNUC__ && defined __PPC64__
 #   define CV_PAUSE(v) do { for (int __delay = (v); __delay > 0; --__delay) { asm volatile("or 27,27,27" ::: "memory"); } } while (0)
 # elif defined __GNUC__ && defined __riscv
