--- src/os_unix.c	2025-12-30 23:24:22.000000000 +0800
+++ src/os_unix.c	2025-12-30 23:44:28.000000000 +0800
@@ -53,6 +53,10 @@
 # endif
 #endif
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 #ifdef __CYGWIN__
 # include <cygwin/version.h>
 # include <sys/cygwin.h>	// for cygwin_conv_to_posix_path() and/or
@@ -3680,7 +3684,7 @@
      * Inform the macOS scheduler that Vim renders UI, and so shouldn’t have its
      * threads’ quality of service classes clamped.
      */
-#ifdef MACOS_X
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
     integer_t policy = TASK_DEFAULT_APPLICATION;
     task_policy_set(mach_task_self(), TASK_CATEGORY_POLICY, &policy, 1);
 #endif
