--- src/os_unix.c.orig	2024-09-11 03:25:14.000000000 +0800
+++ src/os_unix.c	2024-12-28 08:40:47.000000000 +0800
@@ -533,7 +533,7 @@
 	// Mac (Darwin) way of getting the amount of RAM available
 	mach_port_t		host = mach_host_self();
 	kern_return_t		kret;
-#  ifdef HOST_VM_INFO64
+#  if defined(HOST_VM_INFO64) && !defined(__i386__) && !defined(__ppc__)
 	struct vm_statistics64	vm_stat;
 	natural_t		count = HOST_VM_INFO64_COUNT;
 
