--- ui/cocoa.m	2015-04-27 22:08:29.000000000 +0800
+++ ui/cocoa.m	2025-11-30 21:04:40.000000000 +0800
@@ -858,7 +858,7 @@
         [op setMessage:@"Select the disk image you want to boot.\n\nHit the \"Cancel\" button to quit"];
         NSArray *filetypes = [NSArray arrayWithObjects:@"img", @"iso", @"dmg",
                                  @"qcow", @"qcow2", @"cloop", @"vmdk", nil];
-#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_6)
+#if (MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_6) && defined(__clang__)
         [op setAllowedFileTypes:filetypes];
         [op beginSheetModalForWindow:normalWindow
             completionHandler:^(NSInteger returnCode)
