From 02039aad1bdf2b9386261610533a6aef46c52610 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 2 Jun 2025 12:02:53 +0800
Subject: [PATCH 1/5] Do not include endian.h on macOS

---
 src/sqlite3mc_amalgamation.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git src/sqlite3mc_amalgamation.c src/sqlite3mc_amalgamation.c
index 26f549c5..d0532a39 100644
--- src/sqlite3mc_amalgamation.c
+++ src/sqlite3mc_amalgamation.c
@@ -279189,7 +279189,8 @@ void sqlcipher_hmac(int algorithm,
 
 #include <assert.h>
 #include <string.h>
-#if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__clang__) && !defined(__QNX__)
+#if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__clang__) \
+    && !defined(__QNX__) && !defined(__APPLE__)
 #include <endian.h>
 #endif
 
-- 
2.48.0

