From 02166d911c6f6c6cf4feb2f826b26976306fa84f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 2 Jun 2025 12:05:47 +0800
Subject: [PATCH 2/5] Add __POWERPC__ to Big-endian macros

---
 src/sqlite3mc_amalgamation.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git src/sqlite3mc_amalgamation.c src/sqlite3mc_amalgamation.c
index d0532a39..c1cecb8a 100644
--- src/sqlite3mc_amalgamation.c
+++ src/sqlite3mc_amalgamation.c
@@ -15537,7 +15537,8 @@ typedef INT16_TYPE LogEst;
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
      defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
 #   define SQLITE_BYTEORDER 1234
-# elif defined(sparc)   || defined(__ARMEB__)     || defined(__AARCH64EB__)
+# elif defined(sparc)   || defined(__ARMEB__)     || defined(__AARCH64EB__) \
+    || defined(__POWERPC__)
 #   define SQLITE_BYTEORDER 4321
 # else
 #   define SQLITE_BYTEORDER 0
@@ -214020,7 +214021,8 @@ struct RtreeMatchArg {
      defined(_M_AMD64)  || defined(_M_ARM)        || defined(__x86)   ||    \
      defined(__ARMEL__) || defined(__AARCH64EL__) || defined(_M_ARM64)
 #   define SQLITE_BYTEORDER 1234
-# elif defined(sparc)   || defined(__ARMEB__)     || defined(__AARCH64EB__)
+# elif defined(sparc)   || defined(__ARMEB__)     || defined(__AARCH64EB__) \
+    || defined(__POWERPC__)
 #   define SQLITE_BYTEORDER 4321
 # else
 #   define SQLITE_BYTEORDER 0
-- 
2.48.0

