From 62e45c26c9dbe60b723cf5504594df894c2ff825 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 2 Jun 2025 12:12:00 +0800
Subject: [PATCH 3/5] Fix SQLITE_PTRSIZE for ppc64

---
 src/sqlite3mc_amalgamation.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/sqlite3mc_amalgamation.c src/sqlite3mc_amalgamation.c
index c1cecb8a..ce00f9e4 100644
--- src/sqlite3mc_amalgamation.c
+++ src/sqlite3mc_amalgamation.c
@@ -356490,7 +356490,7 @@ sqlite3mc_shutdown(void)
 #     define SQLITE_PTRSIZE __SIZEOF_POINTER__
 #   elif defined(i386)     || defined(__i386__)   || defined(_M_IX86) ||    \
          defined(_M_ARM)   || defined(__arm__)    || defined(__x86)   ||    \
-        (defined(__APPLE__) && defined(__POWERPC__)) ||                     \
+        (defined(__APPLE__) && defined(__ppc__))  ||                        \
         (defined(__TOS_AIX__) && !defined(__64BIT__))
 #     define SQLITE_PTRSIZE 4
 #   else
-- 
2.48.0

