From aeacb688c1c0fd6edd1b043c2d7d06b0e2f56dc5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 2 Jun 2025 12:33:11 +0800
Subject: [PATCH 5/5] Use SecRandom only on 10.7+

---
 src/sqlite3mc_amalgamation.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git src/sqlite3mc_amalgamation.c src/sqlite3mc_amalgamation.c
index b58d0c03..7eee1fc3 100644
--- src/sqlite3mc_amalgamation.c
+++ src/sqlite3mc_amalgamation.c
@@ -279979,6 +279979,8 @@ static size_t entropy(void* buf, size_t n)
 #ifndef RNDGETENTCNT
 #define RNDGETENTCNT _IOR('R', 0x00, int)
 #endif
+#elif defined(__APPLE__)
+#include <AvailabilityMacros.h>
 #endif
 
 /* Returns the number of urandom bytes read (either 0 or n) */
@@ -280041,13 +280043,13 @@ fail:
   return 0;
 }
 
-#if defined(__APPLE__)
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 #include <Security/SecRandom.h>
 #endif
 
 static size_t entropy(void* buf, size_t n)
 {
-#if defined(__APPLE__)
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
   if (SecRandomCopyBytes(kSecRandomDefault, n, (uint8_t*) buf) == 0)
     return n;
 #elif defined(__linux__) && defined(SYS_getrandom)
-- 
2.48.0

