--- src/sqlite3mc_amalgamation.c.orig	2025-07-14 18:36:23.000000000 +0800
+++ src/sqlite3mc_amalgamation.c	2025-07-18 02:43:10.000000000 +0800
@@ -280143,12 +280143,15 @@
 }
 
 #if defined(__APPLE__)
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 #include <CommonCrypto/CommonRandom.h>
 #endif
+#endif
 
 static size_t entropy(void* buf, size_t n)
 {
-#if defined(__APPLE__)
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
   if (CCRandomGenerateBytes(buf, n) == kCCSuccess)
     return n;
 #elif defined(__linux__) && defined(SYS_getrandom)
