From f8d57652d12f9d212f373a81e493eba1a0b058c5 Mon Sep 17 00:00:00 2001
From: Nicholas Corgan <n.corgan@gmail.com>
Date: Sat, 26 Nov 2022 18:07:48 -0600
Subject: [PATCH] ci: added module registration test so we don't depend on
 module CIs

diff --git a/ExampleDriver/CMakeLists.txt b/ExampleDriver/CMakeLists.txt
index 4e83dbe2..4b172d45 100644
--- ExampleDriver/CMakeLists.txt
+++ ExampleDriver/CMakeLists.txt
@@ -17,8 +17,8 @@ set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING "")
 # Header and library resources needed to communicate with the device.
 # These may be found within the build tree or in an external project.
 ########################################################################
-set(MY_DEVICE_INCLUDE_DIRS ...)
-set(MY_DEVICE_LIBRARIES ...)
+set(MY_DEVICE_INCLUDE_DIRS "")
+set(MY_DEVICE_LIBRARIES "")
 
 ########################################################################
 # build the module
diff --git a/ExampleDriver/MyDeviceSupport.cpp b/ExampleDriver/MyDeviceSupport.cpp
index b6f6836d..45995579 100644
--- ExampleDriver/MyDeviceSupport.cpp
+++ ExampleDriver/MyDeviceSupport.cpp
@@ -16,8 +16,11 @@ class MyDevice : public SoapySDR::Device
  **********************************************************************/
 SoapySDR::KwargsList findMyDevice(const SoapySDR::Kwargs &args)
 {
+    (void)args;
     //locate the device on the system...
     //return a list of 0, 1, or more argument maps that each identify a device
+
+    return SoapySDR::KwargsList();
 }
 
 /***********************************************************************
@@ -25,9 +28,10 @@ SoapySDR::KwargsList findMyDevice(const SoapySDR::Kwargs &args)
  **********************************************************************/
 SoapySDR::Device *makeMyDevice(const SoapySDR::Kwargs &args)
 {
+    (void)args;
     //create an instance of the device object given the args
     //here we will translate args into something used in the constructor
-    return new MyDevice(...);
+    return new MyDevice();
 }
 
 /***********************************************************************
diff --git a/cmake/Modules/SoapySDRUtil.cmake b/cmake/Modules/SoapySDRUtil.cmake
index 6d06ff81..04490d53 100644
--- cmake/Modules/SoapySDRUtil.cmake
+++ cmake/Modules/SoapySDRUtil.cmake
@@ -100,7 +100,11 @@ function(SOAPY_SDR_MODULE_UTIL)
     if(CMAKE_COMPILER_IS_GNUCXX)
         #force a compile-time error when symbols are missing
         #otherwise modules will cause a runtime error on load
-        target_link_libraries(${MODULE_TARGET} PRIVATE "-Wl,--no-undefined")
+        if(APPLE)
+            target_link_libraries(${MODULE_TARGET} PRIVATE "-Wl,-undefined,error")
+        else()
+            target_link_libraries(${MODULE_TARGET} PRIVATE "-Wl,--no-undefined")
+        endif()
     endif()
 
     if (NOT MODULE_DESTINATION)
