--- src/output/plugins/OSXOutputPlugin.cxx	2025-03-11 18:05:29.000000000 +0800
+++ src/output/plugins/OSXOutputPlugin.cxx	2025-03-19 21:07:28.000000000 +0800
@@ -28,6 +28,7 @@
 #include <AudioToolbox/AudioToolbox.h>
 #include <CoreServices/CoreServices.h>
 
+#include <unistd.h>
 #include <memory>
 #include <span>
 
@@ -38,6 +39,7 @@
 #else
 	#define KAUDIO_OBJECT_PROPERTY_ELEMENT_MM kAudioObjectPropertyElementMaster
 	#define KAUDIO_HARDWARE_SERVICE_DEVICE_PROPERTY_VV kAudioHardwareServiceDeviceProperty_VirtualMasterVolume
+	#define kAudioObjectPropertyScopeOutput kAudioDevicePropertyScopeOutput
 #endif
 
 static constexpr unsigned MPD_OSX_BUFFER_TIME_MS = 100;
