--- src/local.h	2002-12-22 23:39:35.000000000 +0800
+++ src/local.h	2026-08-01 15:21:06.000000000 +0800
@@ -93,7 +93,7 @@
 # endif
 
 # ifdef PNET_APPLE
-typedef int socklen_t;
+#  include <sys/socket.h>
 # endif
 
 # ifndef STDWin32				/* {{ */

--- src/pkt/bpf-gencode.c	2002-12-22 23:39:33.000000000 +0800
+++ src/pkt/bpf-gencode.c	2026-08-01 15:26:26.000000000 +0800
@@ -4325,7 +4325,7 @@
  * "const char *", so no matter how we declare it here, it'll fail to
  * compile on one of 3.x or 4.x).
  */
-#if !defined(sgi) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if !defined(sgi) && !defined(__NetBSD__) && !defined(__FreeBSD__) && !defined(__APPLE__)
 extern int ether_hostton(char *, struct ether_addr *);
 #endif
 
