--- examples/client/client.c	2026-01-06 09:44:44.000000000 +0800
+++ examples/client/client.c	2026-03-11 14:54:27.000000000 +0800
@@ -79,6 +79,13 @@
     #include <hal/tpm_io.h>
 #endif /* WOLFSSH_TPM */
 
+#if (defined(__OSX__) || defined(__APPLE__))
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#define APPLE_DISPATCH
+#endif
+#endif
+
 #ifndef NO_WOLFSSH_CLIENT
 
 static const char testString[] = "Hello, wolfSSH!";
@@ -240,7 +247,7 @@
 #ifdef WOLFSSH_TERM
 #ifndef _MSC_VER
 
-#if (defined(__OSX__) || defined(__APPLE__))
+#ifdef APPLE_DISPATCH
 #include <dispatch/dispatch.h>
 dispatch_semaphore_t windowSem;
 #else
@@ -251,7 +258,7 @@
 /* capture window change signales */
 static void WindowChangeSignal(int sig)
 {
-#if (defined(__OSX__) || defined(__APPLE__))
+#ifdef APPLE_DISPATCH
     dispatch_semaphore_signal(windowSem);
 #else
     sem_post(&windowSem);
@@ -267,7 +274,7 @@
 
     args = (thread_args*)in;
     do {
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_semaphore_wait(windowSem, DISPATCH_TIME_FOREVER);
     #else
         sem_wait(&windowSem);
@@ -1032,7 +1039,7 @@
         arg.quit = 0;
         wc_InitMutex(&arg.lock);
 #ifdef WOLFSSH_TERM
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         windowSem = dispatch_semaphore_create(0);
     #else
         sem_init(&windowSem, 0, 0);
@@ -1057,7 +1064,7 @@
 #ifdef WOLFSSH_TERM
         /* Wake the windowMonitor thread so it can exit. */
         arg.quit = 1;
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_semaphore_signal(windowSem);
     #else
         sem_post(&windowSem);
@@ -1067,7 +1074,7 @@
         pthread_cancel(thread[1]);
         pthread_join(thread[1], NULL);
 #ifdef WOLFSSH_TERM
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_release(windowSem);
     #else
         sem_destroy(&windowSem);

--- apps/wolfssh/wolfssh.c	2026-01-06 09:44:44.000000000 +0800
+++ apps/wolfssh/wolfssh.c	2026-03-11 15:02:30.000000000 +0800
@@ -77,6 +77,12 @@
     #include <wolfssl/wolfcrypt/asn.h>
 #endif
 
+#if (defined(__OSX__) || defined(__APPLE__))
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+#define APPLE_DISPATCH
+#endif
+#endif
 
 int myoptind = 0;
 char* myoptarg = NULL;
@@ -272,7 +278,7 @@
 
 #ifndef _MSC_VER
 
-#if (defined(__OSX__) || defined(__APPLE__))
+#ifdef APPLE_DISPATCH
 #include <dispatch/dispatch.h>
 dispatch_semaphore_t windowSem;
 #else
@@ -283,7 +289,7 @@
 /* capture window change signales */
 static void WindowChangeSignal(int sig)
 {
-#if (defined(__OSX__) || defined(__APPLE__))
+#ifdef APPLE_DISPATCH
     dispatch_semaphore_signal(windowSem);
 #else
     sem_post(&windowSem);
@@ -299,7 +305,7 @@
 
     args = (thread_args*)in;
     do {
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_semaphore_wait(windowSem, DISPATCH_TIME_FOREVER);
     #else
         sem_wait(&windowSem);
@@ -1060,7 +1066,7 @@
         arg.readError = 0;
 #ifdef WOLFSSH_TERM
         arg.quit = 0;
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         windowSem = dispatch_semaphore_create(0);
     #else
         sem_init(&windowSem, 0, 0);
@@ -1087,7 +1093,7 @@
 #ifdef WOLFSSH_TERM
         /* Wake the windowMonitor thread so it can exit. */
         arg.quit = 1;
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_semaphore_signal(windowSem);
     #else
         sem_post(&windowSem);
@@ -1097,7 +1103,7 @@
         pthread_cancel(thread[1]);
         pthread_join(thread[1], NULL);
 #ifdef WOLFSSH_TERM
-    #if (defined(__OSX__) || defined(__APPLE__))
+    #ifdef APPLE_DISPATCH
         dispatch_release(windowSem);
     #else
         sem_destroy(&windowSem);
