--- src/threads.h	2024-01-19 19:07:44.000000000 +0800
+++ src/threads.h	2024-12-29 15:40:31.000000000 +0800
@@ -88,9 +88,9 @@
 
 #endif //__GNUC__
 
-#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED > 1050 && !defined(__ppc__)
+#if defined(__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED > 1050
 // POSIX semaphores are deprecated on Mac so we use Grand Central Dispatch semaphores instead.
-// However GCD is supported only on 10.6+, and is not supported on any ppc, including 10.6 Rosetta.
+// However GCD is supported only on 10.6+. (Looks like PowerPC on 10.6 can be enabled now.)
 #include <dispatch/dispatch.h>
 typedef dispatch_semaphore_t kvz_sem_t;
 
