--- src/strategyselector.c.orig	2026-09-16 22:31:04.000000000 -0600
+++ src/strategyselector.c	2026-09-16 22:43:24.000000000 -0600
@@ -41,6 +41,13 @@
 #else
 #include <unistd.h>
 #endif
+#if defined(__APPLE__)
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#include <sys/param.h>
+#include <sys/sysctl.h>
+#endif
+#endif
 
 hardware_flags_t kvz_g_hardware_flags;
 hardware_flags_t kvz_g_strategies_in_use;
@@ -487,6 +494,24 @@
     GetSystemInfo(&systeminfo);
 
     kvz_g_hardware_flags.logical_cpu_count = systeminfo.dwNumberOfProcessors;
+#elif defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED < 1050)
+// Adapted from https://github.com/macos-powerpc/powerpc-ports/blob/main/lang/gcc16/files/darwin8-core-count.patch
+    int nm[2];
+    size_t len = 4;
+    uint32_t count;
+ 
+    nm[0] = CTL_HW;
+    nm[1] = HW_AVAILCPU;
+    sysctl(nm, 2, &count, &len, NULL, 0);
+ 
+    if(count < 1) {
+        nm[1] = HW_NCPU;
+        sysctl(nm, 2, &count, &len, NULL, 0);
+    
+        if(count < 1) { count = 1; }
+    }
+    
+    kvz_g_hardware_flags.logical_cpu_count =  count;
 #else
     kvz_g_hardware_flags.logical_cpu_count = sysconf(_SC_NPROCESSORS_ONLN);
 #endif
@@ -514,7 +539,7 @@
       // supported by the compiler.
 #ifdef _XCR_XFEATURE_ENABLED_MASK
       xcr0 = _xgetbv(_XCR_XFEATURE_ENABLED_MASK);
-#elif defined(__GNUC__)
+#elif defined(__GNUC__) && (COMPILE_INTEL_AVX || COMPILE_INTEL_AVX2)
       unsigned eax = 0, edx = 0;
       asm("xgetbv" : "=a"(eax), "=d"(edx) : "c" (0));
       xcr0 = (uint64_t)edx << 32 | eax;
