--- CMakeLists.txt
+++ CMakeLists.txt	2026-01-22 06:57:10.000000000 +0800
@@ -193,7 +193,7 @@
     src/Shared
     src/MACLib)
 
-add_library(libmac
+add_library(libmac SHARED
     ${SHARED_SOURCES}
     ${SHARED_HEADERS}
     ${LIBMAC_SOURCES}
@@ -202,7 +202,9 @@
 target_link_libraries(libmac ${NANOSLEEP_LIBRARY})
 
 set_target_properties(libmac PROPERTIES
-    OUTPUT_NAME mac)
+    OUTPUT_NAME mac
+    VERSION 3.99.6
+    SOVERSION 3)
 
 add_executable(mac ${MAC_SOURCES})
 target_link_libraries(mac libmac)
@@ -210,4 +212,5 @@
 add_executable(Sample1 ${SAMPLE1_SOURCES})
 target_link_libraries(Sample1 libmac)
 
+install(TARGETS libmac DESTINATION lib)
 install(TARGETS mac DESTINATION bin)
