--- src/alcotest/dune.orig	2025-03-13 17:00:14.000000000 +0800
+++ src/alcotest/dune	2025-03-26 06:12:17.000000000 +0800
@@ -3,7 +3,8 @@
  (libraries alcotest.engine astring fmt fmt.tty unix)
  (foreign_stubs
   (language c)
-  (names alcotest_stubs))
+  (names alcotest_stubs)
+  (flags -I@PREFIX@/include/LegacySupport))
  (js_of_ocaml
   (javascript_files runtime.js))
  (preprocess future_syntax))
