--- lib/handle.c	2023-11-16 18:51:29.000000000 +0800
+++ lib/handle.c	2025-04-14 09:54:37.000000000 +0800
@@ -354,11 +354,7 @@
       block = (struct ntreg_hbin_block *) ((char *) h->addr + blkoff);
       int used;
       seg_len = block_len (h, blkoff, &used);
-/* https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78665 */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wstrict-overflow"
       if (seg_len <= 4 || (seg_len & 3) != 0) {
-#pragma GCC diagnostic pop
         if (is_root || !h->unsafe) {
           SET_ERRNO (ENOTSUP,
                      "%s, the block at 0x%zx size %" PRIu32
