--- server/Makefile	2025-01-20 00:02:37.000000000 +0800
+++ server/Makefile	2025-04-05 20:38:16.000000000 +0800
@@ -78,7 +78,7 @@
 
 OBJS = main.o monitor.o mail.o plugins.o utils.o sysdeps-unix.o log.o
 
-all:	gkrellmd$(BINEXT) $(SERVICE_FILE)
+all:	gkrellmd$(BINEXT)
 
 gkrellmd$(BINEXT): $(OBJS) $(EXTRAOBJS)
 	$(CC) $(OBJS) $(EXTRAOBJS) -o $@ \
@@ -152,7 +152,7 @@
 uninstall-service:
 	$(RM) -f $(SERVICE_DIR)/$(SERVICE_FILE)
 
-install: install_bin install_inc install_man install_cfg install-service
+install: install_bin install_inc install_man install_cfg
 
 install-strip:
 	$(MAKE) STRIP="-s" install
