From 96e6ee1f67d39a1286b01a7d97ac2ba8bda00aca Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 24 Jul 2025 23:14:03 +0800
Subject: [PATCH] Makefile: fix strip for macOS

---
 Makefile | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git Makefile Makefile
index 2d7c18d..fae5468 100644
--- Makefile
+++ Makefile
@@ -20,6 +20,8 @@ OS_TYPE = POSIX
 # The 'OS' environment variable is 'Windows_NT' on Windows systems.
 ifeq ($(OS),Windows_NT)
 	OS_TYPE = WINDOWS
+else
+	UNAME_S := $(shell uname -s)
 endif
 
 # --- Windows Build ---
@@ -40,7 +42,11 @@ else
 	# On POSIX, we link against the installed readline library
 	LIBS = -lcurl -lz -lreadline
 	RM = rm -f
-	STRIP = strip -s
+	ifeq ($(UNAME_S),Darwin)
+		STRIP = strip
+	else
+		STRIP = strip -s
+	endif
 endif
 
 OBJ = $(SRC:.c=.o)
