From 374d0812f58da23a5691e1f53c651ba2a8215eac Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 24 Jul 2025 23:15:08 +0800
Subject: [PATCH] Makefile: needs C99

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git Makefile Makefile
index fae5468..38611b9 100644
--- Makefile
+++ Makefile
@@ -8,7 +8,7 @@ TARGET_NAME = gemini-cli
 SRC_COMMON = gemini-cli.c cJSON.c
 OBJ_COMMON = $(SRC_COMMON:.c=.o)
 # Common compiler and linker flags
-CFLAGS = -Wall -Wextra -g -O2 -I.
+CFLAGS += -std=c99 -Wall -Wextra -g -O2 -I.
 #LDFLAGS =
 
 # --- Platform-Specific Configuration ---
