From 5908b35111114dfba48d5da6a7ab434a2bb14b2c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 16 Oct 2025 12:00:17 +0800
Subject: [PATCH] meson.build: fix finding python and other deps

---
 data/meson.build | 4 ++--
 meson.build      | 3 +--
 2 files changed, 3 insertions(+), 4 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index 0c755c9c..4f50d651 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -47,8 +47,8 @@ endif
 #
 dependency('glib-2.0')
 dependency('gobject-introspection-1.0')
-dependency('gtk4', version: '>=4.18')
-dependency('libadwaita-1', version: '>=1.8')
+dependency('gtk4', version: '>=4.14')
+dependency('libadwaita-1', version: '>=1.5')
 
 #
 # gschema XML file
diff --git a/meson.build b/meson.build
index 0189457c..c71d3d0b 100644
--- a/meson.build
+++ b/meson.build
@@ -22,8 +22,7 @@ domainext = 'info'
 gitrepo = 'https://codeberg.org/' + authornickname + '/' + prettyname
 website = 'https://apps.gnome.org/Komikku/'
 
-python = import('python')
-py_installation = python.find_installation('python3')
+py_installation = import('python').find_installation('@PYTHON@')
 if not py_installation.found()
     error('No valid python3 binary found')
 else
