From 2173d31dfc3b1220dc99e2789a70bed99daca940 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Val=C3=A9ry=20Febvre?= <vfebvre@easter-eggs.com>
Date: Sun, 23 Nov 2025 18:55:30 +0100
Subject: [PATCH] Fix #847

---
 komikku/utils.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/komikku/utils.py b/komikku/utils.py
index 023f4ebe..1e59da68 100644
--- a/komikku/utils.py
+++ b/komikku/utils.py
@@ -135,7 +135,7 @@ def convert_and_resize_image(buffer, width, height, keep_aspect_ratio=True, domi
 
 def folder_size(path, exclude=None):
     if not os.path.exists(path):
-        return 0
+        return '0'
 
     cmd = ['du', '-sh', path]
     if exclude is not None:
