From a7e15b9d4e8fa962c93761caf4fc222a9f7cb912 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Val=C3=A9ry=20Febvre?= <vfebvre@easter-eggs.com>
Date: Mon, 2 Mar 2026 22:03:36 +0100
Subject: [PATCH] [Flatpak] Replace pillow-jxl-plugin python module by jxlpy

Close #876
---

diff --git a/komikku/utils.py b/komikku/utils.py
index 85baebc8..ac299caf 100644
--- a/komikku/utils.py
+++ b/komikku/utils.py
@@ -17,9 +17,9 @@ import subprocess
 import traceback
 
 import gi
-from PIL import Image
-import pillow_jxl  # noqa: F401
+from jxlpy import JXLImagePlugin  # noqa: F401
 import magic
+from PIL import Image
 import requests
 from requests.adapters import HTTPAdapter
 from requests.adapters import TimeoutSauce
diff --git a/requirements.txt b/requirements.txt
index 85e550d3..69470ce0 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -12,8 +12,8 @@ keyring >= 21.6.0
 lxml >= 5.2.1
 natsort
 piexif
-pillow >= 12.1.1
-pillow-jxl-plugin
+pillow >= 11.3.0
+jxlpy
 PyGObject
 PyJWT >= 2.10.1
 pypdf >= 6.7.3
