--- a/komikku/preferences.py
+++ b/komikku/preferences.py	2025-11-23 22:59:41
@@ -64,7 +64,6 @@
     clear_cached_data_actionrow = Gtk.Template.Child('clear_cached_data_actionrow')
     clear_cached_data_button = Gtk.Template.Child('clear_cached_data_button')
     clear_cached_data_on_app_close_switch = Gtk.Template.Child('clear_cached_data_on_app_close_switch')
-    clear_webview_data_actionrow = Gtk.Template.Child('clear_webview_data_actionrow')
     clear_webview_data_button = Gtk.Template.Child('clear_webview_data_button')
     external_servers_modules_switch = Gtk.Template.Child('external_servers_modules_switch')
     credentials_storage_plaintext_fallback_switch = Gtk.Template.Child('credentials_storage_plaintext_fallback_switch')
@@ -502,7 +501,6 @@
 
     def update_webview_data_size(self):
         size = folder_size(get_webview_data_dir(), exclude='cookies.sqlite')
-        self.clear_webview_data_actionrow.set_subtitle(size)
 
 
 @Gtk.Template.from_resource('/info/febvre/Komikku/ui/preferences_servers_languages.ui')
