From 1f3a5b6999eca914c1f241ca2e7634f84b1e0996 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 5 Jun 2025 08:56:20 +0800
Subject: [PATCH] tsPlatform.h: fix powerpc macros

---
 src/libtscore/cpp/tsPlatform.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/libtscore/cpp/tsPlatform.h src/libtscore/cpp/tsPlatform.h
index 982b2f751..515ae32b2 100644
--- src/libtscore/cpp/tsPlatform.h
+++ src/libtscore/cpp/tsPlatform.h
@@ -378,11 +378,11 @@
     #if !defined(TS_SPARC)
         #define TS_SPARC 1
     #endif
-#elif defined(__powerpc64__)
+#elif defined(__powerpc64__) || defined(__ppc64__)
     #if !defined(TS_POWERPC64)
         #define TS_POWERPC64 1
     #endif
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__ppc__)
     #if !defined(TS_POWERPC)
         #define TS_POWERPC 1
     #endif
