From c4c6809227de2f9cbbb7c0a62c127a1a9f673148 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 6 Jun 2025 12:27:48 +0800
Subject: [PATCH] tsSysUtils.cpp: fix for macOS when MACH_TASK_BASIC_INFO
 undefined

---
 src/libtscore/system/tsSysUtils.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git src/libtscore/system/tsSysUtils.cpp src/libtscore/system/tsSysUtils.cpp
index 703b33ac9..9e157c311 100644
--- src/libtscore/system/tsSysUtils.cpp
+++ src/libtscore/system/tsSysUtils.cpp
@@ -284,10 +284,17 @@ size_t ts::GetProcessVirtualSize()
     // macOS implementation.
 
     // Get the virtual memory size using task_info (mach kernel).
+#ifdef MACH_TASK_BASIC_INFO
     ::mach_task_basic_info_data_t taskinfo;
     TS_ZERO(taskinfo);
     ::mach_msg_type_number_t count = MACH_TASK_BASIC_INFO_COUNT;
     const ::kern_return_t status1 = ::task_info(::mach_task_self(), MACH_TASK_BASIC_INFO, ::task_info_t(&taskinfo), &count);
+#else
+    ::task_basic_info_data_t taskinfo;
+    TS_ZERO(taskinfo);
+    ::mach_msg_type_number_t count = TASK_BASIC_INFO_COUNT;
+    const ::kern_return_t status1 = ::task_info(::mach_task_self(), TASK_BASIC_INFO, ::task_info_t(&taskinfo), &count);
+#endif
     if (status1 != KERN_SUCCESS) {
         throw ts::Exception(u"task_info error", errno);
     }
