From b15613117bc5122a5fceb6c7c9ddcb003bdc02a0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 6 Jun 2025 12:40:20 +0800
Subject: [PATCH] tsUDPSocket.h: TS_NO_SSM on macOS

---
 src/libtscore/network/tsUDPSocket.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/libtscore/network/tsUDPSocket.h src/libtscore/network/tsUDPSocket.h
index 882e20655..63f419a20 100644
--- src/libtscore/network/tsUDPSocket.h
+++ src/libtscore/network/tsUDPSocket.h
@@ -19,7 +19,7 @@
 #include "tsReport.h"
 #include "tsMemory.h"
 
-#if defined(DOXYGEN) || defined(TS_OPENBSD) || defined(TS_NETBSD) || defined(TS_DRAGONFLYBSD)
+#if defined(DOXYGEN) || defined(TS_OPENBSD) || defined(TS_NETBSD) || defined(TS_DRAGONFLYBSD) || defined(TS_MAC)
     //!
     //! Defined when the operating system does not support UDP/IP source-specific multicast (SSM).
     //!
