From fa7dc64dccccce1d7ef0272fb4ae2b93024885c5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 6 Jun 2025 13:11:08 +0800
Subject: [PATCH] tsEnvironment.cpp: use correct environ for macOS

---
 src/libtscore/system/tsEnvironment.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git src/libtscore/system/tsEnvironment.cpp src/libtscore/system/tsEnvironment.cpp
index 8effd39a2..f24869f4a 100644
--- src/libtscore/system/tsEnvironment.cpp
+++ src/libtscore/system/tsEnvironment.cpp
@@ -8,7 +8,10 @@
 
 #include "tsEnvironment.h"
 
-#if defined(TS_MAC) || defined(TS_BSD)
+#if defined(TS_MAC)
+    #include <crt_externs.h>
+    #define environ (*_NSGetEnviron())
+#elif defined(TS_BSD)
     extern char **environ; // not defined in public headers
 #endif
 
@@ -215,7 +218,7 @@
 
 #else
 
-    for (char** p = ::environ; *p != nullptr; ++p) {
+    for (char** p = environ; *p != nullptr; ++p) {
         AddNameValue(env, UString::FromUTF8(*p), true);
     }
 
