From 7e78acc03e7dda37cb39c74258e5e66f54620038 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 6 Jun 2025 15:04:50 +0800
Subject: [PATCH] Drop html docs, target broken

---
 doc/Makefile | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git doc/Makefile doc/Makefile
index 290419f31..23c86c227 100644
--- doc/Makefile
+++ doc/Makefile
@@ -221,11 +221,9 @@ INSTALLDOC = $(SYSROOT)$(SYSPREFIX)/share/doc/tsduck
 .PHONY: install install-tools install-devel
 install: install-tools install-devel
 
-install-tools: $(if $(NODOC),,$(BINDOC)/tsduck.html)
+install-tools:
 	install -d -m 755 $(INSTALLDOC)
 	install -m 644 $(ROOTDIR)/CHANGELOG.txt $(ROOTDIR)/LICENSE.txt $(ROOTDIR)/OTHERS.txt $(INSTALLDOC)
-	$(if $(NODOC),,install -m 644 $(BINDOC)/tsduck.html $(INSTALLDOC))
 
-install-devel: $(if $(NODOC),,$(BINDOC)/tsduck-dev.html)
+install-devel:
 	install -d -m 755 $(INSTALLDOC)
-	$(if $(NODOC),,install -m 644 $(BINDOC)/tsduck-dev.html $(INSTALLDOC))
