From 7e6ad71985c20dde6fb6db55720d0efa87bbd3a4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 2 Jul 2025 16:07:33 +0800
Subject: [PATCH] tsconfig: accommodate MacPorts prefix case

---
 src/utils/tsconfig | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git src/utils/tsconfig src/utils/tsconfig
index 70d18518d..ddbc6b540 100755
--- src/utils/tsconfig
+++ src/utils/tsconfig
@@ -113,11 +113,15 @@ cmd_lib() {
     dirname $(ls $(cmd_prefix)/lib*/libtscore$(cmd_so) 2>/dev/null | head -1) 2>/dev/null
 }
 
-brewprefix() {
+macosprefix() {
     local before=$1
     local after=$2
     local prefix=
-    [[ $SYSTEM == Darwin && -n $(which brew 2>/dev/null) ]] && prefix=$(brew --prefix)
+    if [[ $SYSTEM == Darwin && -d /opt/local ]]; then
+        prefix=/opt/local
+    elif [[ $SYSTEM == Darwin && -n $(which brew 2>/dev/null) ]]; then
+        prefix=$(brew --prefix)
+    fi
     [[ -n $prefix ]] && echo "$before$prefix$after"
 }
 
@@ -128,7 +132,7 @@ cmd_cflags() {
     [[ -z $TS_TSCORE && $SYSTEM == Linux && -d /usr/include/PCSC ]] && ! $(no_pcsc) && pcsc="-I/usr/include/PCSC"
     [[ -n $TS_NOSTDCPP ]] && std=""
     [[ -z $TS_TSCORE ]] && inc="-I$(cmd_prefix)/include/tsduck"
-    echo "$inc -I$(cmd_prefix)/include/tscore $(brewprefix -I /include) $pcsc $std"
+    echo "$inc -I$(cmd_prefix)/include/tscore $(macosprefix -I /include) $pcsc $std"
 }
 
 cmd_so() {
@@ -149,7 +153,7 @@ libs_common() {
     [[ -z $TS_TSCORE && $SYSTEM == Linux && -d /usr/include/PCSC ]] && ! $(no_pcsc) && pcsc="-lpcsclite"
     [[ $SYSTEM == Linux ]] && rt="-lrt"
     [[ $SYSTEM == OpenBSD ]] || dl="-ldl"
-    echo "$(brewprefix -L /lib) $pcsc -lpthread $rt $dl -lm -lstdc++"
+    echo "$(macosprefix -L /lib) $pcsc -lpthread $rt $dl -lm -lstdc++"
 }
 
 cmd_libs() {
