From 7e0af443da89f2e8d49f2f13a6031aed3706d69f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 2 Jul 2025 16:15:09 +0800
Subject: [PATCH] tsconfig: extra fixes

---
 src/utils/tsconfig | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git src/utils/tsconfig src/utils/tsconfig
index ddbc6b540..654b39970 100755
--- src/utils/tsconfig
+++ src/utils/tsconfig
@@ -175,9 +175,9 @@ cmd_staticlibs() {
     local other=
     [[ -z $TS_TSCORE ]] && lib1="$(cmd_lib)/libtsduck.a"
     $(no_zlib) || zlib="-lz"
-    $(no_srt) || srt=$(ls /usr/lib/libsrt.a /usr/local/lib/libsrt.a /opt/homebrew/lib/libsrt.a 2>/dev/null | head -1)
-    $(no_rist) || rist=$(ls /usr/lib/librist.a /usr/local/lib/librist.a /opt/homebrew/lib/librist.a 2>/dev/null | head -1)
-    [[ $SYSTEM == Linux ]] && other="$other -latomic"
+    $(no_srt) || srt=$(ls /opt/local/lib/libsrt.a /usr/local/lib/libsrt.a 2>/dev/null | head -1)
+    $(no_rist) || rist=$(ls /opt/local/lib/librist.a /usr/local/lib/librist.a 2>/dev/null | head -1)
+    [[ $SYSTEM == Linux || $SYSTEM == Darwin ]] && other="$other -latomic"
     [[ $SYSTEM == FreeBSD ]] && other="$other -lprocstat"
     [[ $SYSTEM == OpenBSD || $SYSTEM == NetBSD ]] && other="$other -lkvm"
     echo "$lib1 $(cmd_lib)/libtscore.a $srt $rist $zlib $(curl-config --static-libs 2>/dev/null) -ledit $other $(libs_common)"
