From 0bd4c918a0fb34ee74a249bc2c34de6356a706ca Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 6 Jun 2025 13:20:58 +0800
Subject: [PATCH] Link to LegacySupport

---
 scripts/make-config.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git scripts/make-config.sh scripts/make-config.sh
index 526c230c0..0c8390d8c 100755
--- scripts/make-config.sh
+++ scripts/make-config.sh
@@ -794,6 +794,7 @@ LIBTSDUCK_LDLIBS="$LIBTSDUCK_LDLIBS $LDLIBS_PCSC"
 [[ -n $FREEBSD ]] && LIBTSCORE_LDLIBS="$LIBTSCORE_LDLIBS -lprocstat"
 [[ -n $OPENBSD$NETBSD$DRAGONFLYBSD ]] && LIBTSCORE_LDLIBS="$LIBTSCORE_LDLIBS -lkvm"
 [[ -n $LINUX$MACOS ]] && LIBTSCORE_LDLIBS="$LIBTSCORE_LDLIBS -latomic"
+[[ -n $MACOS ]] && LIBTSCORE_LDLIBS="$LIBTSCORE_LDLIBS -lMacportsLegacySupport"
 [[ -z $NOOPENSSL ]] && LIBTSCORE_LDLIBS="$LIBTSCORE_LDLIBS -lcrypto"
 [[ -n $NODTAPI ]] && LIBTSDUCK_CXXFLAGS_INCLUDES="$LIBTSDUCK_CXXFLAGS_INCLUDES -DTS_NO_DTAPI=1"
 [[ -n $NOHIDES ]] && LIBTSDUCK_CXXFLAGS_INCLUDES="$LIBTSDUCK_CXXFLAGS_INCLUDES -DTS_NO_HIDES=1"
