--- a/hw/xquartz/xpr/xprScreen.c	2021-04-29 02:02:54.000000000 +0800
+++ b/hw/xquartz/xpr/xprScreen.c	2023-03-04 05:14:16.000000000 +0800
@@ -333,7 +333,7 @@
     DEBUG_LOG("index=%d depth=%d\n", index, depth);

     if (depth == -1) {
-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060 || defined(__ppc__)
         depth = CGDisplaySamplesPerPixel(kCGDirectMainDisplay) *
                 CGDisplayBitsPerSample(kCGDirectMainDisplay);
 #else

--- a/hw/xquartz/quartzRandR.c	2021-04-29 02:02:54.000000000 +0800
+++ b/hw/xquartz/quartzRandR.c	2023-08-06 08:04:38.000000000 +0800
@@ -66,7 +66,7 @@
 typedef int (*QuartzModeCallback)
     (ScreenPtr, QuartzModeInfoPtr, void *);

-#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060 || defined(__ppc__)

 static long
 getDictLong(CFDictionaryRef dictRef, CFStringRef key)

