--- CMakeLists.txt	2022-03-14 10:37:57.000000000 +0800
+++ CMakeLists.txt	2025-04-07 16:34:27.000000000 +0800
@@ -35,18 +35,18 @@
 install(TARGETS dsd DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 # man page
-find_program(HELP2MAN_FOUND help2man)
-if (HELP2MAN_FOUND)
-    add_custom_command(TARGET dsd POST_BUILD
-        COMMAND help2man
-        ARGS -n "Digital Speech Decoder"
-            --version-string=${GIT_TAG}
-            -o ${CMAKE_CURRENT_BINARY_DIR}/dsd.1
-            --no-info
-            $<TARGET_FILE:dsd> 
-    )
-    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dsd.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
-endif()
+#find_program(HELP2MAN_FOUND help2man)
+#if (HELP2MAN_FOUND)
+#    add_custom_command(TARGET dsd POST_BUILD
+#        COMMAND help2man
+#        ARGS -n "Digital Speech Decoder"
+#            --version-string=${GIT_TAG}
+#            -o ${CMAKE_CURRENT_BINARY_DIR}/dsd.1
+#            --no-info
+#            $<TARGET_FILE:dsd> 
+#    )
+#    install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dsd.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1)
+#endif()
 
 # uninstall target
 configure_file(
