--- Makefile	2025-08-08 23:51:24.000000000 +0800
+++ Makefile	2025-08-24 21:58:49.000000000 +0800
@@ -75,7 +75,7 @@
 # ==============================================================================
 # Compiler and Linker Flags
 # ==============================================================================
-COMMON_CFLAGS := -Wall -Wextra -Iinclude -fPIC -I/opt/homebrew/opt/libmagic/include -I/opt/homebrew/opt/libtar/include -I/opt/homebrew/opt/libzip/include -DFAT_VERSION=\"$(VERSION)\" -DINSTALL_PREFIX=\"$(PREFIX)\"
+COMMON_CFLAGS := -Wall -Wextra -Iinclude -fPIC -DFAT_VERSION=\"$(VERSION)\" -DINSTALL_PREFIX=\"$(PREFIX)\"
 
 ifeq ($(DEBUG), 1)
 	CFLAGS := $(COMMON_CFLAGS) -g3 -O0 $(EXTRA_CFLAGS)
@@ -84,7 +84,7 @@
 	CFLAGS := $(COMMON_CFLAGS) -O2 $(STRIP_FLAG) $(EXTRA_CFLAGS)
 endif
 
-LDFLAGS := $(LDFLAGS_NCURSES) -lmagic -L/opt/homebrew/opt/libmagic/lib $(LDFLAGS_PLATFORM) -L$(LIB_DIR) -lfat_utils -lm -lzip -L/opt/homebrew/opt/libzip/lib
+LDFLAGS := $(LDFLAGS_NCURSES) -lmagic $(LDFLAGS_PLATFORM) -L$(LIB_DIR) -lfat_utils -lm -lzip
 ifeq ($(detected_OS),Windows)
     LDFLAGS += -lgnurx
 endif
@@ -141,7 +141,7 @@
 
 $(SHARED_LIB): $(SHARED_LIB_OBJ)
 	@mkdir -p $(LIB_DIR)
-	$(CC) -shared $(SHARED_LIB_OBJ) -o $(SHARED_LIB)
+	$(CC) -shared $(SHARED_LIB_OBJ) -o $(SHARED_LIB) -Wl,-install_name,$(PREFIX)/$(SHARED_LIB)
 
 $(TARGET)$(TARGET_EXT): $(OBJECTS) $(SHARED_LIB)
 	@mkdir -p $(BIN_DIR)
@@ -153,10 +153,10 @@
 	$(CC) $(CFLAGS) -c $< -o $@
 
 $(PLUGIN_DIR)/tar_plugin$(SHARED_LIB_EXT): $(PLUGIN_DIR)/tar_plugin.c | $(SHARED_LIB)
-	$(CC) -shared $(CFLAGS) $< -o $@ -ltar -L/opt/homebrew/opt/libtar/lib -L$(LIB_DIR) -lfat_utils -Wl,-rpath,'$$ORIGIN/../../lib'
+	$(CC) -shared $(CFLAGS) $< -o $@ -ltar -L$(LIB_DIR) -lfat_utils -Wl,-install_name,$(PREFIX)/lib/fat/plugins/tar_plugin$(SHARED_LIB_EXT) -Wl,-rpath,'$$ORIGIN/../../lib'
 
 $(PLUGIN_DIR)/zip_plugin$(SHARED_LIB_EXT): $(PLUGIN_DIR)/zip_plugin.c | $(SHARED_LIB)
-	$(CC) -shared $(CFLAGS) $< -o $@ -lzip -I/opt/homebrew/opt/libzip/include -L/opt/homebrew/opt/libzip/lib -L$(LIB_DIR) -lfat_utils -Wl,-rpath,'$$ORIGIN/../../lib'
+	$(CC) -shared $(CFLAGS) $< -o $@ -lzip -L$(LIB_DIR) -lfat_utils -Wl,-install_name,$(PREFIX)/lib/fat/plugins/zip_plugin$(SHARED_LIB_EXT) -Wl,-rpath,'$$ORIGIN/../../lib'
 
 clean:
 	@$(CLEAN_CMD) obj bin lib
