--- src/EditresCom.c	2026-01-25 07:39:21.000000000 +0800
+++ src/EditresCom.c	2026-01-27 08:28:20.000000000 +0800
@@ -1631,7 +1631,7 @@
      * make sure that they are inserted in the list from parent -> child
      */
     for (i--, temp = w; temp != NULL; temp = XtParent(temp), i--)
-    widget_list[i] = (CARD64)temp;
+    widget_list[i] = (CARD64)(uintptr_t)temp;
 
     _XEditResPut16(stream, num_widgets);	/* insert number of widgets */
     for (i = 0; i < num_widgets; i++)		/* insert Widgets themselves */

--- include/X11/Xmu/EditresP.h	2026-01-25 07:39:21.000000000 +0800
+++ include/X11/Xmu/EditresP.h	2026-01-27 08:27:20.000000000 +0800
@@ -265,6 +265,7 @@
 
 ************************************************************/
 
+#include <stdint.h>
 #include <X11/Intrinsic.h>
 #include <X11/Xfuncproto.h>
 #include <X11/Xmd.h>
@@ -292,11 +293,7 @@
 #define EDITRES_CLIENT_VALUE "EditresClientVal"
 #define EDITRES_PROTOCOL_ATOM "EditresProtocol"
 
-#ifdef LONG64
-#define ID2WIDGET(X) ((Widget)(void *)((X)))
-#else
-#define ID2WIDGET(X) ((Widget)(void *)((X) & 0xffffffff))
-#endif
+#define ID2WIDGET(X) ((Widget)(void *)(uintptr_t)((X)))
 
 typedef enum {
   SendWidgetTree = 0,
