--- GDE/CLUSTAL/clustalv.c	2016-05-02 19:19:07.000000000 +0800
+++ GDE/CLUSTAL/clustalv.c	2026-05-22 04:08:34.000000000 +0800
@@ -125,7 +125,8 @@
 */
 }
 
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-int"
 
 main(int argc,char **argv)
 {
@@ -158,3 +159,5 @@
 		main_menu();
 	}
 }
+
+#pragma GCC diagnostic pop

--- GDE/CLUSTAL/sequence.c	2016-08-22 18:27:49.000000000 +0800
+++ GDE/CLUSTAL/sequence.c	2026-05-22 04:11:09.000000000 +0800
@@ -36,8 +36,14 @@
 extern int *seqlen_array;
 extern char **names,**titles;
 extern char **seq_array;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-int"
+
 extern profile1_empty, profile2_empty;
 
+#pragma GCC diagnostic pop
+
 char *amino_acid_codes   =    "ABCDEFGHIKLMNPQRSTUVWXYZ-"; 
 char *amino_acid_order   =    "XCSTPAGNDEQHRKMILVFYW";
 char *nucleic_acid_order = 	  "NACGTU";
