--- READSEQ/readseq.c	2016-05-02 19:19:03.000000000 +0800
+++ READSEQ/readseq.c	2026-05-22 03:59:10.000000000 +0800
@@ -169,6 +169,7 @@
 
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <string.h>
 #define __NO_CTYPE
 #include <ctype.h>
@@ -711,6 +712,9 @@
 #else
 #define Exit(a)   exit(a)
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-int"
+
 main( int argc, char *argv[])
 #endif
 {
@@ -1145,4 +1149,4 @@
   exit_main(0);
 }
 
-
+#pragma GCC diagnostic pop
