--- GDE/FASTDNAML/fastDNAml.h	2016-05-02 19:19:07.000000000 +0800
+++ GDE/FASTDNAML/fastDNAml.h	2026-05-22 05:22:42.000000000 +0800
@@ -216,7 +216,7 @@
 
 void  exit();
 
-#if  ANSI || MALLOC_VOID
+#if  ANSI || MALLOC_VOID || defined(__APPLE__)
    void *malloc();
 #else
    char *malloc();

--- GDE/FASTDNAML/fastDNAml.c	2016-05-02 19:19:07.000000000 +0800
+++ GDE/FASTDNAML/fastDNAml.c	2026-05-22 05:23:18.000000000 +0800
@@ -203,7 +203,11 @@
 #endif
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
 #include <math.h>
+
 #include "fastDNAml.h"  /*  Requires version 1.2  */
 
 #if Master || Slave
@@ -4461,6 +4465,8 @@
   } /* freopen_pid */
 #endif
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-int"
 
 boolean  showBestTrees (bestlist *bt, tree *tr, analdef *adef, FILE *treefile)
   { /* showBestTrees */
@@ -4480,7 +4486,6 @@
     return TRUE;
   } /* showBestTrees */
 
-
 boolean cmpBestTrees (bestlist *bt, tree *tr)
   { /* cmpBestTrees */
     double  sum, sum2, sd, temp, wtemp, bestscore;
@@ -4954,3 +4959,5 @@
       return 0;
 #   endif
   } /* DNA Maximum Likelihood */
+
+#pragma GCC diagnostic pop
