--- GDE/AxML/axml.c	2016-05-02 19:19:06.000000000 +0800
+++ GDE/AxML/axml.c	2026-05-22 05:03:35.000000000 +0800
@@ -11,6 +11,8 @@
 #include <time.h> 
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
+
 double accTime = 0.0;
 double globTime = 0.0;
 struct timeval ttime;
@@ -2289,6 +2291,8 @@
     return  TRUE;
   } /* setupnodex */
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wimplicit-int"
 
 xarray *getxtip (nodeptr p)
   { /* getxtip */
@@ -5622,3 +5626,5 @@
       return 0;
 #   endif
   } /* DNA Maximum Likelihood */
+
+#pragma GCC diagnostic pop
