--- Makefile
+++ Makefile	2026-05-22 03:39:50.000000000 +0800
@@ -119,12 +119,7 @@
  endif
 endif
 
-ifeq ($(USE_CLANG),1)
-# accept all clang versions:
 ALLOWED_COMPILER_VERSIONS=$(COMPILER_VERSION)
-else
-ALLOWED_COMPILER_VERSIONS=$(ALLOWED_gcc_VERSIONS)
-endif
 
 COMPILER_VERSION_ALLOWED=$(strip $(subst ___,,$(foreach version,$(ALLOWED_COMPILER_VERSIONS),$(findstring ___$(version)___,___$(COMPILER_VERSION)___))))
 
@@ -191,13 +186,7 @@
 
 ifeq ($(DEBUG),0)
 	dflags := -DNDEBUG# defines
-	ifeq ($(USE_CLANG),1)
-		cflags := -O3# compiler flags (C and C++)
-	else
-		cflags := -O3# compiler flags (C and C++)
-		lflags += -O2# linker flags
-		clflags += -Wl,-O2# passthrough linker flags
-	endif
+	cflags := -O3# compiler flags (C and C++)
 endif
 
 ifeq ($(DEBIAN),1)
@@ -489,7 +478,7 @@
 #---------------------- include symbols?
 
 ifeq ($(TRACESYM),1)
-	ifeq ($(USE_CLANG),0)
+	ifeq ($(DARWIN),0)
 		cflags  += -rdynamic
 		lflags  += --export-dynamic
 		clflags += -rdynamic -Wl,--export-dynamic
@@ -527,7 +516,7 @@
 endif
 
 ifeq ('$(USE_GCC_47_OR_HIGHER)','yes')
-cxxflags += -std=gnu++11# see also TEMPLATES/cxxforward.h@USE_Cxx11
+cxxflags += -std=gnu++11 -D_GLIBCXX_USE_CXX11_ABI=0 # see also TEMPLATES/cxxforward.h@USE_Cxx11
 else
 # only use for gcc versions between 4.3 and <4.7 (4.7++ adds -Wc++11-compat above)
 HAVE_GNUPP0X=`SOURCE_TOOLS/requireVersion.pl 4.3 $(COMPILER_VERSION)`
