--- GDE/MUSCLE/Makefile	2016-05-02 19:19:05.000000000 +0800
+++ GDE/MUSCLE/Makefile	2026-05-22 06:10:26.000000000 +0800
@@ -4,7 +4,7 @@
 .PHONY: all clean
 
 CXX=$(A_CXX)
-CXXFLAGS=-c -O3 -msse2 -mfpmath=sse -D_FILE_OFFSET_BITS=64 -DNDEBUG=1
+CXXFLAGS=-c -O3 -D_FILE_OFFSET_BITS=64 -DNDEBUG=1
 LDFLAGS= -g
 
 OBJDIR=obj

--- GDE/PHYML20130708/phyml/configure.fallback	2016-05-02 19:19:09.000000000 +0800
+++ GDE/PHYML20130708/phyml/configure.fallback	2026-05-22 06:04:45.000000000 +0800
@@ -4772,7 +4772,7 @@
 if test "x$enable_debug" = "xyes"; then :
   CFLAGS="-ansi -pedantic -Wall -std=c99 -O0 -g"
 else
-  CFLAGS="-Wall -O2 -msse -fomit-frame-pointer -funroll-loops ${ARCH_flag}"
+  CFLAGS="-Wall -O2 -fomit-frame-pointer -funroll-loops ${ARCH_flag}"
 fi
 
 # Check whether --enable-gprof was given.
@@ -4784,7 +4784,7 @@
   CFLAGS="-ansi -pedantic -Wall -std=c99 -O0 -g -pg"
 else
   if test "x$enable_debug" = "xno"; then :
-  CFLAGS="-Wall -O2 -msse -fomit-frame-pointer -funroll-loops ${ARCH_flag}"
+  CFLAGS="-Wall -O2 -fomit-frame-pointer -funroll-loops ${ARCH_flag}"
 fi
 fi
 
