--- opal/include/opal/sys/powerpc/atomic.h
+++ opal/include/opal/sys/powerpc/atomic.h	2025-01-27 07:50:01.000000000 +0800
@@ -357,6 +357,7 @@
 
 #if OPAL_GCC_INLINE_ASSEMBLY
 
+#if OPAL_ASM_SUPPORT_64BIT
 /* these two functions aren't inlined in the non-gcc case because then
    there would be two function calls (since neither cmpset_64 nor
    atomic_?mb can be inlined).  Instead, we "inline" them by hand in
@@ -378,7 +379,7 @@
     opal_atomic_wmb();
     return opal_atomic_compare_exchange_strong_64 (addr, oldval, newval);
 }
-
+#endif
 
 #define OPAL_ATOMIC_POWERPC_DEFINE_ATOMIC_32(type, instr)               \
 static inline int32_t opal_atomic_fetch_ ## type ## _32(volatile int32_t* v, int val) \
