--- a/meson.build	2025-11-28 04:03:44.000000000 +0800
+++ b/meson.build	2026-08-21 00:38:10.000000000 +0800
@@ -133,7 +133,7 @@
   ['libtiff', 'dependency', 'libtiff-4', [], ['image/tiff', 'image/tiff-fx']],
   ['libpng', 'dependency', 'libpng', [], ['image/png', 'image/x-png']],
   ['libjpeg', 'dependency', 'libturbojpeg', [], ['image/jpeg', 'image/jpg', 'image/pjpeg']],
-  ['librsvg', 'dependency', 'librsvg-2.0', '>= 2.44', ['image/svg+xml']],
+  ['librsvg', 'dependency', 'librsvg-2.0', '>= 2.40', ['image/svg+xml']],
   ['libnsgif', 'dependency', 'libnsgif', '1.0.0', ['image/gif']],
   ['libnsbmp', 'dependency', 'libnsbmp', [], ['image/bmp', 'image/x-bmp']],
   ['libheif', 'dependency', 'libheif', '>= 1.13.0', ['image/heif', 'image/avif']],
