--- a/src/glsl/meson.build	2025-12-22 19:47:15.000000000 +0800
+++ b/src/glsl/meson.build	2025-12-28 02:18:53.000000000 +0800
@@ -30,6 +30,7 @@
 elif not glslang_req.disabled()
 
   glslang_deps = [
+    cxx.find_library('glslang', required: false),
     cxx.find_library('glslang-default-resource-limits', required: false)
   ]
 
@@ -44,7 +45,7 @@
     static   = arg[0]
     required = arg[1]
 
-    spirv = cxx.find_library('SPIRV', required: required, static: static)
+    spirv = cxx.find_library('SPIRV', required: required)
 
     if not spirv.found()
       continue
