--- modules/avcapture/CMakeLists.txt	2025-03-26 15:24:39.000000000 +0800
+++ modules/avcapture/CMakeLists.txt	2025-04-05 18:20:48.000000000 +0800
@@ -1,8 +1,7 @@
 project(avcapture)
 
-if(NOT ${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-  return()
-endif()
+# Broken, uses non-existing framework
+return()
 
 list(APPEND MODULES_DETECTED ${PROJECT_NAME})
 set(MODULES_DETECTED ${MODULES_DETECTED} PARENT_SCOPE)
