From c57389393ce5d99d3695629f6096e551fe18a7d2 Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:31:01 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_variants_conv.ml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ppx_variants_conv.ml b/src/ppx_variants_conv.ml
index 5882f8b..d82d0c4 100644
--- src/ppx_variants_conv.ml
+++ src/ppx_variants_conv.ml
@@ -464,7 +464,7 @@ module Gen_str = struct
           let body =
             if Variant_constructor.is_gadt v
             then locally_abstract_match loc cases ~variant_ty
-            else pexp_function ~loc cases
+            else pexp_function_cases ~loc cases
           in
           [%stri let [%p pvar ~loc name] = [%e body] [@@warning "-4"]]
         in
@@ -677,7 +677,7 @@ module Gen_str = struct
     let body =
       if List.exists variants ~f:Variant_constructor.is_gadt
       then locally_abstract_match loc cases ~variant_ty
-      else pexp_function ~loc cases
+      else pexp_function_cases ~loc cases
     in
     [%stri let to_rank = [%e body]]
   ;;
@@ -689,7 +689,7 @@ module Gen_str = struct
     let body =
       if List.exists variants ~f:Variant_constructor.is_gadt
       then locally_abstract_match loc cases ~variant_ty
-      else pexp_function ~loc cases
+      else pexp_function_cases ~loc cases
     in
     [%stri let to_name = [%e body]]
   ;;
