From 0b204f83e9907456deb1a44ab36f08a8362fd6f0 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Fri, 5 Jul 2019 20:26:06 +0100
Subject: [PATCH] tapi - add ppc and ppc64 archs.

---
 include/tapi/Core/Architecture.def | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git tapi/include/tapi/Core/Architecture.def tapi/include/tapi/Core/Architecture.def
index 32b4bb5..37df40f 100644
--- tapi/include/tapi/Core/Architecture.def
+++ tapi/include/tapi/Core/Architecture.def
@@ -4,6 +4,14 @@
 #define ARCHINFO(arch)
 #endif
 
+#ifdef SUPPORT_ARCH_PPC
+ARCHINFO(ppc, MachO::CPU_TYPE_POWERPC, MachO::CPU_SUBTYPE_POWERPC_ALL)
+#endif
+
+#ifdef SUPPORT_ARCH_PPC64
+ARCHINFO(ppc64, MachO::CPU_TYPE_POWERPC64, MachO::CPU_SUBTYPE_POWERPC_ALL)
+#endif
+
 ///
 /// X86 architectures sorted by cpu type and sub type id.
 ///

--- tapi/CMakeLists.txt	2021-10-06 13:39:00.000000000 +0800
+++ tapi/CMakeLists.txt	2025-01-11 08:15:30.000000000 +0800
@@ -38,11 +38,11 @@
 if(DEFINED ENV{RC_SUPPORTED_ARCHS})
   string(REPLACE " " ";" TAPI_SUPPORTED_ARCHS $ENV{RC_SUPPORTED_ARCHS})
 elseif(NOT DEFINED TAPI_SUPPORTED_ARCHS)
-  set(TAPI_SUPPORTED_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64)
+  set(TAPI_SUPPORTED_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 ppc ppc64)
 endif()
 message(STATUS "Supported Architectures: ${TAPI_SUPPORTED_ARCHS}")
 
-set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64)
+set(KNOWN_ARCHS i386 x86_64 x86_64h armv4t armv6 armv5 armv7 armv7s armv7k armv6m armv7m armv7em arm64 ppc ppc64)
 
 set (CONFIG_FILE ${CMAKE_CURRENT_BINARY_DIR}/include/tapi/Core/ArchitectureConfig.h)
 file(WRITE ${CONFIG_FILE} "#ifndef TAPI_CORE_ARCHITECTURE_CONFIG_H\n")
