From 721b85c0fd3e5a39e5b4cb837ab444c6caefb9d6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:08:05 +0800
Subject: [PATCH] random.hxx: fix macOS version macros

---
 include/vigra/random.hxx | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git include/vigra/random.hxx include/vigra/random.hxx
index e001fa81..69bc4649 100644
--- include/vigra/random.hxx
+++ include/vigra/random.hxx
@@ -153,11 +153,11 @@ void seed(RandomSeedTag, RandomState<EngineTag> & engine)
 
 #ifdef __APPLE__
     seedData.push_back(static_cast<UInt32>(getpid()));
-  #if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_12
+  #if MAC_OS_X_VERSION_MAX_ALLOWED >= 101200
     uint64_t tid64;
     pthread_threadid_np(NULL, &tid64);
     seedData.push_back(static_cast<UInt32>(tid64));
-  #elif defined(SYS_thread_selfid) && (__MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_6)
+  #elif defined(SYS_thread_selfid) && (MAC_OS_X_VERSION_MAX_ALLOWED >= 1060)
     // SYS_thread_selfid was introduced in MacOS 10.6
     seedData.push_back(static_cast<UInt32>(syscall(SYS_thread_selfid)));
   #elif defined(SYS_gettid)
-- 
2.48.0

