From 59bb5a2f817c37ac5961ba9337180825a29210fb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 23 Apr 2025 16:18:41 +0800
Subject: [PATCH] CMakeLists: fix finding boost_python

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git CMakeLists.txt CMakeLists.txt
index 493cbd78..5e5c51db 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -162,6 +162,7 @@ ENDIF()
 
 IF(WITH_VIGRANUMPY)
     set(WITH_BOOST "ON")
+    set(WITH_BOOST_COMPONENTS ${WITH_BOOST_COMPONENTS} python)
 ENDIF()
 
 IF(WITH_BOOST_THREAD)
-- 
2.48.0

