From 32049ac87f869d1b1da3be74d4460ea8ee1c5a0e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 12 Jan 2026 01:05:06 +0800
Subject: [PATCH 02/10] Revert 'suppress IsMenuKeyEvent: in stderr'

---
 RController.m | 44 ++++++++++++++------------------------------
 1 file changed, 14 insertions(+), 30 deletions(-)

diff --git a/RController.m b/RController.m
index fe87c07..c38f1f9 100644
--- a/RController.m
+++ b/RController.m
@@ -2123,24 +2123,6 @@ issuing the newline.
 }
 */
 
-/* A bad hack to remove output from the stream - it is used to remove
-   stderr output caused by buggy Apple code that we have no control over.
-   It is assumed that s is allocated so if a replacement is necessary,
-   it will be released and a new string allocated.
- */
-static NSString *fixBuggyOutput(NSString *s, NSString *regExp) {
-	NSRange r = [s rangeOfString:regExp options:NSRegularExpressionSearch];
-	if (r.location != NSNotFound) {
-		NSString *os = s;
-		NSUInteger start, end;
-		[s getLineStart:&start end:&end contentsEnd:0 forRange:r];
-		s = [os stringByReplacingCharactersInRange:NSMakeRange(start, end-start) withString:@""];
-		[s retain];
-		[os release];
-        }
-	return s;
-}
-
 /* This function is used by two threads to write  stderr and/or stdout to the console
 length: -1 = the string is null-terminated
 outputType: 0 = stdout, 1 = stderr, 2 = stdout/err as root
@@ -2150,18 +2132,20 @@ outputType: 0 = stdout, 1 = stderr, 2 = stdout/err as root
 	NSColor *color=(outputType==0)?[consoleColors objectAtIndex:iStdoutColor]:((outputType==1)?[consoleColors objectAtIndex:iStderrColor]:[consoleColors objectAtIndex:iRootColor]);
 	if (len>=0 && buf[len]!=0) buf[len]=0; /* this MAY be dangerous ... */
 	NSString *s = [[NSString alloc] initWithUTF8String:buf];
-	if (outputType == 1) {
-		// FIXME horrible, horrible hack - this warning seems to be a macOS bug, because
-		// we never create NSPopoverTouchBarItemButton, so we suppress it in the console
-		// by removing a line with that content
-		s = fixBuggyOutput(s, @"Warning: Expected min height of view: .* to be less than or equal to ");
-		/* bug in Monterey and up */
-		s = fixBuggyOutput(s, @"IsMenuKeyEvent: found no unichar data in event; retranslated without deadkeys to produce");
-		/* bug in Sequoia(?) and up (note: we don't use one regex since fixBuggyOutput
-		   only removes one line at a time so we don't want to risk getting both in one s) */
-		s = fixBuggyOutput(s, @"\\[IMKClient subclass\\]: chose ");
-		s = fixBuggyOutput(s, @"\\[IMKInputSession subclass\\]: chose ");
-	}
+    if (outputType == 1) {
+        // FIXME horrible, horrible hack - this warning seems to be a macOS bug, because
+        // we never create NSPopoverTouchBarItemButton, so we suppress it in the console
+        // by removing a line with that content
+        NSRange r = [s rangeOfString:@"Warning: Expected min height of view: .* to be less than or equal to " options:NSRegularExpressionSearch];
+        if (r.location != NSNotFound) {
+            NSString *os = s;
+            NSUInteger start, end;
+            [s getLineStart:&start end:&end contentsEnd:0 forRange:r];
+            s = [os stringByReplacingCharactersInRange:NSMakeRange(start, end-start) withString:@""];
+            [s retain];
+            [os release];
+        }
+    }
 	[self flushROutput];
 	[self writeConsoleDirectly:s withColor:color];
 	[s release];
-- 
2.52.0

