From c75ac28a3b58eab970b66f7421453a77edbbdfc4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Jan 2025 06:50:17 +0800
Subject: [PATCH 08/10] RDocumentWinCtrl.m: fix compilation

---
 RDocumentWinCtrl.m | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/RDocumentWinCtrl.m b/RDocumentWinCtrl.m
index 45059c9..2d5c3fb 100644
--- a/RDocumentWinCtrl.m
+++ b/RDocumentWinCtrl.m
@@ -410,17 +410,6 @@ NSInteger _alphabeticSort(id string1, id string2, void *reverse)
 	
 	// Needed for showing tooltips of folded items
 	[[self window] setAcceptsMouseMovedEvents:YES];
-    
-	// FIXME: this is a hack for layout issues in Big Sur
-	// by forcing re-size we force layout to be updated - is there a better way?
-	NSRect clRect = [[self window] contentLayoutRect];
-	clRect.size.width += 1;
-	[[self window] setContentSize:clRect.size];
-	clRect.size.width -= 1;
-	[[self window] setContentSize:clRect.size];
-
-	// Make the text view fist responder so the user can start typing
-	[[self window] makeFirstResponder:textView];
 
 	SLog(@" - windowDidLoad is done");
 
-- 
2.52.0

