From c103312d9ca53eb44ee78c0c8d6d84dc29f61b88 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Jan 2025 06:56:51 +0800
Subject: [PATCH 09/10] RScriptEditorTextStorage.m: fix compilation

---
 RScriptEditorTextStorage.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/RScriptEditorTextStorage.m b/RScriptEditorTextStorage.m
index 65d9844..869ea72 100644
--- a/RScriptEditorTextStorage.m
+++ b/RScriptEditorTextStorage.m
@@ -350,7 +350,7 @@ static SEL _getlSel;
 
 }
 
-- (void)edited:(NSTextStorageEditActions)mask range:(NSRange)oldRange changeInLength:(NSInteger)lengthChange
+- (void)edited:(NSUInteger)mask range:(NSRange)oldRange changeInLength:(NSInteger)lengthChange
 {
 
 	if(foldedCounter && mask == NSTextStorageEditedCharacters) {
-- 
2.52.0

