From b5d185ce6e65f0a2b09c19eba32a18f9bdd21e63 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 3 Apr 2025 18:40:06 +0800
Subject: [PATCH] picosocks.c: improve fallback code

---
 picoquic/picosocks.c | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git picoquic/picosocks.c picoquic/picosocks.c
index 03b97d10..3658d1f7 100644
--- picoquic/picosocks.c
+++ picoquic/picosocks.c
@@ -448,7 +448,11 @@ void picoquic_socks_cmsg_parse(
                 }
             }
 #endif
-            else if ((cmsg->cmsg_type == IP_TOS || cmsg->cmsg_type == IP_RECVTOS) && cmsg->cmsg_len > 0) {
+            else if ((cmsg->cmsg_type == IP_TOS
+#ifdef IP_RECVTOS
+                || cmsg->cmsg_type == IP_RECVTOS
+#endif
+                ) && cmsg->cmsg_len > 0) {
                 if (received_ecn != NULL) {
                     *received_ecn = *((unsigned char*)CMSG_DATA(cmsg));
                 }
@@ -642,7 +646,13 @@ void picoquic_socks_cmsg_format(
 #else 
             /* The IP_PKTINFO structure is not defined on BSD */
             struct in_addr* pktinfo = (struct in_addr*)cmsg_format_header_return_data_ptr(msg, &last_cmsg,
-                &control_length, IPPROTO_IP, IP_SENDSRCADDR, sizeof(struct in_addr));
+                &control_length, IPPROTO_IP,
+#ifdef IP_SENDSRCADDR
+                IP_SENDSRCADDR
+#else
+                IP_RECVDSTADDR
+#endif
+                , sizeof(struct in_addr));
             if (pktinfo != NULL) {
                 pktinfo->s_addr = ((struct sockaddr_in*)addr_from)->sin_addr.s_addr;
             }
