--- src/Makefile.OCaml	2024-11-05 21:22:18.000000000 +0800
+++ src/Makefile.OCaml	2025-10-20 12:37:46.000000000 +0800
@@ -70,21 +70,6 @@
 	$(info GUI library lablgtk not found. GTK GUI will not be built.)
 endif
 
-.PHONY: macuimaybe
-ifeq ($(OSARCH), Darwin)
-  # If XCode is not installed, xcodebuild is just a placeholder telling that XCode is not installed
-  # and any invocation of xcodebuild results in a non 0 exit code.
-  ifeq ($(shell xcodebuild -version > /dev/null; echo $$?), 0)
-    macuimaybe: macui
-  else
-    macuimaybe:
-	$(info Not building macOS native GUI because XCode is not installed.)
-  endif
-else
-  macuimaybe:
-	$(info Not on macOS. macOS native GUI will not be built.)
-endif
-
 ####################################################################
 ### Default parameters
 
@@ -300,12 +285,6 @@
 
 UIMACDIR=uimac
 
-ifeq ($(OSARCH),Darwin)
-ifeq ($(strip $(XCODEFLAGS)),)
-  XCODEFLAGS=-arch $(shell uname -m)  ## Prevent Xcode from trying to build universal binaries by default
-endif
-endif
-
 # Note: The OCaml library names changed starting with OCaml 5.1.0
 .PHONY: macexecutable
 macexecutable:

--- src/Makefile	2024-11-05 21:22:18.000000000 +0800
+++ src/Makefile	2025-10-20 12:38:21.000000000 +0800
@@ -16,7 +16,7 @@
 # Building installation instructions
 
 .PHONY: all
-all: tui guimaybe macuimaybe fsmonitor
+all: tui guimaybe fsmonitor
 
 ########################################################################
 ## Miscellaneous developer-only switches
