--- src/Makefile.OCaml	2025-11-04 20:21:28.000000000 +0800
+++ src/Makefile.OCaml	2025-12-24 03:34:45.000000000 +0800
@@ -7,7 +7,7 @@
 # (d)make and NMAKE. Do not make any changes in this file unless you are
 # certain that the changes do not break this portability.
 
-all: tui guimaybe macuimaybe fsmonitor
+all: tui guimaybe fsmonitor
 .PHONY: all # This .PHONY definition must come after the first target,
             # so that make implementations which don't support .PHONY
             # don't treat it as the default target (which in turn

--- src/make_tools.ml	2025-11-04 20:21:28.000000000 +0800
+++ src/make_tools.ml	2025-12-24 03:37:31.000000000 +0800
@@ -93,15 +93,8 @@
 
 let build_macGUI =
   if osarch_macos then begin
-    (* If XCode is not installed, xcodebuild is just a placeholder telling
-       that XCode is not installed and any invocation of xcodebuild results
-       in a non-0 exit code. *)
-    if Sys.command "xcodebuild -version > /dev/null" = 0 then
-      true
-    else begin
-      info "Not building macOS native GUI because XCode is not installed.";
-      false
-    end
+    info "Cocoa GUI is not supported.";
+    false
   end else begin
     info "Not on macOS. macOS native GUI will not be built.";
     false
