From faeb35663f23df47241898ff72117866b7eed23a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 18 Mar 2025 07:03:46 +0800
Subject: [PATCH] sexp.h: fix inttypes include

---
 include/sexpp/sexp.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git include/sexpp/sexp.h include/sexpp/sexp.h
index be9cf00..71ec80a 100644
--- include/sexpp/sexp.h
+++ include/sexpp/sexp.h
@@ -28,7 +28,11 @@
 
 #pragma once
 
-#include <inttypes.h>
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
+#include <cinttypes>
 #include <climits>
 #include <limits>
 #include <cctype>
-- 
2.48.0

