--- src/postconf/extract_cfg.sh.orig	2018-02-25 08:44:09.000000000 +0800
+++ src/postconf/extract_cfg.sh	2025-10-29 10:50:37.000000000 +0800
@@ -39,7 +39,7 @@
 #--
 
 # In case not installed.
-m4 </dev/null || exit 1
+gm4 </dev/null || exit 1
 
 # Flags to add db_common parameter names.
 add_legacy_sql_query_params=
@@ -74,7 +74,7 @@
 EOF
 # Convert selected C macro definitions into m4 macro definitions.
 sed 's/^#define[ 	]*\([DICT_MC_NAME_A-Za-z0-9_]*\)[ 	]*\("[^"]*"\)/define(`\1'"'"',`\2'"'"')/' "$@"
-) | m4 | awk -F '// { print $2 }' | (
+) | gm4 | awk -F '// { print $2 }' | (
 test -n "$add_domain_param" && {
 cat <<EOF
 "domain"
