From 65dda45b3011cc1bc2ce138e595142c6d5bc7b7f Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Fri, 27 Jan 2023 15:20:22 -0800
Subject: [PATCH 08/18] darwin: Locate libexpat with cc.find_library() rather
 than dependency()

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 src/meson.build | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 984e34a..ddd553d 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -29,7 +29,12 @@ wayland_util_dep = declare_dependency(
 if get_option('scanner')
 	# wayland-scanner
 
-	scanner_deps = [ dependency('expat') ]
+	if host_machine.system() == 'darwin'
+		scanner_deps = [ cc.find_library('expat') ]
+	else
+		scanner_deps = [ dependency('expat') ]
+	endif
+
 	scanner_args = [ '-include', 'config.h' ]
 
 	if get_option('dtd_validation')
-- 
2.24.3 (Apple Git-128)

