From b43ae6aa20a6ab2d7a54ff82578df9c41804741c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 6 Nov 2025 00:39:27 +0800
Subject: [PATCH 15/18] os-wrappers-test: F_DUPFD_CLOEXEC may not be defined

---
 tests/os-wrappers-test.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/os-wrappers-test.c b/tests/os-wrappers-test.c
index 087d539..40abf48 100644
--- a/tests/os-wrappers-test.c
+++ b/tests/os-wrappers-test.c
@@ -80,12 +80,16 @@ fcntl_wrapper(int fd, int cmd, ...)
 
 	wrapped_calls_fcntl++;
 
+#ifdef F_DUPFD_CLOEXEC
 	if (fall_back && (cmd == F_DUPFD_CLOEXEC)) {
 		errno = EINVAL;
 		return -1;
 	}
+#endif
 	switch (cmd) {
+#ifdef F_DUPFD_CLOEXEC
 	case F_DUPFD_CLOEXEC:
+#endif
 	case F_DUPFD:
 	case F_SETFD:
 		va_start(ap, cmd);
-- 
2.24.3 (Apple Git-128)

