--- src/plugins/macosmediakeys/CMakeLists.txt	2024-07-15 03:37:30.000000000 +0800
+++ src/plugins/macosmediakeys/CMakeLists.txt	2025-08-12 19:08:39.000000000 +0800
@@ -4,12 +4,12 @@
   SPMediaKeyTap.m
 )
 
-if (APPLE)
-  FIND_LIBRARY(APP_KIT_LIBRARY AppKit)
-  FIND_LIBRARY(CARBON_LIBRARY Carbon)
-  MARK_AS_ADVANCED (APP_KIT_LIBRARY CARBON_LIBRARY)
-  add_library(macosmediakeys SHARED ${macosmediakeys_SOURCES})
-  target_link_libraries(macosmediakeys ${APP_KIT_LIBRARY} ${CARBON_LIBRARY})
-else()
+#if (APPLE)
+#  FIND_LIBRARY(APP_KIT_LIBRARY AppKit)
+#  FIND_LIBRARY(CARBON_LIBRARY Carbon)
+#  MARK_AS_ADVANCED (APP_KIT_LIBRARY CARBON_LIBRARY)
+#  add_library(macosmediakeys SHARED ${macosmediakeys_SOURCES})
+#  target_link_libraries(macosmediakeys ${APP_KIT_LIBRARY} ${CARBON_LIBRARY})
+#else()
   disable_plugin(macosmediakeys)
-endif()
\ No newline at end of file
+#endif()
