From 2f9d50f8f80c73ebbdc2eb29e256401a01b960c0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 12 Jan 2026 00:57:20 +0800
Subject: [PATCH 01/12] RDataEditorTableView.m: fix some bad syntax

---
 Tools/RDataEditorTableView.m | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Tools/RDataEditorTableView.m b/Tools/RDataEditorTableView.m
index b8e9f2f..fe30c35 100644
--- a/Tools/RDataEditorTableView.m
+++ b/Tools/RDataEditorTableView.m
@@ -154,7 +154,8 @@
 	NSArray *ds = [[self delegate] objectData];
 	NSArray *columnTypes = [[self delegate] objectColumnTypes];
 
-	NSInteger types[[columnTypes count]];
+	NSUInteger columnCount = [columnTypes count];
+	NSInteger types[columnCount];
 
 	for( i = 0; i < numColumns; i++ )
 		types[i] = [NSArrayObjectAtIndex(columnTypes, i) intValue];
@@ -265,7 +266,8 @@
 	NSArray *ds = [[self delegate] objectData];
 	NSArray *columnTypes = [[self delegate] objectColumnTypes];
 
-	NSInteger types[[columnTypes count]];
+	NSUInteger columnCount = [columnTypes count];
+	NSInteger types[columnCount];
 
 	for( i = 0; i < [columnTypes count]; i++ )
 		types[i] = [NSArrayObjectAtIndex(columnTypes, i) intValue];
-- 
2.52.0

