From e6eb83ad3ca6a75b18eface91a42bce9b1ea84d2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 12 Jan 2026 01:26:16 +0800
Subject: [PATCH 2/2] Fix incorrect REngine.h includes

---
 PrefPanes/QuartzPrefPane.m | 2 +-
 RDocument.m                | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/PrefPanes/QuartzPrefPane.m b/PrefPanes/QuartzPrefPane.m
index 7fc22bb..86a5866 100644
--- a/PrefPanes/QuartzPrefPane.m
+++ b/PrefPanes/QuartzPrefPane.m
@@ -30,7 +30,7 @@
 #import "QuartzPrefPane.h"
 #import "PreferenceKeys.h"
 #import "Preferences.h"
-#import "REngine.h"
+#import "../REngine/REngine.h"
 
 @interface QuartzPrefPane (Private)
 - (void)setIdentifier:(NSString *)newIdentifier;
diff --git a/RDocument.m b/RDocument.m
index a415687..906f1b2 100644
--- a/RDocument.m
+++ b/RDocument.m
@@ -35,7 +35,7 @@
 #import "RController.h"
 #import "Preferences.h"
 #import "RChooseEncodingPopupAccessory.h"
-#import "REngine.h"
+#import "REngine/REngine.h"
 #import "HelpManager.h"
 #import "NSString_RAdditions.h"
 
-- 
2.52.0

