From 39a6c7c47c71f7f76443a8db4e918a8a09a54e75 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Jan 2025 06:56:51 +0800
Subject: [PATCH 5/5] RScriptEditorTextStorage.m: fix compilation

---
 RScriptEditorTextStorage.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git RScriptEditorTextStorage.m RScriptEditorTextStorage.m
index 65d9844..869ea72 100644
--- RScriptEditorTextStorage.m
+++ RScriptEditorTextStorage.m
@@ -350,7 +350,7 @@ static SEL _getlSel;
 
 }
 
-- (void)edited:(NSTextStorageEditActions)mask range:(NSRange)oldRange changeInLength:(NSInteger)lengthChange
+- (void)edited:(NSUInteger)mask range:(NSRange)oldRange changeInLength:(NSInteger)lengthChange
 {
 
 	if(foldedCounter && mask == NSTextStorageEditedCharacters) {
-- 
2.48.0

