From b5910c6b10701a3fda8f94b1dc928e26e9a7cf94 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Jan 2025 06:47:04 +0800
Subject: [PATCH 06/12] Revert "Suppress 'Expected min height of view' macOS
 warning"

This reverts commit 54b786054e8776c386afad7030669890a04898b5.
---
 RController.m | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/RController.m b/RController.m
index 3a2a7b7..196b1c9 100644
--- a/RController.m
+++ b/RController.m
@@ -2123,20 +2123,6 @@ outputType: 0 = stdout, 1 = stderr, 2 = stdout/err as root
 	NSColor *color=(outputType==0)?[consoleColors objectAtIndex:iStdoutColor]:((outputType==1)?[consoleColors objectAtIndex:iStderrColor]:[consoleColors objectAtIndex:iRootColor]);
 	if (len>=0 && buf[len]!=0) buf[len]=0; /* this MAY be dangerous ... */
 	NSString *s = [[NSString alloc] initWithUTF8String:buf];
-    if (outputType == 1) {
-        // FIXME horrible, horrible hack - this warning seems to be a macOS bug, because
-        // we never create NSPopoverTouchBarItemButton, so we suppress it in the console
-        // by removing a line with that content
-        NSRange r = [s rangeOfString:@"Warning: Expected min height of view: .* to be less than or equal to " options:NSRegularExpressionSearch];
-        if (r.location != NSNotFound) {
-            NSString *os = s;
-            NSUInteger start, end;
-            [s getLineStart:&start end:&end contentsEnd:0 forRange:r];
-            s = [os stringByReplacingCharactersInRange:NSMakeRange(start, end-start) withString:@""];
-            [s retain];
-            [os release];
-        }
-    }
 	[self flushROutput];
 	[self writeConsoleDirectly:s withColor:color];
 	[s release];
-- 
2.52.0

