From f1e394a00e3d1b693e10a6ace18dd81f8657a9e8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 12 Jan 2026 01:34:33 +0800
Subject: [PATCH 10/12] Fix NSTableViewDataSource errors

---
 Tools/SortableDataSource.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Tools/SortableDataSource.h b/Tools/SortableDataSource.h
index 086dd45..c255d1c 100644
--- a/Tools/SortableDataSource.h
+++ b/Tools/SortableDataSource.h
@@ -41,6 +41,7 @@ typedef unsigned int NSUInteger;
 #define NSINTEGER_DEFINED 1
 #endif
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_6
 @protocol NSTableViewDataSource <NSObject>
 - (NSInteger)numberOfRowsInTableView:(NSTableView *)tableView;
 - (id)tableView:(NSTableView *)tableView objectValueForTableColumn:(NSTableColumn *)tableColumn row:(NSInteger)row;
@@ -52,6 +53,7 @@ typedef unsigned int NSUInteger;
 - (BOOL)tableView:(NSTableView *)tableView acceptDrop:(id <NSDraggingInfo>)info row:(NSInteger)row dropOperation:(NSTableViewDropOperation)dropOperation;
 - (NSArray *)tableView:(NSTableView *)tableView namesOfPromisedFilesDroppedAtDestination:(NSURL *)dropDestination forDraggedRowsWithIndexes:(NSIndexSet *)indexSet;
 @end
+#endif
 
 @interface SortableDataSource : NSObject <NSTableViewDataSource> {
     NSMutableArray *col;
-- 
2.52.0

