--- RController.m.orig
+++ RController.m
@@ -311,6 +311,7 @@
 	}
 }
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101202
 - (NSTouchBar *)makeTouchBar
 {
 	// Create TouchBar object
@@ -326,6 +327,7 @@
 
 	return touchBar;
 }
+#endif
 
 /* This is user's library path as interpreted by the R GUI. Note that
    during initialization the GUI has no access to R_LIBS_USER, because
@@ -407,10 +409,12 @@
 	[RConsoleWindow setBackgroundColor:[defaultConsoleColors objectAtIndex:iBackgroundColor]]; // we need this, because "update" doesn't touch the color if it's equal - and by default the window has *no* background - not even the default one, so we bring it in sync
 	[RConsoleWindow setDocumentEdited:YES];
 
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101202
 	// Force essentially an empty TouchBar due to performance
 	// problems with Apple's default implemenation
 	[NSApplication sharedApplication].automaticCustomizeTouchBarMenuItemEnabled = YES;
 	consoleTextView.touchBar = [self makeTouchBar];
+#endif
 
 	SLog(@" - working directory setup timer");
 	WDirtimer = [NSTimer scheduledTimerWithTimeInterval:0.5
