--- RDocumentWinCtrl.m.orig
+++ RDocumentWinCtrl.m
@@ -411,13 +411,16 @@
 	// Needed for showing tooltips of folded items
 	[[self window] setAcceptsMouseMovedEvents:YES];
     
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 101000
 	// FIXME: this is a hack for layout issues in Big Sur
 	// by forcing re-size we force layout to be updated - is there a better way?
+	// (contentLayoutRect needs the 10.10 SDK)
 	NSRect clRect = [[self window] contentLayoutRect];
 	clRect.size.width += 1;
 	[[self window] setContentSize:clRect.size];
 	clRect.size.width -= 1;
 	[[self window] setContentSize:clRect.size];
+#endif
 
 	// Make the text view fist responder so the user can start typing
 	[[self window] makeFirstResponder:textView];
