From 7daf054acdb0e88c52494821a660ab4557f84cc8 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Mon, 6 Nov 2023 00:20:39 +0800
Subject: [PATCH] inspector: unbreak build with gcc

---
 deps/v8/src/inspector/string-16.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git deps/v8/src/inspector/string-16.h deps/v8/src/inspector/string-16.h
index c1dd5cb929..2473fb0caa 100644
--- a/deps/v8/src/inspector/string-16.h
+++ b/deps/v8/src/inspector/string-16.h
@@ -149,7 +149,7 @@ String16 String16::concat(T... args) {
 
 }  // namespace v8_inspector
 
-#if !defined(__APPLE__) || defined(_LIBCPP_VERSION)
+#if !(defined(__APPLE__) && defined(__clang__)) || defined(_LIBCPP_VERSION)
 
 namespace std {
 template <>
