--- a/deps/v8/src/execution/isolate-data.h	2020-09-15 15:08:48.000000000 +0800
+++ b/deps/v8/src/execution/isolate-data.h	2025-04-03 13:36:01.000000000 +0800
@@ -112,6 +112,7 @@
 
  private:
 // Static layout definition.
+// FIXME: kBuiltinEntryTableOffset line has an ad hoc hack!
 #define FIELDS(V)                                                             \
   V(kEmbedderDataOffset, Internals::kNumIsolateDataSlots* kSystemPointerSize) \
   V(kExternalMemoryOffset, kInt64Size)                                        \
@@ -120,7 +121,7 @@
   V(kRootsTableOffset, RootsTable::kEntriesCount* kSystemPointerSize)         \
   V(kExternalReferenceTableOffset, ExternalReferenceTable::kSizeInBytes)      \
   V(kThreadLocalTopOffset, ThreadLocalTop::kSizeInBytes)                      \
-  V(kBuiltinEntryTableOffset, Builtins::builtin_count* kSystemPointerSize)    \
+  V(kBuiltinEntryTableOffset, Builtins::builtin_count* kSystemPointerSize + 4) \
   V(kBuiltinsTableOffset, Builtins::builtin_count* kSystemPointerSize)        \
   V(kVirtualCallTargetRegisterOffset, kSystemPointerSize)                     \
   V(kFastCCallCallerFPOffset, kSystemPointerSize)                             \
