--- a/deps/v8/src/base/build_config.h	2025-04-03 20:38:03.000000000 +0800
+++ b/deps/v8/src/base/build_config.h	2025-04-04 09:36:59.000000000 +0800
@@ -33,10 +33,10 @@
 #elif defined(__MIPSEB__) || defined(__MIPSEL__)
 #define V8_HOST_ARCH_MIPS 1
 #define V8_HOST_ARCH_32_BIT 1
-#elif defined(__PPC64__) || defined(_ARCH_PPC64)
+#elif defined(__PPC64__) || defined(_ARCH_PPC64) || defined(__ppc64__)
 #define V8_HOST_ARCH_PPC64 1
 #define V8_HOST_ARCH_64_BIT 1
-#elif defined(__PPC__) || defined(_ARCH_PPC)
+#elif defined(__PPC__) || defined(_ARCH_PPC) || defined(__ppc__)
 #define V8_HOST_ARCH_PPC 1
 #define V8_HOST_ARCH_32_BIT 1
 #elif defined(__s390__) || defined(__s390x__)
@@ -90,9 +90,9 @@
 #define V8_TARGET_ARCH_MIPS64 1
 #elif defined(__MIPSEB__) || defined(__MIPSEL__)
 #define V8_TARGET_ARCH_MIPS 1
-#elif defined(_ARCH_PPC64)
+#elif defined(_ARCH_PPC64) || defined(__ppc64__)
 #define V8_TARGET_ARCH_PPC64 1
-#elif defined(_ARCH_PPC)
+#elif defined(_ARCH_PPC) || defined(__ppc__)
 #define V8_TARGET_ARCH_PPC 1
 #else
 #error Target architecture was not detected as supported by v8
@@ -178,6 +178,8 @@
 #else
 #define V8_TARGET_LITTLE_ENDIAN 1
 #endif
+#elif defined(__POWERPC__)
+#define V8_TARGET_BIG_ENDIAN 1
 #elif __BIG_ENDIAN__  // FOR PPCGR on AIX
 #define V8_TARGET_BIG_ENDIAN 1
 #elif V8_TARGET_ARCH_PPC_LE
@@ -201,7 +203,7 @@
 #endif
 
 // Number of bits to represent the page size for paged spaces.
-#if defined(V8_TARGET_ARCH_PPC) || defined(V8_TARGET_ARCH_PPC64)
+#if V8_TARGET_ARCH_PPC || V8_TARGET_ARCH_PPC64
 // PPC has large (64KB) physical pages.
 const int kPageSizeBits = 19;
 #else
