--- src/TenacityApp.cpp.orig	2025-01-01 20:28:49.000000000 +0800
+++ src/TenacityApp.cpp	2025-10-20 10:08:55.000000000 +0800
@@ -861,7 +861,7 @@
    //
    FilePaths tenacityPathList;
 
-#ifdef __WXGTK__
+#if defined(__WXGTK__) && !defined(__APPLE__)
    wxStandardPaths standardPaths = wxStandardPaths::Get();
    wxString portablePrefix = wxPathOnly(wxPathOnly(standardPaths.GetExecutablePath()));
 
@@ -946,7 +946,7 @@
 #endif //__WXGTK__
 
 // JKC Bug 1220: Use path based on home directory on WXMAC
-#ifdef __WXMAC__
+#ifdef __APPLE__
    wxFileName tmpFile;
    tmpFile.AssignHomeDir();
    wxString tmpDirLoc = tmpFile.GetPath(wxPATH_GET_VOLUME);
@@ -972,7 +972,7 @@
       wxT("%s\\SessionData"), tmpDirLoc ) );
 #endif //__WXWSW__
 
-#ifdef __WXMAC__
+#ifdef __APPLE__
    // On Mac OS X, the path to the Audacity program is in argv[0]
    wxString progPath = wxPathOnly(argv[0]);
 
