--- libraries/lib-strings/Languages.cpp	2025-01-01 20:28:49.000000000 +0800
+++ libraries/lib-strings/Languages.cpp	2025-10-20 05:31:12.000000000 +0800
@@ -215,7 +215,7 @@
       return localLanguageName;
    }();
 
-#if defined(__WXGTK__)
+#if defined(__WXGTK__) && !defined(__APPLE__)
    {
 #ifdef __HAIKU__
       wxFileName pathNorm{ wxStandardPaths::Get().GetInstallPrefix() + L"/data/locale" };
