--- CMakeLists.txt.orig	2025-01-01 20:28:49.000000000 +0800
+++ CMakeLists.txt	2025-10-20 09:59:03.000000000 +0800
@@ -141,18 +141,14 @@
 
 # Definitions that must happen before the project() command
 if( APPLE )
-   set( MIN_MACOS_VERSION 10.15 )
-   set( TARGET_MACOS_VERSION 10.15 )
+   set( MIN_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET} )
+   set( TARGET_MACOS_VERSION ${CMAKE_OSX_DEPLOYMENT_TARGET} )
    # Generate schema files
    set( CMAKE_XCODE_GENERATE_SCHEME ON )
 
    # Define the OSX compatibility parameters
-   set( CMAKE_OSX_DEPLOYMENT_TARGET ${MIN_MACOS_VERSION} CACHE INTERNAL "" )
    set( CMAKE_OSX_SYSROOT macosx CACHE INTERNAL "" )
    set( CMAKE_XCODE_ATTRIBUTE_CODE_SIGN_IDENTITY "" CACHE INTERNAL "" )
-
-   # Shouldn't cmake do this???
-   string( APPEND CMAKE_CXX_FLAGS " -stdlib=libc++" )
 endif()
 
 # Add our module path
@@ -643,10 +639,9 @@
     message(STATUS "VST2 plugin host support disabled.")
 endif()
 
-if(NOT CMAKE_SYSTEM_NAME MATCHES "Darwin|Windows")
+if(NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
     find_package(GLIB REQUIRED)
     find_package(GTK 3.0 REQUIRED)
-
 endif()
 
 # Handle Audio Units option

--- src/CMakeLists.txt.orig	2025-01-01 20:28:49.000000000 +0800
+++ src/CMakeLists.txt	2025-10-20 09:53:41.000000000 +0800
@@ -58,9 +58,7 @@
 #
 # We always assume that GTK is used on platforms other than Windows and macOS,
 # as there is no other implementation avalaible now
-if( CMAKE_SYSTEM_NAME MATCHES "Darwin" )
-   set( wxIS_MAC on )
-elseif( CMAKE_SYSTEM_NAME MATCHES "Windows" )
+if( CMAKE_SYSTEM_NAME MATCHES "Windows" )
    set ( wxIS_WIN on )
 else()
    set ( wxIS_GTK on)
@@ -1085,9 +1083,9 @@
       $<$<BOOL:${USE_SBSMS}>:sbsms::sbsms>
       $<$<BOOL:${USE_SOUNDTOUCH}>:SoundTouch::SoundTouch>
       $<$<BOOL:${USE_VAMP}>:VampHostSDK::VampHostSDK>
-      $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:${GLIB_LIBRARIES}>
-      $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN,Haiku>:GTK::GTK>
-      $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:pthread>
+      $<$<PLATFORM_ID:Darwin,Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:${GLIB_LIBRARIES}>
+      $<$<PLATFORM_ID:Darwin,Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN,Haiku>:GTK::GTK>
+      $<$<PLATFORM_ID:Darwin,Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:pthread>
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:rt>
 )
 
@@ -1151,11 +1149,6 @@
          PRIVATE
             -Wl,-platform_version,macos,${MIN_MACOS_VERSION},${TARGET_MACOS_VERSION}
       )
-   else()
-      list( APPEND LDFLAGS
-         PRIVATE
-            -Wl,-sdk_version,10.13
-      )
    endif()
 
    # Define Mac specific resources
@@ -1189,7 +1182,11 @@
       effects/audiounits/AUControl.mm
       menus/WindowMenus.cpp
       PROPERTIES
-         COMPILE_FLAGS "-ObjC++"
+         if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+            COMPILE_FLAGS "-ObjC++"
+         else()
+            COMPILE_FLAGS "-x objective-c++"
+         endif()
          SKIP_PRECOMPILE_HEADERS YES
    )
 
