--- src/fetch/main.c	2025-10-29 00:02:11.000000000 +0800
+++ src/fetch/main.c	2026-03-29 06:35:11.000000000 +0800
@@ -110,6 +110,10 @@
   return "i386";
 #elif defined(__ia64__)
   return "ia64";
+#elif defined(__ppc64__) || defined(__powerpc64__)
+  return "ppc64";
+#elif defined(__ppc__) || defined(__powerpc__)
+  return "ppc";
 #else
   #warning "unknown target architecture"
   return "unknown";
