From fab8078c2429d65c2ff95718050b2089f8cfd47f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 12 Feb 2025 07:20:01 +0800
Subject: [PATCH 1/2] Add a missing header for basename

---
 filestore_dirlist.c | 1 +
 1 file changed, 1 insertion(+)

diff --git filestore_dirlist.c filestore_dirlist.c
index 748822d..78e0d37 100644
--- filestore_dirlist.c
+++ filestore_dirlist.c
@@ -1,6 +1,7 @@
 #include "filestore_dirlist.h"
 #include "commands.h"
 #include <fnmatch.h>
+#include <libgen.h> // basename on macOS and musl
 
 void FileStoreDirListFree(TFileStore *FS, ListNode *Dir)
 {
-- 
2.48.0

