From bcd91813b389e5a40e0ff500ca89f4eac2561993 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 12 Feb 2025 10:35:54 +0800
Subject: [PATCH 2/2] Fix Makefile

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git Makefile.in Makefile.in
index 657ef71..e861508 100644
--- Makefile.in
+++ Makefile.in
@@ -6,7 +6,7 @@ STATIC_LIBUSEFUL=@LIBUSEFUL_BUNDLED@
 
 
 all: $(OBJ) main.c $(STATIC_LIBUSEFUL)
-	gcc $(FLAGS) -ofileferry $(OBJ) main.c $(STATIC_LIBUSEFUL) $(LIBS)
+	gcc $(FLAGS) -o fileferry $(OBJ) main.c $(STATIC_LIBUSEFUL) $(LIBS)
 
 libUseful-5/libUseful.a:
 	make -C libUseful-5
-- 
2.48.0

