--- .home/.cargo/macports/clap-4.0.18/src/macros.rs	2006-07-24 09:21:28.000000000 +0800
+++ .home/.cargo/macports/clap-4.0.18/src/macros.rs	2026-01-05 08:14:13.000000000 +0800
@@ -42,7 +42,7 @@
 ///             .get_matches();
 /// # }
 /// ```
-#[cfg(feature = "cargo")]
+#[cfg(all(not(feature = "no_cargo"), not(rust_compiler="mrustc")))]
 #[macro_export]
 macro_rules! crate_authors {
     ($sep:expr) => {{
@@ -75,7 +75,7 @@
 ///             .get_matches();
 /// # }
 /// ```
-#[cfg(feature = "cargo")]
+#[cfg(all(not(feature = "no_cargo"), not(rust_compiler="mrustc")))]
 #[macro_export]
 macro_rules! crate_description {
     () => {
@@ -130,17 +130,6 @@
     }};
     ($name:expr) => {{
         let mut cmd = $crate::Command::new($name).version($crate::crate_version!());
-
-        let author = $crate::crate_authors!();
-        if !author.is_empty() {
-            cmd = cmd.author(author)
-        }
-
-        let about = $crate::crate_description!();
-        if !about.is_empty() {
-            cmd = cmd.about(about)
-        }
-
         cmd
     }};
 }
