--- configure	2025-05-08 15:51:13.000000000 +0800
+++ configure	2025-11-30 06:33:10.000000000 +0800
@@ -21544,7 +21544,7 @@
 elif test "$architecture" = "Darwin"
 then
    ARCH="DARWIN"
-   CFLAGS="$CFLAGS -DREENTRANT -no-cpp-precomp -DNEED_SOCKLEN_T_DEFINED"
+   CFLAGS="$CFLAGS -DREENTRANT -DNEED_SOCKLEN_T_DEFINED"
    LDFLAGS="$LDFLAGS -Wl,-search_paths_first"
    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for kvm_open in -lkvm" >&5
 printf %s "checking for kvm_open in -lkvm... " >&6; }
@@ -23241,35 +23241,6 @@
 fi
 
 
-# Determine the minimum macOS version we can build
-if [ $(uname) == "Darwin" ]; then
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for minimum macOS version supported" >&5
-printf %s "checking for minimum macOS version supported... " >&6; }
-
-    # Create a simple C program
-    echo "int main() { return 0; }" > t.c
-
-    # Try to compile it with the specified macOS version
-    $CC -mmacosx-version-min=10.15 -o t t.c || as_fn_error $? "Could not compile test program" "$LINENO" 5
-
-    # Extract the minos value using otool
-    min_macos_version=$(otool -l t | awk '/minos/{print $2}')
-
-    # Clean up the test files
-    rm -f t.c t
-
-    # Check if min_macos_version is set
-    if test -z "$min_macos_version"; then
-        as_fn_error $? "Could not determine minimum macOS version" "$LINENO" 5
-    else
-        # Set CFLAGS with the determined minimum macOS version
-        CFLAGS="$CFLAGS -mmacosx-version-min=$min_macos_version"
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $min_macos_version" >&5
-printf "%s\n" "$min_macos_version" >&6; }
-    fi
-fi
-
-
 # ------------------------------------------------------------------------
 # Outputs
 # ------------------------------------------------------------------------

--- libmonit/configure	2025-05-08 15:51:15.000000000 +0800
+++ libmonit/configure	2025-11-30 06:16:45.000000000 +0800
@@ -19672,34 +19672,6 @@
 # Build options
 # ---------------------------------------------------------------------------
 
-# Determine the minimum macOS version we can build
-if [ $(uname) == "Darwin" ]; then
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for minimum macOS version supported" >&5
-printf %s "checking for minimum macOS version supported... " >&6; }
-
-    # Create a simple C program
-    echo "int main() { return 0; }" > t.c
-
-    # Try to compile it with the specified macOS version
-    $CC -mmacosx-version-min=10.15 -o t t.c || as_fn_error $? "Could not compile test program" "$LINENO" 5
-
-    # Extract the minos value using otool
-    min_macos_version=$(otool -l t | egrep minos | awk '{print $2}')
-
-    # Clean up the test files
-    rm -f t.c t
-
-    # Check if min_macos_version is set
-    if test -z "$min_macos_version"; then
-        as_fn_error $? "Could not determine minimum macOS version" "$LINENO" 5
-    else
-        # Set CFLAGS with the determined minimum macOS version
-        CFLAGS="$CFLAGS -mmacosx-version-min=$min_macos_version"
-        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $min_macos_version" >&5
-printf "%s\n" "$min_macos_version" >&6; }
-    fi
-fi
-
 # ---------------------------------------------------------------------------
 # Outputs
 # ---------------------------------------------------------------------------
