--- boost/process/v2/posix/default_launcher.hpp	2024-12-05 08:53:35.000000000 +0800
+++ boost/process/v2/posix/default_launcher.hpp	2025-01-27 19:22:13.000000000 +0800
@@ -29,7 +29,10 @@
 #include <unistd.h>
 
 
-#if defined(__APPLE__) || defined(__MACH__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun)
+#if defined(__APPLE__) || defined(__MACH__)
+#include <crt_externs.h>
+#define environ *_NSGetEnviron()
+#elif defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun)
 extern "C" { extern char **environ; }
 #endif
 
@@ -293,7 +296,7 @@
 struct default_launcher
 {
     /// The pointer to the environment forwarded to the subprocess.
-    const char * const * env = ::environ;
+    const char * const * env = environ;
     /// The pid of the subprocess - will be assigned after fork.
     int pid = -1;
 
--- boost/process/v2/detail/environment_posix.hpp	2024-12-05 08:53:35.000000000 +0800
+++ boost/process/v2/detail/environment_posix.hpp	2025-01-27 19:20:49.000000000 +0800
@@ -14,7 +14,10 @@
 #include <boost/process/v2/detail/config.hpp>
 #include <boost/process/v2/cstring_ref.hpp>
 
-#if defined(__APPLE__) || defined(__MACH__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun)
+#if defined(__APPLE__) || defined(__MACH__)
+#include <crt_externs.h>
+#define environ *_NSGetEnviron()
+#elif defined(__FreeBSD__) || defined(__DragonFly__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__sun)
 extern "C" { extern char **environ; }
 #endif
 

--- libs/process/src/detail/environment_posix.cpp	2024-12-05 08:53:35.000000000 +0800
+++ libs/process/src/detail/environment_posix.cpp	2025-01-27 19:21:43.000000000 +0800
@@ -51,7 +51,7 @@
 }
 
 
-native_handle_type load_native_handle() { return ::environ; }
+native_handle_type load_native_handle() { return environ; }
 
 
 native_iterator next(native_iterator nh)
