--- test/merge/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/merge/Makefile	2026-07-14 20:14:41.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 EXEC = test1
 
--- test/cpp/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/cpp/Makefile	2026-07-14 20:16:01.000000000 +0800
@@ -1,8 +1,8 @@
-CFLAGS = -std=c17 -fsanitize=address,undefined -Wmissing-declarations -Wall -Wextra -MMD
+CFLAGS = -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
 EXEC = test1
 
 ifdef DEBUG
-    CFLAGS += -O0 -g
+    CFLAGS += -O0 -g -fsanitize=address,undefined
 else
     CFLAGS += -O3 -DNDEBUG
 endif

--- test/parser/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/parser/Makefile	2026-07-14 20:14:04.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD 
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 all: driver
 
--- test/scankey/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/scankey/Makefile	2026-07-14 20:11:49.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD 
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 all: driver
 
--- test/scanvalue/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/scanvalue/Makefile	2026-07-14 20:11:13.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD 
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 all: driver
 
--- test/source/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/source/Makefile	2026-07-14 20:13:23.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 EXEC = test1
 
--- test/stdtest/Makefile.orig	2026-06-19 08:30:03.000000000 +0800
+++ test/stdtest/Makefile	2026-07-14 20:12:30.000000000 +0800
@@ -1,4 +1,8 @@
-CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD 
+ifdef DEBUG
+    CFLAGS := -O0 -g -fsanitize=address,undefined -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+else
+    CFLAGS += -O3 -DNDEBUG -std=c17 -Wmissing-declarations -Wall -Wextra -MMD
+endif
 
 all: driver
 
