--- Makefile	2025-07-13 07:20:32.000000000 +0800
+++ Makefile	2025-10-13 07:58:32.000000000 +0800
@@ -1,6 +1,7 @@
 .NOTPARALLEL:
 
-prefix ?= /usr/local
+prefix ?= $(PREFIX)
+destdir ?= $(DESTDIR)
 # remove trailing /
 override prefix := $(prefix:%/=%)
 DIRS = src simple test
@@ -33,13 +34,13 @@
 	$(MAKE) -C $(@:build-%=%)
 
 install: all
-	install -d ${prefix}/include
-	install -d ${prefix}/lib
-	install -d ${prefix}/lib/pkgconfig
-	install -m 0644 -t ${prefix}/include src/tomlc17.h
-	install -m 0644 -t ${prefix}/include src/tomlcpp.hpp
-	install -m 0644 -t ${prefix}/lib src/libtomlc17.a
-	@echo "$$PCFILE" >> ${prefix}/lib/pkgconfig/tomlc17.pc
+	install -d ${destdir}${prefix}/include
+	install -d ${destdir}${prefix}/lib
+	install -d ${destdir}${prefix}/lib/pkgconfig
+	install -m 0644 src/tomlc17.h ${destdir}${prefix}/include
+	install -m 0644 src/tomlcpp.hpp ${destdir}${prefix}/include
+	install -m 0644 src/libtomlc17.a ${destdir}${prefix}/lib
+	@echo "$$PCFILE" >> ${destdir}${prefix}/lib/pkgconfig/tomlc17.pc
 
 test: $(TESTDIRS)
 
