--- a/src/symbolize/gimli/macho.rs	2026-10-05 14:04:23.915651574 +0000
+++ b/src/symbolize/gimli/macho.rs	2026-10-05 14:04:23.941652220 +0000
@@ -88,6 +88,10 @@
             Some(macho::CPU_TYPE_ARM)
         } else if cfg!(target_arch = "aarch64") {
             Some(macho::CPU_TYPE_ARM64)
+        } else if cfg!(target_arch = "powerpc") {
+            Some(macho::CPU_TYPE_POWERPC)
+        } else if cfg!(target_arch = "powerpc64") {
+            Some(macho::CPU_TYPE_POWERPC64)
         } else {
             None
         }
@@ -161,7 +165,7 @@
                     .filter_map(|nlist: &MachNlist| {
                         let name = nlist.name(endian, symbols.strings()).ok()?;
                         if name.len() > 0 && !nlist.is_undefined() {
-                            Some((name, nlist.n_value(endian)))
+                            Some((name, u64::from(nlist.n_value(endian))))
                         } else {
                             None
                         }
