--- source/posix/thread.c	2025-07-09 02:00:04.000000000 +0800
+++ source/posix/thread.c	2025-08-23 02:52:39.000000000 +0800
@@ -28,6 +28,11 @@
 typedef cpuset_t cpu_set_t;
 #elif defined(__OpenBSD__)
 #    include <pthread_np.h>
+#elif defined(__APPLE__)
+#    include <AvailabilityMacros.h>
+#    if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#        define NO_AWS_PTHREAD_GETNAME
+#    endif
 #endif
 
 #if !defined(AWS_AFFINITY_METHOD)
@@ -505,8 +510,8 @@
 #define THREAD_NAME_BUFFER_SIZE 256
 int aws_thread_name(struct aws_allocator *allocator, aws_thread_id_t thread_id, struct aws_string **out_name) {
     *out_name = NULL;
-#if defined(AWS_PTHREAD_GETNAME_TAKES_2ARGS) || defined(AWS_PTHREAD_GETNAME_TAKES_3ARGS) ||                            \
-    defined(AWS_PTHREAD_GET_NAME_TAKES_2_ARGS)
+#if (defined(AWS_PTHREAD_GETNAME_TAKES_2ARGS) || defined(AWS_PTHREAD_GETNAME_TAKES_3ARGS) ||                            \
+    defined(AWS_PTHREAD_GET_NAME_TAKES_2_ARGS)) && !defined(NO_AWS_PTHREAD_GETNAME)
     char name[THREAD_NAME_BUFFER_SIZE] = {0};
 #    ifdef AWS_PTHREAD_GETNAME_TAKES_3ARGS
     if (pthread_getname_np(thread_id, name, THREAD_NAME_BUFFER_SIZE)) {
