--- dmenu.c	2025-08-09 21:00:49.000000000 +0800
+++ dmenu.c	2026-07-15 18:13:43.000000000 +0800
@@ -1,4 +1,11 @@
 /* See LICENSE file for copyright and license details. */
+
+#ifdef __APPLE__
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#endif
+
 #include <ctype.h>
 #include <locale.h>
 #include <stdio.h>

--- stest.c	2025-08-09 21:00:49.000000000 +0800
+++ stest.c	2026-07-15 18:19:21.000000000 +0800
@@ -1,6 +1,12 @@
 /* See LICENSE file for copyright and license details. */
-#include <sys/stat.h>
 
+#ifdef __APPLE__
+#ifndef _DARWIN_C_SOURCE
+#define _DARWIN_C_SOURCE
+#endif
+#endif
+
+#include <sys/stat.h>
 #include <dirent.h>
 #include <limits.h>
 #include <stdio.h>
