From 2a528ef8c5a9a19e4f796086b780f0620fc325fa Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 15:52:27 +0800
Subject: [PATCH 11/11] mymodel.h: avoid nullptr, use macro

---
 libfm/mymodel.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git libfm/mymodel.h libfm/mymodel.h
index 8c57755..5e3a3c0 100644
--- libfm/mymodel.h
+++ libfm/mymodel.h
@@ -37,7 +37,7 @@ class myModel : public QAbstractItemModel {
   Q_OBJECT
 public:
 
-  myModel(bool realMime, MimeUtils* mimeUtils, QObject* parent = nullptr);
+  myModel(bool realMime, MimeUtils* mimeUtils, QObject* parent = Q_NULLPTR);
   ~myModel();
   void loadMimeTypes() const;
   void cacheInfo();
-- 
2.49.0

