From 211e9bb9bf1ff1d7d85cc8cd45ca14c0814c25e9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 13:20:34 +0800
Subject: [PATCH 03/10] mymodel.cpp: add missing QFileDevice

---
 libfm/mymodel.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git libfm/mymodel.cpp libfm/mymodel.cpp
index 33ef968..09b50b0 100644
--- libfm/mymodel.cpp
+++ libfm/mymodel.cpp
@@ -25,6 +25,12 @@
 #include <sys/ioctl.h>
 #include <QApplication>
 #include <QMessageBox>
+#if QT_VERSION >= 0x050000
+#include <QFileDevice>
+#else
+#include <QFile>
+#define QFileDevice QFile
+#endif
 #include "fileutils.h"
 
 #ifdef WITH_MAGICK
-- 
2.49.0

