From 48525b85da160853643822de74617a091ac35ff1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 13:51:49 +0800
Subject: [PATCH 05/10] Include libfm to find headers

---
 qtfm.pro | 2 ++
 1 file changed, 2 insertions(+)

diff --git qtfm.pro qtfm.pro
index 6ddaa9d..bb3cc8e 100644
--- qtfm.pro
+++ qtfm.pro
@@ -1,6 +1,8 @@
 TEMPLATE = subdirs
 CONFIG -= ordered
 SUBDIRS += libfm fm
+INCLUDEPATH += $$PWD/libfm
+INCLUDEPATH += $$PWD/fm/src
 fm.depends += libfm
 
 CONFIG(v7) {
-- 
2.49.0

