From 4b907e6a1e83ddc82d25a8c352644d8f9e9a4b33 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 14:33:42 +0800
Subject: [PATCH 06/10] settingsdialog: no setClearButtonEnabled

---
 fm/src/settingsdialog.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git fm/src/settingsdialog.cpp fm/src/settingsdialog.cpp
index db915e6..c43ed8b 100644
--- fm/src/settingsdialog.cpp
+++ fm/src/settingsdialog.cpp
@@ -364,7 +364,9 @@ QWidget* SettingsDialog::createMimeSettings() {
   // tree filter
   QLineEdit *mimeSearch = new QLineEdit(grpMimes);
   mimeSearch->setPlaceholderText(tr("Filter ..."));
+#if QT_VERSION >= 0x050000
   mimeSearch->setClearButtonEnabled(true);
+#endif
   connect(mimeSearch, SIGNAL(textChanged(QString)), this, SLOT(filterMimes(QString)));
   layoutMimes->addWidget(mimeSearch);
 
-- 
2.49.0

