From 422db8ac6c3601a7aca7c4fb7f88e6d1cd6dbf36 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 14:35:33 +0800
Subject: [PATCH 07/10] mainwindow: queueFile

---
 fm/src/mainwindow.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git fm/src/mainwindow.cpp fm/src/mainwindow.cpp
index befa7e0..0396e2f 100644
--- fm/src/mainwindow.cpp
+++ fm/src/mainwindow.cpp
@@ -1156,7 +1156,11 @@ void MainWindow::pasteLauncher(const QList<QUrl> &files, const QString &newPath,
 
   // Copy/move files
   for (int i=0;i<files.size();++i) {
+#if QT_VERSION >= 0x050000
       QString queueFile = files.at(i).fileName();
+#else
+      QString queueFile = QFileInfo(files.at(i).toLocalFile()).fileName();
+#endif
       queueFile.prepend(QString("%1/").arg(newPath));
       if (!progressQueue.contains(queueFile)) {
           qDebug() << "add to queue" << queueFile;
-- 
2.49.0

