From 167a9f314f81bcc2e3a6ff62f7e5e812a87e31da Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 18 Jul 2025 14:38:18 +0800
Subject: [PATCH 08/10] mainwindow: BackButton

---
 fm/src/mainwindow.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git fm/src/mainwindow.cpp fm/src/mainwindow.cpp
index 0396e2f..a742e22 100644
--- fm/src/mainwindow.cpp
+++ fm/src/mainwindow.cpp
@@ -1646,7 +1646,11 @@ bool MainWindow::eventFilter(QObject *o, QEvent *e)
     if (e->type() == QEvent::MouseButtonPress) {
         QMouseEvent* me = static_cast<QMouseEvent*>(e);
         switch (me->button()) {
+#if QT_VERSION >= 0x050000
         case Qt::BackButton:
+#else
+        case Qt::XButton1:
+#endif
             goBackDir();
             break;
         default:;
-- 
2.49.0

