Only search the MacPorts prefix for .pc files.

--- vlib/v/pkgconfig/pkgconfig.v.original
+++ vlib/v/pkgconfig/pkgconfig.v
@@ -6,24 +6,7 @@
 const version = '0.3.4'
 
 const default_paths = [
-	'/usr/local/lib/x86_64-linux-gnu/pkgconfig',
-	'/usr/local/lib64/pkgconfig',
-	'/usr/local/lib/pkgconfig',
-	'/usr/local/share/pkgconfig',
-	'/usr/lib/x86_64-linux-gnu/pkgconfig',
-	'/usr/lib/aarch64-linux-gnu/pkgconfig',
-	'/usr/lib64/pkgconfig',
-	'/usr/lib/pkgconfig',
-	'/usr/share/pkgconfig',
-	'/opt/homebrew/lib/pkgconfig', // Brew on macOS
-	'/opt/homebrew/share/pkgconfig', // Brew on macOS. Needed for fish.pc, eigen3.pc, applewmproto.pc, fontsproto.pc, xextproto.pc, SPIRV-Headers.pc etc; seems like a legacy folder.
-	'/opt/homebrew/Library/Homebrew/os/mac/pkgconfig/11', // Brew on macOS. Needed for zlib.pc, libcurl.pc, expat.pc and a few others; all the rest are symlinked in /opt/homebrew/lib/pkgconfig .
-	'/opt/local/lib/pkgconfig', // MacPorts on macOS
-	'/usr/local/libdata/pkgconfig', // FreeBSD
-	'/usr/libdata/pkgconfig', // FreeBSD
-	'/usr/lib/i386-linux-gnu/pkgconfig', // Debian 32bit
-	'/data/data/com.termux/files/usr/lib/pkgconfig', // Termux
-	'/usr/pkg/lib/pkgconfig', // NetBSD
+	'@PREFIX@/lib/pkgconfig', // MacPorts
 ]
 
 pub struct Options {
