--- configure	2026-05-19 09:11:40.000000000 +0800
+++ configure	2026-05-19 09:15:52.000000000 +0800
@@ -23702,8 +23702,8 @@
 
 case "$host_os" in
     darwin*)
-        test x"$x_includes" = x"NONE" && x_includes="/usr/X11R6/include"
-        test x"$x_libraries" = x"NONE" && x_libraries="/usr/X11R6/lib"
+        test x"$x_includes" = x"NONE" && x_includes="@PREFIX@/include"
+        test x"$x_libraries" = x"NONE" && x_libraries="@PREFIX@/lib"
         ;;
 esac
 
@@ -23770,6 +23770,9 @@
 # Standard set of common directories for X headers.
 # Check X11 before X11Rn because it is often a symlink to the current release.
 ac_x_header_dirs='
+@PREFIX@/include/X11
+@PREFIX@/include
+
 /usr/X11/include
 /usr/X11R7/include
 /usr/X11R6/include
@@ -24540,13 +24543,13 @@
 b
 :q
 q"
-x_lib_location="`ls -1 "${x_libraries:-/usr/local/lib}/libX11.$acl_cv_shlibext"* "${x_libraries:-/usr/lib}/libX11.$acl_cv_shlibext"* 2>/dev/null | sed -e \"${soname_script}\"`"
+x_lib_location="`ls -1 "${x_libraries:-@PREFIX@/lib}/libX11.$acl_cv_shlibext"* "${x_libraries:-/usr/lib}/libX11.$acl_cv_shlibext"* 2>/dev/null | sed -e \"${soname_script}\"`"
 
 cat >>confdefs.h <<_ACEOF
 #define LIBX11_SO "${x_lib_location:-libX11.$acl_cv_shlibext}"
 _ACEOF
 
-x_lib_location="`ls -1 "${x_libraries:-/usr/local/lib}/libXv.$acl_cv_shlibext"*  "${x_libraries:-/usr/lib}/libXv.$acl_cv_shlibext"*  2>/dev/null | sed -e \"${soname_script}\"`"
+x_lib_location="`ls -1 "${x_libraries:-@PREFIX@/lib}/libXv.$acl_cv_shlibext"*  "${x_libraries:-/usr/lib}/libXv.$acl_cv_shlibext"*  2>/dev/null | sed -e \"${soname_script}\"`"
 
 cat >>confdefs.h <<_ACEOF
 #define LIBXV_SO "${x_lib_location:-libXv.$acl_cv_shlibext}"
