--- src/common/open_memstream.c	2026-03-23 16:45:44.000000000 +0800
+++ src/common/open_memstream.c	2026-05-30 12:50:30.000000000 +0800
@@ -29,7 +29,6 @@
 
 /* BSD-compatible implementation of open_memstream */
 
-#if HAVE_FUNOPEN
 struct memstream {
   char *buffer;
   size_t size, space;
@@ -84,5 +83,3 @@
                  0,                     /* seek */
                  0);                    /* close */
 }
-#endif
-
