--- configure.ac.orig	2003-09-30 17:58:46
+++ configure.ac	2025-07-04 19:26:24
@@ -72,7 +72,7 @@
 # Programs
 SUBDIRS68="$SUBDIRS68 as68 sc68 info68 sourcer68 debug68"
 # Data
-SUBDIRS68="$SUBDIRS68 data info man"
+SUBDIRS68="$SUBDIRS68 data man"
 
 # C-flags
 MY_CFLAGS="-DHAVE_CONFIG_CONFIG68_H=1 -DNO_FALLBACK_CONFIG=1"
@@ -262,7 +262,6 @@
  sourcer68/Makefile
  unice68/Makefile
  man/Makefile
- info/Makefile
  doc/Makefile
  data/Makefile])
 AC_OUTPUT
