From f871218447916da5eea16e7e64e3dabf598b03b0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 28 May 2025 12:59:05 +0800
Subject: [PATCH 3/4] sandbox.cpp: use sandbox on 10.7+

---
 src/cli/sandbox.cpp | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git src/cli/sandbox.cpp src/cli/sandbox.cpp
index a52fff2bf..b598db267 100644
--- src/cli/sandbox.cpp
+++ src/cli/sandbox.cpp
@@ -10,12 +10,18 @@
 #include <botan/internal/target_info.h>
 #include <memory>
 
+#if defined(BOTAN_TARGET_OS_IS_MACOS)
+   #include <AvailabilityMacros.h>
+#endif
+
 #if defined(BOTAN_TARGET_OS_HAS_CAP_ENTER)
    #include <sys/capsicum.h>
    #include <unistd.h>
 #elif defined(BOTAN_TARGET_OS_HAS_SETPPRIV)
    #include <priv.h>
-#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC)
+// macOS 10.6 has sandbox, but does not have sandbox_init
+#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC) && \
+   !(defined(BOTAN_TARGET_OS_IS_MACOS) && MAC_OS_X_VERSION_MIN_REQUIRED < 1070)
    #include <sandbox.h>
 #endif
 
@@ -35,7 +41,8 @@ Sandbox::Sandbox() {
    m_name = "capsicum";
 #elif defined(BOTAN_TARGET_OS_HAS_SETPPRIV)
    m_name = "privilege";
-#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC)
+#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC) && \
+   !(defined(BOTAN_TARGET_OS_IS_MACOS) && MAC_OS_X_VERSION_MIN_REQUIRED < 1070)
    m_name = "sandbox";
 #else
    m_name = "<none>";
@@ -93,7 +100,8 @@ bool Sandbox::init() {
    }
 
    return true;
-#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC)
+#elif defined(BOTAN_TARGET_OS_HAS_SANDBOX_PROC) && \
+   !(defined(BOTAN_TARGET_OS_IS_MACOS) && MAC_OS_X_VERSION_MIN_REQUIRED < 1070)
 
    BOTAN_DIAGNOSTIC_PUSH
    BOTAN_DIAGNOSTIC_IGNORE_DEPRECATED_DECLARATIONS
-- 
2.48.0

