--- meson.build.orig	2025-07-20 03:59:05.000000000 +0800
+++ meson.build	2025-09-05 03:21:06.000000000 +0800
@@ -5,7 +5,6 @@
 add_project_arguments('-pedantic', language: 'c')
 add_project_arguments('-Wstrict-prototypes', language: 'c')
 add_project_arguments('-Wcast-align', language: 'c')
-add_project_arguments('-Wno-gnu', language: 'c')
 add_project_arguments('-fno-common', language: 'c')
 add_project_arguments('-fvisibility=hidden', language: 'c')
 
