From af7fefea7dd5e719f132ecb262047610c4fc7fc1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 11 Jan 2025 05:24:59 +0800
Subject: [PATCH] Do not hardcode Xcode ld

diff --git a/llvm/cmake/config-ix.cmake b/llvm/cmake/config-ix.cmake
index 92f902af31f..261bb9226e8 100644
--- a/src/llvm/cmake/config-ix.cmake
+++ b/src/llvm/cmake/config-ix.cmake
@@ -642,6 +642,7 @@
 endif()
 
 if(CMAKE_HOST_APPLE AND APPLE)
+  find_program(LD64_EXECUTABLE NAMES ld DOC "The ld64 linker")
   if(NOT LD64_EXECUTABLE)
     if(NOT CMAKE_XCRUN)
       find_program(CMAKE_XCRUN NAMES xcrun)
@@ -650,8 +651,6 @@
       execute_process(COMMAND ${CMAKE_XCRUN} -find ld
         OUTPUT_VARIABLE LD64_EXECUTABLE
         OUTPUT_STRIP_TRAILING_WHITESPACE)
-    else()
-      find_program(LD64_EXECUTABLE NAMES ld DOC "The ld64 linker")
     endif()
   endif()
 
--- a/src/clang/CMakeLists.txt	2026-04-05 16:33:48.000000000 +0800
+++ b/src/clang/CMakeLists.txt	2026-08-01 07:15:26.000000000 +0800
@@ -317,13 +317,6 @@
   endif()
 endif ()
 
-# Determine HOST_LINK_VERSION on Darwin.
-set(HOST_LINK_VERSION)
-if (APPLE AND NOT CMAKE_LINKER MATCHES ".*lld.*")
-  get_darwin_linker_version(HOST_LINK_VERSION)
-  message(STATUS "Host linker version: ${HOST_LINK_VERSION}")
-endif()
-
 include(CMakeParseArguments)
 include(AddClang)
 
