--- a/src/llvm/lib/TargetParser/Host.cpp	2026-04-05 16:33:48.000000000 +0800
+++ b/src/llvm/lib/TargetParser/Host.cpp	2026-08-19 02:53:30.000000000 +0800
@@ -1300,7 +1300,7 @@
   return "generic";
 }
 
-#elif defined(__APPLE__) && defined(__powerpc__)
+#elif defined(__APPLE__) && defined(__POWERPC__)
 StringRef sys::getHostCPUName() {
   host_basic_info_data_t hostInfo;
   mach_msg_type_number_t infoCount;
@@ -1912,7 +1912,10 @@
 #elif defined(__x86_64__)
   T.setArch(Triple::x86_64);
   T.setArchName("x86_64");
-#elif defined(__powerpc__)
+#elif defined(__ppc64__)
+  T.setArch(Triple::ppc64);
+  T.setArchName("powerpc64");
+#elif defined(__ppc__)
   T.setArch(Triple::ppc);
   T.setArchName("powerpc");
 #else
