--- a/src/tapi/CMakeLists.txt	2026-08-01 06:45:29.000000000 +0800
+++ b/src/tapi/CMakeLists.txt	2026-08-19 06:58:51.000000000 +0800
@@ -161,7 +161,7 @@
   add_llvm_executable(${name} ${ARGN})
   set_target_properties(${name} PROPERTIES FOLDER "Tapi executables")
   if(APPLE)
-    target_link_options(${name} PRIVATE "-Wl,-no_exported_symbols")
+    # target_link_options(${name} PRIVATE "-Wl,-no_exported_symbols")
   endif()
 endmacro(add_tapi_executable)
 
