--- libao2/ao_coreaudio.c	2023-04-20 03:42:04.000000000 +0800
+++ libao2/ao_coreaudio.c	2025-05-12 22:01:09.000000000 +0800
@@ -173,7 +173,9 @@
 	// However since this runs in a separate thread, this can
 	// trigger a race condition if MPlayer calls AudioOutputUnitStop
 	// for example during uninit, which then causes a hang.
-	return amt ? noErr : kAudioFileEndOfFileError;
+
+	// Restore this line from mplayer 1.5, since later one breaks compilation.
+	return noErr;
 }
 
 static int control(int cmd,void *arg){
