diff --git a/libao2/ao_coreaudio.c b/libao2/ao_coreaudio.c
index 540a001..b7490fc 100644
--- libao2/ao_coreaudio.c
+++ libao2/ao_coreaudio.c
@@ -118,7 +118,7 @@ typedef struct ao_coreaudio_s
   int paused;
 
   /* Ring-buffer */
-  AVFifoBuffer *buffer;
+  AVFifo *buffer;
   unsigned int buffer_len; ///< must always be num_chunks * chunk_size
   unsigned int num_chunks;
   unsigned int chunk_size;
@@ -130,21 +130,23 @@ static ao_coreaudio_t *ao = NULL;
  * \brief add data to ringbuffer
  */
 static int write_buffer(unsigned char* data, int len){
-  int free = ao->buffer_len - av_fifo_size(ao->buffer);
+  int free = ao->buffer_len - (int)av_fifo_can_read(ao->buffer);
   if (len > free) len = free;
-  return av_fifo_generic_write(ao->buffer, data, len, NULL);
+  if (av_fifo_write(ao->buffer, data, len) < 0)
+    return 0;
+  return len;
 }
 
 /**
  * \brief remove data from ringbuffer
  */
 static int read_buffer(unsigned char* data,int len){
-  int buffered = av_fifo_size(ao->buffer);
+  int buffered = (int)av_fifo_can_read(ao->buffer);
   if (len > buffered) len = buffered;
   if (data)
-    av_fifo_generic_read(ao->buffer, data, len, NULL);
+    av_fifo_read(ao->buffer, data, len);
   else
-    av_fifo_drain(ao->buffer, len);
+    av_fifo_drain2(ao->buffer, len);
   return len;
 }
 
@@ -154,7 +156,7 @@ static OSStatus theRenderProc(void *inRefCon,
                               UInt32 inBusNumber, UInt32 inNumFrames,
                               AudioBufferList *ioData)
 {
-int amt=av_fifo_size(ao->buffer);
+int amt=(int)av_fifo_can_read(ao->buffer);
 int req=(inNumFrames)*ao->packetSize;
 
 	if(amt>req)
@@ -622,7 +624,7 @@ int device_id, display_help = 0;
 
 	ao->num_chunks = (ao_data.bps+ao->chunk_size-1)/ao->chunk_size;
     ao->buffer_len = ao->num_chunks * ao->chunk_size;
-    ao->buffer = av_fifo_alloc(ao->buffer_len);
+    ao->buffer = av_fifo_alloc2(ao->buffer_len, 1, 0);
 
 	ao_msg(MSGT_AO,MSGL_V, "using %5d chunks of %d bytes (buffer len %d bytes)\n", (int)ao->num_chunks, (int)ao->chunk_size, (int)ao->buffer_len);
 
@@ -643,7 +645,7 @@ err_out2:
 err_out1:
     CloseComponent(ao->theOutputUnit);
 err_out:
-    av_fifo_free(ao->buffer);
+    av_fifo_freep2(&ao->buffer);
     free(ao);
     ao = NULL;
     return CONTROL_FALSE;
@@ -854,7 +856,7 @@ static int OpenSPDIF(void)
 
     ao->num_chunks = (ao_data.bps+ao->chunk_size-1)/ao->chunk_size;
     ao->buffer_len = ao->num_chunks * ao->chunk_size;
-    ao->buffer = av_fifo_alloc(ao->buffer_len);
+    ao->buffer = av_fifo_alloc2(ao->buffer_len, 1, 0);
 
     ao_msg(MSGT_AO,MSGL_V, "using %5d chunks of %d bytes (buffer len %d bytes)\n", (int)ao->num_chunks, (int)ao->chunk_size, (int)ao->buffer_len);
 
@@ -899,7 +901,7 @@ err_out:
             ao_msg(MSGT_AO, MSGL_WARN, "Could not release hogmode: [%4.4s]\n",
                    (char *)&err);
     }
-    av_fifo_free(ao->buffer);
+    av_fifo_freep2(&ao->buffer);
     free(ao);
     ao = NULL;
     return CONTROL_FALSE;
@@ -1068,7 +1070,7 @@ static OSStatus RenderCallbackSPDIF( AudioDeviceID inDevice,
                                     const AudioTimeStamp * inOutputTime,
                                     void * threadGlobals )
 {
-    int amt = av_fifo_size(ao->buffer);
+    int amt = (int)av_fifo_can_read(ao->buffer);
     int req = outOutputData->mBuffers[ao->i_stream_index].mDataByteSize;
 
     if (amt > req)
@@ -1124,14 +1126,14 @@ static int play(void* output_samples,int num_bytes,int flags)
 static void reset(void)
 {
   audio_pause();
-  av_fifo_reset(ao->buffer);
+  av_fifo_reset2(ao->buffer);
 }
 
 
 /* return available space */
 static int get_space(void)
 {
-  return ao->buffer_len - av_fifo_size(ao->buffer);
+  return ao->buffer_len - (int)av_fifo_can_read(ao->buffer);
 }
 
 
@@ -1139,7 +1141,7 @@ static int get_space(void)
 static float get_delay(void)
 {
   // inaccurate, should also contain the data buffered e.g. by the OS
-  return (float)av_fifo_size(ao->buffer)/(float)ao_data.bps;
+  return (float)av_fifo_can_read(ao->buffer)/(float)ao_data.bps;
 }
 
 
@@ -1149,8 +1151,8 @@ static void uninit(int immed)
   OSStatus            err = noErr;
 
   if (!immed) {
-    long long timeleft=(1000000LL*av_fifo_size(ao->buffer))/ao_data.bps;
-    ao_msg(MSGT_AO,MSGL_DBG2, "%d bytes left @%d bps (%d usec)\n", av_fifo_size(ao->buffer), ao_data.bps, (int)timeleft);
+    long long timeleft=(1000000LL*(int)av_fifo_can_read(ao->buffer))/ao_data.bps;
+    ao_msg(MSGT_AO,MSGL_DBG2, "%d bytes left @%d bps (%d usec)\n", (int)av_fifo_can_read(ao->buffer), ao_data.bps, (int)timeleft);
     usec_sleep((int)timeleft);
   }
 
@@ -1199,7 +1201,7 @@ static void uninit(int immed)
       }
   }
 
-  av_fifo_free(ao->buffer);
+  av_fifo_freep2(&ao->buffer);
   free(ao);
   ao = NULL;
 }
