--- trunk/libvo/osx_objc_common.m.orig	2026-10-02 13:06:51.000000000 -0600
+++ trunk/libvo/osx_objc_common.m	2026-10-02 13:08:29.000000000 -0600
@@ -31,6 +31,11 @@
 #include <CoreServices/../Frameworks/OSServices.framework/Headers/Power.h>
 #endif
 
+#if (MAC_OS_X_VERSION_MIN_REQUIRED < 1050) && defined(__i386__)
+#import <OpenGL/gl.h>
+#import <OpenGL/OpenGL.h>
+#endif
+
 //MPLAYER
 #include "config.h"
 #include "video_out.h"
