From a2386d5da7a979684f09c99457f2333667237086 Mon Sep 17 00:00:00 2001
From: Maxence Guesdon <maxence.guesdon@inria.fr>
Date: Fri, 9 Jan 2026 10:35:20 +0100
Subject: [PATCH] Add Sdl.Pixel.format_rgba32 (close #117).

diff --git a/src/tsdl.ml b/src/tsdl.ml
index b40e3ee..0c0976b 100644
--- src/tsdl.ml
+++ src/tsdl.ml
@@ -809,6 +809,7 @@ module Pixel = struct
   let format_index4lsb = i sdl_pixelformat_index4lsb
   let format_index4msb = i sdl_pixelformat_index4msb
   let format_index8 = i sdl_pixelformat_index8
+  let format_rgba32 = i sdl_pixelformat_rgba32
   let format_rgb332 = i sdl_pixelformat_rgb332
   let format_rgb444 = i sdl_pixelformat_rgb444
   let format_rgb555 = i sdl_pixelformat_rgb555
diff --git a/src/tsdl.mli b/src/tsdl.mli
index 23ae8ea..2cb2e77 100644
--- src/tsdl.mli
+++ src/tsdl.mli
@@ -578,6 +578,7 @@ module Pixel : sig
   val format_index4lsb : format_enum
   val format_index4msb : format_enum
   val format_index8 : format_enum
+  val format_rgba32 : format_enum
   val format_rgb332 : format_enum
   val format_rgb444 : format_enum
   val format_rgb555 : format_enum
diff --git a/support/consts_stub.c b/support/consts_stub.c
index 29e8af0..8fec45d 100644
--- support/consts_stub.c
+++ support/consts_stub.c
@@ -129,6 +129,7 @@ void consts (FILE *fd)
   int32_v (SDL_PIXELFORMAT_INDEX4LSB);
   int32_v (SDL_PIXELFORMAT_INDEX4MSB);
   int32_v (SDL_PIXELFORMAT_INDEX8);
+  int32_v (SDL_PIXELFORMAT_RGBA32);
   int32_v (SDL_PIXELFORMAT_RGB332);
   int32_v (SDL_PIXELFORMAT_RGB444);
   int32_v (SDL_PIXELFORMAT_RGB555);
