From 8003e8f89013cb95e77ca47935638450caacb4a1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 2 Aug 2025 02:17:52 +0800
Subject: [PATCH 2/4] cocoa.m: fix types

---
 uitoolkit/quartz/cocoa.m | 40 +++++++++++++++++++---------------------
 1 file changed, 19 insertions(+), 21 deletions(-)

diff --git uitoolkit/quartz/cocoa.m uitoolkit/quartz/cocoa.m
index f4f8f830..7ae34673 100644
--- uitoolkit/quartz/cocoa.m
+++ uitoolkit/quartz/cocoa.m
@@ -456,14 +456,12 @@ - (void)viewDidMoveToWindow {
 
   /* Change view size */
 
-  self.frame =
-      CGRectMake(uiwindow->x, y, ACTUAL_WIDTH(uiwindow), ACTUAL_HEIGHT(uiwindow));
+  [self setFrame:NSMakeRect(uiwindow->x, y, ACTUAL_WIDTH(uiwindow), ACTUAL_HEIGHT(uiwindow))];
 
   /* Creating scrollbar */
 
-  CGRect r =
-      CGRectMake(term->scrollbar.window.x, y, sb_width, ACTUAL_HEIGHT(uiwindow));
-  NSScroller *scroller = [[NSScroller alloc] initWithFrame:r];
+  NSRect sr = NSMakeRect(term->scrollbar.window.x, y, sb_width, ACTUAL_HEIGHT(uiwindow));
+  NSScroller *scroller = [[NSScroller alloc] initWithFrame:sr];
   [scroller setEnabled:YES];
   [scroller setTarget:self];
   [scroller setAction:@selector(scrollerAction:)];
@@ -498,15 +496,15 @@ - (void)viewDidMoveToWindow {
 
     uiwindow->parent->my_window = [self window];
 
-    r = [self window].frame;
-    r.size.width = ACTUAL_WIDTH(uiwindow->parent) + diff_x;
+    NSRect wr = [self window].frame;
+    wr.size.width = ACTUAL_WIDTH(uiwindow->parent) + diff_x;
     float new_height = ACTUAL_HEIGHT(uiwindow->parent) + diff_y;
-    r.origin.y += (r.size.height - new_height);
-    r.size.height = new_height;
+    wr.origin.y += (wr.size.height - new_height);
+    wr.size.height = new_height;
 
     [[self window] setResizeIncrements:NSMakeSize(uiwindow->width_inc,
                                                   uiwindow->height_inc)];
-    [[self window] setFrame:r display:NO];
+    [[self window] setFrame:wr display:NO];
 
     if (!IS_OPAQUE) {
       [self bgColorChanged];
@@ -519,17 +517,17 @@ - (void)viewDidMoveToWindow {
      * window position.
      */
     if (uiwindow->parent->x > 0 || uiwindow->parent->y > 0) {
-      r = [self window].frame;
-      r.origin.x += uiwindow->parent->x;
-      r.origin.y -= uiwindow->parent->y;
+      NSRect wr = [[self window] frame];
+      wr.origin.x += uiwindow->parent->x;
+      wr.origin.y -= uiwindow->parent->y;
 
-      [[self window] setFrame:r display:NO];
+      [[self window] setFrame:wr display:NO];
     }
   }
 
-  NSRect sr = [[[self window] screen] visibleFrame];
-  uiwindow->disp->width = sr.size.width;
-  uiwindow->disp->height = sr.size.height;
+  NSRect sr2 = [[[self window] screen] visibleFrame];
+  uiwindow->disp->width = sr2.size.width;
+  uiwindow->disp->height = sr2.size.height;
 
   /* Adjust scroller position */
   [scroller setFrameOrigin:NSMakePoint(term->scrollbar.window.x, y)];
@@ -1479,8 +1477,8 @@ void window_alloc(ui_window_t *root) {
 void window_dealloc(NSWindow *window) { [window close]; }
 
 void window_resize(NSWindow *window, int width, int height) {
-  CGRect wr = window.frame;
-  CGSize vs = ((NSView *)window.contentView).frame.size;
+  NSRect wr = [window frame];
+  NSSize vs = ((NSView *)window.contentView).frame.size;
   int diff_x = wr.size.width - vs.width;
   int diff_y = wr.size.height - vs.height;
 
@@ -1491,8 +1489,8 @@ void window_resize(NSWindow *window, int width, int height) {
 }
 
 void window_move_resize(NSWindow *window, int x, int y, int width, int height) {
-  CGRect wr = window.frame;
-  CGSize vs = ((NSView *)window.contentView).frame.size;
+  NSRect wr = [window frame];
+  NSSize vs = ((NSView *)window.contentView).frame.size;
   int diff_x = wr.size.width - vs.width;
   int diff_y = wr.size.height - vs.height;
 
-- 
2.24.3 (Apple Git-128)

