From 8003e8f89013cb95e77ca47935638450caacb4a1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 2 Aug 2025 02:17:52 +0800
Subject: [PATCH 2/4] cocoa.m: fix types

---
 uitoolkit/quartz/cocoa.m | 40 +++++++++++++++++++---------------------
 1 file changed, 19 insertions(+), 21 deletions(-)

diff --git uitoolkit/quartz/cocoa.m uitoolkit/quartz/cocoa.m
index f4f8f830..7ae34673 100644
--- uitoolkit/quartz/cocoa.m
+++ uitoolkit/quartz/cocoa.m
@@ -456,14 +456,12 @@ - (void)viewDidMoveToWindow {
 
   /* Change view size */
 
-  self.frame =
-      CGRectMake(uiwindow->x, y, ACTUAL_WIDTH(uiwindow), ACTUAL_HEIGHT(uiwindow));
+  [self setFrame:NSMakeRect(uiwindow->x, y, ACTUAL_WIDTH(uiwindow), ACTUAL_HEIGHT(uiwindow))];
 
   /* Creating scrollbar */
 
-  CGRect r =
-      CGRectMake(term->scrollbar.window.x, y, sb_width, ACTUAL_HEIGHT(uiwindow));
-  NSScroller *scroller = [[NSScroller alloc] initWithFrame:r];
+  NSRect sr = NSMakeRect(term->scrollbar.window.x, y, sb_width, ACTUAL_HEIGHT(uiwindow));
+  NSScroller *scroller = [[NSScroller alloc] initWithFrame:sr];
   [scroller setEnabled:YES];
   [scroller setTarget:self];
   [scroller setAction:@selector(scrollerAction:)];
@@ -498,15 +496,15 @@ - (void)viewDidMoveToWindow {
 
     uiwindow->parent->my_window = [self window];
 
-    r = [self window].frame;
-    r.size.width = ACTUAL_WIDTH(uiwindow->parent) + diff_x;
+    NSRect wr = [self window].frame;
+    wr.size.width = ACTUAL_WIDTH(uiwindow->parent) + diff_x;
     float new_height = ACTUAL_HEIGHT(uiwindow->parent) + diff_y;
-    r.origin.y += (r.size.height - new_height);
-    r.size.height = new_height;
+    wr.origin.y += (wr.size.height - new_height);
+    wr.size.height = new_height;
 
     [[self window] setResizeIncrements:NSMakeSize(uiwindow->width_inc,
                                                   uiwindow->height_inc)];
-    [[self window] setFrame:r display:NO];
+    [[self window] setFrame:wr display:NO];
 
     if (!IS_OPAQUE) {
       [self bgColorChanged];
@@ -519,17 +517,17 @@ - (void)viewDidMoveToWindow {
      * window position.
      */
     if (uiwindow->parent->x > 0 || uiwindow->parent->y > 0) {
-      r = [self window].frame;
-      r.origin.x += uiwindow->parent->x;
-      r.origin.y -= uiwindow->parent->y;
+      NSRect wr = [[self window] frame];
+      wr.origin.x += uiwindow->parent->x;
+      wr.origin.y -= uiwindow->parent->y;
 
-      [[self window] setFrame:r display:NO];
+      [[self window] setFrame:wr display:NO];
     }
   }
 
-  NSRect sr = [[[self window] screen] visibleFrame];
-  uiwindow->disp->width = sr.size.width;
-  uiwindow->disp->height = sr.size.height;
+  NSRect sr2 = [[[self window] screen] visibleFrame];
+  uiwindow->disp->width = sr2.size.width;
+  uiwindow->disp->height = sr2.size.height;
 
   /* Adjust scroller position */
   [scroller setFrameOrigin:NSMakePoint(term->scrollbar.window.x, y)];

 
From ed024f0c010a2345e034f7e8789b90ff32ab053a Mon Sep 17 00:00:00 2001
From: arakiken <arakiken@users.sf.net>
Date: Fri, 15 Aug 2025 13:37:57 +0900
Subject: [PATCH] * man/mlterm.1, README.ja, PROTOCOL: Updated. *

diff --git a/uitoolkit/quartz/cocoa.m b/uitoolkit/quartz/cocoa.m
index c75d2fd7..3a1c8e2f 100644
--- uitoolkit/quartz/cocoa.m
+++ uitoolkit/quartz/cocoa.m
@@ -1486,8 +1486,8 @@ void window_resize(NSWindow *window, int width, int height) {
 }
 
 void window_move_resize(NSWindow *window, int x, int y, int width, int height) {
-  CGRect wr = window.frame;
-  CGSize vs = ((NSView *)window.contentView).frame.size;
+  NSRect wr = window.frame;
+  NSSize vs = ((NSView *)window.contentView).frame.size;
   int diff_x = wr.size.width - vs.width;
   int diff_y = wr.size.height - vs.height;
 
From 281c1da207fe6410b1de12957a030923f4cb6f5d Mon Sep 17 00:00:00 2001
From: arakiken <arakiken@users.sf.net>
Date: Sat, 16 Aug 2025 20:18:50 +0900
Subject: [PATCH] * cocoa.m: CGRect -> NSRect

diff --git a/uitoolkit/quartz/cocoa.m b/uitoolkit/quartz/cocoa.m
index 3a1c8e2f..894d8243 100644
--- uitoolkit/quartz/cocoa.m
+++ uitoolkit/quartz/cocoa.m
@@ -1474,8 +1474,8 @@ void window_alloc(ui_window_t *root) {
 void window_dealloc(NSWindow *window) { [window close]; }
 
 void window_resize(NSWindow *window, int width, int height) {
-  CGRect wr = window.frame;
-  CGSize vs = ((NSView *)window.contentView).frame.size;
+  NSRect wr = window.frame;
+  NSSize vs = ((NSView *)window.contentView).frame.size;
   int diff_x = wr.size.width - vs.width;
   int diff_y = wr.size.height - vs.height;
 
diff --git a/uitoolkit/ui_screen.c b/uitoolkit/ui_screen.c
index cb87a7f3..7fe7681b 100644
--- uitoolkit/ui_screen.c
+++ uitoolkit/ui_screen.c
@@ -3725,8 +3725,8 @@ static void button_motion(ui_window_t *win, XMotionEvent *event) {
 
     if (!screen->grab_scroll) {
 #if 0
-      bl_debug_printf("Motion Time %d %d y %d %d\n", screen->flick_time, event->time,
-                      screen->flick_cur_y, event->y);
+      bl_debug_printf("Motion Time %d %d y %d %d state %d.\n", screen->flick_time, event->time,
+                      screen->flick_cur_y, event->y, event->state);
 #endif
       if (abs(diff) >= ui_line_height(screen)) {
         if (screen->flick_time + 500 /* msec */ > event->time) {
