From e0d6b427cdf4d90896ed5d271a5d98e160ed9a49 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 13 Jul 2026 11:12:57 +0800
Subject: [PATCH 4/6] Wayland: minor tweak for macOS

---
 uitoolkit/wayland/ui_display.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git uitoolkit/wayland/ui_display.c uitoolkit/wayland/ui_display.c
index e07aa706..93cee378 100644
--- uitoolkit/wayland/ui_display.c
+++ uitoolkit/wayland/ui_display.c
@@ -402,17 +402,17 @@ static void registry_global(void *data, struct wl_registry *registry, uint32_t n
 #endif
 #endif /* COMPAT_LIBVTE */
   else if (strcmp(interface, "wl_shm") == 0) {
-    wlserv->shm = wl_registry_bind(registry, name, &wl_shm_interface, BL_MIN(version, 2));
+    wlserv->shm = wl_registry_bind(registry, name, &wl_shm_interface, BL_MIN(version, 1));
   } else if (strcmp(interface, "wl_seat") == 0) {
     /* XXX BL_MIN(version, 10) makes 'rate' of keyboard_repeat_info() 0 unexpectedly in kde-plasma 6.6.5. */
     wlserv->seat = wl_registry_bind(registry, name, &wl_seat_interface,
-                                    BL_MIN(version, 4 /* 10 */)); /* >= 4 for keyboard_repeat_info */
+                                    BL_MIN(version, 2 /* 10 */)); /* >= 4 for keyboard_repeat_info */
     wl_seat_add_listener(wlserv->seat, &seat_listener, wlserv);
   } else if (strcmp(interface, "wl_output") == 0) {
-    wlserv->output = wl_registry_bind(registry, name, &wl_output_interface, BL_MIN(version, 4));
+    wlserv->output = wl_registry_bind(registry, name, &wl_output_interface, BL_MIN(version, 1));
   } else if (strcmp(interface, "wl_data_device_manager") == 0) {
     wlserv->data_device_manager = wl_registry_bind(registry, name,
-                                                   &wl_data_device_manager_interface, BL_MIN(version, 3));
+                                                   &wl_data_device_manager_interface, BL_MIN(version, 2));
   }
 #ifdef GTK_PRIMARY
   else if (strcmp(interface, "gtk_primary_selection_device_manager") == 0) {
-- 
2.54.0

