From cafb978f34bafa3631a9fd34b9e4a163b59fb7a9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 13 Jul 2026 11:53:01 +0800
Subject: [PATCH 6/6] ui_font: add macOS case

---
 uitoolkit/fb/ui_font.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git uitoolkit/fb/ui_font.c uitoolkit/fb/ui_font.c
index 266b4838..56375b08 100644
--- uitoolkit/fb/ui_font.c
+++ uitoolkit/fb/ui_font.c
@@ -2032,6 +2032,14 @@ ui_font_t *ui_font_new(Display *display, vt_font_t id, int size_attr, ui_type_en
         font_file = "/usr/X11R6/lib/X11/fonts/misc/10x20.pcf.gz";
         percent = 0;
       }
+#elif defined(__APPLE__)
+      if (stat("/opt/local/share/fonts/X11/misc/unifont.pcf.gz", &st) == 0) {
+        font_file = "/opt/local/share/fonts/X11/misc/unifont.pcf.gz";
+        percent = 100;
+      } else {
+        font_file = "/opt/local/share/fonts/misc/10x20.pcf.gz";
+        percent = 0;
+      }
 #elif defined(__ANDROID__)
       if (stat("/system/fonts/DroidSansMono.ttf", &st) == 0) {
         font_file = "/system/fonts/DroidSansMono.ttf";
-- 
2.54.0

