diff --git a/include/openssl/target.h b/include/openssl/target.h
index 2c03389..dd9d75e 100644
--- a/include/openssl/target.h
+++ b/include/openssl/target.h
@@ -45,6 +45,17 @@
 #define OPENSSL_RISCV64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 4
 #define OPENSSL_32_BIT
+#elif (defined(__ppc64__) || defined(__powerpc64__)) && defined(_LITTLE_ENDIAN)
+#define OPENSSL_64_BIT
+#define OPENSSL_PPC64LE
+#elif defined(__ppc64__) || defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#define OPENSSL_PPC64
+#define OPENSSL_BIGENDIAN
+#elif defined(__ppc__) || defined(__powerpc__)
+#define OPENSSL_32_BIT
+#define OPENSSL_PPC
+#define OPENSSL_BIGENDIAN
 #elif defined(__pnacl__)
 #define OPENSSL_32_BIT
 #define OPENSSL_PNACL
diff --git a/crypto/internal.h b/crypto/internal.h
index f7cc137..2f33b05 100644
--- a/crypto/internal.h
+++ b/crypto/internal.h
@@ -892,79 +892,140 @@ inline void *OPENSSL_memset(void *dst, int c, size_t n) {
 inline uint16_t CRYPTO_load_u16_le(const void *in) {
   uint16_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return CRYPTO_bswap2(v);
+#else
   return v;
+#endif
 }
 
 inline void CRYPTO_store_u16_le(void *out, uint16_t v) {
+#ifdef OPENSSL_BIGENDIAN
+  v = CRYPTO_bswap2(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline uint16_t CRYPTO_load_u16_be(const void *in) {
   uint16_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return v;
+#else
   return CRYPTO_bswap2(v);
+#endif
 }
 
 inline void CRYPTO_store_u16_be(void *out, uint16_t v) {
+#ifndef OPENSSL_BIGENDIAN
   v = CRYPTO_bswap2(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline uint32_t CRYPTO_load_u32_le(const void *in) {
   uint32_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return CRYPTO_bswap4(v);
+#else
   return v;
+#endif
 }
 
 inline void CRYPTO_store_u32_le(void *out, uint32_t v) {
+#ifdef OPENSSL_BIGENDIAN
+  v = CRYPTO_bswap4(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline uint32_t CRYPTO_load_u32_be(const void *in) {
   uint32_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return v;
+#else
   return CRYPTO_bswap4(v);
+#endif
 }
 
 inline void CRYPTO_store_u32_be(void *out, uint32_t v) {
+#ifndef OPENSSL_BIGENDIAN
   v = CRYPTO_bswap4(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline uint64_t CRYPTO_load_u64_le(const void *in) {
   uint64_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return CRYPTO_bswap8(v);
+#else
   return v;
+#endif
 }
 
 inline void CRYPTO_store_u64_le(void *out, uint64_t v) {
+#ifdef OPENSSL_BIGENDIAN
+  v = CRYPTO_bswap8(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline uint64_t CRYPTO_load_u64_be(const void *ptr) {
   uint64_t ret;
   OPENSSL_memcpy(&ret, ptr, sizeof(ret));
+#ifdef OPENSSL_BIGENDIAN
+  return ret;
+#else
   return CRYPTO_bswap8(ret);
+#endif
 }
 
 inline void CRYPTO_store_u64_be(void *out, uint64_t v) {
+#ifndef OPENSSL_BIGENDIAN
   v = CRYPTO_bswap8(v);
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline crypto_word_t CRYPTO_load_word_le(const void *in) {
   crypto_word_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+#if defined(OPENSSL_64_BIT)
+  static_assert(sizeof(v) == 8, "crypto_word_t has unexpected size");
+  return CRYPTO_bswap8(v);
+#else
+  static_assert(sizeof(v) == 4, "crypto_word_t has unexpected size");
+  return CRYPTO_bswap4(v);
+#endif
+#else
   return v;
+#endif
 }
 
 inline void CRYPTO_store_word_le(void *out, crypto_word_t v) {
+#ifdef OPENSSL_BIGENDIAN
+#if defined(OPENSSL_64_BIT)
+  static_assert(sizeof(v) == 8, "crypto_word_t has unexpected size");
+  v = CRYPTO_bswap8(v);
+#else
+  static_assert(sizeof(v) == 4, "crypto_word_t has unexpected size");
+  v = CRYPTO_bswap4(v);
+#endif
+#endif
   OPENSSL_memcpy(out, &v, sizeof(v));
 }
 
 inline crypto_word_t CRYPTO_load_word_be(const void *in) {
   crypto_word_t v;
   OPENSSL_memcpy(&v, in, sizeof(v));
+#ifdef OPENSSL_BIGENDIAN
+  return v;
+#else
 #if defined(OPENSSL_64_BIT)
   static_assert(sizeof(v) == 8, "crypto_word_t has unexpected size");
   return CRYPTO_bswap8(v);
@@ -972,6 +1033,7 @@ inline crypto_word_t CRYPTO_load_word_be(const void *in) {
   static_assert(sizeof(v) == 4, "crypto_word_t has unexpected size");
   return CRYPTO_bswap4(v);
 #endif
+#endif
 }
 
 
diff --git a/crypto/fipsmodule/bn/bytes.cc.inc b/crypto/fipsmodule/bn/bytes.cc.inc
index 7a62d46..649eff8 100644
--- a/crypto/fipsmodule/bn/bytes.cc.inc
+++ b/crypto/fipsmodule/bn/bytes.cc.inc
@@ -109,9 +109,17 @@ BIGNUM *BN_lebin2bn(const uint8_t *in, size_t len, BIGNUM *ret) {
   // Make sure the top bytes will be zeroed.
   ret->d[num_words - 1] = 0;
 
+#ifdef OPENSSL_BIGENDIAN
+  // On big-endian, byte-swap the input data into the internal representation.
+  uint8_t *out = (uint8_t *)ret->d;
+  for (size_t i = 0; i < len; i++) {
+    out[i ^ (BN_BYTES - 1)] = in[i];
+  }
+#else
   // We only support little-endian platforms, so we can simply memcpy the
   // internal representation.
   OPENSSL_memcpy(ret->d, in, len);
+#endif
   return ret;
 }
 
@@ -127,7 +135,11 @@ static int fits_in_bytes(const BN_ULONG *words, size_t num_words,
   size_t tot_bytes = num_words * sizeof(BN_ULONG);
   uint8_t mask = 0;
   for (size_t i = num_bytes; i < tot_bytes; i++) {
+#ifdef OPENSSL_BIGENDIAN
+    mask |= bytes[i ^ (BN_BYTES - 1)];
+#else
     mask |= bytes[i];
+#endif
   }
   return mask == 0;
 }
@@ -138,7 +150,11 @@ void bssl::bn_assert_fits_in_bytes(const BIGNUM *bn, size_t num) {
   if (tot_bytes > num) {
     CONSTTIME_DECLASSIFY(bytes + num, tot_bytes - num);
     for (size_t i = num; i < tot_bytes; i++) {
+#ifdef OPENSSL_BIGENDIAN
+      assert(bytes[i ^ (BN_BYTES - 1)] == 0);
+#else
       assert(bytes[i] == 0);
+#endif
     }
     (void)bytes;
   }
@@ -149,17 +165,24 @@ void bssl::bn_words_to_big_endian(uint8_t *out, size_t out_len,
   // The caller should have selected an output length without truncation.
   declassify_assert(fits_in_bytes(in, in_len, out_len));
 
-  // We only support little-endian platforms, so the internal representation is
-  // also little-endian as bytes. We can simply copy it in reverse.
   const uint8_t *bytes = (const uint8_t *)in;
   size_t num_bytes = in_len * sizeof(BN_ULONG);
   if (out_len < num_bytes) {
     num_bytes = out_len;
   }
 
+#ifdef OPENSSL_BIGENDIAN
+  // On big-endian, internal representation is big-endian, so swap index.
+  for (size_t i = 0; i < num_bytes; i++) {
+    out[out_len - i - 1] = bytes[i ^ (BN_BYTES - 1)];
+  }
+#else
+  // We only support little-endian platforms, so the internal representation is
+  // also little-endian as bytes. We can simply copy it in reverse.
   for (size_t i = 0; i < num_bytes; i++) {
     out[out_len - i - 1] = bytes[i];
   }
+#endif
   // Pad out the rest of the buffer with zeroes.
   OPENSSL_memset(out, 0, out_len - num_bytes);
 }
@@ -175,15 +198,22 @@ int BN_bn2le_padded(uint8_t *out, size_t len, const BIGNUM *in) {
     return 0;
   }
 
-  // We only support little-endian platforms, so we can simply memcpy into the
-  // internal representation.
   const uint8_t *bytes = (const uint8_t *)in->d;
   size_t num_bytes = in->width * BN_BYTES;
   if (len < num_bytes) {
     num_bytes = len;
   }
 
+#ifdef OPENSSL_BIGENDIAN
+  // On big-endian, byte-swap the internal representation to little-endian output.
+  for (size_t i = 0; i < num_bytes; i++) {
+    out[i] = bytes[i ^ (BN_BYTES - 1)];
+  }
+#else
+  // We only support little-endian platforms, so we can simply memcpy into the
+  // internal representation.
   OPENSSL_memcpy(out, bytes, num_bytes);
+#endif
   // Pad out the rest of the buffer with zeroes.
   OPENSSL_memset(out + num_bytes, 0, len - num_bytes);
   return 1;
