From d4ff5c2f3befde2ec55ef8a9ea57e4666b555092 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 1 Jul 2025 11:31:43 +0800
Subject: [PATCH 1/2] target.h: unbreak arch detection

---
 include/openssl/target.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git include/openssl/target.h include/openssl/target.h
index 8d4763d0b..f96456472 100644
--- include/openssl/target.h
+++ include/openssl/target.h
@@ -45,6 +45,12 @@
 #define OPENSSL_RISCV64
 #elif defined(__riscv) && __SIZEOF_POINTER__ == 4
 #define OPENSSL_32_BIT
+#elif defined(__ppc64__) || defined(__powerpc64__)
+#define OPENSSL_64_BIT
+#define OPENSSL_PPC64
+#elif defined(__ppc__) || defined(__powerpc__)
+#define OPENSSL_32_BIT
+#define OPENSSL_PPC
 #elif defined(__pnacl__)
 #define OPENSSL_32_BIT
 #define OPENSSL_PNACL
-- 
2.48.0

