From 3afc488ac3a43c2e14c519d225a435aa178347e3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Oct 2025 14:38:38 +0800
Subject: [PATCH] entropy_cpu_stubs.c: fix powerpc

---
 src/native/entropy_cpu_stubs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/native/entropy_cpu_stubs.c b/src/native/entropy_cpu_stubs.c
index c5af9427..588c57c9 100644
--- src/native/entropy_cpu_stubs.c
+++ src/native/entropy_cpu_stubs.c
@@ -126,7 +126,7 @@ static inline uint64_t read_virtual_count(void)
 }
 #endif /* aarch64 */
 
-#if defined (__powerpc64__)
+#if defined (__powerpc64__) || defined(__POWERPC__)
 /* from clang's builtin version and gperftools at
 https://chromium.googlesource.com/external/gperftools/+/master/src/base/cycleclock.h
 */
@@ -183,7 +183,7 @@ CAMLprim value mc_cycle_counter (value __unused(unit)) {
   return Val_long (__rdtsc ());
 #elif defined (__arm__) || defined (__aarch64__)
   return Val_long (read_virtual_count ());
-#elif defined(__powerpc64__)
+#elif defined(__powerpc64__) || defined(__POWERPC__)
   return Val_long (read_cycle_counter ());
 #elif defined(__riscv) && (64 == __riscv_xlen)
   return Val_long (cycle_count ());

From e4a2c0a4b61e9fdb86b286bec91df4d0a0902319 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 17 Oct 2025 14:31:22 +0800
Subject: [PATCH] mirage_crypto.h: add powerpc macros

---
 src/native/mirage_crypto.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/native/mirage_crypto.h b/src/native/mirage_crypto.h
index 5496a965..f623c349 100644
--- src/native/mirage_crypto.h
+++ src/native/mirage_crypto.h
@@ -50,9 +50,9 @@ extern struct _mc_cpu_features mc_detected_cpu_features;
 
 #endif /* __mc_ACCELERATE__ */
 
-#if defined (__x86_64__) || defined (__aarch64__) || defined (__powerpc64__) || (64 == __riscv_xlen) || defined (__s390x__) || (defined (__mips__) && _MIPS_SIM==_ABI64) || defined (__loongarch_lp64) || (1 == _WIN64)
+#if defined (__x86_64__) || defined (__aarch64__) || defined (__powerpc64__) || defined (__ppc64__) || (64 == __riscv_xlen) || defined (__s390x__) || (defined (__mips__) && _MIPS_SIM==_ABI64) || defined (__loongarch_lp64) || (1 == _WIN64)
 #define ARCH_64BIT
-#elif defined (__i386__) || defined (__arm__) || (32 == __riscv_xlen) || (defined (__mips__) && _MIPS_SIM==_ABIO32) || (1 == _WIN32)
+#elif defined (__i386__) || defined (__arm__) || (32 == __riscv_xlen) || (defined (__mips__) && _MIPS_SIM==_ABIO32) || defined (__ppc__) || (1 == _WIN32)
 #define ARCH_32BIT
 #else
 #error "unsupported platform"
