--- Makefile	2025-02-28 11:20:10.000000000 +0800
+++ Makefile	2025-03-22 05:01:17.000000000 +0800
@@ -25,7 +25,7 @@
 LIBGIT2_VERREQ := libgit2 >= 1.7.0, libgit2 < 1.10.0
 LIBCURL_VERREQ := libcurl >= 7.79.1, libcurl < 9.0.0
 NLOHMANN_JSON_VERREQ := nlohmann_json >= 3.10.5, nlohmann_json < 4.0.0
-TBB_VERREQ := tbb >= 2021.5.0, tbb < 2023.0.0
+TBB_VERREQ := tbb32 >= 2021.5.0, tbb32 < 2023.0.0
 FMT_VERREQ := fmt >= 9.0.0, fmt < 12.0.0
 SPDLOG_VERREQ := spdlog >= 1.8.0, spdlog < 2.0.0
 TOML11_VER := $(shell grep -m1 toml11 cabin.toml | sed 's/.*rev = \(.*\)}/\1/' | tr -d '"')
