--- test/CMakeLists.txt	2021-10-26 14:05:04.000000000 +0800
+++ test/CMakeLists.txt	2024-12-04 12:48:22.000000000 +0800
@@ -6,7 +6,7 @@
 # I'm sure I could do this better but I don't want to spend my time on cmake really
 function(toxext_test test_name)
 	add_executable(${test_name} ${ARGN})
-	if (UNIX)
+	if (UNIX AND NOT APPLE)
 		target_compile_options(${test_name} PRIVATE -fsanitize=address)
 		target_link_libraries(${test_name} "-fsanitize=address")
 	endif()
