--- src/openzl/common/assertion.h.orig	2025-10-06 11:47:11.000000000 +0800
+++ src/openzl/common/assertion.h	2026-02-04 04:25:50.000000000 +0800
@@ -334,7 +334,11 @@
 #if ZL_ENABLE_STATIC_ASSERT
 #    if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
 #        include <assert.h>
-#        define ZL_STATIC_ASSERT_IMPL(expr, msg) static_assert(expr, msg)
+#        ifdef static_assert
+#            define ZL_STATIC_ASSERT_IMPL(expr, msg) static_assert(expr, msg)
+#        else
+#            define ZL_STATIC_ASSERT_IMPL(expr, msg) _Static_assert(expr, msg)
+#        endif
 #    else
 
 /* This fallback static assert idiom was copied from the following URL, where
