From b7e658806cc9d2527c8ee00b1105a4d99809ab74 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 18:10:30 +0800
Subject: [PATCH] kpty.cpp: fix for Apple

---
 kpty/kpty.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kpty/kpty.cpp b/kpty/kpty.cpp
index ff04244c..892e36e0 100644
--- a/kpty/kpty.cpp
+++ b/kpty/kpty.cpp
@@ -301,7 +301,7 @@ void KPty::login(const char *user, const char *remotehost)
 
 #if defined(HAVE_UTIL_LOGINX)
     ::loginx(&l_struct);
-#elif defined(HAVE_LOGIN) || defined(HAVE_UTIL_LOGIN)
+#elif (defined(HAVE_LOGIN) || defined(HAVE_UTIL_LOGIN)) && !defined(__APPLE__)
     ::login(&l_struct);
 #elif defined(HAVE_UTMPX)
 # ifdef _PATH_UTMPX
