From 7c0a181237c8087d052b6b0e64aab19b7b548eb7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 18:11:05 +0800
Subject: [PATCH] karchive.cpp: fix for Apple

---
 kutils/karchive/karchive.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kutils/karchive/karchive.cpp b/kutils/karchive/karchive.cpp
index c861b6c0..ba954df1 100644
--- a/kutils/karchive/karchive.cpp
+++ b/kutils/karchive/karchive.cpp
@@ -45,6 +45,12 @@
 #  define PATH_MAX _POSIX_PATH_MAX
 #endif
 
+#ifdef __APPLE__
+#define st_atim st_atimespec
+#define st_ctim st_ctimespec
+#define st_mtim st_mtimespec
+#endif
+
 // NOTE: many KArchive users are not doing listing and extraction in a thread which means that the
 // UI will be "frozen" while that happens so process events while doing so
 #define KARCHIVE_TIMEOUT 250
