From 7269d03f5e01a0c31c51ebdb41ef03cca600b526 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 18:19:17 +0800
Subject: [PATCH] Fix webp linking

---
 kimgio/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/kimgio/CMakeLists.txt b/kimgio/CMakeLists.txt
index 39cd42d5..5283aaab 100644
--- a/kimgio/CMakeLists.txt
+++ b/kimgio/CMakeLists.txt
@@ -25,7 +25,10 @@ if(WEBP_FOUND)
     kde4_add_plugin(kimg_webp webp.cpp)
     target_link_libraries(kimg_webp
         ${QT_QTGUI_LIBRARY}
-        ${WEBP_LIBRARIES}
+        webp
+        webpdecoder
+        webpdemux
+        webpmux
         kdecore
     )
     set_target_properties(kimg_webp PROPERTIES
