From 4b3eddf68cc93b17244ee49a487c0c7755f84047 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 25 Jun 2025 11:34:17 +0800
Subject: [PATCH] karchive.cpp: include missing <limits.h>

---
 kutils/karchive/karchive.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kutils/karchive/karchive.cpp b/kutils/karchive/karchive.cpp
index ba954df1..9ea7f55b 100644
--- a/kutils/karchive/karchive.cpp
+++ b/kutils/karchive/karchive.cpp
@@ -41,6 +41,8 @@
 #  include <string.h>
 #endif
 
+#include <limits.h>
+
 #ifndef PATH_MAX
 #  define PATH_MAX _POSIX_PATH_MAX
 #endif
