--- Makefile	2024-08-15 00:26:55.000000000 +0800
+++ Makefile	2025-12-24 12:30:11.000000000 +0800
@@ -32,8 +32,8 @@
 FFDIR := ffmpeg-$(FF_VER)
 
 ifeq ($(FF_VER), shared)
-	CXXFLAGS += -isystem/usr/include/ffmpeg
-	LDFLAGS += -lavformat -lavcodec -lavutil
+	CXXFLAGS += -isystem@FFMPEG_ROOT@/include
+	LDFLAGS += -L@FFMPEG_ROOT@/lib -lavformat -lavcodec -lavutil
 else
 	CXXFLAGS += -isystem./$(FFDIR)
 	LDFLAGS += -L$(FFDIR)/libavformat -lavformat
