--- tls/s2n_ktls_io.c
+++ tls/s2n_ktls_io.c	2025-07-22 14:24:20.000000000 +0800
@@ -221,7 +221,8 @@
         .msg_iovlen = msg_iovlen,
     };
 
-    char control_data[S2N_KTLS_CONTROL_BUFFER_SIZE] = { 0 };
+    char control_data[S2N_KTLS_CONTROL_BUFFER_SIZE];
+    memset(control_data, 0, S2N_KTLS_CONTROL_BUFFER_SIZE);
     RESULT_GUARD(s2n_ktls_set_control_data(&msg, control_data, sizeof(control_data),
             S2N_TLS_SET_RECORD_TYPE, record_type));
 
@@ -264,7 +265,8 @@
      * member of the msghdr with the length of the control message
      * buffer.
      */
-    char control_data[S2N_KTLS_CONTROL_BUFFER_SIZE] = { 0 };
+    char control_data[S2N_KTLS_CONTROL_BUFFER_SIZE];
+    memset(control_data, 0, S2N_KTLS_CONTROL_BUFFER_SIZE);
     msg.msg_controllen = sizeof(control_data);
     msg.msg_control = control_data;
 
