--- src/shrpx_log.cc	2025-09-15 20:03:54.000000000 +0800
+++ src/shrpx_log.cc	2025-09-16 12:33:17.000000000 +0800
@@ -418,7 +418,7 @@
     }
 
     auto n = std::min(std::ranges::size(dest),
-                      as_unsigned(std::ranges::distance(safe_first, p)));
+                      static_cast<std::size_t>(as_unsigned(std::ranges::distance(safe_first, p))));
     std::ranges::copy_n(safe_first, as_signed(n), std::ranges::begin(dest));
     dest = dest.subspan(n);
 
@@ -436,7 +436,7 @@
 
   auto n = std::min(
     std::ranges::size(dest),
-    as_unsigned(std::ranges::distance(safe_first, std::ranges::end(src))));
+    static_cast<std::size_t>(as_unsigned(std::ranges::distance(safe_first, std::ranges::end(src)))));
   std::ranges::copy_n(safe_first, as_signed(n), std::ranges::begin(dest));
 
   return dest.subspan(n);
