--- cmake/luametatex.cmake	2025-03-11 01:13:39.000000000 +0800
+++ cmake/luametatex.cmake	2025-03-17 06:26:02.000000000 +0800
@@ -80,7 +80,7 @@
 
 if (DEFINED LMT_OPTIMIZE)
     # we strip anyway
-elseif (CMAKE_HOST_SOLARIS)
+elseif (APPLE OR CMAKE_HOST_SOLARIS)
     # no strip
 elseif (CMAKE_C_COMPILER_ID MATCHES "GNU")
     # -g -S -d : remove all debugging symbols & sections

--- CMakeLists.txt	2026-02-21 17:34:05.000000000 +0800
+++ CMakeLists.txt	2026-03-05 19:05:21.000000000 +0800
@@ -172,8 +172,10 @@
 
       # e.g. mp crashes when we have > 4K currentpicture clips in a row 
 
-    # set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=1000000")
-      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=2621440")
+    if (NOT APPLE)
+        # set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=1000000")
+        set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-z,stack-size=2621440")
+    endif()
 
     endif()
 
