From 29596e6049db74f1a09f65fdc5db95125cb051c5 Mon Sep 17 00:00:00 2001
From: Federico Perini <federico.perini@gmail.com>
Date: Sun, 30 Nov 2025 16:35:08 +0100
Subject: [PATCH] fix(lapack): add missing string module dependencies (#1063)

---
 src/lapack/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lapack/CMakeLists.txt b/src/lapack/CMakeLists.txt
index c988164c0..b8096c787 100644
--- src/lapack/CMakeLists.txt
+++ src/lapack/CMakeLists.txt
@@ -7,6 +7,9 @@ set(lapack_fppFiles
     ../stdlib_io.fypp
     ../stdlib_ascii.fypp
     ../stdlib_string_type.fypp
+    ../stdlib_string_type_constructor.fypp
+    ../stdlib_strings.fypp
+    ../stdlib_strings_to_string.fypp
     stdlib_lapack_base.fypp
     stdlib_lapack_solve.fypp
     stdlib_lapack_others.fypp
