--- setup.ml	2020-03-29 00:19:15.000000000 +0800
+++ setup.ml	2026-09-16 15:30:47.000000000 +0800
@@ -24,11 +24,6 @@
 (* Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA              *)
 (******************************************************************************)
 
-let () =
-  try
-    Topdirs.dir_directory (Sys.getenv "OCAML_TOPLEVEL_PATH")
-  with Not_found -> ()
-;;
 #use "topfind";;
 #require "oasis.dynrun";;
 open OASISDynRun;;
--- src/gufoEngine.ml	2026-09-16 15:29:58.000000000 +0800
+++ src/gufoEngine.ml	2026-09-16 15:48:25.000000000 +0800
@@ -253,7 +253,7 @@
       match resa, resb with
         | Some _, None -> 1
         | None, Some _ -> -1
-        | Some a, Some b when a != b -> compare a b
+        | Some a, Some b when a != b -> Stdlib.compare a b
         | _,_ -> 0
     in
     let cmp_print printa printb = 
@@ -382,7 +382,7 @@
             | MOTypeBoolVal _, MOTypeBoolVal _ ->
                   0
             | MOTypeIntVal aaaa, MOTypeIntVal bbbb ->
-                  compare aaaa bbbb
+                  Stdlib.compare aaaa bbbb
             | MOTypeFloatVal aaaa, MOTypeFloatVal bbbb ->
                 if aaaa = bbbb then 0 
                 else 
@@ -891,7 +891,7 @@
     | MOSimple_val(MOFun_val fv) -> 
       (*apply the code using arg2valMap when needed.*)
       (*we check if it is a partial application *)
-      (match (Pervasives.compare (List.length arglst) (List.length fv.mofv_args_id)) with
+      (match (Stdlib.compare (List.length arglst) (List.length fv.mofv_args_id)) with
         | i when i < 0 -> 
             apply_partial_fun funval arglst
         | 0 -> 
--- src/gufoParsedToOpt.ml	2020-03-29 00:19:15.000000000 +0800
+++ src/gufoParsedToOpt.ml	2026-09-16 15:53:44.000000000 +0800
@@ -1766,7 +1766,7 @@
                       * outbounds. 
                       * This should be possible only if the ret is itself a fun.
                       * *)
-                    (match (Pervasives.compare (List.length args) (List.length args_type) )
+                    (match (Stdlib.compare (List.length args) (List.length args_type) )
                     with 
                       | i when i > 0 ->  
                           (*TODO*)
@@ -2840,7 +2840,7 @@
     let rec do_analyse_funcall_ var_types ftyp args_types allTypeMap = 
       match ftyp with
         | MOFun_type(args_required_type, ret) ->
-          (match (Pervasives.compare (List.length args_types) (List.length args_required_type) )
+          (match (Stdlib.compare (List.length args_types) (List.length args_required_type) )
               with 
                 | i when i > 0 ->  
                   (*This is possible that there are more args than there
