--- _oasis.orig	2020-03-29 00:19:15.000000000 +0800
+++ _oasis	2025-10-18 23:31:32.000000000 +0800
@@ -8,7 +8,7 @@
 Homepage: http://gufo.pvittet.com
 
 AlphaFeatures: ocamlbuild_more_args
-XOCamlbuildExtraArgs: "-tag debug -cflag -g -ocamlc 'ocamlc -g' -ocamlopt 'ocamlopt -g' -use-menhir -menhir 'menhir --dump --explain' -tag thread -use-ocamlfind -tag annot"
+XOCamlbuildExtraArgs: "-tag debug -cflag -g -ocamlc 'ocamlc -g' -use-menhir -menhir 'menhir --dump --explain' -tag thread -use-ocamlfind -tag annot"
 
 OASISFormat: 0.4
 BuildTools: ocamlbuild
@@ -17,7 +17,7 @@
 Executable "gufo"
   Path: src
   MainIs: guforun.ml
-  CompiledObject: native
+  CompiledObject: byte
   BuildTools: menhir
   BuildDepends: 
     camomile,
