--- a/config.mak.uname	2026-04-22 00:12:22.000000000 +0800
+++ b/config.mak.uname	2026-04-22 00:53:27.000000000 +0800
@@ -148,20 +148,9 @@
 	FREAD_READS_DIRECTORIES = UnfortunatelyYes
 	HAVE_NS_GET_EXECUTABLE_PATH = YesPlease
 	CSPRNG_METHOD = arc4random
-	USE_ENHANCED_BASIC_REGULAR_EXPRESSIONS = YesPlease
 	HAVE_PLATFORM_PROCINFO = YesPlease
 	COMPAT_OBJS += compat/darwin/procinfo.o
 
-        ifeq ($(uname_M),arm64)
-		HOMEBREW_PREFIX = /opt/homebrew
-        else
-		HOMEBREW_PREFIX = /usr/local
-        endif
-        ifeq ($(shell test "$(DARWIN_MAJOR_VERSION)" -ge 24 && echo 1),1)
-		USE_HOMEBREW_LIBICONV = UnfortunatelyYes
-		NEEDS_GOOD_LIBICONV = UnfortunatelyYes
-        endif
-
 	# The builtin FSMonitor on MacOS builds upon Simple-IPC.  Both require
 	# Unix domain sockets and PThreads.
         ifndef NO_PTHREADS
