--- a/generate-configlist.sh.orig	2025-05-31 18:16:44.000000000 +0800
+++ b/generate-configlist.sh	2025-05-31 18:19:04.000000000 +0800
@@ -13,18 +13,10 @@
 	cat <<EOF
 static const char *config_name_list[] = {
 EOF
-	sed -E '
-/^`?[a-zA-Z].*\..*`?::$/ {
-	/deprecated/d;
-	s/::$//;
-	s/`//g;
-	s/^.*$/	"&",/;
-	s/,  */",\n	"/g;
-	p;};
-d' \
-	    "$SOURCE_DIR"/Documentation/*config.adoc \
-	    "$SOURCE_DIR"/Documentation/config/*.adoc|
-	sort
+	grep -h '^[a-zA-Z].*\..*::$' "$SOURCE_DIR"/Documentation/*config.adoc "$SOURCE_DIR"/Documentation/config/*.adoc |
+	sed '/deprecated/d; s/::$//; s/,  */\n/g' |
+	sort |
+	sed 's/^.*$/	"&",/'
 	cat <<EOF
 	NULL,
 };
