From 97164ed2120ffa965d711295d41c927743e97d0c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 5 May 2023 23:03:58 +0800
Subject: [PATCH 3/9] mono-config: fix PPC

---
 mono/metadata/mono-config.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mono/metadata/mono-config.c b/mono/metadata/mono-config.c
index eae909e26c1..88c300220b4 100644
--- a/mono/metadata/mono-config.c
+++ b/mono/metadata/mono-config.c
@@ -61,14 +61,14 @@
 #elif defined(sparc) || defined(__sparc__)
 #define CONFIG_CPU "sparc"
 #define CONFIG_WORDSIZE "32"
-#elif defined(__ppc64__) || defined(__powerpc64__) || defined(_ARCH_64) || defined(TARGET_POWERPC)
+#elif defined(__ppc64__) || defined(__powerpc64__) || defined(_ARCH_64) || defined(TARGET_POWERPC64)
 #define CONFIG_WORDSIZE "64"
 #ifdef __mono_ppc_ilp32__ 
 #   define CONFIG_CPU "ppc64ilp32"
 #else
 #   define CONFIG_CPU "ppc64"
 #endif
-#elif defined(__ppc__) || defined(__powerpc__)
+#elif defined(__ppc__) || defined(__powerpc__) || defined(TARGET_POWERPC)
 #define CONFIG_CPU "ppc"
 #define CONFIG_WORDSIZE "32"
 #elif defined(__s390x__)
-- 
2.42.0

