From b20e2273178f7a7c9c9e536c430b21aec772f362 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 5 May 2023 23:40:00 +0800
Subject: [PATCH 5/9] sgen-archdep: fix red zone for ppc64

---
 mono/sgen/sgen-archdep.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mono/sgen/sgen-archdep.h b/mono/sgen/sgen-archdep.h
index 125065ca2bf..f2b4c659d55 100644
--- a/mono/sgen/sgen-archdep.h
+++ b/mono/sgen/sgen-archdep.h
@@ -35,6 +35,10 @@
 #define REDZONE_SIZE	128
 #endif
 
+#elif defined(TARGET_POWERPC64)
+
+#define REDZONE_SIZE	320
+
 #elif defined(TARGET_POWERPC)
 
 #define REDZONE_SIZE	224
-- 
2.42.0

