From 9a2e9d6bb539f7cae003abee8f099683063d3636 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 6 May 2023 00:25:51 +0800
Subject: [PATCH 6/9] configure.ac: add Darwin PPC

---
 configure.ac | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/configure.ac b/configure.ac
index dbc4904d9b2..785f0d37caa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -445,6 +445,14 @@ case "$host" in
 				RID="osx-x64"
 				COREARCH="x64"
 				;;
+			powerpc-*-darwin*)
+				COREARCH="ppc"
+				PPC32_LDFLAGS="-latomic"
+				LDFLAGS="$LDFLAGS $PPC32_LDFLAGS"
+				;;
+			powerpc64-*-darwin*)
+				COREARCH="ppc64"
+				;;
 			arm*-darwin*)
 				platform_ios=yes
 				has_dtrace=no
-- 
2.42.0

