From bce917dee20df184018c9b4fe35f7914925d877e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 30 Jun 2024 21:31:07 +0800
Subject: [PATCH 08/10] configure.ac: minor fixes for powerpc

---
 configure.ac | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index ea0009fb3..23cff3eab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4538,6 +4538,11 @@ case "$host" in
 			BTLS_SUPPORTED=yes
 			BTLS_PLATFORM=powerpc
 			;;
+		  darwin*)
+			BTLS_SUPPORTED=no
+			BTLS_PLATFORM=powerpc
+			with_tls=pthread
+			;;
 		esac
 		;;
 	armv7k-*-darwin*)
-- 
2.43.0

