--- a/configure.ac
+++ b/configure.ac
@@ -3804,45 +3804,74 @@
 	# HAVE_CURLPIPE_MULTIPLEX check skipped because we don't use libcurl in mono
 	# HAVE_CURL_SSLVERSION_TLSv1_012 check skipped because we don't use libcurl in mono
 
-	enable_gss=no;
-	AC_MSG_CHECKING(for GSS/GSS.h)
-	AC_TRY_COMPILE([
-			#include <GSS/GSS.h>
-	], [
-	],[
-			AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_GSSFW_HEADERS, 1, [GSS/GSS.h])
-			AC_DEFINE(HAVE_GSS_SPNEGO_MECHANISM, 1, [GSS_SPNEGO_MECHANISM])
-			enable_gss=yes
-			AC_DEFINE(ENABLE_GSS, 1, [ENABLE_GSS])
-	], [
-			AC_MSG_RESULT(no)
-	])
+	AC_ARG_ENABLE(gss, [  --enable-gss    Enable support for GSS (enabled by default)], enable_gss=$enableval, enable_gss=yes)
+	if test "x$enable_gss" = "xyes"; then
+		AC_MSG_CHECKING(for GSS/GSS.h)
+		AC_TRY_COMPILE([
+				#include <GSS/GSS.h>
+		], [
+		],[
+				AC_MSG_RESULT(yes)
+				AC_DEFINE(HAVE_GSSFW_HEADERS, 1, [GSS/GSS.h])
+				AC_DEFINE(HAVE_GSS_SPNEGO_MECHANISM, 1, [GSS_SPNEGO_MECHANISM])
+				AC_DEFINE(ENABLE_GSS, 1, [ENABLE_GSS])
+		], [
+				AC_MSG_RESULT(no)
 
-	AC_MSG_CHECKING(for gssapi/gssapi_ext.h)
-	AC_TRY_COMPILE([
-			#include <gssapi/gssapi_ext.h>
-	], [
-	],[
-			AC_MSG_RESULT(yes)
-			enable_gss=yes
-			AC_DEFINE(ENABLE_GSS, 1, [ENABLE_GSS])
-	], [
-			AC_MSG_RESULT(no)
-	])
+				AC_MSG_CHECKING(for gssapi/gssapi_ext.h)
+				AC_TRY_COMPILE([
+						#include <gssapi/gssapi_ext.h>
+				], [
+				],[
+						AC_MSG_RESULT(yes)
+						AC_DEFINE(ENABLE_GSS, 1, [ENABLE_GSS])
+				], [
+						AC_MSG_RESULT(no)
+
+						AC_MSG_CHECKING(for gssapi/gssapi.h)
+						AC_TRY_COMPILE([
+								#include <gssapi/gssapi.h>
+						], [
+						],[
+								AC_MSG_RESULT(yes)
+
+								AC_DEFINE(ENABLE_GSS, 1, [ENABLE_GSS])
+								AC_DEFINE(HAVE_HEIMDAL_HEADERS, 1, [HAVE_HEIMDAL_HEADERS])
+						], [
+								AC_MSG_RESULT(no)
+								enable_gss=no
+						])
+				])
+		])
 
-	AC_MSG_CHECKING(for GSS_SPNEGO_MECHANISM)
-	AC_TRY_COMPILE([
-			#include <gssapi/gssapi_ext.h>
-			#include <gssapi/gssapi_krb5.h>
-			gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = GSS_SPNEGO_MECHANISM};
-	], [
-	],[
-			AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_GSS_SPNEGO_MECHANISM, 1, [GSS_SPNEGO_MECHANISM])
-	], [
-			AC_MSG_RESULT(no)
-	])
+		if test "x$enable_gss" = "xyes"; then
+			AC_MSG_CHECKING(for GSS_SPNEGO_MECHANISM)
+			AC_TRY_COMPILE([
+					#if HAVE_HEIMDAL_HEADERS
+					#include <gssapi/gssapi.h>
+					#else
+					#include <gssapi/gssapi_ext.h>
+					#endif
+					#include <gssapi/gssapi_krb5.h>
+					gss_OID_set_desc gss_mech_spnego_OID_set_desc = {.count = 1, .elements = GSS_SPNEGO_MECHANISM};
+			], [
+			],[
+					AC_MSG_RESULT(yes)
+					AC_DEFINE(HAVE_GSS_SPNEGO_MECHANISM, 1, [GSS_SPNEGO_MECHANISM])
+			], [
+					AC_MSG_RESULT(no)
+			])
+
+			gssapi_lib=""
+			AC_CHECK_LIB(gssapi_krb5, gss_acquire_cred, [
+				gssapi_lib="-lgssapi_krb5"
+			], [
+				AC_CHECK_LIB(gssapi, gss_acquire_cred, [
+					gssapi_lib="-lgssapi"
+				])
+			])
+		fi
+	fi
 
 	AM_CONDITIONAL(ENABLE_GSS, test x$enable_gss = xyes)
 
@@ -6494,9 +6523,9 @@
 	MONO_NATIVE_CFLAGS=$CFLAGS
 	MONO_NATIVE_LDFLAGS=$LDFLAGS
 	if test x$enable_gss = xyes; then
-		MONO_NATIVE_LDFLAGS="$MONO_NATIVE_LDFLAGS -lgssapi_krb5"
+		MONO_NATIVE_LDFLAGS="$MONO_NATIVE_LDFLAGS ${gssapi_lib}"
 	else
-		AC_MSG_WARN([GSS is not enabled, it requires libkrb5-dev package])
+		AC_MSG_WARN([GSS is not enabled, it requires libkrb5-dev or libheimdal-dev package])
 	fi
 	mono_native=yes
 	mono_native_compat=no
