--- mono/utils/Makefile.in	2011-04-18 20:15:51.000000000 +0800
+++ mono/utils/Makefile.in	2026-01-23 15:09:36.000000000 +0800
@@ -37,6 +37,7 @@
 @X86_TRUE@am__append_1 = mach-support-x86.c
 @AMD64_TRUE@am__append_2 = mach-support-amd64.c
 @ARM_TRUE@am__append_3 = mach-support-arm.c
+@POWERPC_TRUE@am__append_4 = mach-support-ppc.c
 subdir = mono/utils
 DIST_COMMON = $(libmonoutilsinclude_HEADERS) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in ChangeLog
@@ -86,8 +87,9 @@
 @X86_TRUE@am__objects_2 = mach-support-x86.lo
 @AMD64_TRUE@am__objects_3 = mach-support-amd64.lo
 @ARM_TRUE@am__objects_4 = mach-support-arm.lo
-am__objects_5 = $(am__objects_2) $(am__objects_3) $(am__objects_4)
-am_libmonoutils_la_OBJECTS = $(am__objects_1) $(am__objects_5)
+@POWERPC_TRUE@am__objects_5 = mach-support-ppc.lo
+am__objects_6 = $(am__objects_2) $(am__objects_3) $(am__objects_4) $(am__objects_5)
+am_libmonoutils_la_OBJECTS = $(am__objects_1) $(am__objects_6)
 libmonoutils_la_OBJECTS = $(am_libmonoutils_la_OBJECTS)
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
@@ -376,7 +378,7 @@
 	mach-support.h  \
 	memcheck.h
 
-arch_sources = $(am__append_1) $(am__append_2) $(am__append_3)
+arch_sources = $(am__append_1) $(am__append_2) $(am__append_3) $(am__append_4)
 libmonoutils_la_SOURCES = $(monoutils_sources) $(arch_sources)
 libmonoutilsincludedir = $(includedir)/mono-$(API_VER)/mono/utils
 libmonoutilsinclude_HEADERS = \

--- mono/utils/mach-support.h	2011-04-18 19:23:28.000000000 +0800
+++ mono/utils/mach-support.h	2026-01-23 15:10:03.000000000 +0800
@@ -12,7 +12,7 @@
 #include <mach/thread_act.h>
 #include <mach/thread_status.h>
 
-#if defined(__i386__) || defined (__x86_64__)
+#if defined(__i386__) || defined (__x86_64__) || defined (__ppc__)
 #define MONO_MACH_ARCH_SUPPORTED 1
 #endif
 
--- /dev/null	2026-01-23 14:48:46.000000000 +0800
+++ mono/utils/mach-support-ppc.c	2026-01-23 15:12:00.000000000 +0800
@@ -0,0 +1,82 @@
+/*
+ * mach-support-ppc.c: mach support for ppc
+ *
+ * Authors:
+ *   Geoff Norton (gnorton@novell.com)
+ *   Sergey Fedorov (barracuda@macos-powerpc.org)
+ *
+ * (C) 2026 Novell, Inc.
+ */
+
+#include <config.h>
+
+#if defined(__MACH__)
+#include <stdint.h>
+#include <glib.h>
+#include <pthread.h>
+#include "utils/mono-sigcontext.h"
+#include "mach-support.h"
+
+void *
+mono_mach_arch_get_ip (thread_state_t state)
+{
+	ppc_thread_state_t *arch_state = (ppc_thread_state_t *) state;
+
+	return (void *) arch_state->__srr0;
+}
+
+void *
+mono_mach_arch_get_sp (thread_state_t state)
+{
+	ppc_thread_state_t *arch_state = (ppc_thread_state_t *) state;
+
+	return (void *) arch_state->__r1;
+}
+
+int
+mono_mach_arch_get_mcontext_size ()
+{
+	return sizeof (struct __darwin_mcontext);
+}
+
+void
+mono_mach_arch_thread_state_to_mcontext (thread_state_t state, mcontext_t context)
+{
+	ppc_thread_state_t *arch_state = (ppc_thread_state_t *) state;
+	struct __darwin_mcontext *ctx = (struct __darwin_mcontext *) context;
+
+	ctx->__ss = *arch_state;
+}
+
+int
+mono_mach_arch_get_thread_state_size ()
+{
+	return sizeof (ppc_thread_state_t);
+}
+
+kern_return_t
+mono_mach_arch_get_thread_state (thread_port_t thread, thread_state_t state, mach_msg_type_number_t *count)
+{
+	ppc_thread_state_t *arch_state = (ppc_thread_state_t *) state;
+	kern_return_t ret;
+
+	*count = PPC_THREAD_STATE_COUNT;
+
+	ret = thread_get_state (thread, PPC_THREAD_STATE, (thread_state_t) arch_state, count);
+
+	return ret;
+}
+
+void *
+mono_mach_arch_get_tls_value_from_thread (pthread_t thread, guint32 key)
+{
+	/* OSX stores TLS values in a hidden array inside the pthread_t structure
+	 * They are keyed off a giant array offset 0x48 into the pointer. This value
+	 * is baked into their pthread_getspecific implementation
+	 */
+	intptr_t *p = (intptr_t *) thread;
+	intptr_t **tsd = (intptr_t **) ((char*)p + 0x48);
+
+	return (void *) tsd [key];
+}
+#endif
