From 3d695819cdf2e09e818a664b5a0ef2f49e1c150c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 18 Jan 2026 05:28:07 +0800
Subject: [PATCH 1/2] platform.cpp: add missing errno.h

---
 source/system/platform.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git source/system/platform.cpp source/system/platform.cpp
index 83e803a0..aafbc02f 100644
--- source/system/platform.cpp
+++ source/system/platform.cpp
@@ -25,6 +25,7 @@
 	#include <pthread.h>
 	#include <sys/stat.h>
 	#include <spawn.h>
+	#include <errno.h>
 	#include <cstring>
 	#include <memory>
 
-- 
2.52.0

