From cf750cd39c14e63499b8b7d056a80d182ee74657 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 18 Jan 2026 05:42:22 +0800
Subject: [PATCH 2/2] platform.cpp: use correct environ on macOS

---
 source/system/platform.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git source/system/platform.cpp source/system/platform.cpp
index aafbc02f..51977390 100644
--- source/system/platform.cpp
+++ source/system/platform.cpp
@@ -28,13 +28,20 @@
 	#include <errno.h>
 	#include <cstring>
 	#include <memory>
+	#ifdef __APPLE__
+		#include <crt_externs.h>
+	#endif
 
 static void posix_open_url(const char *url_utf8)
 {
     using nana::system::split_string_type;
     using nana::system::split_string;
 
+#ifdef __APPLE__
+    char **environ = *_NSGetEnviron();
+#else
     extern char **environ;
+#endif
     struct stat exists;
     const split_string_type path = getenv("PATH");
 
-- 
2.52.0

