--- source/detail/text_reshaping.hpp	2020-05-17 06:08:50.000000000 +0800
+++ source/detail/text_reshaping.hpp	2026-01-18 08:42:24.000000000 +0800
@@ -213,7 +213,7 @@
 		                    if (shift_harakat_position)
 		                    {
 		                    	auto & ph = positions_harakat[position];
-		                    	ph.insert(ph.cbegin(), letter);
+		                    	ph.insert(ph.begin(), letter);
 		                    }
 		                    else
 		                        positions_harakat[position] += letter;
