--- include/the_Foundation/atomic.h	2025-11-27 12:59:44.000000000 +0800
+++ include/the_Foundation/atomic.h	2026-02-10 14:02:05.000000000 +0800
@@ -27,7 +27,7 @@
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</small>
 */
 
-#if __STDC_VERSION__ >= 201100L
+#if (__STDC_VERSION__ >= 201100L) && !defined(__OBJC__)
 #  include <stdatomic.h>
 typedef atomic_int iAtomicInt;
 #  define value_Atomic(a)               atomic_load(a)
