--- src/window.h
+++ src/window.h	2026-04-26 19:05:59.354459675 +0000
@@ -7,10 +7,13 @@
 
 #include <GLFW/glfw3.h>
 
-#include <GrBackendSurface.h>
-#include <GrContext.h>
-#include <SkCanvas.h>
-#include <SkSurface.h>
+#include <include/gpu/ganesh/GrBackendSurface.h>
+#include <include/gpu/ganesh/GrDirectContext.h>
+#include <include/gpu/ganesh/gl/GrGLBackendSurface.h>
+#include <include/gpu/ganesh/gl/GrGLTypes.h>
+#include <include/core/SkCanvas.h>
+#include <include/core/SkSurface.h>
+#include <include/core/SkColorSpace.h>
 
 #include <functional>
 
@@ -69,8 +72,7 @@
 
   std::unique_ptr<GLManager> m_gl{new GLManager};
   GrGLFramebufferInfo m_info;
-  sk_sp<const GrGLInterface> m_interface;
-  sk_sp<GrContext> m_context;
+  sk_sp<GrDirectContext> m_context;
   std::unique_ptr<GrBackendRenderTarget> m_target;
   sk_sp<SkSurface> m_surface;
 };
