--- src/uterm.h.orig	2026-04-27 01:49:24.545631687 +0000
+++ src/uterm.h	2026-04-27 02:00:56.055544100 +0000
@@ -57,9 +57,11 @@
   void HandleSelection(Selection state, double mx, double my);
   void HandleScroll(ScrollDirection direction, uint distance);
   void HandleTitle(const string &title);
+  void HandleDrop(const std::vector<string> &paths);
 
   std::mutex m_current_reader_lock;
   ReaderThread *m_current_reader{nullptr};
+  Pty *m_pty{nullptr};
 
   Config m_config;
   Terminal m_term;
