From b01e292944c49184aaccddcf61252001107bd1eb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 22:50:25 +0800
Subject: [PATCH 1/2] ksysguard: do not error out if not supported

---
 ksysguard/ksysguardd/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ksysguard/ksysguardd/CMakeLists.txt b/ksysguard/ksysguardd/CMakeLists.txt
index 79d06ddd..b416fc64 100644
--- a/ksysguard/ksysguardd/CMakeLists.txt
+++ b/ksysguard/ksysguardd/CMakeLists.txt
@@ -25,8 +25,6 @@ elseif(CMAKE_SYSTEM_NAME MATCHES "(NetBSD|OpenBSD)")
     add_definitions(-DOSTYPE_${CMAKE_SYSTEM_NAME})
     include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${CMAKE_SYSTEM_NAME})
     include(${CMAKE_SYSTEM_NAME}/${CMAKE_SYSTEM_NAME}.cmake)
-else()
-    message(FATAL_ERROR "ksysguardd does not support: ${CMAKE_SYSTEM_NAME}")
 endif()
 
 ########### next target ###############
-- 
2.49.0

