--- a/ksysguard/ksysguardd/CMakeLists.txt	2026-03-20 19:27:17
+++ b/ksysguard/ksysguardd/CMakeLists.txt	2026-03-22 18:07:08
@@ -25,6 +25,10 @@
     add_definitions(-DOSTYPE_${CMAKE_SYSTEM_NAME})
     include_directories(${CMAKE_CURRENT_SOURCE_DIR}/${CMAKE_SYSTEM_NAME})
     include(${CMAKE_SYSTEM_NAME}/${CMAKE_SYSTEM_NAME}.cmake)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    add_definitions(-DOSTYPE_Darwin)
+    include_directories(${CMAKE_CURRENT_SOURCE_DIR}/Darwin)
+    include(Darwin/Darwin.cmake)
 else()
     message(FATAL_ERROR "ksysguardd does not support: ${CMAKE_SYSTEM_NAME}")
 endif()
