From dcb923ae96966b11cc0cb1eb5226069b56a66038 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 24 Jun 2025 22:52:10 +0800
Subject: [PATCH 2/2] Fix SharedMimeInfo search

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e6c528c..1823aeec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ include(FeatureSummary)
 find_package(KDELibs4 4.23.0 REQUIRED)
 
 # kdelibs requires 0.91
-find_package(SharedMimeInfo 0.91)
+find_package(SharedMimeInfo)
 set_package_properties(SharedMimeInfo PROPERTIES
     DESCRIPTION "Shared Mime Info"
     URL "http://freedesktop.org/wiki/Software/shared-mime-info"
-- 
2.49.0

