--- ext/nokogiri/extconf.rb	2021-09-15 22:20:13.000000000 +0300
+++ ext/nokogiri/extconf.rb	2025-10-31 07:17:19.000000000 +0800
@@ -15,7 +15,7 @@
 REQUIRED_LIBXML_VERSION = "2.9.2"
 RECOMMENDED_LIBXML_VERSION = "2.12.0"
 
-REQUIRED_MINI_PORTILE_VERSION = "~> 2.8.2" # keep this version in sync with the one in the gemspec
+REQUIRED_MINI_PORTILE_VERSION = "~> 2.8.9" # keep this version in sync with the one in the gemspec
 REQUIRED_PKG_CONFIG_VERSION = "~> 1.1"
 
 # Keep track of what versions of what libraries we build against
@@ -699,10 +699,9 @@
 # these tend to be noisy, but on occasion useful during development
 # append_cflags(["-Wcast-qual", "-Wwrite-strings"])
 
-# Add SDK-specific include path for macOS and brew versions before v2.2.12 (2020-04-08) [#1851, #1801]
-macos_mojave_sdk_include_path = "/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libxml2"
-if config_system_libraries? && darwin? && Dir.exist?(macos_mojave_sdk_include_path) && !nix?
-  append_cppflags("-I#{macos_mojave_sdk_include_path}")
+libxml_include_path = "@@PREFIX@@/include/libxml2"
+if config_system_libraries? && darwin? && Dir.exist?(libxml_include_path) && !nix?
+  append_cppflags("-I#{libxml_include_path}")
 end
 
 # Work around a character escaping bug in MSYS by passing an arbitrary double-quoted parameter to gcc.
