diff --git a/include/GL/internal/dri_interface.h b/include/GL/internal/dri_interface.h
index 87c046b254524eccb856ce29a325a0c8fcd028f2..a4d576ecda2f346ae0b3d9a82901b1d61809ab97 100644
--- include/GL/internal/dri_interface.h
+++ include/GL/internal/dri_interface.h
@@ -442,7 +442,6 @@ struct mesa_glinterop_device_info;
 struct mesa_glinterop_export_in;
 struct mesa_glinterop_export_out;
 struct mesa_glinterop_flush_out;
-typedef struct __GLsync *GLsync;
 
 struct __DRI2interopExtensionRec {
    __DRIextension base;
