--- meson.build.orig	2026-08-17 00:27:08.000000000 -0600
+++ meson.build	2026-08-17 00:28:18.000000000 -0600
@@ -1456,15 +1456,8 @@
   endif
 endif
 
-# Detect __builtin_ia32_clflushopt support
-if cc.has_function('__builtin_ia32_clflushopt', args : '-mclflushopt')
-  pre_args += '-DHAVE___BUILTIN_IA32_CLFLUSHOPT'
-  clflushopt_args = ['-mclflushopt']
-  with_clflushopt = true
-else
-  clflushopt_args = []
-  with_clflushopt = false
-endif
+clflushopt_args = []
+with_clflushopt = false
 
 # Check for GCC style atomics
 dep_atomic = null_dep
