--- src/glx/create_context.c.orig	2026-02-25 23:54:24.000000000 +0800
+++ src/glx/create_context.c	2026-03-07 20:50:37.000000000 +0800
@@ -94,6 +94,7 @@
 
    assert(screen == psc->scr);
 
+#if defined(GLX_DIRECT_RENDERING) && (!defined(GLX_USE_APPLEGL) || defined(GLX_USE_APPLE))
    if (orig_attrib_list != NULL) {
       attrib_list = malloc(sizeof(int) * num_attribs * 2);
 
@@ -114,6 +115,9 @@
          }
       }
    }
+#else
+   attrib_list = (int *)orig_attrib_list;
+#endif
 
    /* Some application may request an indirect context but we may want to force a direct
     * one because Xorg only allows indirect contexts if they were enabled.
