--- meson.build	2025-03-20 21:32:27.000000000 +0800
+++ meson.build	2025-03-30 01:08:51.000000000 +0800
@@ -937,8 +937,7 @@
   pre_args += '-DALLOW_KCMP'
 endif
 
-# On Windows, a venv has no versioned aliased to 'python'.
-prog_python = find_program('python3', 'python', version : '>= 3.8')
+prog_python = find_program('@@python3@@', 'python', version : '>= 3.8')
 
 has_mako = run_command(
   prog_python, '-c',
