--- src/util/os_misc.c
+++ src/util/os_misc.c	2026-08-16 11:34:56.000000000 +0800
@@ -520,7 +521,7 @@
 #endif
 }
 
-#if DETECT_OS_APPLE
+#if DETECT_OS_APPLE && !defined(__POWERPC__)
 
 static bool jit_allowed;
 
@@ -552,7 +553,9 @@
 bool
 os_jit_allowed(void)
 {
-#if DETECT_OS_APPLE
+#ifdef __POWERPC__ /* JIT is broken in LLVM for PowerPC */
+   return false;
+#elif DETECT_OS_APPLE
    static util_once_flag once = UTIL_ONCE_FLAG_INIT;
 
    util_call_once(&once, os_probe_jit_allowed);
