--- src/modules/macosx/module-coreaudio-device.c.orig
+++ src/modules/macosx/module-coreaudio-device.c
@@ -474,7 +474,7 @@ static int ca_device_create_sink(pa_module *m, AudioBuffer *buf, int channel_idx
     }

     ca_sink->ss.rate = u->stream_description.mSampleRate;
-    ca_sink->ss.format = PA_SAMPLE_FLOAT32LE;
+    ca_sink->ss.format = PA_SAMPLE_FLOAT32BE;

     pa_sink_new_data_init(&new_data);
     new_data.card = u->card;
@@ -613,7 +613,7 @@ static int ca_device_create_source(pa_module *m, AudioBuffer *buf, int channel_i
     }

     ca_source->ss.rate = u->stream_description.mSampleRate;
-    ca_source->ss.format = PA_SAMPLE_FLOAT32LE;
+    ca_source->ss.format = PA_SAMPLE_FLOAT32BE;

     pa_source_new_data_init(&new_data);
     new_data.card = u->card;
