--- ./src/pulsecore/core-rtclock.c.orig	2026-09-22 01:50:53.000000000 -0600
+++ ./src/pulsecore/core-rtclock.c	2026-09-22 01:51:12.000000000 -0600
@@ -35,7 +35,12 @@
 #include <sys/prctl.h>
 #endif
 
+// Need to define strings.h for tiger before including CoreServices.h for bzero
 #ifdef OS_IS_DARWIN
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#include <strings.h>
+#endif
 #include <CoreServices/CoreServices.h>
 #include <mach/mach.h>
 #include <mach/mach_time.h>
