--- meson.build	2026-03-17 04:22:41.000000000 +0800
+++ meson.build	2026-05-05 02:23:38.000000000 +0800
@@ -103,6 +103,11 @@
   introspection_dep = []
 endif
 
+legacysupport_dep = declare_dependency(
+  include_directories: include_directories('@PREFIX@/include/LegacySupport'),
+  link_args: ['-lMacportsLegacySupport'],
+)
+
 xmllint_prg = find_program('xmllint', required: false)
 
 glib_version_arr = glib_req_version.split('.')

--- gtksourceview/meson.build	2026-03-17 04:22:41.000000000 +0800
+++ gtksourceview/meson.build	2026-05-05 02:24:05.000000000 +0800
@@ -160,6 +160,7 @@
   pangoft2_dep,
   fontconfig_dep,
   pcre2_dep,
+  legacysupport_dep,
 ]
 
 # We have no way to know from the gtk4_dep whether Vulkan support
