--- Makefile	2025-11-25 21:38:04.000000000 +0800
+++ Makefile	2026-01-20 04:34:06.000000000 +0800
@@ -12,9 +12,9 @@
 
 -include $(shell ocamlc -where)/Makefile.config
 
-PREFIX=/usr
+PREFIX=@PREFIX@
 BINDIR=$(DESTDIR)$(PREFIX)/bin
-LIBDIR=$(DESTDIR)$(PREFIX)/lib/ocaml/cmdliner
+LIBDIR=$(DESTDIR)$(PREFIX)/lib/ocaml/site-lib/cmdliner
 SHAREDIR=$(DESTDIR)$(PREFIX)/share
 DOCDIR=$(SHAREDIR)/doc/cmdliner
 MANDIR=$(SHAREDIR)/man
