--- a/meson.build	2025-12-24 00:48:02.000000000 +0800
+++ b/meson.build	2025-12-24 00:49:59.000000000 +0800
@@ -689,7 +689,7 @@
 endif
 
 if conf_data.get('CONFIG_QUICKJS')
-    quickjsdeps = compiler.find_library('quickjs', static: st, required: false)
+    quickjsdeps = dependency('quickjs', static: st, required: false)
     if not quickjsdeps.found()
         quickjsdeps = compiler.find_library('quickjs/libquickjs', static: true, required: not conf_data.get('CONFIG_OS_DOS'))
     endif
