--- a/src/main/select.c	2025-12-06 16:12:55.000000000 +0800
+++ b/src/main/select.c	2026-01-02 23:52:48.000000000 +0800
@@ -22,6 +22,13 @@
 #ifdef HAVE_INTTYPES_H
 #include <inttypes.h> /* OMG */
 #endif
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+/* poll in 10.5 is available but broken. */
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#undef HAVE_POLL
+#endif
+#endif
 
 #if defined(HAVE_POLL_H) && defined(HAVE_POLL) && !defined(INTERIX) && !defined(__HOS_AIX__) && !defined(CONFIG_OS_DOS)
 #define USE_POLL
