From 3685149d9a7f0a56dd0d52dbdbc0cc04b7d1ee7c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 7 Apr 2025 14:10:19 +0800
Subject: [PATCH] common.h: add a declaration of hosts_ctl for macOS

---
 common.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git common.h common.h
index 3503a95..7243104 100644
--- common.h
+++ common.h
@@ -34,6 +34,10 @@
 #include <sys/capability.h>
 #endif
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 #include "config.h"
 #include "version.h"
 
@@ -182,6 +186,10 @@ extern struct sslhcfg_item cfg;
 extern struct addrinfo *addr_listen;
 extern const char* server_type;
 
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED < 1080)
+extern int hosts_ctl();
+#endif
+
 /* sslh-fork.c */
 void start_shoveler(int);
 
-- 
2.48.0

