--- src/sqlcipher.c	2025-08-01 23:43:28.000000000 +0800
+++ src/sqlcipher.c	2025-09-23 04:51:01.000000000 +0800
@@ -402,7 +402,7 @@
   }
   #endif
 #elif defined(__APPLE__)
-  #if !defined(__has_feature) || !__has_feature(address_sanitizer)
+  #if !defined(__has_feature)
   static void (*const sqlcipher_fini_func)(void) __attribute__((used, section("__DATA,__mod_term_func"))) = sqlcipher_fini;
   #else
   static void sqlcipher_cleanup_destructor(void) __attribute__((destructor));
