--- libavcodec/audiotoolboxdec.c.orig
+++ libavcodec/audiotoolboxdec.c
@@ -36,6 +36,11 @@
 #define kAudioFormatEnhancedAC3 'ec-3'
 #endif

+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#define kAudioFormatMicrosoftGSM 0x6D730031
+#define kAudioFormatiLBC 'ilbc'
+#endif
+
 typedef struct ATDecodeContext {
     AVClass *av_class;

--- libavcodec/audiotoolboxenc.c.orig
+++ libavcodec/audiotoolboxenc.c
@@ -35,6 +35,10 @@
 #include "libavutil/opt.h"
 #include "libavutil/log.h"

+#if __MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#define kAudioFormatiLBC 'ilbc'
+#endif
+
 typedef struct ATDecodeContext {
     AVClass *av_class;
     int mode;
