From ebeed833dafc34465886a2b03b18c1560a6ed4cb Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 5 May 2025 06:54:04 +0800
Subject: [PATCH] Old C++ ABI

---
 SConstruct | 1 +
 1 file changed, 1 insertion(+)

diff --git SConstruct SConstruct
index e3bb68de..7087f586 100644
--- SConstruct
+++ SConstruct
@@ -127,6 +127,7 @@ class FreelanEnvironment(Environment):
         self.Append(LIBPATH=['@OPENSSL@/lib'])
 
         if sys.platform.startswith('darwin'):
+            self.Append(CXXFLAGS=['-D_GLIBCXX_USE_CXX11_ABI=0'])
             self.Append(CXXFLAGS=['-DBOOST_ASIO_DISABLE_KQUEUE'])
 
         if sys.platform.startswith('freebsd'):
