From 92813eb0a7a8f496da8cae050e677152692b83bc Mon Sep 17 00:00:00 2001
From: zsugabubus <zsugabubus@users.noreply.github.com>
Date: Thu, 11 Sep 2025 20:28:35 +0200
Subject: [PATCH 2/6] Replace `ncursesw` with `ncurses`

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git meson.build meson.build
index 3278c9b..5b6e125 100755
--- meson.build
+++ meson.build
@@ -27,7 +27,7 @@ executable('aria2t',
 		command: ['keys.gen', '@INPUT@'],
 		capture: true,
 	),
-	dependencies: dependency('ncursesw'),
+	dependencies: dependency('ncurses'),
 	install: true,
 )
 
-- 
2.51.0

