From 1b756b37b8ce58b9b45793bf63f133044d203526 Mon Sep 17 00:00:00 2001
From: zsugabubus <zsugabubus@users.noreply.github.com>
Date: Thu, 11 Sep 2025 20:29:17 +0200
Subject: [PATCH 3/6] Replace `sed` with `gsed`

---
 keys.gen    | 6 +++---
 manpage.gen | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git keys.gen keys.gen
index d2bbab1..78f6763 100755
--- keys.gen
+++ keys.gen
@@ -8,13 +8,13 @@ echo '#pragma GCC diagnostic push'
 echo '#pragma GCC diagnostic ignored "-Wmissing-field-initializers"'
 echo '#pragma GCC diagnostic ignored "-Wunused-parameter"'
 
-sed -n '/K_[a-zA-Z_.-]*_parse/{s/.*\(K_[a-zA-Z_.-]*\)_parse.*/\1/p}' "$source" |
+gsed -n '/K_[a-zA-Z_.-]*_parse/{s/.*\(K_[a-zA-Z_.-]*\)_parse.*/\1/p}' "$source" |
 sort |
 uniq |
 while read -r name
 do
 	table=$(
-		sed -n '/_\(parse\|none\)/! {
+		gsed -n '/_\(parse\|none\)/! {
 			/'$name'_[a-zA-Z_.-]*/ {
 				s:.*\('$name'_\([a-zA-Z_.-]*\)\).*:\2,\t\1:
 				: subst_dashes
@@ -36,7 +36,7 @@ do
 %{
 enum ${name}_e_ {
 	${name}_none
-$(printf '%s' "$table" | sed 's/.*\t\(.*\)/\t, \1/')
+$(printf '%s' "$table" | gsed 's/.*\t\(.*\)/\t, \1/')
 };
 %}
 %struct-type
diff --git manpage.gen manpage.gen
index 452422e..ee1d451 100755
--- manpage.gen
+++ manpage.gen
@@ -4,10 +4,10 @@ while IFS= read -r line; do
 	(INCLUDE\(*\))
 		name=${line#INCLUDE(}
 		name=${name%)}
-		sed -n '/\/\*MAN('"$name"')/{n;:o;N;s/\*\/$//;To;s/[\t]* \*  *//g;p}' "$@"
+		gsed -n '/\/\*MAN('"$name"')/{n;:o;N;s/\*\/$//;To;s/[\t]* \*  *//g;p}' "$@"
 		;;
 	(SHELL\(*\))
-		git shortlog -sn | cut -f2|sed -e 'i.UR' -e 'a.UE'
+		git shortlog -sn | cut -f2|gsed -e 'i.UR' -e 'a.UE'
 		cmd=${line#SHELL(}
 		cmd=${cmd%)}
 		eval "$cmd"
-- 
2.51.0

