From 9c0edb122b6ae1cc4827db25ddf8a55e37f54782 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 13 Sep 2025 01:35:40 +0800
Subject: [PATCH 4/6] Add macOS-specific headers

---
 aria2t.c  | 5 +++++
 jeezson.c | 4 ++++
 2 files changed, 9 insertions(+)

diff --git aria2t.c aria2t.c
index 54f7ab0..efeff84 100644
--- aria2t.c
+++ aria2t.c
@@ -22,6 +22,11 @@
 #include <time.h>
 #include <unistd.h>
 
+#ifdef __APPLE__
+# include <signal.h>
+# include <xlocale.h>
+#endif
+
 #include "program.h"
 #include "websocket.h"
 #include "b64.h"
diff --git jeezson.c jeezson.c
index 0bb93e8..db72c74 100644
--- jeezson.c
+++ jeezson.c
@@ -25,6 +25,10 @@
 #include <stdlib.h>
 #include <string.h>
 
+#ifdef __APPLE__
+# include <xlocale.h>
+#endif
+
 #include "jeezson.h"
 
 #if __STDC_VERSION__ < 201112L
