From b2bbc390bc41c04d34afe7d6446cfb2585c78af7 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 13 Sep 2025 03:25:42 +0800
Subject: [PATCH 5/6] Add a fallback for ppoll for macOS

---
 aria2t.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git aria2t.c aria2t.c
index efeff84..2d80eb4 100644
--- aria2t.c
+++ aria2t.c
@@ -5167,7 +5167,15 @@ main(int argc, char *argv[])
 	sigemptyset(&ss);
 
 	for (;;) {
+	#ifdef __APPLE__
+		int timeout_ms = -1;
+		if (on_timeout) {
+			timeout_ms = interval.tv_sec * 1000 + interval.tv_nsec / 1000000;
+		}
+		switch (poll(pfds, ARRAY_SIZE(pfds), timeout_ms)) {
+	#else
 		switch (ppoll(pfds, ARRAY_SIZE(pfds), on_timeout ? &interval : NULL, &ss)) {
+	#endif
 		case 0:
 			on_timeout();
 			break;
-- 
2.51.0

