From 25082e80f171ddd38e5fd940c7cc85fcba97d16c Mon Sep 17 00:00:00 2001
From: Anthony Hu <anthony@wolfssl.com>
Date: Fri, 9 May 2025 17:44:38 -0400
Subject: [PATCH] Rename variable index to idx to avoid conflicting
 declaration.

---
 wolfcrypt/benchmark/benchmark.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/wolfcrypt/benchmark/benchmark.c b/wolfcrypt/benchmark/benchmark.c
index 7c7f4702ab..0380b4f84b 100644
--- a/wolfcrypt/benchmark/benchmark.c
+++ b/wolfcrypt/benchmark/benchmark.c
@@ -8630,7 +8630,7 @@ void bench_srtpkdf(void)
     const byte *key = bench_key_buf;
     const byte salt[14] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08,
                            0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e };
-    const byte index[6] = { 0x55, 0xAA, 0x55, 0xAA, 0x55, 0xAA };
+    const byte idx[6] = { 0x55, 0xAA, 0x55, 0xAA, 0x55, 0xAA };
     int kdrIdx = 0;
     int i;
     DECLARE_MULTI_VALUE_STATS_VARS()
@@ -8640,7 +8640,7 @@ void bench_srtpkdf(void)
     do {
         for (i = 0; i < numBlocks * 1000; i++) {
             ret = wc_SRTP_KDF(key, AES_128_KEY_SIZE, salt, sizeof(salt),
-                kdrIdx, index, keyE, AES_128_KEY_SIZE, keyA, sizeof(keyA),
+                kdrIdx, idx, keyE, AES_128_KEY_SIZE, keyA, sizeof(keyA),
                 keyS, sizeof(keyS));
             RECORD_MULTI_VALUE_STATS();
         }
@@ -8663,7 +8663,7 @@ void bench_srtpkdf(void)
     do {
         for (i = 0; i < numBlocks * 1000; i++) {
             ret = wc_SRTP_KDF(key, AES_256_KEY_SIZE, salt, sizeof(salt),
-                kdrIdx, index, keyE, AES_256_KEY_SIZE, keyA, sizeof(keyA),
+                kdrIdx, idx, keyE, AES_256_KEY_SIZE, keyA, sizeof(keyA),
                 keyS, sizeof(keyS));
             RECORD_MULTI_VALUE_STATS();
         }
@@ -8686,7 +8686,7 @@ void bench_srtpkdf(void)
     do {
         for (i = 0; i < numBlocks * 1000; i++) {
             ret = wc_SRTCP_KDF(key, AES_128_KEY_SIZE, salt, sizeof(salt),
-                kdrIdx, index, keyE, AES_128_KEY_SIZE, keyA, sizeof(keyA),
+                kdrIdx, idx, keyE, AES_128_KEY_SIZE, keyA, sizeof(keyA),
                 keyS, sizeof(keyS));
             RECORD_MULTI_VALUE_STATS();
         }
@@ -8709,7 +8709,7 @@ void bench_srtpkdf(void)
     do {
         for (i = 0; i < numBlocks * 1000; i++) {
             ret = wc_SRTCP_KDF(key, AES_256_KEY_SIZE, salt, sizeof(salt),
-                kdrIdx, index, keyE, AES_256_KEY_SIZE, keyA, sizeof(keyA),
+                kdrIdx, idx, keyE, AES_256_KEY_SIZE, keyA, sizeof(keyA),
                 keyS, sizeof(keyS));
             RECORD_MULTI_VALUE_STATS();
         }
