--- Makefile.orig	2026-06-06 00:15:05.000000000 +0800
+++ Makefile	2026-06-09 19:58:32.000000000 +0800
@@ -1,5 +1,6 @@
+PREFIX ?= /opt/local
 EXEC = wayst
-INSTALL_DIR = /usr/local/bin
+INSTALL_DIR = $(PREFIX)/bin
 
 ARGS =
 
@@ -9,13 +10,14 @@
 TGT_DIR = .
 
 ifeq ($(shell uname -s),FreeBSD)
-	INCLUDES = -I/usr/local/include/freetype2/
-	INCLUDES += -I/usr/local/include
-	LDLIBS = -lfreetype -lfontconfig -lutil -L/usr/local/lib -lm
+	INCLUDES = -I$(PREFIX)/include/freetype2/
+	INCLUDES += -I$(PREFIX)/include
+	LDLIBS += -lfreetype -lfontconfig -lutil -L/usr/local/lib -lm
 else
 	CC?= cc
-	INCLUDES = -I/usr/include/freetype2/
-	LDLIBS = -lfreetype -lfontconfig -lutil -L/usr/lib -lm
+	INCLUDES = -I$(PREFIX)/include/freetype2/
+	INCLUDES += -I$(PREFIX)/include
+	LDLIBS += -lfreetype -lfontconfig -lutil -lm
 endif
 
 ifeq ($(mode),sanitized)
@@ -102,7 +103,7 @@
 	$(RM) -f $(EXEC) $(OBJ) $(OBJ:.o=.d)
 
 install:
-	cp $(EXEC) $(INSTALL_DIR)/
+	cp $(EXEC) $(DESTDIR)$(INSTALL_DIR)/
 
 uninstall:
 	$(RM) $(INSTALL_DIR)/$(EXEC)
