--- a/viu_media/assets/scripts/fzf/dynamic_preview.py	2026-01-01 02:43:43.000000000 +0800
+++ b/viu_media/assets/scripts/fzf/dynamic_preview.py	2026-03-29 13:02:44.000000000 +0800
@@ -234,7 +234,7 @@
     """Render using Chafa in auto mode."""
     if which("chafa"):
         subprocess.run(
-            ["chafa", "-s", f"{width}x{height}", file_path],
+            ["chafa", "-f", "symbols", "--symbols", "block", "-s", f"{width}x{height}", file_path],
             stdout=sys.stdout,
             stderr=sys.stderr,
         )

--- a/viu_media/assets/scripts/fzf/preview.py	2026-01-01 02:43:43.000000000 +0800
+++ b/viu_media/assets/scripts/fzf/preview.py	2026-03-29 13:01:37.000000000 +0800
@@ -174,7 +174,7 @@
     """
     if which("chafa"):
         subprocess.run(
-            ["chafa", "-s", f"{width}x{height}", file_path],
+            ["chafa", "-f", "symbols", "--symbols", "block", "-s", f"{width}x{height}", file_path],
             stdout=sys.stdout,
             stderr=sys.stderr,
         )
