--- lib/flow/VaporFlow.cpp.orig	2013-08-24 03:03:49.000000000 +0800
+++ lib/flow/VaporFlow.cpp	2026-06-06 10:35:39.000000000 +0800
@@ -1169,8 +1169,7 @@
 //scaleField indicates whether or not the field is scaled by current steady field scale factor.
 FieldData* VaporFlow::
 setupFieldData(const vector<string>& varnames, 
-			   bool useRakeBounds, int numRefinements, int timestep, bool scaleField){
-	
+			   bool useRakeBounds, int numRefinements, int timestep, bool scaleField) {
 	size_t minInt[3], maxInt[3];
 	size_t ts = (size_t) timestep;
 	const vector<double>& userExts = dataMgr->GetExtents(ts);
@@ -1203,7 +1202,7 @@
 		minInt, maxInt, &pUGrid[0],  &pVGrid[0],  &pWGrid[0]);
 	
 	if (!gotData) 
-		return false;
+		return nullptr;
 	
 	pSolution = new Solution(pUGrid,pVGrid,pWGrid, 1, periodicDim);
 	if (scaleField)
