--- make/config/Darwin.mk	2013-12-09 00:02:26.000000000 -0800
+++ make/config/Darwin.mk	2013-12-09 00:04:26.000000000 -0800
@@ -1,6 +1,6 @@
-G++-INCLUDE-DIR = /usr/include/g++
-CXX = g++ -fno-common
-CC = gcc -fno-common
+G++-INCLUDE-DIR =
+CXX = @@MP_CXX@@ -fno-common
+CC = @@MP_CC@@ -fno-common
 
 ifeq    ($(HAVE_QT_FRAMEWORK),1)
 QT_FRAMEWORK = 1
@@ -16,23 +16,15 @@
 C_DEBUG_FLAGS     += -g
 
 LDFLAGS           += -framework CoreFoundation
-LD_RELEASE_FLAGS  += 
-LD_DEBUG_FLAGS    += 
+LD_RELEASE_FLAGS  +=
+LD_DEBUG_FLAGS    +=
 
 PROFILEFLAGS = -pg -a
 
-ifeq ($(MACHTYPE),x86_64)
-SHARED_LDFLAGS += -m64 
-LDFLAGS += -m64
-CXXFLAGS += -m64
-CFLAGS += -m64
-else 
-SHARED_LDFLAGS += -m32 
-LDFLAGS += -m32
-CXXFLAGS += -m32
-CFLAGS += -m32
-endif
-
+SHARED_LDFLAGS += @@ARCHFLAG@@
+LDFLAGS += @@ARCHFLAG@@
+CXXFLAGS += @@ARCHFLAG@@
+CFLAGS += @@ARCHFLAG@@
 
 CAT = cat
 AS = as
@@ -60,7 +60,7 @@
 LIBSUFFIX = .a
 OBJSUFFIX = .o
 MV = mv
-SHARED_LDFLAGS += -dynamiclib -current_version $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE) -compatibility_version $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE) -undefined dynamic_lookup
+SHARED_LDFLAGS += -dynamiclib -install_name @@MP_PREFIX@@/lib/$(LIB_REALNAME) -current_version $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE) -compatibility_version $(VERSION_MAJOR).$(VERSION_MINOR).$(VERSION_RELEASE) -undefined dynamic_lookup
 PERL = perl
 PYTHON = python
 
