--- Makefile	2013-08-24 03:06:17.000000000 +0800
+++ Makefile	2026-06-06 09:11:00.000000000 +0800
@@ -33,7 +33,7 @@
 install-dep:: 
 	@$(ECHO) "Copying Qt plugins to $(INSTALL_PLUGINSDIR)"
 	@$(MAKE_INSTALL_PLUGINSDIR)
-	$(CP) -R $(QTDIR)/plugins/* $(INSTALL_PLUGINSDIR)
+	$(CP) -R $(QTDIR)/share/plugins/* $(INSTALL_PLUGINSDIR)
 	$(RM) $(INSTALL_PLUGINSDIR)/*/*debug*
 	@$(ECHO) "Copying Qt plugin library dependencies to $(INSTALL_LIBDIR)"
 	$(PERL) $(TOP)/buildutils/copylibdeps.pl -arch $(ARCH) $(LDLIBPATHS) $(CLD_EXCLUDE_FLAGS) $(CLD_INCLUDE_FLAGS) $(INSTALL_PLUGINSDIR)/*/* $(INSTALL_LIBDIR)
