--- make/config/Darwin.mk
+++ make/config/Darwin.mk	2026-06-06 09:53:34.000000000 +0800
@@ -6,7 +6,7 @@
 QT_FRAMEWORK = 1
 endif
 
-CXXFLAGS          += -DDARWIN -Wall -Wno-format -Wno-sign-compare  -fPIC  
+CXXFLAGS          += -DDARWIN -Wall -Wno-format -Wno-sign-compare -fPIC -std=gnu++11 -D_GLIBCXX_USE_CXX11_ABI=0
 CXXFLAGS          += -Wno-overloaded-virtual
 CXX_RELEASE_FLAGS += -O3 -DNDEBUG
 CXX_DEBUG_FLAGS   += -g
