--- apps/vaporgui/Makefile	2026-06-06 08:36:21.000000000 +0800
+++ apps/vaporgui/Makefile	2026-06-06 11:58:03.000000000 +0800
@@ -36,7 +36,7 @@
 MAKEFILE_INCLUDE_DIRS += -I/System/Library/Frameworks/OpenGL.framework/Headers 
 MAKEFILE_INCLUDE_DIRS += -I/System/Library/Frameworks/AGL.framework/Headers
 
-MAKEFILE_LDFLAGS += -framework CoreFoundation -framework CoreServices 
+MAKEFILE_LDFLAGS += -F@@MP_PREFIX@@/libexec/qt4/Library/Frameworks -framework QtCore -framework QtGui
 MAKEFILE_LDFLAGS += -framework Cocoa -framework AppKit -framework 
 MAKEFILE_LDFLAGS += ApplicationServices -framework OpenGL -framework AGL
 MAKEFILE_LDFLAGS += -u _PyMac_Error 

--- apps/vaporgui/main/Makefile	2013-08-24 03:06:48.000000000 +0800
+++ apps/vaporgui/main/Makefile	2026-06-06 11:57:43.000000000 +0800
@@ -17,8 +17,8 @@
 endif
 
 ifneq ($(ARCH),WIN32)
-MAKEFILE_CXXFLAGS += -DPYTHONVERSION=\"$(PYTHONVERSION)\"
-MAKEFILE_CXXFLAGS += -DPYTHONDIR=\"$(PYTHONDIR)\"
+MAKEFILE_CXXFLAGS += -DPYTHONVERSION=\"2.7\"
+MAKEFILE_CXXFLAGS += -DPYTHONDIR=\"@@MP_PREFIX@@/Library/Frameworks/Python.framework/Versions/2.7\"
 endif
 
 
