--- lib/params/flowparams.cpp	2013-08-24 03:03:56.000000000 +0800
+++ lib/params/flowparams.cpp	2026-06-06 10:50:33.000000000 +0800
@@ -1183,7 +1183,7 @@
 	if (numValidTimesteps < 2){
 		MyBase::SetErrMsg(VAPOR_ERROR_FLOW, "Insufficient time steps for unsteady advection");
 		delete [] timeStepList;
-		return false;
+		return 0;
 	}
 	if (numValidTimesteps < numTimesteps && ds->warnIfDataMissing()){
 		
