diff --git numpy/_core/src/highway/hwy/detect_compiler_arch.h numpy/_core/src/highway/hwy/detect_compiler_arch.h
index f4ccf2d8..101a9bab 100644
--- numpy/_core/src/highway/hwy/detect_compiler_arch.h
+++ numpy/_core/src/highway/hwy/detect_compiler_arch.h
@@ -244,13 +244,13 @@
 #define HWY_ARCH_X86 0
 #endif
 
-#if defined(__powerpc64__) || defined(_M_PPC) || defined(__powerpc__)
+#if defined(__powerpc64__) || defined(_M_PPC) || defined(__powerpc__) || defined(__POWERPC__)
 #define HWY_ARCH_PPC 1
 #else
 #define HWY_ARCH_PPC 0
 #endif
 
-#if defined(__powerpc64__) || (HWY_ARCH_PPC && defined(__64BIT__))
+#if defined(__powerpc64__) || (HWY_ARCH_PPC && defined(__64BIT__)) || defined(__ppc64__)
 #define HWY_ARCH_PPC_64 1
 #else
 #define HWY_ARCH_PPC_64 0
