From 2e9c9fb007c241bd54aa1f77e191f697693fe985 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Jul 2024 11:51:06 +0800
Subject: [PATCH 03/10] display: fix

---
 src/osx/core/display.cpp | 31 +++++++++++++++++++++----------
 1 file changed, 21 insertions(+), 10 deletions(-)

diff --git src/osx/core/display.cpp src/osx/core/display.cpp
index 6b8004da4c..4fd775e6a0 100644
--- src/osx/core/display.cpp
+++ src/osx/core/display.cpp
@@ -31,6 +31,8 @@
 
 #include "wx/osx/private.h"
 
+#ifdef __WXMAC__
+
 // ----------------------------------------------------------------------------
 // common helpers compiled even in wxUSE_DISPLAY==0 case
 // ----------------------------------------------------------------------------
@@ -320,7 +322,7 @@ static int wxOSXCGDisplayModeGetBitsPerPixel( CGDisplayModeRef theValue )
         depth = 16;
     else if ( CFStringCompare( pixelEncoding, CFSTR(IO8BitIndexedPixels) , kCFCompareCaseInsensitive) == kCFCompareEqualTo )
         depth = 8;
-    
+
     return depth;
 }
 
@@ -329,17 +331,17 @@ wxArrayVideoModes wxDisplayImplMacOSX::GetModes(const wxVideoMode& mode) const
     wxArrayVideoModes resultModes;
 
     wxCFRef<CFArrayRef> theArray(CGDisplayCopyAllDisplayModes( m_id ,NULL ) );
-    
+
     for (CFIndex i = 0; i < CFArrayGetCount(theArray); ++i)
     {
         CGDisplayModeRef theValue = static_cast<CGDisplayModeRef>(const_cast<void*>(CFArrayGetValueAtIndex(theArray, i)));
-        
+
         wxVideoMode theMode(
                             CGDisplayModeGetWidth(theValue),
                             CGDisplayModeGetHeight(theValue),
                             wxOSXCGDisplayModeGetBitsPerPixel(theValue),
                             int(CGDisplayModeGetRefreshRate(theValue)));
-        
+
         if (theMode.Matches( mode ))
             resultModes.Add( theMode );
     }
@@ -350,7 +352,7 @@ wxArrayVideoModes wxDisplayImplMacOSX::GetModes(const wxVideoMode& mode) const
 wxVideoMode wxDisplayImplMacOSX::GetCurrentMode() const
 {
     wxCFRef<CGDisplayModeRef> theValue( CGDisplayCopyDisplayMode( m_id ) );
-    
+
     return wxVideoMode(
                        CGDisplayModeGetWidth(theValue),
                        CGDisplayModeGetHeight(theValue),
@@ -370,20 +372,20 @@ bool wxDisplayImplMacOSX::ChangeMode( const wxVideoMode& mode )
 
     wxCHECK_MSG( mode.GetWidth() && mode.GetHeight(), false,
                 wxT("at least the width and height must be specified") );
-    
+
     bool bOK = false;
     wxCFRef<CFArrayRef> theArray(CGDisplayCopyAllDisplayModes( m_id ,NULL ) );
-    
+
     for (CFIndex i = 0; i < CFArrayGetCount(theArray); ++i)
     {
         CGDisplayModeRef theValue = static_cast<CGDisplayModeRef>(const_cast<void*>(CFArrayGetValueAtIndex(theArray, i)));
-        
+
         wxVideoMode theMode(
                             CGDisplayModeGetWidth(theValue),
                             CGDisplayModeGetHeight(theValue),
                             wxOSXCGDisplayModeGetBitsPerPixel(theValue),
                             int(CGDisplayModeGetRefreshRate(theValue)));
-        
+
         if ( theMode.GetWidth() == mode.GetWidth() && theMode.GetHeight() == mode.GetHeight() &&
             ( mode.GetDepth() == 0 || theMode.GetDepth() == mode.GetDepth() ) &&
             ( mode.GetRefresh() == 0 || theMode.GetRefresh() == mode.GetRefresh() ) )
@@ -393,7 +395,7 @@ bool wxDisplayImplMacOSX::ChangeMode( const wxVideoMode& mode )
             break;
         }
     }
-    
+
     return bOK;
 }
 
@@ -441,4 +443,13 @@ protected:
     return new wxDisplayFactorySingleMacOSX;
 }
 
+#else
+
+/* static */ wxDisplayFactory *wxDisplay::CreateFactory()
+{
+    return new wxDisplayFactorySingle;
+}
+
+#endif
+
 #endif // wxUSE_DISPLAY
-- 
2.45.2

