From a862398b2bdf9cc968110062874981e46134a09b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Jul 2024 12:01:50 +0800
Subject: [PATCH 06/10] defs.h: fix

---
 include/wx/defs.h | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git include/wx/defs.h include/wx/defs.h
index 39ce129477..ebb75ebd07 100644
--- include/wx/defs.h
+++ include/wx/defs.h
@@ -2811,10 +2811,6 @@ typedef int (* LINKAGEMODE wxListIterateFunction)(void *current);
 #define DECLARE_WXOSX_OPAQUE_CFREF( name ) typedef struct __##name* name##Ref;
 #define DECLARE_WXOSX_OPAQUE_CONST_CFREF( name ) typedef const struct __##name* name##Ref;
 
-#endif
-
-#ifdef __WXMAC__
-
 #define WX_OPAQUE_TYPE( name ) struct wxOpaque##name
 
 typedef void*       WXHCURSOR;
@@ -2832,7 +2828,9 @@ typedef unsigned short  WXWORD;
 
 typedef WX_OPAQUE_TYPE(PicHandle ) * WXHMETAFILE ;
 
+#if !defined(__WXX11__) && !defined(__WXGTK__)
 typedef void*       WXDisplay;
+#endif
 
 /*
  * core frameworks
@@ -2910,7 +2908,7 @@ typedef HIShapeRef WXHRGN;
 
 #endif // __WXMAC__
 
-#if defined(__WXMAC__)
+#if defined(__WXMAC__) || defined(__OBJC__)
 
 /* Objective-C type declarations.
  * These are to be used in public headers in lieu of NSSomething* because
-- 
2.45.2

