From 5cced63f3187d8d908c51d99c69972e7086d4749 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Jul 2024 12:03:47 +0800
Subject: [PATCH 07/10] intl.cpp: fix

---
 src/common/intl.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git src/common/intl.cpp src/common/intl.cpp
index 1a9038a6b9..86f36a82d0 100644
--- src/common/intl.cpp
+++ src/common/intl.cpp
@@ -57,7 +57,7 @@
 
 #ifdef __WIN32__
     #include "wx/msw/private/uilocale.h"
-#elif defined(__WXOSX__)
+#elif defined(__APPLE__)
     #include "wx/osx/core/cfref.h"
     #include "wx/osx/core/cfstring.h"
     #include <CoreFoundation/CFLocale.h>
@@ -556,7 +556,7 @@ wxString wxLocale::GetSystemEncodingName()
     default:
         encname.Printf(wxS("windows-%u"), codepage);
     }
-#elif defined(__WXMAC__)
+#elif defined(__APPLE__)
     encname = wxCFStringRef::AsString(
         CFStringGetNameOfEncoding(CFStringGetSystemEncoding())
     );
@@ -867,7 +867,7 @@ wxString wxLocale::GetHeaderValue(const wxString& header,
 // accessors for locale-dependent data
 // ----------------------------------------------------------------------------
 
-#if defined(__WINDOWS__) || defined(__WXOSX__)
+#if defined(__WINDOWS__) || defined(__APPLE__)
 
 namespace
 {
@@ -1382,7 +1382,7 @@ wxString wxLocale::GetOSInfo(wxLocaleInfo index, wxLocaleCategory cat)
     return wxUILocale::GetCurrent().GetInfo(index, cat);
 }
 
-#elif defined(__WXOSX__)
+#elif defined(__APPLE__)
 
 // This function is also used by wxUILocaleImpl, so don't make it private.
 extern wxString
-- 
2.45.2

