From 441e4346717ceb8e0b5db49dd6889f8b7c24f839 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Jul 2024 12:05:06 +0800
Subject: [PATCH 08/10] webrequest_urlsession.h: fix

---
 include/wx/osx/private/webrequest_urlsession.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git include/wx/osx/private/webrequest_urlsession.h include/wx/osx/private/webrequest_urlsession.h
index dad169703e..4ec7b3bf7b 100644
--- include/wx/osx/private/webrequest_urlsession.h
+++ include/wx/osx/private/webrequest_urlsession.h
@@ -10,6 +10,15 @@
 #ifndef _WX_OSX_WEBREQUEST_URLSESSION_H
 #define _WX_OSX_WEBREQUEST_URLSESSION_H
 
+#if defined(__OBJC__)
+#define DECLARE_WXCOCOA_OBJC_CLASS(klass) \
+@class klass; \
+typedef klass *WX_##klass
+#else /*  not defined(__OBJC__) */
+#define DECLARE_WXCOCOA_OBJC_CLASS(klass) \
+typedef struct klass *WX_##klass
+#endif /*  defined(__OBJC__) */
+
 #if wxUSE_WEBREQUEST_URLSESSION
 
 #include "wx/private/webrequest.h"
@@ -18,6 +27,7 @@ DECLARE_WXCOCOA_OBJC_CLASS(NSURLCredential);
 DECLARE_WXCOCOA_OBJC_CLASS(NSURLSession);
 DECLARE_WXCOCOA_OBJC_CLASS(NSURLSessionTask);
 DECLARE_WXCOCOA_OBJC_CLASS(wxWebSessionDelegate);
+DECLARE_WXCOCOA_OBJC_CLASS(NSData);
 
 class wxWebSessionURLSession;
 class wxWebRequestURLSession;
-- 
2.45.2

