From fe89060d6bfc443659cf405335154ca4d05a7d35 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 18 Mar 2025 01:18:45 +0800
Subject: [PATCH 12/13] xlocale

---
 src/unix/uilocale.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/unix/uilocale.cpp b/src/unix/uilocale.cpp
index 4aaebee4a5..b3d64ed2bc 100644
--- a/src/unix/uilocale.cpp
+++ b/src/unix/uilocale.cpp
@@ -21,6 +21,10 @@
 #if wxUSE_INTL
 
 #include <locale.h>
+#ifdef __APPLE__
+    #include <xlocale.h>
+#endif
+
 #ifdef HAVE_LANGINFO_H
     #include <langinfo.h>
 #endif
-- 
2.49.0

