From 0c9bc33143c279b70074615da25923591e060326 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Jul 2024 16:38:59 +0800
Subject: [PATCH 4/4] utilsunx: no memset_s

---
 src/unix/utilsunx.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/unix/utilsunx.cpp src/unix/utilsunx.cpp
index 8a944cd626..9d356ebb95 100644
--- src/unix/utilsunx.cpp
+++ src/unix/utilsunx.cpp
@@ -225,7 +225,7 @@ void wxSecureZeroMemory(void* v, size_t n)
     // but may be found in a non-standard header file, or in a library that is
     // not linked by default.
     explicit_bzero(v, n);
-#elif defined(__DARWIN__) || defined(__STDC_LIB_EXT1__)
+#elif defined(__STDC_LIB_EXT1__)
     // memset_s() is available since OS X 10.9, and may be available on
     // other platforms.
     memset_s(v, n, 0, n);
-- 
2.45.2

