--- DarwinProcess.c
+++ DarwinProcess.c	2025-07-08 12:43:33.000000000 +0800
@@ -444,6 +444,7 @@
    const bool hideUserlandThreads = dpt->super.super.host->settings->hideUserlandThreads;
    bool isProcessStuck = false;
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
    for (mach_msg_type_number_t i = 0; i < thread_count; i++) {
 
       thread_identifier_info_data_t identifer_info;
@@ -512,6 +513,7 @@
       if (!preExisting)
          ProcessTable_add(&dpt->super, tprocess);
    }
+#endif
 
    if (isProcessStuck) {
       dp->super.state = UNINTERRUPTIBLE_WAIT;
