From 0966d4114e16a121f5c406b255b52db7006adc8e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 1 Jul 2025 05:06:51 +0800
Subject: [PATCH] Fix build with gcc13+

---
 lib/TerminalDisplay.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/TerminalDisplay.cpp b/lib/TerminalDisplay.cpp
index 1399c32..5a21730 100644
--- a/lib/TerminalDisplay.cpp
+++ b/lib/TerminalDisplay.cpp
@@ -23,6 +23,8 @@
 // Own
 #include "TerminalDisplay.h"
 
+#include <cstdint>
+
 // Qt
 #include <QAbstractButton>
 #include <QApplication>
