From e19cdcb915b0b7e6c0c1e86a33da51e8a0164630 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 7 Jan 2025 10:26:33 +0800
Subject: [PATCH] mobical.cpp: define MAP_ANONYMOUS if undefined

---
 src/mobical.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git src/mobical.cpp src/mobical.cpp
index 9339e96..7f60c71 100644
--- src/mobical.cpp
+++ src/mobical.cpp
@@ -4581,6 +4581,10 @@ extern "C" {
 #include <sys/mman.h>
 }
 
+#ifndef MAP_ANONYMOUS
+#define MAP_ANONYMOUS MAP_ANON
+#endif
+
 Mobical::Mobical () {
   const int prot = PROT_READ | PROT_WRITE;
   const int flags = MAP_ANONYMOUS | MAP_SHARED;
