diff --git a/3rd/luamake/bee.lua/binding/lua_time.cpp b/3rd/luamake/bee.lua/binding/lua_time.cpp
index bdaa79d..9e0cba5 100644
--- a/3rd/luamake/bee.lua/binding/lua_time.cpp
+++ b/3rd/luamake/bee.lua/binding/lua_time.cpp
@@ -81,15 +81,15 @@ namespace bee::lua_time {
 #elif defined(__APPLE__)
     template <uint32_t Numer, uint32_t Denom>
     static int lua_monotonic(lua_State* L) {
-        static_assert((NSEC_PER_MSEC * Denom) % Numer == 0);
-        constexpr uint64_t Freq = (NSEC_PER_MSEC * Denom) / Numer;
+        static_assert((NSecPerMSec * Denom) % Numer == 0);
+        constexpr uint64_t Freq = (NSecPerMSec * Denom) / Numer;
         uint64_t now            = mach_continuous_time();
         lua_pushinteger(L, now / Freq);
         return 1;
     }
     static int lua_monotonic(lua_State* L) {
         uint64_t now   = mach_continuous_time();
-        uint64_t freq  = NSEC_PER_MSEC * G.timebase.denom;
+        uint64_t freq  = NSecPerMSec * G.timebase.denom;
         uint64_t whole = (now / freq) * G.timebase.numer;
         uint64_t part  = (now % freq) * G.timebase.numer / freq;
         lua_pushinteger(L, whole + part);
