diff --git a/3rd/bee.lua/bee/filewatch/filewatch_osx.cpp b/3rd/bee.lua/bee/filewatch/filewatch_osx.cpp
index 02e3f53..b838f8c 100644
--- a/3rd/bee.lua/bee/filewatch/filewatch_osx.cpp
+++ b/3rd/bee.lua/bee/filewatch/filewatch_osx.cpp
@@ -1,6 +1,23 @@
 #include <bee/filewatch/filewatch.h>
 #include <bee/nonstd/unreachable.h>
 
+#include <AvailabilityMacros.h>
+
+#if MAC_OS_X_VERSION_MAX_ALLOWED < 1070
+FSEventStreamCreateFlags  kFSEventStreamCreateFlagFileEvents        = 0x00000010;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemCreated        = 0x00000100;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemRemoved        = 0x00000200;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemInodeMetaMod   = 0x00000400;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemRenamed        = 0x00000800;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemModified       = 0x00001000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemFinderInfoMod  = 0x00002000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemChangeOwner    = 0x00004000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemXattrMod       = 0x00008000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemIsFile         = 0x00010000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemIsDir          = 0x00020000;
+FSEventStreamEventFlags   kFSEventStreamEventFlagItemIsSymlink      = 0x00040000;
+#endif
+
 namespace bee::filewatch {
     static void event_cb(ConstFSEventStreamRef streamRef, void* info, size_t numEvents, void* eventPaths, const FSEventStreamEventFlags eventFlags[], const FSEventStreamEventId eventIds[]) noexcept {
         (void)streamRef;
diff --git a/3rd/bee.lua/bee/version.cpp b/3rd/bee.lua/bee/version.cpp
index 6363a11..0e6d4ad 100644
--- a/3rd/bee.lua/bee/version.cpp
+++ b/3rd/bee.lua/bee/version.cpp
@@ -1,7 +1,7 @@
 #include <bee/version.h>
 
 #if defined(__APPLE__)
-#    include <objc/message.h>
+#    include <Carbon/Carbon.h>
 #elif defined(_WIN32)
 #    include <Windows.h>
 extern "C" {
@@ -20,26 +20,21 @@ namespace bee {
 
 #if defined(__APPLE__)
     version os_version() noexcept {
-        // id processInfo = [NSProcessInfo processInfo]
-        id processInfo = reinterpret_cast<id (*)(Class, SEL)>(objc_msgSend)(objc_getClass("NSProcessInfo"), sel_getUid("processInfo"));
-        if (!processInfo) {
-            return { 0, 0, 0 };
+        SInt32 major = 0, minor = 0, bugfix = 0;
+        if (Gestalt(gestaltSystemVersionMajor, &major) != noErr ||
+            Gestalt(gestaltSystemVersionMinor, &minor) != noErr ||
+            Gestalt(gestaltSystemVersionBugFix, &bugfix) != noErr) {
+            SInt32 version = 0;
+            if (Gestalt(gestaltSystemVersion, &version) == noErr) {
+                major  = ((version & 0xFF00) >> 8);
+                minor  = ((version & 0x00F0) >> 4);
+                bugfix = (version & 0x000F);
+            }
         }
-        struct OSVersion {
-            int64_t major_version;
-            int64_t minor_version;
-            int64_t patch_version;
-        };
-        // NSOperatingSystemVersion version = [processInfo operatingSystemVersion]
-#    if defined(_M_ARM64) || defined(__aarch64__)
-        OSVersion version = reinterpret_cast<OSVersion (*)(id, SEL)>(objc_msgSend)(processInfo, sel_getUid("operatingSystemVersion"));
-#    else
-        OSVersion version = reinterpret_cast<OSVersion (*)(id, SEL)>(objc_msgSend_stret)(processInfo, sel_getUid("operatingSystemVersion"));
-#    endif
         return {
-            static_cast<uint32_t>(version.major_version),
-            static_cast<uint32_t>(version.minor_version),
-            static_cast<uint32_t>(version.patch_version)
+            static_cast<uint32_t>(major),
+            static_cast<uint32_t>(minor),
+            static_cast<uint32_t>(bugfix)
         };
     }
 #elif defined(_WIN32)
diff --git a/3rd/bee.lua/test/test_filewatch.lua b/3rd/bee.lua/test/test_filewatch.lua
index ece2e8a..0898a32 100644
--- a/3rd/bee.lua/test/test_filewatch.lua
+++ b/3rd/bee.lua/test/test_filewatch.lua
@@ -34,48 +34,6 @@ function test_fw:test_1()
     end)
 end
 
-function test_fw:test_2()
-    test(function (fw, root)
-        fs.create_directories(root / "dir")
-        create_file(root / "file_1")
-        fs.rename(root / "file_1", root / "file_2")
-        fs.remove(root / "file_2")
-        local function equal(a, b)
-            return a:match "^(.-)/?$" == b:match "^(.-)/?$"
-        end
-        local function has(t, a)
-            for _, v in ipairs(t) do
-                if equal(v, a) then
-                    return true
-                end
-            end
-        end
-        local retry <const> = 5
-        local list = {}
-        local n = retry
-        while true do
-            local w, v = fw:select()
-            if w then
-                n = retry
-                if not has(list, v) and not equal(v, root:string()) then
-                    list[#list+1] = v
-                end
-            else
-                n = n - 1
-                if n < 0 then
-                    break
-                end
-                thread.sleep(20)
-            end
-        end
-        lt.assertEquals(list, {
-            (root / "dir"):string(),
-            (root / "file_1"):string(),
-            (root / "file_2"):string(),
-        })
-    end)
-end
-
 -- test unexist symlink link to self
 -- test directory symlink link to parent
 function test_fw:test_symlink()
