--- src/imageformat.hpp.orig	2026-09-05 16:45:20.000000000 +0800
+++ src/imageformat.hpp	2026-09-09 14:40:44.000000000 +0800
@@ -164,7 +164,7 @@
 
 protected:
     // Name of "enable" decoder parameter
-    static constexpr const std::string enable_param_name = "enable";
+    static constexpr const char* enable_param_name = "enable";
 
     /**
      * Check signature existence in source data buffer.
