--- CMakeLists.txt	2024-03-19 11:40:45.000000000 +0800
+++ CMakeLists.txt	2026-06-20 03:41:33.000000000 +0800
@@ -201,7 +201,9 @@
 add_executable(fish_test_helper src/fish_test_helper.cpp)
 
 # Set up tests.
-include(cmake/Tests.cmake)
+if (BUILD_TESTING)
+  include(cmake/Tests.cmake)
+endif()
 
 # Benchmarking support.
 include(cmake/Benchmark.cmake)

--- cmake/Install.cmake	2024-03-19 11:40:45.000000000 +0800
+++ cmake/Install.cmake	2026-06-20 03:47:20.000000000 +0800
@@ -175,7 +175,6 @@
 
 # Group install targets into a InstallTargets folder
 set_property(TARGET build_fish_pc CHECK-FISH-BUILD-VERSION-FILE
-                    tests_buildroot_target
              PROPERTY FOLDER cmake/InstallTargets)
 
 # Make a target build_root that installs into the buildroot directory, for testing.
