--- configure.ac	2026-08-24 03:43:07.000000000 +0800
+++ configure.ac	2026-10-03 06:23:41.000000000 +0800
@@ -351,8 +351,6 @@
   src/maple2c/gga_vxc/Makefile
   src/maple2c/mgga_exc/Makefile
   src/maple2c/mgga_vxc/Makefile
-  testsuite/Makefile
-  testsuite/general/Makefile
   pkgconfig/Makefile
   pkgconfig/libxc.pc
   pkgconfig/libxcf03.pc
@@ -366,8 +364,6 @@
   maple/mgga_vxc/Makefile
   pylibxc/Makefile
   cmake/Makefile
-  examples/Makefile
-  examples/fortran/Makefile
 ])
 
 AC_OUTPUT

--- Makefile.am	2026-08-24 03:43:07.000000000 +0800
+++ Makefile.am	2026-10-03 06:22:36.000000000 +0800
@@ -8,7 +8,7 @@
 ##
 
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = pkgconfig scripts maple src examples testsuite cmake pylibxc
+SUBDIRS = pkgconfig scripts maple src cmake pylibxc
 
 include_HEADERS = xc_version.h libxc.bib
 EXTRA_DIST = ChangeLog.md PACKAGING setup.py config.h.cmake.in CMakeLists.txt CITATION
