--- meson.build	2025-11-21 13:42:58.000000000 +0800
+++ meson.build	2026-07-11 18:04:58.000000000 +0800
@@ -45,7 +45,6 @@
 absl_libs = [
   # Via Base:
   'absl_raw_logging_internal',
-  'absl_borrowed_fixup_buffer',
 
   # Via Strings:
   'absl_int128',
@@ -70,7 +70,6 @@
   'absl_cordz_update_tracker',
   'absl_crc32c',
   'absl_crc_cord_state',
-  'absl_crc_cpu_detect',
   'absl_crc_internal',
   'absl_debugging_internal',
   'absl_decode_rust_punycode',
