--- c/CMakeLists.txt.orig	2026-03-24 15:49:17.000000000 +0800
+++ c/CMakeLists.txt	2026-03-24 15:51:35.000000000 +0800
@@ -149,8 +149,10 @@
 )
 target_compile_features(blake3 PUBLIC c_std_99)
 if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.12)
-  target_compile_features(blake3 PUBLIC cxx_std_20)
-  # else: add it further below through `BLAKE3_CMAKE_CXXFLAGS_*`
+  if(BLAKE3_USE_TBB)
+    target_compile_features(blake3 PUBLIC cxx_std_20)
+    # else: add it further below through `BLAKE3_CMAKE_CXXFLAGS_*`
+  endif()
 endif()
 
 # ensure C_EXTENSIONS OFF is respected without overriding CMAKE_C_STANDARD
