--- libobjc/objc/runtime.h	1999-06-04 05:37:10.000000000 +0800
+++ libobjc/objc/runtime.h	2025-08-16 02:24:17.000000000 +0800
@@ -30,8 +30,9 @@
 
 #include <stdio.h>
 #include <ctype.h>
-
+#include <stdlib.h>		/* abort */
 #include <stddef.h>		/* so noone else will get system versions */
+
 #include "assert.h"
 
 #include "objc/objc.h"		/* core data types */

--- src/defobj/defobj.h	2008-11-02 12:19:20.000000000 +0800
+++ src/defobj/defobj.h	2025-08-16 02:50:26.000000000 +0800
@@ -26,6 +26,8 @@
 #ifndef __swarm_defobj_h
 #define __swarm_defobj_h
 
+#include <stdio.h>
+
 #if defined(__cplusplus) && !defined(__OBJC__)
 extern "C" {
 #endif
