--- src/crispy/utils.h	2025-01-01 00:15:38.000000000 +0800
+++ src/crispy/utils.h	2025-08-18 03:56:34.000000000 +0800
@@ -3,8 +3,10 @@
 #include <crispy/defines.h>
 #include <crispy/escape.h>
 
-#include <range/v3/view/iota.hpp>
-#include <range/v3/view/transform.hpp>
+#ifndef APPLE_OBJCPLUSPLUS
+    #include <range/v3/view/iota.hpp>
+    #include <range/v3/view/transform.hpp>
+#endif
 
 #include <algorithm>
 #include <filesystem>
@@ -28,6 +30,7 @@
 
 namespace views
 {
+#ifndef APPLE_OBJCPLUSPLUS
     template <typename T>
     auto as()
     {
@@ -39,6 +42,7 @@
     {
         return ranges::views::ints(0, n) | as<T>();
     }
+#endif
 } // namespace views
 
 constexpr std::string_view trimRight(std::string_view value) noexcept
