--- configure.in	2010-12-14 01:24:39.000000000 +0800
+++ configure.in	2026-06-06 17:06:24.000000000 +0800
@@ -1561,7 +1561,6 @@
   doc/Makefile
   doc/goops/Makefile
   doc/r5rs/Makefile
-  doc/ref/Makefile
   doc/tutorial/Makefile
   emacs/Makefile
   examples/Makefile

--- Makefile.in	2010-12-14 03:29:40.000000000 +0800
+++ Makefile.in	2026-06-06 17:30:48.000000000 +0800
@@ -357,7 +357,7 @@
 #
 AUTOMAKE_OPTIONS = 1.10
 SUBDIRS = oop libguile ice-9 guile-config guile-readline emacs \
-	  scripts srfi doc examples test-suite benchmark-suite lang am
+	  scripts srfi examples test-suite benchmark-suite lang am
 
 bin_SCRIPTS = guile-tools
 include_HEADERS = libguile.h
