--- Modules/posixmodule.c
+++ Modules/posixmodule.c	2024-10-31 03:05:47.000000000 +0800
@@ -13212,6 +13212,11 @@
  * os.statvfs is implemented in terms of statfs(2).
  */
 
+#if defined(__i386__) || defined(__ppc__)
+#define statfs statfs64
+#define fstatfs fstatfs64
+#endif
+
 static PyObject*
 _pystatvfs_fromstructstatfs(PyObject *module, struct statfs st) {
     PyObject *StatVFSResultType = get_posix_state(module)->StatVFSResultType;
