--- usrsctplib/usrsctp.h	2012-05-27 17:18:24.000000000 +0800
+++ usrsctplib/usrsctp.h	2025-08-20 18:04:14.000000000 +0800
@@ -49,6 +49,11 @@
 #define MSG_NOTIFICATION 0x2000         /* SCTP notification */
 #endif
 
+#ifndef IPPROTO_SCTP
+/* This is the IANA assigned protocol number of SCTP. */
+#define IPPROTO_SCTP 132
+#endif
+
 #if defined(__Userspace_os_Windows)
 #define uint8_t   unsigned __int8
 #define uint16_t  unsigned __int16
