--- usrsctplib/netinet/sctp_os_userspace.h	2021-01-13 22:37:19.000000000 +0800
+++ usrsctplib/netinet/sctp_os_userspace.h	2025-08-28 00:11:13.000000000 +0800
@@ -283,9 +283,7 @@
 
 #else /* !defined(Userspace_os_Windows) */
 #include <sys/socket.h>
-#if defined(__DragonFly__) || defined(__FreeBSD__) || defined(__linux__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__native_client__) || defined(__Fuchsia__) || defined(__EMSCRIPTEN_PTHREADS__)
 #include <pthread.h>
-#endif
 typedef pthread_mutex_t userland_mutex_t;
 typedef pthread_cond_t userland_cond_t;
 typedef pthread_t userland_thread_t;
