--- src/Makevars.in	2024-12-26 21:42:13.000000000 +0800
+++ src/Makevars.in	2025-01-15 04:05:25.000000000 +0800
@@ -1,2 +1,2 @@
-PKG_LIBS= -L@RHIGHS_LIB_DIR@/lib -L@RHIGHS_LIB_DIR@/lib64 -lhighs
-PKG_CPPFLAGS=-I@RHIGHS_LIB_DIR@/include -I@RHIGHS_LIB_DIR@/include/highs
+PKG_LIBS= -L@RHIGHS_LIB_DIR@/lib -lhighs -latomic
+PKG_CPPFLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -I@RHIGHS_LIB_DIR@/include -I@RHIGHS_LIB_DIR@/include/highs

--- inst/HiGHS/CMakeLists.txt	2025-01-07 21:23:34.000000000 +0800
+++ inst/HiGHS/CMakeLists.txt	2025-01-15 04:04:34.000000000 +0800
@@ -416,6 +416,11 @@
 
 configure_file(${HIGHS_SOURCE_DIR}/src/HConfig.h.in ${HIGHS_BINARY_DIR}/HConfig.h)
 
+if (CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_GLIBCXX_USE_CXX11_ABI=0 -Wl,-latomic")
+  set (CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} -latomic")
+endif()
+
 if (DEBUG_MEMORY STREQUAL "Address")
   set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} \
       -fsanitize=address,undefined \
