--- vpx_util/vpx_thread.c
+++ vpx_util/vpx_thread.c	2025-08-24 19:11:22.000000000 +0800
@@ -25,6 +25,10 @@
 #include "vpx_mem/vpx_mem.h"
 #include "vpx_util/vpx_pthread.h"
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 #if CONFIG_MULTITHREAD
 
 struct VPxWorkerImpl {
@@ -39,7 +43,7 @@
 
 static THREADFN thread_loop(void *ptr) {
   VPxWorker *const worker = (VPxWorker *)ptr;
-#ifdef __APPLE__
+#if defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1060)
   if (worker->thread_name != NULL) {
     // Apple's version of pthread_setname_np takes one argument and operates on
     // the current thread only. The maximum size of the thread_name buffer was
