From 5908b35111114dfba48d5da6a7ab434a2bb14b2c Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 16 Oct 2025 12:00:17 +0800
Subject: [PATCH] meson.build: fix finding python

diff --git a/meson.build b/meson.build
index 0189457c..c71d3d0b 100644
--- a/meson.build
+++ b/meson.build
@@ -22,8 +22,7 @@ domainext = 'info'
 gitrepo = 'https://codeberg.org/' + authornickname + '/' + prettyname
 website = 'https://apps.gnome.org/Komikku/'
 
-python = import('python')
-py_installation = python.find_installation('python3')
+py_installation = import('python').find_installation('@PYTHON@')
 if not py_installation.found()
     error('No valid python3 binary found')
 else
