From efd565b664bfe8c17234efb0b990350b76d502ee Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 16 Oct 2025 14:10:26 +0800
Subject: [PATCH] Really not glycin

---
 komikku/reader/pager/image.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/komikku/reader/pager/image.py b/komikku/reader/pager/image.py
index 2b7d5f0c..f14cb67e 100644
--- a/komikku/reader/pager/image.py
+++ b/komikku/reader/pager/image.py
@@ -11,8 +11,6 @@ from PIL import Image
 from PIL import ImageChops
 
 gi.require_version('Gdk', '4.0')
-gi.require_version('Gly', '2')
-gi.require_version('GlyGtk4', '2')
 gi.require_version('Gtk', '4.0')
 gi.require_version('GdkPixbuf', '2.0')
 gi.require_version('Graphene', '1.0')
