--- meson.build.orig	2021-03-22 19:25:50.000000000 +0800
+++ meson.build	2024-11-14 16:55:49.000000000 +0800
@@ -39,17 +39,7 @@
 endif
 add_project_arguments(global_cflags, language: 'c')
 
-test_ldflags = [
-  '-pie',
-  '-Wl,-z,relro',
-  '-Wl,-z,now'
-]
 global_ldflags = []
-foreach ldflag : test_ldflags
-  if cc.has_argument(ldflag)
-    global_ldflags += ldflag
-  endif
-endforeach
 add_project_link_arguments(global_ldflags, language: 'c')
 
 if not cc.has_function('setlocale')
