--- src/molecule-class-info.cc	2026-04-01 08:29:03.000000000 +0800
+++ src/molecule-class-info.cc	2026-05-07 04:37:54.000000000 +0800
@@ -8527,8 +8527,16 @@
    coot::backup_file_info_t bfi(file_name, description);
    bfi.imol = imol_no;
    bfi.valid_status = true;
+#if defined(__APPLE__) && !defined(CLOCK_REALTIME)
+   {
+      struct timeval tv;
+      gettimeofday(&tv, nullptr);
+      bfi.ctime.tv_sec = tv.tv_sec;
+      bfi.ctime.tv_nsec = tv.tv_usec * 1000;
+   }
+#else
    clock_gettime(CLOCK_REALTIME, &bfi.ctime);
-
+#endif
    history_filename_vec.push_back(bfi);
 }
 
