--- ideal/Makefile.am	2026-04-01 08:29:03.000000000 +0800
+++ ideal/Makefile.am	2026-05-07 00:22:50.000000000 +0800
@@ -45,7 +45,7 @@
 AM_CXXFLAGS = $(gsl_CFLAGS) $(COOT_WINDOWS_CFLAGS) \
      $(CLIPPER_CXXFLAGS) $(MMDB_CXXFLAGS) $(LIBPNG_CFLAGS) $(RDKIT_CXXFLAGS)
 
-AM_LDFLAGS = $(MMDB_LIBS) $(CLIPPER_LIBS) $(GLS_LIBS)
+AM_LDFLAGS = $(MMDB_LIBS) $(CLIPPER_LIBS) $(GLS_LIBS) -latomic
 
 #libcoot_ideal_la_LIBADD = ../coot-utils/libcoot-utils.la                       \
 #	../coot-utils/libcoot-coord-utils.la                                   \
