--- ligand/Makefile.am	2026-04-01 08:29:03.000000000 +0800
+++ ligand/Makefile.am	2026-05-07 00:41:59.000000000 +0800
@@ -310,7 +310,7 @@
         $(top_builddir)/analysis/libcoot-analysis.la	   \
 	$(top_builddir)/compat/libcoot-compat.la           \
         $(GLOB_LIBS) \
-	$(CLIPPER_LIBS) $(MMDB_LIBS) $(gsl_LIBS)
+	$(CLIPPER_LIBS) $(MMDB_LIBS) $(gsl_LIBS) -latomic
 
 # 	$(top_builddir)/coords/libcoot-coords.la
 
