--- libkexiv2/kexiv2.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2.cpp	2026-02-04 12:28:11.000000000 +0800
@@ -241,7 +241,7 @@
 
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((Exiv2::byte*)imgData.data(), imgData.size());
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((Exiv2::byte*)imgData.data(), imgData.size());
 
         d->filePath.clear();
         image->readMetadata();
@@ -297,7 +297,7 @@
 
     try
     {
-        Exiv2::Image::AutoPtr image;
+        Exiv2::Image::UniquePtr image;
 
         image        = Exiv2::ImageFactory::open((const char*)(QFile::encodeName(filePath)));
 
@@ -346,7 +346,7 @@
             QString xmpSidecarPath = sidecarFilePathForFile(filePath);
             QFileInfo xmpSidecarFileInfo(xmpSidecarPath);
 
-            Exiv2::Image::AutoPtr xmpsidecar;
+            Exiv2::Image::UniquePtr xmpsidecar;
             if (xmpSidecarFileInfo.exists() && xmpSidecarFileInfo.isReadable())
             {
                 // Read sidecar data
@@ -354,7 +354,7 @@
                 xmpsidecar->readMetadata();
 
                 // Merge
-                d->loadSidecarData(xmpsidecar);
+                d->loadSidecarData(std::move(xmpsidecar));
                 hasLoaded = true;
             }
         }

--- libkexiv2/kexiv2_p.h.orig	2026-02-04 11:43:29.000000000 +0800
+++ libkexiv2/kexiv2_p.h	2026-02-04 12:21:22.000000000 +0800
@@ -149,7 +149,7 @@
 
     bool saveToXMPSidecar(const QFileInfo& finfo)    const;
     bool saveToFile(const QFileInfo& finfo)          const;
-    bool saveOperations(const QFileInfo& finfo, Exiv2::Image::AutoPtr image) const;
+    bool saveOperations(const QFileInfo& finfo, Exiv2::Image::UniquePtr image) const;
 
     /** Wrapper method to convert a Comments content to a QString.
      */
@@ -176,7 +176,7 @@
 #ifdef _XMP_SUPPORT_
     Exiv2::XmpData&        xmpMetadata()         { return data.data()->xmpMetadata;        }
 
-    void loadSidecarData(Exiv2::Image::AutoPtr xmpsidecar);
+    void loadSidecarData(Exiv2::Image::UniquePtr xmpsidecar);
 #endif
 
 public:

--- libkexiv2/kexiv2_p.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2_p.cpp	2026-02-04 12:30:56.000000000 +0800
@@ -74,9 +74,9 @@
 
     try
     {
-        Exiv2::Image::AutoPtr image;
+        Exiv2::Image::UniquePtr image;
         image = Exiv2::ImageFactory::create(Exiv2::ImageType::xmp, (const char*)(QFile::encodeName(filePath)));
-        return saveOperations(finfo, image);
+        return saveOperations(finfo, std::move(image));
     }
     catch( Exiv2::Error& e )
     {
@@ -150,9 +150,9 @@
 
     try
     {
-        Exiv2::Image::AutoPtr image;
+        Exiv2::Image::UniquePtr image;
         image = Exiv2::ImageFactory::open((const char*)(QFile::encodeName(finfo.filePath())));
-        return saveOperations(finfo, image);
+        return saveOperations(finfo, std::move(image));
     }
     catch( Exiv2::Error& e )
     {
@@ -166,7 +166,7 @@
     }
 }
 
-bool KExiv2::Private::saveOperations(const QFileInfo& finfo, Exiv2::Image::AutoPtr image) const
+bool KExiv2::Private::saveOperations(const QFileInfo& finfo, Exiv2::Image::UniquePtr image) const
 {
     try
     {
@@ -331,7 +331,8 @@
 {
     std::string s(e.what());
     kError() << msg.toAscii().constData() << " (Error #"
-             << e.code() << ": " << s.c_str();
+             << Exiv2::Error(e.code()).what()
+             << ")";
 }
 
 void KExiv2::Private::printExiv2MessageHandler(int lvl, const char* msg)
@@ -445,7 +446,7 @@
 }
 
 #ifdef _XMP_SUPPORT_
-void KExiv2::Private::loadSidecarData(Exiv2::Image::AutoPtr xmpsidecar)
+void KExiv2::Private::loadSidecarData(Exiv2::Image::UniquePtr xmpsidecar)
 {
     // Having a sidecar is a special situation.
     // The sidecar data often "dominates", see in particular bug 309058 for important aspects:

--- libkexiv2/kexiv2comments.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2comments.cpp	2026-02-04 12:33:02.000000000 +0800
@@ -37,7 +37,7 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((const char*)
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((const char*)
                                       (QFile::encodeName(filePath)));
 
         Exiv2::AccessMode mode = image->checkMode(Exiv2::mdComment);
@@ -47,7 +47,8 @@
     {
         std::string s(e.what());
         kError() << "Cannot check Comment access mode using Exiv2 (Error #"
-                 << e.code() << ": " << s.c_str() << ")";
+                 << Exiv2::Error(e.code()).what()
+                 << ")";
     }
     catch(...)
     {

--- libkexiv2/kexiv2exif.cpp.orig	2026-02-04 11:43:29.000000000 +0800
+++ libkexiv2/kexiv2exif.cpp	2026-02-04 13:05:55.000000000 +0800
@@ -44,7 +44,7 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((const char*)
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((const char*)
                                       (QFile::encodeName(filePath)));
 
         Exiv2::AccessMode mode = image->checkMode(Exiv2::mdExif);
@@ -55,7 +55,8 @@
     {
         std::string s(e.what());
         kError() << "Cannot check Exif access mode using Exiv2 (Error #"
-                 << e.code() << ": " << s.c_str() << ")";
+                 << Exiv2::Error(e.code()).what()
+                 << ")";
     }
     catch(...)
     {
@@ -690,7 +691,7 @@
 
         if (it != exifData.end() && it->count() > 0)
         {
-            val = it->toLong(component);
+            val = it->toUint32(component);
             return true;
         }
     }
@@ -756,7 +757,7 @@
                 case Exiv2::signedShort:
                 case Exiv2::signedLong:
                     if (it->count() > component)
-                        return QVariant((int)it->toLong(component));
+                        return QVariant((int)it->toUint32(component));
                     else
                         return QVariant(QVariant::Int);
                 case Exiv2::unsignedRational:
@@ -890,7 +891,7 @@
     {
         Exiv2::ExifThumbC thumb(d->exifMetadata());
         Exiv2::DataBuf const c1 = thumb.copy();
-        thumbnail.loadFromData(c1.pData_, c1.size_);
+        thumbnail.loadFromData(c1.c_data(), c1.size());
 
         if (!thumbnail.isNull())
         {
@@ -906,7 +907,7 @@
 
                 if (it != exifData.end() && it->count())
                 {
-                    long orientation = it->toLong();
+                    long orientation = it->toUint32();
                     kDebug() << "Exif Thumbnail Orientation: " << (int)orientation;
                     rotateExifQImage(thumbnail, (ImageOrientation)orientation);
                 }
@@ -984,9 +985,9 @@
         // Make sure IFD0 is explicitly marked as a main image
         Exiv2::ExifData::const_iterator pos = d->exifMetadata().findKey(Exiv2::ExifKey("Exif.Image.NewSubfileType"));
 
-        if (pos == d->exifMetadata().end() || pos->count() != 1 || pos->toLong() != 0)
+        if (pos == d->exifMetadata().end() || pos->count() != 1 || pos->toUint32() != 0)
         {
-            throw Exiv2::Error(Exiv2::kerErrorMessage, "Exif.Image.NewSubfileType missing or not set as main image");
+            throw Exiv2::Error(Exiv2::ErrorCode::kerErrorMessage, "Exif.Image.NewSubfileType missing or not set as main image");
         }
 
         // Remove sub-IFD tags
@@ -1011,9 +1012,9 @@
             Exiv2::DataBuf buf((Exiv2::byte *)data.data(), data.size());
             Exiv2::ULongValue val;
             val.read("0");
-            val.setDataArea(buf.pData_, buf.size_);
+            val.setDataArea(buf.c_data(), buf.size());
             d->exifMetadata()["Exif.SubImage1.JPEGInterchangeFormat"]       = val;
-            d->exifMetadata()["Exif.SubImage1.JPEGInterchangeFormatLength"] = uint32_t(buf.size_);
+            d->exifMetadata()["Exif.SubImage1.JPEGInterchangeFormatLength"] = uint32_t(buf.size());
             d->exifMetadata()["Exif.SubImage1.Compression"]                 = uint16_t(6); // JPEG (old-style)
             d->exifMetadata()["Exif.SubImage1.NewSubfileType"]              = uint32_t(1); // Thumbnail image
             return true;

--- libkexiv2/kexiv2gps.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2gps.cpp	2026-02-04 13:07:51.000000000 +0800
@@ -340,7 +340,7 @@
         // Do all the easy constant ones first.
         // GPSVersionID tag: standard says is should be four bytes: 02 00 00 00
         // (and, must be present).
-        Exiv2::Value::AutoPtr value = Exiv2::Value::create(Exiv2::unsignedByte);
+        Exiv2::Value::UniquePtr value = Exiv2::Value::create(Exiv2::unsignedByte);
         value->read("2 0 0 0");
         d->exifMetadata().add(Exiv2::ExifKey("Exif.GPSInfo.GPSVersionID"), value.get());
 
@@ -395,7 +395,7 @@
         if (altitude)
         {
             // Altitude reference: byte "00" meaning "above sea level", "01" mening "behing sea level".
-            Exiv2::Value::AutoPtr value = Exiv2::Value::create(Exiv2::unsignedByte);
+            Exiv2::Value::UniquePtr value = Exiv2::Value::create(Exiv2::unsignedByte);
 
             if ((*altitude) >= 0) value->read("0");
             else               value->read("1");

--- libkexiv2/kexiv2xmp.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2xmp.cpp	2026-02-04 13:13:02.000000000 +0800
@@ -36,7 +36,7 @@
 #ifdef _XMP_SUPPORT_
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((const char*)
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((const char*)
                                       (QFile::encodeName(filePath)));
 
         Exiv2::AccessMode mode = image->checkMode(Exiv2::mdXmp);
@@ -46,7 +46,8 @@
     {
         std::string s(e.what());
         kError() << "Cannot check Xmp access mode using Exiv2 (Error #"
-                 << e.code() << ": " << s.c_str() << ")";
+                 << Exiv2::Error(e.code()).what()
+                 << ")";
     }
     catch(...)
     {
@@ -391,7 +392,7 @@
     try
     {
         const std::string &txt(value.toUtf8().constData());
-        Exiv2::Value::AutoPtr xmpTxtVal = Exiv2::Value::create(Exiv2::xmpText);
+        Exiv2::Value::UniquePtr xmpTxtVal = Exiv2::Value::create(Exiv2::xmpText);
         xmpTxtVal->read(txt);
         d->xmpMetadata()[xmpTagName].setValue(xmpTxtVal.get());
         return true;
@@ -530,7 +531,7 @@
 
         if (!values.isEmpty())
         {
-            Exiv2::Value::AutoPtr xmpTxtVal = Exiv2::Value::create(Exiv2::langAlt);
+            Exiv2::Value::UniquePtr xmpTxtVal = Exiv2::Value::create(Exiv2::langAlt);
 
             for (AltLangMap::const_iterator it = values.constBegin(); it != values.constEnd(); ++it)
             {
@@ -635,7 +636,7 @@
         QString txtLangAlt = QString("lang=%1 %2").arg(language).arg(value);
 
         const std::string &txt(txtLangAlt.toUtf8().constData());
-        Exiv2::Value::AutoPtr xmpTxtVal = Exiv2::Value::create(Exiv2::langAlt);
+        Exiv2::Value::UniquePtr xmpTxtVal = Exiv2::Value::create(Exiv2::langAlt);
 
         // Search if an Xmp tag already exist.
 
@@ -748,7 +749,7 @@
         else
         {
             const QStringList list = seq;
-            Exiv2::Value::AutoPtr xmpTxtSeq = Exiv2::Value::create(Exiv2::xmpSeq);
+            Exiv2::Value::UniquePtr xmpTxtSeq = Exiv2::Value::create(Exiv2::xmpSeq);
 
             for (QStringList::const_iterator it = list.constBegin(); it != list.constEnd(); ++it )
             {
@@ -847,7 +848,7 @@
         else
         {
             QStringList list = bag;
-            Exiv2::Value::AutoPtr xmpTxtBag = Exiv2::Value::create(Exiv2::xmpBag);
+            Exiv2::Value::UniquePtr xmpTxtBag = Exiv2::Value::create(Exiv2::xmpBag);
 
             for (QStringList::const_iterator it = list.constBegin(); it != list.constEnd(); ++it )
             {
@@ -939,7 +940,7 @@
                 case Exiv2::unsignedLong:
                 case Exiv2::signedShort:
                 case Exiv2::signedLong:
-                    return QVariant((int)it->toLong());
+                    return QVariant((int)it->toUint32());
                 case Exiv2::unsignedRational:
                 case Exiv2::signedRational:
                     if (rationalAsListOfInts)

--- libkexiv2/kexiv2iptc.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2iptc.cpp	2026-02-04 14:29:04.000000000 +0800
@@ -27,6 +27,7 @@
 
 #include "kexiv2.h"
 #include "kexiv2_p.h"
+#include <exiv2/photoshop.hpp>
 
 namespace KExiv2Iface
 {
@@ -35,7 +36,7 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((const char*)
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((const char*)
                                       (QFile::encodeName(filePath)));
 
         Exiv2::AccessMode mode = image->checkMode(Exiv2::mdIptc);
@@ -45,7 +46,8 @@
     {
         std::string s(e.what());
         kError() << "Cannot check Iptc access mode using Exiv2 (Error #"
-                 << e.code() << ": " << s.c_str() << ")";
+                 << Exiv2::Error(e.code()).what()
+                 << ")";
     }
     catch(...)
     {
@@ -90,14 +92,14 @@
 
             if (addIrbHeader)
             {
-                c2 = Exiv2::Photoshop::setIptcIrb(0, 0, iptc);
+                c2 = Exiv2::Photoshop::setIptcIrb(nullptr, 0, iptc);
             }
             else
             {
                 c2 = Exiv2::IptcParser::encode(d->iptcMetadata());
             }
 
-            QByteArray data((const char*)c2.pData_, c2.size_);
+            QByteArray data((const char*)c2.c_data(), c2.size());
             return data;
 
         }
@@ -530,7 +532,7 @@
             QString key = *it;
             key.truncate(maxSize);
 
-            Exiv2::Value::AutoPtr val = Exiv2::Value::create(Exiv2::string);
+            Exiv2::Value::UniquePtr val = Exiv2::Value::create(Exiv2::string);
             val->read(key.toUtf8().constData());
             iptcData.add(iptcTag, val.get());
         }
@@ -632,7 +634,7 @@
             QString key = *it;
             key.truncate(64);
 
-            Exiv2::Value::AutoPtr val = Exiv2::Value::create(Exiv2::string);
+            Exiv2::Value::UniquePtr val = Exiv2::Value::create(Exiv2::string);
             val->read(key.toUtf8().constData());
             iptcData.add(iptcTag, val.get());
         }
@@ -726,7 +728,7 @@
             QString key = *it;
             key.truncate(236);
 
-            Exiv2::Value::AutoPtr val = Exiv2::Value::create(Exiv2::string);
+            Exiv2::Value::UniquePtr val = Exiv2::Value::create(Exiv2::string);
             val->read(key.toUtf8().constData());
             iptcData.add(iptcTag, val.get());
         }
@@ -821,7 +823,7 @@
             QString key = *it;
             key.truncate(32);
 
-            Exiv2::Value::AutoPtr val = Exiv2::Value::create(Exiv2::string);
+            Exiv2::Value::UniquePtr val = Exiv2::Value::create(Exiv2::string);
             val->read(key.toUtf8().constData());
             iptcData.add(iptcTag, val.get());
         }

--- libkexiv2/kexiv2image.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2image.cpp	2026-02-04 13:19:43.000000000 +0800
@@ -106,12 +106,12 @@
         Exiv2::ExifKey key("Exif.Photo.PixelXDimension");
         Exiv2::ExifData::iterator it = exifData.findKey(key);
         if (it != exifData.end() && it->count())
-            width = it->toLong();
+            width = it->toUint32();
 
         Exiv2::ExifKey key2("Exif.Photo.PixelYDimension");
         Exiv2::ExifData::iterator it2 = exifData.findKey(key2);
         if (it2 != exifData.end() && it2->count())
-            height = it2->toLong();
+            height = it2->toUint32();
 
         if (width != -1 && height != -1)
             return QSize(width, height);
@@ -124,12 +124,12 @@
         Exiv2::ExifKey key3("Exif.Image.ImageWidth");
         Exiv2::ExifData::iterator it3 = exifData.findKey(key3);
         if (it3 != exifData.end() && it3->count())
-            width = it3->toLong();
+            width = it3->toUint32();
 
         Exiv2::ExifKey key4("Exif.Image.ImageLength");
         Exiv2::ExifData::iterator it4 = exifData.findKey(key4);
         if (it4 != exifData.end() && it4->count())
-            height = it4->toLong();
+            height = it4->toUint32();
 
         if (width != -1 && height != -1)
             return QSize(width, height);
@@ -264,7 +264,7 @@
 
         if (it != exifData.end() && it->count())
         {
-            orientation = it->toLong();
+            orientation = it->toUint32();
             kDebug() << "Orientation => Exif.MinoltaCs7D.Rotation => " << (int)orientation;
             switch(orientation)
             {
@@ -283,7 +283,7 @@
 
         if (it != exifData.end() && it->count())
         {
-            orientation = it->toLong();
+            orientation = it->toUint32();
             kDebug() << "Orientation => Exif.MinoltaCs5D.Rotation => " << (int)orientation;
             switch(orientation)
             {
@@ -304,7 +304,7 @@
 
         if (it != exifData.end() && it->count())
         {
-            orientation = it->toLong();
+            orientation = it->toUint32();
             kDebug() << "Orientation => Exif.Image.Orientation => " << (int)orientation;
             return (ImageOrientation)orientation;
         }
@@ -377,7 +377,7 @@
         it = d->exifMetadata().findKey(thumbKey);
         if (it != d->exifMetadata().end() && it->count())
         {
-            RotationMatrix operation((KExiv2Iface::KExiv2::ImageOrientation)it->toLong());
+            RotationMatrix operation((KExiv2Iface::KExiv2::ImageOrientation)it->toUint32());
             operation *= orientation;
             (*it) = static_cast<uint16_t>(operation.exifOrientation());
         }
@@ -778,7 +778,7 @@
 
         const std::string &xmpdatetime(dateTime.toString(Qt::ISODate).toAscii().constData());
 
-        Exiv2::Value::AutoPtr xmpTxtVal = Exiv2::Value::create(Exiv2::xmpText);
+        Exiv2::Value::UniquePtr xmpTxtVal = Exiv2::Value::create(Exiv2::xmpText);
         xmpTxtVal->read(xmpdatetime);
         d->xmpMetadata().add(Exiv2::XmpKey("Xmp.exif.DateTimeOriginal"), xmpTxtVal.get());
         d->xmpMetadata().add(Exiv2::XmpKey("Xmp.photoshop.DateCreated"), xmpTxtVal.get());

--- libkexiv2/kexiv2previews.cpp.orig	2014-10-07 14:07:22.000000000 +0800
+++ libkexiv2/kexiv2previews.cpp	2026-02-04 14:33:53.000000000 +0800
@@ -49,9 +49,9 @@
         delete manager;
     }
 
-    void load(Exiv2::Image::AutoPtr image_)
+    void load(Exiv2::Image::UniquePtr image_)
     {
-        image                              = image_;
+        image                              = std::move(image_);
 
         image->readMetadata();
 
@@ -69,7 +69,7 @@
 
 public:
 
-    Exiv2::Image::AutoPtr           image;
+    Exiv2::Image::UniquePtr         image;
     Exiv2::PreviewManager*          manager;
     QList<Exiv2::PreviewProperties> properties;
 };
@@ -79,8 +79,8 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((const char*)(QFile::encodeName(filePath)));
-        d->load(image);
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((const char*)(QFile::encodeName(filePath)));
+        d->load(std::move(image));
     }
     catch( Exiv2::Error& e )
     {
@@ -97,8 +97,8 @@
 {
     try
     {
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open((Exiv2::byte*)imgData.data(), imgData.size());
-        d->load(image);
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open((Exiv2::byte*)imgData.data(), imgData.size());
+        d->load(std::move(image));
     }
     catch( Exiv2::Error& e )
     {
