From f7877fc8e643b8f1d04b590738304c784bfb01b3 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 26 Dec 2023 17:25:15 +0800
Subject: [PATCH 1/6] GNUmakefile for ppc

---
 GNUmakefile | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git GNUmakefile GNUmakefile
index e72005222b..f30c5a7545 100644
--- GNUmakefile
+++ GNUmakefile
@@ -1,6 +1,5 @@
 ifdef CONFIG
 	VERSION = 0.99
-	GIT_LABEL = $(shell echo `git describe --all`-`git rev-parse HEAD`)
 	BUNDLE = Factor.app
 	DEBUG ?= 0
 	REPRODUCIBLE ?= 0
@@ -13,16 +12,12 @@
 		CXX=$(SHELL_CXX)
 	endif
 
-	XCODE_PATH ?= /Applications/Xcode.app
-	MACOSX_32_SDK ?= MacOSX10.11.sdk
-
 	include $(CONFIG)
 
 	CFLAGS += -Wall \
 		-Wextra \
 		-pedantic \
 		-DFACTOR_VERSION="$(VERSION)" \
-		-DFACTOR_GIT_LABEL="$(GIT_LABEL)" \
 		$(SITE_CFLAGS)
 
 	CXXFLAGS += -std=c++11
@@ -169,6 +164,8 @@
 	@echo "macosx-x86-64"
 	@echo "macosx-x86-fat"
 	@echo "macosx-arm64"
+	@echo "macosx-ppc-32"
+	@echo "macosx-ppc-64"
 	@echo "windows-x86-32"
 	@echo "windows-x86-64"
 	@echo ""
@@ -187,6 +184,12 @@
 freebsd-x86-64:
 	$(MAKE) $(ALL) CONFIG=vm/Config.freebsd.x86.64
 
+macosx-ppc-32:
+	$(MAKE) $(ALL) macosx.app CONFIG=vm/Config.macosx.ppc.32
+
+macosx-ppc-64:
+	$(MAKE) $(ALL) macosx.app CONFIG=vm/Config.macosx.ppc.64
+
 macosx-x86-32:
 	$(MAKE) $(ALL) macosx.app CONFIG=vm/Config.macosx.x86.32
 
