From b224b4f2bfe586aed4a144ca932e157eaf9fb247 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 26 Dec 2023 18:06:05 +0800
Subject: [PATCH 5/6] master.hpp: fix ppc64 defines

---
 vm/master.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git vm/master.hpp vm/master.hpp
index 653b9fda9c..88ccc0ff53 100644
--- vm/master.hpp
+++ vm/master.hpp
@@ -73,7 +73,7 @@
 #elif defined(i386) || defined(__i386) || defined(__i386__) || defined(_M_IX86)
 #define FACTOR_X86
 #elif(defined(__POWERPC__) || defined(__ppc__) || defined(_ARCH_PPC)) && \
-    (defined(__PPC64__) || defined(__64BIT__))
+    (defined(__PPC64__) || defined(__ppc64__) || defined(__64BIT__))
 #define FACTOR_PPC64
 #define FACTOR_PPC
 #define FACTOR_64
-- 
2.43.0

