From 05ab21cc956230b3ad3fb4544651d646ff64d980 Mon Sep 17 00:00:00 2001
From: barracuda156 <vital.had@gmail.com>
Date: Tue, 26 Dec 2023 18:10:26 +0800
Subject: [PATCH 6/6] platform.hpp: add powerpc-darwin targets

---
 vm/platform.hpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git vm/platform.hpp vm/platform.hpp
index c3ae476345..54f94ce79e 100644
--- vm/platform.hpp
+++ vm/platform.hpp
@@ -24,6 +24,10 @@
       #include "os-macosx-x86.64.hpp"
     #elif defined(FACTOR_ARM64)
       #include "os-macosx-arm64.hpp"
+    #elif defined(FACTOR_PPC64)
+      #include "os-macosx-ppc.64.hpp"
+    #elif defined(FACTOR_PPC32)
+      #include "os-macosx-ppc.32.hpp"
     #else
       #error "Unsupported Mac OS X flavor"
     #endif
-- 
2.43.0

