--- source/CMakeLists.txt.orig	2026-08-29 17:18:15.000000000 -0600
+++ source/CMakeLists.txt	2026-08-29 17:30:13.000000000 -0600
@@ -35,8 +35,8 @@
 include_directories("decore")
 set(UAVS3D_ASM_FILES "")
 
-if("${UAVS3D_TARGET_CPU}" MATCHES "x86" OR
-   "${UAVS3D_TARGET_CPU}" MATCHES "x86_64")
+# Require x86_64 for SSE4.2/AVX2. Should go more in depth, but better.
+if("${UAVS3D_TARGET_CPU}" MATCHES "x86_64")
   aux_source_directory(./decore/sse DIR_X86_SRC)
   aux_source_directory(./decore/avx2 DIR_X86_256_SRC)
   set_source_files_properties(${DIR_X86_SRC} PROPERTIES COMPILE_FLAGS "${CMAKE_C_FLAGS} -msse4.2")
