--- Makefile.in	2019-11-15 18:09:13.000000000 +0800
+++ Makefile.in	2025-01-01 22:04:02.000000000 +0800
@@ -75,10 +75,10 @@
 distclean: dist-clean
 
 .c.lo:
-	$(SILENTCCLT) $(LIBTOOL) $(SILENCELT) --mode=compile --tag=CC $(CC) -shared $(CFLAGS) $(CONFIGFLAGS) $(GETTEXTFLAGS) -Isrc -DT3_CONFIG_BUILD_DSO -c -o $@ $<
+	$(SILENTCCLT) $(LIBTOOL) --mode=compile --tag=CC $(CC) -shared $(CFLAGS) $(CONFIGFLAGS) $(GETTEXTFLAGS) -Isrc -DT3_CONFIG_BUILD_DSO -c -o $@ $<
 
 src/libt3config.la: $(OBJECTS)
-	$(SILENTLDLT) $(LIBTOOL) $(SILENCELT) --mode=link --tag=CC $(CC) -shared -version-info 0:1:0 $(CFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) $(LDLIBS) $(GETTEXTLIBS) -rpath $(libdir)
+	$(SILENTLDLT) $(LIBTOOL) --mode=link --tag=CC $(CC) -shared -version-info 0:1:0 $(CFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) $(LDLIBS) $(GETTEXTLIBS) -rpath $(libdir)
 
 # Macros to make DESTDIR support more readable
 _libdir=$(DESTDIR)$(libdir)
