From 2f15a66135b19e6805b722898d2dbba7e94286b3 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 16 Nov 2025 14:28:59 +0800
Subject: [PATCH] Use lower-case module name


diff --git AnimeSnap/__main__.py AnimeSnap/__main__.py
index f4a79a3..9a8d231 100644
--- AnimeSnap/__main__.py
+++ AnimeSnap/__main__.py
@@ -9,8 +9,8 @@ except ImportError:
         """Dummy install function if rich is not available."""
         pass
 
-from AnimeSnap.cli import get_parsed_args
-from AnimeSnap.gui import QApplication, Window
+from animesnap.cli import get_parsed_args
+from animesnap.gui import QApplication, Window
 
 
 def main() -> None:
diff --git AnimeSnap/cli.py AnimeSnap/cli.py
index 78e7082..e47f3f6 100644
--- AnimeSnap/cli.py
+++ AnimeSnap/cli.py
@@ -6,9 +6,9 @@ from argparse import Namespace
 if sys.version_info > (3, 5):
     from core_helpers.cli import ArgparseColorThemes, setup_parser
 
-from AnimeSnap.consts import PACKAGE
-from AnimeSnap.consts import __desc__ as DESC
-from AnimeSnap.consts import __version__ as VERSION
+from animesnap.consts import PACKAGE
+from animesnap.consts import __desc__ as DESC
+from animesnap.consts import __version__ as VERSION
 
 
 def get_parsed_args() -> Namespace:
diff --git AnimeSnap/consts.py AnimeSnap/consts.py
index 25f4672..dae6aab 100644
--- AnimeSnap/consts.py
+++ AnimeSnap/consts.py
@@ -28,8 +28,8 @@ if sys.version_info > (3, 5):
 else:
     __version__ = "0.1.0"
     __desc__ = "Unlock a world of anime information from a single scene screenshot including episode, timestamps, and much more!"
-    PACKAGE = "AnimeSnap"
-    GITHUB = "https://github.com/YisusChrist/AnimeSnap"
+    PACKAGE = "animesnap"
+    GITHUB = "https://github.com/YisusChrist/animesnap"
     AUTHOR = "Alejandro González Momblán"
 
     CONFIG_PATH = "."
diff --git AnimeSnap/gui.py AnimeSnap/gui.py
index e5157f2..e98de69 100644
--- AnimeSnap/gui.py
+++ AnimeSnap/gui.py
@@ -85,7 +85,7 @@
 
 from qdarkstyle import load_stylesheet  # type: ignore
 
-from AnimeSnap.consts import (
+from animesnap.consts import (
     AUTHOR,
     GITHUB,
     HEIGHT,
@@ -97,11 +97,11 @@
     X,
     Y,
 )
-from AnimeSnap.consts import __desc__ as DESC
-from AnimeSnap.consts import __version__ as VERSION
-from AnimeSnap.files import get_image_extensions
-from AnimeSnap.json_operations import json_to_tabular, save_to_json
-from AnimeSnap.search import search_anime
+from animesnap.consts import __desc__ as DESC
+from animesnap.consts import __version__ as VERSION
+from animesnap.files import get_image_extensions
+from animesnap.json_operations import json_to_tabular, save_to_json
+from animesnap.search import search_anime
 
 
 def get_open_file_name(
diff --git AnimeSnap/search.py AnimeSnap/search.py
index eee8590..8779f93 100644
--- AnimeSnap/search.py
+++ AnimeSnap/search.py
@@ -5,7 +5,7 @@ import urllib
 import requests  # type: ignore
 from validators import url as validate_url  # type: ignore
 
-from AnimeSnap.consts import API_URL, MAX_TIMEOUT
+from animesnap.consts import API_URL, MAX_TIMEOUT
 
 
 def search_anime(img_source: str, anilist_id) -> dict:
diff --git pyproject.toml pyproject.toml
index 3c25417..734242a 100644
--- pyproject.toml
+++ pyproject.toml
@@ -1,5 +1,5 @@
 [project]
-name = "AnimeSnap"
+name = "animesnap"
 version = "0.1.0"
 description = "Unlock a world of anime information from a single scene screenshot including episode, timestamps, and much more!"
 license = "MIT"
@@ -21,7 +21,7 @@ dependencies = [
 repository = "https://github.com/YisusChrist/AnimeSnap"
 
 [project.scripts]
-AnimeSnap = "AnimeSnap.__main__:main"
+animesnap = "animesnap.__main__:main"
 
 [project.optional-dependencies]
 qt5 = ["pyqt5", "PyQt5-Qt5==5.15.2"]
