diff --git a/configure.orig b/configure
index 39e7f03..3cb9721 100755
--- configure..in.orig
+++ configure.in
@@ -1938,7 +1938,7 @@ OS=`uname`
 OS="${OS}"

 MACHINE=$(uname -m)
-CPU="CPU_INTEL"
+CPU="CPU_ARM"
 if test "$MACHINE" = "armv5l"; then
   CPU="CPU_ARM"
 fi
diff --git a/libfind2.c.orig b/libfind2.c
index d5c8c32..2dc00b0 100644
--- libfind2.c.orig
+++ libfind2.c
@@ -192,12 +192,11 @@ if(argv[2][i  ] == '.' &&
   while(argv[2][j+1] != '/' && argv[2][j+1] !=0)j++;
   while(argv[2][k-1] != ' ' && argv[2][k-1] != 10 && k>0)k--;
 findmac:;
-  if(argv[2][i  ] == 'x' && 
-     argv[2][i+1] == '8' && 
-     argv[2][i+2] == '6' && 
-     argv[2][i+3] == '_' && 
-     argv[2][i+4] == '6' && 
-     argv[2][i+5] == '4')
+  if(argv[2][i  ] == 'p' && 
+     argv[2][i+1] == 'p' && 
+     argv[2][i+2] == 'c' && 
+     argv[2][i+3] == '6' && 
+     argv[2][i+4] == '4')
     {
     if(argv[1][0] == '0')
       {  
@@ -240,10 +239,9 @@ findmac:;
       return 0;
       }
     }
-  if(argv[2][i  ] == 'i' && 
-     argv[2][i+1] == '3' && 
-     argv[2][i+2] == '8' && 
-     argv[2][i+3] == '6')
+  if(argv[2][i  ] == 'p' && 
+     argv[2][i+1] == 'p' && 
+     argv[2][i+2] == 'c')
     {  
     if(argv[1][0] == '0')
       {  
