diff --git a/configure.in.orig b/configure.in
index a5092b1..fa4fa7a 100644
--- configure.in.orig
+++ configure.in
@@ -676,7 +676,7 @@ if test "$DARWIN" = "-DDARWIN=0"; then
     ST="-s"
   fi
 else
-  ST="-Wl,-no_pie"
+  ST="-Wl"
 fi
 AC_SUBST(ST)
 # ********************************************************************
diff --git a/lxsys.c.orig b/lxsys.c
index 1ab89ba..5344b07 100644
--- lxsys.c.orig
+++ lxsys.c
@@ -336,7 +336,7 @@ thread_pid[no]=no;
 thread_pid[no]=syscall(SYS_gettid);
 #else
 uint64_t thread_id;
-pthread_threadid_np(NULL, &thread_id);
+thread_id = pthread_mach_thread_np(pthread_self());
 thread_pid[no]=(int)thread_id;
 #endif
 #endif
