--- makefiles/Makefile.tools	2023-12-25 07:06:18.000000000 +0800
+++ makefiles/Makefile.tools	2025-12-31 02:46:11.000000000 +0800
@@ -62,6 +62,12 @@
 ###############################################################################
 
 # Autodetect host
+
+# On ppc64 arch detection is broken with the system gcc, so assign these:
+HOST  := powerpc64-apple-darwin9
+BUILD := powerpc64-apple-darwin9
+ARCHFLAG = -arch ppc64
+
 ifeq ($(HOST),)
   HOST := $(shell $(CC) -dumpmachine)
   ifeq ($(HOST),)
@@ -113,9 +119,9 @@
 
   ifeq ($(findstring arwin,$(HOST)),arwin)
     # Building on+for Mac OS X (Darwin) with MAC ports
-    CFLAGS := $(CFLAGS) -I/opt/local/include
-    CXXFLAGS := $(CXXFLAGS) -I/opt/local/include
-    LDFLAGS := $(LDFLAGS) -L/opt/local/lib
+    CFLAGS := $(CFLAGS) $(ARCHFLAG) -I/opt/local/include
+    CXXFLAGS := $(CXXFLAGS) $(ARCHFLAG) -I/opt/local/include
+    LDFLAGS := $(LDFLAGS) $(ARCHFLAG) -L/opt/local/lib
   endif
 
 else
