From 896fc62631c86ab9ec22ace3da544debb1e20366 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 18 Aug 2024 12:30:20 +0800
Subject: [PATCH 1/6] arch.cpp: fix powerpc macros

---
 cmake/arch.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/arch.cpp b/cmake/arch.cpp
index 887fb9f4b..b9c08759a 100644
--- a/cmake/arch.cpp
+++ b/cmake/arch.cpp
@@ -13,9 +13,9 @@ int main() {
     printf("MIPS64");
 #elif defined(__mips__)
     printf("MIPS");
-#elif defined(__powerpc64__)
+#elif defined(__powerpc64__) || defined(__ppc64__)
     printf("POWERPC64");
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__ppc__)
     printf("POWERPC");
 #elif defined(__s390__)
     printf("S390");
-- 
2.49.0

