From 0ed968d9b5215a10c2f622d18cc34b735e31c706 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 18 Aug 2024 12:29:24 +0800
Subject: [PATCH 2/6] CMakeLists: add Apple

---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfcb01385..03146daac 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -63,6 +63,8 @@ if(CMAKE_SYSTEM_NAME MATCHES "(Linux|GNU)")
     katie_definition(-D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE)
 elseif(CMAKE_SYSTEM_NAME MATCHES "(FreeBSD|DragonFly)")
     katie_definition(-D_THREAD_SAFE)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    katie_definition(-D_DARWIN_C_SOURCE)
 elseif(NOT CMAKE_SYSTEM_NAME MATCHES "(OpenBSD|NetBSD|Solaris|SunOS)")
     # if it is not open-source it is not supported, don't even bother requesting support for such
     message(FATAL_ERROR "Unknown platform '${CMAKE_SYSTEM_NAME}'")
@@ -276,6 +278,8 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux")
     set(KATIE_OS "LINUX")
 elseif(CMAKE_SYSTEM_NAME MATCHES "GNU")
     set(KATIE_OS "HURD")
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+    set(KATIE_OS "MAC")
 elseif(CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
     set(KATIE_OS "FREEBSD")
 elseif(CMAKE_SYSTEM_NAME MATCHES "DragonFly")
-- 
2.49.0

