From d9d15576c5bc417ca6cbfe908f3b3574c902a61f Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 2 May 2026 21:24:21 +0800
Subject: [PATCH] common.h: define __STDC_FORMAT_MACROS prior to including
 inttypes.h

---
 src/libnetdata/common.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/libnetdata/common.h src/libnetdata/common.h
index 10ec398ad..634929bec 100644
--- src/libnetdata/common.h
+++ src/libnetdata/common.h
@@ -247,6 +247,9 @@ typedef uint32_t uid_t;
 #include <float.h>
 
 #if defined(HAVE_INTTYPES_H)
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <inttypes.h>
 #elif defined(HAVE_STDINT_H)
 #include <stdint.h>
