--- meson.build	2021-08-25 06:40:50.000000000 +0800
+++ meson.build	2025-05-10 00:36:07.000000000 +0800
@@ -192,7 +192,6 @@
 install_subdir('haskell', install_dir: 'lib/bali-phy')
 install_subdir('bindings', install_dir: 'lib/bali-phy')
 install_subdir('help', install_dir: 'lib/bali-phy')
-install_subdir('examples', install_dir: get_option('datadir')/'doc/bali-phy')
 
 # Print a summary of the configuration
 
