--- src/util/include/util/range.H	2021-08-25 06:40:50.000000000 +0800
+++ src/util/include/util/range.H	2025-05-10 00:02:33.000000000 +0800
@@ -1,6 +1,7 @@
 #ifndef UTIL_RANGE_H
 #define UTIL_RANGE_H
 
+#include <algorithm>
 #include <vector>
 #include <functional>
 
--- src/util/include/util/set.H	2021-08-25 06:40:50.000000000 +0800
+++ src/util/include/util/set.H	2025-05-10 00:09:05.000000000 +0800
@@ -1,6 +1,7 @@
 #ifndef UTIL_SET_H
 #define UTIL_SET_H
 
+#include <algorithm>
 #include <vector>
 #include <set>
 #include <map>
