--- xVideoServiceThief.pro.orig	2013-03-02 07:40:57.000000000 +0800
+++ xVideoServiceThief.pro	2025-06-21 15:26:31.000000000 +0800
@@ -198,10 +198,6 @@
 	HEADERS += \
 		src/webkit_mac/WebKitClass.h \
 		src/mac_only/mac_tools.h
-    CONFIG(release, debug|release) { 
-        message(Release build! Archs: 32 and 64 bits)
-        CONFIG += x86 x86_64
-    }
     CONFIG(debug, debug|release):message(Debug build! Archs: Current architecture only)
 }
 win32 { 
