--- common/cmake_aux/NMFLIB_FindARMA.cmake	2023-06-22 11:40:07.000000000 +0800
+++ common/cmake_aux/NMFLIB_FindARMA.cmake	2025-04-21 23:57:49.000000000 +0800
@@ -25,10 +25,10 @@
   add_definitions(-D__WITH__BARRIER__TIMING__=1)
 endif()
 
-#C++11 standard
-set (CMAKE_CXX_STANDARD 11)
+#C++14 standard
+set (CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
-#set (CMAKE_CXX_EXTENSIONS OFF) # use -std=c++11 instead of -std=gnu++11
+#set (CMAKE_CXX_EXTENSIONS OFF) # use -std=c++14 instead of -std=gnu++14
 
 #while exception get the stack trace
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -rdynamic -g3 -O0 ")
