--- common/utils.hpp	2023-06-22 11:40:07.000000000 +0800
+++ common/utils.hpp	2025-04-22 00:18:49.000000000 +0800
@@ -19,7 +19,7 @@
 #include <cblas.h>
 #endif
 
-static ULONG powersof10[16] = {1,
+static ULONGLONG powersof10[16] = {1,
                                10,
                                100,
                                1000,

--- common/utils.h	2023-06-22 11:40:07.000000000 +0800
+++ common/utils.h	2025-04-22 00:15:02.000000000 +0800
@@ -76,6 +76,7 @@
 typedef unsigned int UINT;
 typedef unsigned int uint;
 typedef unsigned long ULONG;
+typedef unsigned long long ULONGLONG;
 
 void absmat(const FMAT *X);
 
