--- common/utils.hpp	2023-06-22 11:40:07.000000000 +0800
+++ common/utils.hpp	2025-04-21 23:46:24.000000000 +0800
@@ -16,7 +16,7 @@
 #ifdef MKL_FOUND
 #include <mkl.h>
 #else
-#include <cblas.h>
+#include <cblas_openblas.h>
 #endif
 
 static ULONG powersof10[16] = {1,

--- common/tensor.hpp	2023-06-22 11:40:07.000000000 +0800
+++ common/tensor.hpp	2025-04-21 23:55:25.000000000 +0800
@@ -6,7 +6,7 @@
 #ifdef MKL_FOUND
 #include <mkl.h>
 #else
-#include <cblas.h>
+#include <cblas_openblas.h>
 #endif
 #include <armadillo>
 #include <fstream>

--- ntf/auntf.hpp	2023-06-22 11:40:07.000000000 +0800
+++ ntf/auntf.hpp	2025-04-22 00:04:43.000000000 +0800
@@ -9,7 +9,7 @@
 #ifdef MKL_FOUND
 #include <mkl.h>
 #else
-#include <cblas.h>
+#include <cblas_openblas.h>
 #endif
 
 #include <armadillo>

--- dimtree/ddttensor.h	2023-06-22 11:40:07.000000000 +0800
+++ dimtree/ddttensor.h	2025-04-22 00:06:02.000000000 +0800
@@ -6,7 +6,7 @@
 #ifdef MKL_FOUND
 #include <mkl.h>
 #else
-#include <cblas.h>
+#include <cblas_openblas.h>
 #endif
 #include <math.h>
 #include <omp.h>
