--- configure	2011-11-06 21:07:12.000000000 +0800
+++ configure	2026-06-06 16:52:19.000000000 +0800
@@ -2316,7 +2315,6 @@
 
 
 BUILD_LIBS='libH5Part'
-BUILD_TESTS='test'
 
 ###############################################################################
 ################# --enable-xxx and --with-xxx Argument ########################
@@ -4078,7 +4076,6 @@
 echo "${ECHO_T}yes" >&6; }
 	LIB_FORTRAN="libH5PartF.la"
         BUILD_LIBS="$BUILD_LIBS libH5PartF"
-	BUILD_TESTS="$BUILD_TESTS testf"
 
 	if test "X$USE_PARALLEL" = "Xyes"; then
 		ac_ext=${ac_fc_srcext-f}
