--- libs/locale/src/shared/mo_lambda.cpp
+++ libs/locale/src/shared/mo_lambda.cpp
@@ -211,9 +211,9 @@
             size_t pos;
             int next_tocken;
             int int_value;
             bool is_blank(char c) { return c == ' ' || c == '\r' || c == '\n' || c == '\t'; }
-            bool isdigit(char c) { return '0' <= c && c <= '9'; }
+            bool is_digit(char c) { return '0' <= c && c <= '9'; }
             void step()
             {
                 while(text[pos] && is_blank(text[pos]))
                     pos++;
@@ -245,9 +245,9 @@
                     next_tocken = NEQ;
                 } else if(*ptr == 'n') {
                     pos++;
                     next_tocken = VARIABLE;
-                } else if(isdigit(*ptr)) {
+                } else if(is_digit(*ptr)) {
                     int_value = strtol(text + pos, &tmp_ptr, 0);
                     pos = tmp_ptr - text;
                     next_tocken = NUM;
                 } else if(*ptr == '\0') {
