--- src/shared/CMakeLists.txt
+++ src/shared/CMakeLists.txt	2026-04-26 17:03:32.057361362 +0000
@@ -3,11 +3,19 @@
 # The SHL subsystem contains several small code pieces used all over libtsm and
 # other applications.
 #
-add_library(shl OBJECT
+
+set(SHL_SOURCES
     shl-htable.c
-    shl-pty.c
     shl-ring.c
 )
+# shl-pty uses Linux-specific epoll, exclude on macOS
+if(NOT APPLE)
+    list(APPEND SHL_SOURCES shl-pty.c)
+endif()
+
+add_library(shl OBJECT
+    ${SHL_SOURCES}
+)
 
 target_include_directories(shl
     PUBLIC
