From 954345de7c216e2271ef36c1092d2a0214c9cd5b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 12 May 2025 17:42:43 +0800
Subject: [PATCH 1/4] btop_shared.hpp: fix headers for Apple

---
 src/btop_shared.hpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git src/btop_shared.hpp src/btop_shared.hpp
index ca8cdcb..8c3a874 100644
--- src/btop_shared.hpp
+++ src/btop_shared.hpp
@@ -32,6 +32,9 @@ tab-size = 4
 
 // From `man 3 getifaddrs`: <net/if.h> must be included before <ifaddrs.h>
 // clang-format off
+#if defined(__APPLE__)
+# include <sys/socket.h>
+#endif
 #include <net/if.h>
 #include <ifaddrs.h>
 // clang-format on
-- 
2.48.0

