--- src/osx/btop_collect.cpp
+++ src/osx/btop_collect.cpp	2026-06-20 22:43:34.000000000 +0800
@@ -65,6 +65,10 @@
 #endif
 #include "smc.hpp"
 
+#ifndef kIOMainPortDefault
+#define kIOMainPortDefault kIOMasterPortDefault
+#endif
+
 using std::clamp, std::string_literals::operator""s, std::cmp_equal, std::cmp_less, std::cmp_greater;
 using std::ifstream, std::numeric_limits, std::streamsize, std::round, std::max, std::min;
 namespace fs = std::filesystem;
