--- Makefile	2026-05-02 00:05:43.000000000 +0800
+++ Makefile	2026-05-03 14:50:22.000000000 +0800
@@ -44,7 +44,7 @@
 	endif
 endif
 ifeq ($(PLATFORM_LC),macos)
-	GPU_SUPPORT := true
+#	GPU_SUPPORT := true
 endif
 ifneq ($(GPU_SUPPORT),true)
 	GPU_SUPPORT := false
@@ -118,7 +118,7 @@
 else ifeq ($(PLATFORM_LC),macos)
 	PLATFORM_DIR := osx
 	THREADS	:= $(shell sysctl -n hw.ncpu || echo 1)
-	override ADDFLAGS += -framework IOKit -framework CoreFoundation -lIOReport -Wno-format-truncation
+	override ADDFLAGS += -framework IOKit -framework CoreFoundation -Wno-format-truncation
 	SU_GROUP := wheel
 else ifeq ($(PLATFORM_LC),openbsd)
 	PLATFORM_DIR := openbsd
