From 00bf38799fd126114be1344a2efc45fe74db117e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Mon, 27 Oct 2025 13:33:07 +0800
Subject: [PATCH] CMakeLists: check if libatomic is needed

---
 CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6225cdda..fb5267b4 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -365,6 +365,19 @@ if(CMAKE_DL_LIBS)
     target_link_libraries(cjsh PRIVATE ${CMAKE_DL_LIBS})
 endif()
 
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles("
+    #include <atomic>
+    int main() {
+        std::atomic<uint64_t> w{0};
+        return ++w;
+    }
+" CJSH_BUILDS_WITHOUT_LIBATOMIC)
+
+if(NOT CJSH_BUILDS_WITHOUT_LIBATOMIC)
+    target_link_libraries(cjsh PRIVATE atomic)
+endif()
+
 if(CJSH_STRIP_BINARY)
     find_program(CJSH_STRIP_TOOL strip)
     if(CJSH_STRIP_TOOL)
