--- Makefile.options	2024-07-31 16:07:15.000000000 +0800
+++ Makefile.options	2025-06-20 23:17:30.000000000 +0800
@@ -16,27 +16,27 @@
 LINK = $(CC) $(CFLAGS) $(LDFLAGS) -o $@
 
 CFLAGS ?= -g -O2
-CFLAGS += -DD_DNS_THREADED -D_REENTRANT -DHAVE_CONFIG -D_THREAD_SAFE -Wall -W -Wno-unused-parameter -Waggregate-return -Wl,--no-as-needed
-INCLUDES ?= -I. -I.. -I/usr/local/include
-LDFLAGS ?= -L/usr/local/lib
+CFLAGS += -DD_DNS_THREADED -D_REENTRANT -DHAVE_CONFIG -D_THREAD_SAFE -Wall -W -Wno-unused-parameter -Waggregate-return
+INCLUDES ?= -I. -I.. -I@PREFIX@/include
+LDFLAGS ?= -L@PREFIX@/lib
 
 CXXCOMPILE = $(CXX) $(INCLUDES) $(CXXFLAGS)
 CXXFLAGS = $(CFLAGS) -fno-rtti -fno-exceptions
 
-LIBFLTK_CFLAGS = -I/usr/X11R6/include/freetype2 -pipe -I/usr/X11R6/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_THREAD_SAFE -D_REENTRANT
+LIBFLTK_CFLAGS = -I@PREFIX@/include/freetype2 -pipe -I@PREFIX@/include -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_THREAD_SAFE -D_REENTRANT
 LIBFLTK_CXXFLAGS = $(LIBFLTK_CFLAGS) -fvisibility-inlines-hidden
-LIBFLTK_LDFLAGS = $(LDFLAGS) -Wl,-rpath,/usr/local/lib -L/usr/X11R6/lib -lfltk -lXcursor -lXfixes -lXext -lXft -lfontconfig -lXinerama -lpthread -lm -lX11 -lXdmcp -lXau
+LIBFLTK_LDFLAGS = $(LDFLAGS) -Wl,-rpath,@PREFIX@/lib -L@PREFIX@/lib -lfltk -lXcursor -lXfixes -lXext -lXft -lfontconfig -lXinerama -lpthread -lm -lX11 -lXdmcp -lXau
 
-HTTPS_LDFLAGS = -lcrypto -lssl
+HTTPS_LDFLAGS = -L@SSLROOT@/lib -lcrypto -lssl
 
-LIBPNG16_CXXFLAGS = -I/usr/local/include/libpng16
+LIBPNG16_CXXFLAGS = -I@PREFIX@/include/libpng16
 
-DILLO_LDFLAGS = -ljpeg -L/usr/local/lib -lpng -Wl,-rpath,/usr/local/lib -L/usr/X11R6/lib -lfltk -lXcursor -lXfixes -lXext -lXft -lfontconfig -lXinerama -lpthread -lm -lX11 -lXdmcp -lXau -lz -lX11
+DILLO_LDFLAGS = -L@PREFIX@/lib -ljpeg -lpng -Wl,-rpath,@PREFIX@/lib -L@PREFIX@/lib -lfltk -lXcursor -lXfixes -lXext -lXft -lfontconfig -lXinerama -lpthread -lm -lX11 -lXdmcp -lXau -lz -liconv -lX11
 
 # == HTTPS CERTS ==
 
-CA_CERTS_FILE ?= /etc/ssl/cert.pem
-CA_CERTS_DIR ?= /etc/ssl/
+CA_CERTS_FILE ?= @PREFIX@/etc/openssl/cert.pem
+CA_CERTS_DIR ?= @PREFIX@/etc/openssl/
 
 # == EXTERNAL TOOLS ==
 
