--- poppler/Function.cc.orig	2026-01-07 06:13:29.000000000 +0800
+++ poppler/Function.cc	2026-02-02 00:39:53.000000000 +0800
@@ -34,6 +34,8 @@
 #include <cstring>
 #include <cctype>
 #include <cmath>
+#include <climits>
+
 #include "goo/gmem.h"
 #include "goo/gstrtod.h"
 #include "Object.h"

--- poppler/Gfx.cc.orig	2026-01-07 06:13:29.000000000 +0800
+++ poppler/Gfx.cc	2026-02-02 00:39:32.000000000 +0800
@@ -64,7 +64,9 @@
 #include <cstdio>
 #include <cstring>
 #include <cmath>
+#include <climits>
 #include <memory>
+
 #include "goo/GooTimer.h"
 #include "GlobalParams.h"
 #include "CharTypes.h"

--- poppler/Parser.cc.orig	2026-01-07 06:13:29.000000000 +0800
+++ poppler/Parser.cc	2026-02-02 00:39:15.000000000 +0800
@@ -33,6 +33,8 @@
 
 #include <config.h>
 
+#include <climits>
+
 #include "Object.h"
 #include "Array.h"
 #include "Dict.h"

--- poppler/SplashOutputDev.cc.orig	2026-01-07 06:13:29.000000000 +0800
+++ poppler/SplashOutputDev.cc	2026-02-02 00:38:59.000000000 +0800
@@ -55,7 +55,10 @@
 
 #include <cstring>
 #include <cmath>
+#include <climits>
+#include <algorithm>
 #include <vector>
+
 #include "Stream.h"
 #include "goo/gfile.h"
 #include "GlobalParams.h"
@@ -81,7 +84,6 @@
 #include "splash/SplashMath.h"
 #include "splash/Splash.h"
 #include "SplashOutputDev.h"
-#include <algorithm>
 
 static const double s_minLineWidth = 0.0;
 
