--- configure	2025-04-08 22:15:29.000000000 +0800
+++ configure	2025-04-12 17:43:31.000000000 +0800
@@ -7811,10 +7811,10 @@
           # Any changes made here should be reflected in the GCC+Darwin case below
           if test $Py_LTO_POLICY = default
           then
-            LTOFLAGS="-flto -Wl,-export_dynamic"
+            LTOFLAGS="-flto"
             LTOCFLAGS="-flto"
           else
-            LTOFLAGS="-flto=${Py_LTO_POLICY} -Wl,-export_dynamic"
+            LTOFLAGS="-flto=${Py_LTO_POLICY}"
             LTOCFLAGS="-flto=${Py_LTO_POLICY}"
           fi
           ;;
@@ -7843,7 +7843,7 @@
             LDFLAGS_NOLTO="-fno-lto"
       case $ac_sys_system in
         Darwin*)
-          LTOFLAGS="-flto -Wl,-export_dynamic"
+          LTOFLAGS="-flto"
           LTOCFLAGS="-flto"
           ;;
         *)
