--- configure.orig	2025-04-05 13:47:10.000000000 +0800
+++ configure	2025-04-05 13:55:41.000000000 +0800
@@ -18312,11 +18312,6 @@
         LUA_CFLAGS="$luajit_CFLAGS"
         LUA_LIBS="$luajit_LIBS"
 
-
-        if test "x$macos" != "xno"; then
-            LDFLAGS="${LDFLAGS} -pagezero_size 10000 -image_base 100000000"
-        fi
-
     else
         echo
         echo "   ERROR!  LuaJIT library not found. Go get it from http://www.luajit.org/ (or)"
