diff --git a/src/database.nim b/src/database.nim
index ad2f855..2c52365 100644
--- src/database.nim
+++ src/database.nim
@@ -1,4 +1,4 @@
-import db_sqlite
+import db_connector/db_sqlite
 import os
 import strutils
 
diff --git a/src/torrents.nim b/src/torrents.nim
index eb7d0bf..1a4398f 100644
--- src/torrents.nim
+++ src/torrents.nim
@@ -1,4 +1,4 @@
-import db_sqlite
+import db_connector/db_sqlite
 import strformat
 import times
 import strutils
diff --git a/torrentinim.nimble b/torrentinim.nimble
index 8f594ee..4e8b16c 100644
--- torrentinim.nimble
+++ torrentinim.nimble
@@ -14,4 +14,5 @@ bin           = @["torrentinim"]
 requires "nim >= 1.6.12"
 requires "nimquery >= 1.2.3"
 requires "filesize >= 1.0.0"
-requires "prologue >= 0.6.4"
\ No newline at end of file
+requires "prologue >= 0.6.4"
+requires "db_connector >= 0.1.0"
