From c7c6914ba283745c530c04c1590c4f98338f47c4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Apr 2025 04:04:50 +0800
Subject: [PATCH] dirctrlg.cpp: minor fix to macOS macro

---
 src/generic/dirctrlg.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/generic/dirctrlg.cpp b/src/generic/dirctrlg.cpp
index ae855a738b..2f41cb31df 100644
--- a/src/generic/dirctrlg.cpp
+++ b/src/generic/dirctrlg.cpp
@@ -97,7 +97,7 @@ size_t wxGetAvailableDrives(wxArrayString &paths, wxArrayString &names, wxArrayI
 {
 #if defined(wxHAS_FILESYSTEM_VOLUMES) || defined(__APPLE__)
 
-#if (defined(__WIN32__) || defined(__WXOSX__)) && wxUSE_FSVOLUME
+#if (defined(__WIN32__) || defined(__APPLE__)) && wxUSE_FSVOLUME
     // TODO: this code (using wxFSVolumeBase) should be used for all platforms
     //       but unfortunately wxFSVolumeBase is not implemented everywhere
     const wxArrayString as = wxFSVolumeBase::GetVolumes();
