From 6b7cef7a11ef14e5ac7fd232143b824bc8cbe79e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Apr 2025 03:46:47 +0800
Subject: [PATCH] defs.h: minor change to macOS macros

---
 include/wx/defs.h | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/include/wx/defs.h b/include/wx/defs.h
index 27804b8571..5800343481 100644
--- a/include/wx/defs.h
+++ b/include/wx/defs.h
@@ -2600,10 +2600,6 @@ typedef int (* LINKAGEMODE wxListIterateFunction)(void *current);
 #define DECLARE_WXOSX_OPAQUE_CFREF( name ) typedef struct __##name* name##Ref;
 #define DECLARE_WXOSX_OPAQUE_CONST_CFREF( name ) typedef const struct __##name* name##Ref;
 
-#endif
-
-#ifdef __WXMAC__
-
 #define WX_OPAQUE_TYPE( name ) struct wxOpaque##name
 
 typedef void*       WXHCURSOR;
@@ -2621,8 +2617,6 @@ typedef unsigned short  WXWORD;
 
 typedef WX_OPAQUE_TYPE(PicHandle ) * WXHMETAFILE ;
 
-typedef void*       WXDisplay;
-
 /*
  * core frameworks
  */
@@ -2699,7 +2693,7 @@ typedef HIShapeRef WXHRGN;
 
 #endif // __WXMAC__
 
-#if defined(__WXMAC__)
+#if defined(__WXMAC__) || defined(__APPLE__)
 
 /* Objective-C type declarations.
  * These are to be used in public headers in lieu of NSSomething* because
