From 54cbb0d5600afbe9c1a2ba5ecdca1ea19f99f9bf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Apr 2025 03:22:38 +0800
Subject: [PATCH] uilocale.cpp: include xlocale.h on macOS

---
 src/unix/uilocale.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/unix/uilocale.cpp b/src/unix/uilocale.cpp
index 36cedbf16d..fd5232f677 100644
--- a/src/unix/uilocale.cpp
+++ b/src/unix/uilocale.cpp
@@ -21,6 +21,10 @@
 #if wxUSE_INTL
 
 #include <locale.h>
+#ifdef __APPLE__
+    #include <xlocale.h>
+#endif
+
 #ifdef HAVE_LANGINFO_H
     #include <langinfo.h>
 #endif
