From a8343fa37cbb66a24210b1fa5312979febf4acfc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 4 Apr 2025 03:49:23 +0800
Subject: [PATCH] toolkit.cmake: fix condition for X11

---
 build/cmake/toolkit.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/cmake/toolkit.cmake b/build/cmake/toolkit.cmake
index 29c7531da64c..a2c696bd138b 100644
--- a/build/cmake/toolkit.cmake
+++ b/build/cmake/toolkit.cmake
@@ -119,7 +119,7 @@ endif()
 # support, but not for Wayland-only GTK (necessarily 3 or later), which is why
 # we have to do this after find_package(GTKx) above, as this is what sets
 # wxHAVE_GDK_X11.
-if(UNIX AND NOT APPLE AND NOT WIN32 AND (WXX11 OR WXGTK2 OR (WXGTK AND wxHAVE_GDK_X11)))
+if(UNIX AND NOT WIN32 AND (WXX11 OR WXGTK2 OR (WXGTK AND wxHAVE_GDK_X11)))
     find_package(X11 REQUIRED)
     list(APPEND wxTOOLKIT_INCLUDE_DIRS ${X11_INCLUDE_DIR})
     list(APPEND wxTOOLKIT_LIBRARIES ${X11_LIBRARIES})
