From 3dabf8fffbd82989c77184936ee630f6c4eb639d Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 22 Oct 2025 22:55:25 +0200
Subject: [PATCH 04/48] fixing warnings in __DARWIN__ builds

---
 include/wx/osx/core/private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/wx/osx/core/private.h b/include/wx/osx/core/private.h
index 9c22a5a922..b4f82ba4df 100644
--- a/include/wx/osx/core/private.h
+++ b/include/wx/osx/core/private.h
@@ -39,7 +39,7 @@
 #   endif
 #endif
 
-#if ( !wxUSE_GUI && !wxOSX_USE_IPHONE ) || wxOSX_USE_COCOA_OR_CARBON
+#if ( !wxUSE_GUI && !defined(__WXOSX_IPHONE__) ) || defined(__WXOSX_COCOA__)
 
 // Carbon functions are currently still used in wxOSX/Cocoa too (including
 // wxBase part of it).
-- 
2.24.3 (Apple Git-128)

