From 6d8a9483a61c22540e665fec65a28c525fa7bafa Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 12 Apr 2025 13:49:55 +0800
Subject: [PATCH] files.cmake: sort uilocale and stdpaths

---
 build/cmake/files.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/build/cmake/files.cmake b/build/cmake/files.cmake
index b53ac4a246..4a1e047291 100644
--- a/build/cmake/files.cmake
+++ b/build/cmake/files.cmake
@@ -34,6 +34,7 @@ set(BASE_UNIX_AND_DARWIN_HDR
 set(BASE_UNIX_AND_DARWIN_NOTWXMAC_SRC
     ${BASE_UNIX_AND_DARWIN_SRC}
     src/unix/mimetype.cpp
+    src/unix/uilocale.cpp
 )
 
 set(BASE_UNIX_AND_DARWIN_NOTWXMAC_HDR
@@ -46,7 +47,6 @@ set(BASE_UNIX_SRC
     src/unix/fswatcher_inotify.cpp
     src/unix/secretstore.cpp
     src/unix/stdpaths.cpp
-    src/unix/uilocale.cpp
 )
 
 set(BASE_UNIX_HDR
@@ -113,8 +113,8 @@ set(BASE_COREFOUNDATION_SRC
     src/osx/core/evtloop_cf.cpp
     src/osx/core/secretstore.cpp
     src/osx/core/strconv_cf.cpp
+    src/osx/cocoa/stdpaths.mm
     src/osx/cocoa/utils_base.mm
-    src/osx/core/uilocale.mm
 )
 
 set(BASE_COREFOUNDATION_HDR
@@ -137,8 +137,8 @@ set(BASE_COREFOUNDATION_HDR
 
 set(BASE_OSX_SHARED_SRC
     src/osx/core/mimetype.cpp
+    src/osx/core/uilocale.mm
     src/osx/fswatcher_fsevents.cpp
-    src/osx/cocoa/stdpaths.mm
     ${BASE_COREFOUNDATION_SRC}
     ${BASE_UNIX_AND_DARWIN_SRC}
 )
