From 3bb089af1c7b1622f0d42e8e8ebd84aa0ff0bf54 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 6 Apr 2025 03:30:43 +0800
Subject: [PATCH] utils_base.mm: fix for wxGTK build

---
 src/osx/cocoa/utils_base.mm | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/osx/cocoa/utils_base.mm b/src/osx/cocoa/utils_base.mm
index 126178e044..9b09259bd1 100644
--- a/src/osx/cocoa/utils_base.mm
+++ b/src/osx/cocoa/utils_base.mm
@@ -24,6 +24,10 @@
 #include "wx/osx/private.h"
 #include "wx/osx/private/available.h"
 
+#ifndef __WXOSX_IPHONE__
+#include <AppKit/AppKit.h>
+#endif
+
 #if wxUSE_SOCKETS
 // global pointer which lives in the base library, set from the net one (see
 // sockosx.cpp) and used from the GUI code (see utilsexc_cf.cpp) -- ugly but
