From c3141dfe3e065c6913f9c3655a649240375f408d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 12 Apr 2025 16:13:31 +0800
Subject: [PATCH] osx/core/private.h: fix parsing included CarbonCore header on
 powerpc

---
 include/wx/osx/core/private.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/wx/osx/core/private.h b/include/wx/osx/core/private.h
index a771611296..18f1d34d13 100644
--- a/include/wx/osx/core/private.h
+++ b/include/wx/osx/core/private.h
@@ -12,6 +12,12 @@
 #ifndef _WX_PRIVATE_CORE_H_
 #define _WX_PRIVATE_CORE_H_
 
+// Without this MachineExceptions.h of CarbonCore cannot be parsed
+// when C++11 or higher standard is used.
+#ifdef __APPLE_ALTIVEC__
+#undef __APPLE_ALTIVEC__
+#endif
+
 #include "wx/defs.h"
 
 #include <CoreFoundation/CoreFoundation.h>
