From 7c490262904591963d6158a403b8e8c295d8c9c9 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 22 Oct 2025 23:06:32 +0200
Subject: [PATCH 05/48] more precise macro name allows former order

---
 include/wx/defs.h | 21 ++++++++++-----------
 1 file changed, 10 insertions(+), 11 deletions(-)

diff --git a/include/wx/defs.h b/include/wx/defs.h
index 49d750e22c..73e89b5930 100644
--- a/include/wx/defs.h
+++ b/include/wx/defs.h
@@ -2756,16 +2756,7 @@ DECLARE_WXCOCOA_OBJC_CLASS(NSString);
 DECLARE_WXCOCOA_OBJC_CLASS(NSObject);
 DECLARE_WXCOCOA_OBJC_CLASS(NSSet);
 
-#if defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
-
-DECLARE_WXCOCOA_OBJC_CLASS(UIColor);
-DECLARE_WXCOCOA_OBJC_CLASS(UIImage);
-DECLARE_WXCOCOA_OBJC_CLASS(UIFont);
-
-typedef WX_UIColor WXColor;
-typedef WX_UIImage WXImage;
-
-#elif defined(TARGET_OS_MAC) && TARGET_OS_MAC
+#if defined(TARGET_OS_OSX) && TARGET_OS_OSX
 
 DECLARE_WXCOCOA_OBJC_CLASS(NSBitmapImageRep);
 DECLARE_WXCOCOA_OBJC_CLASS(NSColor);
@@ -2778,7 +2769,15 @@ DECLARE_WXCOCOA_OBJC_CLASS(NSThread);
 typedef WX_NSColor WXColor;
 typedef WX_NSImage WXImage;
 
-#elif
+#elif defined(TARGET_OS_IPHONE) && TARGET_OS_IPHONE
+
+DECLARE_WXCOCOA_OBJC_CLASS(UIColor);
+DECLARE_WXCOCOA_OBJC_CLASS(UIImage);
+DECLARE_WXCOCOA_OBJC_CLASS(UIFont);
+
+typedef WX_UIColor WXColor;
+typedef WX_UIImage WXImage;
+
 #else
 
 #endif
-- 
2.24.3 (Apple Git-128)

