From 80418be5082a5472a97baf251f7e09631d1d66e1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 6 Apr 2025 06:08:39 +0800
Subject: [PATCH] utilsunx: no memset_s before 10.9

---
 src/unix/utilsunx.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/unix/utilsunx.cpp b/src/unix/utilsunx.cpp
index f394059062..79996ef504 100644
--- a/src/unix/utilsunx.cpp
+++ b/src/unix/utilsunx.cpp
@@ -148,6 +148,7 @@
 
 #if defined(__DARWIN__)
     #include <sys/sysctl.h>
+    #include <AvailabilityMacros.h>
 #endif
 
 // ----------------------------------------------------------------------------
@@ -223,7 +224,8 @@ void wxSecureZeroMemory(void* v, size_t n)
     // but may be found in a non-standard header file, or in a library that is
     // not linked by default.
     explicit_bzero(v, n);
-#elif defined(__DARWIN__) || defined(__STDC_LIB_EXT1__)
+#elif (defined(__DARWIN__) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1090)) || \
+    defined(__STDC_LIB_EXT1__)
     // memset_s() is available since OS X 10.9, and may be available on
     // other platforms.
     memset_s(v, n, 0, n);
