From 38fa53c1063f5b42d0e6ba085acbda095a227dba Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Thu, 23 Oct 2025 23:51:30 +0200
Subject: [PATCH 07/14] fix warnings

---
 src/common/utilscmn.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/common/utilscmn.cpp b/src/common/utilscmn.cpp
index f75150161c..d8ab1bf7bf 100644
--- a/src/common/utilscmn.cpp
+++ b/src/common/utilscmn.cpp
@@ -509,7 +509,7 @@ wxString wxGetCurrentDir()
 // ----------------------------------------------------------------------------
 
 #ifdef __WXOSX__
-#if wxOSX_USE_COCOA_OR_CARBON
+#if defined(wxOSX_USE_COCOA_OR_CARBON) && wxOSX_USE_COCOA_OR_CARBON
     #include <crt_externs.h>
 #endif
 #endif
@@ -530,7 +530,7 @@ bool wxGetEnvMap(wxEnvVariableHashMap *map)
    // TODO : should we do something with logicals?
     char **env=nullptr;
 #elif defined(__DARWIN__)
-#if wxOSX_USE_COCOA_OR_CARBON
+#if defined(wxOSX_USE_COCOA_OR_CARBON) && wxOSX_USE_COCOA_OR_CARBON
     // Under Mac shared libraries don't have access to the global environ
     // variable so use this Mac-specific function instead as advised by
     // environ(7) under Darwin
@@ -1512,7 +1512,7 @@ wxWindowDisabler::wxWindowDisabler(bool disable)
     {
         DoDisable();
 
-#if defined(__WXOSX__) && wxOSX_USE_COCOA
+#if defined(__WXOSX__) && defined(wxOSX_USE_COCOA) && wxOSX_USE_COCOA
         AfterDisable(nullptr);
 #endif
     }
@@ -1529,7 +1529,7 @@ wxWindowDisabler::wxWindowDisabler(wxWindow *winToSkip, wxWindow *winToSkip2)
 
     DoDisable();
 
-#if defined(__WXOSX__) && wxOSX_USE_COCOA
+#if defined(__WXOSX__) && defined(wxOSX_USE_COCOA) && wxOSX_USE_COCOA
     AfterDisable(winToSkip);
 #endif
 }
@@ -1562,7 +1562,7 @@ wxWindowDisabler::~wxWindowDisabler()
     if ( !m_disabled )
         return;
 
-#if defined(__WXOSX__) && wxOSX_USE_COCOA
+#if defined(__WXOSX__) && defined(wxOSX_USE_COCOA) && wxOSX_USE_COCOA
     BeforeEnable();
 #endif
 
-- 
2.51.0

