From b9862a349a3c7d5d0122e155cdd1ee80b426a597 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 6 Apr 2025 06:14:42 +0800
Subject: [PATCH] utils_base.mm: fix for < 10.10

---
 src/osx/cocoa/utils_base.mm | 49 +++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/src/osx/cocoa/utils_base.mm b/src/osx/cocoa/utils_base.mm
index 126178e044..6b488e5f94 100644
--- a/src/osx/cocoa/utils_base.mm
+++ b/src/osx/cocoa/utils_base.mm
@@ -33,9 +33,15 @@ extern WXDLLIMPEXP_BASE wxSocketManager *wxOSXSocketManagerCF;
 wxSocketManager *wxOSXSocketManagerCF = nullptr;
 #endif // wxUSE_SOCKETS
 
+#if (defined(__APPLE__) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 101000) \
+    || (defined(__WXOSX_IPHONE__) && defined(__IPHONE_8_0))
+    #define wxHAS_NSPROCESSINFO 1
+#endif
+
 // our OS version is the same in non GUI and GUI cases
 wxOperatingSystemId wxGetOsVersion(int *verMaj, int *verMin, int *verMicro)
 {
+#if wxHAS_NSPROCESSINFO
     NSOperatingSystemVersion osVer = [NSProcessInfo processInfo].operatingSystemVersion;
 
     if ( verMaj != nullptr )
@@ -46,18 +52,42 @@ wxOperatingSystemId wxGetOsVersion(int *verMaj, int *verMin, int *verMicro)
 
     if ( verMicro != nullptr )
         *verMicro = osVer.patchVersion;
+#else
+    SInt32 maj, min, micro;
+
+    Gestalt(gestaltSystemVersionMajor, &maj);
+    Gestalt(gestaltSystemVersionMinor, &min);
+    Gestalt(gestaltSystemVersionBugFix, &micro);
 
+    if ( verMaj != NULL )
+        *verMaj = maj;
+
+    if ( verMin != NULL )
+        *verMin = min;
+
+    if ( verMicro != NULL )
+        *verMicro = micro;
+#endif
     return wxOS_MAC_OSX_DARWIN;
 }
 
 bool wxCheckOsVersion(int majorVsn, int minorVsn, int microVsn)
 {
+#if wxHAS_NSPROCESSINFO
     NSOperatingSystemVersion osVer;
     osVer.majorVersion = majorVsn;
     osVer.minorVersion = minorVsn;
     osVer.patchVersion = microVsn;
 
     return [[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:osVer] != NO;
+#else
+    int majorCur, minorCur, microCur;
+    wxGetOsVersion(&majorCur, &minorCur, &microCur);
+
+    return majorCur > majorVsn
+        || (majorCur == majorVsn && minorCur >= minorVsn)
+        || (majorCur == majorVsn && minorCur == minorVsn && microCur >= microVsn);
+#endif
 }
 
 wxString wxGetOsDescription()
@@ -77,6 +107,25 @@ wxString wxGetOsDescription()
     {
         switch (minorVer)
         {
+            case 5:
+                osName = "Leopard";
+                osBrand = "Mac OS X";
+                break;
+            case 6:
+                osName = "Snow Leopard";
+                osBrand = "Mac OS X";
+                break;
+            case 7:
+                osName = "Lion";
+                // 10.7 was the last version where the "Mac" prefix was used
+                osBrand = "Mac OS X";
+                break;
+            case 8:
+                osName = "Mountain Lion";
+                break;
+            case 9:
+                osName = "Mavericks";
+                break;
             case 10:
                 osName = "Yosemite";
                 break;
