From 07439079404edfd1f4fdee398e647f554ec598e1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 17 Feb 2025 14:16:01 +0800
Subject: [PATCH] uilocale

diff --git a/build/cmake/files.cmake b/build/cmake/files.cmake
index 0e9e841228..813727f4e3 100644
--- a/build/cmake/files.cmake
+++ b/build/cmake/files.cmake
@@ -36,6 +36,7 @@ set(BASE_UNIX_AND_DARWIN_NOTWXMAC_SRC
     ${BASE_UNIX_AND_DARWIN_SRC}
     src/unix/mimetype.cpp
     src/unix/stdpaths.cpp
+    src/unix/uilocale.cpp
 )
 
 set(BASE_UNIX_AND_DARWIN_NOTWXMAC_HDR
@@ -48,7 +49,6 @@ set(BASE_UNIX_SRC
     ${BASE_UNIX_AND_DARWIN_NOTWXMAC_SRC}
     src/unix/fswatcher_inotify.cpp
     src/unix/secretstore.cpp
-    src/unix/uilocale.cpp
 )
 
 set(BASE_UNIX_HDR
@@ -116,7 +116,6 @@ set(BASE_COREFOUNDATION_SRC
     src/osx/core/secretstore.cpp
     src/osx/core/strconv_cf.cpp
     src/osx/cocoa/utils_base.mm
-    src/osx/core/uilocale.mm
 )
 
 set(BASE_COREFOUNDATION_HDR
@@ -139,6 +138,7 @@ set(BASE_COREFOUNDATION_HDR
 
 set(BASE_OSX_SHARED_SRC
     src/osx/core/mimetype.cpp
+    src/osx/core/uilocale.mm
     src/osx/fswatcher_fsevents.cpp
     src/osx/cocoa/stdpaths.mm
     ${BASE_COREFOUNDATION_SRC}
