From 21daf2489d3fb99d08e16ee66d7afa6d25b3b74c Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Fri, 24 Oct 2025 17:40:49 +0200
Subject: [PATCH 13/14] correct condition

---
 include/wx/osx/private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/wx/osx/private.h b/include/wx/osx/private.h
index afc35adc61..113e244f14 100644
--- a/include/wx/osx/private.h
+++ b/include/wx/osx/private.h
@@ -7,7 +7,7 @@
     #include "wx/osx/iphone/private.h"
 #elif defined(__WXOSX_COCOA__)
     #include "wx/osx/cocoa/private.h"
-#elif defined(__WXOSX__)
+#elif wxUSE_GUI && defined(__WXOSX__)
     #error "Must include wx/defs.h first"
 #else
     #include <ApplicationServices/ApplicationServices.h>
-- 
2.51.0

