From e7b2fba7986ef971fd55cece7f9dba52a462e1cc Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Fri, 24 Oct 2025 18:21:17 +0200
Subject: [PATCH 14/14] guard code with wxUSE_GUI properly again

---
 include/wx/osx/core/private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/wx/osx/core/private.h b/include/wx/osx/core/private.h
index 05ac7878e6..57f3b0d917 100644
--- a/include/wx/osx/core/private.h
+++ b/include/wx/osx/core/private.h
@@ -66,7 +66,7 @@ WXDLLIMPEXP_BASE NSString* wxNSStringWithWxString(const wxString &wxstring);
 
 WXDLLIMPEXP_BASE CFURLRef wxOSXCreateURLFromFileSystemPath( const wxString& path);
 
-#ifdef __WXOSX__
+#if wxUSE_GUI && defined(__WXOSX__)
 
 #if !defined(__WXOSX_IPHONE__)
 #include <ApplicationServices/ApplicationServices.h>
-- 
2.51.0

