From 3240a19874994cafdd5421796ea2cc46d41f33af Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Mon, 27 Oct 2025 13:10:52 +0100
Subject: [PATCH 20/48] add missing frameworks

---
 build/cmake/lib/base/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/build/cmake/lib/base/CMakeLists.txt b/build/cmake/lib/base/CMakeLists.txt
index 526136812b..c98f732213 100644
--- a/build/cmake/lib/base/CMakeLists.txt
+++ b/build/cmake/lib/base/CMakeLists.txt
@@ -58,8 +58,11 @@ endif()
 
 if(APPLE)
     wx_lib_link_libraries(wxbase
+        PRIVATE
+        "-framework Security"
         PUBLIC
         "-framework CoreFoundation"
+        "-framework AppKit"
     )
     if(WXOSX_COCOA)
         wx_lib_link_libraries(wxbase
-- 
2.24.3 (Apple Git-128)

