From 7a1fb96554ce123c9d0b2922c9942e71a0afc827 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Tue, 28 Oct 2025 20:37:38 +0100
Subject: [PATCH 23/48] using backwards compatibly OBJC_FOR

---
 src/osx/core/uilocale.mm | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/osx/core/uilocale.mm b/src/osx/core/uilocale.mm
index 8fa0e137f7..7bd026a161 100644
--- a/src/osx/core/uilocale.mm
+++ b/src/osx/core/uilocale.mm
@@ -103,7 +103,7 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
         // completely invalid strings, so we need to check if the name is
         // actually in the list of the supported locales ourselves.
         bool isAvailable = false;
-        for ( id nsLocId in [NSLocale availableLocaleIdentifiers] )
+        wxOBJC_FOR_LOOP( id nsLocId, [NSLocale availableLocaleIdentifiers] )
         {
             // We can't simply compare the names here because the list returned
             // by NSLocale is incomplete and doesn't contain all synonyms, e.g.
@@ -141,6 +141,7 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
             if ( isAvailable )
                 break;
         }
+        wxOBJC_END_FOR_LOOP
 
         if ( !isAvailable )
             return nullptr;
-- 
2.24.3 (Apple Git-128)

