From 0d5190fb11517593fd3db9faa52ec24e6e8f563c Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 29 Oct 2025 09:19:20 +0100
Subject: [PATCH 24/48] test to bring locale resolution in line with unix

---
 src/osx/core/uilocale.mm | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/osx/core/uilocale.mm b/src/osx/core/uilocale.mm
index 7bd026a161..796a7ce4e4 100644
--- a/src/osx/core/uilocale.mm
+++ b/src/osx/core/uilocale.mm
@@ -133,10 +133,15 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
                         isAvailable = availId.GetRegion() == locId.GetRegion();
                     }
                 }
-                else
+                else if (!locId.GetRegion().empty() )
                 {
                     isAvailable = availId.GetRegion() == locId.GetRegion();
                 }
+                else
+                {
+                    // No script or region specified, language match is enough.
+                    isAvailable = true;
+                }
             }
             if ( isAvailable )
                 break;
-- 
2.24.3 (Apple Git-128)

