From 1ce6e5a27477288a32cf7067f791766523fd469d Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 29 Oct 2025 12:03:23 +0100
Subject: [PATCH 27/48] NSNumber Literals (clang 4) not available under gcc

---
 src/osx/core/uilocale.mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osx/core/uilocale.mm b/src/osx/core/uilocale.mm
index 68607197ed..d3f7197c56 100644
--- a/src/osx/core/uilocale.mm
+++ b/src/osx/core/uilocale.mm
@@ -399,7 +399,7 @@ int CompareStrings(const wxString& lhs, const wxString& rhs,
     formatter.locale = m_nsloc;
     formatter.numberStyle = NSNumberFormatterCurrencyStyle;
 
-    NSString* formatted = [formatter stringFromNumber:@123];
+    NSString* formatted = [formatter stringFromNumber:[NSNumber numberWithInt:123]];
     NSString* symbol = formatter.currencySymbol;
 
     NSRange symbolRange = [formatted rangeOfString:symbol];
-- 
2.24.3 (Apple Git-128)

