From b498b7c6a70f3007d9c7c26c2b576a8963044d65 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 29 Oct 2025 14:26:37 +0100
Subject: [PATCH 29/48] guard code properly for non iPhone builds

---
 src/osx/webrequest_urlsession.mm | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/osx/webrequest_urlsession.mm b/src/osx/webrequest_urlsession.mm
index 1ba44f0a0e..412371fba0 100644
--- a/src/osx/webrequest_urlsession.mm
+++ b/src/osx/webrequest_urlsession.mm
@@ -506,7 +506,7 @@ long WXUNUSED(dataTimeoutMs))
 {
     [m_session release];
     [m_delegate release];
-#if !wxOSX_USE_IPHONE
+#if !defined(wxOSX_USE_IPHONE) || !wxOSX_USE_IPHONE
     [m_proxyURL release];
 #endif // !wxOSX_USE_IPHONE
 }
@@ -542,7 +542,7 @@ long WXUNUSED(dataTimeoutMs))
 
 bool wxWebSessionURLSession::SetProxy(const wxWebProxy& proxy)
 {
-#if wxOSX_USE_IPHONE
+#if defined(wxOSX_USE_IPHONE) && wxOSX_USE_IPHONE
     // Setting the proxy doesn't seem to be supported under iOS.
     return false;
 #else // !wxOSX_USE_IPHONE
@@ -641,7 +641,7 @@ long WXUNUSED(dataTimeoutMs))
              [NSURLSessionConfiguration defaultSessionConfiguration] :
              [NSURLSessionConfiguration ephemeralSessionConfiguration];
 
-#if !wxOSX_USE_IPHONE
+#if !defined(wxOSX_USE_IPHONE) || !wxOSX_USE_IPHONE
         switch ( GetProxy().GetType() )
         {
             case wxWebProxy::Type::URL:
-- 
2.24.3 (Apple Git-128)

