From b15f3fde5adddcbcafd4f42c9b286658b6c83e42 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 29 Oct 2025 21:28:40 +0100
Subject: [PATCH 34/48] remove debug info, tests passing now

---
 src/osx/core/uilocale.mm | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/osx/core/uilocale.mm b/src/osx/core/uilocale.mm
index 2efcb77cd9..da597080b7 100644
--- a/src/osx/core/uilocale.mm
+++ b/src/osx/core/uilocale.mm
@@ -116,7 +116,6 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
             const auto availId = wxLocaleIdent::FromTag(strId);
             if ( availId.IsEmpty() )
             {
-                // wxLogTrace("Failed to parse locale identifier %s" , [nsLocId UTF8String]);
                 wxLogDebug("Failed to parse locale identifier \"%s\"", strId);
                 continue;
             }
@@ -147,10 +146,7 @@ explicit wxUILocaleImplCF(NSLocale* nsloc)
         wxOBJC_END_FOR_LOOP
 
         if ( !isAvailable )
-        {
-            wxLogTrace("locale", "no match found for %s" ,  locId.GetTag().c_str());
             return nullptr;
-        }
 
         wxCFStringRef cfName(locId.GetName());
         NSLocale* nsloc = [[NSLocale alloc] initWithLocaleIdentifier:cfName.AsNSString()];
-- 
2.24.3 (Apple Git-128)

