From e0682edbbebcf2f7b0612eda1362725d5a7fd509 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Thu, 30 Oct 2025 20:26:58 +0100
Subject: [PATCH 35/38] look for gstreamer when using gtk also under macos

---
 build/cmake/init.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/cmake/init.cmake b/build/cmake/init.cmake
index 4ed7b264ef..04756956b0 100644
--- a/build/cmake/init.cmake
+++ b/build/cmake/init.cmake
@@ -645,7 +645,7 @@ if(wxUSE_GUI)
         endif()
     endif()
 
-    if(wxUSE_MEDIACTRL AND WXGTK AND NOT APPLE AND NOT WIN32)
+    if(wxUSE_MEDIACTRL AND WXGTK AND NOT WIN32)
         find_package(GSTREAMER 1.0 COMPONENTS video)
         if(NOT GSTREAMER_FOUND)
             find_package(GSTREAMER 0.10 COMPONENTS interfaces)
-- 
2.24.3 (Apple Git-128)

