From 7acb1d859ddb0e4b13859e8b607cd491ebc3d2ee Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Fri, 31 Oct 2025 10:01:10 +0100
Subject: [PATCH 36/48] make sure, wxHAS_NSPROCESSINFO is always defined

---
 src/osx/cocoa/utils_base.mm | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/osx/cocoa/utils_base.mm b/src/osx/cocoa/utils_base.mm
index 784b9463e9..599a4fa413 100644
--- a/src/osx/cocoa/utils_base.mm
+++ b/src/osx/cocoa/utils_base.mm
@@ -36,6 +36,8 @@
 #if (defined(__APPLE__) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 101000) \
     || (defined(__WXOSX_IPHONE__) && defined(__IPHONE_8_0))
     #define wxHAS_NSPROCESSINFO 1
+#else
+    #define wxHAS_NSPROCESSINFO 0
 #endif
 
 // our OS version is the same in non GUI and GUI cases
-- 
2.24.3 (Apple Git-128)

