From 72309b225301a125e26e7dec00ea2ef657a9b8bc Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Mon, 3 Nov 2025 10:00:59 +0100
Subject: [PATCH 39/48] use backwards compatible APIs

---
 src/osx/volume.mm | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/osx/volume.mm b/src/osx/volume.mm
index 3c4eb732dd..f0d5737246 100644
--- a/src/osx/volume.mm
+++ b/src/osx/volume.mm
@@ -67,7 +67,7 @@
     {
         wxOBJC_FOR_LOOP(NSURL* url, nativeVolumes)
         {
-            wxFSVolumeBase volume(url.fileSystemRepresentation);
+            wxFSVolumeBase volume([[url path] fileSystemRepresentation]);
             int flags = volume.GetFlags();
             if ((flags & flagsSet) == flagsSet && !(flags & flagsUnset))
                 volumePaths.push_back(volume.GetName());
-- 
2.24.3 (Apple Git-128)

