From ceb58f2bf1c51bda301dca3174bf21f0b7a9d784 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Mon, 3 Nov 2025 11:35:46 +0100
Subject: [PATCH 41/48] fix two checks problems

---
 src/osx/core/mimetype.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/osx/core/mimetype.cpp b/src/osx/core/mimetype.cpp
index 0aec5f1dd8..9c6fb07728 100644
--- a/src/osx/core/mimetype.cpp
+++ b/src/osx/core/mimetype.cpp
@@ -429,8 +429,8 @@ void wxMimeTypesManagerImpl::LoadDisplayDataForUti(const wxString& uti)
     wxCFRef<CFURLRef> appUrl;
 #if MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_10
     CFURLRef aurl;
-    // THIS FUNCTION, DESPITE ITS NAME, RETAINS THE URL REFERENCE ON BEHALF OF THE CALLER. 
-    OSStatus status = LSGetApplicationForInfo( kLSUnknownType, kLSUnknownCreator, ext, kLSRolesAll, NULL, &aurl );
+    // THIS FUNCTION, DESPITE ITS NAME, RETAINS THE URL REFERENCE ON BEHALF OF THE CALLER.
+    OSStatus status = LSGetApplicationForInfo( kLSUnknownType, kLSUnknownCreator, ext, kLSRolesAll, nullptr, &aurl );
     if( status != noErr )
         return;
     appUrl.reset(aurl);
-- 
2.24.3 (Apple Git-128)

