--- Makefile
+++ Makefile	2026-08-05 18:06:52.000000000 +0800
@@ -157,9 +157,9 @@
 # Homebrew's OpenSSL instead of the system one by default.
 ifdef IS_MACOS
 	ifeq ($(CRYPTO_ENGINE_PATH),openssl)
-		ifneq ($(HOMEBREW_OPENSSL_PATH),)
-			CRYPTO_ENGINE_INCLUDE_PATH = $(HOMEBREW_OPENSSL_PATH)/include
-			CRYPTO_ENGINE_LIB_PATH = $(HOMEBREW_OPENSSL_PATH)/lib
+		ifneq ($(OPENSSL_INSTALL_PATH),)
+			CRYPTO_ENGINE_INCLUDE_PATH = $(OPENSSL_INSTALL_PATH)/include
+			CRYPTO_ENGINE_LIB_PATH = $(OPENSSL_INSTALL_PATH)/lib
 		endif
 	endif
 endif

--- configure
+++ configure	2026-08-05 18:13:24.000000000 +0800
@@ -232,11 +232,7 @@
 
 find_macos_openssl() {
     [ -z "${IS_MACOS:-}" ] && return
-    local path="$(brew --prefix openssl@1.1 2> /dev/null || true)"
-    if [ -n "$path" ]
-    then
-        set_variable HOMEBREW_OPENSSL_PATH "$path"
-    fi
+        set_variable OPENSSL_INSTALL_PATH "$OPENSSL_INSTALL_PATH"
 }
 
 detect_language_runtimes() {
