--- xlgui/__init__.py	2025-05-24 06:15:49.000000000 +0800
+++ xlgui/__init__.py	2025-06-13 03:10:15.000000000 +0800
@@ -151,9 +151,6 @@
         logger.info("Done loading main window...")
         Main._main = self
 
-        if sys.platform == 'darwin':
-            self._setup_osx()
-
     def open_uris(self, uris, play=True):
         if len(uris) > 0:
             self.open_uri(uris[0], play=play)
