--- Makefile	2025-05-24 06:15:49.000000000 +0800
+++ Makefile	2025-06-13 00:41:48.000000000 +0800
@@ -2,7 +2,7 @@
 PYTEST        ?= py.test-3
 BLACK         ?= black
 
-PREFIX         = /usr/local
+PREFIX         ?= /opt/local
 EPREFIX        = $(PREFIX)
 
 LIBINSTALLDIR  = $(EPREFIX)/lib
@@ -10,7 +10,7 @@
 MANPREFIX      = $(PREFIX)/share
 # /etc if PREFIX is /usr, $PREFIX/etc otherwise.
 ETCDIR        := $(shell [ "$(PREFIX)" = "/usr" ] && echo /etc || echo "$(PREFIX)/etc")
-XDGCONFDIR     = $(ETCDIR)/xdg
+XDGCONFDIR     = $(ETCDIR)
 
 # Find bash-completion's completions directory, first by checking pkg-config,
 # then using a hard-coded path. Override BASHCOMPDIR if it's still wrong for
