--- src/qt-console/bat.pro.in	2024-03-22 17:48:41.000000000 +0800
+++ src/qt-console/bat.pro.in	2025-05-10 04:08:22.000000000 +0800
@@ -18,6 +18,11 @@
 help.path = /$(DESTDIR)@docdir@
 help.files = help/*.html images/status.png images/mail-message-new.png
 
+QMAKE_CC     = @CC@
+QMAKE_CFLAGS    += "@CFLAGS@"
+QMAKE_CXX    = @CXX@
+QMAKE_CXXFLAGS  += "@CXXFLAGS@"
+QMAKE_LINK_SHLIB = @CXX@
 TEMPLATE     = app
 TARGET       = bat
 DEPENDPATH  += .
