--- plugins/perl-loader/Makefile.am	2026-04-30 10:17:20.000000000 +0800
+++ plugins/perl-loader/Makefile.am	2026-08-06 20:50:51.000000000 +0800
@@ -41,11 +41,10 @@
 perl-gnumeric.lo: $(some_deps)
 
 
-# Use the right compiler.  This is not 100% correct: we use GNUMERIC_CFLAGS
-# here which might contain stuff we don't like.  It looks good enough,
-# though.
+# Using the right compiler. Requires CC here, not PERL_CC.
+# The code here uses C11 features, it fails to compile with PERL_CC that expands to /usr/bin/cc.
 dist_noinst_SCRIPTS=perl-cc-wrapper
-COMPILE=$(PERL) $(srcdir)/perl-cc-wrapper $(PERL_CC) $(PERL_CCOPTS) \
+COMPILE=$(PERL) $(srcdir)/perl-cc-wrapper $(CC) $(PERL_CCOPTS) \
 	$(PERL_CCCDLFLAGS) $(AM_CPPFLAGS) --sanitize=1 $(GNUMERIC_CFLAGS)
 LTCOMPILE=$(LIBTOOL) --tag=CC --mode=compile $(COMPILE)
 
