--- plugins/psiconv/psiconv-read.c	2026-04-30 10:17:20.000000000 +0800
+++ plugins/psiconv/psiconv-read.c	2026-08-06 19:49:49.000000000 +0800
@@ -56,6 +56,17 @@
 		g_utf16_to_utf8 (data, -1, NULL, NULL, NULL));
 }
 
+static GnmCellRef *
+p_cellref_init (GnmCellRef *cr, int row_offset, gboolean row_absolute,
+		int col_offset, gboolean col_absolute)
+{
+	cr->sheet = NULL;
+	cr->col   = col_offset;
+	cr->row   = row_offset;
+	cr->col_relative = !col_absolute;
+	cr->row_relative = !row_absolute;
+	return cr;
+}
 
 static void
 set_format (GnmStyle *style, const psiconv_sheet_numberformat psi_numberformat)
