--- boost/asio/detail/config.hpp	2025-12-03 21:46:38.000000000 +0800
+++ boost/asio/detail/config.hpp	2026-09-12 20:26:43.000000000 +0800
@@ -389,9 +389,9 @@
 #       define BOOST_ASIO_HAS_STD_ALIGNED_ALLOC 1
 #      elif defined(__APPLE__)
 #       if defined(__MAC_OS_X_VERSION_MIN_REQUIRED)
-#        if (__MAC_OS_X_VERSION_MIN_REQUIRED >= 101500)
+#        if (__MAC_OS_X_VERSION_MIN_REQUIRED >= 110000)
 #         define BOOST_ASIO_HAS_STD_ALIGNED_ALLOC 1
-#        endif // (__MAC_OS_X_VERSION_MIN_REQUIRED >= 101500)
+#        endif // (__MAC_OS_X_VERSION_MIN_REQUIRED >= 110000)
 #       elif defined(__IPHONE_OS_VERSION_MIN_REQUIRED)
 #        if (__IPHONE_OS_VERSION_MIN_REQUIRED >= 130000)
 #         define BOOST_ASIO_HAS_STD_ALIGNED_ALLOC 1
