From 4844ce200cb06d68743ea7a6810b5ea975d557b1 Mon Sep 17 00:00:00 2001
From: Michael Schwendt <mschwendt@fedoraproject.org>
Date: Thu, 30 Oct 2025 22:29:25 +0100
Subject: [PATCH] for Clang, fixes #4

---
 src/LiveFix.h | 2 +-
 src/TFMX.cpp  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/LiveFix.h b/src/LiveFix.h
index 5894dc8..b65461e 100644
--- src/LiveFix.h
+++ src/LiveFix.h
@@ -364,7 +364,7 @@ if ( !LiveFixShaolin5 ) {
     };
     if ( havePattern(0x1a,pattOrig1a) && havePattern(0x17,pattOrig17) ) {
 
-        const int t = 2*trackStepLen;
+        const int t = 2*(TFMX_TRACKTAB_STEP_SIZE+1);
         const ubyte trackTabOrig[t] = {
             0x12, 0x0c, 0x04, 0x14, 0xf4, 0x00, 0x1a, 0x00, 0x0c, 0x00, 0x00, 0x00, 0x00,
             0x16, 0x0c, 0x04, 0x15, 0xf4, 0x00, 0x00, 0x00, 0x0c, 0x00, 0x00, 0x00, 0x00
diff --git a/src/TFMX.cpp b/src/TFMX.cpp
index bd95567..51ac304 100644
--- src/TFMX.cpp
+++ src/TFMX.cpp
@@ -179,7 +179,7 @@ void FC::TFMX_nextNote(VoiceVars& voiceX) {
         voiceX.pattStart = offsets.patterns+(pattern<<6);
         voiceX.transpose = fcBufS[trackOffs++];  // TR
         voiceX.soundTranspose = fcBufS[trackOffs++];  // ST
-        if (trackColumnSize==3 && voiceX.soundTranspose == 0x80) {
+        if (trackColumnSize==3 && (ubyte)voiceX.soundTranspose == 0x80) {
         // ST 0x80 is "player off".
             songEnd = true;
             off();
