--- setup.py	2025-06-23 21:00:21
+++ setup.py	2025-06-27 10:01:23
@@ -129,9 +129,9 @@
 if pkg_flag:
 
     # Icons/sounds must be copied into the right place
-    icon_path = '/usr/share/tartube/icons/'
-    locale_path = '/usr/share/locale/'
-    sound_path = '/usr/share/tartube/sounds/'
+    icon_path = '/share/tartube/icons/'
+    locale_path = '/share/locale/'
+    sound_path = '/share/tartube/sounds/'
     # Add a desktop file
     param_list.append(('share/applications', ['pack/tartube.desktop']))
     param_list.append(('share/pixmaps', ['pack/tartube.png']))
@@ -162,12 +162,14 @@
 
 #for path in glob.glob('locale/*'):
 #    param_list.append((locale_path, [path]))
-for root, dirs, files in os.walk('locale'):
-    for file in files:
-        param_list.append((
-            os.path.join('/usr/share/locale', os.path.relpath(root, 'locale')),
-            [os.path.join(root, file)]
-        ))
+with_locale = os.environ.get('WITH_LOCALE', None)
+if with_locale is not None:
+    for root, dirs, files in os.walk('locale'):
+        for file in files:
+            param_list.append((
+                os.path.join('@PYROOT@/share/locale', os.path.relpath(root, 'locale')),
+                [os.path.join(root, file)]
+            ))
 
 for path in glob.glob('sounds/*'):
     param_list.append((sound_path, [path]))
