--- source3/lib/sysquotas_4B.c.orig	2026-01-20 23:42:54.000000000 +0800
+++ source3/lib/sysquotas_4B.c	2026-08-23 08:28:14.000000000 +0800
@@ -135,7 +135,7 @@
 	become_root();
 #endif  /* HFS_QUOTACTL_WAR */
 
-	ret = quotactl(path, cmd, id, qblk);
+	ret = quotactl(path, cmd, id, (caddr_t)qblk);
 	if (ret == -1) {
 		/* ENOTSUP means quota support is not compiled in. EINVAL
 		 * means that quotas are not configured (commonly).
