From 9b088652b2ecdcc2e132431b37772d1307c48822 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 17 Sep 2025 20:41:06 +0800
Subject: [PATCH] Allow using pre-defined wx config on macOS

---
 CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fc09951964..56afdff432 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -232,7 +232,10 @@ else()
         endif()
 
     elseif (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-        if (EXISTS "/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.2/bin/wx-config")
+        if (DEFINED wxWidgets_CONFIG_EXECUTABLE)
+            message(STATUS "Using pre-defined wx config: ${wxWidgets_CONFIG_EXECUTABLE}")
+
+        elseif (EXISTS "/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.2/bin/wx-config")
             # MacPorts
             set(wxWidgets_CONFIG_EXECUTABLE "/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.2/bin/wx-config")
             set(wxWidgets_wxrc_EXECUTABLE "/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.2/bin/wxrc")


--- CMakeLists.txt	2025-09-18 01:15:34.000000000 +0800
+++ CMakeLists.txt	2025-09-18 02:01:37.000000000 +0800
@@ -86,13 +86,7 @@
 message(STATUS "Build Type: ${CMAKE_BUILD_TYPE} Version: ${VERSION}")
 
 if(NOT DEFINED CMAKE_INSTALL_PREFIX)
-   if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-      set(CMAKE_INSTALL_PREFIX "/usr/local")
-   elseif (${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
-      set(CMAKE_INSTALL_PREFIX "/usr/local")
-   else()
-      set(CMAKE_INSTALL_PREFIX "/usr")
-   endif()
+   set(CMAKE_INSTALL_PREFIX "@PREFIX@")
 endif()
 
 ##############################
@@ -156,9 +150,8 @@
 set(CMAKE_C_FLAGS_RELEASE "-O2")
 
 if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    set(APP_DIR ${CMAKE_BINARY_DIR}/install)
-    set(INSTALL_DIR ${APP_DIR}/${APP_NAME}.app/Contents/MacOS)
-    set(EXECUTABLE_NAME ${APP_NAME})
+    set(INSTALL_DIR ${CMAKE_BINARY_DIR}/install)
+    set(EXECUTABLE_NAME far2l)
 
 else()
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ffunction-sections -fdata-sections")
@@ -408,22 +401,8 @@
     pkg_search_module(LibArchive QUIET libarchive)
     find_package(LibArchive)
     if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-        # workaround for MacOS brew's/macports' libarchive
-        execute_process(COMMAND brew --prefix
-            OUTPUT_VARIABLE BREW_PREFIX
-            OUTPUT_STRIP_TRAILING_WHITESPACE
-            ERROR_QUIET)
-        if("${BREW_PREFIX}" STREQUAL "")
-            set(BREW_PREFIX "/usr/local")
-            message(STATUS "Fallback BREW_PREFIX=${BREW_PREFIX}")
-        else()
-            message(STATUS "Detected BREW_PREFIX=${BREW_PREFIX}")
-        endif()
-        if(IS_DIRECTORY "${BREW_PREFIX}/opt/libarchive/include")
-            set(LibArchive_INCLUDE_DIR "${BREW_PREFIX}/opt/libarchive/include")
-            set(LibArchive_LIBRARY "${BREW_PREFIX}/opt/libarchive/lib/libarchive.dylib")
-        elseif(EXISTS "/opt/local/lib/libarchive.dylib")
-            set(LibArchive_LIBRARY "/opt/local/lib/libarchive.dylib")
+        if(EXISTS "@PREFIX@/lib/libarchive.dylib")
+            set(LibArchive_LIBRARY "@PREFIX@/lib/libarchive.dylib")
         endif()
         find_package(LibArchive)
         if(NOT LibArchive_FOUND)
@@ -534,13 +513,6 @@
 add_subdirectory(man)
 add_subdirectory(bash-completion)
 
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    set(FIXUP_SCRIPT packaging/osx/FixupBundle.cmake)
-    configure_file(${FIXUP_SCRIPT}.in ${PROJECT_BINARY_DIR}/${FIXUP_SCRIPT} @ONLY)
-    install(DIRECTORY ${APP_DIR}/${APP_NAME}.app DESTINATION . USE_SOURCE_PERMISSIONS COMPONENT app EXCLUDE_FROM_ALL)
-    install(SCRIPT ${PROJECT_BINARY_DIR}/${FIXUP_SCRIPT} COMPONENT app EXCLUDE_FROM_ALL)
-endif()
-
 install(PROGRAMS "${INSTALL_DIR}/${EXECUTABLE_NAME}" DESTINATION "bin" RENAME far2l COMPONENT base)
 
 install(DIRECTORY "${INSTALL_DIR}/" DESTINATION "lib/far2l" USE_SOURCE_PERMISSIONS COMPONENT base FILES_MATCHING
