From 63fe8996b2901a65d87cc2af3b5c2fd083268115 Mon Sep 17 00:00:00 2001
From: Alexey Shelaev <75270873+exkrexpexfex@users.noreply.github.com>
Date: Wed, 17 Sep 2025 18:52:55 +0300
Subject: [PATCH] Adjust colors for big-endian

fix #2870
---
 far2l/src/color.hpp | 67 ++++++++++++++++++++++++++++++---------------
 1 file changed, 45 insertions(+), 22 deletions(-)

diff --git a/far2l/src/color.hpp b/far2l/src/color.hpp
index e8930f6777..56284ae9e7 100644
--- far2l/src/color.hpp
+++ far2l/src/color.hpp
@@ -32,30 +32,53 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
-typedef struct rgbcolor_s
-{
-	union {
-		struct {
-			uint8_t r;
-			uint8_t g;
-			uint8_t b;
-			uint8_t a;
-		};
-		uint32_t rgba;
-	};
+#include <stdint.h>
+
+typedef struct rgbcolor_s {
+    union {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+        struct {
+            uint8_t r;
+            uint8_t g;
+            uint8_t b;
+            uint8_t a;
+        };
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        struct {
+            uint8_t a;
+            uint8_t b;
+            uint8_t g;
+            uint8_t r;
+        };
+#else
+#   error "Unsupported endianness"
+#endif
+        uint32_t rgba;
+    };
 } rgbcolor_t;
 
-typedef struct hsvcolor_s
-{
-	union {
-		struct {
-			uint8_t h;
-			uint8_t s;
-			uint8_t v;
-			uint8_t a;
-		};
-		uint32_t hsva;
-	};
+
+typedef struct hsvcolor_s {
+    union {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+        struct {
+            uint8_t h;
+            uint8_t s;
+            uint8_t v;
+            uint8_t a;
+        };
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+        struct {
+            uint8_t a;
+            uint8_t v;
+            uint8_t s;
+            uint8_t h;
+        };
+#else
+#   error "Unsupported endianness"
+#endif
+        uint32_t hsva;
+    };
 } hsvcolor_t;
 
 #ifndef RGB
