--- WinPort/src/Backend/WX/CMakeLists.txt	2026-03-23 16:04:51.000000000 +0800
+++ WinPort/src/Backend/WX/CMakeLists.txt	2026-04-25 01:37:58.000000000 +0800
@@ -33,7 +33,6 @@
         Mac/hide.mm
         Mac/init.mm
         Mac/displaynotify.mm
-        Mac/printing.mm
     )
 endif()
 
--- WinPort/src/Backend/WX/wxPrinterSupport.cpp	2026-03-23 16:04:51.000000000 +0800
+++ WinPort/src/Backend/WX/wxPrinterSupport.cpp	2026-04-25 01:41:29.000000000 +0800
@@ -2,7 +2,7 @@
 #include <wx/display.h>
 
 #if defined(__APPLE__)
-#define MAC_NATIVE_PRINTING // comment this define if you want wxWidgets-aware dialogs
+// #define MAC_NATIVE_PRINTING // uncomment this define if you want Cocoa dialogs and use clang.
 #endif
 
 #ifndef MAC_NATIVE_PRINTING
