diff --git plugins/multiarc/src/formats/rar/rar.cpp plugins/multiarc/src/formats/rar/rar.cpp
index 330297293..ea18a9e7c 100644
--- plugins/multiarc/src/formats/rar/rar.cpp
+++ plugins/multiarc/src/formats/rar/rar.cpp
@@ -16,7 +16,7 @@
 using namespace oldfar;
 #include "fmt.hpp"
 #include "marclng.hpp"
-#include "dll.hpp"
+#include <unrar/dll.hpp>
 
 // TODO: we have full-featured unrar code on board, use it to unpack instead of unrar executable!
 
