--- far2l/CMakeLists.txt
+++ far2l/CMakeLists.txt	2025-09-18 02:18:40.000000000 +0800
@@ -195,23 +195,8 @@
 endif()
 
 # Set target names for the executables
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    set(MACOSX_BUNDLE_LONG_VERSION_STRING ${VERSION})
-    set(MACOSX_BUNDLE_SHORT_VERSION_STRING ${VERSION})
-    set(MACOSX_BUNDLE_GUI_IDENTIFIER ${APP_IDENTIFIER})
-    set(MACOSX_BUNDLE_ICON_FILE "Far.icns")
-    set(MACOSX_BUNDLE_BUNDLE_NAME ${APP_NAME})
-    add_executable(far2l MACOSX_BUNDLE ${SOURCES} "DE/Far.icns")
-    set_target_properties(far2l PROPERTIES
-        OUTPUT_NAME ${APP_NAME}
-        RUNTIME_OUTPUT_DIRECTORY ${APP_DIR})
-    set_source_files_properties("DE/Far.icns" PROPERTIES
-        MACOSX_PACKAGE_LOCATION Resources)
-else()
-    add_executable(far2l ${SOURCES})
-    set_target_properties(far2l PROPERTIES
-        RUNTIME_OUTPUT_DIRECTORY "${INSTALL_DIR}")
-endif()
+add_executable(far2l ${SOURCES})
+set_target_properties(far2l PROPERTIES RUNTIME_OUTPUT_DIRECTORY "${INSTALL_DIR}")
 
 if (NOT DEFINED USEUCD)
     set(USEUCD "YES")
@@ -281,19 +266,16 @@
     COMMAND ln -sf ${EXECUTABLE_NAME} ${INSTALL_DIR}/far2l_sudoapp
     COMMAND ln -sf ${EXECUTABLE_NAME} ${INSTALL_DIR}/far2ledit)
 
-if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-#TODO
-else()
-    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l-tty.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l-tty.desktop")
-    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2ledit-tty.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit-tty.desktop")
-    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l.desktop")
-    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2ledit.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit.desktop")
-
-    install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/DE/icons" DESTINATION "share" USE_SOURCE_PERMISSIONS COMPONENT desktop FILES_MATCHING PATTERN "*")
-
-	# Have to make desktop files executable, see:
-	# https://wiki.ubuntu.com/SecurityTeam/Policies#Execute-Permission_Bit_Required
-    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l-tty.desktop"
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l-tty.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l-tty.desktop")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2ledit-tty.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit-tty.desktop")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2l.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l.desktop")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/DE/far2ledit.desktop" "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit.desktop")
+
+install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/DE/icons" DESTINATION "share" USE_SOURCE_PERMISSIONS COMPONENT desktop FILES_MATCHING PATTERN "*")
+
+# Have to make desktop files executable, see:
+# https://wiki.ubuntu.com/SecurityTeam/Policies#Execute-Permission_Bit_Required
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l-tty.desktop"
                   "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit-tty.desktop"
                   "${CMAKE_CURRENT_BINARY_DIR}/DE/far2l.desktop"
                   "${CMAKE_CURRENT_BINARY_DIR}/DE/far2ledit.desktop"
@@ -301,9 +283,9 @@
         COMPONENT desktop
         PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ GROUP_EXECUTE GROUP_READ WORLD_READ WORLD_EXECUTE)
 
-    find_package(Python3 COMPONENTS Interpreter)
+find_package(Python3 COMPONENTS Interpreter)
 
-    if(Python3_Interpreter_FOUND)
+if(Python3_Interpreter_FOUND)
         execute_process(COMMAND "${Python3_EXECUTABLE}" -c "import markdown"
                         RESULT_VARIABLE PYTHON_MARKDOWN_STATUS
                         OUTPUT_QUIET
@@ -326,8 +308,6 @@
         else()
             message(WARNING "Python-Markdown not found. Skipping AppStream metainfo generation.")
         endif()
-    else()
+else()
         message(WARNING "Python3 interpreter not found. Skipping AppStream metainfo generation.")
-    endif()
-
 endif()
