--- libavcodec/audiotoolboxenc.c	2026-05-05 23:50:52.000000000 +0800
+++ libavcodec/audiotoolboxenc.c	2026-05-10 01:44:39.000000000 +0800
@@ -69,8 +69,10 @@
             return kAudioFormatMPEG4AAC_HE_V2;
         case FF_PROFILE_AAC_LD:
             return kAudioFormatMPEG4AAC_LD;
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 1070
         case FF_PROFILE_AAC_ELD:
             return kAudioFormatMPEG4AAC_ELD;
+#endif
         }
     case AV_CODEC_ID_ADPCM_IMA_QT:
         return kAudioFormatAppleIMA4;
