--- libavcodec/audiotoolboxdec.c.orig
+++ libavcodec/audiotoolboxdec.c
@@ -31,11 +31,17 @@
 #include "libavutil/avassert.h"
 #include "libavutil/opt.h"
 #include "libavutil/log.h"
+#include <AvailabilityMacros.h>

 #if __MAC_OS_X_VERSION_MIN_REQUIRED < 101100
 #define kAudioFormatEnhancedAC3 'ec-3'
 #endif

+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#define kAudioFormatMicrosoftGSM 0x6D730031
+#define kAudioFormatiLBC 'ilbc'
+#endif
+
 typedef struct ATDecodeContext {
     AVClass *av_class;

--- libavcodec/audiotoolboxenc.c.orig
+++ libavcodec/audiotoolboxenc.c
@@ -34,6 +34,11 @@
 #include "libavutil/avassert.h"
 #include "libavutil/opt.h"
 #include "libavutil/log.h"
+#include <AvailabilityMacros.h>
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+#define kAudioFormatiLBC 'ilbc'
+#endif

 typedef struct ATDecodeConte
