From fbb188a148d5b9ea32ceba8f54b4763c65dc24d6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 8 Feb 2025 01:09:59 +0800
Subject: [PATCH] Re-enable dispatch on 10.6 powerpc

---
 src/lib/IlmThread/IlmThreadSemaphore.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/lib/IlmThread/IlmThreadSemaphore.h src/lib/IlmThread/IlmThreadSemaphore.h
index 18638615..0ba881d6 100644
--- src/lib/IlmThread/IlmThreadSemaphore.h
+++ src/lib/IlmThread/IlmThreadSemaphore.h
@@ -31,7 +31,7 @@
 #        define ILMTHREAD_SEMAPHORE_POSIX 1
 #    elif defined(__APPLE__)
 #        include <AvailabilityMacros.h>
-#        if MAC_OS_X_VERSION_MIN_REQUIRED > 1050 && !defined(__ppc__)
+#        if MAC_OS_X_VERSION_MIN_REQUIRED > 1050
 #            include <dispatch/dispatch.h>
 #            define ILMTHREAD_SEMAPHORE_OSX 1
 #        else
