From 78cd229657187269ae6d94226840690eed9d9c51 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 15 Mar 2025 07:27:14 +0800
Subject: [PATCH] exrmetrics: fix isinf

---
 src/bin/exrmetrics/exrmetrics.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/bin/exrmetrics/exrmetrics.cpp src/bin/exrmetrics/exrmetrics.cpp
index b0cfc707..9b6df5d8 100644
--- src/bin/exrmetrics/exrmetrics.cpp
+++ src/bin/exrmetrics/exrmetrics.cpp
@@ -43,6 +43,7 @@
 using std::string;
 using std::to_string;
 using std::vector;
+using std::isinf;
 
 double
 timing (steady_clock::time_point start, steady_clock::time_point end)
