--- modules/videoio/CMakeLists.txt	2021-10-09 02:31:56.000000000 +0800
+++ modules/videoio/CMakeLists.txt	2024-07-17 01:25:47.000000000 +0800
@@ -164,9 +164,6 @@
   if(UNIX AND BZIP2_LIBRARIES)
     list(APPEND VIDEOIO_LIBRARIES ${BZIP2_LIBRARIES})
   endif()
-  if(APPLE)
-    list(APPEND VIDEOIO_LIBRARIES "-framework VideoDecodeAcceleration" bz2)
-  endif()
   if(HAVE_FFMPEG_WRAPPER)
     add_definitions(-DHAVE_FFMPEG_WRAPPER=1)
   endif()
