--- cmake/OpenCVCompilerOptions.cmake	2023-06-27 19:29:13.000000000 +0800
+++ cmake/OpenCVCompilerOptions.cmake	2026-05-21 10:50:18.000000000 +0800
@@ -121,7 +121,6 @@
   endif()
   add_extra_compiler_option(-Wreturn-type)
   add_extra_compiler_option(-Wnon-virtual-dtor)
-  add_extra_compiler_option(-Waddress)
   add_extra_compiler_option(-Wsequence-point)
   add_extra_compiler_option(-Wformat)
   add_extra_compiler_option(-Wformat-security -Wformat)
