--- cmake/OpenCVCompilerOptions.cmake	2017-10-23 23:41:12.000000000 +0800
+++ cmake/OpenCVCompilerOptions.cmake	2024-07-15 15:39:53.000000000 +0800
@@ -115,7 +115,6 @@
   add_extra_compiler_option(-Wall)
   add_extra_compiler_option(-Werror=return-type)
   add_extra_compiler_option(-Werror=non-virtual-dtor)
-  add_extra_compiler_option(-Werror=address)
   add_extra_compiler_option(-Werror=sequence-point)
   add_extra_compiler_option(-Wformat)
   add_extra_compiler_option(-Werror=format-security -Wformat)
