--- cmake/OpenCVDetectCXXCompiler.cmake	2021-10-09 02:31:56.000000000 +0800
+++ cmake/OpenCVDetectCXXCompiler.cmake	2024-07-17 01:36:56.000000000 +0800
@@ -9,7 +9,7 @@
 # - X86 / X86_64
 # - ARM - ARM CPU, not defined for AArch64
 # - AARCH64 - ARMv8+ (64-bit)
-# - PPC64 / PPC64LE - PowerPC
+# - PPC / PPC64 / PPC64LE - PowerPC
 # - MIPS
 #
 # OS:
@@ -96,6 +96,8 @@
   set(PPC64LE 1)
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)64")
   set(PPC64 1)
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(powerpc|ppc)")
+  set(PPC 1)
 elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "^(mips.*|MIPS.*)")
   set(MIPS 1)
 else()
