From 9fc8295d55e02fc90911d070b12a0e3778b24bab Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 8 Apr 2025 03:31:57 +0800
Subject: [PATCH] cmGlobalXCodeGenerator: use specific CF headers

---
 Source/cmGlobalXCodeGenerator.cxx | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git Source/cmGlobalXCodeGenerator.cxx Source/cmGlobalXCodeGenerator.cxx
index 69acd1c777..ab8f395c83 100644
--- Source/cmGlobalXCodeGenerator.cxx
+++ Source/cmGlobalXCodeGenerator.cxx
@@ -58,7 +58,10 @@
 #include "cmake.h"
 
 #if !defined(CMAKE_BOOTSTRAP) && defined(__APPLE__)
-#  include <CoreFoundation/CoreFoundation.h>
+#  include <CoreFoundation/CFBundle.h>
+#  include <CoreFoundation/CFString.h>
+#  include <CoreFoundation/CFURL.h>
+#  include <CoreFoundation/CFUUID.h>
 #  if !TARGET_OS_IPHONE
 #    define HAVE_APPLICATION_SERVICES
 #    include <ApplicationServices/ApplicationServices.h>
