From f9e34a311f149830f5b3da8ee0b7939e1aa58d4e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 8 Apr 2025 03:30:50 +0800
Subject: [PATCH] cmCPackDragNDropGenerator.cxx: use specific CF headers

---
 Source/CPack/cmCPackDragNDropGenerator.cxx | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git Source/CPack/cmCPackDragNDropGenerator.cxx Source/CPack/cmCPackDragNDropGenerator.cxx
index 8d2f516e1c..709ea8a595 100644
--- Source/CPack/cmCPackDragNDropGenerator.cxx
+++ Source/CPack/cmCPackDragNDropGenerator.cxx
@@ -10,7 +10,8 @@
 #include <cm/string_view>
 #include <cmext/string_view>
 
-#include <CoreFoundation/CoreFoundation.h>
+#include <CoreFoundation/CFLocale.h>
+#include <CoreFoundation/CFString.h>
 #include <cm3p/kwiml/abi.h>
 
 #include "cmsys/Base64.h"
