--- CMakeLists.txt	2025-07-09 00:01:09.000000000 +0800
+++ CMakeLists.txt	2026-03-08 09:45:10.000000000 +0800
@@ -23,21 +23,11 @@
 add_library(OpenCL::Headers ALIAS Headers)
 
 if(CMAKE_SYSTEM_NAME MATCHES Darwin)
-  execute_process(
-    COMMAND ln -snf ${CMAKE_CURRENT_SOURCE_DIR}/CL ${CMAKE_CURRENT_BINARY_DIR}/CL
-    COMMAND ln -snf ${CMAKE_CURRENT_SOURCE_DIR}/CL ${CMAKE_CURRENT_BINARY_DIR}/OpenCL
-  )
   target_include_directories(Headers
     INTERFACE
-      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
       $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   )
-  install(CODE "
-    file(MAKE_DIRECTORY  ${CMAKE_INSTALL_FULL_INCLUDEDIR}/CL)
-    execute_process(
-      COMMAND ln -snf CL ${CMAKE_INSTALL_FULL_INCLUDEDIR}/OpenCL
-    )"
-  )
 else()
   target_include_directories(Headers
     INTERFACE
