From 6b659fa1af7194b59a6bfb2b5b21df899ed8e46e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Thu, 26 Mar 2026 05:41:19 +0800
Subject: [PATCH] FS.h: add missing <cstdint>

---
 libi2pd/FS.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libi2pd/FS.h b/libi2pd/FS.h
index 7af8f494623..a48a86957f7 100644
--- a/src/i2pd/libi2pd/FS.h
+++ b/src/i2pd/libi2pd/FS.h
@@ -14,6 +14,7 @@
 #include <iostream>
 #include <sstream>
 #include <functional>
+#include <cstdint>
 
 #ifndef STD_FILESYSTEM
 #	if (_WIN32 && __GNUG__) // MinGW GCC somehow incorrectly converts paths
