--- a/i2pd_qt.pro
+++ b/i2pd_qt.pro	2026-03-26 05:57:02.000000000 +0800
@@ -135,6 +135,7 @@
 	LIBS += -L$$SSLROOT/lib -lssl -lcrypto
 	LIBS += -L$$BOOSTROOT/lib -lboost_system$$BOOST_SUFFIX -lboost_date_time$$BOOST_SUFFIX -lboost_filesystem$$BOOST_SUFFIX -lboost_program_options$$BOOST_SUFFIX
 	LIBS += -L$$UPNPROOT/lib -lminiupnpc
+	LIBS += -latomic
 	LIBS += -Wl,-dead_strip
 	LIBS += -Wl,-dead_strip_dylibs
 	LIBS += -Wl,-bind_at_load
