--- src/platform/posix/posix_pollq_kqueue.c	2025-06-02 22:33:39.000000000 +0800
+++ src/platform/posix/posix_pollq_kqueue.c	2025-07-22 14:57:49.000000000 +0800
@@ -180,7 +180,10 @@
 {
 	struct kevent    ev[2];
 	int              nev   = 0;
-	unsigned         flags = EV_ENABLE | EV_DISPATCH | EV_CLEAR;
+	unsigned         flags = EV_ENABLE | EV_CLEAR;
+#ifdef EV_DISPATCH
+    flags |= EV_DISPATCH;
+#endif
 	nni_posix_pollq *pq    = pf->pq;
 
 	nni_mtx_lock(&pf->mtx);
