--- meson.build	2023-03-23 22:53:00.000000000 +0800
+++ meson.build	2024-09-26 23:04:01.000000000 +0800
@@ -312,8 +312,12 @@
 cxx = meson.get_compiler('cpp')
 
 c_opts = [
-  '-D_ISOC99_SOURCE', '-D_ISOC11_SOURCE', '-D_GNU_SOURCE', '-D_XOPEN_SOURCE=700',
-  '-U__STRICT_ANSI__', '-Wmissing-prototypes',
+  '-D_ISOC99_SOURCE', '-D_ISOC11_SOURCE', '-D_DARWIN_C_SOURCE', '-D_XOPEN_SOURCE=700',
+  '-U__STRICT_ANSI__',
+
+  '-I@PREFIX@/include/LegacySupport',
+
+  '-Wmissing-prototypes',
 
   # Warnings to ignore
   '-Wno-sign-compare', '-Wno-unused-parameter',
