--- src/src.pro
+++ src/src.pro	2024-07-07 08:19:46.000000000 +0800
@@ -38,8 +38,6 @@
 
 SOURCES += \
 	qjackctl.cpp \
-	qjackctlAlsaConnect.cpp \
-	qjackctlAlsaGraph.cpp \
 	qjackctlConnect.cpp \
 	qjackctlConnectAlias.cpp \
 	qjackctlGraph.cpp \


--- src/qjackctlGraphForm.cpp
+++ src/qjackctlGraphForm.cpp	2024-07-07 08:21:37.000000000 +0800
@@ -217,7 +217,9 @@
 
 	m_ui.viewColorsJackAudioAction->setData(qjackctlJackGraph::audioPortType());
 	m_ui.viewColorsJackMidiAction->setData(qjackctlJackGraph::midiPortType());
+#ifdef CONFIG_ALSA_SEQ
 	m_ui.viewColorsAlsaMidiAction->setData(qjackctlAlsaGraph::midiPortType());
+#endif
 
 	QObject::connect(m_ui.viewColorsJackAudioAction,
 		SIGNAL(triggered(bool)),
@@ -225,9 +227,11 @@
 	QObject::connect(m_ui.viewColorsJackMidiAction,
 		SIGNAL(triggered(bool)),
 		SLOT(viewColorsAction()));
+#ifdef CONFIG_ALSA_SEQ
 	QObject::connect(m_ui.viewColorsAlsaMidiAction,
 		SIGNAL(triggered(bool)),
 		SLOT(viewColorsAction()));
+#endif
 	QObject::connect(m_ui.viewColorsResetAction,
 		SIGNAL(triggered(bool)),
 		SLOT(viewColorsReset()));
@@ -387,7 +391,9 @@
 {
 	m_ui.graphCanvas->clearPortTypeColors();
 	m_jack->resetPortTypeColors();
+#ifdef CONFIG_ALSA_SEQ
 	m_alsa->resetPortTypeColors();
+#endif
 	m_ui.graphCanvas->updatePortTypeColors();
 
 	updateViewColors();
@@ -716,7 +722,9 @@
 {
 	updateViewColorsAction(m_ui.viewColorsJackAudioAction);
 	updateViewColorsAction(m_ui.viewColorsJackMidiAction);
+#ifdef CONFIG_ALSA_SEQ
 	updateViewColorsAction(m_ui.viewColorsAlsaMidiAction);
+#endif
 }
 
 
