From 6790234a907e23f91f5d1eb870fe23c3b42d3496 Mon Sep 17 00:00:00 2001
From: Chris Robinson <chris.kcat@gmail.com>
Date: Wed, 11 Dec 2024 02:02:07 -0800
Subject: [PATCH] Overload OSStatus instead of int for FourCCPrinter

---
 alc/backends/coreaudio.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/alc/backends/coreaudio.cpp b/alc/backends/coreaudio.cpp
index 9e355a6a50..2e309b67ef 100644
--- alc/backends/coreaudio.cpp
+++ alc/backends/coreaudio.cpp
@@ -104,7 +104,7 @@ struct FourCCPrinter {
             code >>= 8;
         }
     }
-    constexpr FourCCPrinter(int code) noexcept : FourCCPrinter{static_cast<UInt32>(code)} { }
+    constexpr FourCCPrinter(OSStatus code) noexcept : FourCCPrinter{static_cast<UInt32>(code)} { }
 
     constexpr const char *c_str() const noexcept { return mString; }
 };
