--- alc/backends/coreaudio.cpp.orig
+++ alc/backends/coreaudio.cpp
@@ -55,6 +55,15 @@
 #define CAN_ENUMERATE 1
 #endif

+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+typedef Component AudioComponent;
+typedef ComponentDescription AudioComponentDescription;
+#define AudioComponentFindNext FindNextComponent
+#define AudioComponentInstanceNew OpenAComponent
+#define AudioComponentInstanceDispose CloseComponent
+#define kAudioUnitProperty_ShouldAllocateBuffer 0
+#endif
+
 namespace {

 constexpr auto OutputElement = 0;
