--- CMakeLists.txt	2026-01-20 10:56:03.000000000 +0800
+++ CMakeLists.txt	2026-04-21 15:21:38.000000000 +0800
@@ -1578,7 +1578,7 @@
     set(MODULES_OBJS  ${MODULES_OBJS} alsoft-modules/gsl.cppm)
     target_sources(alsoft.common PUBLIC FILE_SET CXX_MODULES FILES alsoft-modules/gsl.cppm)
 endif()
-if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND DEFINED CMAKE_OSX_DEPLOYMENT_TARGET
+if(APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang" AND DEFINED CMAKE_OSX_DEPLOYMENT_TARGET
     AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS "10.13")
     # Targeting versions prior to macOS 10.13 requires specifying custom
     # alignment-aware operators, which we do. This lets the compiler know they
