--- alc/backends/coreaudio.cpp	2026-05-12 10:21:07.000000000 +0800
+++ alc/backends/coreaudio.cpp	2026-06-24 03:22:07.000000000 +0800
@@ -93,15 +102,6 @@
         kAudioChannelLabel_LeftSurround, kAudioChannelLabel_RightSurround,
         kAudioChannelLabel_LeftCenter, kAudioChannelLabel_RightCenter
 };
-constexpr std::array<AudioChannelLabel, 12> X714ChanMap {
-        kAudioChannelLabel_Left, kAudioChannelLabel_Right, kAudioChannelLabel_Center, kAudioChannelLabel_LFEScreen,
-    kAudioChannelLabel_LeftSurround, kAudioChannelLabel_RightSurround,
-    kAudioChannelLabel_LeftSideSurround,
-    kAudioChannelLabel_RightSideSurround,
-    kAudioChannelLabel_LeftTopFront,
-    kAudioChannelLabel_RightTopFront,
-    kAudioChannelLabel_LeftTopRear, kAudioChannelLabel_RightTopRear
-};
 
 struct FourCCPrinter {
     char mString[sizeof(UInt32) + 1]{};
@@ -545,8 +545,7 @@
         bool is_51rear;
     };
 
-    static constexpr std::array<ChannelMap, 8> chanmaps{{
-        { DevFmtX714, X714ChanMap, false },
+    static constexpr std::array<ChannelMap, 7> chanmaps{{
         { DevFmtX71, X71ChanMap, false },
         { DevFmtX61, X61ChanMap, false },
         { DevFmtX51, X51ChanMap, false },
