--- src/meson.build	2020-06-23 09:04:15.000000000 +0800
+++ src/meson.build	2026-03-19 00:13:30.000000000 +0800
@@ -11,7 +11,7 @@
 	'gelbooru.c', 'konachan.c', 'mangadrawing.c', 'sankakucomplex.c',
 	'yandere.c' , 'zerochan.c']
 
-helpers = ['helpers.c', 'interface.c', 'parser.c', 'structs.c', 'udload.c']
+helpers = ['interface.c', 'parser.c', 'structs.c', 'udload.c', 'util.c']
 
 #add_project_arguments(
 # ruiji executable
