--- tiled.pri	2013-07-28 04:34:23.000000000 +0800
+++ tiled.pri	2025-06-10 16:56:07.000000000 +0800
@@ -4,9 +4,4 @@
 isEmpty(RPATH):RPATH = yes
 isEmpty(INSTALL_HEADERS):INSTALL_HEADERS = no
 
-macx {
-    # Do a universal build when possible
-    contains(QT_CONFIG, ppc):CONFIG += x86 ppc
-}
-
 CONFIG += depend_includepath

--- src/libtiled/libtiled.pro	2013-07-28 04:34:23.000000000 +0800
+++ src/libtiled/libtiled.pro	2025-06-10 17:02:53.000000000 +0800
@@ -91,8 +91,3 @@
     headers.path = $${PREFIX}/include/tiled
     INSTALLS += headers
 }
-
-macx {
-    contains(QT_CONFIG, ppc):CONFIG += x86 \
-        ppc
-}
