--- CMakeLists.txt	2019-02-07 00:46:19.000000000 +0800
+++ CMakeLists.txt	2025-12-03 17:32:40.000000000 +0800
@@ -97,7 +97,8 @@
 endif()
 
 if(EVHTP_ALLOCATOR STREQUAL "jemalloc")
-    find_package(jemalloc)
+    find_package(PkgConfig)
+    pkg_check_modules(JEMALLOC jemalloc)
     if(JEMALLOC_FOUND)
         list(APPEND LIBEVHTP_EXTERNAL_LIBS jemalloc)
         list(APPEND package_deps jemalloc)
