--- videodl/modules/utils/misc.py.orig	2026-01-16 19:29:05.000000000 +0800
+++ videodl/modules/utils/misc.py	2026-01-17 05:03:03.000000000 +0800
@@ -244,7 +244,8 @@
         outputs = {'ext': 'NULL', 'sniffer': 'NULL', 'ok': False}
         # urllib.parse
         if not skip_urllib_parse:
-            ext = os.path.splitext(urlparse(url).path)[1].removeprefix('.')
+            ext_with_dot = os.path.splitext(urlparse(url).path)[1]
+            ext = ext_with_dot.removeprefix('. ') if ext_with_dot else ''
             if ext:
                 outputs.update(dict(ext=ext, sniffer='urllib.parse', ok=True))
                 return outputs
@@ -252,7 +253,8 @@
         resp = requests.head(url, allow_redirects=True, **request_overrides)
         content_type = resp.headers.get('Content-Type', '').split(';')[0]
         if content_type:
-            ext = mimetypes.guess_extension(content_type).removeprefix('.')
+            ext_raw = mimetypes.guess_extension(content_type)
+            ext = ext_raw.removeprefix('. ') if ext_raw else ''
             if ext:
                 outputs.update(dict(ext=ext, sniffer='requests.head', ok=True))
                 return outputs
@@ -260,7 +262,8 @@
         resp = requests.get(url, allow_redirects=True, stream=True, **request_overrides)
         content_type = resp.headers.get('Content-Type', '').split(';')[0]
         if content_type:
-            ext = mimetypes.guess_extension(content_type).removeprefix('.')
+            ext_raw = mimetypes.guess_extension(content_type)
+            ext = ext_raw.removeprefix('. ') if ext_raw else ''
             if ext:
                 outputs.update(dict(ext=ext, sniffer='requests.get.stream', ok=True))
                 return outputs
