--- third_party/libyuv/include/libyuv/row.h.orig	2026-09-15 19:13:34.000000000 -0600
+++ third_party/libyuv/include/libyuv/row.h	2026-09-15 19:22:26.000000000 -0600
@@ -44,7 +44,9 @@
 // GCC >= 4.7.0 required for AVX2.
 #if defined(__GNUC__) && (defined(__x86_64__) || defined(__i386__))
 #if (__GNUC__ > 4) || (__GNUC__ == 4 && (__GNUC_MINOR__ >= 7))
+#if !defined(DISABLE_AVX2)
 #define GCC_HAS_AVX2 1
+#endif  // DISABLE_AVX2
 #endif  // GNUC >= 4.7
 #endif  // __GNUC__
 
@@ -277,7 +279,6 @@
 #define HAS_I400TOARGBROW_SSE2
 #define HAS_I422TOAR30ROW_SSSE3
 #define HAS_MERGERGBROW_SSSE3
-#define HAS_MIRRORUVROW_AVX2
 #define HAS_MIRRORUVROW_SSSE3
 #define HAS_RAWTORGBAROW_SSSE3
 #define HAS_RGB24MIRRORROW_SSSE3
@@ -307,6 +308,7 @@
 #define HAS_I422TOUYVYROW_AVX2
 #define HAS_I422TOYUY2ROW_AVX2
 #define HAS_MERGEUVROW_16_AVX2
+#define HAS_MIRRORUVROW_AVX2
 #define HAS_MULTIPLYROW_16_AVX2
 #define HAS_RGBATOYJROW_AVX2
 #define HAS_SWAPUVROW_AVX2
--- third_party/libyuv/include/libyuv/scale_row.h.orig	2026-09-15 20:10:07.000000000 -0600
+++ third_party/libyuv/include/libyuv/scale_row.h	2026-09-15 20:11:08.000000000 -0600
@@ -36,7 +36,9 @@
 // GCC >= 4.7.0 required for AVX2.
 #if defined(__GNUC__) && (defined(__x86_64__) || defined(__i386__))
 #if (__GNUC__ > 4) || (__GNUC__ == 4 && (__GNUC_MINOR__ >= 7))
+#if !defined(DISABLE_AVX2)
 #define GCC_HAS_AVX2 1
+#endif  // DISABLE_AVX2
 #endif  // GNUC >= 4.7
 #endif  // __GNUC__
 
--- third_party/libyuv/source/row_gcc.cc.orig	2026-09-15 19:35:07.000000000 -0600
+++ third_party/libyuv/source/row_gcc.cc	2026-09-15 19:36:53.000000000 -0600
@@ -7100,6 +7100,7 @@
       : "memory", "cc", "xmm0", "xmm1", "xmm2", "xmm3", "xmm4", "xmm5");
 }
 
+#if defined(HAS_HALFMERGEUVROW_AVX2)
 void HalfMergeUVRow_AVX2(const uint8_t* src_u,
                          int src_stride_u,
                          const uint8_t* src_v,
@@ -7146,6 +7147,7 @@
         "r"((intptr_t)(src_stride_v))   // %5
       : "memory", "cc", "xmm0", "xmm1", "xmm2", "xmm3", "xmm4", "xmm5");
 }
+#endif
 
 void ClampFloatToZero_SSE2(const float* src_x, float* dst_y, int width) {
   asm volatile(
