--- similarity_search/test/testdataset.h	2026-01-06 07:39:56.000000000 +0800
+++ similarity_search/test/testdataset.h	2026-01-11 23:05:35.000000000 +0800
@@ -24,7 +24,7 @@
 
 using std::string;
 
-const string sampleDataPrefix = string("..") + PATH_SEPARATOR + string("sample_data") + PATH_SEPARATOR;
+const string sampleDataPrefix = string("..") + PATH_SEPARATOR + string("..") + PATH_SEPARATOR + string("sample_data") + PATH_SEPARATOR;
 
 class TestDataset {
  public:
