--- src/filezilla/rundown_protection.hpp.orig	2025-11-18 22:40:06.000000000 +0800
+++ src/filezilla/rundown_protection.hpp	2026-01-29 19:21:11.000000000 +0800
@@ -9,7 +9,9 @@
 class rundown_protection
 {
 	using state_type = uint64_t;
+#if !defined(__i386__) && !defined(__ppc__)
 	static_assert(std::atomic<state_type>::is_always_lock_free, "std::atomic<state_type> must be always lock free");
+#endif
 
 public:
 	rundown_protection() = default;
