--- Makefile.am	2025-07-16 21:07:57.000000000 +0800
+++ Makefile.am	2026-03-19 01:56:31.000000000 +0800
@@ -21,7 +21,7 @@
 endif
 
 if FZ_MAC
-PKG_DIR = mac
+PKG_DIR = unix
 endif
 
 PKG_TARGETS = \

--- pkg/Makefile.am	2026-03-19 02:01:54.000000000 +0800
+++ pkg/Makefile.am	2026-03-19 01:56:50.000000000 +0800
@@ -7,7 +7,7 @@
 endif
 
 if FZ_MAC
-SUBDIRS = mac
+SUBDIRS = unix
 endif
 
 dist_noinst_DATA = \

--- res/share/icons/hicolor/Makefile.am	2023-04-24 21:53:39.000000000 +0800
+++ res/share/icons/hicolor/Makefile.am	2026-03-19 01:57:51.000000000 +0800
@@ -1,4 +1,4 @@
-if FZ_UNIX
+if FZ_MAC
 
 logo16dir = $(datadir)/icons/hicolor
 nobase_dist_logo16_DATA = 16x16/apps/filezilla-server-gui.png

--- src/filezilla/webui/app/Makefile.am	2024-08-07 23:13:13.000000000 +0800
+++ src/filezilla/webui/app/Makefile.am	2026-03-19 03:11:10.000000000 +0800
@@ -52,9 +52,9 @@
 
 if ENABLE_FZ_WEBUI
 install-data-hook:
-	@mkdir -p $(datadir)/filezilla-server/
-	@rm -rf $(datadir)/filezilla-server/webui
-	@cp -a $(srcdir)/dist/spa $(datadir)/filezilla-server/webui
+	@mkdir -p $(DESTDIR)$(datadir)/filezilla-server/
+	@rm -rf $(DESTDIR)$(datadir)/filezilla-server/webui
+	@cp -a $(srcdir)/dist/spa $(DESTDIR)$(datadir)/filezilla-server/webui
 
 uninstall-hook:
 	@rm -rf $(datadir)/filezilla-server/webui

--- src/gui/main.cpp	2025-08-12 21:24:32.000000000 +0800
+++ src/gui/main.cpp.new	2026-03-19 02:00:06.000000000 +0800
@@ -181,7 +181,7 @@
 
 static wxIcon get_logo(wxCoord coord)
 {
-#if defined(FZ_UNIX)
+#if defined(FZ_UNIX) || defined (FZ_MAC)
 	static auto search_dirs =
 		(fz::known_paths::data::home() +
 		(fz::util::get_own_executable_directory().parent() / fzT("share")) +
@@ -194,8 +194,6 @@
 		return {};
 
 	return wxIcon(fz::to_wxString(path.str()), wxBITMAP_TYPE_PNG);
-#elif defined (FZ_MAC)
-	return wxIcon(wxT("filezilla-server-gui"), wxBITMAP_TYPE_ICON_RESOURCE, coord, coord);
 #elif defined (FZ_WINDOWS)
 	return wxIcon(wxT("appicon"), wxBITMAP_TYPE_ICO_RESOURCE, coord, coord);
 #else

--- src/server/main.cpp	2025-11-18 22:40:06.000000000 +0800
+++ src/server/main.cpp	2026-03-19 02:00:38.000000000 +0800
@@ -371,7 +371,7 @@
 			#endif
 
 			if (webui_app_root_list.empty()) {
-				#ifdef FZ_UNIX
+				#if defined(FZ_UNIX) || defined (FZ_MAC)
 					webui_app_root_list = (fz::known_paths::data::dirs()/fzT("filezilla-server")/fzT("webui"));
 				#else
 					webui_app_root_list = (fz::util::get_own_executable_directory()/fzT("webui"));

--- src/filezilla/known_paths.cpp	2024-08-07 23:13:13.000000000 +0800
+++ src/filezilla/known_paths.cpp	2026-03-19 01:59:11.000000000 +0800
@@ -157,6 +157,8 @@
 	static const path_list pl = get(var("XDG_DATA_DIRS", "XDG_DATA_DIRS", folderid::local_appdata), absolute_filter, []() -> path_list {
 	#if defined(FZ_UNIX)
 		return absolute_filter("/usr/local/share/:/usr/share/");
+	#elif defined(FZ_MAC)
+		return absolute_filter("@PREFIX@/share/:/usr/share/");
 	#elif defined(FZ_WINDOWS)
 		return absolute_filter(util::get_own_executable_directory());
 	#else
