--- src/filezilla/authentication/system_identity_provider.cpp.orig	2025-11-18 22:40:06.000000000 +0800
+++ src/filezilla/authentication/system_identity_provider.cpp	2026-09-10 16:54:55.000000000 +0800
@@ -20,7 +20,7 @@
 	auto account = fz::to_native_from_utf8(username);
 #endif
 
-	if (auto uid = fz::get_user_uid(account); !uid.empty()) {
+	if (auto uid = fz::get_user_id(account); !uid.empty()) {
 		uinfo.uid = std::move(uid);
 		uinfo.account = std::move(account);
 
