From 1021c6cfd20dd6ed44de430c56f85ccd0f75b41b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 26 Mar 2025 03:52:55 +0800
Subject: [PATCH 2/4] readpassphrase: fix for undefined _NSIG

---
 src/lib/libc/gen/readpassphrase.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git src/lib/libc/gen/readpassphrase.c src/lib/libc/gen/readpassphrase.c
index 3837b08..9a16198 100644
--- src/lib/libc/gen/readpassphrase.c
+++ src/lib/libc/gen/readpassphrase.c
@@ -32,6 +32,12 @@
 #include <unistd.h>
 #include <readpassphrase.h>
 
+#ifdef __APPLE__
+#ifndef _NSIG
+#define _NSIG NSIG
+#endif
+#endif
+
 static volatile sig_atomic_t signo[_NSIG];
 
 static void handler(int);
-- 
2.48.0

