From d9ba0eafaaa136ee9f3953208ff18cf4282ebd6b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 26 Mar 2025 03:56:04 +0800
Subject: [PATCH 3/4] Makefile: add LDFLAGS

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git Makefile Makefile
index 1ea5d52..ecf00fd 100644
--- Makefile
+++ Makefile
@@ -65,13 +65,14 @@ CFLAGS+= -Wall -Wextra
 CFLAGS+= -Wno-pointer-sign
 CFLAGS+= -Wno-sign-compare
 CFLAGS+= -Wno-unused-parameter
+LDFLAGS =
 
 ### targets
 
 .PHONY: fetch hash-helpers clean install check test up check-updates ls
 
 signify: ${LOCAL_SRCS} ${SRCS} ${INCL}
-	${CC} ${CFLAGS} -o $@ ${SRCS} ${LOCAL_SRCS}
+	${CC} ${CFLAGS} -o $@ ${SRCS} ${LOCAL_SRCS} ${LDFLAGS}
 	cp src/usr.bin/signify/signify.1 .
 
 hash-helpers: ${HASH_HELPERS}
-- 
2.48.0

