--- wscript
+++ wscript	2025-12-02 00:19:11.000000000 +0800
@@ -307,6 +295,8 @@
         conf.env.CFLAGS_WXPY.append('-I@PREFIX@/include')
         conf.env.CXXFLAGS_WXPY.append('-I@PREFIX@/include')
 
+        conf.env.CXXFLAGS_WXPY.append('-D_GLIBCXX_USE_CXX11_ABI=0')
+
         # And if --debug is set turn on more detailed debug info and turn off optimization
         if conf.env.debug:
             for flags in [conf.env.CFLAGS_PYEXT, conf.env.CXXFLAGS_PYEXT]:
