--- wscript
+++ wscript	2025-11-24 18:29:59
@@ -126,9 +126,6 @@
         _copyEnvGroup(conf.env, '_WX', '_WXGL')
         conf.env.LIB_WXGL += cfg.makeLibName('gl')
 
-        _copyEnvGroup(conf.env, '_WX', '_WXWEBVIEW')
-        conf.env.LIB_WXWEBVIEW += cfg.makeLibName('webview')
-
         _copyEnvGroup(conf.env, '_WX', '_WXXML')
         conf.env.LIB_WXXML += cfg.makeLibName('xml', isMSWBase=True)
 
@@ -226,16 +223,7 @@
                        uselib_store='WXGL', mandatory=True,
                        msg='Finding libs for WXGL')
 
-        if cfg.checkSetup(wxConfigDir, 'wxUSE_WEBVIEW'):
-            wv_libs = '--libs webview,core,net'
-        else:
-            wv_libs = '--libs core,net'
         conf.check_cfg(path=conf.options.wx_config, package='',
-                       args='--cxxflags ' + wv_libs + rpath,
-                       uselib_store='WXWEBVIEW', mandatory=True,
-                       msg='Finding libs for WXWEBVIEW')
-
-        conf.check_cfg(path=conf.options.wx_config, package='',
                        args='--cxxflags --libs xml,core,net' + rpath,
                        uselib_store='WXXML', mandatory=True,
                        msg='Finding libs for WXXML')
@@ -564,7 +552,6 @@
     makeETGRule(bld, 'etg/_stc.py',        '_stc',       'WXSTC')
     makeETGRule(bld, 'etg/_html.py',       '_html',      'WXHTML')
     makeETGRule(bld, 'etg/_glcanvas.py',   '_glcanvas',  'WXGL')
-    makeETGRule(bld, 'etg/_html2.py',      '_html2',     'WXWEBVIEW')
     makeETGRule(bld, 'etg/_xml.py',        '_xml',       'WXXML')
     makeETGRule(bld, 'etg/_xrc.py',        '_xrc',       'WXXRC')
     makeETGRule(bld, 'etg/_richtext.py',   '_richtext',  'WXHTML WXRICHTEXT')
