From 52e127395a1d1f057e82090b2ff4483626ad5a5e Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:35:58 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_inline_test.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ppx_inline_test.ml b/src/ppx_inline_test.ml
index 4ee75fe..2ffa92f 100644
--- src/ppx_inline_test.ml
+++ src/ppx_inline_test.ml
@@ -269,6 +269,7 @@ module E = struct
       (pstr_value
          nonrecursive
          (value_binding
+            ~constraint_:drop
             ~pat:
               (map
                  (Attribute.pattern tags (opt_name ()))
