From 137b069525e798386b186886e01b435b656e248a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 5 May 2025 08:48:58 +0800
Subject: [PATCH] Freelan_gui_settings.cpp: minor fix

---
 src/Freelan_gui_settings.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/Freelan_gui_settings.cpp src/Freelan_gui_settings.cpp
index 307c6c3..654f6ed 100644
--- src/Freelan_gui_settings.cpp
+++ src/Freelan_gui_settings.cpp
@@ -823,7 +823,7 @@ bool Freelan_gui_settings::write( QIODevice* const settings_io ) const
 			settings_stream << '[' << prefix << "]\n";
 		}
 
-		const QHash < QString, QList < QVariant >>& value_hash = m_values[ prefix ];
+		const QHash < QString, QList < QVariant > >& value_hash = m_values[ prefix ];
 
 		// For each key
 		const QList< QString > keys = value_hash.uniqueKeys();
