--- gdb/macosx/macosx-nat-dyld-process.c	2021-10-06 13:19:07.000000000 +0800
+++ gdb/macosx/macosx-nat-dyld-process.c	2025-08-28 12:27:47.000000000 +0800
@@ -404,7 +404,9 @@
     case MH_DYLIB:
       break;
     case MH_BUNDLE:
+#ifdef MH_KEXT_BUNDLE
     case MH_KEXT_BUNDLE:
+#endif
       break;
     default:
       return;

--- gdb/macosx/macosx-nat-dyld.c	2021-10-06 13:19:07.000000000 +0800
+++ gdb/macosx/macosx-nat-dyld.c	2025-08-28 12:28:13.000000000 +0800
@@ -2066,7 +2066,9 @@
     case MH_DYLIB:
     case MH_DYLINKER:
     case MH_BUNDLE:
+#ifdef MH_KEXT_BUNDLE
     case MH_KEXT_BUNDLE:
+#endif
       break;
     case MH_FVMLIB:
     case MH_PRELOAD:
@@ -2390,7 +2392,9 @@
       break;
     case MH_DYLINKER:
     case MH_BUNDLE:
+#ifdef MH_KEXT_BUNDLE
     case MH_KEXT_BUNDLE:
+#endif
       entry->reason = dyld_reason_dyld;
       break;
     default:
