--- bfd/config.bfd	2021-10-06 13:19:07.000000000 +0800
+++ bfd/config.bfd	2025-08-28 14:20:17.000000000 +0800
@@ -1046,13 +1046,13 @@
     targ_selvecs="bfd_elf32_powerpcle_vec ppcboot_vec"
     targ64_selvecs="bfd_elf64_powerpc_vec bfd_elf64_powerpcle_vec"
     ;;
-  powerpc-*-darwin* | powerpc-*-macos10* | powerpc-*-rhapsody*)
+  powerpc*-*-darwin* | powerpc*-*-macos10* | powerpc*-*-rhapsody*)
     targ_defvec=mach_o_be_vec
     targ_selvecs="mach_o_be_vec mach_o_le_vec mach_o_fat_vec pef_vec pef_xlib_vec sym_vec"
     targ_archs="bfd_powerpc_arch bfd_rs6000_arch bfd_i386_arch"
     targ_underscore=yes
     ;;
-  powerpc-*-macos*)
+  powerpc*-*-macos*)
     targ_defvec=pmac_xcoff_vec
     ;;
   powerpc-*-lynxos*)
