From 13b2316b67db24489e319c26f75b5d75ce44f1f2 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 26 Sep 2025 21:20:31 +0800
Subject: [PATCH] mum.h: fix incorrect bswap macros

---
 mum.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/mum.h b/mum.h
index f3715bc..10b3dd5 100644
--- contrib/mumhash/mum.h
+++ contrib/mumhash/mum.h
@@ -126,12 +126,12 @@ static _MUM_INLINE uint64_t _mum (uint64_t v, uint64_t p) {
 }
 
 #if defined(_MSC_VER)
-#define _mum_bswap_32(x) _byteswap_uint32_t (x)
-#define _mum_bswap_64(x) _byteswap_uint64_t (x)
+#define _mum_bswap32(x) _byteswap_uint32_t (x)
+#define _mum_bswap64(x) _byteswap_uint64_t (x)
 #elif defined(__APPLE__)
 #include <libkern/OSByteOrder.h>
-#define _mum_bswap_32(x) OSSwapInt32 (x)
-#define _mum_bswap_64(x) OSSwapInt64 (x)
+#define _mum_bswap32(x) OSSwapInt32 (x)
+#define _mum_bswap64(x) OSSwapInt64 (x)
 #elif defined(__GNUC__)
 #define _mum_bswap32(x) __builtin_bswap32 (x)
 #define _mum_bswap64(x) __builtin_bswap64 (x)
