--- src/libserver/rspamd_control.c	2025-09-17 18:36:36.000000000 +0800
+++ src/libserver/rspamd_control.c	2025-09-26 22:30:27.000000000 +0800
@@ -768,7 +768,7 @@
 	static struct rspamd_control_command cmd;
 	static struct msghdr msg;
 	static struct iovec iov;
-	static unsigned char fdspace[CMSG_SPACE(sizeof(int))];
+	unsigned char fdspace[CMSG_SPACE(sizeof(int))];
 	int rfd = -1;
 	gssize r;
 
@@ -987,7 +987,7 @@
 	struct msghdr msg;
 	struct cmsghdr *cmsg;
 	static struct iovec iov;
-	static unsigned char fdspace[CMSG_SPACE(sizeof(int))];
+	unsigned char fdspace[CMSG_SPACE(sizeof(int))];
 	int *spair, rfd = -1;
 	char *nid;
 	struct rspamd_control_command wcmd;
