From fd907d3842e312a668c1fcbf37bfb7a747332014 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 2 Feb 2025 00:09:31 +0800
Subject: [PATCH] Fix linking on macOS: needs zlib

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git configure.ac configure.ac
index f223a24..62975cb 100755
--- configure.ac
+++ configure.ac
@@ -141,7 +141,7 @@ dnl At least staticly include OpenSSL libs
 if test x"${STATIC}" = xyes; then
 	case "$host" in
 	*-*-darwin*)
-		LDADD_STATIC="${STATIC_LIBSSLDIR}/libssl.a ${STATIC_LIBSSLDIR}/libcrypto.a"
+		LDADD_STATIC="${STATIC_LIBSSLDIR}/libssl.a ${STATIC_LIBSSLDIR}/libcrypto.a @PREFIX@/lib/libz.a"
 		AC_DEFINE(HAVE_LIBSSL, 1, [Define to 1 if you have the `ssl' library (-lssl)])
 		AC_DEFINE(HAVE_LIBCRYPTO, 1, [Define to 1 if you have the `crypto' library (-lcrypto)])
 		STATIC_SSL="yes"
