--- CMakeLists.txt	2025-03-24 07:50:19.000000000 +0800
+++ CMakeLists.txt	2025-03-25 22:11:07.000000000 +0800
@@ -5,17 +5,9 @@
 )
 
 # --- Dependencies--------------------------------------------------------------
-include(cmake/find_or_fetch_package.cmake)
 
-find_or_fetch_package(ftxui v6.0.0
-  https://github.com/ArthurSonzogni/ftxui
-  v6.0.0
-)
-
-find_or_fetch_package(subprocess v0.4.0
-  https://github.com/benman64/subprocess
-  e1cae5e86e5d089e65e906f7c13917b7bbb75d04
-)
+find_package(ftxui)
+find_package(subprocess)
 
 # ------------------------------------------------------------------------------
 