From 229efa42ffc2d0623062f1646f72ad9beb71f2cc Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 5 Jan 2025 00:31:27 +0800
Subject: [PATCH 2/2] scal.S: use r11 on 32-bit Darwin on powerpc

---
 kernel/power/scal.S | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git kernel/power/scal.S kernel/power/scal.S
index 5e92a88aa..eceb9fe8e 100644
--- kernel/power/scal.S
+++ kernel/power/scal.S
@@ -59,7 +59,7 @@
 #if !defined(__64BIT__) && defined(DOUBLE)
 #define X r8
 #define INCX r9
-#define FLAG r13
+#define FLAG r11
 #else
 #define X r7
 #define INCX r8
@@ -91,7 +91,7 @@
 	fcmpu	cr0, FZERO, ALPHA
 	bne-	cr0, LL(A1I1)
 
-	ld      FLAG,    48+64+8(SP)
+	LDLONG  FLAG,    48+64+8(SP)
 	cmpwi   cr0, FLAG, 1
 	beq-   cr0, LL(A1I1)
 
-- 
2.47.1

