From e1caff6772bca5fa9f3449d17a5a2d2edf9c206a Mon Sep 17 00:00:00 2001
From: Erik Doernenburg <erik@doernenburg.com>
Date: Sun, 28 Jul 2019 22:11:35 +0200
Subject: [PATCH] Preferences are always brought to the front now.

---
 AppController.m | 1 +
 1 file changed, 1 insertion(+)

diff --git a/AppController.m b/AppController.m
index 3c4af7c..c5309a1 100644
--- AppController.m
+++ AppController.m
@@ -58,6 +58,7 @@ - (void)applicationWillTerminate:(NSNotification *)notification
 
 - (void)openPreferences:(id)sender
 {
+    [NSApp activateIgnoringOtherApps:YES];
     [[PreferencesController sharedInstance] showWindow:sender];
 }
 