--- docs/reference/dist-docs.sh	2025-10-14 02:52:32.000000000 +0800
+++ docs/reference/dist-docs.sh	2026-05-07 13:24:07.000000000 +0800
@@ -5,4 +5,4 @@
 ninja
 
 # And copy the resulting devhelp documentation into the dist location
-cp -a $docs_path ${MESON_DIST_ROOT}/docs/reference/doc/
+cp -pRP $docs_path ${MESON_DIST_ROOT}/docs/reference/doc/

--- src/libtinysparql/symlink-libtracker-sparql.sh	2025-10-14 02:52:32.000000000 +0800
+++ src/libtinysparql/symlink-libtracker-sparql.sh	2026-05-07 13:31:45.000000000 +0800
@@ -15,5 +15,5 @@
 
   rm ${srcdir}/${libtracker_sparql_so_filename}
   ln -s ${so_filename} ${srcdir}/${libtracker_sparql_so_filename}
-  cp -a ${srcdir}/${libtracker_sparql_so_filename} ${MESON_INSTALL_DESTDIR_PREFIX}/${libdir}/
+  cp -pRP ${srcdir}/${libtracker_sparql_so_filename} ${MESON_INSTALL_DESTDIR_PREFIX}/${libdir}/
 done
