From 3f21033b86c096eb788e5932b045200d9dab3bc4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Wed, 15 Jan 2025 05:37:23 +0800
Subject: [PATCH] filesystem-disk-unix.c++: add a missing header for Apple

---
 c++/src/kj/filesystem-disk-unix.c++ | 4 ++++
 1 file changed, 4 insertions(+)

diff --git c++/src/kj/filesystem-disk-unix.c++ c++/src/kj/filesystem-disk-unix.c++
index 74d59fd9..d70324f4 100644
--- src/kj/filesystem-disk-unix.c++
+++ src/kj/filesystem-disk-unix.c++
@@ -53,6 +53,10 @@
 #include <sys/sendfile.h>
 #endif
 
+#if __APPLE__
+#include <sys/stdio.h> // renameat
+#endif
+
 namespace kj {
 namespace {
 
