--- orc/orccompiler.c.orig	2026-08-11 17:25:11.000000000 -0600
+++ orc/orccompiler.c	2026-08-11 17:32:52.000000000 -0600
@@ -15,7 +15,11 @@
 /* TARGET_OS_OSX is not defined on older macOS, but is always defined
    to 0 for Apple mobile targets. */
 #if !defined(TARGET_OS_OSX) || TARGET_OS_OSX
-#include <libkern/OSCacheControl.h>
+#include <AvailabilityMacros.h>
+// Not available in Tiger.
+#if MAC_OS_X_VERSION_MIN_REQUIRED > 1040
+#include  <libkern/OSCacheControl.h>
+#endif
 #endif
 #endif
 
