--- orc/orccpu-x86.c.orig	2026-08-11 22:06:52.000000000 -0600
+++ orc/orccpu-x86.c	2026-08-11 22:12:09.000000000 -0600
@@ -248,7 +248,15 @@
 static orc_uint32 ORC_TARGET_XSAVE
 get_xcr0 (void)
 {
-  return _xgetbv(0);
+#if defined(__APPLE__) && defined(__i386__)
+// This is expecting _xgetbv in some form to exist so that features can be
+// checked for, but on older Intel CPUs it does not, so return feature can
+// not be used by returning zero always.
+// https://stackoverflow.com/questions/72522885/are-the-xgetbv-and-cpuid-checks-sufficient-to-guarantee-avx2-support
+    return 0;
+#else  
+    return _xgetbv(0);
+#endif
 }
 #endif
 
