--- Makefile.in	2022-01-31 20:36:30.000000000 +0800
+++ Makefile.in	2025-01-02 00:08:39.000000000 +0800
@@ -83,21 +83,21 @@
 distclean: dist-clean
 
 .cc.lo:
-	$(SILENTCXXLT) $(LIBTOOL) $(SILENCELT) --mode=compile --tag=CXX $(CXX) -shared $(CXXFLAGS) $(CONFIGFLAGS) \
+	$(SILENTCXXLT) $(LIBTOOL) --mode=compile --tag=CXX $(CXX) -shared $(CXXFLAGS) $(CONFIGFLAGS) \
 		$(GETTEXTFLAGS) -DLOCALEDIR=\"$(LOCALEDIR)\" -Isrc -DT3_WIDGET_BUILD_DSO -D_T3_WIDGET_INTERNAL \
 		-DX11_MOD_NAME=\"$(libdir)/libt3widget/libt3widget2-x11.mod\" -c -o $@ $<
 
 src/x11.lo: src/x11.cc
-	$(SILENTCXXLT) $(LIBTOOL) $(SILENCELT) --mode=compile --tag=CXX $(CXX) -shared $(CXXFLAGS) $(CONFIGFLAGS) \
+	$(SILENTCXXLT) $(LIBTOOL) --mode=compile --tag=CXX $(CXX) -shared $(CXXFLAGS) $(CONFIGFLAGS) \
 		$(GETTEXTFLAGS) -DLOCALEDIR=\"$(LOCALEDIR)\" -Isrc -DT3_WIDGET_BUILD_DSO -D_T3_WIDGET_INTERNAL \
 		$(X11_FLAGS) -c -o $@ $<
 
 src/libt3widget.la: $(OBJECTS)
-	$(SILENTLDLT) $(LIBTOOL) $(SILENCELT) --mode=link --tag=CXX $(CXX) -shared -version-info 2:0:0 \
+	$(SILENTLDLT) $(LIBTOOL) --mode=link --tag=CXX $(CXX) -shared -version-info 2:0:0 \
 		$(CXXFLAGS) $(LDFLAGS) -o $@ $(OBJECTS) $(LDLIBS) $(CONFIGLIBS) $(GETTEXTLIBS) -rpath $(libdir)
 
 src/libt3widget2-x11.la: src/x11.lo src/libt3widget.la
-	$(SILENTLDLT) $(LIBTOOL) $(SILENCELT) --mode=link --tag=CXX $(CXX) -shared -module -avoid-version \
+	$(SILENTLDLT) $(LIBTOOL) --mode=link --tag=CXX $(CXX) -shared -module -avoid-version \
 		-shrext .mod $(CXXFLAGS) $(LDFLAGS) -o $@ src/x11.lo $(LDLIBS) $(CONFIGLIBS) $(X11_LIBS) $(GETTEXTLIBS) \
 		-rpath $(libdir)
 
