--- libs-0.21.0/userspace/libsinsp/event.cpp	2025-05-19 17:24:01.000000000 +0800
+++ libs-0.21.0/userspace/libsinsp/event.cpp	2026-05-20 16:45:29.000000000 +0800
@@ -19,6 +19,9 @@
 #include <libsinsp/sinsp_errno.h>
 
 #ifndef _WIN32
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <inttypes.h>
 #include <sys/socket.h>
 #include <algorithm>

--- libs-0.21.0/userspace/libsinsp/version.h	2025-05-19 17:24:01.000000000 +0800
+++ libs-0.21.0/userspace/libsinsp/version.h	2026-05-20 16:49:15.000000000 +0800
@@ -18,6 +18,10 @@
 
 #pragma once
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
 #include <vector>
 #include <string>
 #include <unordered_map>
