--- libs-0.21.0/driver/syscall_compat.h	2025-05-19 17:24:01.000000000 +0800
+++ libs-0.21.0/driver/syscall_compat.h	2026-05-20 16:38:55.000000000 +0800
@@ -6,7 +6,7 @@
 #include "syscall_compat_aarch64.h"
 #elif defined(__s390x__)
 #include "syscall_compat_s390x.h"
-#elif defined(__powerpc__)
+#elif defined(__powerpc__) || defined(__POWERPC__)
 #include "syscall_compat_ppc64le.h"
 #elif defined(__riscv)
 #include "syscall_compat_riscv64.h"
