--- a/image/decode/segdec.c	2026-05-21 13:19:41.000000000 +0800
+++ b/image/decode/segdec.c	2026-05-23 12:51:03.000000000 +0800
@@ -52,7 +52,7 @@
 //================================================================
 // Memory access functions
 //================================================================
-#if defined(__unix__) || defined(__APPLE__) // fix compilation error under macOS and Linux
+#ifndef _byteswap_ulong
 extern U32 _byteswap_ulong(U32 bits);
 #endif
 
