--- setup.py.orig	2023-11-19 18:04:43.000000000 +0800
+++ setup.py	2026-04-29 02:56:03.000000000 +0800
@@ -7,6 +7,9 @@
 CONFIG_VERSION = open("upstream-quickjs/VERSION").read().strip()
 extra_link_args: List[str] = []
 
+if sys.platform == "darwin":
+    extra_link_args = ["-lMacportsLegacySupport"]
+
 if sys.platform == "win32":
     # To build for Windows:
     # 1. Install MingW-W64-builds from https://mingw-w64.org/doku.php/download
