--- a/configure.py
+++ b/configure.py
@@ -991,7 +991,10 @@
                 pysh_lib = "python%d%d%s.dll" % (py_major, py_minor, abi)
             else:
                 # Use distutils to get the additional configuration.
-                from distutils.sysconfig import get_config_vars
+                try:
+                    from distutils.sysconfig import get_config_vars
+                except ImportError:
+                    from sysconfig import get_config_vars
                 ducfg = get_config_vars()
 
                 config_args = ducfg.get("CONFIG_ARGS", "")
