From c7726e00545ef5a0c81208455cc54628107a05bf Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 11 Oct 2025 12:33:30 +0800
Subject: [PATCH] auhal.c: fix compilation

---
 modules/audio_output/auhal.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git modules/audio_output/auhal.c modules/audio_output/auhal.c
index 38dbf39c9e..afa6fec199 100644
--- modules/audio_output/auhal.c
+++ modules/audio_output/auhal.c
@@ -36,6 +36,13 @@
 #pragma mark -
 #pragma mark local prototypes & module descriptor
 
+#ifndef AudioFormatID
+typedef UInt32 AudioFormatID;
+#endif
+
+#ifndef kAudioObjectPropertyScopeOutput
+#define kAudioObjectPropertyScopeOutput kAudioDevicePropertyScopeOutput
+#endif
 
 #define AOUT_VOLUME_DEFAULT             256
 #define AOUT_VOLUME_MAX                 512
@@ -296,7 +303,11 @@ IsAudioFormatDigital(AudioFormatID id)
         case 'iac3':
         case kAudioFormat60958AC3:
         case kAudioFormatAC3:
+#ifdef kAudioFormatEnhancedAC3
         case kAudioFormatEnhancedAC3:
+#else
+        case 'ec-3':
+#endif
             return true;
         default:
             return false;
