From 0385fff8995cd0c62429fd313b4c21ba64c301e5 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sat, 11 Oct 2025 11:09:53 +0800
Subject: [PATCH 3/4] gui/minimal_macosx/misc.m

---
 modules/gui/minimal_macosx/misc.m | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git modules/gui/minimal_macosx/misc.m modules/gui/minimal_macosx/misc.m
index e57e9fe261..63fda0c28c 100644
--- modules/gui/minimal_macosx/misc.m
+++ modules/gui/minimal_macosx/misc.m
@@ -24,6 +24,8 @@
  *****************************************************************************/
 
 #import "misc.h"
+#import <AppKit/AppKit.h>
+#import <ApplicationServices/ApplicationServices.h>
 
 @implementation NSScreen (VLCAdditions)
 
@@ -36,12 +38,14 @@
 {
     NSRect screen_frame = [self frame];
     NSRect screen_visible_frame = [self visibleFrame];
-    CGFloat f_menu_bar_thickness = [self hasMenuBar] ? [[NSStatusBar systemStatusBar] thickness] : 0.0;
+
+    // NSStatusBar thickness only available 10.7+, hardcode fallback for 10.6
+    CGFloat f_menu_bar_thickness = 22.0;
 
     BOOL b_found_dock = NO;
     if (screen_visible_frame.size.width < screen_frame.size.width)
         b_found_dock = YES;
-    else if (screen_visible_frame.size.height + f_menu_bar_thickness < screen_frame.size.height)
+    else if (screen_visible_frame.size.height + ( [self hasMenuBar] ? f_menu_bar_thickness : 0.0 ) < screen_frame.size.height)
         b_found_dock = YES;
 
     return b_found_dock;
@@ -49,7 +53,8 @@
 
 - (CGDirectDisplayID)displayID
 {
-    return (CGDirectDisplayID)[[[self deviceDescription] objectForKey: @"NSScreenNumber"] intValue];
+    NSNumber *screenNumber = [[self deviceDescription] objectForKey:@"NSScreenNumber"];
+    return (CGDirectDisplayID)[screenNumber intValue];
 }
 
 @end
-- 
2.51.0

