--- .home/.cargo/macports/nix-0.19.1/src/sys/socket/mod.rs	1973-11-30 05:33:09.000000000 +0800
+++ .home/.cargo/macports/nix-0.19.1/src/sys/socket/mod.rs	2026-01-04 22:47:48.000000000 +0800
@@ -1697,12 +1697,12 @@
 /// [Further reading](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getpeername.html)
 pub fn getpeername(fd: RawFd) -> Result<SockAddr> {
     unsafe {
-        let mut addr = mem::MaybeUninit::uninit();
+        let mut addr: mem::MaybeUninit<sockaddr_storage> = mem::MaybeUninit::uninit();
         let mut len = mem::size_of::<sockaddr_storage>() as socklen_t;
 
         let ret = libc::getpeername(
             fd,
-            addr.as_mut_ptr() as *mut libc::sockaddr,
+            addr.as_mut_ptr() as *mut sockaddr_storage as *mut libc::sockaddr,
             &mut len
         );
 
@@ -1717,12 +1717,12 @@
 /// [Further reading](https://pubs.opengroup.org/onlinepubs/9699919799/functions/getsockname.html)
 pub fn getsockname(fd: RawFd) -> Result<SockAddr> {
     unsafe {
-        let mut addr = mem::MaybeUninit::uninit();
+        let mut addr: mem::MaybeUninit<sockaddr_storage> = mem::MaybeUninit::uninit();
         let mut len = mem::size_of::<sockaddr_storage>() as socklen_t;
 
         let ret = libc::getsockname(
             fd,
-            addr.as_mut_ptr() as *mut libc::sockaddr,
+            addr.as_mut_ptr() as *mut sockaddr_storage as *mut libc::sockaddr,
             &mut len
         );
 
