--- .home/.cargo/macports/rayon-1.4.1/src/iter/flatten_iter.rs
+++ .home/.cargo/macports/rayon-1.4.1/src/iter/flatten_iter.rs	2026-01-05 23:01:11.000000000 +0800
@@ -117,8 +117,10 @@
     where
         I: IntoIterator<Item = T>,
     {
-        let iter = iter.into_iter().flatten();
-        let base = self.base.consume_iter(iter);
+        let mut base = self.base;
+        for item in iter {
+            base = base.consume_iter(item);
+        }
         FlattenIterFolder { base }
     }
 
