From da7fd9f9c1715a4391fa2e0074c1f3f42d50c49b Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 11 Apr 2026 17:53:07 +0700
Subject: [PATCH] curlnetworkmanager.cpp: avoid non-existing macros

---
 src/httpnetworkmanager/curlnetworkmanager.cpp | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/src/httpnetworkmanager/curlnetworkmanager.cpp b/src/httpnetworkmanager/curlnetworkmanager.cpp
index f71e552..fc1ef69 100644
--- a/src/httpnetworkmanager/curlnetworkmanager.cpp
+++ b/src/httpnetworkmanager/curlnetworkmanager.cpp
@@ -354,8 +354,19 @@ bool CurlNetworkManager::setupOptions(RequestInfo *requestInfo, const std::share
     }
 
     if (request->isExtraTLSPadding()) {
-        if (curl_easy_setopt(requestInfo->curlEasyHandle, CURLOPT_SSL_OPTIONS, CURLSSLOPT_TLSEXT_PADDING | CURLSSLOPT_TLSEXT_PADDING_SUPER) != CURLE_OK)
-            return false;
+        long ssl_opts = 0;
+
+#ifdef CURLSSLOPT_TLSEXT_PADDING
+        ssl_opts |= CURLSSLOPT_TLSEXT_PADDING;
+#endif
+#ifdef CURLSSLOPT_TLSEXT_PADDING_SUPER
+        ssl_opts |= CURLSSLOPT_TLSEXT_PADDING_SUPER;
+#endif
+
+        if (ssl_opts != 0) {
+            if (curl_easy_setopt(requestInfo->curlEasyHandle, CURLOPT_SSL_OPTIONS, ssl_opts) != CURLE_OK)
+                return false;
+        }
     }
 
     curl_easy_setopt(requestInfo->curlEasyHandle, CURLOPT_PRIVATE, new std::uint64_t(requestInfo->id));    // our user data, must be deleted in the RequestInfo destructor
