From ee2237a6d531b0ce50face317f94ee3d2f1099a9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 11 Apr 2026 18:34:57 +0700
Subject: [PATCH] CMakeLists.txt: fix linking for macOS

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 167e7a8..e486f30 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -102,7 +102,7 @@ endif()
 
 # Set platform specific dependencies
 if (APPLE)
-    set(OS_SPECIFIC_LIBRARIES "-framework Foundation" "-framework CFNetwork")
+    set(OS_SPECIFIC_LIBRARIES "-framework Foundation" "-framework CFNetwork" "-framework SystemConfiguration")
     target_compile_options(wsnet PRIVATE "-fobjc-arc")
 endif()
 
