--- tools/propcc.ml4	2024-06-10 11:22:38.000000000 +0800
+++ tools/propcc.ml4	2026-08-07 23:21:45.000000000 +0800
@@ -289,7 +289,7 @@
 let ooutfile = ref ""
 
 let process_file f =
-  let base = Filename.chop_extension f in
+  let base = Filename.chop_extension (Filename.basename f) in
   let baseM = String.capitalize_ascii base in
   prefix := baseM;
   (* Input *)

--- tools/propcc.ml	2024-06-10 11:22:38.000000000 +0800
+++ tools/propcc.ml	2026-08-07 23:34:24.000000000 +0800
@@ -549,7 +549,7 @@
 let ooutfile = ref ""
 
 let process_file f =
-  let base = Filename.chop_extension f in
+  let base = Filename.chop_extension (Filename.basename f) in
   let baseM = String.capitalize_ascii base in
   prefix := baseM;
   (* Input *)
