--- src/coreaudio.c	2026-03-13 04:00:50.000000000 +0800
+++ src/coreaudio.c	2026-03-31 05:17:20.000000000 +0800
@@ -33,6 +33,11 @@
 # define kAudioObjectPropertyElementMain kAudioObjectPropertyElementMaster
 #endif
 
+#ifndef MAC_OS_X_VERSION_10_8
+# define kAudioObjectPropertyScopeInput kAudioDevicePropertyScopeInput
+# define kAudioObjectPropertyScopeOutput kAudioDevicePropertyScopeOutput
+#endif
+
 #define Buffactor 4
 
 typedef struct {
