--- Modules/posixmodule.c
+++ Modules/posixmodule.c
@@ -50,6 +50,7 @@
 #ifdef __APPLE__
    /* Needed for the implementation of os.statvfs */
 #  include <sys/param.h>
+#include "AvailabilityMacros.h"
 #  include <sys/mount.h>
 #endif
 
@@ -13363,10 +13364,12 @@
     if (st.f_flags & MNT_NOSUID) {
         flags |= ST_NOSUID;
     }
 
+#ifdef MAC_OS_X_VERSION_10_5
     _Static_assert(sizeof(st.f_blocks) == sizeof(long long), "assuming large file");
+#endif
 
 #define SET_ITEM(SEQ, INDEX, EXPR)                       \
     do {                                                 \
         PyObject *obj = (EXPR);                          \
         if (obj == NULL) {                               \
