--- Modules/posixmodule.c
+++ Modules/posixmodule.c
@@ -13363,10 +13363,13 @@
     if (st.f_flags & MNT_NOSUID) {
         flags |= ST_NOSUID;
     }
 
+#include "AvailabilityMacros.h"
+#ifdef MAC_OS_X_VERSION_10_5
     _Static_assert(sizeof(st.f_blocks) == sizeof(long long), "assuming large file");
+#endif /* !MAC_OS_X_VERSION_10_5 */
 
 #define SET_ITEM(SEQ, INDEX, EXPR)                       \
     do {                                                 \
         PyObject *obj = (EXPR);                          \
         if (obj == NULL) {                               \
