--- lib/sigsegv.c	2025-05-10 20:41:50.000000000 +0800
+++ lib/sigsegv.c	2025-05-18 18:37:55.000000000 +0800
@@ -660,7 +660,12 @@
      - 'ucontext_t' and 'struct __darwin_ucontext' in <sys/_types/_ucontext.h>,
      - 'struct __darwin_mcontext64' in <i386/_mcontext.h>, and
      - 'struct __darwin_x86_thread_state64' in <mach/i386/_structs.h>.  */
-#  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__rsp
+#  if !(defined _STRUCT_MCONTEXT || defined _STRUCT_MCONTEXT32 || defined _STRUCT_MCONTEXT64)
+/* Mac OS X 10.4 and earlier omitted the underscores.  */
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->ss.rsp
+#  else
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__rsp
+#  endif
 
 # elif defined __i386__
 
@@ -668,7 +673,12 @@
      - 'ucontext_t' and 'struct __darwin_ucontext' in <sys/_types/_ucontext.h>,
      - 'struct __darwin_mcontext32' in <i386/_mcontext.h>, and
      - 'struct __darwin_i386_thread_state' in <mach/i386/_structs.h>.  */
-#  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__esp
+#  if !(defined _STRUCT_MCONTEXT || defined _STRUCT_MCONTEXT32 || defined _STRUCT_MCONTEXT64)
+/* Mac OS X 10.4 and earlier omitted the underscores.  */
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->ss.esp
+#  else
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__esp
+#  endif
 
 # elif defined __arm64__
 
@@ -678,7 +688,7 @@
      - 'struct __darwin_arm_thread_state64' in <mach/arm/_structs.h>.  */
 #  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__sp
 
-# elif defined __powerpc__
+# elif defined __POWERPC__
 
 /* See the definitions of
      - 'ucontext_t' and 'struct __darwin_ucontext' in <sys/_structs.h>,
