--- lib/sigsegv.c.orig	2026-07-26 19:42:43.000000000 -0600
+++ lib/sigsegv.c	2026-07-26 20:03:02.000000000 -0600
@@ -668,7 +668,12 @@
      - 'ucontext_t' and 'struct __darwin_ucontext' in <sys/_types/_ucontext.h>,
      - 'struct __darwin_mcontext32' in <i386/_mcontext.h>, and
      - 'struct __darwin_i386_thread_state' in <mach/i386/_structs.h>.  */
-#  define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__esp
+#  if !(defined _STRUCT_MCONTEXT || defined _STRUCT_MCONTEXT32 || defined _STRUCT_MCONTEXT64)
+/* Mac OS X 10.4 and earlier omitted the underscores.  */
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->ss.esp
+#  else
+#   define SIGSEGV_FAULT_STACKPOINTER  ((ucontext_t *) ucp)->uc_mcontext->__ss.__esp
+#  endif
 
 # elif defined __arm64__
 
