--- src/unix/thread.c.orig	2026-03-07 03:35:52.000000000 +0800
+++ src/unix/thread.c	2026-09-22 20:13:42.000000000 +0800
@@ -900,6 +900,7 @@
   return UV_ENOSYS;
 }
 #elif defined(__APPLE__)
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 int uv__thread_setname(const char* name) {
   char namebuf[UV_PTHREAD_MAX_NAMELEN_NP];
   strncpy(namebuf, name, sizeof(namebuf) - 1);
@@ -909,6 +910,11 @@
     return UV__ERR(errno);
   return 0;
 }
+#else
+int uv__thread_setname(const char* name) {
+  return UV_ENOSYS;
+}
+#endif
 #elif defined(__NetBSD__)
 int uv__thread_setname(const char* name) {
   char namebuf[UV_PTHREAD_MAX_NAMELEN_NP];
@@ -949,6 +955,7 @@
   return 0;
 }
 #elif defined(__APPLE__)
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
 int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
   char thread_name[UV_PTHREAD_MAX_NAMELEN_NP];
   if (pthread_getname_np(*tid, thread_name, sizeof(thread_name)) != 0)
@@ -960,6 +967,11 @@
 }
 #else
 int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
+  return UV_ENOSYS;
+}
+#endif
+#else
+int uv__thread_getname(uv_thread_t* tid, char* name, size_t size) {
   int r;
   char thread_name[UV_PTHREAD_MAX_NAMELEN_NP];
   r = pthread_getname_np(*tid, thread_name, sizeof(thread_name));
