--- ../ruby-4.0.4.org/configure	2026-05-12 06:07:54
+++ ./configure	2026-05-12 23:13:13
@@ -8524,7 +8524,8 @@ then :
 	   >/dev/null
 then :
 
-	suppress_ld_waring=yes
+	CC_WRAPPER=`cd -P "${tooldir}" && pwd`/darwin-cc
+	CC="$CC_WRAPPER $CC"
 
 fi
     rm -fr conftest*
@@ -34981,21 +34982,8 @@ fi
 then :
   withval=$with_destdir; DESTDIR="$withval"
 fi
-
-
-
-if test "x$load_relative:$DESTDIR" = xyes:
-then :
 
-    if test "x$prefix" = xNONE
-then :
-  DESTDIR="$ac_default_prefix"
-else $as_nop
-  DESTDIR="$prefix"
-fi
-    prefix=/.
 
-fi
 
 cat >confcache <<\_ACEOF
 # This file is a shell script that caches the results of configure
