From 218e3cd9e642aaa40a668cfc29510baad0090b00 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 14 Apr 2025 16:10:46 +0800
Subject: [PATCH 1/2] CommandBuffer.cpp: add missing header

---
 src/buffers/CommandBuffer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/buffers/CommandBuffer.cpp src/buffers/CommandBuffer.cpp
index a14a3b4..c85e628 100644
--- src/buffers/CommandBuffer.cpp
+++ src/buffers/CommandBuffer.cpp
@@ -1,4 +1,5 @@
 #include "CommandBuffer.hpp"
+#include <cstring>
 
 CommandBuffer::CommandBuffer(std::string command) : Buffer(command, BufferTypeCommand) {
   this->command = command;
-- 
2.48.0

