From bb25f37adca59f56f60bf583b194da9c58ee86d6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 14 Apr 2025 16:26:10 +0800
Subject: [PATCH 2/2] Git.cpp: add missing header

---
 src/Git.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git src/Git.cpp src/Git.cpp
index 0dceadf..3c13633 100644
--- src/Git.cpp
+++ src/Git.cpp
@@ -4,6 +4,7 @@
 #include <Path.hpp>
 #include <cstdio>
 #include <stdlib.h>
+#include <cstring>
 #include <cerrno>
 
 bool Git::Command(std::string command, std::string &result) {
-- 
2.48.0

