--- Makefile	2024-07-25 07:32:20
+++ Makefile	2025-04-14 15:17:39
@@ -1,7 +1,8 @@
 # makefile for Phred
 
-CC=clang++
-LD=clang++
+CXX=@CXX@
+LD=@CXX@
+PREFIX=@PREFIX@
 
 BUILD_DIR = build
 SRC_DIR = src
@@ -20,9 +21,9 @@
 
 DEBUGOPTS = 
 
-CCOPTS = -Isrc -g  -std=c++20 $(FULL_WARNINGS) $(EXTRA_CCOPTS) $(DEBUGOPTS)
-CCOPTS+=-I/opt/homebrew/opt/ncurses/include
-CCOPTS+=-I/opt/homebrew/include/lua
+CCOPTS += -Isrc -g  -std=c++20 $(FULL_WARNINGS) $(EXTRA_CCOPTS) $(DEBUGOPTS)
+CCOPTS+=-I$(PREFIX)/include
+# CCOPTS+=-I$(PREFIX)/include/lua
 
 
 #######################################
@@ -30,36 +31,34 @@
 all: phred
 
 $(BUILD_DIR)/%_cpp.o: $(SRC_DIR)/%.cpp
-	@echo "CC" $<
+	@echo "CXX" $<
 	@mkdir -p $(@D)
 	@echo ""
-	$(CC) $(CCOPTS) -MMD -c $< -o $@
+	$(CXX) $(CCOPTS) -MMD -c $< -o $@
 
 PHRED_FILES= $(shell find src -name '*.cpp')
 PHRED_OBJS=$(PHRED_FILES:$(SRC_DIR)/%.cpp=$(BUILD_DIR)/%_cpp.o)
 
 DEP_FILES = $(PHRED_OBJS:%.o=%.d)
 
-LD_LIBRARIES=
-LD_LIBRARIES+=-L/opt/homebrew/lib -llua
-#LD_LIBRARIES+=-L/opt/homebrew/opt/ncurses/lib -lncursesw
+LD_LIBRARIES+=-L$(PREFIX)/lib -llua -lncursesw
 
 phred: $(PHRED_OBJS) 
 	@echo ""
 	$(LD) -o $@ $(PHRED_OBJS)  $(LDADD) $(LD_LIBRARIES)
 
 install: phred
-	@echo "Installing phred to /usr/local/bin"
+#	@echo "Installing phred to $(PREFIX)/bin"
 	@strip phred
-	@mkdir -p /usr/local/bin
-	@install -m 755 phred /usr/local/bin
-	@echo "Installing phred/share to /usr/local/share/phred"
-	@mkdir -p /usr/local/share/phred
-	@cp -rp share/ /usr/local/share/phred
-	@echo
-	@echo Size compared to nvim
-	@ls -Llh /usr/local/bin/phred `which nvim`
-#        install -d -m 644 share/*.lua /usr/local/share/phred/
+	@mkdir -p $(DESTDIR)$(PREFIX)/bin
+	@install -m 755 phred $(DESTDIR)$(PREFIX)/bin
+#	@echo "Installing phred/share to $(PREFIX)/share/phred"
+	@mkdir -p $(DESTDIR)$(PREFIX)/share/phred
+	@cp -rp share/ $(DESTDIR)$(PREFIX)/share/phred
+#	@echo
+#	@echo Size compared to nvim
+#	@ls -Llh $(PREFIX)/bin/phred `which nvim`
+#        install -d -m 644 share/*.lua $(PREFIX)/share/phred/
 
 clean:
 	rm -rf $(BUILD_DIR) newvm
