--- src/main.cpp	2024-07-25 07:32:20
+++ src/main.cpp	2025-04-14 15:28:19
@@ -45,12 +45,12 @@
 
   bool lua_ok = true;
   std::string *lua_ret;
-  setenv("LUA_PATH", "./share/?.lua;/usr/local/share/phred/?.lua", 0);
+  setenv("LUA_PATH", "./share/?.lua;@PREFIX@/share/phred/?.lua", 0);
   lua_ret = lua.RunFile("./share/Phred.lua");
   if (lua_ret) {
     console->printf("%s\n", lua_ret->c_str());
     delete lua_ret;
-    lua_ret = lua.RunFile("/usr/local/share/phred/Phred.lua");
+    lua_ret = lua.RunFile("@PREFIX@/share/phred/Phred.lua");
     if (lua_ret) {
       console->printf("%s\n", lua_ret->c_str());
       delete lua_ret;

--- src/EmbeddedLua.cpp	2024-07-25 07:32:20
+++ src/EmbeddedLua.cpp	2025-04-14 15:26:49
@@ -58,7 +58,7 @@
 static std::vector<std::string> require_path = {
   "./",
   "./share/",
-  "/usr/local/share/phred/",
+  "@PREFIX@/share/phred/",
 };
 
 static int PhredRequire(lua_State *L) {
