From e749148b6347a2dd1e763456cb9fe9c7cfa2cee4 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 14 Feb 2025 09:33:50 +0800
Subject: [PATCH] CMakeLists.txt: use standard syntax for cp

---
 test/CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git test/CMakeLists.txt test/CMakeLists.txt
index 4553ef8..610de3f 100644
--- test/CMakeLists.txt
+++ test/CMakeLists.txt
@@ -18,9 +18,9 @@ target_link_libraries(
 add_custom_command(
     TARGET ${tester} POST_BUILD
     COMMAND
-        cp -a ${CMAKE_CURRENT_SOURCE_DIR}/run_tests.py
-              ${CMAKE_CURRENT_SOURCE_DIR}/ref
-              ${CMAKE_CURRENT_BINARY_DIR}/
+        cp -pPR ${CMAKE_CURRENT_SOURCE_DIR}/run_tests.py
+                ${CMAKE_CURRENT_SOURCE_DIR}/ref
+                ${CMAKE_CURRENT_BINARY_DIR}/
 )
 
 if (testsweeper_is_project)
