diff --git a/uppsrc/plugin/bmp/bmphdr.h b/uppsrc/plugin/bmp/bmphdr.h
index 1a4cfaef6..c5e75c759 100644
--- a/uppsrc/plugin/bmp/bmphdr.h
+++ b/uppsrc/plugin/bmp/bmphdr.h
@@ -15,9 +15,15 @@ struct BMP_FILEHEADER {
 	void    SwapEndian()
 	{
 #ifdef CPU_BIG_ENDIAN
-		bfType = UPP::SwapEndian(bfType);
-		bfSize = UPP::SwapEndian(bfSize);
-		bfOffBits = UPP::SwapEndian(bfOffBits);
+		word tmp_bfType = bfType;
+		dword tmp_bfSize = bfSize;
+		dword tmp_bfOffBits = bfOffBits;
+		Upp::EndianSwap(tmp_bfType);
+		Upp::EndianSwap(tmp_bfSize);
+		Upp::EndianSwap(tmp_bfOffBits);
+		bfType = tmp_bfType;
+		bfSize = tmp_bfSize;
+		bfOffBits = tmp_bfOffBits;
 #endif
 	}
 }
@@ -43,17 +49,39 @@ struct BMP_INFOHEADER
 	void    SwapEndian()
 	{
 #ifdef CPU_BIG_ENDIAN
-		biSize = UPP::SwapEndian(biSize);
-		biWidth = UPP::SwapEndian(biWidth);
-		biHeight = UPP::SwapEndian(biHeight);
-		biPlanes = UPP::SwapEndian(biPlanes);
-		biBitCount = UPP::SwapEndian(biBitCount);
-		biCompression = UPP::SwapEndian(biCompression);
-		biSizeImage = UPP::SwapEndian(biSizeImage);
-		biXPelsPerMeter = UPP::SwapEndian(biXPelsPerMeter);
-		biYPelsPerMeter = UPP::SwapEndian(biYPelsPerMeter);
-		biClrUsed = UPP::SwapEndian(biClrUsed);
-		biClrImportant = UPP::SwapEndian(biClrImportant);
+		dword tmp_biSize = biSize;
+		int32 tmp_biWidth = biWidth;
+		int32 tmp_biHeight = biHeight;
+		word tmp_biPlanes = biPlanes;
+		word tmp_biBitCount = biBitCount;
+		dword tmp_biCompression = biCompression;
+		dword tmp_biSizeImage = biSizeImage;
+		int32 tmp_biXPelsPerMeter = biXPelsPerMeter;
+		int32 tmp_biYPelsPerMeter = biYPelsPerMeter;
+		dword tmp_biClrUsed = biClrUsed;
+		dword tmp_biClrImportant = biClrImportant;
+		Upp::EndianSwap(tmp_biSize);
+		Upp::EndianSwap(tmp_biWidth);
+		Upp::EndianSwap(tmp_biHeight);
+		Upp::EndianSwap(tmp_biPlanes);
+		Upp::EndianSwap(tmp_biBitCount);
+		Upp::EndianSwap(tmp_biCompression);
+		Upp::EndianSwap(tmp_biSizeImage);
+		Upp::EndianSwap(tmp_biXPelsPerMeter);
+		Upp::EndianSwap(tmp_biYPelsPerMeter);
+		Upp::EndianSwap(tmp_biClrUsed);
+		Upp::EndianSwap(tmp_biClrImportant);
+		biSize = tmp_biSize;
+		biWidth = tmp_biWidth;
+		biHeight = tmp_biHeight;
+		biPlanes = tmp_biPlanes;
+		biBitCount = tmp_biBitCount;
+		biCompression = tmp_biCompression;
+		biSizeImage = tmp_biSizeImage;
+		biXPelsPerMeter = tmp_biXPelsPerMeter;
+		biYPelsPerMeter = tmp_biYPelsPerMeter;
+		biClrUsed = tmp_biClrUsed;
+		biClrImportant = tmp_biClrImportant;
 #endif
 	}
 }
