--- src/CMakeLists.txt	2025-12-10 02:40:40.000000000 +0800
+++ src/CMakeLists.txt	2025-12-11 07:08:57.000000000 +0800
@@ -21,7 +21,6 @@
 	retroshare/rsgxstrans.h
 	retroshare/rsgxstunnel.h
 	retroshare/rsids.h
-	retroshare/rsnotify.h
 	retroshare/rsphoto.h
 	retroshare/rsplugin.h
 	retroshare/rsreputations.h
@@ -273,7 +272,6 @@
 	pqi/pqistore.cc
 	pqi/authgpg.cc
 	pqi/p3cfgmgr.cc
-	pqi/p3notify.cc
 	pqi/p3servicecontrol.cc
 	pqi/pqifdbin.cc
 	pqi/pqinetstatebox.cc
@@ -302,7 +300,6 @@
 	pqi/p3historymgr.h
 	pqi/p3linkmgr.h
 	pqi/p3netmgr.h
-	pqi/p3notify.h
 	pqi/p3peermgr.h
 	pqi/p3servicecontrol.h
 	pqi/p3upnpmgr.h
