--- .home/.cargo/macports/clap_derive-4.0.21/src/lib.rs	2006-07-24 09:21:28.000000000 +0800
+++ .home/.cargo/macports/clap_derive-4.0.21/src/lib.rs	2026-01-06 22:51:47.000000000 +0800
@@ -30,7 +30,7 @@
 
 /// Generates the `ValueEnum` impl.
 #[proc_macro_derive(ValueEnum, attributes(clap, value))]
-#[proc_macro_error]
+#[proc_macro_error(allow_not_macro)]
 pub fn value_enum(input: TokenStream) -> TokenStream {
     let input: DeriveInput = parse_macro_input!(input);
     derives::derive_value_enum(&input).into()
@@ -43,7 +43,7 @@
 /// implementing a conversion code to instantiate an instance of the user
 /// context struct.
 #[proc_macro_derive(Parser, attributes(clap, structopt, command, arg, group))]
-#[proc_macro_error]
+#[proc_macro_error(allow_not_macro)]
 pub fn parser(input: TokenStream) -> TokenStream {
     let input: DeriveInput = parse_macro_input!(input);
     derives::derive_parser(&input).into()
@@ -51,7 +51,7 @@
 
 /// Generates the `Subcommand` impl.
 #[proc_macro_derive(Subcommand, attributes(clap, command, arg, group))]
-#[proc_macro_error]
+#[proc_macro_error(allow_not_macro)]
 pub fn subcommand(input: TokenStream) -> TokenStream {
     let input: DeriveInput = parse_macro_input!(input);
     derives::derive_subcommand(&input).into()
@@ -59,7 +59,7 @@
 
 /// Generates the `Args` impl.
 #[proc_macro_derive(Args, attributes(clap, command, arg, group))]
-#[proc_macro_error]
+#[proc_macro_error(allow_not_macro)]
 pub fn args(input: TokenStream) -> TokenStream {
     let input: DeriveInput = parse_macro_input!(input);
     derives::derive_args(&input).into()
