--- .home/.cargo/macports/clap-2.33.3/src/macros.rs	2020-08-13 05:36:10.000000000 +0800
+++ .home/.cargo/macports/clap-2.33.3/src/macros.rs	2026-01-04 16:23:16.000000000 +0800
@@ -461,7 +461,7 @@
 ///             .get_matches();
 /// # }
 /// ```
-#[cfg(not(feature = "no_cargo"))]
+#[cfg(all(not(feature = "no_cargo"), not(rust_compiler="mrustc")))]
 #[macro_export]
 macro_rules! crate_authors {
     ($sep:expr) => {{
@@ -576,13 +576,11 @@
     () => {
         $crate::App::new(crate_name!())
             .version(crate_version!())
-            .author(crate_authors!())
             .about(crate_description!())
     };
     ($sep:expr) => {
         $crate::App::new(crate_name!())
             .version(crate_version!())
-            .author(crate_authors!($sep))
             .about(crate_description!())
     };
 }
