--- .home/.cargo/macports/derive_builder-0.9.0/src/lib.rs
+++ .home/.cargo/macports/derive_builder-0.9.0/src/lib.rs	2026-01-05 23:17:01.000000000 +0800
@@ -552,7 +552,7 @@
 //! [builder pattern]: https://aturon.github.io/ownership/builders.html
 //! [`derive_builder_core`]: https://crates.io/crates/derive_builder_core
 
-#![crate_type = "proc-macro"]
+// #![crate_type = "proc-macro"]
 #![deny(warnings)]
 
 #[macro_use]
