See: https://github.com/rust-onig/rust-onig/issues/229

--- .home/.cargo/macports/onig_sys-69.8.1/build.rs
+++ .home/.cargo/macports/onig_sys-69.8.1/build.rs	2026-07-23 13:29:13.000000000 +0800
@@ -193,6 +193,8 @@
     }
 
     cc.warnings(false); // not actionable by the end user
+    cc.flag_if_supported("-Wno-incompatible-pointer-types");
+    cc.flag_if_supported("-std=gnu89");
     cc.compile("onig");
 }
 
