--- .home/.cargo/macports/time-macros-0.1.1/src/lib.rs	2020-09-29 06:28:18.000000000 +0800
+++ .home/.cargo/macports/time-macros-0.1.1/src/lib.rs	2026-01-05 23:47:05.000000000 +0800
@@ -3,4 +3,8 @@
 use proc_macro_hack::proc_macro_hack;
 
 #[proc_macro_hack]
-pub use time_macros_impl::{date, offset, time};
+pub use time_macros_impl::date;
+#[proc_macro_hack]
+pub use time_macros_impl::offset;
+#[proc_macro_hack]
+pub use time_macros_impl::time;
