--- Makefile	2025-05-21 23:32:53.000000000 +0800
+++ Makefile	2025-10-06 19:28:17.000000000 +0800
@@ -127,7 +127,9 @@
 
   # NOTE: libxlsxwriter is required for xlsx file export support
   ifneq ($(shell pkg-config --exists xlsxwriter || echo 'no'),no)
-    CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter)
+# Cannot use pkg-config here, since it pulls in /opt/local/include/minizip/zip.h,
+# which conflicts with libzip header, /opt/local/include/zip.h
+#    CFLAGS += -DXLSX_EXPORT $(shell pkg-config --cflags xlsxwriter)
     LDLIBS += $(shell pkg-config --libs xlsxwriter)
   endif
 
@@ -216,12 +218,12 @@
 	doxygen Doxyfile
 
 man_install:
-	@cp -r ../docs/man/man3/ /usr/local/share/man/
+	@cp -r ../docs/man/man3/ ${prefix}/share/man/
 	mandb
 
 # "sc-im" MUST match what is in Doxyfile `MAN_EXTENSION = .sc-im.3`
 man_uninstall:
-	@rm -rf /usr/local/share/man/man3/*sc-im.3
+	@rm -rf ${prefix}/share/man/man3/*sc-im.3
 	@mandb
 
 clean:
