From d94ef98d26b97266af42d9eef081cd199d45efef Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 5 Sep 2025 09:22:27 +0800
Subject: [PATCH] Add missing limits.h for PATH_MAX

diff --git a/src/slash-utils/ls.cpp b/src/slash-utils/ls.cpp
index 791fdc3..a212b00 100644
--- src/slash-utils/ls.cpp
+++ src/slash-utils/ls.cpp
@@ -8,6 +8,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <dirent.h>
+#include <limits.h> // PATH_MAX
 #include <sys/stat.h>
 #include <csignal>
 
