From 058f762f85b48ee641204051ebe446f3366284c8 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 5 Sep 2025 08:54:06 +0800
Subject: [PATCH] slash-utils/move.cpp: fix timespec for Apple

diff --git a/src/slash-utils/move.cpp b/src/slash-utils/move.cpp
index 281fe3a..328d60a 100644
--- src/slash-utils/move.cpp
+++ src/slash-utils/move.cpp
@@ -9,6 +9,11 @@
 #include <sstream>
 #include "../help_helper.h"
 
+#ifdef __APPLE__
+#define st_atim st_atimespec
+#define st_mtim st_mtimespec
+#endif
+
 class Move {
   private:
     int clear_history() {
