From 08af977c30850b4b3e783daadf96cb98354d96c9 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 12 May 2025 23:49:39 +0800
Subject: [PATCH 3/4] Revert af4df8a0ed4111286783387e2d3a940a10f9da29

---
 src/file/mmc_device_darwin.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git src/file/mmc_device_darwin.c src/file/mmc_device_darwin.c
index 1f7f87d..a8f70be 100644
--- src/file/mmc_device_darwin.c
+++ src/file/mmc_device_darwin.c
@@ -229,6 +229,8 @@ static int iokit_unmount(MMCDEV *mmc) {
 
     DADiskUnmount(mmc->disk, kDADiskUnmountOptionForce, iokit_unmount_complete, mmc);
 
+    CFRunLoopRunInMode (kCFRunLoopDefaultMode, 10, true);
+
     return mmc->is_mounted ? -1 : 0;
 }
 
@@ -236,6 +238,8 @@ static int iokit_mount(MMCDEV *mmc) {
     if (0 == mmc->is_mounted) {
         if (mmc->disk && mmc->session) {
             DADiskMount(mmc->disk, NULL, kDADiskMountOptionDefault, iokit_mount_complete, mmc);
+
+            CFRunLoopRunInMode (kCFRunLoopDefaultMode, 10, true);
         }
     }
     return mmc->is_mounted ? 0 : -1;
@@ -337,14 +341,15 @@ static int iokit_da_init(MMCDEV *mmc) {
         return -1;
     }
 
-    dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
-    DASessionSetDispatchQueue(mmc->session, queue);
+    DAApprovalSessionScheduleWithRunLoop (mmc->session, CFRunLoopGetCurrent (),
+                                          kCFRunLoopDefaultMode);
 
     return 0;
 }
 
 static void iokit_da_destroy(MMCDEV *mmc) {
-    DASessionSetDispatchQueue(mmc->session, NULL);
+    DAApprovalSessionUnscheduleFromRunLoop (mmc->session, CFRunLoopGetCurrent (),
+                                            kCFRunLoopDefaultMode);
 
     if (mmc->disk) {
         CFRelease(mmc->disk);
-- 
2.48.0

