From 8e0547013fe44f171f0579c67e4253829e81ee78 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 13 May 2025 00:05:19 +0800
Subject: [PATCH 4/4] Revert 5f58595ff4a19d645c482ded6877e9ac10e23952

---
 src/file/mmc_device_darwin.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git src/file/mmc_device_darwin.c src/file/mmc_device_darwin.c
index a8f70be..4e3de6b 100644
--- src/file/mmc_device_darwin.c
+++ src/file/mmc_device_darwin.c
@@ -463,12 +463,6 @@ void device_close(MMCDEV **pp)
             IODestroyPlugInInterface(mmc->plugInInterface);
         }
 
-        if (!mmc->sync_sem) {
-            /* open failed before iokit_da_init() */
-            X_FREE(*pp);
-            return;
-        }
-
         (void) iokit_mount (mmc);
 
         iokit_da_destroy(mmc);
-- 
2.48.0

