--- cmake/config.cmake.in	2025-06-20 00:54:05.000000000 +0800
+++ cmake/config.cmake.in	2025-07-16 02:56:53.000000000 +0800
@@ -28,7 +28,7 @@
     find_library(IOKIT_LIBRARY IOKit)
     find_dependency(Threads REQUIRED)
     find_dependency(PkgConfig REQUIRED)
-    pkg_check_modules(sqlcipher REQUIRED IMPORTED_TARGET sqlcipher)
+    pkg_check_modules(sqlite3 REQUIRED IMPORTED_TARGET sqlite3)
     pkg_check_modules(glib REQUIRED IMPORTED_TARGET glib-2.0)
     pkg_check_modules(gio REQUIRED IMPORTED_TARGET gio-2.0)
     pkg_check_modules(gmodule REQUIRED IMPORTED_TARGET gmodule-2.0)
@@ -42,7 +42,7 @@
 else()
     find_dependency(Threads REQUIRED)
     find_dependency(PkgConfig REQUIRED)
-    pkg_check_modules(sqlcipher REQUIRED IMPORTED_TARGET sqlcipher)
+    pkg_check_modules(sqlite3 REQUIRED IMPORTED_TARGET sqlite3)
     pkg_check_modules(glib REQUIRED IMPORTED_TARGET glib-2.0)
     pkg_check_modules(gio REQUIRED IMPORTED_TARGET gio-2.0)
     pkg_check_modules(gmodule REQUIRED IMPORTED_TARGET gmodule-2.0)
