From 1fdf06b67369dc140e402a5481253962a45bea45 Mon Sep 17 00:00:00 2001
From: barracuda156 <barracuda@macos-powerpc.org>
Date: Fri, 28 Aug 2026 03:50:45 +0000
Subject: [PATCH] port: inspector-unbreak-build-with-gcc

---
 v8/src/inspector/string-16.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8/src/inspector/string-16.h b/v8/src/inspector/string-16.h
index 76c6849..aed378d 100644
--- a/v8/src/inspector/string-16.h
+++ b/v8/src/inspector/string-16.h
@@ -147,7 +147,7 @@ String16 String16::concat(T... args) {
 
 }  // namespace v8_inspector
 
-#if !defined(__APPLE__) || defined(_LIBCPP_VERSION)
+#if !(defined(__APPLE__) && defined(__clang__)) || defined(_LIBCPP_VERSION)
 
 namespace std {
 template <>
-- 
2.43.0

