From 1637d091eb308fb8ec2df5768056d09d5bd93981 Mon Sep 17 00:00:00 2001
From: barracuda156 <barracuda@macos-powerpc.org>
Date: Fri, 28 Aug 2026 03:50:45 +0000
Subject: [PATCH] port: cpu.cc-support-powerpc-darwin

---
 v8/src/base/cpu.cc | 28 +++++++++++++++++++++++++++-
 1 file changed, 27 insertions(+), 1 deletion(-)

diff --git a/v8/src/base/cpu.cc b/v8/src/base/cpu.cc
index bbdae52..a9a2a26 100644
--- a/v8/src/base/cpu.cc
+++ b/v8/src/base/cpu.cc
@@ -28,6 +28,15 @@
 #define POWER_9 0x20000
 #endif
 #endif
+#if V8_OS_MACOSX
+#include <sys/sysctl.h>  // sysctlbyname
+#if V8_HOST_ARCH_PPC || V8_HOST_ARCH_PPC64
+#include <mach/mach.h>
+#include <mach/mach_host.h>
+#include <mach/host_info.h>
+#include <mach/machine.h>
+#endif
+#endif
 #if V8_OS_POSIX
 #include <unistd.h>  // sysconf()
 #endif
@@ -683,7 +692,24 @@ CPU::CPU()
       part_ = PPC_POWER5;
       break;
   }
-#endif  // V8_OS_AIX
+
+#elif V8_OS_MACOSX
+
+struct host_basic_info host_basic_info;
+  switch(host_basic_info.cpu_subtype) {
+    case CPU_SUBTYPE_POWERPC_970:
+    default:
+      part_ = PPC_G5;
+      break;
+    case CPU_SUBTYPE_POWERPC_7450:
+      part_ = PPC_G4;
+      break;
+    case CPU_SUBTYPE_POWERPC_7400:
+      part_ = PPC_G4;
+      break;
+   }
+
+#endif  // V8_OS_MACOSX
 #endif  // !USE_SIMULATOR
 #endif  // V8_HOST_ARCH_PPC || V8_HOST_ARCH_PPC64
 }
-- 
2.43.0

