From aba4ebaa6f2515f4352bdd48331f280cf34d40da Mon Sep 17 00:00:00 2001
From: barracuda156 <barracuda@macos-powerpc.org>
Date: Fri, 28 Aug 2026 03:50:45 +0000
Subject: [PATCH] port: runtime-utils.h-fix-static_cast

---
 v8/src/runtime/runtime-utils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8/src/runtime/runtime-utils.h b/v8/src/runtime/runtime-utils.h
index 170c0bc..a7e02ae 100644
--- a/v8/src/runtime/runtime-utils.h
+++ b/v8/src/runtime/runtime-utils.h
@@ -136,7 +136,7 @@ static inline ObjectPair MakePair(Object x, Object y) {
 #if defined(V8_TARGET_LITTLE_ENDIAN)
   return x.ptr() | (static_cast<ObjectPair>(y.ptr()) << 32);
 #elif defined(V8_TARGET_BIG_ENDIAN)
-  return y->ptr() | (static_cast<ObjectPair>(x->ptr()) << 32);
+  return y.ptr() | (static_cast<ObjectPair>(x.ptr()) << 32);
 #else
 #error Unknown endianness
 #endif
-- 
2.43.0

