From 0ad0fd4669796a2f99c7cd657bbcfacd8d099b4f Mon Sep 17 00:00:00 2001
From: barracuda156 <barracuda@macos-powerpc.org>
Date: Fri, 28 Aug 2026 03:50:46 +0000
Subject: [PATCH] port: amend-constants-ppc

---
 v8/src/codegen/ppc/constants-ppc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/v8/src/codegen/ppc/constants-ppc.h b/v8/src/codegen/ppc/constants-ppc.h
index b75c3e3..a9f6a6d 100644
--- a/v8/src/codegen/ppc/constants-ppc.h
+++ b/v8/src/codegen/ppc/constants-ppc.h
@@ -29,7 +29,7 @@
 #endif
 
 #if !(V8_HOST_ARCH_PPC || V8_HOST_ARCH_PPC64) || V8_OS_AIX || \
-    V8_TARGET_ARCH_PPC64
+    V8_TARGET_ARCH_PPC64 || V8_OS_MACOSX
 #define ABI_PASSES_HANDLES_IN_REGS 1
 #else
 #define ABI_PASSES_HANDLES_IN_REGS 0
-- 
2.43.0

