--- CMakeLists.txt	2026-03-07 21:56:40.000000000 +0800
+++ CMakeLists.txt	2026-09-14 17:49:10.000000000 +0800
@@ -66,13 +66,9 @@
 
 # Compiler-specific flags
 if (CMAKE_CXX_COMPILER_ID MATCHES "^(Apple)?Clang$")
-    if (NOT APPLE)
-        add_compile_options(-march=native)
-    endif ()
-
     add_compile_options(-pedantic -ferror-limit=1)
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-    add_compile_options(-march=native -pedantic -fmax-errors=1 -Wno-tautological-constant-compare)
+    add_compile_options(-pedantic -fmax-errors=1 -Wno-tautological-constant-compare)
 elseif (CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
     add_compile_options(-w -ferror-limit=1)
 endif ()
