From d93b3a597f3668a9b75f12c33a7580b142845b78 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 21 Jun 2025 12:04:45 +0800
Subject: [PATCH 2/3] util.h: add missing <stdexcept>

---
 include/salticidae/util.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git include/salticidae/util.h include/salticidae/util.h
index 1ad81c1..a9b8ed0 100644
--- include/salticidae/util.h
+++ include/salticidae/util.h
@@ -28,6 +28,7 @@
 #include <getopt.h>
 #include <unistd.h>
 #include <fcntl.h>
+
 #include "salticidae/config.h"
 
 typedef struct SalticidaeCError {
@@ -64,6 +65,8 @@ const char *salticidae_strerror(int code);
 #include <vector>
 #include <unordered_map>
 #include <functional>
+#include <stdexcept>
+
 #include "salticidae/ref.h"
 
 namespace salticidae {
