From caf21b682198a698dc11e814792c6e642bf8847a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 21 Jun 2025 12:26:03 +0800
Subject: [PATCH 3/3] test_msgnet_c.c: add missing salticidae/endian.h

---
 test/test_msgnet_c.c | 1 +
 1 file changed, 1 insertion(+)

diff --git test/test_msgnet_c.c test/test_msgnet_c.c
index 2ee9443..0c3021e 100644
--- test/test_msgnet_c.c
+++ test/test_msgnet_c.c
@@ -28,6 +28,7 @@
 #include <signal.h>
 
 #include "salticidae/event.h"
+#include "salticidae/endian.h"
 #include "salticidae/network.h"
 #include "salticidae/stream.h"
 
