--- include/salticidae/config.h	2020-05-11 20:44:00.000000000 +0800
+++ include/salticidae/config.h	2025-06-21 11:44:36.000000000 +0800
@@ -1,11 +1,8 @@
 #ifndef _SALTICIDAE_CONFIG_H
 #define _SALTICIDAE_CONFIG_H
 
-#cmakedefine SALTICIDAE_DEBUG_LOG
-#cmakedefine SALTICIDAE_NORMAL_LOG
-#cmakedefine SALTICIDAE_MSG_STAT
-#cmakedefine SALTICIDAE_NOCHECK
-#cmakedefine SALTICIDAE_NOCHECKSUM
-#cmakedefine SALTICIDAE_CBINDINGS
+#define SALTICIDAE_NORMAL_LOG 1
+#define SALTICIDAE_MSG_STAT 1
+#define SALTICIDAE_CBINDINGS 1
 
 #endif

--- CMakeLists.txt	2020-05-11 20:44:00.000000000 +0800
+++ CMakeLists.txt	2025-06-21 11:48:15.000000000 +0800
@@ -71,7 +71,8 @@
 option(SALTICIDAE_NOCHECKSUM " disable checksum in messages" OFF)
 option(SALTICIDAE_CBINDINGS "enable C bindings" ON)
 
-configure_file(src/config.h.in include/salticidae/config.h @ONLY)
+# This does not work.
+#configure_file(src/config.h.in include/salticidae/config.h @ONLY)
 
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -W -Wall -Wextra -pedantic -Wsuggest-override")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -W -Wall -Wextra -pedantic -Wsuggest-override")
