--- a/plugin/group_replication/CMakeLists.txt	2025-09-24 21:35:03.000000000 +0800
+++ b/plugin/group_replication/CMakeLists.txt	2025-12-15 06:31:41.000000000 +0800
@@ -219,6 +219,9 @@
   OpenSSL::SSL OpenSSL::Crypto
   ${LINK_WHOLE_ARCHIVE} ext::libprotobuf-lite ${LINK_NO_WHOLE_ARCHIVE}
   ${GR_PROTOBUF_LITE_LIB}
+  absl_hash
+  absl_log_internal_check_op
+  absl_log_internal_message
   mysqlgcs
   extra::rapidjson
   mysql_gtid
