--- a/mysys/my_systime.cc	2026-07-10 20:37:46.000000000 +0800
+++ b/mysys/my_systime.cc	2026-08-13 02:44:13.000000000 +0800
@@ -71,7 +71,7 @@
     return;
   }
   const unsigned long long int now = my_getsystime() + (nsec / 100);
-  unsigned long long int const tv_sec = now / 10000000ULL;
+  unsigned long long int tv_sec = now / 10000000ULL;
 #if SIZEOF_TIME_T < SIZEOF_LONG_LONG
   /* Ensure that the number of seconds don't overflow. */
   tv_sec = std::min(tv_sec, static_cast<unsigned long long int>(

--- a/components/connection_control/connection_delay.cc	2026-07-10 20:37:46.000000000 +0800
+++ b/components/connection_control/connection_delay.cc	2026-08-13 04:52:45.000000000 +0800
@@ -169,7 +169,7 @@
     return;
   }
   const unsigned long long int now = my_getsystime() + (nsec / 100);
-  unsigned long long int const tv_sec = now / 10000000ULL;
+  unsigned long long int tv_sec = now / 10000000ULL;
 #if SIZEOF_TIME_T < SIZEOF_LONG_LONG
   /* Ensure that the number of seconds don't overflow. */
   tv_sec = std::min(tv_sec, static_cast<unsigned long long int>(
