--- a/cmake/os/Darwin.cmake	2025-09-24 21:35:03.000000000 +0800
+++ b/cmake/os/Darwin.cmake	2025-12-14 19:04:06.000000000 +0800
@@ -48,6 +48,12 @@
 # This is used for the version_compile_machine variable.
 IF(APPLE_ARM)
   SET(MYSQL_MACHINE_TYPE "arm64")
+ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "ppc64")
+  SET(MYSQL_MACHINE_TYPE "ppc64")
+ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "ppc")
+  SET(MYSQL_MACHINE_TYPE "ppc")
+ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "i386")
+  SET(MYSQL_MACHINE_TYPE "i386")
 ELSE()
   SET(MYSQL_MACHINE_TYPE "x86_64")
 ENDIF()

--- a/cmake/package_name.cmake	2025-09-24 21:35:03.000000000 +0800
+++ b/cmake/package_name.cmake	2025-12-14 19:27:51.000000000 +0800
@@ -86,6 +86,12 @@
 
       IF(APPLE_ARM)
         SET(DEFAULT_MACHINE "arm64")
+      ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "ppc64")
+        SET(DEFAULT_MACHINE "ppc64")
+      ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "ppc")
+        SET(DEFAULT_MACHINE "ppc")
+      ELSEIF(CMAKE_OSX_ARCHITECTURES MATCHES "i386")
+        SET(DEFAULT_MACHINE "i386")
       ELSE()
         SET(DEFAULT_MACHINE "x86_64")
       ENDIF()

--- a/storage/innobase/include/ut0cpu_cache.h	2025-09-24 21:35:03.000000000 +0800
+++ b/storage/innobase/include/ut0cpu_cache.h	2025-12-15 00:03:12.000000000 +0800
@@ -35,7 +35,7 @@
 namespace ut {
 
 /** CPU cache line size */
-#ifdef __powerpc__
+#if defined(__powerpc__) || defined(__POWERPC__)
 constexpr size_t INNODB_CACHE_LINE_SIZE = 128;
 #else
 constexpr size_t INNODB_CACHE_LINE_SIZE = 64;
