--- lib/fuse_misc.h	2025-08-26 18:59:36.000000000 +0800
+++ lib/fuse_misc.h	2026-02-20 12:47:05.000000000 +0800
@@ -41,7 +41,7 @@
 #define ST_MTIM_NSEC(stbuf) ((stbuf)->st_mtim.tv_nsec)
 #define ST_ATIM_NSEC_SET(stbuf, val) (stbuf)->st_atim.tv_nsec = (val)
 #define ST_MTIM_NSEC_SET(stbuf, val) (stbuf)->st_mtim.tv_nsec = (val)
-#elif defined(HAVE_STRUCT_STAT_ST_ATIMESPEC)
+#elif defined(HAVE_STRUCT_STAT_ST_ATIMESPEC) && !defined(__APPLE__)
 /* FreeBSD */
 #define ST_ATIM_NSEC(stbuf) ((stbuf)->st_atimespec.tv_nsec)
 #define ST_CTIM_NSEC(stbuf) ((stbuf)->st_ctimespec.tv_nsec)
