--- cone/cone.sh.in	2025-12-29 01:24:06.000000000 +0800
+++ cone/cone.sh.in	2026-07-02 21:02:14.000000000 +0800
@@ -5,6 +5,13 @@
 sysconfdir="@sysconfdir@"
 libexecdir="@libexecdir@"
 
+if [ -n "$DYLD_LIBRARY_PATH" ]; then
+   DYLD_LIBRARY_PATH=${prefix}/lib/libgcc:$DYLD_LIBRARY_PATH
+else
+   DYLD_LIBRARY_PATH=${prefix}/lib/libgcc
+fi
+export DYLD_LIBRARY_PATH
+
 if test -f ${sysconfdir}/cone
 then
 	set -a
