From 10d50fe4c49f5098334f714f61b1500b965be8a6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Fri, 2 Jan 2026 23:14:34 +0800
Subject: [PATCH] __stack_chk_fail.c: add missing string.h for memcpy

---
 src/env/x86_64-xnu/__stack_chk_fail.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/env/x86_64-xnu/__stack_chk_fail.c b/src/env/x86_64-xnu/__stack_chk_fail.c
index 3f85ef2b..a2a2b04a 100644
--- a/src/env/x86_64-xnu/__stack_chk_fail.c
+++ b/src/env/x86_64-xnu/__stack_chk_fail.c
@@ -1,5 +1,6 @@
 #include <stdint.h>
 #include <stdlib.h>
+#include <string.h>
 #include "atomic.h"
  
 uintptr_t __stack_chk_guard;
