From 91a2817e9ed26ad614a56930399772a6d2ba1a04 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 30 Mar 2025 03:01:59 +0800
Subject: [PATCH] getiface.c: fix headers order for Apple

---
 libatalk/util/getiface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git libatalk/util/getiface.c libatalk/util/getiface.c
index bdf478c7..59cdd5d4 100644
--- libatalk/util/getiface.c
+++ libatalk/util/getiface.c
@@ -9,6 +9,7 @@
 #endif
 
 #include <errno.h>
+#include <sys/socket.h>
 #include <net/if.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
@@ -17,7 +18,6 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/ioctl.h>
-#include <sys/socket.h>
 
 #ifdef __svr4__
 #include <sys/sockio.h>
