--- setup.py	2022-10-26 23:55:58.000000000 +0800
+++ setup.py	2025-06-18 18:21:27.000000000 +0800
@@ -14,10 +14,7 @@
         return fp.read()
 
 
-extra_compile_args = ['-Wall', '-g', '-x', 'c++', '-std=c++11']
-
-if platform.system() == 'Darwin':
-    extra_compile_args += ['-stdlib=libc++']
+extra_compile_args = ['-Wall', '-g', '-x', 'c++', '-std=c++11', '-D_GLIBCXX_USE_CXX11_ABI=0']
 
 ext_modules = [
     Extension(
