From: barracuda156 <vital.had@gmail.com>
Date: Thu, 5 Oct 2023 08:11:51 +0800
Subject: [PATCH] Fix config for pre-release 10.6

--- gcc/config/darwin.h
+++ gcc/config/darwin.h
@@ -419,7 +419,6 @@
       %{%:sanitize(address): -lasan } \
       %{%:sanitize(undefined): -lubsan } \
       %(link_ssp) \
-      %:version-compare(>< 10.6 10.7 mmacosx-version-min= -ld10-uwfef) \
       %(link_gcc_c_sequence) \
       %{!nodefaultexport: \
 	%{%:version-compare(>= 10.11 asm_macosx_version_min= -U): \
@@ -551,17 +550,17 @@
 #undef REAL_LIBGCC_SPEC
 #define REAL_LIBGCC_SPEC \
 "%{static-libgcc|static:						  \
-    %:version-compare(!> 10.6 mmacosx-version-min= -lgcc_eh);		  \
+    %:version-compare(!> 10.7 mmacosx-version-min= -lgcc_eh);		  \
    shared-libgcc|fexceptions|fobjc-exceptions|fgnu-runtime:		  \
    " DARWIN_SHARED_LIBGCC "						  \
     %:version-compare(!> 10.3.9 mmacosx-version-min= -lgcc_eh)		  \
     %:version-compare(>< 10.3.9 10.5 mmacosx-version-min= -lgcc_s.10.4)   \
-    %:version-compare(>< 10.5 10.6 mmacosx-version-min= -lgcc_s.10.5)	  \
+    %:version-compare(>< 10.5 10.7 mmacosx-version-min= -lgcc_s.10.5)	  \
   } -lgcc "
 
 #define DARWIN_WEAK_CRTS \
 "%{static-libgcc|static:						  \
-    %{%:version-compare(>= 10.6 mmacosx-version-min= -lemutls_w):	  \
+    %{%:version-compare(>= 10.7 mmacosx-version-min= -lemutls_w):	  \
       " DARWIN_HEAP_T_LIB "} ;						  \
    shared-libgcc|fexceptions|fobjc-exceptions|fgnu-runtime: " \
      DARWIN_SHARED_WEAK_ADDS " ; \
