--- src/CMakeLists.txt	2025-07-19 14:49:41.000000000 +0800
+++ src/CMakeLists.txt	2025-12-07 15:54:37.000000000 +0800
@@ -96,10 +96,6 @@
     endif()
 endif()
 
-if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
-    target_compile_options(curlcpp PUBLIC -stdlib=libc++)
-endif()
-
 if(IPO_SUPPORTED AND CMAKE_BUILD_TYPE STREQUAL "Release")
     set_target_properties(curlcpp PROPERTIES INTERPROCEDURAL_OPTIMIZATION TRUE)
 endif()
