--- uhubctl.c	2024-09-01 07:30:35.000000000 +0800
+++ uhubctl.c	2025-06-14 22:21:52.000000000 +0800
@@ -9,7 +9,7 @@
  *
  */
 
-#define _XOPEN_SOURCE 500
+#define _DARWIN_C_SOURCE
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -29,13 +29,13 @@
 #include <unistd.h>
 #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__NetBSD__) || defined(_WIN32)
+#if defined(__FreeBSD__) || defined(__NetBSD__) || defined(_WIN32)
 #include <libusb.h>
 #else
 #include <libusb-1.0/libusb.h>
 #endif
 
-#if defined(__APPLE__) || defined(__FreeBSD__) /* snprintf is not available in pure C mode */
+#if defined(__FreeBSD__) /* snprintf is not available in pure C mode */
 int snprintf(char * __restrict __str, size_t __size, const char * __restrict __format, ...) __printflike(3, 4);
 #endif
 
