--- src/tap/Makefile.orig	2026-09-01 13:09:38.000000000 -0600
+++ src/tap/Makefile	2026-09-01 13:10:29.000000000 -0600
@@ -23,16 +23,16 @@
 	-I/System/Library/Frameworks/Kernel.framework/Headers \
 	-I/System/Library/Frameworks/Kernel.framework/Headers/bsd
 CFLAGS = -static -nostdinc -Wall -msoft-float -mlong-branch -force_cpusubtype_ALL -fno-builtin \
-	-arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk/ \
+	@ARCHFLAGS@ -isysroot /Developer/SDKs/MacOSX10.4u.sdk/ \
 	-DKERNEL -D__APPLE__ -DKERNEL_PRIVATE -DTUNTAP_VERSION=\"$(TUNTAP_VERSION)\" \
 	-DTAP_KEXT_VERSION=\"$(TAP_KEXT_VERSION)\"
 CCFLAGS = $(CFLAGS) -nostdinc++ -fapple-kext -fno-exceptions -fno-rtti
 LDFLAGS = -static -Wall -nostdlib -r -lcc_kext \
-	-arch ppc -arch i386 -isysroot /Developer/SDKs/MacOSX10.4u.sdk/
+	@ARCHFLAGS@ -isysroot /Developer/SDKs/MacOSX10.4u.sdk/
 
 
-CCP = g++
-CC = gcc
+CCP = @CXX@
+CC = @CC@
 
 all: $(KMOD_BIN) bundle
 
