From 0bd3122ee4230212f1c94f7b917538cdd1ac80e0 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sun, 1 Jun 2025 01:53:48 +0800
Subject: [PATCH 6/6] No git-version

---
 CMakeLists.txt  | 5 -----
 src/Tsunami.cpp | 3 +--
 2 files changed, 1 insertion(+), 7 deletions(-)

diff --git CMakeLists.txt CMakeLists.txt
index fe2a64ae..11c3ea1d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -290,11 +290,6 @@ set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 list(APPEND DEPS Threads::Threads)
 
-# git version
-include(.cmake/CheckGit.cmake)
-CheckGitSetup()
-list(APPEND DEPS git-version)
-
 
 #=======================================================================================
 #    target tsunami
diff --git src/Tsunami.cpp src/Tsunami.cpp
index a4994bba..4a981f23 100755
--- src/Tsunami.cpp
+++ src/Tsunami.cpp
@@ -37,12 +37,11 @@
 #include "view/module/ConfigPanel.h"
 #include "view/module/ConfigurationDialog.h"
 #endif
-#include "git-version.h"
 
 namespace tsunami {
 
 const string AppName = "Tsunami";
-const string AppVersion = GitHash;
+const string AppVersion = "@HASH@";
 const string AppNickname = "absolute 2er0";
 
 Tsunami *Tsunami::instance = nullptr;
-- 
2.48.0

