--- kitty/main.py	2022-09-05 13:19:50.000000000 +0800
+++ kitty/main.py	2026-01-07 23:25:09.000000000 +0800
@@ -103,7 +103,7 @@
 
 
 def init_glfw(opts: Options, debug_keyboard: bool = False, debug_rendering: bool = False) -> str:
-    glfw_module = 'cocoa' if is_macos else ('wayland' if is_wayland(opts) else 'x11')
+    glfw_module = 'x11' if is_macos else ('wayland' if is_wayland(opts) else 'x11')
     init_glfw_module(glfw_module, debug_keyboard, debug_rendering)
     return glfw_module
 
@@ -119,19 +119,6 @@
     candidates = []
     qkey = tuple(defn.split())
     candidates = func_map[qkey]
-    if candidates:
-        from .fast_data_types import cocoa_set_global_shortcut
-        alt_mods = GLFW_MOD_ALT, GLFW_MOD_ALT | GLFW_MOD_SHIFT
-        # Reverse list so that later defined keyboard shortcuts take priority over earlier defined ones
-        for candidate in reversed(candidates):
-            if candidate.mods in alt_mods:
-                # Option based shortcuts dont work in the global menubar,
-                # presumably because Apple reserves them for IME, see
-                # https://github.com/kovidgoyal/kitty/issues/3515
-                continue
-            if cocoa_set_global_shortcut(lookup_name or qkey[0], candidate[0], candidate[2]):
-                ans = candidate
-                break
     return ans
 
 
@@ -143,32 +130,6 @@
 
 def _run_app(opts: Options, args: CLIOptions, prewarm: PrewarmProcess, bad_lines: Sequence[BadLine] = ()) -> None:
     global_shortcuts: Dict[str, SingleKey] = {}
-    if is_macos:
-        from collections import defaultdict
-        func_map = defaultdict(list)
-        for k, v in opts.keymap.items():
-            parts = tuple(v.split())
-            func_map[parts].append(k)
-
-        for ac in ('new_os_window', 'close_os_window', 'close_tab', 'edit_config_file', 'previous_tab',
-                   'next_tab', 'new_tab', 'new_window', 'close_window', 'toggle_macos_secure_keyboard_entry', 'toggle_fullscreen'):
-            val = get_macos_shortcut_for(func_map, ac)
-            if val is not None:
-                global_shortcuts[ac] = val
-        val = get_macos_shortcut_for(func_map, 'clear_terminal reset active', lookup_name='reset_terminal')
-        if val is not None:
-            global_shortcuts['reset_terminal'] = val
-        val = get_macos_shortcut_for(func_map, 'clear_terminal to_cursor active', lookup_name='clear_terminal_and_scrollback')
-        if val is not None:
-            global_shortcuts['clear_terminal_and_scrollback'] = val
-        val = get_macos_shortcut_for(func_map, 'load_config_file', lookup_name='reload_config')
-        if val is not None:
-            global_shortcuts['reload_config'] = val
-        val = get_macos_shortcut_for(func_map, f'open_url {website_url()}', lookup_name='open_kitty_website')
-        if val is not None:
-            global_shortcuts['open_kitty_website'] = val
-    if is_macos and opts.macos_custom_beam_cursor:
-        set_custom_ibeam_cursor()
     if not is_wayland() and not is_macos:  # no window icons on wayland
         set_x11_window_icon()
     with cached_values_for(run_app.cached_values_name) as cached_values:
@@ -206,11 +167,6 @@
         finally:
             set_options(None)
             free_font_data()  # must free font data before glfw/freetype/fontconfig/opengl etc are finalized
-            if is_macos:
-                from kitty.fast_data_types import (
-                    cocoa_set_notification_activated_callback
-                )
-                cocoa_set_notification_activated_callback(None)
 
 
 run_app = AppRunner()
