--- kitty/data-types.c	2022-09-05 13:19:50.000000000 +0800
+++ kitty/data-types.c	2026-01-08 03:16:40.000000000 +0800
@@ -278,13 +278,10 @@
 extern bool init_png_reader(PyObject *module);
 extern bool init_utmp(PyObject *module);
 extern bool init_loop_utils(PyObject *module);
-#ifdef __APPLE__
-extern int init_CoreText(PyObject *);
-extern bool init_cocoa(PyObject *module);
-extern bool init_macos_process_info(PyObject *module);
-#else
 extern bool init_freetype_library(PyObject*);
 extern bool init_freetype_render_ui_text(PyObject*);
+#ifdef __APPLE__
+extern bool init_macos_process_info(PyObject *module);
 #endif
 
 static unsigned
@@ -335,14 +332,11 @@
     if (!init_png_reader(m)) return NULL;
 #ifdef __APPLE__
     if (!init_macos_process_info(m)) return NULL;
-    if (!init_CoreText(m)) return NULL;
-    if (!init_cocoa(m)) return NULL;
-#else
+#endif
     if (!init_freetype_library(m)) return NULL;
     if (!init_fontconfig_library(m)) return NULL;
     if (!init_desktop(m)) return NULL;
     if (!init_freetype_render_ui_text(m)) return NULL;
-#endif
     if (!init_fonts(m)) return NULL;
     if (!init_utmp(m)) return NULL;
     if (!init_loop_utils(m)) return NULL;
