--- kitty/main.py.orig	2022-09-05 13:19:50.000000000 +0800
+++ kitty/main.py	2026-01-07 23:25:09.000000000 +0800
@@ -103,7 +103,7 @@
 
 
 def init_glfw(opts: Options, debug_keyboard: bool = False, debug_rendering: bool = False) -> str:
-    glfw_module = 'cocoa' if is_macos else ('wayland' if is_wayland(opts) else 'x11')
+    glfw_module = 'x11' if is_macos else ('wayland' if is_wayland(opts) else 'x11')
     init_glfw_module(glfw_module, debug_keyboard, debug_rendering)
     return glfw_module
 
