--- kitty/boss.py.orig	2022-09-05 13:19:50.000000000 +0800
+++ kitty/boss.py	2026-01-08 03:49:03.000000000 +0800
@@ -281,11 +281,6 @@
         self.global_shortcuts = global_shortcuts
         self.mouse_handler: Optional[Callable[[WindowSystemMouseEvent], None]] = None
         self.update_keymap()
-        if is_macos:
-            from .fast_data_types import (
-                cocoa_set_notification_activated_callback
-            )
-            cocoa_set_notification_activated_callback(notification_activated)
 
     def update_keymap(self) -> None:
         self.keymap = get_options().keymap.copy()
