--- kitty/fonts/render.py.orig	2022-09-05 13:19:50.000000000 +0800
+++ kitty/fonts/render.py	2026-01-07 23:23:01.000000000 +0800
@@ -24,30 +24,19 @@
 from kitty.types import _T
 from kitty.utils import log_error
 
-if is_macos:
-    from .core_text import (
-        find_font_features, font_for_family as font_for_family_macos,
-        get_font_files as get_font_files_coretext
-    )
-else:
-    from .fontconfig import (
-        find_font_features, font_for_family as font_for_family_fontconfig,
-        get_font_files as get_font_files_fontconfig
-    )
+from .fontconfig import (
+    find_font_features, font_for_family as font_for_family_fontconfig,
+    get_font_files as get_font_files_fontconfig
 
 FontObject = Union[CoreTextFont, FontConfigPattern]
 current_faces: List[Tuple[FontObject, bool, bool]] = []
 
 
 def get_font_files(opts: Options) -> Dict[str, Any]:
-    if is_macos:
-        return get_font_files_coretext(opts)
     return get_font_files_fontconfig(opts)
 
 
 def font_for_family(family: str) -> Tuple[FontObject, bool, bool]:
-    if is_macos:
-        return font_for_family_macos(family)
     return font_for_family_fontconfig(family)
 
 
