--- setup.py	2022-09-05 13:19:50.000000000 +0800
+++ setup.py	2026-01-07 22:00:34.000000000 +0800
@@ -746,9 +746,9 @@
     ans, headers = [], []
     d = 'kitty'
     exclude = {
-        'fontconfig.c', 'freetype.c', 'desktop.c', 'freetype_render_ui_text.c'
+        'core_text.m', 'cocoa_window.m'
     } if is_macos else {
-        'core_text.m', 'cocoa_window.m', 'macos_process_info.c'
+        'macos_process_info.c'
     }
     for x in sorted(os.listdir(d)):
         ext = os.path.splitext(x)[1]
@@ -761,7 +761,7 @@
 
 
 def compile_glfw(compilation_database: CompilationDatabase) -> None:
-    modules = 'cocoa' if is_macos else 'x11 wayland'
+    modules = 'x11' if is_macos else 'x11 wayland'
     for module in modules.split():
         try:
             genv = glfw.init_env(env, pkg_config, pkg_version, at_least_version, test_compile, module)
