--- kitty/fontconfig.c	2022-09-05 13:19:50.000000000 +0800
+++ kitty/fontconfig.c	2026-01-08 03:39:48.000000000 +0800
@@ -83,6 +83,7 @@
 #if defined(_KITTY_FONTCONFIG_LIBRARY)
             _KITTY_FONTCONFIG_LIBRARY,
 #else
+            "@PREFIX@/lib/libfontconfig.dylib",
             "libfontconfig.so",
             // some installs are missing the .so symlink, so try the full name
             "libfontconfig.so.1",
