--- src/common/file_location.c.orig	2025-10-08 03:57:41.000000000 +0800
+++ src/common/file_location.c	2025-10-22 10:07:03.000000000 +0800
@@ -31,10 +31,6 @@
 #include <config.h>
 #endif
 
-#ifdef __APPLE__
-#include "osx/osx.h"
-#endif
-
 #include "common/grealpath.h"
 #include "darktable.h"
 #include "file_location.h"
@@ -133,7 +129,7 @@
       // default_value is relative.
       // combine basename (application_directory) and relative path (default_value).
       gchar complete_path[PATH_MAX] = { 0 };
-#if defined(__APPLE__)
+#if 0
       char *bundle_path = dt_osx_get_bundle_res_path();
       if(bundle_path)
       {
