--- src/osx/osx.mm	2025-10-08 03:57:41.000000000 +0800
+++ src/osx/osx.mm	2025-10-22 09:32:39.000000000 +0800
@@ -22,8 +22,11 @@
 #include <ApplicationServices/ApplicationServices.h>
 #include <CoreServices/CoreServices.h>
 #include <AppKit/AppKit.h>
+#include <Foundation/Foundation.h>
 #include <gtk/gtk.h>
+#ifdef GDK_WINDOWING_QUARTZ
 #include <gdk/gdkquartz.h>
+#endif
 #include <gio/gio.h>
 #include <glib.h>
 #include <string.h>

--- src/osx/osx.h	2025-10-08 03:57:41.000000000 +0800
+++ src/osx/osx.h	2025-10-22 09:41:03.000000000 +0800
@@ -22,8 +22,10 @@
 
 #ifdef __cplusplus
 extern "C" {
+#ifdef GDK_WINDOWING_QUARTZ
 #include <gdk/quartz/gdkquartz-cocoa-access.h>
 #endif
+#endif
 
 void dt_osx_autoset_dpi(GtkWidget *widget);
 float dt_osx_get_ppd();
