--- src/external/rawspeed/cmake/compiler-versions.cmake	2025-04-06 15:28:29.000000000 +0800
+++ src/external/rawspeed/cmake/compiler-versions.cmake	2025-10-22 08:33:40.000000000 +0800
@@ -52,10 +52,6 @@
   message(SEND_ERROR "XCode (Apple clang) C++ compiler version ${CMAKE_CXX_COMPILER_VERSION} is too old and is unsupported. XCode version 15.2+ is required.")
 endif()
 
-if(CMAKE_OSX_DEPLOYMENT_TARGET AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 13.5)
-  message(SEND_ERROR "Targeting OSX version ${CMAKE_OSX_DEPLOYMENT_TARGET} older than 13.5 is unsupported.")
-endif()
-
 include(CheckCXXSourceCompiles)
 
 if(NOT DEFINED RAWSPEED_LIBSTDCXX_MIN)

--- cmake/compiler-versions.cmake	2025-10-08 03:57:41.000000000 +0800
+++ cmake/compiler-versions.cmake	2025-10-22 08:40:41.000000000 +0800
@@ -20,7 +20,3 @@
 if(CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 14.0.0.14000029)
   message(SEND_ERROR "XCode (Apple clang) C++ compiler version ${CMAKE_CXX_COMPILER_VERSION} is too old and is unsupported. XCode version 14.2+ is required.")
 endif()
-
-if(CMAKE_OSX_DEPLOYMENT_TARGET AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_LESS 12.5)
-  message(SEND_ERROR "Targeting OSX version ${CMAKE_OSX_DEPLOYMENT_TARGET} older than 12.5 is unsupported.")
-endif()
\ No newline at end of file
