From 59344dd08800fc7210f3b2ad2a563a26534a225d Mon Sep 17 00:00:00 2001
From: Arxtage <tsaturyanarmann@gmail.com>
Date: Tue, 1 Apr 2025 14:13:09 +0100
Subject: [PATCH] fix reels download

---
 instaloader/structures.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/instaloader/structures.py b/instaloader/structures.py
index 29a80df5..127e0cb3 100644
--- a/instaloader/structures.py
+++ b/instaloader/structures.py
@@ -1265,7 +1265,7 @@ def get_reels(self) -> NodeIterator[Post]:
             query_referer = 'https://www.instagram.com/{0}/'.format(self.username),
             is_first = Profile._make_is_newest_checker(),
             # fb_api_req_friendly_name=PolarisProfileReelsTabContentQuery_connection
-            doc_id = '7845543455542541',
+            doc_id = '8471403382962070',
             query_hash = None,
         )
 
