--- test/CMakeLists.txt	2024-08-12 15:42:19.000000000 +0800
+++ test/CMakeLists.txt	2025-01-02 02:14:07.000000000 +0800
@@ -36,14 +36,6 @@
   add_executable(${target} ${ARGN})
   target_link_libraries(${target} ${PROJECT_NAME} GTest::gtest)
 
-  if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
-    # GCC is not strict enough by default, so enable most of the warnings.
-    target_compile_options(${target} PRIVATE
-      -Werror=all
-      -Werror=extra
-      )
-  endif()
-
   add_test(NAME ${target} COMMAND ${target})
 endfunction()
 
