--- src/dune	2023-07-28 00:15:26.000000000 +0800
+++ src/dune	2025-10-17 16:52:36.000000000 +0800
@@ -2,9 +2,12 @@
  (deps (:first_dep ../stubgen/ffi_stubgen.exe))
  (action (bash "%{first_dep} -ml > ffi_generated.ml")))
 
-(rule (targets ffi_generated_stubs.c)
+(rule
+ (targets ffi_generated_stubs.c)
  (deps (:first_dep ../stubgen/ffi_stubgen.exe))
- (action (bash "%{first_dep} -c > ffi_generated_stubs.c")))
+ (action
+  (bash
+   "printf '%s\n' '#if defined(__GNUC__)' '#pragma GCC diagnostic push' '#pragma GCC diagnostic ignored \"-Wincompatible-pointer-types\"' '#endif' > ffi_generated_stubs.c && %{first_dep} -c >> ffi_generated_stubs.c && printf '%s\n' '#if defined(__GNUC__)' '#pragma GCC diagnostic pop' '#endif' >> ffi_generated_stubs.c")))
 
 (rule (targets ffi_generated_types.ml)
  (deps (:first_dep ../stubgen/ffi_ml_types_stubgen.exe))
