--- src/texteditor.cpp.orig	2014-08-02 06:58:39.000000000 +0800
+++ src/texteditor.cpp	2026-06-14 07:07:48.000000000 +0800
@@ -25,7 +25,17 @@
 ********************************************************************************/
 
 
-#include <QtWidgets>
+#include <QtGui>
+#include <QFile>
+#include <QFileInfo>
+#include <QMenu>
+#include <QMenuBar>
+#include <QMessageBox>
+#include <QSettings>
+#include <QStatusBar>
+#include <QTextEdit>
+#include <QToolBar>
+
 #include "texteditor.h"
 
 TextEditor::TextEditor(const QString &fileName)
