--- socnetv.pro	2014-08-02 06:58:39.000000000 +0800
+++ socnetv.pro	2026-06-14 06:24:39.000000000 +0800
@@ -32,6 +32,7 @@
 LANGUAGE = C++
 
 # support
+QT += webkit
 QT += webkitwidgets
 QT += xml 
 QT += network
@@ -85,11 +86,11 @@
 
 
 # Extra optimization flags
-QMAKE_CXXFLAGS += -msse -mfpmath=sse -ffast-math 
+QMAKE_CXXFLAGS += -ffast-math 
 
 #LIBS    += @ac_libs@
-INCLUDEPATH +=  /usr/local/include /usr/include /usr/include/qt5 /usr/share/qt5/include 
-INCLUDEPATH +=  /usr/local/include /usr/include /usr/include/qt  /usr/include/qt5 /usr/share/qt5/include
+#INCLUDEPATH +=  /usr/local/include /usr/include /usr/include/qt5 /usr/share/qt5/include 
+#INCLUDEPATH +=  /usr/local/include /usr/include /usr/include/qt  /usr/include/qt5 /usr/share/qt5/include
 
 RESOURCES = src/src.qrc
 win32 {
