--- src/mainwindow.cpp	2014-08-02 06:58:39.000000000 +0800
+++ src/mainwindow.cpp	2026-06-14 07:00:37.000000000 +0800
@@ -29,10 +29,24 @@
 #endif
 
 
-#include <QtWidgets>
+#include <QtGui>
 #include <QtGlobal>
 #include <QtDebug>
+#include <QFileDialog>
+#include <QInputDialog>
 #include <QPrintDialog>
+#include <QProgressDialog>
+#include <QMenu>
+#include <QMenuBar>
+#include <QStatusBar>
+#include <QToolBar>
+#include <QGroupBox>
+#include <QMessageBox>
+#include <QByteArray>
+#include <QString>
+#include <qevent.h>
+#include <qlcdnumber.h>
+#include <qpushbutton.h>
 
 #include <ctime> 
 
@@ -55,8 +69,9 @@
 bool printDebug=false;
 
 
-void myMessageOutput( QtMsgType type, const QMessageLogContext &context, const QString &msg )     {
-    QByteArray localMsg = msg.toLocal8Bit();
+void myMessageOutput( QtMsgType type, const char *msg ) {
+    QString message = QString::fromLocal8Bit(msg);
+    QByteArray localMsg = message.toLocal8Bit();
     if (printDebug)
         switch ( type ) {
         case QtDebugMsg:
@@ -78,7 +93,7 @@
 /** MainWindow contruction method **/
 MainWindow::MainWindow(const QString & m_fileName) {
 
-    qInstallMessageHandler( myMessageOutput );
+    qInstallMsgHandler( myMessageOutput );
 
     setWindowIcon (QIcon(":/images/socnetv.png"));
 
