--- fuzzing/CMakeLists.txt
+++ fuzzing/CMakeLists.txt	2025-12-21 09:29:27.000000000 +0800
@@ -32,7 +32,7 @@
         target_link_libraries(fuzz_${testname} PRIVATE main)
     endif()
 
-    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         target_compile_options(fuzz_${testname} PUBLIC -fsanitize=address,undefined -fno-sanitize-recover=all )
         target_link_options(fuzz_${testname} PUBLIC -fsanitize=address,undefined -fno-sanitize-recover=all )
     else()
@@ -65,7 +65,7 @@
     target_link_libraries(${testname} PRIVATE glaze::glaze)
     # do not set sanitizers here - CI needs to run as fast as possible,
     # manual running may want sanitizers and debug.
-    if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
+    if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
         #target_compile_options(${testname} PRIVATE -O2)
         #target_compile_options(${testname} PRIVATE -fsanitize=address,undefined  -fno-sanitize-recover=all )
         #target_link_options(${testname} PRIVATE -fsanitize=address,undefined  -fno-sanitize-recover=all )
