--- configure	2025-08-11 17:10:38.000000000 +0800
+++ configure	2025-10-20 06:13:58.000000000 +0800
@@ -19061,7 +19061,7 @@
 if test "x$GCC" = xyes
 then :
       CFLAGS="$CFLAGS -Wall"
-    CFLAGS="$CFLAGS -Wmissing-declarations -Wmissing-prototypes -Werror=incompatible-pointer-types -Werror=implicit -Werror=int-conversion"
+    CFLAGS="$CFLAGS -Wmissing-declarations -Wmissing-prototypes -Werror=implicit"
     # FIXME: Check for --enable-debug here
     CFLAGS="$CFLAGS -g"
     LDFLAGS="$LDFLAGS -g"
