--- mz.h.orig	2026-06-30 10:44:28.000000000 -0600
+++ mz.h	2026-09-06 14:52:02.000000000 -0600
@@ -54,6 +54,13 @@
 #define MZ_OPEN_MODE_CREATE    (0x08)
 #define MZ_OPEN_MODE_EXISTING  (0x10)
 #define MZ_OPEN_MODE_NOFOLLOW  (0x20)
+/* Mac OS X 10.4 and below does not have O_NOFOLLOW */
+#if defined(__APPLE__) 
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#define NO_O_NOFOLLOW
+#endif
+#endif
 
 /* MZ_SEEK */
 #define MZ_SEEK_SET (0)
