--- mz_zip_rw.c.orig	2026-09-06 13:41:48.000000000 -0600
+++ mz_zip_rw.c	2026-09-06 13:53:26.000000000 -0600
@@ -1704,8 +1704,10 @@
         stream = mz_stream_os_create();
         if (!stream)
             return MZ_STREAM_ERROR;
+        #ifndef NO_O_NOFOLLOW
         if (!writer->follow_links)
             mode |= MZ_OPEN_MODE_NOFOLLOW;
+        #endif
         err = mz_stream_os_open(stream, path, mode);
     }
 
