--- mz.h.orig	2026-06-30 10:44:28.000000000 -0600
+++ mz.h	2026-09-06 14:52:02.000000000 -0600
@@ -54,6 +54,13 @@
 #define MZ_OPEN_MODE_CREATE    (0x08)
 #define MZ_OPEN_MODE_EXISTING  (0x10)
 #define MZ_OPEN_MODE_NOFOLLOW  (0x20)
+/* Mac OS X 10.4 and below does not have O_NOFOLLOW */
+#if defined(__APPLE__)
+#include <AvailabilityMacros.h>
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1050
+#define NO_O_NOFOLLOW
+#endif
+#endif
 
 /* MZ_SEEK */
 #define MZ_SEEK_SET (0)
--- mz_strm_os_posix.c.orig	2026-09-06 13:09:30.000000000 -0600
+++ mz_strm_os_posix.c	2026-09-06 13:55:06.000000000 -0600
@@ -89,8 +89,10 @@
     } else
         return MZ_OPEN_ERROR;
 
+    #ifndef NO_O_NOFOLLOW
     if (mode & MZ_OPEN_MODE_NOFOLLOW)
         mode_open |= O_NOFOLLOW;
+    #endif
 
     fd = open(path, mode_open, S_IRUSR | S_IWUSR | S_IRGRP);
     if (fd != -1) {
--- mz_zip_rw.c.orig	2026-09-06 13:41:48.000000000 -0600
+++ mz_zip_rw.c	2026-09-06 13:53:26.000000000 -0600
@@ -1704,8 +1704,10 @@
         stream = mz_stream_os_create();
         if (!stream)
             return MZ_STREAM_ERROR;
+        #ifndef NO_O_NOFOLLOW
         if (!writer->follow_links)
             mode |= MZ_OPEN_MODE_NOFOLLOW;
+        #endif
         err = mz_stream_os_open(stream, path, mode);
     }
 
