From a04c027917c0f5efbb12bec563ec5469c7d3eb86 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Tue, 2 Dec 2025 23:10:58 +0800
Subject: [PATCH] fsck.c: add missing header

---
 server/gc/fsck.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/server/gc/fsck.c b/server/gc/fsck.c
index e47d0e4e..3a104650 100644
--- server/gc/fsck.c
+++ server/gc/fsck.c
@@ -3,6 +3,7 @@
 #include <fcntl.h>
 
 #include "seafile-session.h"
+#include "seaf-utils.h"
 #include "log.h"
 #include "utils.h"
 
