diff --git a/lib/pure/hashes.nim b/lib/pure/hashes.nim
index c0171237d..c1a092b9a 100644
--- a/lib/pure/hashes.nim
+++ b/lib/pure/hashes.nim
@@ -304,6 +304,8 @@ else:
 proc rotl32(x: uint32, r: int): uint32 {.inline.} =
   (x shl r) or (x shr (32 - r))
 
+from std/endians import littleEndian64, littleEndian32
+
 proc murmurHash(x: openArray[byte]): Hash =
   # https://github.com/PeterScott/murmur3/blob/master/murmur3.c
   const
@@ -335,7 +337,7 @@ proc murmurHash(x: openArray[byte]): Hash =
       impl()
     else:
       when declared(copyMem):
-        copyMem(addr k1, addr x[i], 4)
+        littleEndian32 addr k1, addr x[i]
       else:
         impl()
     inc i, stepSize
@@ -399,7 +401,7 @@ proc load4(s: openArray[byte], o=0): uint32 {.inline.} =
   else:
     when declared copyMem:
       result = uint32(0)
-      copyMem result.addr, s[o].addr, result.sizeof
+      littleEndian32 addr result, addr s[o]
     else: result = load4e(s, o)
 
 proc load8(s: openArray[byte], o=0): uint64 {.inline.} =
@@ -407,7 +409,7 @@ proc load8(s: openArray[byte], o=0): uint64 {.inline.} =
   else:
     when declared copyMem:
       result = uint64(0)
-      copyMem result.addr, s[o].addr, result.sizeof
+      littleEndian64 addr result, addr s[o]
     else: result = load8e(s, o)
 
 proc lenU(s: openArray[byte]): uint64 {.inline.} = s.len.uint64
