--- a/compiler/lexer.nim
+++ b/compiler/lexer.nim	2025-12-27 00:42:01.000000000 +0800
@@ -484,11 +484,12 @@
         of tkUInt16Lit: setNumber result.iNumber, xi and 0xffff
         of tkUInt32Lit: setNumber result.iNumber, xi and 0xffffffff
         of tkFloat32Lit:
-          setNumber result.fNumber, (cast[ptr float32](addr(xi)))[]
+          let xi2 = uint32(xi)
+          setNumber result.fNumber, (cast[float32](xi2))
           # note: this code is endian neutral!
           # XXX: Test this on big endian machine!
         of tkFloat64Lit, tkFloatLit:
-          setNumber result.fNumber, (cast[ptr float64](addr(xi)))[]
+          setNumber result.fNumber, (cast[float64](xi))
         else: internalError(L.config, getLineInfo(L), "getNumber")
 
         # Bounds checks. Non decimal literals are allowed to overflow the range of
