--- libs/indibase/dsp/manager.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ libs/indibase/dsp/manager.h	2025-08-30 04:34:45.000000000 +0800
@@ -27,6 +27,7 @@
 #include <fitsio.h>
 #include <functional>
 #include <string>
+#include <cstdint>
 
 namespace INDI
 {

--- libs/indibase/dsp/transforms.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ libs/indibase/dsp/transforms.h	2025-08-30 04:34:59.000000000 +0800
@@ -24,6 +24,7 @@
 #include "dsp.h"
 
 #include <string>
+#include <cstdint>
 
 namespace DSP
 {

--- libs/indibase/dsp/convolution.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ libs/indibase/dsp/convolution.h	2025-08-30 04:33:29.000000000 +0800
@@ -25,6 +25,7 @@
 #include <fitsio.h>
 #define N_WAVELETS 7
 #include <string>
+#include <cstdint>
 
 namespace DSP
 {

--- libs/indibase/dsp/dspinterface.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ libs/indibase/dsp/dspinterface.h	2025-08-30 04:32:46.000000000 +0800
@@ -26,6 +26,7 @@
 #include <fitsio.h>
 #include <functional>
 #include <string>
+#include <cstdint>
 
 namespace INDI
 {

--- libs/indibase/connectionplugins/connectionserial.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ libs/indibase/connectionplugins/connectionserial.h	2025-08-30 04:32:11.000000000 +0800
@@ -24,6 +24,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace Connection
 {

--- drivers/telescope/celestrondriver.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ drivers/telescope/celestrondriver.h	2025-08-30 04:49:28.000000000 +0800
@@ -31,6 +31,8 @@
 #ifdef __FreeBSD__
 #include <stdint.h>
 typedef uint8_t u_int8_t;
+#else
+#include <cstdint>
 #endif
 
 //#include <thread>

--- drivers/telescope/ieqdriverbase.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ drivers/telescope/ieqdriverbase.h	2025-08-30 04:57:54.000000000 +0800
@@ -22,6 +22,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace iEQ
 {

--- drivers/telescope/ieqprolegacydriver.h.orig	2022-01-17 18:21:47.000000000 +0800
+++ drivers/telescope/ieqprolegacydriver.h	2025-08-30 04:58:04.000000000 +0800
@@ -21,6 +21,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 typedef enum { GPS_OFF, GPS_ON, GPS_DATA_OK } IEQ_GPS_STATUS;
 typedef enum {

--- drivers/auxiliary/flip_flat.cpp.orig	2022-01-17 18:21:47.000000000 +0800
+++ drivers/auxiliary/flip_flat.cpp	2025-08-30 05:08:03.000000000 +0800
@@ -27,12 +27,16 @@
 #include "indicom.h"
 #include "connectionplugins/connectionserial.h"
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
 #include <cerrno>
 #include <cstring>
 #include <memory>
 #include <termios.h>
 #include <unistd.h>
-#include <inttypes.h>
+#include <cinttypes>
 #include <sys/ioctl.h>
 
 // We declare an auto pointer to FlipFlat.
