--- src/slirp.c	2025-05-28 06:38:05.000000000 +0800
+++ src/slirp.c	2026-05-02 00:55:25.000000000 +0800
@@ -966,14 +966,7 @@
         fd = (int) so->s;
         if ((slirp_os_socket) fd != so->s)
             g_warning_once("Truncating socket to int failed!");
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
         slirp->cb->register_poll_fd(fd, slirp->opaque);
-#ifdef __GNUC__
-#pragma GCC diagnostic pop
-#endif
     }
 }
 
@@ -987,14 +980,7 @@
         fd = (int) so->s;
         if ((slirp_os_socket) fd != so->s)
             g_warning_once("Truncating socket to int failed!");
-#ifdef __GNUC__
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#endif
         slirp->cb->unregister_poll_fd(fd, slirp->opaque);
-#ifdef __GNUC__
-#pragma GCC diagnostic pop
-#endif
     }
 }
 
