--- src/CMakeLists.txt	2026-05-17 21:52:16.000000000 +0800
+++ src/CMakeLists.txt	2026-05-18 03:20:00.000000000 +0800
@@ -166,6 +166,7 @@
 		PRIVATE mulecommon
 		PRIVATE mulesocket
 		PRIVATE wxWidgets::NET
+		PRIVATE atomic
 	)
 
 	if (GLIB_FOUND)
@@ -253,6 +254,7 @@
 		PRIVATE mulesocket
 		PRIVATE $<$<VERSION_LESS:${wxWidgets_VERSION_STRING},3.1.2>:wxWidgets::ADV>
 		PRIVATE wxWidgets::NET
+		PRIVATE atomic
 	)
 
 	if (HAVE_BFD)
