--- Code/GraphMol/SynthonSpaceSearch/CMakeLists.txt	2026-01-30 14:55:37.000000000 +0800
+++ Code/GraphMol/SynthonSpaceSearch/CMakeLists.txt	2026-02-04 03:48:55.000000000 +0800
@@ -4,7 +4,7 @@
         SynthonSpaceSearcher.cpp SynthonSpaceSubstructureSearcher.cpp SynthonSpaceFingerprintSearcher.cpp
         SynthonSpaceRascalSearcher.cpp SynthonSpaceHitSet.cpp SearchResults.cpp MemoryMappedFileReader.cpp
         LINK_LIBRARIES SmilesParse FileParsers ChemTransforms Fingerprints SubstructMatch GraphMol RascalMCES
-        GeneralizedSubstruct Descriptors)
+        GeneralizedSubstruct Descriptors atomic)
 target_compile_definitions(SynthonSpaceSearch PRIVATE RDKIT_SYNTHONSPACESEARCH_BUILD)
 
 rdkit_headers(SynthonSpace.h SynthonSet.h Synthon.h SearchResults.h
