From acbc65312e70b92bf4c4027ec84a49a9b6804846 Mon Sep 17 00:00:00 2001
From: Brian Kelley <fustigator@gmail.com>
Date: Thu, 5 Feb 2026 16:24:14 -0500
Subject: [PATCH] Fixes #9077

---
 External/ChemDraw/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/External/ChemDraw/CMakeLists.txt b/External/ChemDraw/CMakeLists.txt
index 1e7f19d808d..dd26e23a25d 100644
--- External/ChemDraw/CMakeLists.txt
+++ External/ChemDraw/CMakeLists.txt
@@ -10,8 +10,8 @@ include(CMakePrintHelpers)
 # and will need to be fixed in the future
 if(RDK_BUILD_CHEMDRAW_SUPPORT)
   if(NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/chemdraw/chemdraw/CDXIO.h")
-    set(RELEASE_NO "1.0.11")
-    set(MD5 "8f2a65313eac027258423bd46feda5bf")
+    set(RELEASE_NO "1.0.12")
+    set(MD5 "8abbd599d798e980b63394a939a08b38")
     downloadAndCheckMD5("https://codeload.github.com/Glysade/chemdraw/tar.gz/refs/tags/v${RELEASE_NO}"
       "${CMAKE_CURRENT_SOURCE_DIR}/chemdraw-v${RELEASE_NO}.tar.gz" ${MD5})
 
