From 88eedd32d11b1f600d201c93cb33d4e6b7caef6e Mon Sep 17 00:00:00 2001
From: ManoloFLTK <41016272+ManoloFLTK@users.noreply.github.com>
Date: Thu, 30 Apr 2026 09:03:16 +0200
Subject: [PATCH] Restore building for macos with SDK 10.6 (#1424)

---
 src/Fl_cocoa.mm | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/Fl_cocoa.mm b/src/Fl_cocoa.mm
index 725d2c2010..6a21dac780 100644
--- src/Fl_cocoa.mm
+++ src/Fl_cocoa.mm
@@ -165,6 +165,7 @@
 const NSUInteger NSEventModifierFlagControl = NSControlKeyMask;
 const NSUInteger NSEventModifierFlagShift = NSShiftKeyMask;
 const NSUInteger NSEventModifierFlagCapsLock = NSAlphaShiftKeyMask;
+const NSUInteger NSEventModifierFlagFunction = NSFunctionKeyMask;
 
 const NSEventType NSEventTypeLeftMouseDown = NSLeftMouseDown;
 const NSEventType NSEventTypeRightMouseDown = NSRightMouseDown;
