--- a/CMakeLists.txt	2026-05-13 02:27:10.000000000 +0800
+++ b/CMakeLists.txt	2026-05-15 10:35:57.000000000 +0800
@@ -50,7 +50,7 @@
 # define the FLTK project and version
 #######################################################################
 
-project(FLTK VERSION 1.5.0)
+project(FLTK VERSION 1.5.0 LANGUAGES C CXX OBJCXX)
 
 #######################################################################
 # set the required C++ standard -- may be overridden by the user
