--- src/Fl_cocoa.mm	2025-04-04 21:54:36.000000000 +0800
+++ src/Fl_cocoa.mm	2025-04-06 04:41:57.000000000 +0800
@@ -2758,7 +2758,7 @@
     NSData *data = [pboard dataForType:UTF8_pasteboard_type];
     DragData = (char *)malloc([data length] + 1);
     [data getBytes:DragData length:[data length]];
-    DragData[[data length]] = 0;
+    DragData[([data length])] = 0;
     Fl_Screen_Driver::convert_crlf(DragData, strlen(DragData));
   }
   else {

--- src/Fl_Native_File_Chooser_MAC.mm	2025-04-04 21:54:36.000000000 +0800
+++ src/Fl_Native_File_Chooser_MAC.mm	2025-04-06 04:44:42.000000000 +0800
@@ -498,7 +498,7 @@
   BOOL isdir = NO;
   [[NSFileManager defaultManager] fileExistsAtPath:filename isDirectory:&isdir];
   if (isdir) return YES;
-  if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[ [nspopup indexOfSelectedItem] ]) ) return YES;
+  if ( fl_filename_match([filename fileSystemRepresentation], filter_pattern[([nspopup indexOfSelectedItem])]) ) return YES;
   return NO;
 }
 - (BOOL)panel:(id)sender shouldEnableURL:(NSURL *)url
