--- pyproject.toml	1970-01-01 08:00:00.000000000 +0800
+++ pyproject.toml	2024-01-05 02:03:11.000000000 +0800
@@ -1,39 +1,34 @@
-[project]
-name = "xmod"
-version = "1.9.0"
-description = "🌱 Turn any object into a module 🌱"
-authors = [{ name = "Tom Ritchford", email = "tom@swirly.com" }]
-requires-python = ">=3.10"
-readme = "README.md"
-license = "MIT"
-classifiers = ["Programming Language :: Python :: 3", "Programming Language :: Python :: 3.10", "Programming Language :: Python :: 3.11", "Programming Language :: Python :: 3.12", "Programming Language :: Python :: 3.13", "Programming Language :: Python :: 3.14"]
-
-[project.urls]
-Homepage = "https://github.com/rec/xmod"
-Repository = "https://github.com/rec/xmod"
-Documentation = "https://rec.github.io/xmod"
+[tool.black]
+line-length = 79
+skip-string-normalization = true
+target-version = ['py310']
 
 [tool.doks]
 auto = true
 
-[dependency-groups]
-dev = [
-    "coverage",
-    "pytest",
-    "pyupgrade>=3.21.2",
-    "ruff",
-    "ty>=0.0.14",
-]
-
-[tool.uv]
-
-[tool.uv.build-backend]
-module-root = ""
-source-include = ["xmod.pyi/**"]
-
-[build-system]
-requires = ["uv_build>=0.9.0,<0.10.0"]
-build-backend = "uv_build"
+[tool.poetry]
+name = "xmod"
+version = "1.9.0"
+description = "🌱 Turn any object into a module 🌱"
+authors = ["Tom Ritchford <tom@swirly.com>"]
+license = "MIT"
+readme = "README.md"
+include = ["xmod.pyi"]
+repository = "https://github.com/rec/xmod"
+homepage = "https://github.com/rec/xmod"
+documentation = "https://rec.github.io/xmod"
+
+[tool.poetry.dependencies]
+python = ">=3.10"
+
+[tool.poetry.group.dev.dependencies]
+black = "*"
+coverage = "*"
+flake8 = "*"
+isort = "*"
+mypy = "*"
+pytest = "*"
+ruff = "*"
 
 [tool.coverage.run]
 branch = true
@@ -43,10 +38,6 @@
 fail_under = "91"
 skip_covered = true
 exclude_lines = ["pragma: no cover", "if False:", "if __name__ == .__main__.:", "raise NotImplementedError"]
-
-[tool.ruff]
-line-length = 88
-
-[tool.ruff.format]
-quote-style = "single"
-
+[build-system]
+requires = ["poetry-core"]
+build-backend = "poetry.core.masonry.api"
