--- Makefile	2025-02-15 00:39:10.000000000 +0800
+++ Makefile	2026-04-26 00:40:53.000000000 +0800
@@ -36,10 +36,10 @@
 # Make it easier to find the OpenCL headers on systems
 # that don't ship them by default; the user can just clone
 # them on a parallel directory from the official repository
-CPPFLAGS += -I../OpenCL-Headers
+# CPPFLAGS += -I../OpenCL-Headers
 
 CFLAGS ?= -g -pedantic -Werror
-CFLAGS += -std=c99 -Wall -Wextra
+CFLAGS += -std=c11 -Wall -Wextra
 
 SPARSE ?= sparse
 SPARSEFLAGS=-Wsparse-all -Wno-decl
@@ -48,7 +48,7 @@
 RM ?= rm -f
 
 # Installation paths and modes
-PREFIX ?= /usr/local
+PREFIX ?= /opt/local
 BINDIR ?= $(PREFIX)/bin
 BINMODE ?= 555
 MANDIR ?= $(PREFIX)/man
@@ -65,8 +65,8 @@
 LDLIBS__common = -lOpenCL -ldl
 
 # OS-specific library includes
-LDLIBS_Darwin = -framework OpenCL
-LDLIBS_Darwin_exclude = -lOpenCL
+# LDLIBS_Darwin = -framework OpenCL
+# LDLIBS_Darwin_exclude = -lOpenCL
 
 LDLIBS += $(LDLIBS_${OS}) $(LDLIBS__common:$(LDLIBS_${OS}_exclude)=)
 
