--- src/ext.h	2025-02-15 00:39:10.000000000 +0800
+++ src/ext.h	2026-04-26 00:49:35.000000000 +0800
@@ -14,7 +14,7 @@
  * of the 3.0 headers lying around, which may prevent clinfo from being
  * compilable.
  */
-#define CL_TARGET_OPENCL_VERSION 220
+#define CL_TARGET_OPENCL_VERSION 120
 
 /* We will use the deprecated clGetExtensionFunctionAddress,
  * so let the headers know that we don't care about it being deprecated.
@@ -28,13 +28,8 @@
  * ocl-icd as OpenCL dispatcher on macOS?
  */
 
-#ifdef __APPLE__
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-#include <OpenCL/opencl.h>
-#else
 #define CL_USE_DEPRECATED_OPENCL_1_1_APIS
 #include <CL/cl.h>
-#endif
 
 /* Very old headers will be missing these defines */
 #ifndef CL_VERSION_1_1
