--- libs/locale/src/shared/mo_lambda.cpp
+++ libs/locale/src/shared/mo_lambda.cpp
@@ -253,9 +253,9 @@
         bool is_blank(char c)
         {
             return c==' ' || c=='\r' || c=='\n' || c=='\t';
         }
-        bool isdigit(char c) 
+        bool is_digit(char c) 
         {
             return '0'<=c && c<='9'; 
         }
         void step() 
@@ -271,9 +271,9 @@
             else if(strncmp(ptr,">=",2)==0) { pos+=2; next_tocken=GTE; }
             else if(strncmp(ptr,"==",2)==0) { pos+=2; next_tocken=EQ; }
             else if(strncmp(ptr,"!=",2)==0) { pos+=2; next_tocken=NEQ; }
             else if(*ptr=='n') { pos++; next_tocken=VARIABLE; }
-            else if(isdigit(*ptr)) { int_value=strtol(text+pos,&tmp_ptr,0); pos=tmp_ptr-text; next_tocken=NUM; }
+            else if(is_digit(*ptr)) { int_value=strtol(text+pos,&tmp_ptr,0); pos=tmp_ptr-text; next_tocken=NUM; }
             else if(*ptr=='\0') { next_tocken=0; }
             else { next_tocken=*ptr; pos++; }
         }
     };
