--- src/modelHandler_OpenCL.cpp
+++ src/modelHandler_OpenCL.cpp	2026-03-04 18:12:44.000000000 +0800
@@ -92,7 +92,11 @@
 #ifdef _WIN32
 		handle = LoadLibraryA("OpenCL.dll");
 #elif defined __APPLE__
-		handle = dlopen("/System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL", RTLD_LAZY);
+        handle = dlopen("/opt/local/lib/libOpenCL.dylib", RTLD_LAZY);
+        if (handle == nullptr)
+		{
+		    handle = dlopen("/System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL", RTLD_LAZY);
+		}
 #define GetProcAddress dlsym
 #define FreeLibrary dlclose
 
