From ce00df1a8f28d9ebffe11590f67a324d79e1f67b Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:41:27 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_expect.ml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ppx_expect.ml b/src/ppx_expect.ml
index 42789f7..1349af2 100644
--- src/main.ml
+++ src/main.ml
@@ -145,6 +145,7 @@
          (Attribute.pattern
             uncaught_exn
             (value_binding
+               ~constraint_:drop
                ~pat:
                  (map
                     (Attribute.pattern Ppx_inline_test.tags (opt_name ()))
