From 3fb20acd04fbdabff7afe924028be7f9a12532a1 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 1 Jun 2026 17:38:15 +0800
Subject: [PATCH] libcdatetime_extern.h: fix incorrect macro usage

---
 libcdatetime/libcdatetime_extern.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libcdatetime/libcdatetime_extern.h b/libcdatetime/libcdatetime_extern.h
index 857d742..74a8b1f 100644
--- libcdatetime/libcdatetime_extern.h
+++ libcdatetime/libcdatetime_extern.h
@@ -24,11 +24,15 @@
 
 #include <common.h>
 
-#if defined( __has_attribute ) && __has_attribute( visibility ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if defined( __has_attribute ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if __has_attribute( visibility )
 #define LIBCDATETIME_INTERNAL	__attribute__((visibility("hidden"))) extern
 #else
 #define LIBCDATETIME_INTERNAL	extern
 #endif
+#else
+#define LIBCDATETIME_INTERNAL	extern
+#endif
 
 /* Define HAVE_LOCAL_LIBCDATETIME for local use of libcdatetime
  */
