--- src/main.cpp	2025-11-16 00:10:03.000000000 +0800
+++ src/main.cpp.new	2026-01-08 08:21:11.000000000 +0800
@@ -168,8 +168,9 @@
     {
         auto className = config.className.c_str();
 
+#if GLFW_VERSION_MAJOR >= 3 && GLFW_VERSION_MINOR >= 4
         glfwWindowHintString(GLFW_X11_CLASS_NAME, className);
-
+#endif
         if (helpers::getPlatform() == helpers::Platform::Wayland)
         {
 #if GLFW_VERSION_MAJOR >= 3 && GLFW_VERSION_MINOR >= 4
