--- CMakeLists.txt	2025-11-16 00:10:03.000000000 +0800
+++ CMakeLists.txt	2026-01-08 08:47:21.000000000 +0800
@@ -63,7 +63,7 @@
 if(CMAKE_BUILD_TYPE STREQUAL "Release")
     pretty_print("*" "Release Build" "*" " " "-")
     add_definitions("-DNDEBUG" )
-    add_definitions("-Wall -Wextra -pedantic -pedantic-errors -Wno-unused-function -Wno-deprecated-register -O2")
+    add_definitions("-Wall -Wextra -Wno-unused-function -Wno-deprecated-register")
 else()
     pretty_print("*" "Debug Build" "*" " " "-")
     add_definitions("-DDEBUG" )
