From 7248194a2ce0106ae647b70d0526a96dc9d6ad60 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Wed, 26 Mar 2025 18:17:49 +0800
Subject: [PATCH] animations: minor follow-up fix for std::string (#49)

---
 src/animation/AnimatedVariable.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/animation/AnimatedVariable.cpp b/src/animation/AnimatedVariable.cpp
index 261c00f..02ff25c 100644
--- src/animation/AnimatedVariable.cpp
+++ src/animation/AnimatedVariable.cpp
@@ -6,6 +6,7 @@ using namespace Hyprutils::Animation;
 using namespace Hyprutils::Memory;
 
 static const std::string DEFAULTBEZIERNAME = "default";
+static const std::string DEFAULTSTYLE = "";
 
 #define SP CSharedPointer
 #define WP CWeakPointer
@@ -54,8 +55,6 @@ const std::string& CBaseAnimatedVariable::getBezierName() const {
 }
 
 const std::string& CBaseAnimatedVariable::getStyle() const {
-    static constexpr const std::string DEFAULTSTYLE = "";
-
     if (const auto PCONFIG = m_pConfig.lock()) {
         const auto PVALUES = PCONFIG->pValues.lock();
         return PVALUES ? PVALUES->internalStyle : DEFAULTSTYLE;
