--- a/src/smplayer.pro
+++ b/src/smplayer.pro	2025-06-15 09:40:40.000000000 +0800
@@ -649,11 +649,8 @@
 		RESOURCES += glsl.qrc
 		QT += opengl
 	} else {
-		LIBS += -lswscale
-		mac {
-		    INCLUDEPATH += /usr/local/Cellar/ffmpeg/4.4_2/include/
-			LIBS += -L/usr/local/Cellar/ffmpeg/4.4_2/lib/
-		}
+		INCLUDEPATH += @FFMPG_ROOT@/include/
+		LIBS += -L@FFMPG_ROOT@/lib -lswscale
 	}
 
 	unix:!macx {
