--- src/myapplication.h
+++ src/myapplication.h	2025-06-15 08:45:38.000000000 +0800
@@ -20,6 +20,7 @@
 #define MYAPPLICATION_H
 
 #include <QtGlobal>
+#include <QStringList>
 
 #ifdef SINGLE_INSTANCE
 #include "QtSingleApplication"

--- src/mconnection.mm
+++ src/mconnection.mm	2025-06-15 09:26:59.000000000 +0800
@@ -53,7 +53,7 @@
 }
 
 void MConnection::startConnection() {
-	NSString * name = buffer_name.toNSString();
+	NSString * name = [NSString stringWithUTF8String:buffer_name.toUtf8().constData()];
 	mpc = [[MPlayerConnection alloc] initWithName:name];
 }
 
--- src/smplayer.pro
+++ src/smplayer.pro	2025-06-15 09:40:40.000000000 +0800
@@ -153,7 +153,6 @@
 	#DEFINES -= MPRIS2
 	DEFINES += USE_SHM
 	DEFINES += USE_COREVIDEO_BUFFER
-	DEFINES += USE_GL_WINDOW
 	message("Some features are disabled on macx.")
 }
 
@@ -620,6 +619,7 @@
 	mac {
 		HEADERS += powersaving_mac.h
 		SOURCES += powersaving_mac.cpp
+		LIBS += -framework IOKit
 	}
 }
 
@@ -640,7 +640,7 @@
 	}
 
 	isEqual(QT_MAJOR_VERSION, 5) {
-		#DEFINES += USE_GL_WINDOW
+		DEFINES += USE_GL_WINDOW
 	}
 
 	contains( DEFINES, USE_GL_WINDOW ) {
@@ -649,11 +649,8 @@
 		RESOURCES += glsl.qrc
 		QT += opengl
 	} else {
-		LIBS += -lswscale
-		mac {
-		    INCLUDEPATH += /usr/local/Cellar/ffmpeg/4.4_2/include/
-			LIBS += -L/usr/local/Cellar/ffmpeg/4.4_2/lib/
-		}
+        INCLUDEPATH += @FFMPG_ROOT@/include/
+        LIBS += -L@FFMPG_ROOT@/lib -lswscale
 	}
 
 	unix:!macx {
