From b59d0c84ec028944fe9a0dc79d500dc2ce49422a Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Fri, 23 May 2025 02:52:20 +0800
Subject: [PATCH 2/2] platform.c: fix powerpc macros

---
 src/tup/platform.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git src/tup/platform.c src/tup/platform.c
index 3ccc47cd..6e24f6d8 100644
--- src/tup/platform.c
+++ src/tup/platform.c
@@ -44,9 +44,9 @@ const char *tup_platform = "netbsd";
 const char *tup_arch = "x86_64";
 #elif __i386__
 const char *tup_arch = "i386";
-#elif __powerpc__
+#elif __powerpc__ || __ppc__
 const char *tup_arch = "powerpc";
-#elif __powerpc64__
+#elif __powerpc64__ || __ppc64__
 const char *tup_arch = "powerpc64";
 #elif __ia64__
 const char *tup_arch = "ia64";
-- 
2.48.0

