--- Makefile	2026-04-09 05:19:26.000000000 +0800
+++ Makefile	2026-05-23 11:50:50.000000000 +0800
@@ -4,6 +4,14 @@
 -include config.mak
 endif
 
+PREFIX ?= /usr/local
+prefix ?= $(PREFIX)
+exec_prefix = $(prefix)
+bindir = $(exec_prefix)/bin
+libdir = $(exec_prefix)/lib
+includedir = $(prefix)/include
+pkgconfigdir = $(libdir)/pkgconfig
+
 INSTALL ?= install
 TAR ?= tar
 CONFIG_CFLAGS ?= -O3
@@ -20,15 +28,15 @@
 
 XATTR_NAMESPACE ?= user.
 ifeq ($(OS), Darwin)
-	APP_FLAGS += -I/usr/local/include
-	APP_LIB += -L/usr/local/lib
+	APP_FLAGS += -I$(PREFIX)/include
+	APP_LIB += -L$(PREFIX)/lib
 	XATTR_NAMESPACE = #no namespaces on macOS
-	ifeq ($(shell [ -e /usr/local/lib/libosxfuse.dylib ] && echo 1), 1)
-		FUSE_FLAGS += -I/usr/local/include/osxfuse
+	ifeq ($(shell [ -e $(PREFIX)/lib/libosxfuse.dylib ] && echo 1), 1)
+		FUSE_FLAGS += -I$(PREFIX)/include/osxfuse
 		FUSE_LIB = -losxfuse
-	else ifeq ($(shell [ -e /usr/local/lib/libfuse.dylib -o -e /usr/local/lib/libfuse3.dylib ] && echo 1), 1)
-		FUSE_FLAGS += -I/usr/local/include -Wno-language-extension-token
-	else ifeq ($(shell [ -e /usr/local/lib/libfuse-t.dylib ] && echo 1), 1)
+	else ifeq ($(shell [ -e $(PREFIX)/lib/libfuse.dylib -o -e $(PREFIX)/lib/libfuse3.dylib ] && echo 1), 1)
+		FUSE_FLAGS += -I$(PREFIX)/include -Wno-language-extension-token
+	else ifeq ($(shell [ -e $(PREFIX)/lib/libfuse-t.dylib ] && echo 1), 1)
 		FUSE_FLAGS += -I/Library/Application\ Support/fuse-t/include/fuse
 		FUSE_LIB = -lfuse-t
 	else
@@ -80,14 +88,6 @@
 	endif
 endif
 
-PREFIX ?= /usr/local
-prefix ?= $(PREFIX)
-exec_prefix = $(prefix)
-bindir = $(exec_prefix)/bin
-libdir = $(exec_prefix)/lib
-includedir = $(prefix)/include
-pkgconfigdir = $(libdir)/pkgconfig
-
 WITH_UBLIO ?= local
 WITH_UTF8PROC ?= local
 WITH_LZVN ?= local
@@ -205,7 +205,9 @@
 $(error Invalid option "$(WITH_LZVN)" for WITH_LZVN. Use one of: none, system, local)
 	endif
 endif
-ifneq ($(WITH_UTHASH), system)
+ifeq ($(WITH_UTHASH), system)
+	UTHASH_FLAGS = -I$(PREFIX)/include/uthash
+else
 	UTHASH_FLAGS = -Ilib/uthash
 endif
 
@@ -222,8 +224,7 @@
 	VERSION_STRING = \"$(VERSION)-$(GIT_HEAD_SHA)\"
 	CFLAGS += -DHFSFUSE_VERSION_STRING=$(VERSION_STRING)
 else ifeq ($(wildcard src/version.h), )
-    $(warning Warning: git repo nor prepackaged version.h found, hfsfuse will be built without version information)
-	CFLAGS += -DHFSFUSE_VERSION_STRING=\"omitted\"
+	CFLAGS += -DHFSFUSE_VERSION_STRING=\"@VER@\"
 endif
 
 define pkg_config_file
@@ -269,7 +270,7 @@
 hfsfuse: src/hfsfuse.o $(LIBS)
 	$(CC) -o $@ $(LDFLAGS) $^ $(LDLIBS)
 ifeq ($(OS), Darwin)
-	if ! otool -l hfsfuse | grep -q LC_RPATH; then install_name_tool -add_rpath /usr/local/lib hfsfuse; fi
+	if ! otool -l hfsfuse | grep -q LC_RPATH; then install_name_tool -add_rpath $(PREFIX)/lib hfsfuse; fi
 endif
 
 hfsdump: src/hfsdump.o $(LIBS)
