--- CMakeLists.txt	2025-02-07 08:55:02.000000000 +0800
+++ CMakeLists.txt	2025-02-07 23:43:15.000000000 +0800
@@ -36,7 +36,6 @@
 add_compile_options (-Wall)
 add_compile_options (-Wextra)
 add_compile_options (-Wconversion)
-add_compile_options (-Wno-unused-but-set-variable)
 add_compile_options (-Wno-unused-parameter)
 add_compile_options (-Wno-unknown-pragmas)
 add_compile_options (-Wno-float-equal)
@@ -54,7 +53,7 @@
 
 #### compiler-specific compile options
 
-if (CMAKE_C_COMPILER_ID STREQUAL "GNU")
+if (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND NOT APPLE)
   add_compile_options (-Wmaybe-uninitialized)
   add_compile_options (-Wno-unused-but-set-variable)
   add_compile_options (-Wno-stringop-overflow)
