--- src/libunicode/convert_simd_impl.h	2026-07-19 08:22:09.000000000 +0800
+++ src/libunicode/convert_simd_impl.h	2026-07-19 13:38:23.000000000 +0800
@@ -8,7 +8,7 @@
 #include <cstdint>
 
 // clang-format off
-#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(_LIBCPP_VERSION)
     #define USE_STD_SIMD_CONVERT
     #include <experimental/simd>
     namespace convert_stdx = std::experimental;

--- src/libunicode/scan_simd_impl.h	2026-07-19 08:22:09.000000000 +0800
+++ src/libunicode/scan_simd_impl.h	2026-07-19 13:44:41.000000000 +0800
@@ -6,7 +6,7 @@
 #include <string_view>
 
 // clang-format off
-#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(__APPLE__) && !defined(__FreeBSD__)
+#if __has_include(<experimental/simd>) && defined(LIBUNICODE_USE_STD_SIMD) && !defined(_LIBCPP_VERSION)
     #define USE_STD_SIMD
     #include <experimental/simd>
     namespace stdx = std::experimental;
