--- scribus/plugins/gettext/sxwim/contentreader.h	2019-03-06 06:44:40.000000000 +0800
+++ scribus/plugins/gettext/sxwim/contentreader.h	2026-04-23 18:11:47.000000000 +0800
@@ -33,11 +33,8 @@
 
 #include <utility>
 #include <vector>
-#ifdef HAVE_XML26
- #include <libxml/SAX2.h>
-#else
- #include <libxml/SAX.h>
-#endif
+#include <libxml/parser.h>
+#include <libxml/tree.h>
 #include <QXmlAttributes>
 #include <QMap>
 #include <gtstyle.h>

--- scribus/plugins/gettext/sxwim/stylereader.h	2019-03-06 06:44:40.000000000 +0800
+++ scribus/plugins/gettext/sxwim/stylereader.h	2026-04-23 18:11:57.000000000 +0800
@@ -29,11 +29,8 @@
 
 #include "scconfig.h"
 
-#ifdef HAVE_XML26
- #include <libxml/SAX2.h>
-#else
- #include <libxml/SAX.h>
-#endif
+#include <libxml/parser.h>
+#include <libxml/tree.h>
 #include <QMap>
 #include <QXmlAttributes>
 #include <gtstyle.h>

--- scribus/plugins/gettext/odtim/contentreader.h	2019-03-06 06:44:40.000000000 +0800
+++ scribus/plugins/gettext/odtim/contentreader.h	2026-04-23 18:10:59.000000000 +0800
@@ -31,11 +31,8 @@
 
 #include <utility>
 #include <vector>
-#ifdef HAVE_XML26
- #include <libxml/SAX2.h>
-#else
- #include <libxml/SAX.h>
-#endif
+#include <libxml/parser.h>
+#include <libxml/tree.h>
 #include <QXmlAttributes>
 #include <QMap>
 #include <gtstyle.h>

--- scribus/plugins/gettext/odtim/stylereader.h	2019-03-06 06:44:40.000000000 +0800
+++ scribus/plugins/gettext/odtim/stylereader.h	2026-04-23 18:10:48.000000000 +0800
@@ -29,11 +29,8 @@
 
 #include "scconfig.h"
 
-#ifdef HAVE_XML26
- #include <libxml/SAX2.h>
-#else
- #include <libxml/SAX.h>
-#endif
+#include <libxml/parser.h>
+#include <libxml/tree.h>
 #include <QMap>
 #include <QXmlAttributes>
 #include <gtstyle.h>
