--- a/include/ext_OpenCL.h	2025-08-23 16:43:56.000000000 +0800
+++ b/include/ext_OpenCL.h	2026-04-25 02:54:15.000000000 +0800
@@ -12,6 +12,10 @@
 #define CL_USE_DEPRECATED_OPENCL_2_0_APIS
 
 #if defined (__APPLE__)
+#include <AvailabilityMacros.h>
+#endif
+
+#if defined (__APPLE__) && MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
 #include <OpenCL/cl.h>
 #else
 #include <CL/cl.h>

--- a/include/types.h	2025-08-23 16:43:56.000000000 +0800
+++ b/include/types.h	2026-04-25 18:12:12.000000000 +0800
@@ -1766,7 +1766,7 @@
 
   bool              is_metal;
 
-  #if defined (__APPLE__)
+  #if defined (__APPLE__) && defined (WITH_METAL)
 
   //int               mtl_major;
   //int               mtl_minor;
