--- a/src/Makefile	2026-04-25 19:19:11.000000000 +0800
+++ b/src/Makefile	2026-04-25 20:25:14.000000000 +0800
@@ -398,21 +398,21 @@
 ifeq ($(UNAME),Darwin)
 export MACOSX_DEPLOYMENT_TARGET=10.6
 CFLAGS_NATIVE           := $(CFLAGS)
-# Disabled for now, since linking fails. (However, it not even respected by the build...)
+LFLAGS_NATIVE           := $(LFLAGS)
+# Disabled for now, since linking fails.
 # CFLAGS_NATIVE           += -DWITH_HWMON
 
 ifeq ($(shell test $(DARWIN_VERSION) -le 15; echo $$?), 0)
-CFLAGS_NATIVE           += -DMISSING_CLOCK_GETTIME
+CFLAGS_NATIVE           += -isystem@PREFIX@/include/LegacySupport
+LFLAGS_NATIVE           += -lMacportsLegacySupport
 endif
 
-LFLAGS_NATIVE           := $(LFLAGS)
 LFLAGS_NATIVE           += -framework CoreFoundation
 LFLAGS_NATIVE           += -framework ApplicationServices
 LFLAGS_NATIVE           += -framework Foundation
 LFLAGS_NATIVE           += -framework IOKit
 LFLAGS_NATIVE           += -lpthread
 LFLAGS_NATIVE           += -liconv
-
 endif # Darwin
 
 ifeq ($(IS_ARM),1)
