--- a/src/Makefile	2025-08-23 16:43:56.000000000 +0800
+++ b/src/Makefile	2026-04-25 03:48:36.000000000 +0800
@@ -396,9 +396,10 @@
 endif # NetBSD
 
 ifeq ($(UNAME),Darwin)
-export MACOSX_DEPLOYMENT_TARGET=15.0
+export MACOSX_DEPLOYMENT_TARGET=10.6
 CFLAGS_NATIVE           := $(CFLAGS)
-CFLAGS_NATIVE           += -DWITH_HWMON
+# Disabled for now, since linking fails. (However, it not even respected by the build...)
+# CFLAGS_NATIVE           += -DWITH_HWMON
 
 ifeq ($(shell test $(DARWIN_VERSION) -le 15; echo $$?), 0)
 CFLAGS_NATIVE           += -DMISSING_CLOCK_GETTIME
@@ -406,23 +407,12 @@
 
 LFLAGS_NATIVE           := $(LFLAGS)
 LFLAGS_NATIVE           += -framework CoreFoundation
-LFLAGS_NATIVE           += -framework CoreGraphics
+LFLAGS_NATIVE           += -framework ApplicationServices
 LFLAGS_NATIVE           += -framework Foundation
 LFLAGS_NATIVE           += -framework IOKit
-LFLAGS_NATIVE           += -framework Metal
-LFLAGS_NATIVE           += -lIOReport
 LFLAGS_NATIVE           += -lpthread
 LFLAGS_NATIVE           += -liconv
 
-ifeq ($(IS_APPLE_SILICON),1)
-CFLAGS_NATIVE           += -arch arm64
-CFLAGS_NATIVE           += -arch x86_64
-ifeq ($(SHARED),1)
-LFLAGS_NATIVE           += -arch arm64
-LFLAGS_NATIVE           += -arch x86_64
-endif
-endif
-
 endif # Darwin
 
 ifeq ($(IS_ARM),1)
@@ -478,12 +468,6 @@
 LINUX_OBJS              := $(foreach OBJ,$(OBJS_ALL),obj/$(OBJ).LINUX.o)
 WIN_OBJS                := $(foreach OBJ,$(OBJS_ALL),obj/$(OBJ).WIN.o)
 
-ifeq ($(UNAME),Darwin)
-OBJS_METAL              := ext_metal ext_iokit
-
-NATIVE_OBJS             += $(foreach OBJ,$(OBJS_METAL),obj/$(OBJ).METAL.NATIVE.o)
-endif
-
 ifeq ($(USE_SYSTEM_LZMA),0)
 OBJS_LZMA               := 7zCrc 7zCrcOpt 7zFile 7zStream Alloc Bra Bra86 BraIA64 CpuArch Delta LzmaDec Lzma2Dec MtDec Sha256 Sha256Opt Threads Xz XzCrc64 XzCrc64Opt XzDec XzIn
 
@@ -705,11 +689,6 @@
 obj/%.NATIVE.o: src/%.c
 	$(CC) -c $(CCFLAGS) $(CFLAGS_NATIVE) $< -o $@ -fpic
 
-ifeq ($(UNAME),Darwin)
-obj/%.METAL.NATIVE.o: src/%.m
-	$(CC) -c $(CCFLAGS) $(CFLAGS_NATIVE) $< -o $@ -fpic
-endif
-
 ifeq ($(USE_SYSTEM_LZMA),0)
 obj/%.LZMA.NATIVE.o: $(DEPS_LZMA_PATH)/%.c
 	$(CC) -c $(CCFLAGS) $(CFLAGS_NATIVE) $(CFLAGS_LZMA) $< -o $@ -fpic
