--- mhs.conf.in	2026-06-20 06:33:54.000000000 +0800
+++ mhs.conf.in	2026-06-22 06:35:47.000000000 +0800
@@ -18,17 +18,17 @@
 -- Typical Unix-like flags
 -- Assumes gcc or clang
 [unix]
-cc = "cc"
-ccflags = "-w -Wall -O3 %GMPFLAGS"
-cclibs = "%GMPLIBS -lm"
+cc = "@CC@"
+ccflags = "-w -Wall -O3 %GMPFLAGS @EXTRA_CFLAGS@"
+cclibs = "%GMPLIBS -lm @EXTRA_LIBS@"
 conf = "unix"
 
 -- As above, but with -flto
 -- This flag improves performance, but generates a warning with gcc.
 [unix_lto]
-cc = "cc"
-ccflags = "-w -Wall -O3 -flto %GMPFLAGS"
-cclibs = "%GMPLIBS -lm"
+cc = "@CC@"
+ccflags = "-w -Wall -O3 -flto %GMPFLAGS @EXTRA_CFLAGS@"
+cclibs = "%GMPLIBS -lm @EXTRA_LIBS@"
 conf = "unix"
 
 -- As above, but with the non-portable flag -march=native
@@ -40,9 +40,9 @@
 
 -- Compile with debugging instead of optimization
 [debug]
-cc = "cc"
-ccflags = "-w -Wall -g"
-cclibs = "-lm"
+cc = "@CC@"
+ccflags = "-w -Wall -g @EXTRA_CFLAGS@"
+cclibs = "-lm @EXTRA_LIBS@"
 conf = "unix"
 
 -- Generate JavaScript&Wasm
