--- mhs.conf.in	2026-05-30 21:07:04.000000000 +0800
+++ mhs.conf.in	2026-06-01 22:41:17.000000000 +0800
@@ -17,16 +17,16 @@
 
 -- Typical Unix-like flags
 [unix]
-cc = "cc"
-ccflags = "-w -Wall -O3 %GMPFLAGS"
-cclibs = "%GMPLIBS -lm"
+cc = "@CC@"
+ccflags = "-w -Wall -O3 %GMPFLAGS @EXTRA_CFLAGS@"
+cclibs = "%GMPLIBS -lm @EXTRA_LIBS@"
 conf = "unix"
 
 -- Compile with debugging instead of optimization
 [debug]
-cc = "cc"
-ccflags = "-w -Wall -g"
-cclibs = "-lm"
+cc = "@CC@"
+ccflags = "-w -Wall -g @EXTRA_CFLAGS@"
+cclibs = "-lm @EXTRA_LIBS@"
 conf = "unix"
 
 -- Generate JavaScript&Wasm
