--- meson.build	2025-10-12 16:35:02.000000000 +0800
+++ meson.build	2026-01-31 13:23:35.000000000 +0800
@@ -6,7 +6,6 @@
 gnome = import('gnome')
 i18n = import('i18n')
 pkg = import('pkgconfig')
-python = import('python')
 
 # plugin API versioning
 # Increment on interface addition, reset on removal
@@ -145,7 +144,7 @@
 
 enable_python = false
 enable_gir2 = false
-python_install = python.find_installation('python3', required: get_option('plugins_python'))
+python_install = import('python').find_installation('@PYTHON@', required: get_option('plugins_python'))
 pygobject = dependency('pygobject-3.0', version: '>= 3.0.0', required: get_option('plugins_python'))
 if python_install.found() and pygobject.found()
   enable_python = true
