--- a/include/my_rdtsc.h	2025-05-20 00:36:43.000000000 +0800
+++ b/include/my_rdtsc.h	2025-07-12 12:55:17.000000000 +0800
@@ -171,7 +171,7 @@
     __asm __volatile__ ("mov %0=ar.itc" : "=r" (result));
     return result;
   }
-#elif defined __GNUC__ && defined __powerpc__
+#elif defined __GNUC__ && (defined(__powerpc__) || defined(__POWERPC__))
   #define MY_TIMER_ROUTINE_CYCLES MY_TIMER_ROUTINE_PPC_GET_TIMEBASE
   return __builtin_ppc_get_timebase();
 #elif defined(__GNUC__) && defined(__sparcv9) && defined(_LP64)
