--- httplib.h	2026-02-23 06:27:02.000000000 +0800
+++ httplib.h	2026-02-24 01:34:36.000000000 +0800
@@ -285,6 +285,7 @@
 
 #else // not _WIN32
 
+#include <sys/socket.h>
 #include <arpa/inet.h>
 #if !defined(_AIX) && !defined(__MVS__)
 #include <ifaddrs.h>
@@ -307,7 +308,6 @@
 #include <poll.h>
 #include <pthread.h>
 #include <sys/mman.h>
-#include <sys/socket.h>
 #include <sys/un.h>
 #include <unistd.h>
 
