--- include/fmt/format.h.orig	2026-09-16 23:02:42.000000000 -0600
+++ include/fmt/format.h	2026-09-16 23:04:28.000000000 -0600
@@ -322,6 +322,9 @@
       -> uint128 {
     return {lhs.hi_ | rhs.hi_, lhs.lo_ | rhs.lo_};
   }
+  friend constexpr auto operator~(const uint128& n) -> uint128 {
+    return {~n.hi_, ~n.lo_};
+  }
   friend constexpr auto operator&(const uint128& lhs, const uint128& rhs)
       -> uint128 {
     return {lhs.hi_ & rhs.hi_, lhs.lo_ & rhs.lo_};
--- test/format-test.cc.orig	2026-09-16 23:05:54.000000000 -0600
+++ test/format-test.cc	2026-09-16 23:06:29.000000000 -0600
@@ -76,6 +76,11 @@
   EXPECT_EQ(uint128(0x1000000000000, 0) >> 112, uint128(1));
 }
 
+TEST(uint128_test, bitwise_not) {
+  auto n = ~uint128(0x123456789abcdef0, 0x0fedcba987654321);
+  EXPECT_EQ(n, uint128(0xedcba9876543210f, 0xf0123456789abcde));
+}
+
 TEST(uint128_test, minus) {
   auto n = uint128(42);
   EXPECT_EQ(n - 2, 40);
