From aa0954496d1c6781105ba1b78a0639481c6f195f Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:05:16 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 ppx_globalize.ml | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/ppx_globalize.ml b/ppx_globalize.ml
index 9f01c47..4bcbeb4 100644
--- ppx_globalize.ml
+++ ppx_globalize.ml
@@ -164,7 +164,7 @@
          | Ptyp_constr (_, args) when List.length params = List.length args ->
            List.fold2_exn params args ~init:vars ~f:(fun vars param arg ->
              match arg.ptyp_desc with
-             | Ptyp_var name | Ptyp_alias (_, name) ->
+             | Ptyp_var name | Ptyp_alias (_, { txt = name; loc = _ }) ->
                (match Map.add vars ~key:name ~data:(Globalize (evar param)) with
                 | `Duplicate -> vars
                 | `Ok vars -> vars)
@@ -248,7 +248,7 @@
   | Ptyp_class (lid, args) ->
     let args = List.map ~f:(fun _ -> ptyp_any) args in
     ptyp_class (Located.mk lid.txt) args
-  | Ptyp_poly _ -> assert false
+  | Ptyp_poly _ | Ptyp_open _ -> assert false
 ;;
 
 let mode_crossing_attr_name = "globalized"
@@ -388,7 +388,7 @@
        error
          ~loc:typ.ptyp_loc
          "Cannot generate globalize function for partial variant type"
-     | Ptyp_alias (typ, name) ->
+     | Ptyp_alias (typ, { txt = name; loc = _ }) ->
        (match Env.lookup env name with
         | Some (Globalize fn) -> eapply fn [ exp ]
         | Some Universal | None ->
@@ -409,7 +409,8 @@
          ~loc:typ.ptyp_loc
          "Cannot generate globalize function for first-class module type"
      | Ptyp_extension _ ->
-       error ~loc:typ.ptyp_loc "Cannot generate globalize function for unknown extension")
+       error ~loc:typ.ptyp_loc "Cannot generate globalize function for unknown extension"
+     | Ptyp_open _ -> assert false)
 
 (* Generate code for a function to globalize values of type [type]. *)
 and generate_globalized_for_typ_as_function builder env name_opt typ =
