--- meson.build.orig	2025-06-02 22:52:30.000000000 +0800
+++ meson.build	2025-10-20 07:00:34.000000000 +0800
@@ -46,7 +46,7 @@
     version : run_command('./version.sh', check : true).stdout().strip(),
     license : ['GPL-2.0-or-later'],
     meson_version : '>=1.0.0',
-    default_options : ['cpp_std=c++14', 'warning_level=3', 'buildtype=debugoptimized', 'cpp_link_args=-rdynamic']
+    default_options : ['cpp_std=c++14', 'warning_level=3', 'buildtype=debugoptimized']
 )
 
 # To inhibit warnings from the generated file icons.h
