From 5f2220c9fdad5aaaf03c8acef1399d7a89e4c5ef Mon Sep 17 00:00:00 2001
From: Joachim Bauch <mail@joachim-bauch.de>
Date: Tue, 3 Feb 2026 21:47:26 +0100
Subject: [PATCH] Undefine "PPC" to prevent compiler errors on PowerPC.

---
 src/pylzma/pylzma.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/pylzma/pylzma.c b/src/pylzma/pylzma.c
index 978044b..8fc1f98 100644
--- src/pylzma/pylzma.c
+++ src/pylzma/pylzma.c
@@ -198,6 +198,10 @@ pylzma_bcj_##id##_convert(PyObject *self, PyObject *args) \
     return result; \
 }
 
+// Some PowerPC compilers have a builtin define "PPC" that generates invalid
+// code from the "DEFINE_BCJ_CONVERTER" macro.
+#undef PPC
+
 DEFINE_BCJ_CONVERTER(arm, ARM);
 DEFINE_BCJ_CONVERTER(armt, ARMT);
 DEFINE_BCJ_CONVERTER(arm64, ARM64);
