--- tests/rsa_test.c	2026-05-20 07:22:31.000000000 +0800
+++ tests/rsa_test.c	2026-06-14 04:43:26.000000000 +0800
@@ -669,7 +669,8 @@
 
     aws_cal_library_test_init(allocator);
 
-#if defined(AWS_OS_MACOS)
+#if defined(AWS_OS_MACOS) && !defined(OPENSSL_IS_OPENSSL)
+#ifdef __builtin_available
     if (__builtin_available(macOS 10.12, *)) {
         ASSERT_SUCCESS(s_rsa_signing_roundtrip_from_user(allocator, AWS_CAL_RSA_SIGNATURE_PSS_SHA256, NULL));
     } else {
@@ -678,6 +679,11 @@
             s_rsa_signing_roundtrip_from_user(allocator, AWS_CAL_RSA_SIGNATURE_PSS_SHA256, NULL));
     }
 #else
+    ASSERT_ERROR(
+        AWS_ERROR_CAL_UNSUPPORTED_ALGORITHM,
+        s_rsa_signing_roundtrip_from_user(allocator, AWS_CAL_RSA_SIGNATURE_PSS_SHA256, NULL));
+#endif
+#else
     ASSERT_SUCCESS(s_rsa_signing_roundtrip_from_user(allocator, AWS_CAL_RSA_SIGNATURE_PSS_SHA256, NULL));
 #endif
 
