--- tests/CMakeLists.txt	2026-06-04 01:57:49.000000000 +0800
+++ tests/CMakeLists.txt	2026-06-11 19:46:05.000000000 +0800
@@ -306,7 +306,7 @@
     add_net_test_case(test_ecc_cert_import)
     add_net_test_case(test_tls_cipher_preference)
 
-if(NOT AWS_USE_SECITEM)
+if(NOT APPLE)
     # These tests require the test binary to be codesigned with an Apple Developer account with entitlements.
     # The entitlements also require a provisioning profile and require the binary to be run from within XCode or a
     # valid app bundle.
