--- include/aws/io/private/pki_utils.h	2026-01-14 05:53:02.000000000 +0800
+++ include/aws/io/private/pki_utils.h	2026-01-15 07:11:52.000000000 +0800
@@ -23,6 +23,7 @@
 
 #ifdef AWS_OS_APPLE
 /* It's ok to include external headers because this is a PRIVATE header file */
+#    include <AvailabilityMacros.h>
 #    include <CoreFoundation/CFArray.h>
 #    include <Security/Security.h>
 #endif /* AWS_OS_APPLE */
@@ -39,7 +40,7 @@
 AWS_IO_API const char *aws_determine_default_pki_dir(void);
 AWS_IO_API const char *aws_determine_default_pki_ca_file(void);
 
-#ifdef AWS_OS_APPLE
+#if defined(AWS_OS_APPLE) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
 /**
  * Imports a PEM armored PKCS#7 public/private key pair
  * into identity for use with SecurityFramework.
