--- include/aws/io/private/pki_utils.h	2025-07-29 04:45:52.000000000 +0800
+++ include/aws/io/private/pki_utils.h	2025-08-18 16:02:02.000000000 +0800
@@ -14,6 +14,7 @@
 
 #ifdef AWS_OS_APPLE
 /* It's ok to include external headers because this is a PRIVATE header file */
+#    include <AvailabilityMacros.h>
 #    include <CoreFoundation/CFArray.h>
 #    include <Security/Security.h>
 #endif /* AWS_OS_APPLE */
@@ -30,7 +31,7 @@
 AWS_IO_API const char *aws_determine_default_pki_dir(void);
 AWS_IO_API const char *aws_determine_default_pki_ca_file(void);
 
-#ifdef AWS_OS_APPLE
+#if defined(AWS_OS_APPLE) && (MAC_OS_X_VERSION_MIN_REQUIRED >= 1070)
 /**
  * Imports a PEM armored PKCS#7 public/private key pair
  * into identity for use with SecurityFramework.
