--- source/tls_channel_handler.c	2026-06-04 01:57:49.000000000 +0800
+++ source/tls_channel_handler.c	2026-06-06 19:08:23.000000000 +0800
@@ -811,6 +811,9 @@
 #    if defined(__APPLE__) && defined(AWS_USE_SECITEM)
     secure_transport_init_tls_vtable(&s_tls_channel_handler_vtable);
 
+#    elif defined(__APPLE__) && defined(AWS_APPLE_USE_S2N_ONLY)
+    s2n_init_tls_vtable(&s_tls_channel_handler_vtable);
+
 #    elif defined(__APPLE__) && defined(USE_S2N)
     struct aws_string *use_non_fips_13 = aws_get_env_nonempty(allocator, "AWS_CRT_USE_NON_FIPS_TLS_13");
     if (use_non_fips_13) {
