--- docs/reference/gtk/meson.build	2025-09-30 18:19:25.000000000 +0800
+++ docs/reference/gtk/meson.build	2025-10-26 01:09:22.000000000 +0800
@@ -79,8 +79,8 @@
   )
 endif
 
-rst2man = find_program('rst2man', 'rst2man.py', required: get_option('man-pages'))
-rst2html5 = find_program('rst2html5', 'rst2html5.py', required: get_option('documentation'))
+rst2man = find_program('rst2man-@@PYTHON_VERSION@@', 'rst2man-@@PYTHON_VERSION@@', required: false)
+rst2html5 = find_program('rst2html5', 'rst2html5.py', required: false)
 
 rst_files = [
   [ 'gtk4-broadwayd', '1' ],
