--- client/SDL/SDL3/dialogs/sdl_dialogs.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/dialogs/sdl_dialogs.cpp	2025-05-08 20:50:46.000000000 +0800
@@ -17,9 +17,13 @@
  * limitations under the License.
  */
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <vector>
 #include <string>
 #include <cassert>
+#include <cinttypes>
 
 #include <freerdp/log.h>
 #include <freerdp/utils/smartcardlogon.h>

--- client/SDL/SDL3/sdl_clip.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/sdl_clip.cpp	2025-05-08 21:06:55.000000000 +0800
@@ -18,12 +18,16 @@
  * limitations under the License.
  */
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <string>
 #include <sstream>
 #include <mutex>
 #include <iterator>
 #include <algorithm>
 #include <utility>
+#include <cinttypes>
 
 #include <winpr/wlog.h>
 #include <winpr/image.h>

--- client/SDL/SDL3/sdl_kbd.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/sdl_kbd.cpp	2025-05-08 21:14:10.000000000 +0800
@@ -17,6 +17,11 @@
  * limitations under the License.
  */
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
+
 #include "sdl_kbd.hpp"
 #include "sdl_disp.hpp"
 #include "sdl_freerdp.hpp"

--- client/SDL/SDL3/sdl_freerdp.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/sdl_freerdp.cpp	2025-05-08 21:26:01.000000000 +0800
@@ -17,6 +17,10 @@
  * limitations under the License.
  */
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
 #include <iostream>
 #include <memory>
 #include <mutex>
@@ -26,6 +30,7 @@
 #include <cerrno>
 #include <cstdio>
 #include <cstring>
+#include <cinttypes>
 
 #include <freerdp/constants.h>
 #include <freerdp/freerdp.h>

--- client/SDL/SDL3/sdl_monitor.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/sdl_monitor.cpp	2025-05-08 21:25:23.000000000 +0800
@@ -21,10 +21,14 @@
 
 #include <freerdp/config.h>
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <algorithm>
+#include <cinttypes>
 
 #include <SDL3/SDL.h>
 
--- client/SDL/SDL3/sdl_disp.cpp	2025-04-14 21:11:41.000000000 +0800
+++ client/SDL/SDL3/sdl_disp.cpp	2025-05-08 21:24:52.000000000 +0800
@@ -17,6 +17,10 @@
  * limitations under the License.
  */
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+#include <cinttypes>
 #include <vector>
 #include <winpr/sysinfo.h>
 #include <winpr/assert.h>
