--- winpr/libwinpr/sysinfo/sysinfo.c.orig	2026-08-26 21:07:50.000000000 +0800
+++ winpr/libwinpr/sysinfo/sysinfo.c	2026-09-19 00:50:04.000000000 +0800
@@ -152,10 +152,22 @@
 			return PROCESSOR_ARCHITECTURE_AMD64;
 		case CPU_TYPE_ARM:
 			return PROCESSOR_ARCHITECTURE_ARM;
+#ifdef CPU_TYPE_ARM64
 		case CPU_TYPE_ARM64:
 			return PROCESSOR_ARCHITECTURE_ARM64;
+#endif
+#ifdef CPU_TYPE_ARM64_32
 		case CPU_TYPE_ARM64_32:
 			return PROCESSOR_ARCHITECTURE_ARM;
+#endif
+#ifdef CPU_TYPE_POWERPC
+		case CPU_TYPE_POWERPC:
+			return PROCESSOR_ARCHITECTURE_PPC;
+#endif
+#ifdef CPU_TYPE_POWERPC64
+		case CPU_TYPE_POWERPC64:
+			return PROCESSOR_ARCHITECTURE_PPC;
+#endif
 		default:
 			return PROCESSOR_ARCHITECTURE_UNKNOWN;
 	}
