--- channels/audin/client/mac/CMakeLists.txt	2025-04-14 21:11:41.000000000 +0800
+++ channels/audin/client/mac/CMakeLists.txt	2025-05-08 19:47:28.000000000 +0800
@@ -18,7 +18,6 @@
 
 define_channel_client_subsystem("audin" "mac" "")
 find_library(CORE_AUDIO CoreAudio)
-find_library(AVFOUNDATION AVFoundation)
 find_library(AUDIO_TOOL AudioToolbox)
 find_library(APP_SERVICES ApplicationServices)
 

--- channels/audin/client/mac/audin_mac.m	2025-04-14 21:11:41.000000000 +0800
+++ channels/audin/client/mac/audin_mac.m	2025-05-08 19:47:17.000000000 +0800
@@ -31,8 +31,6 @@
 #include <winpr/debug.h>
 #include <winpr/cmdline.h>
 
-#import <AVFoundation/AVFoundation.h>
-
 #define __COREFOUNDATION_CFPLUGINCOM__ 1
 #define IUNKNOWN_C_GUTS   \
 	void *_reserved;      \
