From 1101cef51b64607191a50d8cb0f3d89195d30bf6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Mon, 1 Jun 2026 17:49:30 +0800
Subject: [PATCH] libclocale_extern.h: fix incorrect macro usage

---
 libclocale/libclocale_extern.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libclocale/libclocale_extern.h b/libclocale/libclocale_extern.h
index b7c9795..8e228c2 100644
--- libclocale/libclocale_extern.h
+++ libclocale/libclocale_extern.h
@@ -24,11 +24,15 @@
 
 #include <common.h>
 
-#if defined( __has_attribute ) && __has_attribute( visibility ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if defined( __has_attribute ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if __has_attribute( visibility )
 #define LIBCLOCALE_INTERNAL	__attribute__((visibility("hidden"))) extern
 #else
 #define LIBCLOCALE_INTERNAL	extern
 #endif
+#else
+#define LIBCLOCALE_INTERNAL	extern
+#endif
 
 /* Define HAVE_LOCAL_LIBCLOCALE for local use of libclocale
  */
