--- a/main/streams/cast.c	2026-03-11 07:15:23.000000000 +0800
+++ b/main/streams/cast.c	2026-03-15 05:36:34.000000000 +0800
@@ -79,7 +79,7 @@
 	return php_stream_write((php_stream *)cookie, (char *)buffer, size);
 }
 
-static PHP_FPOS_T stream_cookie_seeker(void *cookie, zend_off_t position, int whence)
+static PHP_FPOS_T stream_cookie_seeker(void *cookie, PHP_FPOS_T position, int whence)
 {
 
 	return (PHP_FPOS_T)php_stream_seek((php_stream *)cookie, position, whence);
