diff --git a/modules/gui/minimal_macosx/VLCMinimalVoutWindow.m b/modules/gui/minimal_macosx/VLCMinimalVoutWindow.m
index bd58f7d6e1..0a9e49a2fb 100644
--- modules/gui/minimal_macosx/VLCMinimalVoutWindow.m
+++ modules/gui/minimal_macosx/VLCMinimalVoutWindow.m
@@ -34,10 +34,10 @@
 @implementation VLCMinimalVoutWindow
 - (id)initWithContentRect:(NSRect)contentRect
 {
-    if( self = [super initWithContentRect:contentRect
-                                styleMask:NSBorderlessWindowMask
-                                  backing:NSBackingStoreBuffered
-                                    defer:NO])
+    if ((self = [super initWithContentRect:contentRect
+                                 styleMask:NSBorderlessWindowMask
+                                   backing:NSBackingStoreBuffered
+                                     defer:NO]))
     {
         initialFrame = contentRect;
         [self setBackgroundColor:[NSColor blackColor]];
