--- modules/video_output/aa.c	2025-12-23 14:55:38.000000000 +0800
+++ modules/video_output/aa.c	2026-03-12 22:42:17.000000000 +0800
@@ -36,7 +36,7 @@
 #include <assert.h>
 #include <aalib.h>
 
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__APPLE__)
 # ifdef X_DISPLAY_MISSING
 #  error Xlib required due to XInitThreads
 # endif
@@ -92,7 +92,7 @@
     vout_display_t *vd = (vout_display_t *)object;
     vout_display_sys_t *sys;
 
-#ifndef _WIN32
+#if !defined(_WIN32) && !defined(__APPLE__)
     if (!vlc_xlib_init (object))
         return VLC_EGENERIC;
 #endif
