From 8e759c3b9e82ceb5c1c764559571feaca0ced866 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:45:42 +0800
Subject: [PATCH 02/13] version.c: add a missing macro for powerpc

---
 src/detection/version/version.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/detection/version/version.c b/src/detection/version/version.c
index a979bde5..9919771f 100644
--- a/src/detection/version/version.c
+++ b/src/detection/version/version.c
@@ -10,7 +10,7 @@
     #define FF_ARCHITECTURE "arm"
 #elif defined(__mips__)
     #define FF_ARCHITECTURE "mips"
-#elif defined(__powerpc__) || defined(__powerpc)
+#elif defined(__powerpc__) || defined(__powerpc) || defined(__POWERPC__)
     #define FF_ARCHITECTURE "powerpc"
 #elif defined(__riscv__) || defined(__riscv)
     #define FF_ARCHITECTURE "riscv"
-- 
2.48.0

