From cf7f2080fbf1ee2f9a49ee5c0c6bb10cc0b4600e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:50:28 +0800
Subject: [PATCH 06/13] memory_apple: fix for 32-bit

---
 src/detection/memory/memory_apple.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/detection/memory/memory_apple.c b/src/detection/memory/memory_apple.c
index cabdab26..b4d608bf 100644
--- a/src/detection/memory/memory_apple.c
+++ b/src/detection/memory/memory_apple.c
@@ -4,6 +4,7 @@
 #include <mach/mach.h>
 #include <sys/sysctl.h>
 #include <unistd.h>
+#include <AvailabilityMacros.h>
 
 const char* ffDetectMemory(FFMemoryResult* ram)
 {
@@ -11,20 +12,33 @@ const char* ffDetectMemory(FFMemoryResult* ram)
     if (sysctl((int[]){ CTL_HW, HW_MEMSIZE }, 2, &ram->bytesTotal, &length, NULL, 0))
         return "Failed to read hw.memsize";
 
+#if defined(__i386__) || defined(__ppc__)
+    mach_msg_type_number_t count = HOST_VM_INFO_COUNT;
+    vm_statistics_data_t vmstat;
+    if(host_statistics(mach_host_self(), HOST_VM_INFO, (host_info_t) (&vmstat), &count) != KERN_SUCCESS)
+        return "Failed to read host_statistics";
+#else
     mach_msg_type_number_t count = HOST_VM_INFO64_COUNT;
     vm_statistics64_data_t vmstat;
     if(host_statistics64(mach_host_self(), HOST_VM_INFO64, (host_info64_t) (&vmstat), &count) != KERN_SUCCESS)
         return "Failed to read host_statistics64";
+#endif
 
     // https://github.com/exelban/stats/blob/master/Modules/RAM/readers.swift#L56
     ram->bytesUsed = ((uint64_t)
         + vmstat.active_count
         + vmstat.inactive_count
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1050
         + vmstat.speculative_count
+#endif
         + vmstat.wire_count
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1070
         + vmstat.compressor_page_count
-        - vmstat.purgeable_count
         - vmstat.external_page_count
+#endif
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
+        - vmstat.purgeable_count
+#endif
     ) * instance.state.platform.sysinfo.pageSize;
 
     return NULL;
-- 
2.48.0

