From cf7f2080fbf1ee2f9a49ee5c0c6bb10cc0b4600e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:50:28 +0800
Subject: [PATCH 06/13] memory_apple: fix for 32-bit

---
 src/detection/memory/memory_apple.c | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/src/detection/memory/memory_apple.c b/src/detection/memory/memory_apple.c
index cabdab26..b4d608bf 100644
--- a/src/detection/memory/memory_apple.c
+++ b/src/detection/memory/memory_apple.c
@@ -4,6 +4,7 @@
 #include <mach/mach.h>
 #include <sys/sysctl.h>
 #include <unistd.h>
+#include <AvailabilityMacros.h>
 
 const char* ffDetectMemory(FFMemoryResult* ram)
 {
@@ -15,18 +16,32 @@
     if (sysctlbyname("hw.memsize_usable", &usableMemory, &length, NULL, 0) != 0)
         usableMemory = ram->bytesTotal;
 
+#if defined(__i386__) || defined(__ppc__) || MAC_OS_X_VERSION_MIN_REQUIRED < 1060
+    mach_msg_type_number_t count = HOST_VM_INFO_COUNT;
+    vm_statistics_data_t vmstat;
+    if(host_statistics(mach_host_self(), HOST_VM_INFO, (host_info_t) (&vmstat), &count) != KERN_SUCCESS)
+        return "Failed to read host_statistics";
+#else
     mach_msg_type_number_t count = HOST_VM_INFO64_COUNT;
     vm_statistics64_data_t vmstat;
     if(host_statistics64(mach_host_self(), HOST_VM_INFO64, (host_info64_t) (&vmstat), &count) != KERN_SUCCESS)
         return "Failed to read host_statistics64";
-
+#endif
     uint64_t pageSize = instance.state.platform.sysinfo.pageSize;
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1090
     uint64_t appMemory = vmstat.internal_page_count * pageSize;
-    uint64_t wiredMemory = vmstat.wire_count * pageSize;
     uint64_t compressed = vmstat.compressor_page_count * pageSize;
+#endif
+    uint64_t active = vmstat.active_count * pageSize;
+    uint64_t inactive = vmstat.inactive_count * pageSize;
+    uint64_t wiredMemory = vmstat.wire_count * pageSize;
     uint64_t reserved = ram->bytesTotal - usableMemory;
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1090
     ram->bytesUsed = appMemory + wiredMemory + compressed + reserved;
+#else
+    ram->bytesUsed = active + inactive + wiredMemory + reserved;
+#endif
 
     return NULL;
 }
