From 0fcb1f603f6c1df9fd68ab758f9b08a9dec8d372 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:52:55 +0800
Subject: [PATCH] swap_fat_arch_64 does not exist in 10.6

diff --git a/src/common/impl/binary_apple.c b/src/common/impl/binary_apple.c
index 6efeb5d5..ccdc9a9a 100644
--- a/src/common/impl/binary_apple.c
+++ b/src/common/impl/binary_apple.c
@@ -9,6 +9,19 @@
 #include <mach-o/swap.h>
 #include <mach-o/fat.h>
 
+#ifndef FAT_MAGIC_64
+#define FAT_MAGIC_64 0xcafebabf
+#endif
+#ifndef FAT_CIGAM_64
+#define FAT_CIGAM_64 0xbfbafeca
+#endif
+#ifndef fat_arch_64
+#define fat_arch_64 fat_arch
+#endif
+#ifndef swap_fat_arch_64
+#define swap_fat_arch_64 swap_fat_arch
+#endif
+
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations" // swap_fat_arch
 
 // Ref: https://github.com/AlexDenisov/segment_dumper/blob/master/main.c

--- a/src/detection/swap/swap_apple.c	2026-04-23 20:12:45.000000000 +0800
+++ b/src/detection/swap/swap_apple.c	2026-04-25 17:37:44.000000000 +0800
@@ -11,13 +11,13 @@
     }
 
     if (xsw.xsu_total == 0) {
-        if (__builtin_available(macOS 26.0, *)) {
+//        if (__builtin_available(macOS 26.0, *)) {
             // "vm.compressor_mode" no longer exists in macOS 26.0
-        } else {
+//        } else {
             if (ffSysctlGetInt("vm.compressor_mode", 4) <= 2) {
                 return NULL; // Swap is disabled
             }
-        }
+//        }
     }
 
     FFSwapResult* swap = FF_LIST_ADD(FFSwapResult, *result);
