From 0fcb1f603f6c1df9fd68ab758f9b08a9dec8d372 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:52:55 +0800
Subject: [PATCH 10/13] swap_fat_arch_64 does not exist in 10.6

---
 src/util/binary_apple.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/util/binary_apple.c b/src/util/binary_apple.c
index 6efeb5d5..ccdc9a9a 100644
--- a/src/common/impl/binary_apple.c
+++ b/src/common/impl/binary_apple.c
@@ -10,6 +10,13 @@
 #include <mach-o/swap.h>
 #include <mach-o/fat.h>
 
+#ifndef FAT_MAGIC_64
+#define FAT_MAGIC_64 0xcafebabf
+#endif
+#ifndef FAT_CIGAM_64
+#define FAT_CIGAM_64 0xbfbafeca
+#endif
+
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations" // swap_fat_arch
 
 // Ref: https://github.com/AlexDenisov/segment_dumper/blob/master/main.c
@@ -142,13 +149,12 @@ static const char* dumpFatHeader(FILE *objFile, bool (*cb)(const char *str, uint
         }
         else
         {
-            struct fat_arch_64 arch;
+            struct fat_arch arch;
             if (!readData(objFile, &arch, sizeof(arch), (off_t) (sizeof(header) + i * sizeof(arch))))
                 continue;
 
             if (needSwap)
-                swap_fat_arch_64(&arch, 1, NX_UnknownByteOrder);
-
+                swap_fat_arch(&arch, 1, NX_UnknownByteOrder);
             machHeaderOffset = (off_t)arch.offset;
         }
 
