From b712568fb0e16979301ea5e1ed49ed6494e20487 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Thu, 6 Feb 2025 09:53:55 +0800
Subject: [PATCH 11/13] disk_bsd: no support for creation time on ppc

---
 src/detection/disk/disk_bsd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/detection/disk/disk_bsd.c b/src/detection/disk/disk_bsd.c
index 957314f1..ad7c4404 100644
--- a/src/detection/disk/disk_bsd.c
+++ b/src/detection/disk/disk_bsd.c
@@ -17,6 +17,10 @@
 #if __has_include(<libgeom.h>)
 #include <libgeom.h>
 
+#ifdef __APPLE__
+#include <AvailabilityMacros.h>
+#endif
+
 static const char* detectFsLabel(struct statfs* fs, FFDisk* disk)
 {
     if (!ffStrStartsWith(fs->f_mntfromname, "/dev/"))
@@ -182,7 +186,7 @@
         #ifdef __OpenBSD__
         #define st_birthtimespec __st_birthtim
         #endif
-        #ifndef __DragonFly__
+        #if !defined(__DragonFly__) && !(defined(__APPLE__) && (MAC_OS_X_VERSION_MIN_REQUIRED < 1060))
         struct stat st;
         if(stat(fs->f_mntonname, &st) == 0 && st.st_birthtimespec.tv_sec > 0)
             disk->createTime = (uint64_t)(((uint64_t) st.st_birthtimespec.tv_sec * 1000) + ((uint64_t) st.st_birthtimespec.tv_nsec / 1000000));
