--- a/src/detection/wm/wm_apple.m	2026-03-07 11:08:19.000000000 +0800
+++ b/src/detection/wm/wm_apple.m	2026-03-17 12:33:00.000000000 +0800
@@ -5,9 +5,22 @@
 #include "common/stringUtils.h"
 
 #include <ctype.h>
+#include <Foundation/Foundation.h>
+#include <AvailabilityMacros.h>
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1050
 #include <libproc.h>
-#import <Foundation/Foundation.h>
+#endif
+
+#if MAC_OS_X_VERSION_MIN_REQUIRED < 1090
+#define POOLSTART NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
+#define POOLEND   [pool release];
+#else
+#define POOLSTART
+#define POOLEND
+#endif
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1050
 const char* ffDetectWMPlugin(FFstrbuf* pluginName)
 {
     int request[] = {CTL_KERN, KERN_PROC, KERN_PROC_ALL};
@@ -56,22 +69,26 @@
                     {
                         lastSlash -= strlen("MacOS");
                         strcpy(lastSlash, "Info.plist"); // X.app/Contents/Info.plist
-                        NSError* error;
-                        NSDictionary* dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL fileURLWithPath:@(buf)]
-                                                        error:&error];
+
+                        POOLSTART
+                        NSDictionary* dict = [NSDictionary dictionaryWithContentsOfFile:[NSString stringWithUTF8String:buf]];
                         if (dict)
                         {
-                            NSString* name = dict[@"CFBundleDisplayName"] ?: dict[@"CFBundleName"];
-                            ffStrbufSetS(pluginName, name.UTF8String ?: comm);
+                            NSString* name = [dict objectForKey:@"CFBundleDisplayName"];
+                            if (!name) name = [dict objectForKey:@"CFBundleName"];
+                            ffStrbufSetS(pluginName, name ? [name UTF8String] : comm);
 
-                            NSString* version = dict[@"CFBundleShortVersionString"];
+                            NSString* version = [dict objectForKey:@"CFBundleShortVersionString"];
                             if (version)
                             {
                                 ffStrbufAppendC(pluginName, ' ');
                                 ffStrbufAppendS(pluginName, version.UTF8String);
                             }
 
-                            break;
+                            POOLEND
+
+                            if (dict)
+                                break;
                         }
                     }
                 }
@@ -85,7 +102,14 @@
 
     return NULL;
 }
-
+#else
+const char* ffDetectWMPlugin(FF_A_UNUSED FFstrbuf* pluginName)
+{
+    return "Not supported on this platform";
+}
+#endif
+ 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1050
 const char* ffDetectWMVersion(const FFstrbuf* wmName, FFstrbuf* result, FF_A_UNUSED FFWMOptions* options)
 {
     if (!wmName)
@@ -93,18 +117,27 @@
 
     if (ffStrbufEqualS(wmName, "WindowServer"))
     {
-        NSError* error;
-        NSDictionary* dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL fileURLWithPath:@"/System/Library/PrivateFrameworks/SkyLight.framework/Resources/version.plist" isDirectory:NO]
-                                           error:&error];
+        POOLSTART
+        NSDictionary* dict = [NSDictionary dictionaryWithContentsOfFile:@"/System/Library/PrivateFrameworks/SkyLight.framework/Resources/version.plist"];
         if (!dict)
         {
-            dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL fileURLWithPath:@"/System/Library/Frameworks/ApplicationServices.framework/Frameworks/CoreGraphics.framework/Resources/version.plist" isDirectory:NO]
-                                           error:&error];
+            dict = [NSDictionary dictionaryWithContentsOfFile:@"/System/Library/Frameworks/ApplicationServices.framework/Frameworks/CoreGraphics.framework/Resources/version.plist"];
         }
 
         if (dict)
-            ffStrbufSetS(result, ((NSString*) dict[@"CFBundleShortVersionString"]).UTF8String);
+        {
+            NSString* versionString = [dict objectForKey:@"CFBundleShortVersionString"];
+            if (versionString)
+                ffStrbufSetS(result, [versionString UTF8String]);
+        }
+        POOLEND
     }
 
     return NULL;
 }
+#else
+const char* ffDetectWMVersion(FF_A_UNUSED const FFstrbuf* wmName, FF_A_UNUSED FFstrbuf* result, FF_A_UNUSED FFWMOptions* options)
+{
+    return "Not supported on this platform";
+}
+#endif
