From 68c1a876c82827354114b100bfd7c5caa313f24d Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 11 Feb 2025 01:34:58 +0800
Subject: [PATCH 13/13] wm_apple: unbreak

diff --git a/src/detection/wm/wm_apple.m b/src/detection/wm/wm_apple.c
similarity index 66%
rename from src/detection/wm/wm_apple.m
rename to src/detection/wm/wm_apple.c
index c2395d81..dca3b600 100644
--- a/src/detection/wm/wm_apple.m
+++ b/src/detection/wm/wm_apple.m
@@ -5,7 +5,6 @@
 #include "common/stringUtils.h"
 
 #include <ctype.h>
-#import <Foundation/Foundation.h>
 
 const char* ffDetectWMPlugin(FFstrbuf* pluginName)
 {
@@ -42,25 +41,7 @@
     return NULL;
 }
 
-const char* ffDetectWMVersion(const FFstrbuf* wmName, FFstrbuf* result, FF_MAYBE_UNUSED FFWMOptions* options)
+const char* ffDetectWMVersion(FF_MAYBE_UNUSED const FFstrbuf* wmName, FF_MAYBE_UNUSED FFstrbuf* result, FF_MAYBE_UNUSED FFWMOptions* options)
 {
-    if (!wmName)
-        return "No WM detected";
-
-    if (ffStrbufEqualS(wmName, "WindowServer"))
-    {
-        NSError* error;
-        NSDictionary* dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL URLWithString:@"file:///System/Library/PrivateFrameworks/SkyLight.framework/Resources/version.plist"]
-                                           error:&error];
-        if (!dict)
-        {
-            dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL URLWithString:@"file:///System/Library/Frameworks/ApplicationServices.framework/Frameworks/CoreGraphics.framework/Resources/version.plist"]
-                                           error:&error];
-        }
-
-        if (dict)
-            ffStrbufInitS(result, ((NSString*) dict[@"CFBundleShortVersionString"]).UTF8String);
-    }
-
-    return NULL;
+    return "Not supported on this platform";
 }
