From 9eff388297ccacd1e4216d7b0e05333fd89ce29e Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Tue, 11 Feb 2025 03:17:13 +0800
Subject: [PATCH 14/14] displayserver_apple: minor fix

---
 .../displayserver/displayserver_apple.c       | 25 +++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/src/detection/displayserver/displayserver_apple.c b/src/detection/displayserver/displayserver_apple.c
index 355d4047..b34cc1cb 100644
--- a/src/detection/displayserver/displayserver_apple.c
+++ b/src/detection/displayserver/displayserver_apple.c
@@ -104,7 +104,31 @@ static void detectDisplays(FFDisplayServerResult* ds)
             uint32_t physicalWidth = 0, physicalHeight = 0;
             uint32_t preferredWidth = 0, preferredHeight = 0;
             double preferredRefreshRate = 0;
+#ifdef CG_LEGACY_API
+            if(displayInfo)
+            {
+                CFDictionaryRef productNames;
+                if(!ffCfDictGetDict(displayInfo, CFSTR(kDisplayProductName), &productNames))
+                    ffCfDictGetString(productNames, CFSTR("en_US"), &buffer);
+
+                // CGDisplayScreenSize reports invalid result for external displays on old Intel MacBook Pro
+                CFDataRef edidRef = (CFDataRef) CFDictionaryGetValue(displayInfo, CFSTR(kIODisplayEDIDKey));
+                if (edidRef && CFGetTypeID(edidRef) == CFDataGetTypeID())
+                {
+                    const uint8_t* edidData = CFDataGetBytePtr(edidRef);
+                    uint32_t edidLength = (uint32_t) CFDataGetLength(edidRef);
+                    if (edidLength >= 128)
+                        ffEdidGetPhysicalSize(edidData, &physicalWidth, &physicalHeight);
+                }
+            }
 
+            if (!physicalWidth || !physicalHeight)
+            {
+                CGSize size = CGDisplayScreenSize(screen);
+                physicalWidth = (uint32_t) (size.width + 0.5);
+                physicalHeight = (uint32_t) (size.height + 0.5);
+            }
+#else
             if(displayInfo)
             {
                 CFDictionaryRef productNames;
@@ -154,6 +178,7 @@ static void detectDisplays(FFDisplayServerResult* ds)
                 physicalWidth = (uint32_t) (size.width + 0.5);
                 physicalHeight = (uint32_t) (size.height + 0.5);
             }
+#endif
 
             FFDisplayResult* display = ffdsAppendDisplay(ds,
 #ifdef CG_LEGACY_API
-- 
2.48.0

