--- a/src/common/impl/networking_linux.c
+++ b/src/common/impl/networking_linux.c	2025-08-29 20:46:35.000000000 +0800
@@ -61,7 +61,6 @@
             #endif
         }
 
-        #ifndef __APPLE__
         FF_DEBUG("Using sendto() + MSG_DONTWAIT to send %u bytes of data", state->command.length);
         ssize_t sent = sendto(state->sockfd,
                                 state->command.chars,
@@ -75,29 +74,6 @@
                                 MSG_DONTWAIT,
                                 state->addr->ai_addr,
                                 state->addr->ai_addrlen);
-        #else
-        if (fcntl(state->sockfd, F_SETFL, O_NONBLOCK) == -1) {
-            FF_DEBUG("fcntl(F_SETFL) failed: %s", strerror(errno));
-            return "fcntl(F_SETFL) failed";
-        }
-        FF_DEBUG("Using connectx() to send %u bytes of data", state->command.length);
-        // Use connectx to establish connection and send data in one call
-        size_t sent;
-        if (connectx(state->sockfd,
-            &(sa_endpoints_t) {
-                .sae_dstaddr = state->addr->ai_addr,
-                .sae_dstaddrlen = state->addr->ai_addrlen,
-            },
-            SAE_ASSOCID_ANY, state->tfo ? CONNECT_DATA_IDEMPOTENT : 0,
-            &(struct iovec) {
-                .iov_base = state->command.chars,
-                .iov_len = state->command.length,
-            }, 1, &sent, NULL) != 0) sent = 0;
-        if (fcntl(state->sockfd, F_SETFL, 0) == -1) {
-            FF_DEBUG("fcntl(F_SETFL) failed: %s", strerror(errno));
-            return "fcntl(F_SETFL) failed";
-        }
-        #endif
         if (sent > 0 || (errno == EAGAIN || errno == EWOULDBLOCK
             #ifdef __APPLE__
             // On macOS EINPROGRESS means the connection cannot be completed immediately
@@ -107,11 +83,7 @@
         ))
         {
             FF_DEBUG(
-                #ifdef __APPLE__
-                "connectx()"
-                #else
                 "sendto()"
-                #endif
                 " %s (sent=%zd, errno=%d: %s)", errno == 0 ? "succeeded" : "was in progress",
                 sent, errno, strerror(errno));
             freeaddrinfo(state->addr);
@@ -121,17 +93,9 @@
         }
 
         FF_DEBUG(
-            #ifdef __APPLE__
-            "connectx()"
-            #else
             "sendto()"
-            #endif
             " failed: %s (errno=%d)", strerror(errno), errno);
-        #ifdef __APPLE__
-        return "connectx() failed";
-        #else
         return "sendto() failed";
-        #endif
     #else
         FF_UNUSED(state);
         return "TFO support is not available";
