--- a/src/common/apple/smc_temps.c	2026-05-13 19:54:48.000000000 +0800
+++ b/src/common/apple/smc_temps.c	2026-05-19 06:22:33.000000000 +0800
@@ -3,7 +3,18 @@
 #include "common/stringUtils.h"
 
 #include <stdint.h>
-#include <IOKit/IOKitLib.h>
+#include <machine/endian.h>
+#include <libkern/OSByteOrder.h>
+
+#ifndef ntohll
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#define ntohll(x) (x)
+#elif defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#define ntohll(x) OSSwapInt64(x)
+#else
+#error "Unknown endianness"
+#endif
+#endif
 
 static const char kSmcCmdReadBytes = 5;
 static const char kSmcCmdReadKeyInfo = 9;
