--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -942,12 +942,12 @@ elseif(APPLE)
         src/common/impl/processing_linux.c
         src/common/impl/sysctl.c
         src/common/impl/FFPlatform_unix.c
-        src/common/impl/binary_apple.c
+        src/common/impl/binary_linux.c
         src/common/impl/kmod_apple.c
         src/common/apple/cf_helpers.c
         src/common/apple/osascript.m
         src/common/apple/smc_temps.c
-        src/detection/battery/battery_apple.c
+        src/detection/battery/battery_nosupport.c
         src/detection/bluetooth/bluetooth_nosupport.c
         src/detection/bluetoothradio/bluetoothradio_nosupport.c
         src/detection/bootmgr/bootmgr_apple.c
@@ -959,7 +959,7 @@ elseif(APPLE)
         src/detection/cursor/cursor_apple.m
         src/detection/disk/disk_bsd.c
         src/detection/dns/dns_apple.c
-        src/detection/physicaldisk/physicaldisk_apple.c
+        src/detection/physicaldisk/physicaldisk_nosupport.c
         src/detection/diskio/diskio_apple.c
         src/detection/displayserver/displayserver_apple.c
         src/detection/font/font_apple.m
@@ -1015,7 +1015,7 @@ elseif(APPLE)
             src/detection/board/board_apple.c
             src/detection/chassis/chassis_apple.c
             src/detection/host/host_apple.c
-            src/detection/physicalmemory/physicalmemory_apple.m
+            src/detection/physicalmemory/physicalmemory_nosupport.c
         )
     endif()
 elseif(WIN32)
@@ -1711,7 +1711,6 @@ elseif(APPLE)
         PRIVATE "-framework Cocoa"
         PRIVATE "-framework CoreFoundation"
         PRIVATE "-framework CoreAudio"
-        PRIVATE "-framework CoreMedia"
         PRIVATE "-framework CoreVideo"
         PRIVATE "-framework IOBluetooth"
         PRIVATE "-framework IOKit"
