--- a/src/common/apple/smc_temps.c	2026-01-14 13:58:22.000000000 +0800
+++ b/src/common/apple/smc_temps.c	2026-01-18 15:16:39.000000000 +0800
@@ -4,8 +4,20 @@
 
 #include <stdint.h>
 #include <math.h>
+#include <machine/endian.h>
+#include <libkern/OSByteOrder.h>
 #include <IOKit/IOKitLib.h>
 
+#ifndef ntohll
+#if defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_BIG_ENDIAN__)
+#define ntohll(x) (x)
+#elif defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
+#define ntohll(x) OSSwapInt64(x)
+#else
+#error "Unknown endianness"
+#endif
+#endif
+
 static const char kSmcCmdReadBytes = 5;
 static const char kSmcCmdReadKeyInfo = 9;
 static const uint32_t kKernelIndexSmc = 2;
