--- a/src/detection/lm/lm_apple.m	2026-09-01 19:44:59.000000000 +0800
+++ b/src/detection/lm/lm_apple.m	2026-09-03 10:59:47.000000000 +0800
@@ -11,7 +11,8 @@
                                                                 error:&error];
 
     if (dict) {
-        ffStrbufSetS(&result->version, ((NSString*) dict[@"CFBundleShortVersionString"]).UTF8String);
+        NSString* version = [dict objectForKey:@"CFBundleShortVersionString"];
+        ffStrbufSetS(&result->version, version.UTF8String);
     }
 
     return nullptr;
