diff --git a/src/detection/lm/lm_apple.m b/src/detection/lm/lm_apple.m
index d600058d4..af5e7cc3f 100644
--- a/src/detection/lm/lm_apple.m
+++ b/src/detection/lm/lm_apple.m
@@ -6,12 +6,12 @@ const char* ffDetectLM(FFLMResult* result) {
     ffStrbufSetStatic(&result->service, "loginwindow");
     ffStrbufSetStatic(&result->prettyName, "Login Window");
 
-    NSError* error;
-    NSDictionary* dict = [NSDictionary dictionaryWithContentsOfURL:[NSURL fileURLWithPath:@"/System/Library/CoreServices/loginwindow.app/Contents/Info.plist" isDirectory:NO]
-                                                                error:&error];
+    // +dictionaryWithContentsOfURL:error: is 10.13+
+    NSDictionary* dict = [NSDictionary dictionaryWithContentsOfFile:@"/System/Library/CoreServices/loginwindow.app/Contents/Info.plist"];
 
     if (dict) {
-        ffStrbufSetS(&result->version, ((NSString*) dict[@"CFBundleShortVersionString"]).UTF8String);
+        NSString* version = [dict objectForKey:@"CFBundleShortVersionString"];
+        ffStrbufSetS(&result->version, version.UTF8String);
     }
 
     return nullptr;
