--- qtbase/mkspecs/common/gcc-base.conf	2023-11-07 02:10:54.000000000 +0800
+++ qtbase/mkspecs/common/gcc-base.conf	2023-11-07 02:11:27.000000000 +0800
@@ -34,7 +34,7 @@
 QMAKE_CFLAGS_OPTIMIZE      = -O2
 QMAKE_CFLAGS_OPTIMIZE_FULL = -O3
 
-QMAKE_CFLAGS               += -pipe
+QMAKE_CFLAGS               += -pipe -isysroot / -mmacosx-version-min=10.6
 QMAKE_CFLAGS_DEPS          += -M
 QMAKE_CFLAGS_WARN_ON       += -Wall -W
 QMAKE_CFLAGS_WARN_OFF      += -w
@@ -63,7 +63,7 @@
 QMAKE_CXXFLAGS_EXCEPTIONS_OFF += $$QMAKE_CFLAGS_EXCEPTIONS_OFF
 QMAKE_CXXFLAGS_SPLIT_SECTIONS += $$QMAKE_CFLAGS_SPLIT_SECTIONS
 
-QMAKE_LFLAGS           +=
+QMAKE_LFLAGS           += -Wl,-syslibroot,/ -mmacosx-version-min=10.6
 QMAKE_LFLAGS_DEBUG     +=
 QMAKE_LFLAGS_APP       +=
 QMAKE_LFLAGS_RELEASE   +=
