--- qtbase/mkspecs/macx-g++-32/qmake.conf	2014-09-11 18:48:07.000000000 +0800
+++ qtbase/mkspecs/macx-g++-32/qmake.conf	2023-11-06 22:22:53.000000000 +0800
@@ -16,9 +16,9 @@
 
 QMAKE_MACOSX_DEPLOYMENT_TARGET = 10.6
 
-QMAKE_CFLAGS           += -arch i386
-QMAKE_OBJECTIVE_CFLAGS += -arch i386
-QMAKE_CXXFLAGS         += -arch i386
-QMAKE_LFLAGS           += -arch i386
+QMAKE_CFLAGS           += -arch ppc
+QMAKE_OBJECTIVE_CFLAGS += -arch ppc
+QMAKE_CXXFLAGS         += -arch ppc
+QMAKE_LFLAGS           += -arch ppc
 
 load(qt_config)
