From 7bd3fa991675c2c71d0384cd576a095b9a30ecc4 Mon Sep 17 00:00:00 2001
From: Diana Kalinichenko <dkalinichenko@janestreet.com>
Date: Thu, 29 May 2025 18:17:00 -0400
Subject: [PATCH] ppxlib.0.36.0

---
 src/ppx_optcomp.ml | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/ppx_optcomp.ml b/src/ppx_optcomp.ml
index 0e8b9af..c07b444 100644
--- src/ppx_optcomp.ml
+++ src/ppx_optcomp.ml
@@ -481,7 +481,8 @@ let map =
       in
       let x =
         match x with
-        | Pexp_function cs -> Pexp_function (List.filter_map cs ~f)
+        | Pexp_function (params, constr, Pfunction_cases (cs, loc, attr)) ->
+          Pexp_function (params, constr, Pfunction_cases (List.filter_map cs ~f, loc, attr))
         | Pexp_match (e, cs) -> Pexp_match (super#expression env e, List.filter_map cs ~f)
         | Pexp_try (e, cs) -> Pexp_try (super#expression env e, List.filter_map cs ~f)
         | _ -> x
