--- src/audio/coreaudio/SDL_coreaudio.m	2025-01-01 23:48:45.000000000 +0800
+++ src/audio/coreaudio/SDL_coreaudio.m	2025-01-05 13:30:31.000000000 +0800
@@ -882,11 +887,11 @@
         break;
     case 7:
         // L R C LFE Cs Ls Rs
-        layout.mChannelLayoutTag = kAudioChannelLayoutTag_WAVE_6_1;
+        layout.mChannelLayoutTag = kAudioChannelLayoutTag_MPEG_6_1_A;
         break;
     case 8:
         // L R C LFE Rls Rrs Ls Rs
-        layout.mChannelLayoutTag = kAudioChannelLayoutTag_WAVE_7_1;
+        layout.mChannelLayoutTag = kAudioChannelLayoutTag_MPEG_7_1_A;
         break;
     default:
         return SDL_SetError("Unsupported audio channels");
