--- configure
+++ configure	2025-04-01 20:33:38.000000000 +0800
@@ -23318,14 +23318,14 @@
 
             case "$host" in
                 *)
-                    wayland_client_lib=`find_lib "libwayland-client.so.*" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
-                    wayland_egl_lib=`find_lib "libwayland-egl.so.*" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
+                    wayland_client_lib=`find_lib "libwayland-client.dylib" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
+                    wayland_egl_lib=`find_lib "libwayland-egl.dylib" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
                     if test x$wayland_egl_lib = x; then
                         # This works in Ubuntu 13.10, maybe others
-                        wayland_egl_lib=`find_lib "mesa-egl/libwayland-egl.so.*" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
+                        wayland_egl_lib=`find_lib "mesa-egl/libwayland-egl.dylib" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
                     fi
-                    wayland_cursor_lib=`find_lib "libwayland-cursor.so.*" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
-                    xkbcommon_lib=`find_lib "libxkbcommon.so.*" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
+                    wayland_cursor_lib=`find_lib "libwayland-cursor.dylib" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
+                    xkbcommon_lib=`find_lib "libxkbcommon.dylib" "$WAYLAND_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`
                     ;;
             esac
 
