From 7fdca7ef2153b3c1f2d5d6bf019067255e339f18 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <vital.had@gmail.com>
Date: Sun, 13 Apr 2025 04:45:11 +0800
Subject: [PATCH] CMakeLists: fix detection of netinet6/ipsec.h

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git CMakeLists.txt CMakeLists.txt
index c4e3b4b..d20a46b 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -134,7 +134,7 @@ if(HAVE_NET_IPSEC_H)
 	add_definitions(-DHAVE_NET_IPSEC_H)
 endif()
 
-check_include_files("netinet6/ipsec.h" HAVE_NETINET6_IPSEC_H)
+check_include_files("sys/types.h;netinet6/ipsec.h" HAVE_NETINET6_IPSEC_H)
 if(HAVE_NETINET6_IPSEC_H)
 	add_definitions(-DHAVE_NETINET6_IPSEC_H)
 endif()
