--- CMakeLists.txt	2026-05-01 03:56:01.000000000 +0800
+++ CMakeLists.txt	2026-05-16 08:12:15.000000000 +0800
@@ -86,16 +86,6 @@
   find_package(SQLite3 REQUIRED)
 endif()
 
-if(NOT DEFINED MACOS_VERSION_MIN) # to allow overruling with -DMACOS_VERSION_MIN
-  set(MACOS_VERSION_MIN 10.14)
-  if(build_search AND CMAKE_SYSTEM MATCHES "Darwin")
-    set(MACOS_VERSION_MIN 10.15) # needs std::filesystem
-  endif()
-  if(build_wizard AND CMAKE_SYSTEM MATCHES "Darwin")
-    set(MACOS_VERSION_MIN 13.0) # needs Qt6
-  endif()
-endif()
-
 # use C++17 standard for compiling (unless very new Clang is present)
 if(
   (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND
@@ -123,9 +113,9 @@
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if(CMAKE_SYSTEM MATCHES "Darwin")
-  set(CMAKE_OSX_DEPLOYMENT_TARGET "${MACOS_VERSION_MIN}" CACHE STRING "Minimum OS X deployment version")
-  set(CMAKE_CXX_FLAGS "-Wno-deprecated-register ${CMAKE_CXX_FLAGS}")
-  set(CMAKE_C_FLAGS "-Wno-deprecated-register ${CMAKE_C_FLAGS}")
+#  set(CMAKE_OSX_DEPLOYMENT_TARGET "${MACOS_VERSION_MIN}" CACHE STRING "Minimum OS X deployment version")
+#  set(CMAKE_CXX_FLAGS "-Wno-deprecated-register ${CMAKE_CXX_FLAGS}")
+#  set(CMAKE_C_FLAGS "-Wno-deprecated-register ${CMAKE_C_FLAGS}")
   find_library(CORESERVICES_LIB CoreServices)
   set(EXTRA_LIBS ${CORESERVICES_LIB})
 endif()
@@ -155,8 +145,8 @@
 
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DSQLITE_OMIT_LOAD_EXTENSION=1")
 
-# Use 64-bit off_t on 32-bit Linux
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SIZEOF_VOID_P EQUAL 4)
+# Use 64-bit off_t on 32-bit Unix
+if(CMAKE_SIZEOF_VOID_P EQUAL 4)
   # ensure 64bit offsets are used for filesystem accesses for 32bit compilation
   add_compile_definitions(_FILE_OFFSET_BITS=64)
 endif()
