--- src/ports/SkFontMgr_mac_ct.cpp.orig
+++ src/ports/SkFontMgr_mac_ct.cpp
@@ -307,7 +307,12 @@
 
 /** Use CTFontManagerCopyAvailableFontFamilyNames if available, simulate if not. */
 SkUniqueCFRef<CFArrayRef> SkCTFontManagerCopyAvailableFontFamilyNames() {
-#ifdef SK_BUILD_FOR_IOS
+#if defined(SK_BUILD_FOR_MAC) && MAC_OS_X_VERSION_MAX_ALLOWED < 1060
+    // CTFontManagerCopyAvailableFontFamilyNames is 10.6+.
+    SkUniqueCFRef<CTFontCollectionRef> collection(
+        CTFontCollectionCreateFromAvailableFonts(nullptr));
+    return SkUniqueCFRef<CFArrayRef>(SkCopyAvailableFontFamilyNames(collection.get()));
+#elif defined(SK_BUILD_FOR_IOS)
     using CTFontManagerCopyAvailableFontFamilyNamesProc = CFArrayRef (*)(void);
     CTFontManagerCopyAvailableFontFamilyNamesProc ctFontManagerCopyAvailableFontFamilyNames;
     *(void**)(&ctFontManagerCopyAvailableFontFamilyNames) =
@@ -440,8 +445,13 @@
                 kCFAllocatorDefault, reinterpret_cast<const UInt8 *>(locale), strlen(locale),
                 kCFStringEncodingUTF8, false));
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1090
         SkUniqueCFRef<CTFontRef> fallbackFont(CTFontCreateForStringWithLanguage(
                 familyFont.get(), string.get(), range, cfLocale.get()));
+#else
+        SkUniqueCFRef<CTFontRef> fallbackFont(CTFontCreateForString(
+                familyFont.get(), string.get(), range));
+#endif
         if (!fallbackFont || !has_character(fallbackFont.get(), character)) {
             return nullptr;
         }
