--- src/core/Sk4px.h.orig	2026-04-26 19:44:03.025294901 +0000
+++ src/core/Sk4px.h	2026-04-26 19:44:15.771631987 +0000
@@ -27,8 +27,11 @@
 
     // RGBA rgba XYZW xyzw -> AAAA aaaa WWWW wwww
     Sk4px alphas() const {
-        static_assert(SK_A32_SHIFT == 24, "This method assumes little-endian.");
+#ifdef SK_CPU_BENDIAN
+        return Sk4px(skvx::shuffle<0,0,0,0, 4,4,4,4, 8,8,8,8, 12,12,12,12>(fV));
+#else
         return Sk4px(skvx::shuffle<3,3,3,3, 7,7,7,7, 11,11,11,11, 15,15,15,15>(fV));
+#endif
     }
     Sk4px inv() const { return Sk4px(skvx::byte16(255) - fV); }
 
