--- include/private/base/SkFeatures.h	2026-04-22 01:31:39.000000000 +0800
+++ include/private/base/SkFeatures.h	2026-04-26 22:07:42.000000000 +0800
@@ -48,11 +48,11 @@
     #elif defined(__BYTE_ORDER__) && (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
         #define SK_CPU_LENDIAN
     #elif defined(__sparc) || defined(__sparc__) || \
-      defined(_POWER) || defined(__powerpc__) || \
+      defined(__POWERPC__) || defined(__powerpc__) || \
       defined(__ppc__) || defined(__hppa) || \
       defined(__PPC__) || defined(__PPC64__) || \
       defined(_MIPSEB) || defined(__ARMEB__) || \
-      defined(__s390__) || \
+      defined(__s390__) || defined(_POWER) || \
       (defined(__sh__) && defined(__BIG_ENDIAN__)) || \
       (defined(__ia64) && defined(__BIG_ENDIAN__))
          #define SK_CPU_BENDIAN
@@ -69,7 +69,7 @@
   #define SK_CPU_LOONGARCH 1
 #endif
 
-#if defined(__powerpc__) || defined (__powerpc64__)
+#if defined(__powerpc__) || defined (__powerpc64__) || defined (__POWERPC__)
   #define SK_CPU_PPC 1
 #endif
 

--- modules/skcms/src/skcms_internals.h	2026-04-22 01:31:39.000000000 +0800
+++ modules/skcms/src/skcms_internals.h	2026-04-26 22:08:29.000000000 +0800
@@ -52,6 +52,7 @@
                                                  && !defined(__arm__) \
                                                  && !defined(__riscv) \
                                                  && !defined(__powerpc__) \
+                                                 && !defined(__POWERPC__) \
                                                  && !defined(__loongarch__) \
                                                  && !defined(_WIN32) && !defined(__SYMBIAN32__)
             #define SKCMS_HAS_MUSTTAIL 1
