--- src/ports/SkFontMgr_mac_ct.cpp.orig	2026-04-29 04:15:59.000000000 +0800
+++ src/ports/SkFontMgr_mac_ct.cpp	2026-05-08 07:47:53.000000000 +0800
@@ -440,8 +440,13 @@
                 kCFAllocatorDefault, reinterpret_cast<const UInt8 *>(locale), strlen(locale),
                 kCFStringEncodingUTF8, false));
 
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= 1090
         SkUniqueCFRef<CTFontRef> fallbackFont(CTFontCreateForStringWithLanguage(
                 familyFont.get(), string.get(), range, cfLocale.get()));
+#else
+        SkUniqueCFRef<CTFontRef> fallbackFont(CTFontCreateForString(
+                familyFont.get(), string.get(), range));
+#endif
         if (!fallbackFont || !has_character(fallbackFont.get(), character)) {
             return nullptr;
         }
