--- include/private/base/SkLoadUserConfig.h.orig	2026-04-26 17:23:43.024287216 +0000
+++ include/private/base/SkLoadUserConfig.h	2026-04-26 17:23:48.211426974 +0000
@@ -55,9 +55,8 @@
 #endif
 
 #if defined(SK_CPU_BENDIAN) && !defined(I_ACKNOWLEDGE_SKIA_DOES_NOT_SUPPORT_BIG_ENDIAN)
-    #error "The Skia team is not endian-savvy enough to support big-endian CPUs."
-    #error "If you still want to use Skia,"
-    #error "please define I_ACKNOWLEDGE_SKIA_DOES_NOT_SUPPORT_BIG_ENDIAN."
+    #warning "The Skia team is not endian-savvy enough to support big-endian CPUs."
+    #warning "Big-endian support is experimental and may have issues."
 #endif
 
 #define SK_USER_CONFIG_WAS_LOADED
