https://bugs.gentoo.org/800824

--- include/mesos/csi/v0.hpp	2020-11-17 21:52:12.000000000 +0800
+++ include/mesos/csi/v0.hpp	2025-05-10 22:04:17.000000000 +0800
@@ -94,7 +94,7 @@
     google::protobuf::util::MessageToJsonString(message, &output);
 
   CHECK(status.ok())
-    << "Could not convert messages to string: " << status.error_message();
+    << "Could not convert messages to string: " << status.message();
 
   return stream << output;
 }

--- include/mesos/csi/v1.hpp	2020-11-17 21:52:12.000000000 +0800
+++ include/mesos/csi/v1.hpp	2025-05-10 22:55:45.000000000 +0800
@@ -94,7 +94,7 @@
     google::protobuf::util::MessageToJsonString(message, &output);
 
   CHECK(status.ok())
-    << "Could not convert messages to string: " << status.error_message();
+    << "Could not convert messages to string: " << status.message();
 
   return stream << output;
 }
