--- Makefile.in	2026-07-10 03:44:45.000000000 +0800
+++ Makefile.in	2026-08-05 11:00:53.000000000 +0800
@@ -463,7 +463,9 @@
 endif
 	${INSTALL} -p -m 644 base/*.asy $(addprefix base/,$(EXTRA)) \
 		asy-keywords.el $(asydir)
-	${INSTALL} -p -m 755 $(SHIMLIBS) $(asydir)
+	-if test -n "$(SHIMLIBS)"; then \
+		${INSTALL} -p -m 755 $(SHIMLIBS) $(asydir); \
+	fi
 	${INSTALL} -p -m 644 base/collections/*.asy $(asydir)/collections
 	${INSTALL} -p -m 644 base/shaders/*.glsl $(shaderdir)
 	${INSTALL} -p -m 644 base/shaders/GL/*.glsl $(shaderdir)/GL
