--- libcnotify/libcnotify_extern.h	2026-06-01 13:47:29.000000000 +0800
+++ libcnotify/libcnotify_extern.h	2026-06-01 18:38:28.000000000 +0800
@@ -24,11 +24,15 @@
 
 #include <common.h>
 
-#if defined( __has_attribute ) && __has_attribute( visibility ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if defined( __has_attribute ) && !defined( __CYGWIN__ ) && !defined( _WIN32 )
+#if __has_attribute( visibility )
 #define LIBCNOTIFY_INTERNAL	__attribute__((visibility("hidden"))) extern
 #else
 #define LIBCNOTIFY_INTERNAL	extern
 #endif
+#else
+#define LIBCNOTIFY_INTERNAL	extern
+#endif
 
 /* Define HAVE_LOCAL_LIBCNOTIFY for local use of libcnotify
  */
