--- src/CMakeLists.txt.orig	2026-09-02 06:12:38.000000000 +0800
+++ src/CMakeLists.txt	2026-09-10 15:03:17.000000000 +0800
@@ -42,10 +42,10 @@
 endif()
 
 if(TARGET lbug_shared)
-        target_link_libraries(lbug_shared PUBLIC lbug_link_deps)
+        target_link_libraries(lbug_shared PUBLIC lbug_link_deps atomic)
 endif()
 if(TARGET lbug)
-        target_link_libraries(lbug PUBLIC lbug_link_deps)
+        target_link_libraries(lbug PUBLIC lbug_link_deps atomic)
 endif()
 
 if(TARGET lbug)
