--- third_party/antlr4_runtime/src/misc/Interval.h	2026-09-02 06:12:38.000000000 +0800
+++ third_party/antlr4_runtime/src/misc/Interval.h	2026-09-10 12:17:46.000000000 +0800
@@ -5,6 +5,8 @@
 
 #pragma once
 
+#include <sys/types.h>
+
 #include "antlr4-common.h"
 
 namespace antlr4 {

--- third_party/antlr4_runtime/src/misc/IntervalSet.h	2026-09-02 06:12:38.000000000 +0800
+++ third_party/antlr4_runtime/src/misc/IntervalSet.h	2026-09-10 12:04:01.000000000 +0800
@@ -5,6 +5,8 @@
 
 #pragma once
 
+#include <sys/types.h>
+
 #include "misc/Interval.h"
 #include "Exceptions.h"
 
--- third_party/antlr4_runtime/src/IntStream.h	2026-09-02 06:12:38.000000000 +0800
+++ third_party/antlr4_runtime/src/IntStream.h	2026-09-10 11:56:47.000000000 +0800
@@ -5,6 +5,8 @@
 
 #pragma once
 
+#include <sys/types.h>
+
 #include "antlr4-common.h"
 
 namespace antlr4 {

--- third_party/thrift/protocol/TProtocol.h	2026-09-02 06:12:38.000000000 +0800
+++ third_party/thrift/protocol/TProtocol.h	2026-09-10 12:14:59.000000000 +0800
@@ -28,15 +28,17 @@
 #include "transport/TTransport.h"
 #include "protocol/TProtocolException.h"
 
-#include <memory>
-
+#include <sys/types.h>
+#include <sys/socket.h>
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif
-#include <sys/types.h>
+
 #include <string>
 #include <map>
+#include <memory>
 #include <climits>
+#include <cstdint>
 
 // Use this to get around strict aliasing rules.
 // For example, uint64_t i = bitwise_cast<uint64_t>(returns_double());

--- third_party/thrift/transport/TTransport.h	2026-09-02 06:12:38.000000000 +0800
+++ third_party/thrift/transport/TTransport.h	2026-09-10 12:21:48.000000000 +0800
@@ -22,6 +22,8 @@
 
 #include "Thrift.h"
 #include "transport/TTransportException.h"
+
+#include <cstdint>
 #include <memory>
 #include <string>
 
--- tools/shell/embedded_shell.cpp	2026-09-02 06:12:38.000000000 +0800
+++ tools/shell/embedded_shell.cpp	2026-09-10 15:07:49.000000000 +0800
@@ -2,6 +2,10 @@
 
 #include "binder/visitor/confidential_statement_analyzer.h"
 
+#ifndef __STDC_FORMAT_MACROS
+#define __STDC_FORMAT_MACROS
+#endif
+
 #ifndef _WIN32
 #include <termios.h>
 #include <unistd.h>
@@ -10,6 +14,7 @@
 #endif
 #include <algorithm>
 #include <array>
+#include <cinttypes>
 #include <cmath>
 #include <csignal>
 #include <iomanip>
